// @platformscode/core/dist/collection/components/dga-footer/dga-footer.js
import { h } from "@stencil/core";
import { guid } from "../../utils";
export class DgaFooter {
    constructor() {
        this.mainTitle = '';
        this.mainDescription = '';
        this.mainLinkTilte = '';
        this.mainImage = '';
        this.copyright = '';
        this.groupLinks = [];
        this.socialMediaTitle = 'Social Media';
        this.socialMediaLinks = [];
        this.accessibilityTitle = 'Accessibility Tools';
        this.accessibilityLinks = [];
        this.basicLinks = [];
        this.extraLinks = [];
        this.bottomImages = [];
    }
    render() {
        return (h("footer", { key: '894f06673ce67740003f28988491544364445c4d', class: `footer ${this.background === 'DarkGreen' ? 'footer--bg-brand' : ''}` }, h("section", { key: '55ce6b04f49b5e4d63133f3c9914380e24fee931', class: "footer-content" }, (this.mainLinkTilte || this.mainImage) && (h("div", { key: '53f83c947d1d89d05fbf04fe6b93f00f218dc24c', class: "footer-icon-section" }, h("a", { key: 'a9219de3a81767729ec48117b4bb201472942f92', class: this.background === 'DarkGreen' ? 'link--on-color' : '', href: this.mainLink || '#', target: "_blank", rel: "noopener noreferrer" }, this.mainLinkTilte))), (this.mainTitle || this.mainDescription) && (h("div", { key: 'd81a67565ecf434fd32aa90c95c439ad2b506bdc', class: "footer-title-description" }, h("h3", { key: '52b7ed9886b88d120f144e6e2a86ef1f05bac227', class: this.background === 'DarkGreen' ? 'link--on-color' : '' }, this.mainTitle), h("p", { key: 'e2475a35765551f4e2b6efa9ef828bf98445b607', class: this.background === 'DarkGreen' ? 'link--on-color' : '' }, this.mainDescription))), this.NavLinks && (h("section", { key: '839007719103820bafcd05e0f7db449c04155606', class: "footer__nav-links" }, this.groupLinks.map(group => (h("div", { class: "footer__nav-links-column" }, h("div", { class: "footer__nav-links-column-title" }, group.title), h("ul", { class: "footer__nav-links-column-body" }, group.links.map(link => (h("li", null, h("a", { href: link.target, class: `link link--sm link--neutral ${this.background === 'DarkGreen' ? 'link--on-color' : ''}` }, h("span", { class: "link__label" }, link.name))))))))), h("div", { key: '30b4d393a12953833ff9b337085e7ed458619912', class: "footer__nav-links-column" }, this.socialMediaLinks.length > 0 && (h("div", { key: 'c5f6d614eabd0b586227b8be529afa95699c4185' }, this.socialMediaTitle && h("div", { key: '93df78aa9b88ce731a11ebde8a1f268077392a82', class: "footer__nav-links-column-title" }, this.socialMediaTitle), h("div", { key: '776e3e3c288d6ebc24842482f96d5f6136c79eea', class: "dga-flex icons-container" }, this.socialMediaLinks.map(social => (h("a", { href: social.target, title: social.title, class: `dga-btn dga-btn--md dga-btn--secondary-outline dga-btn--icon ${this.background} ${this.background === 'DarkGreen' ? 'dga-btn--secondary-outline--on-color' : ''}` }, h("span", { class: "btn-icon" }, h("dga-icon", Object.assign({}, social.icon, { size: 20 }))))))))), this.accessibilityLinks.length > 0 && (h("div", { key: 'a8b918254a8fe2c016d6ad749487880e20705844' }, h("div", { key: '455df941abb6b9a6424c0d541af92054b4f59e8c', class: "footer__nav-links-column-title" }, this.accessibilityTitle), h("div", { key: '8ea39ab6d11c50c33eaa2a9902b369df5138b319', class: "dga-flex icons-container" }, this.accessibilityLinks.map(access => (h("a", { href: access.target, title: access.title, class: `dga-btn dga-btn--md dga-btn--secondary-outline dga-btn--icon ${this.background} ${this.background === 'DarkGreen' ? 'dga-btn--secondary-outline--on-color' : ''}` }, h("span", { class: "btn-icon" }, h("dga-icon", Object.assign({}, access.icon, { size: 20 })))))))))))), h("section", { key: '69904fdd90b44700f0791c4562fbbbe77607dba9', class: "footer__basic" }, h("div", { key: '413192a5b34e4ddf25c73b48be57f1059b286fa7', class: "footer__actions-copyright" }, h("ul", { key: '3e95b668e0b309027b55d137faf2aa3ba9ddf5b5', class: "footer__actions" }, this.basicLinks.map(link => (h("li", null, h("a", { href: link.target, class: `link link--sm link--neutral link--inline ${this.background === 'DarkGreen' ? 'link--on-color' : ''}` }, h("span", { class: "link__label" }, link.name)))))), this.copyright && h("p", { key: 'dde63098c18dafd5c8bbb1f3b000bc9e49ca11ae', class: "footer__copyright text-sm-semibold" }, this.copyright), h("ul", { key: '92e9df35950c8e1c82c0d05e9c5b470acd0eb927', class: "footer__extra-actions" }, this.extraLinks.map(link => (h("li", null, h("a", { href: link.target, class: `link link--sm link--neutral ${this.background === 'DarkGreen' ? 'link--on-color' : ''}` }, h("span", { class: "link__label" }, link.name))))))), h("div", { key: 'cd09704b93ad2696f2a7597cb43af390d8dd9f22', class: "footer__logos" }, this.bottomImages.map(image => (h("img", { src: image, key: guid() }))))))));
    }
    static get is() { return "dga-footer"; }
    static get encapsulation() { return "shadow"; }
    static get originalStyleUrls() {
        return {
            "$": ["./footer.scss"]
        };
    }
    static get styleUrls() {
        return {
            "$": ["footer.css"]
        };
    }
    static get properties() {
        return {
            "background": {
                "type": "string",
                "attribute": "background",
                "mutable": false,
                "complexType": {
                    "original": "'DarkGreen' | 'Light'",
                    "resolved": "\"DarkGreen\" | \"Light\"",
                    "references": {}
                },
                "required": false,
                "optional": true,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false
            },
            "NavLinks": {
                "type": "boolean",
                "attribute": "nav-links",
                "mutable": false,
                "complexType": {
                    "original": "boolean",
                    "resolved": "boolean",
                    "references": {}
                },
                "required": false,
                "optional": true,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false
            },
            "mainTitle": {
                "type": "string",
                "attribute": "main-title",
                "mutable": false,
                "complexType": {
                    "original": "string",
                    "resolved": "string",
                    "references": {}
                },
                "required": false,
                "optional": false,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false,
                "defaultValue": "''"
            },
            "mainDescription": {
                "type": "string",
                "attribute": "main-description",
                "mutable": false,
                "complexType": {
                    "original": "string",
                    "resolved": "string",
                    "references": {}
                },
                "required": false,
                "optional": false,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false,
                "defaultValue": "''"
            },
            "mainLink": {
                "type": "string",
                "attribute": "main-link",
                "mutable": false,
                "complexType": {
                    "original": "string",
                    "resolved": "string",
                    "references": {}
                },
                "required": false,
                "optional": true,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false
            },
            "mainLinkTilte": {
                "type": "string",
                "attribute": "main-link-tilte",
                "mutable": false,
                "complexType": {
                    "original": "string",
                    "resolved": "string",
                    "references": {}
                },
                "required": false,
                "optional": false,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false,
                "defaultValue": "''"
            },
            "mainImage": {
                "type": "string",
                "attribute": "main-image",
                "mutable": false,
                "complexType": {
                    "original": "string",
                    "resolved": "string",
                    "references": {}
                },
                "required": false,
                "optional": false,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false,
                "defaultValue": "''"
            },
            "copyright": {
                "type": "string",
                "attribute": "copyright",
                "mutable": false,
                "complexType": {
                    "original": "string",
                    "resolved": "string",
                    "references": {}
                },
                "required": false,
                "optional": false,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false,
                "defaultValue": "''"
            },
            "groupLinks": {
                "type": "unknown",
                "attribute": "group-links",
                "mutable": false,
                "complexType": {
                    "original": "{ title: string; links: { name: string; target: string }[] }[]",
                    "resolved": "{ title: string; links: { name: string; target: string; }[]; }[]",
                    "references": {}
                },
                "required": false,
                "optional": false,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "defaultValue": "[]"
            },
            "socialMediaTitle": {
                "type": "string",
                "attribute": "social-media-title",
                "mutable": false,
                "complexType": {
                    "original": "string",
                    "resolved": "string",
                    "references": {}
                },
                "required": false,
                "optional": false,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false,
                "defaultValue": "'Social Media'"
            },
            "socialMediaLinks": {
                "type": "unknown",
                "attribute": "social-media-links",
                "mutable": false,
                "complexType": {
                    "original": "{ title?: string; target: string; icon: Icon }[]",
                    "resolved": "{ title?: string; target: string; icon: Icon; }[]",
                    "references": {
                        "Icon": {
                            "location": "import",
                            "path": "../dga-featured-icon/dga-featured-icon",
                            "id": "src/components/dga-featured-icon/dga-featured-icon.tsx::Icon"
                        }
                    }
                },
                "required": false,
                "optional": false,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "defaultValue": "[]"
            },
            "accessibilityTitle": {
                "type": "string",
                "attribute": "accessibility-title",
                "mutable": false,
                "complexType": {
                    "original": "string",
                    "resolved": "string",
                    "references": {}
                },
                "required": false,
                "optional": false,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false,
                "defaultValue": "'Accessibility Tools'"
            },
            "accessibilityLinks": {
                "type": "unknown",
                "attribute": "accessibility-links",
                "mutable": false,
                "complexType": {
                    "original": "{ title?: string; target: string; icon: Icon }[]",
                    "resolved": "{ title?: string; target: string; icon: Icon; }[]",
                    "references": {
                        "Icon": {
                            "location": "import",
                            "path": "../dga-featured-icon/dga-featured-icon",
                            "id": "src/components/dga-featured-icon/dga-featured-icon.tsx::Icon"
                        }
                    }
                },
                "required": false,
                "optional": false,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "defaultValue": "[]"
            },
            "basicLinks": {
                "type": "unknown",
                "attribute": "basic-links",
                "mutable": false,
                "complexType": {
                    "original": "{ name: string; target: string }[]",
                    "resolved": "{ name: string; target: string; }[]",
                    "references": {}
                },
                "required": false,
                "optional": false,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "defaultValue": "[]"
            },
            "extraLinks": {
                "type": "unknown",
                "attribute": "extra-links",
                "mutable": false,
                "complexType": {
                    "original": "{ name: string; target: string }[]",
                    "resolved": "{ name: string; target: string; }[]",
                    "references": {}
                },
                "required": false,
                "optional": false,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "defaultValue": "[]"
            },
            "bottomImages": {
                "type": "unknown",
                "attribute": "bottom-images",
                "mutable": false,
                "complexType": {
                    "original": "string[]",
                    "resolved": "string[]",
                    "references": {}
                },
                "required": false,
                "optional": false,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "defaultValue": "[]"
            }
        };
    }
}
