// @platformscode/core/dist/collection/components/dga-metric/dga-metric.js
import { h, Fragment } from "@stencil/core";
export class DgaMetric {
    constructor() {
        this.label = '';
        this.Style = 'small-chart';
        this.percentage = '0%';
        this.showFeaturedIcon = true;
        this.featuredIcon = (h("dga-icon", { name: "trade-up", variant: "stroke", size: 24 })); // Default icon
        this.showChart = true;
        this.showInfo = true;
        this.showActions = true;
        this.showSecondaryAction = true;
        this.showMoreButton = true;
        this.changeProps = {
            type: '01',
            trend: 'positive',
        };
        this.onPrimaryAction = () => { };
        this.onSecondaryAction = () => { };
        this.onExpandAction = () => { };
        this.changeIcons = {
            '01': {
                positive: 'arrow-up-02',
                negative: 'arrow-down-02',
            },
            '02': {
                positive: 'trade-up',
                negative: 'trade-down',
            },
        };
    }
    render() {
        return (h("div", { key: '314cdf92f905dbf8a3ae6b63dc9fdf0e493542c3', class: `dga-metric`, "data-chart-size": this.Style }, h("div", { key: '200315a62f15f2aa0b4fb9f192e69d5cd47ad21c', class: "dga-metric__head" }, h("div", { key: 'e3b34506cdd7400be7a46c5194c070f42710125a', class: "dga-metric__head-content" }, this.showFeaturedIcon && h("div", { key: '3068d2addfa736f01ec2d50df4a4e3293a38707a', class: "featured-icon featured-icon--light-brand featured-icon--circle featured-icon--sm" }, this.featuredIcon), this.label && h("span", { key: 'bda97e35abad96b0b1d7a8dc1e3f6fe97f3e6fad', class: "dga-metric__head-content-text" }, this.label)), this.showMoreButton && (h("dga-button-v2", { key: '7b83a8aeef2b452b115e98b4b345852bdb6bd200', variant: "subtle", iconOnly: true, leadIcon: true, leadIconProps: {
                name: 'more-vertical',
                variant: 'stroke',
                type: 'rounded',
            }, onClick: this.onExpandAction }))), h("div", { key: 'f3b2f6290453a9b4ce1d3f399e7609a768af5383', class: "dga-metric__body" }, h("div", { key: '503084c257592859c1249a77460462f2f689f1eb', class: "dga-metric__content" }, h("h3", { key: '7f26e13d786a042e2214dc9caaf7e41c915a3429', class: "dga-metric__content-percentage" }, this.percentage), this.showInfo && (h("div", { key: 'e6b06e62dd0ad6bfdfdd5455417fa95b03b2d0d7', class: "dga-metric__content-change", "data-change-trend": this.changeProps.trend }, h("span", { key: 'd8d1e0ba9f73f2640aa414ac23db3871756275cd', class: "dga-metric__content-change-icon" }, h("dga-icon", { key: '565f756fc4f6f3d2d2907b4fc3ada1c76016da09', icon: this.changeIcons[this.changeProps.type][this.changeProps.trend], type: "rounded", variant: "stroke", size: 24 })), h("span", { key: 'd94c1b1831454cb0295715a522c1419909a614f3', class: "dga-metric__content-change-value" }, Math.abs(100), "%"), h("span", { key: 'ea76897604bd61574f635b872275f68f9724ff3f', class: "dga-metric__content-change-period" }, "vs last month")))), this.showChart && (h("div", { key: 'e64a526ba47e91ace2a36aa58a86f84f75bb9ac1', class: "dga-metric__chart" }, h("dga-metric-chart", Object.assign({ key: '7e85020d66426de8a03be8e7d8e07371b7fdde76' }, this.chartProps, { size: this.Style === 'small-chart' ? 'sm' : 'lg' }))))), this.showActions && (h(Fragment, { key: '50d85ae9db0e4e76b1cb4fcb220aadb511523a2a' }, h("div", { key: '9d0cefe611c0d7c150cfbec676129cfa6d1c1681', class: "dga-metric__divider" }), h("div", { key: 'b5d42a5f863ff14702cb65786e290de146a4e551', class: "dga-metric__actions" }, this.showSecondaryAction && (h("dga-button-v2", { key: '718e9ad0e6c9490acdd2037859903dd7fe7b793b', variant: "subtle", iconOnly: true, leadIcon: true, leadIconProps: {
                name: 'settings-02',
                variant: 'stroke',
                type: 'rounded',
            }, onClick: this.onSecondaryAction })), h("dga-button-v2", { key: '19627305bc5456f4e9b9e772e2810fb5871101cb', variant: "secondary-outline", class: "dga-metric__actions--primary", label: "View Report", onClick: this.onPrimaryAction }))))));
    }
    static get is() { return "dga-metric"; }
    static get originalStyleUrls() {
        return {
            "$": ["dga-metric.scss"]
        };
    }
    static get styleUrls() {
        return {
            "$": ["dga-metric.css"]
        };
    }
    static get properties() {
        return {
            "label": {
                "type": "string",
                "attribute": "label",
                "mutable": false,
                "complexType": {
                    "original": "string",
                    "resolved": "string",
                    "references": {}
                },
                "required": false,
                "optional": false,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false,
                "defaultValue": "''"
            },
            "Style": {
                "type": "string",
                "attribute": "style",
                "mutable": false,
                "complexType": {
                    "original": "'small-chart' | 'large-chart'",
                    "resolved": "\"large-chart\" | \"small-chart\"",
                    "references": {}
                },
                "required": false,
                "optional": true,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false,
                "defaultValue": "'small-chart'"
            },
            "percentage": {
                "type": "string",
                "attribute": "percentage",
                "mutable": false,
                "complexType": {
                    "original": "string",
                    "resolved": "string",
                    "references": {}
                },
                "required": false,
                "optional": false,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false,
                "defaultValue": "'0%'"
            },
            "showFeaturedIcon": {
                "type": "boolean",
                "attribute": "show-featured-icon",
                "mutable": false,
                "complexType": {
                    "original": "boolean",
                    "resolved": "boolean",
                    "references": {}
                },
                "required": false,
                "optional": true,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false,
                "defaultValue": "true"
            },
            "featuredIcon": {
                "type": "any",
                "attribute": "featured-icon",
                "mutable": false,
                "complexType": {
                    "original": "HTMLElement | any",
                    "resolved": "any",
                    "references": {
                        "HTMLElement": {
                            "location": "global",
                            "id": "global::HTMLElement"
                        }
                    }
                },
                "required": false,
                "optional": true,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false,
                "defaultValue": "(<dga-icon name=\"trade-up\" variant=\"stroke\" size={24} />)"
            },
            "showChart": {
                "type": "boolean",
                "attribute": "show-chart",
                "mutable": false,
                "complexType": {
                    "original": "boolean",
                    "resolved": "boolean",
                    "references": {}
                },
                "required": false,
                "optional": true,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false,
                "defaultValue": "true"
            },
            "showInfo": {
                "type": "boolean",
                "attribute": "show-info",
                "mutable": false,
                "complexType": {
                    "original": "boolean",
                    "resolved": "boolean",
                    "references": {}
                },
                "required": false,
                "optional": true,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false,
                "defaultValue": "true"
            },
            "showActions": {
                "type": "boolean",
                "attribute": "show-actions",
                "mutable": false,
                "complexType": {
                    "original": "boolean",
                    "resolved": "boolean",
                    "references": {}
                },
                "required": false,
                "optional": true,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false,
                "defaultValue": "true"
            },
            "showSecondaryAction": {
                "type": "boolean",
                "attribute": "show-secondary-action",
                "mutable": false,
                "complexType": {
                    "original": "boolean",
                    "resolved": "boolean",
                    "references": {}
                },
                "required": false,
                "optional": true,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false,
                "defaultValue": "true"
            },
            "showMoreButton": {
                "type": "boolean",
                "attribute": "show-more-button",
                "mutable": false,
                "complexType": {
                    "original": "boolean",
                    "resolved": "boolean",
                    "references": {}
                },
                "required": false,
                "optional": true,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false,
                "defaultValue": "true"
            },
            "changeProps": {
                "type": "unknown",
                "attribute": "change-props",
                "mutable": false,
                "complexType": {
                    "original": "{\n    type: '01' | '02';\n    trend: 'positive' | 'negative';\n  }",
                    "resolved": "{ type: \"01\" | \"02\"; trend: \"negative\" | \"positive\"; }",
                    "references": {}
                },
                "required": false,
                "optional": true,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "defaultValue": "{\n    type: '01',\n    trend: 'positive',\n  }"
            },
            "chartProps": {
                "type": "unknown",
                "attribute": "chart-props",
                "mutable": false,
                "complexType": {
                    "original": "IChartProps",
                    "resolved": "IChartProps",
                    "references": {
                        "IChartProps": {
                            "location": "local",
                            "path": "/Users/mohamedaboelezz/projects/Azm-X/platform-code-cross-platform-storybook/packages/core/src/components/dga-metric/dga-metric.tsx",
                            "id": "src/components/dga-metric/dga-metric.tsx::IChartProps"
                        }
                    }
                },
                "required": false,
                "optional": true,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false
            },
            "trend": {
                "type": "unknown",
                "attribute": "trend",
                "mutable": false,
                "complexType": {
                    "original": "ITrend",
                    "resolved": "ITrend",
                    "references": {
                        "ITrend": {
                            "location": "local",
                            "path": "/Users/mohamedaboelezz/projects/Azm-X/platform-code-cross-platform-storybook/packages/core/src/components/dga-metric/dga-metric.tsx",
                            "id": "src/components/dga-metric/dga-metric.tsx::ITrend"
                        }
                    }
                },
                "required": false,
                "optional": false,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false
            },
            "onPrimaryAction": {
                "type": "unknown",
                "attribute": "on-primary-action",
                "mutable": false,
                "complexType": {
                    "original": "() => void",
                    "resolved": "() => void",
                    "references": {}
                },
                "required": false,
                "optional": true,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "defaultValue": "() => {}"
            },
            "onSecondaryAction": {
                "type": "unknown",
                "attribute": "on-secondary-action",
                "mutable": false,
                "complexType": {
                    "original": "() => void",
                    "resolved": "() => void",
                    "references": {}
                },
                "required": false,
                "optional": true,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "defaultValue": "() => {}"
            },
            "onExpandAction": {
                "type": "unknown",
                "attribute": "on-expand-action",
                "mutable": false,
                "complexType": {
                    "original": "() => void",
                    "resolved": "() => void",
                    "references": {}
                },
                "required": false,
                "optional": true,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "defaultValue": "() => {}"
            }
        };
    }
}
