// @platformscode/core/dist/collection/components/dga-header/dga-nav-header-logos.js
import { h } from "@stencil/core";
export class DgaNavHeaderLogos {
    constructor() {
        this.logoLink = '#';
        this.govLink = '#';
    }
    render() {
        return (h("div", { key: '098d25036ae5d4c77491d4fea30733a256f3f1b9', class: "header-nav__branding" }, this.govSrc && (h("div", { key: 'fcd1d1f5e43719980b551fa1d7dca61c5dbf0b4c', class: "flex xs-hide" }, h("a", { key: '6f6999e9e2b4624f5d21f8cbb56f8f273edbdf8e', href: this.govLink }, h("img", { key: '34f4bff7ba4fa739bd1cc80b7f3df52f9b5a7269', src: this.govSrc, alt: this.govAlt })), h("div", { key: '0accff810850770751c4300d87fb5fc30f2d8350', class: "dga-ver-divider" }))), h("a", { key: 'd44ceb11341d76cf773bd6156dd3dde763f4c854', href: this.logoLink, class: "header__logo" }, h("img", { key: '71ab46a217896bc8fe719f52cd4c71113ae7b145', src: this.logoSrc, alt: this.logoAlt }))));
    }
    static get is() { return "dga-nav-header-logos"; }
    static get encapsulation() { return "shadow"; }
    static get originalStyleUrls() {
        return {
            "$": ["dga-nav-header-sub-menu.scss"]
        };
    }
    static get styleUrls() {
        return {
            "$": ["dga-nav-header-sub-menu.css"]
        };
    }
    static get properties() {
        return {
            "logoSrc": {
                "type": "string",
                "attribute": "logo-src",
                "mutable": false,
                "complexType": {
                    "original": "string",
                    "resolved": "string",
                    "references": {}
                },
                "required": false,
                "optional": false,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false
            },
            "logoAlt": {
                "type": "string",
                "attribute": "logo-alt",
                "mutable": false,
                "complexType": {
                    "original": "string",
                    "resolved": "string",
                    "references": {}
                },
                "required": false,
                "optional": true,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false
            },
            "logoLink": {
                "type": "string",
                "attribute": "logo-link",
                "mutable": false,
                "complexType": {
                    "original": "string",
                    "resolved": "string",
                    "references": {}
                },
                "required": false,
                "optional": false,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false,
                "defaultValue": "'#'"
            },
            "govSrc": {
                "type": "string",
                "attribute": "gov-src",
                "mutable": false,
                "complexType": {
                    "original": "string",
                    "resolved": "string",
                    "references": {}
                },
                "required": false,
                "optional": true,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false
            },
            "govAlt": {
                "type": "string",
                "attribute": "gov-alt",
                "mutable": false,
                "complexType": {
                    "original": "string",
                    "resolved": "string",
                    "references": {}
                },
                "required": false,
                "optional": true,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false
            },
            "govLink": {
                "type": "string",
                "attribute": "gov-link",
                "mutable": false,
                "complexType": {
                    "original": "string",
                    "resolved": "string",
                    "references": {}
                },
                "required": false,
                "optional": false,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false,
                "defaultValue": "'#'"
            }
        };
    }
}
