// @platformscode/core/dist/collection/components/dga-header/dga-nav-header-sub-menu-column.js
import { h } from "@stencil/core";
export class DgaNavHeaderSubMenuColumn {
    constructor() {
        this.label = '';
    }
    render() {
        return (h("div", { key: 'c177b87c9075cf517d169d395374009c0c357d42', class: "sub-menu__column" }, this.label && h("h2", { key: '642af1dea89f17c70a2e83112ad365110158f017', class: "sub-menu__column__label" }, this.label), h("slot", { key: '6432d519aa14e1f87c0ad2b164c1f90538ab6af6' })));
    }
    static get is() { return "dga-nav-header-sub-menu-column"; }
    static get encapsulation() { return "shadow"; }
    static get originalStyleUrls() {
        return {
            "$": ["dga-nav-header-sub-menu.scss"]
        };
    }
    static get styleUrls() {
        return {
            "$": ["dga-nav-header-sub-menu.css"]
        };
    }
    static get properties() {
        return {
            "label": {
                "type": "string",
                "attribute": "label",
                "mutable": false,
                "complexType": {
                    "original": "string",
                    "resolved": "string",
                    "references": {}
                },
                "required": false,
                "optional": true,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false,
                "defaultValue": "''"
            }
        };
    }
}
