// @platformscode/core/dist/collection/components/dga-header/dga-header-mobile-actions.js
import { h, Fragment } from "@stencil/core";
export class DgaHeaderMobileActions {
    constructor() {
        this.actions = [];
        this.isMenuOpen = false; // State to control menu visibility
        this.anchorEl = null;
        // private buttonRef: HTMLElement | null = null; // Reference to the button
        this.toggleMenu = () => {
            this.isMenuOpen = !this.isMenuOpen;
            if (this.isMenuOpen) {
                document.addEventListener('click', this.handleOutsideClick);
            }
            else {
                document.removeEventListener('click', this.handleOutsideClick);
            }
        };
        this.closeMenu = () => {
            this.isMenuOpen = false;
            document.removeEventListener('click', this.handleOutsideClick);
        };
        this.handleOutsideClick = (event) => {
            const target = event.target;
            if (!this.el.contains(target)) {
                this.closeMenu();
            }
        };
    }
    render() {
        return (h(Fragment, { key: '3f308c3ad8c285b216a01c090528d77f0c78a87a' }, h("div", { key: '3112e8aac9c3e03ba8843d87086760e3feba2f5a', style: { position: 'relative' } }), h("ul", { key: '2672ead1d0e195cb91fe72649b93e3fc1b60c2a6', class: "header-nav__mobile-actions" }, h("slot", { key: '43190fc1bad4d2280e2ef9d7bb0db7738f2e51c1' }))));
    }
    static get is() { return "dga-header-mobile-actions"; }
    static get encapsulation() { return "shadow"; }
    static get originalStyleUrls() {
        return {
            "$": ["dga-nav-header-sub-menu.scss"]
        };
    }
    static get styleUrls() {
        return {
            "$": ["dga-nav-header-sub-menu.css"]
        };
    }
    static get properties() {
        return {
            "actions": {
                "type": "unknown",
                "attribute": "actions",
                "mutable": false,
                "complexType": {
                    "original": "Action[]",
                    "resolved": "Action[]",
                    "references": {
                        "Action": {
                            "location": "local",
                            "path": "/home/mamoud/Documents/temp/components/platform-code-cross-platform-storybook/packages/core/src/components/dga-header/dga-header-mobile-actions.tsx",
                            "id": "src/components/dga-header/dga-header-mobile-actions.tsx::Action"
                        }
                    }
                },
                "required": false,
                "optional": true,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "defaultValue": "[]"
            }
        };
    }
    static get states() {
        return {
            "isMenuOpen": {},
            "anchorEl": {}
        };
    }
    static get elementRef() { return "el"; }
}
