// @platformscode/core/dist/collection/components/dga-header/dga-nav-header-actions.js
import { h, Fragment } from "@stencil/core";
export class DgaNavHeaderActions {
    constructor() {
        this.isMenuOpen = false; // State to control menu visibility
        this.toggleMenu = () => {
            this.isMenuOpen = !this.isMenuOpen;
            if (this.isMenuOpen) {
                document.addEventListener('click', this.handleOutsideClick);
            }
            else {
                document.removeEventListener('click', this.handleOutsideClick);
            }
        };
        this.closeMenu = () => {
            this.isMenuOpen = false;
            document.removeEventListener('click', this.handleOutsideClick);
        };
        this.handleOutsideClick = (event) => {
            const target = event.target;
            if (!this.el.contains(target)) {
                this.closeMenu();
            }
        };
    }
    render() {
        return (h(Fragment, { key: 'a2cced5c123afa2df1aa2f005abcb3f56eeb057c' }, h("div", { key: '12bc1cd4e77b43f9515d96f938354242030cb57d', style: { position: 'relative' } }, h("dga-button-v2", { key: '185d7916d6871c2cde9e6b0c1864de85e81dfd24', "aria-haspopup": "true", "aria-expanded": this.isMenuOpen ? 'true' : 'false', iconOnly: true, leadIcon: true, leadIconProps: {
                name: 'more-horizontal-circle-01',
                size: 24,
                variant: 'solid',
                type: 'rounded',
            }, variant: "transparent", onClick: this.toggleMenu, class: "header-menu__btn" }), this.isMenuOpen && (h("ul", { key: '5bb262698036ccc55d13bac1a6a3e719e55c88f5', class: "dga-breadcrumb-dropdown" }, [{
                label: 'Item 1', path: '/item1'
            },
            { label: 'Item 2', path: '/item2' },
        ].map(ellipsedItem => (h("li", { key: ellipsedItem === null || ellipsedItem === void 0 ? void 0 : ellipsedItem.label, class: "dga-breadcrumb-dropdown-item" }, h("a", null, h("span", { class: "link__label" }, ellipsedItem.label)))))))), h("ul", { key: '589573db09e7942955567f1126b660e00029ad30', class: "header-nav__actions" }, h("slot", { key: 'bf49bcf5611e6e55aa41070e3334727912771d9c' }))));
    }
    static get is() { return "dga-nav-header-actions"; }
    static get encapsulation() { return "shadow"; }
    static get originalStyleUrls() {
        return {
            "$": ["dga-nav-header-sub-menu.scss"]
        };
    }
    static get styleUrls() {
        return {
            "$": ["dga-nav-header-sub-menu.css"]
        };
    }
    static get states() {
        return {
            "isMenuOpen": {}
        };
    }
    static get elementRef() { return "el"; }
}
