// @platformscode/core/dist/collection/components/dga-notification-toast/dga-notification-toast.js
import { h, Fragment } from "@stencil/core";
import { guid } from "../../utils";
export class DgaNotificationToast {
    constructor() {
        this.type = 'neutral'; // Default to "neutral" if not provided
        this.hasHelperText = false; // Default to false if not provided
        this.vPostion = 'top';
        this.hPostion = 'right';
        this.buttonsList = [];
        this.open = true;
        this.icons = {
            success: {
                name: 'checkmark-circle-02',
                variant: 'stroke',
                type: 'rounded',
            },
            info: {
                name: 'information-circle',
                variant: 'stroke',
                type: 'rounded',
            },
            neutral: {
                name: 'information-circle',
                variant: 'stroke',
                type: 'rounded',
            },
            warning: {
                name: 'alert-02',
                variant: 'stroke',
                type: 'rounded',
            },
            error: {
                name: 'alert-circle',
                variant: 'stroke',
                type: 'rounded',
            },
        };
    }
    render() {
        return (h("div", { key: '80fea6e4aeef5111faa5791c4dc8025e4380eab4', class: `notification-toast notification-toast--${this.type} dga-vpostion-${this.vPostion} dga-hpostion-${this.hPostion} ${this.open ? 'active' : ''}` }, h("div", { key: '046992b43822b7b085397f876f8e9a2f850c2c53', class: `notification-toast__header ${this.hasHelperText ? '' : 'dga-flex-center-y'}` }, h("dga-featured-icon", { key: '2472254ebc14cca09f3d0ad218bf62e8e2681fe8', icon: Object.assign(Object.assign({}, this.icons[this.type]), { size: 20 }), color: this.type, size: "lg" }), h("div", { key: 'b99af2fd29bcfb5bfae4c3d2fa68dbff06a39b98', class: "notification-toast__body" }, h("h4", { key: '502779fa20820698198135b633880a64b72abcc1', class: "notification-toast__title text-md-semibold" }, this.leadText), this.hasHelperText && this.helperText && h("p", { key: '59f1a273153e2c79fcfe2314b919c91c75791796', class: "notification-toast__content text-sm-regular" }, this.helperText)), this.closeButton && (h("div", { key: '188e3cafc7620e29fa14c689ee448f78b46c7578', class: "notification-toast__close-btn" }, h("button", { key: 'f4d1365e195955c795b5fd15e1fb8811cc6391c7', class: "dga-btn dga-btn--close dga-btn--icon dga-btn--md", onClick: e => { var _a; return (_a = this.onClose) === null || _a === void 0 ? void 0 : _a.call(this, e); } }, h("span", { key: 'adbefc04ffecd488ed52fc37ed7cd117f90241d1', class: "dga-btn-icon" }, h("dga-icon", { key: '0f60828e0ea90c1359de1a1f950b323b45d323c5', name: "multiplication-sign", variant: "stroke", type: "sharp", size: 20 })))))), this.action && (h(Fragment, { key: 'ad8b9d11094f3b35a0cab7c2668360f8c1b6c18f' }, h("div", { key: '983a64fb3c4704d939685b023173f309d92ddabb', class: "notification-toast__action" }, this.buttonsList.length > 0 &&
            this.buttonsList.map(({ id, onClick, extraClasses, title, label }) => (h("button", Object.assign({ key: id ? id : guid(), id: id ? id : guid(), onClick: e => onClick(e), class: `dga-btn dga-btn--subtle dga-btn--md ${extraClasses}` }, (title && { title: title })), label)))), h("div", { key: 'e9567ceafb5b9b0ea5fefcddab680136bf12748c', class: "notification-toast__action-mobile" }, this.buttonsList.length > 0 &&
            this.buttonsList.map(({ id, onClick, extraClasses, title, label }) => (h("button", Object.assign({ key: id ? id : guid(), id: id ? id : guid(), onClick: e => onClick(e), class: `dga-btn dga-btn--secondary dga-btn--lg ${extraClasses}` }, (title && { title: title })), label))))))));
    }
    static get is() { return "dga-notification-toast"; }
    static get encapsulation() { return "shadow"; }
    static get originalStyleUrls() {
        return {
            "$": ["dga-notification-toast.scss"]
        };
    }
    static get styleUrls() {
        return {
            "$": ["dga-notification-toast.css"]
        };
    }
    static get properties() {
        return {
            "type": {
                "type": "string",
                "attribute": "type",
                "mutable": false,
                "complexType": {
                    "original": "'error' | 'warning' | 'success' | 'info' | 'neutral'",
                    "resolved": "\"error\" | \"info\" | \"neutral\" | \"success\" | \"warning\"",
                    "references": {}
                },
                "required": false,
                "optional": true,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false,
                "defaultValue": "'neutral'"
            },
            "leadText": {
                "type": "string",
                "attribute": "lead-text",
                "mutable": false,
                "complexType": {
                    "original": "string",
                    "resolved": "string",
                    "references": {}
                },
                "required": false,
                "optional": true,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false
            },
            "hasHelperText": {
                "type": "boolean",
                "attribute": "has-helper-text",
                "mutable": false,
                "complexType": {
                    "original": "boolean",
                    "resolved": "boolean",
                    "references": {}
                },
                "required": false,
                "optional": true,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false,
                "defaultValue": "false"
            },
            "helperText": {
                "type": "string",
                "attribute": "helper-text",
                "mutable": false,
                "complexType": {
                    "original": "string",
                    "resolved": "string",
                    "references": {}
                },
                "required": false,
                "optional": true,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false
            },
            "action": {
                "type": "boolean",
                "attribute": "action",
                "mutable": false,
                "complexType": {
                    "original": "boolean",
                    "resolved": "boolean",
                    "references": {}
                },
                "required": false,
                "optional": true,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false
            },
            "closeButton": {
                "type": "boolean",
                "attribute": "close-button",
                "mutable": false,
                "complexType": {
                    "original": "boolean",
                    "resolved": "boolean",
                    "references": {}
                },
                "required": false,
                "optional": true,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false
            },
            "vPostion": {
                "type": "string",
                "attribute": "v-postion",
                "mutable": false,
                "complexType": {
                    "original": "'top' | 'bottom'",
                    "resolved": "\"bottom\" | \"top\"",
                    "references": {}
                },
                "required": false,
                "optional": true,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false,
                "defaultValue": "'top'"
            },
            "hPostion": {
                "type": "string",
                "attribute": "h-postion",
                "mutable": false,
                "complexType": {
                    "original": "'left' | 'right'",
                    "resolved": "\"left\" | \"right\"",
                    "references": {}
                },
                "required": false,
                "optional": true,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false,
                "defaultValue": "'right'"
            },
            "onClose": {
                "type": "unknown",
                "attribute": "on-close",
                "mutable": false,
                "complexType": {
                    "original": "(event: any) => void",
                    "resolved": "(event: any) => void",
                    "references": {}
                },
                "required": false,
                "optional": true,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false
            },
            "buttonsList": {
                "type": "unknown",
                "attribute": "buttons-list",
                "mutable": false,
                "complexType": {
                    "original": "{\n    id?: string;\n    label: string;\n    onClick: (event?: any) => void;\n    extraClasses?: string;\n    title?: string;\n  }[]",
                    "resolved": "{ id?: string; label: string; onClick: (event?: any) => void; extraClasses?: string; title?: string; }[]",
                    "references": {}
                },
                "required": false,
                "optional": true,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "defaultValue": "[]"
            },
            "open": {
                "type": "boolean",
                "attribute": "open",
                "mutable": true,
                "complexType": {
                    "original": "boolean",
                    "resolved": "boolean",
                    "references": {}
                },
                "required": false,
                "optional": true,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": true,
                "defaultValue": "true"
            }
        };
    }
}
