// @platformscode/core/dist/collection/components/dga-grid/dga-grid-container.js
// // First, let's create the Grid Container component
// import { Component, Prop, h, Host } from '@stencil/core';
// export type GridSpacing = 0 | 1 | 2 | 3 | 4 | 5 | 6 | 7 | 8 | 9 | 10;
// @Component({
//   tag: 'dga-grid-container',
//   styleUrl: 'dga-grid.scss',
//   shadow: true,
// })
// export class DgaGridContainer {
//   /** Set to true to use the full width of the viewport */
//   @Prop() fluid: boolean = false;
//   /** Spacing between grid items */
//   @Prop() spacing: GridSpacing = 0;
//   /** Spacing between grid items on xs screens */
//   @Prop() spacingXs: GridSpacing;
//   /** Spacing between grid items on sm screens */
//   @Prop() spacingSm: GridSpacing;
//   /** Spacing between grid items on md screens */
//   @Prop() spacingMd: GridSpacing;
//   /** Spacing between grid items on lg screens */
//   @Prop() spacingLg: GridSpacing;
//   /** Spacing between grid items on xl screens */
//   @Prop() spacingXl: GridSpacing;
//   /** Set to true to disable gutters */
//   @Prop() disableGutters: boolean = false;
//   render() {
//     const classes = {
//       'dga-grid-container': true,
//       'dga-grid-container--fluid': this.fluid,
//       'dga-grid-container--no-gutters': this.disableGutters,
//       [`dga-grid-spacing-${this.spacing}`]: this.spacing !== undefined,
//       [`dga-grid-spacing-xs-${this.spacingXs}`]: this.spacingXs !== undefined,
//       [`dga-grid-spacing-sm-${this.spacingSm}`]: this.spacingSm !== undefined,
//       [`dga-grid-spacing-md-${this.spacingMd}`]: this.spacingMd !== undefined,
//       [`dga-grid-spacing-lg-${this.spacingLg}`]: this.spacingLg !== undefined,
//       [`dga-grid-spacing-xl-${this.spacingXl}`]: this.spacingXl !== undefined,
//     };
//     return (
//       <Host class={Object.keys(classes).filter(key => classes[key]).join(' ')}>
//         <slot />
//       </Host>
//     );
//   }
// }
import { h, Host } from "@stencil/core";
export class DgaGridContainer {
    constructor() {
        /** Set to true to use the full width of the viewport */
        this.fluid = false;
        /** Set to true to disable gutters */
        this.disableGutters = false;
    }
    // Watch for changes to apply styles
    updateSpacing() {
        this.applyGapStyles();
    }
    componentDidLoad() {
        this.applyGapStyles();
    }
    parseGapValue(value) {
        if (value === undefined || value === null)
            return null;
        // If it's already a string with units, return as is
        if (typeof value === 'string' && (value.includes('px') || value.includes('rem') || value.includes('em') || value.includes('%') || value === '0')) {
            return value;
        }
        // If it's a number or string number, convert to px
        if (typeof value === 'number' || !isNaN(Number(value))) {
            return `${value}px`;
        }
        return null;
    }
    applyGapStyles() {
        const style = this.el.style;
        // First handle the numbered spacing prop (legacy)
        if (this.spacing !== undefined) {
            const spacingValue = `${this.spacing * 8}px`;
            style.setProperty('--grid-gap', spacingValue);
            style.setProperty('--grid-column-gap', spacingValue);
            style.setProperty('--grid-row-gap', spacingValue);
        }
        // Then handle the new flexible gap props, which override the legacy spacing
        if (this.gap !== undefined) {
            if (typeof this.gap === 'object') {
                // Responsive object format
                if (this.gap.xs !== undefined) {
                    const value = this.parseGapValue(this.gap.xs);
                    if (value) {
                        style.setProperty('--grid-gap-xs', value);
                        style.setProperty('--grid-column-gap-xs', value);
                        style.setProperty('--grid-row-gap-xs', value);
                    }
                }
                if (this.gap.sm !== undefined) {
                    const value = this.parseGapValue(this.gap.sm);
                    if (value) {
                        style.setProperty('--grid-gap-sm', value);
                        style.setProperty('--grid-column-gap-sm', value);
                        style.setProperty('--grid-row-gap-sm', value);
                    }
                }
                if (this.gap.md !== undefined) {
                    const value = this.parseGapValue(this.gap.md);
                    if (value) {
                        style.setProperty('--grid-gap-md', value);
                        style.setProperty('--grid-column-gap-md', value);
                        style.setProperty('--grid-row-gap-md', value);
                    }
                }
                if (this.gap.lg !== undefined) {
                    const value = this.parseGapValue(this.gap.lg);
                    if (value) {
                        style.setProperty('--grid-gap-lg', value);
                        style.setProperty('--grid-column-gap-lg', value);
                        style.setProperty('--grid-row-gap-lg', value);
                    }
                }
                if (this.gap.xl !== undefined) {
                    const value = this.parseGapValue(this.gap.xl);
                    if (value) {
                        style.setProperty('--grid-gap-xl', value);
                        style.setProperty('--grid-column-gap-xl', value);
                        style.setProperty('--grid-row-gap-xl', value);
                    }
                }
            }
            else {
                // Simple string or number format
                const value = this.parseGapValue(this.gap);
                if (value) {
                    style.setProperty('--grid-gap', value);
                    style.setProperty('--grid-column-gap', value);
                    style.setProperty('--grid-row-gap', value);
                }
            }
        }
        // Handle column and row gap separately (these override the general gap)
        if (this.columnGap !== undefined) {
            if (typeof this.columnGap === 'object') {
                // Responsive object format
                if (this.columnGap.xs !== undefined) {
                    const value = this.parseGapValue(this.columnGap.xs);
                    if (value)
                        style.setProperty('--grid-column-gap-xs', value);
                }
                if (this.columnGap.sm !== undefined) {
                    const value = this.parseGapValue(this.columnGap.sm);
                    if (value)
                        style.setProperty('--grid-column-gap-sm', value);
                }
                if (this.columnGap.md !== undefined) {
                    const value = this.parseGapValue(this.columnGap.md);
                    if (value)
                        style.setProperty('--grid-column-gap-md', value);
                }
                if (this.columnGap.lg !== undefined) {
                    const value = this.parseGapValue(this.columnGap.lg);
                    if (value)
                        style.setProperty('--grid-column-gap-lg', value);
                }
                if (this.columnGap.xl !== undefined) {
                    const value = this.parseGapValue(this.columnGap.xl);
                    if (value)
                        style.setProperty('--grid-column-gap-xl', value);
                }
            }
            else {
                // Simple string or number format
                const value = this.parseGapValue(this.columnGap);
                if (value)
                    style.setProperty('--grid-column-gap', value);
            }
        }
        if (this.rowGap !== undefined) {
            if (typeof this.rowGap === 'object') {
                // Responsive object format
                if (this.rowGap.xs !== undefined) {
                    const value = this.parseGapValue(this.rowGap.xs);
                    if (value)
                        style.setProperty('--grid-row-gap-xs', value);
                }
                if (this.rowGap.sm !== undefined) {
                    const value = this.parseGapValue(this.rowGap.sm);
                    if (value)
                        style.setProperty('--grid-row-gap-sm', value);
                }
                if (this.rowGap.md !== undefined) {
                    const value = this.parseGapValue(this.rowGap.md);
                    if (value)
                        style.setProperty('--grid-row-gap-md', value);
                }
                if (this.rowGap.lg !== undefined) {
                    const value = this.parseGapValue(this.rowGap.lg);
                    if (value)
                        style.setProperty('--grid-row-gap-lg', value);
                }
                if (this.rowGap.xl !== undefined) {
                    const value = this.parseGapValue(this.rowGap.xl);
                    if (value)
                        style.setProperty('--grid-row-gap-xl', value);
                }
            }
            else {
                // Simple string or number format
                const value = this.parseGapValue(this.rowGap);
                if (value)
                    style.setProperty('--grid-row-gap', value);
            }
        }
    }
    render() {
        const classes = {
            'dga-grid-container': true,
            'dga-grid-container--fluid': this.fluid,
            'dga-grid-container--no-gutters': this.disableGutters,
        };
        return (h(Host, { key: '87b902fde60f936900e342dab8169f3fc336d86a', class: Object.keys(classes).filter(key => classes[key]).join(' ') }, h("slot", { key: '58d8a47a48415e02428855703379837d904973dc' })));
    }
    static get is() { return "dga-grid-container"; }
    static get originalStyleUrls() {
        return {
            "$": ["dga-grid.scss"]
        };
    }
    static get styleUrls() {
        return {
            "$": ["dga-grid.css"]
        };
    }
    static get properties() {
        return {
            "fluid": {
                "type": "boolean",
                "attribute": "fluid",
                "mutable": false,
                "complexType": {
                    "original": "boolean",
                    "resolved": "boolean",
                    "references": {}
                },
                "required": false,
                "optional": false,
                "docs": {
                    "tags": [],
                    "text": "Set to true to use the full width of the viewport"
                },
                "getter": false,
                "setter": false,
                "reflect": false,
                "defaultValue": "false"
            },
            "gap": {
                "type": "any",
                "attribute": "gap",
                "mutable": false,
                "complexType": {
                    "original": "FlexibleGap",
                    "resolved": "ResponsiveGap | number | string",
                    "references": {
                        "FlexibleGap": {
                            "location": "local",
                            "path": "/Users/mohamedaboelezz/projects/Azm-X/platform-code-cross-platform-storybook/packages/core/src/components/dga-grid/dga-grid-container.tsx",
                            "id": "src/components/dga-grid/dga-grid-container.tsx::FlexibleGap"
                        }
                    }
                },
                "required": false,
                "optional": false,
                "docs": {
                    "tags": [],
                    "text": "Gap between grid items (can be a string like \"12px\" or a responsive object)"
                },
                "getter": false,
                "setter": false,
                "reflect": false
            },
            "columnGap": {
                "type": "any",
                "attribute": "column-gap",
                "mutable": false,
                "complexType": {
                    "original": "FlexibleGap",
                    "resolved": "ResponsiveGap | number | string",
                    "references": {
                        "FlexibleGap": {
                            "location": "local",
                            "path": "/Users/mohamedaboelezz/projects/Azm-X/platform-code-cross-platform-storybook/packages/core/src/components/dga-grid/dga-grid-container.tsx",
                            "id": "src/components/dga-grid/dga-grid-container.tsx::FlexibleGap"
                        }
                    }
                },
                "required": false,
                "optional": false,
                "docs": {
                    "tags": [],
                    "text": "Column gap between grid items (can be a string like \"12px\" or a responsive object)"
                },
                "getter": false,
                "setter": false,
                "reflect": false
            },
            "rowGap": {
                "type": "any",
                "attribute": "row-gap",
                "mutable": false,
                "complexType": {
                    "original": "FlexibleGap",
                    "resolved": "ResponsiveGap | number | string",
                    "references": {
                        "FlexibleGap": {
                            "location": "local",
                            "path": "/Users/mohamedaboelezz/projects/Azm-X/platform-code-cross-platform-storybook/packages/core/src/components/dga-grid/dga-grid-container.tsx",
                            "id": "src/components/dga-grid/dga-grid-container.tsx::FlexibleGap"
                        }
                    }
                },
                "required": false,
                "optional": false,
                "docs": {
                    "tags": [],
                    "text": "Row gap between grid items (can be a string like \"12px\" or a responsive object)"
                },
                "getter": false,
                "setter": false,
                "reflect": false
            },
            "spacing": {
                "type": "number",
                "attribute": "spacing",
                "mutable": false,
                "complexType": {
                    "original": "GridSpacing",
                    "resolved": "0 | 1 | 10 | 2 | 3 | 4 | 5 | 6 | 7 | 8 | 9",
                    "references": {
                        "GridSpacing": {
                            "location": "local",
                            "path": "/Users/mohamedaboelezz/projects/Azm-X/platform-code-cross-platform-storybook/packages/core/src/components/dga-grid/dga-grid-container.tsx",
                            "id": "src/components/dga-grid/dga-grid-container.tsx::GridSpacing"
                        }
                    }
                },
                "required": false,
                "optional": false,
                "docs": {
                    "tags": [],
                    "text": "Spacing between grid items (legacy prop using 8px multiplier)"
                },
                "getter": false,
                "setter": false,
                "reflect": false
            },
            "disableGutters": {
                "type": "boolean",
                "attribute": "disable-gutters",
                "mutable": false,
                "complexType": {
                    "original": "boolean",
                    "resolved": "boolean",
                    "references": {}
                },
                "required": false,
                "optional": false,
                "docs": {
                    "tags": [],
                    "text": "Set to true to disable gutters"
                },
                "getter": false,
                "setter": false,
                "reflect": false,
                "defaultValue": "false"
            }
        };
    }
    static get elementRef() { return "el"; }
    static get watchers() {
        return [{
                "propName": "gap",
                "methodName": "updateSpacing"
            }, {
                "propName": "columnGap",
                "methodName": "updateSpacing"
            }, {
                "propName": "rowGap",
                "methodName": "updateSpacing"
            }, {
                "propName": "spacing",
                "methodName": "updateSpacing"
            }];
    }
}
