// @platformscode/core/dist/collection/components/dga-grid/dga-grid-item.js
// Now, let's create the Grid Item component
import { h, Host } from "@stencil/core";
export class DgaGridItem {
    constructor() {
        /** Controls if the item should fill the available space */
        this.grow = false;
        /** Controls if the item should shrink */
        this.shrink = false;
    }
    render() {
        const classes = {
            'dga-grid-item': true,
            [`dga-grid-xs-${this.xs}`]: this.xs !== undefined,
            [`dga-grid-sm-${this.sm}`]: this.sm !== undefined,
            [`dga-grid-md-${this.md}`]: this.md !== undefined,
            [`dga-grid-lg-${this.lg}`]: this.lg !== undefined,
            [`dga-grid-xl-${this.xl}`]: this.xl !== undefined,
            'dga-grid-item--grow': this.grow,
            'dga-grid-item--shrink': this.shrink,
            [`dga-grid-order-${this.order}`]: this.order !== undefined,
            [`dga-grid-order-xs-${this.orderXs}`]: this.orderXs !== undefined,
            [`dga-grid-order-sm-${this.orderSm}`]: this.orderSm !== undefined,
            [`dga-grid-order-md-${this.orderMd}`]: this.orderMd !== undefined,
            [`dga-grid-order-lg-${this.orderLg}`]: this.orderLg !== undefined,
            [`dga-grid-order-xl-${this.orderXl}`]: this.orderXl !== undefined,
        };
        const styles = {
            alignSelf: this.alignSelf,
        };
        return (h(Host, { key: 'a352d2ee50f1555cb7895ab0dcbd3612cdce985e', class: Object.keys(classes).filter(key => classes[key]).join(' '), style: styles }, h("slot", { key: 'e3ed677b0aa211f2d07c0ac1eb39624ce094133c' })));
    }
    static get is() { return "dga-grid-item"; }
    static get encapsulation() { return "shadow"; }
    static get originalStyleUrls() {
        return {
            "$": ["dga-grid.scss"]
        };
    }
    static get styleUrls() {
        return {
            "$": ["dga-grid.css"]
        };
    }
    static get properties() {
        return {
            "xs": {
                "type": "any",
                "attribute": "xs",
                "mutable": false,
                "complexType": {
                    "original": "GridSize",
                    "resolved": "\"auto\" | 1 | 10 | 11 | 12 | 2 | 3 | 4 | 5 | 6 | 7 | 8 | 9 | boolean",
                    "references": {
                        "GridSize": {
                            "location": "local",
                            "path": "/Users/mohamedaboelezz/projects/Azm-X/platform-code-cross-platform-storybook/packages/core/src/components/dga-grid/dga-grid-item.tsx",
                            "id": "src/components/dga-grid/dga-grid-item.tsx::GridSize"
                        }
                    }
                },
                "required": false,
                "optional": false,
                "docs": {
                    "tags": [],
                    "text": "Number of columns to span (1-12)"
                },
                "getter": false,
                "setter": false,
                "reflect": false
            },
            "sm": {
                "type": "any",
                "attribute": "sm",
                "mutable": false,
                "complexType": {
                    "original": "GridSize",
                    "resolved": "\"auto\" | 1 | 10 | 11 | 12 | 2 | 3 | 4 | 5 | 6 | 7 | 8 | 9 | boolean",
                    "references": {
                        "GridSize": {
                            "location": "local",
                            "path": "/Users/mohamedaboelezz/projects/Azm-X/platform-code-cross-platform-storybook/packages/core/src/components/dga-grid/dga-grid-item.tsx",
                            "id": "src/components/dga-grid/dga-grid-item.tsx::GridSize"
                        }
                    }
                },
                "required": false,
                "optional": false,
                "docs": {
                    "tags": [],
                    "text": "Number of columns to span on sm screens"
                },
                "getter": false,
                "setter": false,
                "reflect": false
            },
            "md": {
                "type": "any",
                "attribute": "md",
                "mutable": false,
                "complexType": {
                    "original": "GridSize",
                    "resolved": "\"auto\" | 1 | 10 | 11 | 12 | 2 | 3 | 4 | 5 | 6 | 7 | 8 | 9 | boolean",
                    "references": {
                        "GridSize": {
                            "location": "local",
                            "path": "/Users/mohamedaboelezz/projects/Azm-X/platform-code-cross-platform-storybook/packages/core/src/components/dga-grid/dga-grid-item.tsx",
                            "id": "src/components/dga-grid/dga-grid-item.tsx::GridSize"
                        }
                    }
                },
                "required": false,
                "optional": false,
                "docs": {
                    "tags": [],
                    "text": "Number of columns to span on md screens"
                },
                "getter": false,
                "setter": false,
                "reflect": false
            },
            "lg": {
                "type": "any",
                "attribute": "lg",
                "mutable": false,
                "complexType": {
                    "original": "GridSize",
                    "resolved": "\"auto\" | 1 | 10 | 11 | 12 | 2 | 3 | 4 | 5 | 6 | 7 | 8 | 9 | boolean",
                    "references": {
                        "GridSize": {
                            "location": "local",
                            "path": "/Users/mohamedaboelezz/projects/Azm-X/platform-code-cross-platform-storybook/packages/core/src/components/dga-grid/dga-grid-item.tsx",
                            "id": "src/components/dga-grid/dga-grid-item.tsx::GridSize"
                        }
                    }
                },
                "required": false,
                "optional": false,
                "docs": {
                    "tags": [],
                    "text": "Number of columns to span on lg screens"
                },
                "getter": false,
                "setter": false,
                "reflect": false
            },
            "xl": {
                "type": "any",
                "attribute": "xl",
                "mutable": false,
                "complexType": {
                    "original": "GridSize",
                    "resolved": "\"auto\" | 1 | 10 | 11 | 12 | 2 | 3 | 4 | 5 | 6 | 7 | 8 | 9 | boolean",
                    "references": {
                        "GridSize": {
                            "location": "local",
                            "path": "/Users/mohamedaboelezz/projects/Azm-X/platform-code-cross-platform-storybook/packages/core/src/components/dga-grid/dga-grid-item.tsx",
                            "id": "src/components/dga-grid/dga-grid-item.tsx::GridSize"
                        }
                    }
                },
                "required": false,
                "optional": false,
                "docs": {
                    "tags": [],
                    "text": "Number of columns to span on xl screens"
                },
                "getter": false,
                "setter": false,
                "reflect": false
            },
            "alignSelf": {
                "type": "string",
                "attribute": "align-self",
                "mutable": false,
                "complexType": {
                    "original": "GridAlignment",
                    "resolved": "\"center\" | \"flex-end\" | \"flex-start\" | \"stretch\"",
                    "references": {
                        "GridAlignment": {
                            "location": "local",
                            "path": "/Users/mohamedaboelezz/projects/Azm-X/platform-code-cross-platform-storybook/packages/core/src/components/dga-grid/dga-grid-item.tsx",
                            "id": "src/components/dga-grid/dga-grid-item.tsx::GridAlignment"
                        }
                    }
                },
                "required": false,
                "optional": false,
                "docs": {
                    "tags": [],
                    "text": "Vertical alignment"
                },
                "getter": false,
                "setter": false,
                "reflect": false
            },
            "grow": {
                "type": "boolean",
                "attribute": "grow",
                "mutable": false,
                "complexType": {
                    "original": "boolean",
                    "resolved": "boolean",
                    "references": {}
                },
                "required": false,
                "optional": false,
                "docs": {
                    "tags": [],
                    "text": "Controls if the item should fill the available space"
                },
                "getter": false,
                "setter": false,
                "reflect": false,
                "defaultValue": "false"
            },
            "shrink": {
                "type": "boolean",
                "attribute": "shrink",
                "mutable": false,
                "complexType": {
                    "original": "boolean",
                    "resolved": "boolean",
                    "references": {}
                },
                "required": false,
                "optional": false,
                "docs": {
                    "tags": [],
                    "text": "Controls if the item should shrink"
                },
                "getter": false,
                "setter": false,
                "reflect": false,
                "defaultValue": "false"
            },
            "order": {
                "type": "number",
                "attribute": "order",
                "mutable": false,
                "complexType": {
                    "original": "number",
                    "resolved": "number",
                    "references": {}
                },
                "required": false,
                "optional": false,
                "docs": {
                    "tags": [],
                    "text": "Order of the grid item"
                },
                "getter": false,
                "setter": false,
                "reflect": false
            },
            "orderXs": {
                "type": "number",
                "attribute": "order-xs",
                "mutable": false,
                "complexType": {
                    "original": "number",
                    "resolved": "number",
                    "references": {}
                },
                "required": false,
                "optional": false,
                "docs": {
                    "tags": [],
                    "text": "Order on xs screens"
                },
                "getter": false,
                "setter": false,
                "reflect": false
            },
            "orderSm": {
                "type": "number",
                "attribute": "order-sm",
                "mutable": false,
                "complexType": {
                    "original": "number",
                    "resolved": "number",
                    "references": {}
                },
                "required": false,
                "optional": false,
                "docs": {
                    "tags": [],
                    "text": "Order on sm screens"
                },
                "getter": false,
                "setter": false,
                "reflect": false
            },
            "orderMd": {
                "type": "number",
                "attribute": "order-md",
                "mutable": false,
                "complexType": {
                    "original": "number",
                    "resolved": "number",
                    "references": {}
                },
                "required": false,
                "optional": false,
                "docs": {
                    "tags": [],
                    "text": "Order on md screens"
                },
                "getter": false,
                "setter": false,
                "reflect": false
            },
            "orderLg": {
                "type": "number",
                "attribute": "order-lg",
                "mutable": false,
                "complexType": {
                    "original": "number",
                    "resolved": "number",
                    "references": {}
                },
                "required": false,
                "optional": false,
                "docs": {
                    "tags": [],
                    "text": "Order on lg screens"
                },
                "getter": false,
                "setter": false,
                "reflect": false
            },
            "orderXl": {
                "type": "number",
                "attribute": "order-xl",
                "mutable": false,
                "complexType": {
                    "original": "number",
                    "resolved": "number",
                    "references": {}
                },
                "required": false,
                "optional": false,
                "docs": {
                    "tags": [],
                    "text": "Order on xl screens"
                },
                "getter": false,
                "setter": false,
                "reflect": false
            }
        };
    }
}
