// @platformscode/core/dist/collection/components/dga-circular-progress-bar/dga-circular-progress-bar.js
import { h } from "@stencil/core";
export class DgaCircularProgressBar {
    constructor() {
        this.RTL = false;
        this.size = '64px';
        this.variant = 'primary';
        this.percentage = 0;
        this.sizes = {
            '200px': 200,
            '160px': 160,
            '120px': 120,
            '80px': 80,
            '64px': 64,
        };
    }
    render() {
        const strokeWidth = this.sizes[this.size] / 10;
        const radius = (this.sizes[this.size] - strokeWidth) / 2;
        const dashArray = 2 * Math.PI * radius;
        const dashOffset = dashArray * (1 - this.percentage / 100);
        return (h("div", { key: '070667ec9a0e72631b65378bcaf9fb09d5cbbd7b', class: "circular-progress-bar", "aria-size": this.size, dir: this.RTL ? 'rtl' : 'ltr' }, h("div", { key: '859b2ecb05fd37c0aa30336796bbac9894ec14d8', class: "circular-progress-bar__label" }, this.variant === 'success' && h("i", { key: '07d34627d42e8456ec06b6d6ef531898f594698b', class: `circular-progress-bar__label--icon hgi-solid hgi-checkmark-circle-02` }), this.variant === 'error' && h("i", { key: '095e4483cad22d878d7a1b9df9be94163a457f9c', class: `circular-progress-bar__label--icon hgi-solid hgi-multiplication-sign-circle` }), this.variant !== 'success' && this.variant !== 'error' && h("h4", { key: '382ecb8376d1e099ec7566270971ca63efd66b60', class: "circular-progress-bar__label--percentage" }, this.percentage, "%"), this.text && h("h5", { key: '9e98c94452f6b00c9273725943aa0f438ced1331', class: `circular-progress-bar__label--text ${this.variant}` }, this.text)), h("svg", { key: '7c8f7542cd0488e2b6bb63f1859a28ebd532fb75', width: this.sizes[this.size], height: this.sizes[this.size], viewBox: `0 0 ${this.sizes[this.size]} ${this.sizes[this.size]}` }, h("circle", { key: 'e95f6b028bc5ec60d88d2421c42b5a9c5b1db098', class: "track", fill: "none", stroke: "#e6e6e6", r: radius, cx: this.sizes[this.size] / 2, cy: this.sizes[this.size] / 2, "stroke-width": strokeWidth }), h("circle", { key: '941e34ca8d6f9ea3b1d67f4f3f27c79ca0218de0', class: `tail tail--${this.variant}`, fill: "none", stroke: "currentColor", r: radius, cx: this.sizes[this.size] / 2, cy: this.sizes[this.size] / 2, "stroke-width": strokeWidth, "stroke-linecap": 'round', "stroke-dasharray": dashArray, "stroke-dashoffset": dashOffset, transform: `rotate(-180 ${this.sizes[this.size] / 2} ${this.sizes[this.size] / 2})`, style: {
                transition: 'stroke-dashoffset 0.5s ease',
            } }))));
    }
    static get is() { return "dga-circular-progress-bar"; }
    static get encapsulation() { return "shadow"; }
    static get originalStyleUrls() {
        return {
            "$": ["dga-circular-progress-bar.scss"]
        };
    }
    static get styleUrls() {
        return {
            "$": ["dga-circular-progress-bar.css"]
        };
    }
    static get properties() {
        return {
            "RTL": {
                "type": "boolean",
                "attribute": "r-t-l",
                "mutable": false,
                "complexType": {
                    "original": "boolean",
                    "resolved": "boolean",
                    "references": {}
                },
                "required": false,
                "optional": false,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false,
                "defaultValue": "false"
            },
            "size": {
                "type": "string",
                "attribute": "size",
                "mutable": false,
                "complexType": {
                    "original": "'200px' | '160px' | '120px' | '80px' | '64px'",
                    "resolved": "\"120px\" | \"160px\" | \"200px\" | \"64px\" | \"80px\"",
                    "references": {}
                },
                "required": false,
                "optional": false,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false,
                "defaultValue": "'64px'"
            },
            "text": {
                "type": "string",
                "attribute": "text",
                "mutable": false,
                "complexType": {
                    "original": "string",
                    "resolved": "string",
                    "references": {}
                },
                "required": false,
                "optional": true,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false
            },
            "variant": {
                "type": "string",
                "attribute": "variant",
                "mutable": false,
                "complexType": {
                    "original": "'primary' | 'neutral' | 'success' | 'error'",
                    "resolved": "\"error\" | \"neutral\" | \"primary\" | \"success\"",
                    "references": {}
                },
                "required": false,
                "optional": true,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false,
                "defaultValue": "'primary'"
            },
            "percentage": {
                "type": "number",
                "attribute": "percentage",
                "mutable": false,
                "complexType": {
                    "original": "number",
                    "resolved": "number",
                    "references": {}
                },
                "required": false,
                "optional": true,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false,
                "defaultValue": "0"
            }
        };
    }
}
