// @platformscode/core/dist/collection/components/dga-skeleton/dga-circle-skeleton.js
import { h } from "@stencil/core";
export class CircleSkeleton {
    constructor() {
        this.type = '1';
        this.width = '80px';
    }
    render() {
        return (h("div", { key: '1c4aba90588b8e23561c6999447db5a2bceccff4', style: { width: this.width, height: this.width }, class: `circle_skeleton type${this.type}` }));
    }
    static get is() { return "dga-circle-skeleton"; }
    static get encapsulation() { return "shadow"; }
    static get originalStyleUrls() {
        return {
            "$": ["./style.scss"]
        };
    }
    static get styleUrls() {
        return {
            "$": ["style.css"]
        };
    }
    static get properties() {
        return {
            "type": {
                "type": "string",
                "attribute": "type",
                "mutable": false,
                "complexType": {
                    "original": "'1' | '2'",
                    "resolved": "\"1\" | \"2\"",
                    "references": {}
                },
                "required": false,
                "optional": false,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false,
                "defaultValue": "'1'"
            },
            "width": {
                "type": "string",
                "attribute": "width",
                "mutable": false,
                "complexType": {
                    "original": "'24px' | '64px' | '48px' | '80px' | '120px' | '170px' | '240px'",
                    "resolved": "\"120px\" | \"170px\" | \"240px\" | \"24px\" | \"48px\" | \"64px\" | \"80px\"",
                    "references": {}
                },
                "required": false,
                "optional": false,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false,
                "defaultValue": "'80px'"
            }
        };
    }
}
