// @platformscode/core/dist/collection/components/dga-slideout-menu/dga-slideout-menu-items.js
import { h, Host } from "@stencil/core";
export class DgaSlideoutMenuItems {
    render() {
        return (h(Host, { key: 'f4c67827f20a53f82116256e37dda0d8a86d4a71' }, h("slot", { key: 'bb9f03cb866a3e08a91fe83cd9681f4dbfabd213' })));
    }
    static get is() { return "dga-slideout-menu-items"; }
    static get encapsulation() { return "shadow"; }
    static get originalStyleUrls() {
        return {
            "$": ["dga-slideout-menu.scss"]
        };
    }
    static get styleUrls() {
        return {
            "$": ["dga-slideout-menu.css"]
        };
    }
    static get properties() {
        return {
            "title": {
                "type": "string",
                "attribute": "title",
                "mutable": false,
                "complexType": {
                    "original": "string",
                    "resolved": "string",
                    "references": {}
                },
                "required": false,
                "optional": false,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false
            },
            "description": {
                "type": "string",
                "attribute": "description",
                "mutable": false,
                "complexType": {
                    "original": "string",
                    "resolved": "string",
                    "references": {}
                },
                "required": false,
                "optional": false,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false
            }
        };
    }
}
