// @platformscode/core/dist/collection/components/dga-slideout-menu/dga-slideout-menu-header.js
import { h } from "@stencil/core";
export class DgaSlideoutMenuHeader {
    constructor() {
        // @Prop() onClose?: () => void;
        this.handleClose = () => {
            // Find parent slideout menu
            const slideoutMenu = this.el.closest('dga-slideout-menu');
            if (slideoutMenu) {
                // Access parent's onClose
                const parentClose = slideoutMenu['onClose'];
                if (typeof parentClose === 'function') {
                    parentClose();
                }
            }
        };
    }
    render() {
        return (
        // <div class="dga-slideout-menu">
        h("div", { key: '2182354df46ee4fb78c0a96d80194d95ffa3a5c7', class: "dga-slideout-menu__header" }, h("dga-featured-icon", { key: 'afff9bef48efae4c7b3c9035ebccda801b583659', icon: {
                name: 'add-circle',
                variant: 'stroke',
                type: 'standard',
            }, size: 'sm', color: 'brand', variant: 'light' }), h("dga-flex", { key: 'e39ad7b1a9eb6c12480fc17588a61fe272d0b56c', direction: "column", spacing: 4, class: "dga-slideout-menu__header-content" }, h("h3", { key: '6aa3a2c38da28e2e0d7489ca18ad48363cdcca5f', class: "dga-slideout-menu__header-title" }, this.title), h("p", { key: '95da6284f46fe878108c4a134e485d5b3aaffb34', class: "dga-slideout-menu__header-description" }, this.description)), h("dga-button-v2", { key: '5a6ceb040ef1fc27710694cf842bc25eebebc68e', iconOnly: true, leadIcon: true, leadIconProps: {
                name: 'multiplication-sign',
                variant: 'stroke',
                type: 'standard',
                color: "#161616"
            }, variant: "transparent", onClick: this.handleClose }))
        // </div>
        );
    }
    static get is() { return "dga-slideout-menu-header"; }
    static get encapsulation() { return "shadow"; }
    static get originalStyleUrls() {
        return {
            "$": ["dga-slideout-menu-header.scss"]
        };
    }
    static get styleUrls() {
        return {
            "$": ["dga-slideout-menu-header.css"]
        };
    }
    static get properties() {
        return {
            "title": {
                "type": "string",
                "attribute": "title",
                "mutable": false,
                "complexType": {
                    "original": "string",
                    "resolved": "string",
                    "references": {}
                },
                "required": false,
                "optional": false,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false
            },
            "description": {
                "type": "string",
                "attribute": "description",
                "mutable": false,
                "complexType": {
                    "original": "string",
                    "resolved": "string",
                    "references": {}
                },
                "required": false,
                "optional": false,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false
            }
        };
    }
    static get elementRef() { return "el"; }
}
