// @platformscode/core/dist/collection/components/dga-codeSnippet/dga-codeSnippet.js
import { h } from "@stencil/core";
export class CodeSnippet {
    constructor() {
        var _a;
        this.snippets = {};
        this.showButton = false;
        this.type = 'multi-line';
        this.showSidebar = false;
        this.showTabList = false;
        this.showMore = false;
        this.activeTab = ((_a = Object.keys(this.snippets || {})) === null || _a === void 0 ? void 0 : _a[0]) || '';
    }
    componentDidLoad() {
        var _a;
        this.activeTab = ((_a = Object.keys(this.snippets || {})) === null || _a === void 0 ? void 0 : _a[0]) || '';
    }
    handleTabClick(tab) {
        this.activeTab = tab;
        this.showMore = false;
    }
    handleCopy() {
        navigator.clipboard.writeText(this.snippets[this.activeTab]);
    }
    renderWithLineNumbers(code) {
        var _a;
        const isMultiLine = this.type === 'multi-line';
        if (isMultiLine) {
            return (_a = code === null || code === void 0 ? void 0 : code.split('\n')) === null || _a === void 0 ? void 0 : _a.map((line, index) => (h("div", { key: index, class: "code-snippet-code-line" }, this.showSidebar && (h("div", { class: "code-snippet-number_container" }, h("span", { class: "code-snippet-line-number" }, index + 1))), h("span", { class: "code-snippet-line-content" }, line))));
        }
        else {
            return code
                .split('\n')
                .slice(0, 1)
                .map((line, index) => (h("div", { key: index, class: "code-snippet-code-line" }, h("span", { class: "code-snippet-line-content green" }, line))));
        }
    }
    render() {
        return (h("div", { key: 'd956081ad5e9447527e19bb54230239351be3591', class: "code-snippet-container" }, this.showTabList && (h("div", { key: 'cbaac6ed73b1d3a04afb33c19ee9bbe082403bbe', class: "code-snippet-tabs" }, h("dga-tabs", { key: '11e74c6f975da810bb4091a1db6b117f9acac275', divider: true, orientation: "horizontal", size: "lg", tabsList: Object.keys(this.snippets).map(tab => ({
                label: tab,
                onClick: () => this.handleTabClick(tab),
            })) }))), h("pre", { key: '72b752f74ca5b3dd3eb4f75a191102b48964d193', class: "code-snippet-code-display" }, h("dga-button-v2", { key: '07d7d58f13132c4a0dd5ab44bcf01b88bd857137', size: "lg", iconOnly: true, variant: 'subtle', leadIcon: true, leadIconProps: {
                name: 'copy-01',
                size: 24,
                color: '#161616',
                variant: 'stroke',
            }, class: `code-snippet-icon ${this.type}`, onClick: () => this.handleCopy(), onColor: false }), h("code", { key: '83ff0cc16564fe090046c18e128ca0c7a634716a', style: {
                height: this.showButton && this.type === 'multi-line' && !this.showMore ? '150px' : 'auto',
            } }, this.renderWithLineNumbers(this.snippets[this.activeTab])), this.showButton && this.type === 'multi-line' && (h("div", { key: '9d130a5dfaf6e51cd496ddc6e086b0c4f9ce55b5', class: "code-snippet-wrapper" }, h("dga-button-v2", { key: '84d734acb82a90cbcb7410bcf7f692ae89695207', label: this.showMore ? 'Show Less' : 'Show More', size: "md", trailIcon: true, variant: 'subtle', trailIconProps: { name: !this.showMore ? 'arrow-down-01' : 'arrow-up-01', size: 20, color: '#161616', variant: 'stroke' }, onClick: () => (this.showMore = !this.showMore), onColor: false }))))));
    }
    static get is() { return "dga-codesnippet"; }
    static get originalStyleUrls() {
        return {
            "$": ["./style.scss"]
        };
    }
    static get styleUrls() {
        return {
            "$": ["style.css"]
        };
    }
    static get properties() {
        return {
            "snippets": {
                "type": "unknown",
                "attribute": "snippets",
                "mutable": false,
                "complexType": {
                    "original": "{ [key: string]: string }",
                    "resolved": "{ [key: string]: string; }",
                    "references": {}
                },
                "required": false,
                "optional": false,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "defaultValue": "{}"
            },
            "showButton": {
                "type": "boolean",
                "attribute": "show-button",
                "mutable": false,
                "complexType": {
                    "original": "boolean",
                    "resolved": "boolean",
                    "references": {}
                },
                "required": false,
                "optional": false,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false,
                "defaultValue": "false"
            },
            "type": {
                "type": "string",
                "attribute": "type",
                "mutable": false,
                "complexType": {
                    "original": "'multi-line' | 'single'",
                    "resolved": "\"multi-line\" | \"single\"",
                    "references": {}
                },
                "required": false,
                "optional": false,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false,
                "defaultValue": "'multi-line'"
            },
            "showSidebar": {
                "type": "boolean",
                "attribute": "show-sidebar",
                "mutable": false,
                "complexType": {
                    "original": "boolean",
                    "resolved": "boolean",
                    "references": {}
                },
                "required": false,
                "optional": false,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false,
                "defaultValue": "false"
            },
            "showTabList": {
                "type": "boolean",
                "attribute": "show-tab-list",
                "mutable": false,
                "complexType": {
                    "original": "boolean",
                    "resolved": "boolean",
                    "references": {}
                },
                "required": false,
                "optional": false,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false,
                "defaultValue": "false"
            }
        };
    }
    static get states() {
        return {
            "showMore": {},
            "activeTab": {}
        };
    }
}
