// @platformscode/core/dist/collection/components/dga-menu/dga-menu-group.js
import { h } from "@stencil/core";
export class DgaMenuGroup {
    render() {
        return (h("div", { key: 'f83f05369c4927b2081e1b180480c59e9d73e540', class: "menu-group" }, this.groupLabel && h("span", { key: '557db4ae2b7333a6bec917979c657e6392f984ca', class: "menu-group__label" }, this.groupLabel), h("ul", { key: '0b66140bfde01c8f8df1c2d59433f7ccbe6f37b4', class: "menu-group__items" }, h("slot", { key: '720f7613be74e425cdb4204fbaa57e61c98ce153' }))));
    }
    static get is() { return "dga-menu-group"; }
    static get encapsulation() { return "shadow"; }
    static get originalStyleUrls() {
        return {
            "$": ["dga-menu-group.scss"]
        };
    }
    static get styleUrls() {
        return {
            "$": ["dga-menu-group.css"]
        };
    }
    static get properties() {
        return {
            "groupLabel": {
                "type": "string",
                "attribute": "group-label",
                "mutable": false,
                "complexType": {
                    "original": "string",
                    "resolved": "string",
                    "references": {}
                },
                "required": false,
                "optional": true,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false
            }
        };
    }
}
