// @platformscode/core/dist/collection/components/dga-menu/dga-menu-item.js
import { h } from "@stencil/core";
export class DgaMenuItem {
    constructor() {
        this.disabled = false;
        this.hasTrailElement = false;
        this.hasLeadIcon = false;
        this.pressed = false;
    }
    handleClick(event) {
        if (!this.disabled) {
            this.onClick.emit(event);
        }
    }
    render() {
        return (h("li", { key: '653c9c0bd9e3f3b38f4d414bf55fdfc1f78f9751', class: `menu-item ${this.disabled ? 'disabled' : ''} ${this.pressed ? "pressed" : ""}`, onMouseDown: () => this.pressed = true, onMouseUp: () => this.pressed = false, tabindex: "0", onClick: this.handleClick }, this.hasLeadIcon && h("span", { key: '92e16bee31b7f248363dbfe9b9600db9283d4dd1', class: "menu-item__icon" }, h("slot", { key: '565479b56817e98ad2f454fab7e5d485a3d6029b', name: "lead-icon" })), h("span", { key: '53a53f7a22b50c3fa30e60b759e051787048c8c6', class: "menu-item__label" }, this.label), this.hasTrailElement && h("div", { key: '1bad8270fb653e6211e3e1fc9005d0fd25bdaf09', class: "menu-item__trail-element" }, h("slot", { key: '371654c563a50b14a90cdd42a4789e9e5ce26fcf', name: "trail-element" }))));
    }
    static get is() { return "dga-menu-item"; }
    static get encapsulation() { return "shadow"; }
    static get originalStyleUrls() {
        return {
            "$": ["dga-menu-item.scss"]
        };
    }
    static get styleUrls() {
        return {
            "$": ["dga-menu-item.css"]
        };
    }
    static get properties() {
        return {
            "label": {
                "type": "string",
                "attribute": "label",
                "mutable": false,
                "complexType": {
                    "original": "string",
                    "resolved": "string",
                    "references": {}
                },
                "required": false,
                "optional": true,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false
            },
            "disabled": {
                "type": "boolean",
                "attribute": "disabled",
                "mutable": false,
                "complexType": {
                    "original": "boolean",
                    "resolved": "boolean",
                    "references": {}
                },
                "required": false,
                "optional": true,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false,
                "defaultValue": "false"
            },
            "hasTrailElement": {
                "type": "boolean",
                "attribute": "has-trail-element",
                "mutable": false,
                "complexType": {
                    "original": "boolean",
                    "resolved": "boolean",
                    "references": {}
                },
                "required": false,
                "optional": true,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false,
                "defaultValue": "false"
            },
            "hasLeadIcon": {
                "type": "boolean",
                "attribute": "has-lead-icon",
                "mutable": false,
                "complexType": {
                    "original": "boolean",
                    "resolved": "boolean",
                    "references": {}
                },
                "required": false,
                "optional": true,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false,
                "defaultValue": "false"
            }
        };
    }
    static get states() {
        return {
            "pressed": {}
        };
    }
    static get events() {
        return [{
                "method": "onClick",
                "name": "onClick",
                "bubbles": true,
                "cancelable": true,
                "composed": true,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "complexType": {
                    "original": "MouseEvent",
                    "resolved": "MouseEvent",
                    "references": {
                        "MouseEvent": {
                            "location": "global",
                            "id": "global::MouseEvent"
                        }
                    }
                }
            }];
    }
}
