// @platformscode/core/dist/collection/components/dga-tooltip/dga-tooltip.js
import { Host, h } from "@stencil/core";
import { SxHandler } from "../../utils/sx-handler";
export class DgaTooltip {
    constructor() {
        this.beakPlacement = 'none';
        this.beakAlignment = 'center';
        this.direction = 'top';
        this.icon = false;
        this.theme = 'light';
        this.isHovered = false;
        this.icons = {
            'success-check': {
                icon: 'hgi-checkmark-circle-02',
                color: '#067647',
            },
            'blue-info': {
                icon: 'hgi-information-circle',
                color: '#175CD3',
            },
            'neutral-info': {
                icon: 'hgi-information-circle',
                color: '#384250',
            },
            'neutral-?': {
                icon: 'hgi-help-circle',
                color: '#384250',
            },
            'warning-!-circle': {
                icon: 'hgi-alert-circle',
                color: '#B54708',
            },
            'warning-!-triangle': {
                icon: 'hgi-sharp hgi-alert-02',
                color: '#B54708',
            },
            'error-!': {
                icon: 'hgi-alert-circle',
                color: '#B42318',
            },
            'error-x': {
                icon: 'hgi-cancel-circle',
                color: '#B42318',
            },
        };
        this.handleMouseEnter = () => {
            this.isHovered = true;
        };
        this.handleMouseLeave = () => {
            this.isHovered = false;
        };
    }
    sxChanged() {
        this.sxHandler.processSx(this.sx);
    }
    componentWillLoad() {
        this.sxHandler = new SxHandler(this.el, 'dga-tooltip');
    }
    componentDidLoad() {
        this.sxHandler.processSx(this.sx);
    }
    disconnectedCallback() {
        this.sxHandler.cleanup();
    }
    render() {
        return (h(Host, { key: 'dac010cf2949f3a2bb55f4a04db412518c7881a7', id: this.sxHandler.getId(), style: this.sxHandler.getStyles() }, h("div", { key: '5638f89d9b5fc1e89b4453513f2065b795afef38', class: "dga-tooltip", "data-theme": this.theme, "data-beak-direction": `${this.beakPlacement}-${this.beakAlignment}` }, this.isHovered && (h("div", { key: '7f897fd2fe2bfe6e8dbf5b73da12619b16f9de74', class: "dga-tooltip-main", "data-direction": this.direction }, h("div", { key: 'd60d1a78d88557e5b74b236025bcee97e9649fdb', class: "dga-tooltip-main__wrapper" }, this.beakPlacement !== 'none' && h("span", { key: '5006b4c69d9d321c612e924ebb8b81135001ce4a', class: "dga-tooltip-main__beak", "data-beak-direction": `${this.beakPlacement}-${this.beakAlignment}` }), this.icon && (h("span", { key: '34458cddbee70eacd42be9c52fb6b30d9142ae8f', class: "dga-tooltip-main__icon" }, h("i", { key: '0cc42460234f3181ebc03482ddb4d063f6a5a76f', class: `${this.feedbackIcon.style === 'icon-only' ? 'hgi-stroke' : 'hgi-solid'} ${this.icons[this.feedbackIcon.type].icon}`, style: { color: this.icons[this.feedbackIcon.type].color, fontSize: `${this.feedbackIcon.size}px` } }))), h("div", { key: '3a07d704cc3e0159e2512cc4437678b4979aff8b', class: "dga-tooltip-main__content" }, this.tooltipTitle && h("span", { key: '26f43b95813f6dc0421bc10c77759fc4a2bcdd66', class: "dga-tooltip-main__content__title" }, this.tooltipTitle), this.helperText && h("span", { key: 'a97ce2ff26adee4dea840ef372a613b2c087c9ed', class: "dga-tooltip-main__content__helper-text" }, this.helperText))))), h("div", { key: '713a91f9198fa6c9ff5695fa651ffd26a69a14b5', class: "dga-tooltip-item", onMouseEnter: this.handleMouseEnter, onMouseLeave: this.handleMouseLeave }, h("slot", { key: '022ccaf8b9b97bde444700784a1a40a41435684b' })))));
    }
    static get is() { return "dga-tooltip"; }
    static get originalStyleUrls() {
        return {
            "$": ["dga-tooltip.scss"]
        };
    }
    static get styleUrls() {
        return {
            "$": ["dga-tooltip.css"]
        };
    }
    static get properties() {
        return {
            "sx": {
                "type": "string",
                "attribute": "sx",
                "mutable": false,
                "complexType": {
                    "original": "SxProps | string | undefined",
                    "resolved": "SxProps | string",
                    "references": {
                        "SxProps": {
                            "location": "import",
                            "path": "../../utils/sx-handler",
                            "id": "src/utils/sx-handler.ts::SxProps"
                        }
                    }
                },
                "required": false,
                "optional": true,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false
            },
            "beakPlacement": {
                "type": "string",
                "attribute": "beak-placement",
                "mutable": false,
                "complexType": {
                    "original": "BeakPlacement",
                    "resolved": "\"bottom\" | \"left\" | \"none\" | \"right\" | \"top\"",
                    "references": {
                        "BeakPlacement": {
                            "location": "local",
                            "path": "/home/mamoud/Documents/temp/components/platform-code-cross-platform-storybook/packages/core/src/components/dga-tooltip/dga-tooltip.tsx",
                            "id": "src/components/dga-tooltip/dga-tooltip.tsx::BeakPlacement"
                        }
                    }
                },
                "required": false,
                "optional": false,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false,
                "defaultValue": "'none'"
            },
            "beakAlignment": {
                "type": "string",
                "attribute": "beak-alignment",
                "mutable": false,
                "complexType": {
                    "original": "BeakAlignment",
                    "resolved": "\"center\" | \"end\" | \"start\"",
                    "references": {
                        "BeakAlignment": {
                            "location": "local",
                            "path": "/home/mamoud/Documents/temp/components/platform-code-cross-platform-storybook/packages/core/src/components/dga-tooltip/dga-tooltip.tsx",
                            "id": "src/components/dga-tooltip/dga-tooltip.tsx::BeakAlignment"
                        }
                    }
                },
                "required": false,
                "optional": false,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false,
                "defaultValue": "'center'"
            },
            "direction": {
                "type": "string",
                "attribute": "direction",
                "mutable": false,
                "complexType": {
                    "original": "Direction",
                    "resolved": "\"bottom\" | \"left\" | \"right\" | \"top\"",
                    "references": {
                        "Direction": {
                            "location": "local",
                            "path": "/home/mamoud/Documents/temp/components/platform-code-cross-platform-storybook/packages/core/src/components/dga-tooltip/dga-tooltip.tsx",
                            "id": "src/components/dga-tooltip/dga-tooltip.tsx::Direction"
                        }
                    }
                },
                "required": false,
                "optional": false,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false,
                "defaultValue": "'top'"
            },
            "tooltipTitle": {
                "type": "string",
                "attribute": "title",
                "mutable": false,
                "complexType": {
                    "original": "string",
                    "resolved": "string",
                    "references": {}
                },
                "required": false,
                "optional": true,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false
            },
            "helperText": {
                "type": "string",
                "attribute": "helper-text",
                "mutable": false,
                "complexType": {
                    "original": "string",
                    "resolved": "string",
                    "references": {}
                },
                "required": false,
                "optional": true,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false
            },
            "icon": {
                "type": "boolean",
                "attribute": "icon",
                "mutable": true,
                "complexType": {
                    "original": "boolean",
                    "resolved": "boolean",
                    "references": {}
                },
                "required": false,
                "optional": false,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": true,
                "defaultValue": "false"
            },
            "feedbackIcon": {
                "type": "unknown",
                "attribute": "feedback-icon",
                "mutable": false,
                "complexType": {
                    "original": "FeedbackIcon",
                    "resolved": "FeedbackIcon",
                    "references": {
                        "FeedbackIcon": {
                            "location": "local",
                            "path": "/home/mamoud/Documents/temp/components/platform-code-cross-platform-storybook/packages/core/src/components/dga-tooltip/dga-tooltip.tsx",
                            "id": "src/components/dga-tooltip/dga-tooltip.tsx::FeedbackIcon"
                        }
                    }
                },
                "required": false,
                "optional": false,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false
            },
            "theme": {
                "type": "string",
                "attribute": "theme",
                "mutable": false,
                "complexType": {
                    "original": "Theme",
                    "resolved": "\"dark\" | \"light\"",
                    "references": {
                        "Theme": {
                            "location": "local",
                            "path": "/home/mamoud/Documents/temp/components/platform-code-cross-platform-storybook/packages/core/src/components/dga-tooltip/dga-tooltip.tsx",
                            "id": "src/components/dga-tooltip/dga-tooltip.tsx::Theme"
                        }
                    }
                },
                "required": false,
                "optional": false,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false,
                "defaultValue": "'light'"
            }
        };
    }
    static get states() {
        return {
            "isHovered": {}
        };
    }
    static get elementRef() { return "el"; }
    static get watchers() {
        return [{
                "propName": "sx",
                "methodName": "sxChanged"
            }];
    }
}
