// @platformscode/core/dist/collection/components/dga-radial-stepper/dga-radial-stepper.js
import { Host, h } from "@stencil/core";
export class DgaRadialStepper {
    constructor() {
        this.RTL = false;
        this.size = '64px';
        this.showText = true;
        this.steps = [];
        this.activeStep = 1;
        this.variant = 'primary';
        this.showStepName = true;
        this.showStepDescription = true;
        this.sizes = {
            '120px': 120,
            '80px': 80,
            '64px': 64,
            '48px': 48,
            '40px': 40,
        };
    }
    updateInternalPercentage() {
        if ((this.activeStep >= this.steps.length + 1) || (this.activeStep < 1)) {
            return;
        }
        this.internalPercentage = (this.activeStep / this.steps.length) * 100;
        this.internalActiveStep = this.activeStep;
    }
    componentDidLoad() {
        this.updateInternalPercentage();
    }
    render() {
        var _a, _b, _c, _d, _e, _f, _g, _h;
        const strokeWidth = this.sizes[this.size] / 10;
        const radius = (this.sizes[this.size] - strokeWidth) / 2;
        const dashArray = 2 * Math.PI * radius;
        const dashOffset = (this.RTL ? 1 : -1) * dashArray * (1 - this.internalPercentage / 100);
        return (h(Host, { key: '64ed9b018cc60a6df620a01c6407ed0b91fb386f' }, h("dga-flex", { key: '6e97c1979c02957f59c8e7648de03997e264e71d', spacing: 8, align: 'center', dir: this.RTL ? "rtl" : "ltr", class: `dga-radial-stepper--${this.size} dga-radial-stepper--${this.variant}`, "aria-size": this.size }, h("div", { key: '0bdc1f4abc89c6b9db6d6cafb7ae39033fae9be6', class: "dga-circular-stepper", "aria-size": this.size }, h("div", { key: 'd1b21815ead4687ea17e1d191bb8c6fb7ecd21d0', class: "dga-circular-stepper__label" }, this.showText && this.steps.length > 0 && h("h4", { key: '71e155fc8b6dd10065a6bfdda2065f4da2a105d5', class: `dga-circular-stepper__label--text ${this.variant}` }, this.RTL ? `${this.internalActiveStep} من ${this.steps.length}` : `${this.internalActiveStep} of ${this.steps.length}`)), h("svg", { key: 'ba0e123d3016ad6e29bb19cb781c6dd2d4f46c42', width: this.sizes[this.size], height: this.sizes[this.size], viewBox: `0 0 ${this.sizes[this.size]} ${this.sizes[this.size]}` }, h("circle", { key: 'fab4d864d6e356192c329e5f53304e0ce0cbb063', class: `track track--${this.variant}`, fill: "none", stroke: "#e6e6e6",
            // stroke="currentColor"
            r: radius, cx: this.sizes[this.size] / 2, cy: this.sizes[this.size] / 2, "stroke-width": strokeWidth }), h("circle", { key: '19b203de05f2ef9051437a6f74fefbfe45c38928', class: `tail tail--${this.variant}`, fill: "none", stroke: "currentColor", r: radius, cx: this.sizes[this.size] / 2, cy: this.sizes[this.size] / 2, "stroke-width": strokeWidth, "stroke-linecap": 'round', "stroke-dasharray": dashArray, "stroke-dashoffset": dashOffset, transform: `rotate(-90 ${this.sizes[this.size] / 2} ${this.sizes[this.size] / 2})`, style: {
                transition: 'stroke-dashoffset 0.5s ease',
            } }))), h("dga-flex", { key: '85b855c467c5418e95b6ebf94e143e7e29b48723', direction: 'column', spacing: 4 }, ((_a = this.steps[this.internalActiveStep - 1]) === null || _a === void 0 ? void 0 : _a.showPreStep) && h("span", { key: '88fe0e0bb5265377a581517c98cc9bdc6b2337a1', class: "dga-radial-stepper__pre-name" }, "Previous: ", (_b = this.steps[this.internalActiveStep - 1]) === null || _b === void 0 ? void 0 :
            _b.stepName), this.showStepName && ((_c = this.steps[this.internalActiveStep - 1]) === null || _c === void 0 ? void 0 : _c.stepName) && h("h3", { key: '19c5390d637e90fe754b171a671a37c877a657f1', class: "dga-radial-stepper__name" }, (_d = this.steps[this.internalActiveStep - 1]) === null || _d === void 0 ? void 0 : _d.stepName), this.showStepDescription && ((_e = this.steps[this.internalActiveStep - 1]) === null || _e === void 0 ? void 0 : _e.stepDescription) && h("h5", { key: 'ac296823e7dafc242f2f7e430796b67121b61b3c', class: "dga-radial-stepper__desc" }, (_f = this.steps[this.internalActiveStep - 1]) === null || _f === void 0 ? void 0 : _f.stepDescription), ((_g = this.steps[this.internalActiveStep - 1]) === null || _g === void 0 ? void 0 : _g.showNextStep) && h("span", { key: 'f996e0a11cf7ac90fcf4b15a1619b8186178b7e9', class: "dga-radial-stepper__next-name" }, "Next: ", (_h = this.steps[this.internalActiveStep]) === null || _h === void 0 ? void 0 :
            _h.stepName)))));
    }
    static get is() { return "dga-radial-stepper"; }
    static get encapsulation() { return "shadow"; }
    static get originalStyleUrls() {
        return {
            "$": ["dga-radial-stepper.scss"]
        };
    }
    static get styleUrls() {
        return {
            "$": ["dga-radial-stepper.css"]
        };
    }
    static get properties() {
        return {
            "RTL": {
                "type": "boolean",
                "attribute": "r-t-l",
                "mutable": false,
                "complexType": {
                    "original": "boolean",
                    "resolved": "boolean",
                    "references": {}
                },
                "required": false,
                "optional": false,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false,
                "defaultValue": "false"
            },
            "size": {
                "type": "string",
                "attribute": "size",
                "mutable": false,
                "complexType": {
                    "original": "'120px' | '80px' | '64px' | '48px' | '40px'",
                    "resolved": "\"120px\" | \"40px\" | \"48px\" | \"64px\" | \"80px\"",
                    "references": {}
                },
                "required": false,
                "optional": false,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false,
                "defaultValue": "'64px'"
            },
            "showText": {
                "type": "boolean",
                "attribute": "show-text",
                "mutable": false,
                "complexType": {
                    "original": "boolean",
                    "resolved": "boolean",
                    "references": {}
                },
                "required": false,
                "optional": false,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false,
                "defaultValue": "true"
            },
            "steps": {
                "type": "unknown",
                "attribute": "steps",
                "mutable": false,
                "complexType": {
                    "original": "Step[]",
                    "resolved": "Step[]",
                    "references": {
                        "Step": {
                            "location": "local",
                            "path": "/home/mamoud/Documents/temp/components/platform-code-cross-platform-storybook/packages/core/src/components/dga-radial-stepper/dga-radial-stepper.tsx",
                            "id": "src/components/dga-radial-stepper/dga-radial-stepper.tsx::Step"
                        }
                    }
                },
                "required": false,
                "optional": false,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "defaultValue": "[]"
            },
            "activeStep": {
                "type": "number",
                "attribute": "active-step",
                "mutable": false,
                "complexType": {
                    "original": "number",
                    "resolved": "number",
                    "references": {}
                },
                "required": false,
                "optional": false,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false,
                "defaultValue": "1"
            },
            "variant": {
                "type": "string",
                "attribute": "variant",
                "mutable": false,
                "complexType": {
                    "original": "'primary' | 'neutral' | 'on-color'",
                    "resolved": "\"neutral\" | \"on-color\" | \"primary\"",
                    "references": {}
                },
                "required": false,
                "optional": true,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false,
                "defaultValue": "'primary'"
            },
            "showStepName": {
                "type": "boolean",
                "attribute": "show-step-name",
                "mutable": false,
                "complexType": {
                    "original": "boolean",
                    "resolved": "boolean",
                    "references": {}
                },
                "required": false,
                "optional": false,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false,
                "defaultValue": "true"
            },
            "showStepDescription": {
                "type": "boolean",
                "attribute": "show-step-description",
                "mutable": false,
                "complexType": {
                    "original": "boolean",
                    "resolved": "boolean",
                    "references": {}
                },
                "required": false,
                "optional": false,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false,
                "defaultValue": "true"
            }
        };
    }
    static get states() {
        return {
            "internalPercentage": {},
            "internalActiveStep": {}
        };
    }
    static get watchers() {
        return [{
                "propName": "steps",
                "methodName": "updateInternalPercentage"
            }, {
                "propName": "activeStep",
                "methodName": "updateInternalPercentage"
            }];
    }
}
