// @platformscode/core/dist/collection/components/dga-inline-alert/dga-inline-alert.js
import { h, Fragment } from "@stencil/core";
import { guid } from "../../utils";
export class DgaInlineAlert {
    constructor() {
        this.type = 'neutral';
        this.isactionButtons = true;
        this.isCloseButton = true;
        this.colored = false;
        this.buttonsList = [];
        this.icons = {
            error: 'help-circle',
            warning: 'alert-02',
            success: 'tick-02',
            info: 'help-circle',
            neutral: 'help-circle',
        };
    }
    render() {
        return (h("div", { key: 'cf31fdd990ac2405d139c84b31103753e3a3cc86', class: `alert alert--${this.type} ${this.colored ? `alert--${this.type}-bg` : ''}` }, h("div", { key: '79dd235b6e16da480894aec2659d0b951296bee9', class: `alert__header ${this.helperText ? '' : 'dga-flex-center-y'}` }, h("div", { key: '1762f1290159341040b50b2ff7a35504b47beb0c', class: `featured-icon featured-icon--circle featured-icon--md featured-icon--light-${this.type === 'neutral' ? 'gray' : this.type}` }, h("dga-icon", { key: '53cd5e9ddfa85a93c49a8fd223447ac12d2cb670', name: this.icons[this.type], variant: "stroke", size: 20 })), h("div", { key: 'fc0f70fadcf82aee08de90b2377ed9faed07dcfa', class: "alert__body" }, h("h4", { key: '5e30c0ccc81f9d1e9f80afbacb8f44bb59c83b99', class: "alert__title text-md-semibold" }, this.leadText), this.helperText && h("p", { key: 'c449f978f57877792783c3ba8f48c5de47cd1fd9', class: "alert__content text-sm-regular" }, this.helperText)), this.isCloseButton && (h("div", { key: 'd6de8d168f45954d9759543d23f55ac77292ee31', class: "alert__close-btn" }, h("button", { key: '5dd488e3c55dd653135c56cd23d0e73f08a7dede', class: "dga-btn dga-btn--close dga-btn--icon dga-btn--md", onClick: e => this.onClose && this.onClose(e) }, h("span", { key: 'df2ac196ba828d02985da95357e8618f87b9aa66', class: "dga-btn-icon" }, h("dga-icon", { key: 'dfe8a7eb3a8f3bf859f3b0e22f1a912469eeccc0', name: "multiplication-sign", variant: "stroke", size: 20 })))))), this.isactionButtons && this.buttonsList.length > 0 && (h(Fragment, { key: '461782a42956e4df6150c6a2236bfa079273d0ed' }, h("div", { key: 'bdbf9703075d02b4e1bed0d08746f104bce33ea1', class: "alert__action" }, this.buttonsList.map(({ id, onClick, exraClasses, title, label }) => {
            return (h("button", Object.assign({ key: id ? id : guid(), id: id ? id : guid(), onClick: e => onClick(e), class: `dga-btn dga-btn--subtle dga-btn--md ${exraClasses}` }, (title && { title: title })), label));
        })), h("div", { key: 'cab7651a92e0cd1345b527cdb6ade1902b98cda7', class: "alert__action-mobile" }, this.buttonsList.map(({ id, onClick, exraClasses, title, label }) => {
            return (h("button", Object.assign({ key: id ? id : guid(), id: id ? id : guid(), onClick: e => onClick(e), class: `btn btn--secondary btn--lg ${exraClasses}` }, (title && { title: title })), label));
        }))))));
    }
    static get is() { return "dga-inline-alert"; }
    static get originalStyleUrls() {
        return {
            "$": ["dga-inline-alert.scss"]
        };
    }
    static get styleUrls() {
        return {
            "$": ["dga-inline-alert.css"]
        };
    }
    static get properties() {
        return {
            "type": {
                "type": "string",
                "attribute": "type",
                "mutable": false,
                "complexType": {
                    "original": "'neutral' | 'info' | 'error' | 'success' | 'warning'",
                    "resolved": "\"error\" | \"info\" | \"neutral\" | \"success\" | \"warning\"",
                    "references": {}
                },
                "required": false,
                "optional": false,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false,
                "defaultValue": "'neutral'"
            },
            "leadText": {
                "type": "string",
                "attribute": "lead-text",
                "mutable": false,
                "complexType": {
                    "original": "string",
                    "resolved": "string",
                    "references": {}
                },
                "required": false,
                "optional": false,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false
            },
            "helperText": {
                "type": "string",
                "attribute": "helper-text",
                "mutable": false,
                "complexType": {
                    "original": "string",
                    "resolved": "string",
                    "references": {}
                },
                "required": false,
                "optional": true,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false
            },
            "isactionButtons": {
                "type": "boolean",
                "attribute": "isaction-buttons",
                "mutable": false,
                "complexType": {
                    "original": "boolean",
                    "resolved": "boolean",
                    "references": {}
                },
                "required": false,
                "optional": false,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false,
                "defaultValue": "true"
            },
            "isCloseButton": {
                "type": "boolean",
                "attribute": "is-close-button",
                "mutable": false,
                "complexType": {
                    "original": "boolean",
                    "resolved": "boolean",
                    "references": {}
                },
                "required": false,
                "optional": false,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false,
                "defaultValue": "true"
            },
            "onClose": {
                "type": "unknown",
                "attribute": "on-close",
                "mutable": false,
                "complexType": {
                    "original": "(event: any) => void",
                    "resolved": "(event: any) => void",
                    "references": {}
                },
                "required": false,
                "optional": true,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false
            },
            "colored": {
                "type": "boolean",
                "attribute": "colored",
                "mutable": false,
                "complexType": {
                    "original": "boolean",
                    "resolved": "boolean",
                    "references": {}
                },
                "required": false,
                "optional": false,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false,
                "defaultValue": "false"
            },
            "buttonsList": {
                "type": "unknown",
                "attribute": "buttons-list",
                "mutable": false,
                "complexType": {
                    "original": "{\n    id?: string;\n    label: string;\n    onClick: (event?: any) => void;\n    exraClasses?: string;\n    title?: string;\n  }[]",
                    "resolved": "{ id?: string; label: string; onClick: (event?: any) => void; exraClasses?: string; title?: string; }[]",
                    "references": {}
                },
                "required": false,
                "optional": false,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "defaultValue": "[]"
            }
        };
    }
}
