// @platformscode/core/dist/collection/components/dga-radial-stepper/dga-radial-stepper.js
import { Host, h } from "@stencil/core";
export class DgaRadialStepper {
    constructor() {
        this.RTL = false;
        this.size = '64px';
        this.showText = true;
        this.steps = [];
        this.activeStep = 1;
        this.variant = 'primary';
        this.showStepName = true;
        this.showStepDescription = true;
        this.sizes = {
            '120px': 120,
            '80px': 80,
            '64px': 64,
            '48px': 48,
            '40px': 40,
        };
    }
    updateInternalPercentage() {
        if ((this.activeStep >= this.steps.length + 1) || (this.activeStep < 1)) {
            return;
        }
        this.internalPercentage = (this.activeStep / this.steps.length) * 100;
        this.internalActiveStep = this.activeStep;
    }
    componentDidLoad() {
        this.updateInternalPercentage();
    }
    render() {
        var _a, _b, _c, _d, _e, _f, _g, _h;
        const strokeWidth = this.sizes[this.size] / 10;
        const radius = (this.sizes[this.size] - strokeWidth) / 2;
        const dashArray = 2 * Math.PI * radius;
        const dashOffset = (this.RTL ? 1 : -1) * dashArray * (1 - this.internalPercentage / 100);
        return (h(Host, { key: 'a1e701310694201dfc2e8209165dcfc0b510a1c8' }, h("dga-flex", { key: '3f4f28374a807d2b55d52bf57458ae9e779b60bb', spacing: 8, align: 'center', dir: this.RTL ? "rtl" : "ltr", class: `dga-radial-stepper--${this.size} dga-radial-stepper--${this.variant}`, "aria-size": this.size }, h("div", { key: 'f8cae0ec45e4a96a21c074c988fdf6f1339cc3b0', class: "dga-circular-stepper", "aria-size": this.size }, h("div", { key: 'f6ed3e6723fbbd6ae3cb89b4b8a9079bce680ba9', class: "dga-circular-stepper__label" }, this.showText && this.steps.length > 0 && h("h4", { key: '1f02f978eaff9f7f5c05900439bf4a3e379e9910', class: `dga-circular-stepper__label--text ${this.variant}` }, this.RTL ? `${this.internalActiveStep} من ${this.steps.length}` : `${this.internalActiveStep} of ${this.steps.length}`)), h("svg", { key: 'eac4bf8b1d6ba62eccd2d2da95112316619ad4cc', width: this.sizes[this.size], height: this.sizes[this.size], viewBox: `0 0 ${this.sizes[this.size]} ${this.sizes[this.size]}` }, h("circle", { key: 'e1b57a2b23b1d959328f8a7dcc62abb5ef697c84', class: `track track--${this.variant}`, fill: "none", stroke: "#e6e6e6",
            // stroke="currentColor"
            r: radius, cx: this.sizes[this.size] / 2, cy: this.sizes[this.size] / 2, "stroke-width": strokeWidth }), h("circle", { key: '6c5e32d7a23a618daf226df2c399f622a774448c', class: `tail tail--${this.variant}`, fill: "none", stroke: "currentColor", r: radius, cx: this.sizes[this.size] / 2, cy: this.sizes[this.size] / 2, "stroke-width": strokeWidth, "stroke-linecap": 'round', "stroke-dasharray": dashArray, "stroke-dashoffset": dashOffset, transform: `rotate(-90 ${this.sizes[this.size] / 2} ${this.sizes[this.size] / 2})`, style: {
                transition: 'stroke-dashoffset 0.5s ease',
            } }))), h("dga-flex", { key: 'cff3d7f857f93027267d02dffe7a7b1507048deb', direction: 'column', spacing: 4 }, ((_a = this.steps[this.internalActiveStep - 1]) === null || _a === void 0 ? void 0 : _a.showPreStep) && h("span", { key: 'ac511010e1a7d6ec63632971265ff7a7e318258c', class: "dga-radial-stepper__pre-name" }, "Previous: ", (_b = this.steps[this.internalActiveStep - 1]) === null || _b === void 0 ? void 0 :
            _b.stepName), this.showStepName && ((_c = this.steps[this.internalActiveStep - 1]) === null || _c === void 0 ? void 0 : _c.stepName) && h("h3", { key: '6dcdb3e616b3dc914317ebe7fa6c880422daeba3', class: "dga-radial-stepper__name" }, (_d = this.steps[this.internalActiveStep - 1]) === null || _d === void 0 ? void 0 : _d.stepName), this.showStepDescription && ((_e = this.steps[this.internalActiveStep - 1]) === null || _e === void 0 ? void 0 : _e.stepDescription) && h("h5", { key: 'bd2ed78883628401f31a4437dd214192fb57924a', class: "dga-radial-stepper__desc" }, (_f = this.steps[this.internalActiveStep - 1]) === null || _f === void 0 ? void 0 : _f.stepDescription), ((_g = this.steps[this.internalActiveStep - 1]) === null || _g === void 0 ? void 0 : _g.showNextStep) && h("span", { key: '2c1bff47812768c1c8a662079d40c19100d97435', class: "dga-radial-stepper__next-name" }, "Next: ", (_h = this.steps[this.internalActiveStep]) === null || _h === void 0 ? void 0 :
            _h.stepName)))));
    }
    static get is() { return "dga-radial-stepper"; }
    static get encapsulation() { return "shadow"; }
    static get originalStyleUrls() {
        return {
            "$": ["dga-radial-stepper.scss"]
        };
    }
    static get styleUrls() {
        return {
            "$": ["dga-radial-stepper.css"]
        };
    }
    static get properties() {
        return {
            "RTL": {
                "type": "boolean",
                "attribute": "r-t-l",
                "mutable": false,
                "complexType": {
                    "original": "boolean",
                    "resolved": "boolean",
                    "references": {}
                },
                "required": false,
                "optional": false,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false,
                "defaultValue": "false"
            },
            "size": {
                "type": "string",
                "attribute": "size",
                "mutable": false,
                "complexType": {
                    "original": "'120px' | '80px' | '64px' | '48px' | '40px'",
                    "resolved": "\"120px\" | \"40px\" | \"48px\" | \"64px\" | \"80px\"",
                    "references": {}
                },
                "required": false,
                "optional": false,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false,
                "defaultValue": "'64px'"
            },
            "showText": {
                "type": "boolean",
                "attribute": "show-text",
                "mutable": false,
                "complexType": {
                    "original": "boolean",
                    "resolved": "boolean",
                    "references": {}
                },
                "required": false,
                "optional": false,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false,
                "defaultValue": "true"
            },
            "steps": {
                "type": "unknown",
                "attribute": "steps",
                "mutable": false,
                "complexType": {
                    "original": "Step[]",
                    "resolved": "Step[]",
                    "references": {
                        "Step": {
                            "location": "local",
                            "path": "/Users/mohamedaboelezz/projects/Azm-X/platform-code-cross-platform-storybook/packages/core/src/components/dga-radial-stepper/dga-radial-stepper.tsx",
                            "id": "src/components/dga-radial-stepper/dga-radial-stepper.tsx::Step"
                        }
                    }
                },
                "required": false,
                "optional": false,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "defaultValue": "[]"
            },
            "activeStep": {
                "type": "number",
                "attribute": "active-step",
                "mutable": false,
                "complexType": {
                    "original": "number",
                    "resolved": "number",
                    "references": {}
                },
                "required": false,
                "optional": false,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false,
                "defaultValue": "1"
            },
            "variant": {
                "type": "string",
                "attribute": "variant",
                "mutable": false,
                "complexType": {
                    "original": "'primary' | 'neutral' | 'on-color'",
                    "resolved": "\"neutral\" | \"on-color\" | \"primary\"",
                    "references": {}
                },
                "required": false,
                "optional": true,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false,
                "defaultValue": "'primary'"
            },
            "showStepName": {
                "type": "boolean",
                "attribute": "show-step-name",
                "mutable": false,
                "complexType": {
                    "original": "boolean",
                    "resolved": "boolean",
                    "references": {}
                },
                "required": false,
                "optional": false,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false,
                "defaultValue": "true"
            },
            "showStepDescription": {
                "type": "boolean",
                "attribute": "show-step-description",
                "mutable": false,
                "complexType": {
                    "original": "boolean",
                    "resolved": "boolean",
                    "references": {}
                },
                "required": false,
                "optional": false,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false,
                "defaultValue": "true"
            }
        };
    }
    static get states() {
        return {
            "internalPercentage": {},
            "internalActiveStep": {}
        };
    }
    static get watchers() {
        return [{
                "propName": "steps",
                "methodName": "updateInternalPercentage"
            }, {
                "propName": "activeStep",
                "methodName": "updateInternalPercentage"
            }];
    }
}
