// @platformscode/core/dist/components/dga-header.js
import { p as proxyCustomElement, H, h } from './p-JCGAt0My.js';
import { D as DgaNavHeader } from './p-CsKEXIdY.js';
import { D as DgaNavHeaderMain } from './p-CJ1Dn2qN.js';
import { D as DgaNavHeaderLogos } from './p-D9-dGUrV.js';
import { D as DgaNavHeaderMenu } from './p-DbBEr5PW.js';
import { D as DgaNavHeaderLink } from './p-KLIoWA81.js';
import { D as DgaNavHeaderSubMenuColumn } from './p-B4kOHBcQ.js';
import { D as DgaNavHeaderSubMenuLink } from './p-DBxuY-4b.js';
import { D as DgaNavHeaderActions } from './p-DS3wOmQO.js';
import { D as DgaHeaderActionBtn } from './p-Dxf245we.js';

const DgaHeader$1 = /*@__PURE__*/ proxyCustomElement(class DgaHeader extends H {
    constructor() {
        super();
        this.__registerHost();
    }
    render() {
        return (h(DgaNavHeader, { key: 'de59f73fc8b94c75267b8b24f0f7d93f06656a5c', fullWidth: this.config.fullWidth, divider: this.config.divider }, h(DgaNavHeaderMain, { key: 'd35f058aee7ff64e136f96dad016b7859e9c1d98', collapsed: true }, h(DgaNavHeaderLogos, { key: '2473f086ff8155042e83d5443c64c0b1fdb3af5b', logoSrc: this.config.logoSrc, govSrc: this.config.govSrc, logoLink: this.config.logoLink, govLink: this.config.govLink }), h(DgaNavHeaderMenu, { key: '6aa461900c3bc1cbba91c6b0aa00b9b803759282' }, this.config.headerLinks.map((link, index) => (h(DgaNavHeaderLink, { key: index, active: false, onClick: () => console.log('clicked'), label: link.label, icon: link.icon, subMenuBackground: this.config.subMenuBackground, subMenuFullWidth: this.config.subMenuFullWidth }, link.subMenu.map((subMenu, subMenuIndex) => (h(DgaNavHeaderSubMenuColumn, { key: subMenuIndex }, subMenu.columns.map((column, columnIndex) => (h(DgaNavHeaderSubMenuColumn, { key: columnIndex }, column.links.map((subMenuLink, linkIndex) => (h(DgaNavHeaderSubMenuLink, { key: linkIndex, background: this.config.subMenuBackground, label: subMenuLink.label, helperText: subMenuLink.helperText, icon: subMenuLink.icon, linkStyle: subMenuLink.linkStyle })))))))))))))), h(DgaNavHeaderActions, { key: '6fecf1aa7b080c6b17106dafb0c99008c42a2a3b' }, this.config.actions.map((action, index) => (h(DgaHeaderActionBtn, { key: index, onClick: () => console.log('clicked'), label: action.label, icon: action.icon }))))));
    }
}, [0, "dga-header", {
        "config": [16]
    }]);
function defineCustomElement$1() {
    if (typeof customElements === "undefined") {
        return;
    }
    const components = ["dga-header"];
    components.forEach(tagName => { switch (tagName) {
        case "dga-header":
            if (!customElements.get(tagName)) {
                customElements.define(tagName, DgaHeader$1);
            }
            break;
    } });
}

const DgaHeader = DgaHeader$1;
const defineCustomElement = defineCustomElement$1;

export { DgaHeader, defineCustomElement };
