// @platformscode/core/dist/components/dga-file-upload-usage.js
import { p as proxyCustomElement, H, h } from './p-JCGAt0My.js';
import { d as defineCustomElement$4 } from './p-CSDgLEi4.js';
import { d as defineCustomElement$3 } from './p-CcNS0CYN.js';
import { d as defineCustomElement$2 } from './p-FjQbvIG7.js';

const DgaFileUploadUsage$1 = /*@__PURE__*/ proxyCustomElement(class DgaFileUploadUsage extends H {
    constructor() {
        super();
        this.__registerHost();
    }
    render() {
        return (h("div", { key: '3ce335f8b1c04a28cec30f101b594e523492fb8d' }, h("h1", { key: '0276d54d601cdd6bb421d7fef51d80f0131ed313' }, "File Upload Example"), h("div", { key: '2cd28e401fd58631d2a3bd4beb86dc5a3b5b7a65', style: { padding: "50px", display: "flex", gap: "20px" } }, h("div", { key: '647e3106fc501238cc3b99496ab62f83cc11fcad', style: { display: "flex", flexDirection: "column", gap: "20px" } }, h("dga-file-upload", { key: 'ab2445d5f43bcbecc56543f3b98541c7c5d56a80' })), h("div", { key: 'f51b1d46e5c0a7b03a0b5cba068e1310cda45428', style: { display: "flex", flexDirection: "column", gap: "20px" } }))));
    }
}, [0, "dga-file-upload-usage"]);
function defineCustomElement$1() {
    if (typeof customElements === "undefined") {
        return;
    }
    const components = ["dga-file-upload-usage", "dga-file-upload", "dga-icon", "dga-loading"];
    components.forEach(tagName => { switch (tagName) {
        case "dga-file-upload-usage":
            if (!customElements.get(tagName)) {
                customElements.define(tagName, DgaFileUploadUsage$1);
            }
            break;
        case "dga-file-upload":
            if (!customElements.get(tagName)) {
                defineCustomElement$4();
            }
            break;
        case "dga-icon":
            if (!customElements.get(tagName)) {
                defineCustomElement$3();
            }
            break;
        case "dga-loading":
            if (!customElements.get(tagName)) {
                defineCustomElement$2();
            }
            break;
    } });
}

const DgaFileUploadUsage = DgaFileUploadUsage$1;
const defineCustomElement = defineCustomElement$1;

export { DgaFileUploadUsage, defineCustomElement };
