// @platformscode/core/dist/components/dga-header-usage.js
import { p as proxyCustomElement, H, h, F as Fragment } from './p-JCGAt0My.js';
import { d as defineCustomElement$t } from './p-BLDzt_Uq.js';
import { d as defineCustomElement$s } from './p-Bs_sPiZE.js';
import { d as defineCustomElement$r } from './p-Don4li2O.js';
import { d as defineCustomElement$q } from './p-Cg2ruBZP.js';
import { d as defineCustomElement$p } from './p-Dxf245we.js';
import { d as defineCustomElement$o } from './p-DLER3hMX.js';
import { d as defineCustomElement$n } from './p-ByOC3T5X.js';
import { d as defineCustomElement$m } from './p-CcNS0CYN.js';
import { d as defineCustomElement$l } from './p-CsKEXIdY.js';
import { d as defineCustomElement$k } from './p-DS3wOmQO.js';
import { d as defineCustomElement$j } from './p-KLIoWA81.js';
import { d as defineCustomElement$i } from './p-D9-dGUrV.js';
import { d as defineCustomElement$h } from './p-CJ1Dn2qN.js';
import { d as defineCustomElement$g } from './p-DbBEr5PW.js';
import { d as defineCustomElement$f } from './p-TR-8p4D-.js';
import { d as defineCustomElement$e } from './p-B4kOHBcQ.js';
import { d as defineCustomElement$d } from './p-DBxuY-4b.js';
import { d as defineCustomElement$c } from './p-Cmg--ZO6.js';
import { d as defineCustomElement$b } from './p-ChOKtheJ.js';
import { d as defineCustomElement$a } from './p-gOotOaIt.js';
import { d as defineCustomElement$9 } from './p-DJjQrzWf.js';
import { d as defineCustomElement$8 } from './p-B0P1tNYn.js';
import { d as defineCustomElement$7 } from './p-DMuRo8EU.js';
import { d as defineCustomElement$6 } from './p-D9uV831l.js';
import { d as defineCustomElement$5 } from './p-SDJFYO-x.js';
import { d as defineCustomElement$4 } from './p-Pml5NKmL.js';
import { d as defineCustomElement$3 } from './p-CgCfk7bZ.js';
import { d as defineCustomElement$2 } from './p-ZhmbbVrl.js';

// import { IRoute } from '../dga-drawer/dga-drawer';
const SidePanelRoutes = [
    {
        name: 'Link',
        path: '/',
        icon: {
            name: 'home-01',
            variant: 'stroke',
            size: 16,
        },
        disabled: false,
        children: [
            {
                name: 'Link',
                path: '1',
                // icon: <i class='hgi-stroke hgi-arrow-down-01' style={{fontSize:"16px"}} />,
                // badge: 39,
            },
            {
                name: 'Link',
                path: '2',
                // icon: <i class='hgi-stroke hgi-arrow-down-01' style={{fontSize:"16px"}} />,
                // badge: 21,
            },
            {
                name: 'Link',
                path: '3',
                // icon: <i class='hgi-stroke hgi-arrow-down-01' style={{fontSize:"16px"}} />,
                // badge: 2,
            },
            // {
            //   name: "Home 4",
            //   path: "4",
            //   icon: <i class='hgi-stroke hgi-arrow-down-01' style={{fontSize:"16px"}} />,
            //   badge: 20,
            // },
            // {
            //   name: "Home 5",
            //   path: "5",
            //   icon: <i class='hgi-stroke hgi-arrow-down-01' style={{fontSize:"16px"}} />,
            //   badge: 150,
            //   children: [
            //     {
            //       name: "Home 1",
            //       path: "1",
            //       icon: <i class='hgi-stroke hgi-arrow-down-01' style={{fontSize:"16px"}} />,
            //       badge: 39,
            //       level: 2,
            //     },
            //     {
            //       name: "Home 4",
            //       path: "4",
            //       icon: <i class='hgi-stroke hgi-arrow-down-01' style={{fontSize:"16px"}} />,
            //       badge: 20,
            //       level: 2,
            //     },
            //     {
            //       name: "Home 5",
            //       path: "5",
            //       icon: <i class='hgi-stroke hgi-arrow-down-01' style={{fontSize:"16px"}} />,
            //       badge: 150,
            //       level: 2,
            //     },
            //     {
            //       name: "Home 6",
            //       path: "6",
            //       icon: <i class='hgi-stroke hgi-arrow-down-01' style={{fontSize:"16px"}} />,
            //       badge: 200,
            //       level: 2,
            //     },
            //     {
            //       name: "Home 7",
            //       path: "7",
            //       icon: <i class='hgi-stroke hgi-arrow-down-01' style={{fontSize:"16px"}} />,
            //       badge: 24,
            //       level: 2,
            //     },
            //   ],
            // },
            // {
            //   name: "Home 6",
            //   path: "6",
            //   icon: <i class='hgi-stroke hgi-arrow-down-01' style={{fontSize:"16px"}} />,
            //   badge: 200,
            // },
            // {
            //   name: "Home 7",
            //   path: "7",
            //   icon: <i class='hgi-stroke hgi-arrow-down-01' style={{fontSize:"16px"}} />,
            //   badge: 24,
            // },
        ],
    },
    {
        name: 'Components',
        path: '/components',
        icon: {
            name: 'home-01',
            variant: 'stroke',
            size: 16,
        },
        disabled: false,
        children: [
            {
                name: 'Buttons',
                path: '/buttons',
                icon: {
                    name: 'home-01',
                    variant: 'stroke',
                    size: 16,
                },
                badge: 100,
            },
            {
                name: 'Badges',
                path: '/badges',
                icon: {
                    name: 'home-01',
                    variant: 'stroke',
                    size: 16,
                },
                badge: 100,
            },
            {
                name: 'Paths',
                path: '/Paths',
                children: [
                    {
                        name: 'Buttons',
                        path: '/buttons',
                        icon: {
                            name: 'home-01',
                            variant: 'stroke',
                            size: 16,
                        },
                        badge: 100,
                    },
                    {
                        name: 'Badges',
                        path: '/badges',
                        icon: {
                            name: 'home-01',
                            variant: 'stroke',
                            size: 16,
                        },
                        badge: 100,
                    },
                    {
                        name: 'Paths',
                        path: '/Paths',
                        icon: {
                            name: 'home-01',
                            variant: 'stroke',
                            size: 16,
                        },
                        badge: 100,
                    },
                ],
            },
            {
                name: 'Inputs',
                path: '/inputs',
                icon: {
                    name: 'home-01',
                    variant: 'stroke',
                    size: 16,
                },
                badge: 100,
            },
            {
                name: 'Accordions',
                path: '/accordions',
                icon: {
                    name: 'home-01',
                    variant: 'stroke',
                    size: 16,
                },
                badge: 100,
            },
            {
                name: 'Tabs',
                path: '/tabs',
                icon: {
                    name: 'home-01',
                    variant: 'stroke',
                    size: 16,
                },
                badge: 100,
            },
        ],
    },
    {
        name: 'Components 2',
        path: '/components',
        icon: {
            name: 'home-01',
            variant: 'stroke',
            size: 16,
        },
        disabled: false,
        children: [
            {
                name: 'Buttons',
                path: '/buttons',
                icon: {
                    name: 'home-01',
                    variant: 'stroke',
                    size: 16,
                },
                badge: 100,
            },
            {
                name: 'Badges',
                path: '/badges',
                icon: {
                    name: 'home-01',
                    variant: 'stroke',
                    size: 16,
                },
                badge: 100,
            },
            {
                name: 'Paths',
                path: '/Paths',
                children: [
                    {
                        name: 'Buttons',
                        path: '/buttons',
                        badge: 100,
                    },
                    {
                        name: 'Badges',
                        path: '/badges',
                        badge: 100,
                    },
                    {
                        name: 'Paths',
                        path: '/Paths',
                        badge: 100,
                    },
                ],
            },
            {
                name: 'Inputs',
                path: '/inputs',
                badge: 100,
            },
            {
                name: 'Accordions',
                path: '/accordions',
                badge: 100,
            },
            {
                name: 'Tabs',
                path: '/tabs',
                badge: 100,
            },
        ],
    },
];
const DgaHeaderUsage$1 = /*@__PURE__*/ proxyCustomElement(class DgaHeaderUsage extends H {
    constructor() {
        super();
        this.__registerHost();
        this.activeLink = 'link-1';
        this.collapsed = false;
        this.open = false;
    }
    render() {
        return (h(Fragment, { key: '546171693b856a444cf4d46ede707efdc503ab34' }, h("dga-second-nav-header", { key: '2e16de393ea7961c086741f81080361fa7d533aa', hideDivider: true }, h("dga-second-nav-header-actions", { key: '45fde60210f5aa5830937b047feffce7f04e8714' }, h("dga-button-v2", { key: '4e0188bdc3df129d08276174b7b71dcf8e8ce3a5', label: "Button", variant: "transparent", size: "sm", iconOnly: true, leadIcon: true, leadIconProps: {
                name: 'mic-01',
                type: 'rounded',
                variant: 'stroke',
                color: '#FFFFFF',
            } }), h("dga-button-v2", { key: '1419e5f0ec92d72cc9c3ba43b43775f4d8e5a59c', label: "Button", variant: "transparent", size: "sm", iconOnly: true, leadIcon: true, leadIconProps: {
                name: 'zoom-in-area',
                type: 'rounded',
                variant: 'stroke',
                color: '#FFFFFF',
            } }), h("dga-button-v2", { key: '6450f3a8089accc81e6a02ad7538781844e2342d', label: "Button", variant: "transparent", size: "sm", iconOnly: true, leadIcon: true, leadIconProps: {
                name: 'zoom-out-area',
                type: 'rounded',
                variant: 'stroke',
                color: '#FFFFFF',
            } }), h("dga-button-v2", { key: '63716e75bdb27162e937b6b96a5bc89fc24e991a', label: "Button", variant: "transparent", size: "sm", iconOnly: true, leadIcon: true, leadIconProps: {
                name: 'view',
                type: 'rounded',
                variant: 'stroke',
                color: '#FFFFFF',
            } })), h("dga-second-nav-header-content", { key: '484272baf65bda7c1fc9188b3de289f82fce9cbf' }, h("dga-second-nav-header-item", { key: '5406f7a1d51df3f60cfa65571ae259c7960696e9', label: "Cloudy" }, h("dga-icon", { key: 'a5508dae7cd6522559a9b441aa374cb2719cb16f', icon: "cloud", variant: "stroke", type: "rounded" })), h("dga-second-nav-header-item", { key: 'b073617030d0d6aff6093d9017f25d080114df60', label: "21 Jan 2024" }, h("dga-icon", { key: '68bbd7d1c3a24d652423d63b1d62f22f678d4a88', icon: "calendar-04", variant: "stroke", type: "rounded" })))), h("dga-nav-header", { key: '7c8fd9019a9bdd8be65198c0968129d43f62d895', fullWidth: true, divider: true, sticky: true }, h("dga-nav-header-main", { key: 'c4abbf319405576dbd564195b2c5752c06bb9ae0', collapsed: this.collapsed, onToggleCollapsed: e => {
                this.collapsed = e.detail;
                console.log('collapsed', e.detail);
            } }, h("dga-button", { key: '570f2596c47aa847fe7afddb57da71cf9252a4fa', iconOnly: true, leadIcon: true, leadIconType: this.collapsed ? 'cancel01' : 'menu01', leadIconProps: {
                name: this.collapsed ? 'menu01' : 'cancel01',
                size: 24,
                variant: 'stroke',
                type: 'rounded',
            }, variant: "transparent", onClick: () => (this.collapsed = !this.collapsed), class: "header-menu__btn", sx: {
                display: { xs: 'block', md: 'none' },
            } }), h("dga-nav-header-logos", { key: '2762f7e23f2b672de0e74adc96592be3c8b5c80f', logoSrc: "https://dga-nds-fbhtx.ondigitalocean.app/mobile-logo.svg", govSrc: "https://dga-nds-fbhtx.ondigitalocean.app/mobile-logo.svg", logoLink: "#", govLink: "#" }), h("dga-nav-header-menu", { key: 'fa6c8733d72f29f8f040bd66e72ee93bd340a632' }, h("dga-nav-header-link", { key: 'b15379f2f60857ae0566d0dbe73878f519fd9e2b', label: "Item 1", icon: "arrow-down-01", active: this.activeLink === 'link-1', subMenuBackground: "white", onClick: () => {
                this.activeLink = 'link-1';
            } }, h("dga-nav-header-sub-menu", { key: '42247a9abe3260fe7221af522175f40b455ebcf9', "sub-menu-full-width": true, background: 'white' }, h("dga-nav-header-sub-menu-column", { key: '655a69458b12df2b51d2f59498e86c6c227aa898', label: "Group Label" }, h("dga-nav-header-sub-menu-link", { key: '4abab5ac0f331bf9b8809291e132dd21a6107189', label: "Submenu Link", link: "#", icon: "document", helperText: "This is a description" }), h("dga-nav-header-sub-menu-link", { key: '067c3340b3830ce8cc1f15c5b98cb268492cea01', label: "Submenu Link", link: "#", icon: "document", helperText: "This is a description" })), h("dga-nav-header-sub-menu-column", { key: '240642d35355d75631055689e49ec2f47fef7a7a', label: "Group Label" }, h("dga-nav-header-sub-menu-link", { key: '40ca18e7ab4f9e66c63fc0cef5033ff8e162acec', label: "Submenu Link", link: "#", icon: "document", helperText: "This is a description" }), h("dga-nav-header-sub-menu-link", { key: 'e8094fe1baec0826e52a5a3f01f13cd701bc52a1', label: "Submenu Link", link: "#", icon: "document", helperText: "This is a description" })), h("dga-nav-header-sub-menu-column", { key: 'bbd6ca6046a8a1149117116afeacac6d3d824a93', label: "Group Label" }, h("dga-nav-header-sub-menu-link", { key: '2d4e69e229895adb279e72306438fc2536282930', label: "Submenu Link", link: "#", icon: "document", helperText: "This is a description" }), h("dga-nav-header-sub-menu-link", { key: '9479ad5451a3c2324f89906750adb5db4fcc0e5b', label: "Submenu Link", link: "#", icon: "document", helperText: "This is a description" })), h("dga-nav-header-sub-menu-column", { key: 'a38ad1b8f5daa58843cc0961c74307ac4816532e', label: "Group Label" }, h("dga-nav-header-sub-menu-link", { key: '3651204b1351e53d630e168e43ab81018a7a5dbe', label: "Submenu Link", link: "#", icon: "document", helperText: "This is a description" }), h("dga-nav-header-sub-menu-link", { key: 'bc6bffb75bb2494938b1f715c0235517f3e7a19a', label: "Submenu Link", link: "#", icon: "document", helperText: "This is a description" })))), h("dga-nav-header-link", { key: '7ea6a387144bce5148020ae076e60dda3690d873', label: "Item  2", icon: "arrow-down-01", active: this.activeLink === 'link-2', subMenuBackground: "white", subMenuFullWidth: true, onClick: () => {
                this.activeLink = 'link-2';
            } }, h("dga-nav-header-sub-menu", { key: '0f40baad9372b612cc990cc9408717e8f64da45b', "sub-menu-full-width": true, background: 'white' }, h("dga-nav-header-sub-menu-column", { key: '054394b4af742d0a45e15e1914530fc19dc88119', label: "Group Label" }, h("dga-nav-header-sub-menu-link", { key: 'a8bc8ffbda2a0b13096b0d5c1ff1896461fb8d7b', label: "Submenu Link", link: "#", icon: "document", helperText: "This is a description" }), h("dga-nav-header-sub-menu-link", { key: 'e69fcefea55e325135d45e8daf11eebf4d013c77', label: "Submenu Link", link: "#", icon: "document", helperText: "This is a description" })), h("dga-nav-header-sub-menu-column", { key: 'eccfe352b1fb394ca423de144ed88833f75e1863', label: "Group Label" }, h("dga-nav-header-sub-menu-link", { key: '779c56250336b920048235ed3a1245f492f7b234', label: "Submenu Link", link: "#", icon: "document", helperText: "This is a description" }), h("dga-nav-header-sub-menu-link", { key: '2ca59d886b7a5d9ed0bb33275573f52144082b0d', label: "Submenu Link", link: "#", icon: "document", helperText: "This is a description" })))), h("dga-nav-header-link", { key: '2a84aba2b304c44769f745d64a271ba5905af167', label: "Item  3", icon: "arrow-down-01", active: this.activeLink === 'link-3', subMenuBackground: "white", subMenuFullWidth: true, onClick: () => {
                this.activeLink = 'link-3';
            } }), h("dga-nav-header-link", { key: '7593a053b73badbdca8ed3b9fffd3e0d95e6cec8', label: "Item  4", icon: "arrow-down-01", active: this.activeLink === 'link-4', subMenuBackground: "white", subMenuFullWidth: true, onClick: () => {
                this.activeLink = 'link-4';
            } }, h("dga-nav-header-sub-menu", { key: '9cc849a54ac5499deda1453b62a8fd997e6519a2', "sub-menu-full-width": true, background: 'white' }, h("dga-nav-header-sub-menu-column", { key: '265b1075aefc65192a19b480667eb040781dc508', label: "Group Label" }, h("dga-nav-header-sub-menu-link", { key: '8c34f9dc94812f44aae8c4b283291be22c1513c7', label: "Submenu Link", link: "#", icon: "document", helperText: "This is a description" }), h("dga-nav-header-sub-menu-link", { key: '3847ccb25c5de230c6c696fecaf2d008d2cc0157', label: "Submenu Link", link: "#", icon: "document", helperText: "This is a description" })), h("dga-nav-header-sub-menu-column", { key: 'f63ecdd4b50396826015b2611ed2d2edcc24536f', label: "Group Label" }, h("dga-nav-header-sub-menu-link", { key: '9626364dc07de26a6a289ac1452ddb8652945a9e', label: "Submenu Link", link: "#", icon: "document", helperText: "This is a description" }), h("dga-nav-header-sub-menu-link", { key: 'ea64cd9b4ccc9f2cae598ef9869560c57fb6a584', label: "Submenu Link", link: "#", icon: "document", helperText: "This is a description" })), h("dga-nav-header-sub-menu-column", { key: 'd004aac96543a4f90b7a3ad0f7b3f84a5aa1dc3d', label: "Group Label" }, h("dga-nav-header-sub-menu-link", { key: '14e5fb19a4d0af2a5ed3a94a71fc521ada03bc28', label: "Submenu Link", link: "#", icon: "document", helperText: "This is a description" }), h("dga-nav-header-sub-menu-link", { key: '22d13a7272a8a053a87ba5950fade24259b5d6e5', label: "Submenu Link", link: "#", icon: "document", helperText: "This is a description" })), h("dga-nav-header-sub-menu-column", { key: '93e423927e520c16896e809b33c57c5b2f179c96', label: "Group Label" }, h("dga-nav-header-sub-menu-link", { key: 'd4bacbb705cd5aa9d70b2f3365dc812621c49e82', label: "Submenu Link", link: "#", icon: "document", helperText: "This is a description" }), h("dga-nav-header-sub-menu-link", { key: '61f45f0bcd18cbaf2e239fb278bb4dfd8c4700b1', label: "Submenu Link", link: "#", icon: "document", helperText: "This is a description" })), h("dga-nav-header-sub-menu-column", { key: '40cc3bfb6c9e5c4dc7809b5d21dd2f5aa97db5b5', label: "Group Label" }, h("dga-nav-header-sub-menu-link", { key: 'c16d08f5908e6ba258920701cbe6f56a72a75f22', label: "Submenu Link", link: "#", icon: "document", helperText: "This is a description" }), h("dga-nav-header-sub-menu-link", { key: '54fe22f4a1a8f5ce8685ad73ca62be75513fdbb6', label: "Submenu Link", link: "#", icon: "document", helperText: "This is a description" })), h("dga-nav-header-sub-menu-column", { key: '8efc07301a1633f79f0a50d9125416b3b2cb690d', label: "Group Label" }, h("dga-nav-header-sub-menu-link", { key: 'e518e24b52fcf94da2b924cebdd0b68dae9b3818', label: "Submenu Link", link: "#", icon: "document", helperText: "This is a description" }), h("dga-nav-header-sub-menu-link", { key: '3a81d707608e8855d61a51178d152c4cbcc81187', label: "Submenu Link", link: "#", icon: "document", helperText: "This is a description" })))))), h("dga-nav-header-actions", { key: '3ba32063cca29165c46d06f1eac57b7d1b99d13b' }, h("dga-header-action-btn", { key: 'dee3461677518da01311badd3cd3c12e1d941923', label: "Search", icon: "search-01", subMenuBackground: "white" }, h("dga-nav-header-sub-menu", { key: 'c7af33e115a526374a0cb7084f52ba801d5e9855', "sub-menu-full-width": true, background: 'white' }, h("dga-flex", { key: '722cee268d0e81c1ec32f62883f229ae72c6f2a3', spacing: 16 }, h("dga-search-box", { key: '96c9760591883ee6d454f9537b2bc7b9b46d43b4', placeholder: "Search" }), h("dga-button", { key: 'b63e26a0187fcf0f2d859e5f40a697fee6092e96', label: "Search" })))), h("dga-header-action-btn", { key: 'c6875a6e5f8ceaee1ce2166f0aabf31c4a77a9ce', label: "\u0639\u0631\u0628\u064A", icon: "translation" }), h("dga-header-action-btn", { key: 'e15f0c5a74e7d77767d1b90dfcecfa7b254c11ee', label: "Login", icon: "user" })), h("dga-header-mobile-actions", { key: '80f797b857439363ecdbc47500cc2a39ced44e35' }, h("dga-header-menu", { key: '6791fc019b957540f82c488826b1b92dba104af5', actions: [
                {
                    label: 'Search',
                    icon: 'search-01',
                    onClick: () => {
                        console.log('clicked');
                    },
                },
                {
                    label: 'عربي',
                    icon: 'translation',
                    onClick: () => {
                        console.log('clicked');
                    },
                },
                {
                    label: 'Login',
                    icon: 'user',
                    onClick: () => {
                        console.log('clicked');
                    },
                }
            ] }))), h("dga-drawer", { key: 'b933b8df9569309c05a273693bd6918ea35d5fff', routes: SidePanelRoutes, open: this.collapsed, background: 'gray', headerLogo: "https://casamedia.com/wp-content/uploads/2023/04/adidas-1024x683.png", sx: {
                '& > .sidepanel': {
                    top: '112px',
                },
            } }), h("div", { key: '0d84117b208149f39a8c50784b7d22d5464fbc62' }, "Test")));
    }
}, [0, "dga-header-usage", {
        "activeLink": [32],
        "collapsed": [32],
        "open": [32]
    }]);
function defineCustomElement$1() {
    if (typeof customElements === "undefined") {
        return;
    }
    const components = ["dga-header-usage", "dga-button", "dga-button-v2", "dga-drawer", "dga-flex", "dga-header-action-btn", "dga-header-menu", "dga-header-mobile-actions", "dga-icon", "dga-nav-header", "dga-nav-header-actions", "dga-nav-header-link", "dga-nav-header-logos", "dga-nav-header-main", "dga-nav-header-menu", "dga-nav-header-sub-menu", "dga-nav-header-sub-menu-column", "dga-nav-header-sub-menu-link", "dga-search-box", "dga-second-nav-header", "dga-second-nav-header-actions", "dga-second-nav-header-content", "dga-second-nav-header-item", "dga-tag", "dga-tooltip", "side-panel-menu", "submenu-component", "tab-component", "tab-link"];
    components.forEach(tagName => { switch (tagName) {
        case "dga-header-usage":
            if (!customElements.get(tagName)) {
                customElements.define(tagName, DgaHeaderUsage$1);
            }
            break;
        case "dga-button":
            if (!customElements.get(tagName)) {
                defineCustomElement$t();
            }
            break;
        case "dga-button-v2":
            if (!customElements.get(tagName)) {
                defineCustomElement$s();
            }
            break;
        case "dga-drawer":
            if (!customElements.get(tagName)) {
                defineCustomElement$r();
            }
            break;
        case "dga-flex":
            if (!customElements.get(tagName)) {
                defineCustomElement$q();
            }
            break;
        case "dga-header-action-btn":
            if (!customElements.get(tagName)) {
                defineCustomElement$p();
            }
            break;
        case "dga-header-menu":
            if (!customElements.get(tagName)) {
                defineCustomElement$o();
            }
            break;
        case "dga-header-mobile-actions":
            if (!customElements.get(tagName)) {
                defineCustomElement$n();
            }
            break;
        case "dga-icon":
            if (!customElements.get(tagName)) {
                defineCustomElement$m();
            }
            break;
        case "dga-nav-header":
            if (!customElements.get(tagName)) {
                defineCustomElement$l();
            }
            break;
        case "dga-nav-header-actions":
            if (!customElements.get(tagName)) {
                defineCustomElement$k();
            }
            break;
        case "dga-nav-header-link":
            if (!customElements.get(tagName)) {
                defineCustomElement$j();
            }
            break;
        case "dga-nav-header-logos":
            if (!customElements.get(tagName)) {
                defineCustomElement$i();
            }
            break;
        case "dga-nav-header-main":
            if (!customElements.get(tagName)) {
                defineCustomElement$h();
            }
            break;
        case "dga-nav-header-menu":
            if (!customElements.get(tagName)) {
                defineCustomElement$g();
            }
            break;
        case "dga-nav-header-sub-menu":
            if (!customElements.get(tagName)) {
                defineCustomElement$f();
            }
            break;
        case "dga-nav-header-sub-menu-column":
            if (!customElements.get(tagName)) {
                defineCustomElement$e();
            }
            break;
        case "dga-nav-header-sub-menu-link":
            if (!customElements.get(tagName)) {
                defineCustomElement$d();
            }
            break;
        case "dga-search-box":
            if (!customElements.get(tagName)) {
                defineCustomElement$c();
            }
            break;
        case "dga-second-nav-header":
            if (!customElements.get(tagName)) {
                defineCustomElement$b();
            }
            break;
        case "dga-second-nav-header-actions":
            if (!customElements.get(tagName)) {
                defineCustomElement$a();
            }
            break;
        case "dga-second-nav-header-content":
            if (!customElements.get(tagName)) {
                defineCustomElement$9();
            }
            break;
        case "dga-second-nav-header-item":
            if (!customElements.get(tagName)) {
                defineCustomElement$8();
            }
            break;
        case "dga-tag":
            if (!customElements.get(tagName)) {
                defineCustomElement$7();
            }
            break;
        case "dga-tooltip":
            if (!customElements.get(tagName)) {
                defineCustomElement$6();
            }
            break;
        case "side-panel-menu":
            if (!customElements.get(tagName)) {
                defineCustomElement$5();
            }
            break;
        case "submenu-component":
            if (!customElements.get(tagName)) {
                defineCustomElement$4();
            }
            break;
        case "tab-component":
            if (!customElements.get(tagName)) {
                defineCustomElement$3();
            }
            break;
        case "tab-link":
            if (!customElements.get(tagName)) {
                defineCustomElement$2();
            }
            break;
    } });
}

const DgaHeaderUsage = DgaHeaderUsage$1;
const defineCustomElement = defineCustomElement$1;

export { DgaHeaderUsage, defineCustomElement };
