// @platformscode/core/dist/components/dga-table-usage.js
import { p as proxyCustomElement, H, h } from './p-JCGAt0My.js';
import { d as defineCustomElement$4 } from './p-C-62j229.js';
import { d as defineCustomElement$3 } from './p-CKrmM6in.js';
import { d as defineCustomElement$2 } from './p-CcNS0CYN.js';

const DgaTableUsage$1 = /*@__PURE__*/ proxyCustomElement(class DgaTableUsage extends H {
    constructor() {
        super();
        this.__registerHost();
    }
    render() {
        return (h("div", { key: '7e60df79d94e5d55c73b7915bcac778c416aac22' }, h("h1", { key: '718ce9785dc840c85ae414dad04be45a477cbe4f' }, "Table Example"), h("div", { key: 'd03f6e246ef28c1b3d82d9cd05f2de32b1300460', style: { padding: '50px', display: 'flex', gap: '20px' } }, h("div", { key: '0467ef3fdac4aba61ad52480d635c4a83df066ae', style: { display: 'flex', flexDirection: 'column', gap: '20px' } }, h("dga-data-table", { key: 'a496f4ee79e1e89f535fdc165c51e78a5e1a2611', columns: [
                {
                    label: 'Header',
                },
            ], cells: [
                {
                    title: 'city',
                    sort: 'up',
                    isSort: true,
                    isFilter: true,
                    type: 'propertyName',
                    propertyName: 'city',
                    colSpan: 1, // Set colSpan to 1 (default is 1)
                },
                {
                    title: 'price',
                    sort: 'down',
                    isSort: true,
                    isFilter: true,
                    type: 'propertyName',
                    propertyName: 'price',
                    colSpan: 1,
                },
                {
                    title: 'state',
                    isSort: true,
                    isFilter: true,
                    type: 'propertyName',
                    propertyName: 'state',
                    colSpan: 1,
                },
            ], data: [
                {
                    h_id: '3',
                    city: 'Dallas',
                    state: 'TX',
                    zip: '75201',
                    price: '162500',
                    selected: false, // Add selected property for checkbox functionality
                },
                {
                    h_id: '4',
                    city: 'Bevery Hills',
                    state: 'CA',
                    zip: '90210',
                    price: '319250',
                    selected: false,
                },
                {
                    h_id: '5',
                    city: 'New York',
                    state: 'NY',
                    zip: '00010',
                    price: '962500',
                    selected: false,
                },
            ] })), h("div", { key: 'db1536d669fe9add6f1792540259e1bf5dfbc785', style: { display: 'flex', flexDirection: 'column', gap: '20px' } }))));
    }
}, [0, "dga-table-usage"]);
function defineCustomElement$1() {
    if (typeof customElements === "undefined") {
        return;
    }
    const components = ["dga-table-usage", "dga-checkbox", "dga-data-table", "dga-icon"];
    components.forEach(tagName => { switch (tagName) {
        case "dga-table-usage":
            if (!customElements.get(tagName)) {
                customElements.define(tagName, DgaTableUsage$1);
            }
            break;
        case "dga-checkbox":
            if (!customElements.get(tagName)) {
                defineCustomElement$4();
            }
            break;
        case "dga-data-table":
            if (!customElements.get(tagName)) {
                defineCustomElement$3();
            }
            break;
        case "dga-icon":
            if (!customElements.get(tagName)) {
                defineCustomElement$2();
            }
            break;
    } });
}

const DgaTableUsage = DgaTableUsage$1;
const defineCustomElement = defineCustomElement$1;

export { DgaTableUsage, defineCustomElement };
