// @platformscode/core/dist/components/dga-search-box-usage.js
import { p as proxyCustomElement, H, h } from './p-JCGAt0My.js';
import { d as defineCustomElement$4 } from './p-CcNS0CYN.js';
import { d as defineCustomElement$3 } from './p-Cmg--ZO6.js';
import { d as defineCustomElement$2 } from './p-D9uV831l.js';

const DgaSearchBoxUsage$1 = /*@__PURE__*/ proxyCustomElement(class DgaSearchBoxUsage extends H {
    constructor() {
        super();
        this.__registerHost();
        this.value = 'test';
    }
    render() {
        return (h("div", { key: 'c470e35bbe0426ca814e07234cfeda1c7dc0aee5' }, h("h1", { key: '25b99581887c5739c07105b22b590d51fda26bb6' }, "Search Box Example"), h("div", { key: '0e7a2d8dc1e5b598f3ea5ece01b49b6f5d504861', style: { padding: '50px' } }, h("div", { key: '7dbfdea7041582285a17e43c3f2f580471be39a4', style: { display: "flex", flexDirection: "column", gap: "16px" } }, h("dga-search-box", { key: 'f2cc6f9a4d32283004199758900653f2c8d4ecae', variant: 'default', value: this.value, placeholder: 'test placeholder', onChange: (e) => {
                const target = e.target;
                console.log("search changed");
                this.value = target.value;
            }, onInput: (e) => {
                console.log("search changed", e.target.value);
                this.value = e.target.value;
            } })))));
    }
}, [0, "dga-search-box-usage", {
        "value": [32]
    }]);
function defineCustomElement$1() {
    if (typeof customElements === "undefined") {
        return;
    }
    const components = ["dga-search-box-usage", "dga-icon", "dga-search-box", "dga-tooltip"];
    components.forEach(tagName => { switch (tagName) {
        case "dga-search-box-usage":
            if (!customElements.get(tagName)) {
                customElements.define(tagName, DgaSearchBoxUsage$1);
            }
            break;
        case "dga-icon":
            if (!customElements.get(tagName)) {
                defineCustomElement$4();
            }
            break;
        case "dga-search-box":
            if (!customElements.get(tagName)) {
                defineCustomElement$3();
            }
            break;
        case "dga-tooltip":
            if (!customElements.get(tagName)) {
                defineCustomElement$2();
            }
            break;
    } });
}

const DgaSearchBoxUsage = DgaSearchBoxUsage$1;
const defineCustomElement = defineCustomElement$1;

export { DgaSearchBoxUsage, defineCustomElement };
