// @platformscode/core/dist/components/dga-tabs-usage.js
import { p as proxyCustomElement, H, h } from './p-JCGAt0My.js';
import { d as defineCustomElement$4 } from './p-BLDzt_Uq.js';
import { d as defineCustomElement$3 } from './p-CcNS0CYN.js';
import { d as defineCustomElement$2 } from './p-BfhhEgg-.js';

const DgaTabsUsage$1 = /*@__PURE__*/ proxyCustomElement(class DgaTabsUsage extends H {
    constructor() {
        super();
        this.__registerHost();
        this.activeTab = 'Home';
        this.handleTabClick = (tabName) => {
            console.log(`Tab clicked: ${tabName}`);
            this.activeTab = tabName;
            // Handle tab navigation logic here
        };
    }
    render() {
        return (h("div", { key: '0591d9bf25a4050197a1c3326d1bcebf795d264d', style: { padding: '20px', fontFamily: 'Arial, sans-serif' } }, h("h1", { key: 'b305e94f70cf35f7778e42028d2539a108d02fd4' }, "DGA Tabs Component Examples"), h("p", { key: '6b37a3c44b8f96ebbf80b487fc2cb1998408d334' }, "Comprehensive examples of the tabs component in various real-world scenarios."), h("section", { key: 'e7ba8adbf4f5417e3e81a9db0f3d5e3411aa1456', style: { marginBottom: '40px' } }, h("h2", { key: '25b0e0e2f02d70f4008b62ce6a57d78c7ec861a1' }, "1. Website Navigation Menu"), h("p", { key: 'f13bd68cfd51c141d7276bf446f576d9a3f310c1' }, "Main navigation for a website with different sections"), h("dga-tabs", { key: '37772a5d4796954b12b7c4e6141787c4a7d997bf', size: "md", orientation: "horizontal", flush: false, external: false, divider: true, tabsList: [
                {
                    label: 'Home',
                    onClick: () => this.handleTabClick('Home'),
                    tabIcon: "home-01",
                    iconProps: { variant: "stroke", type: "sharp" }
                },
                {
                    label: 'Products',
                    onClick: () => this.handleTabClick('Products'),
                    tabIcon: "package-01",
                    iconProps: { variant: "stroke", type: "sharp" }
                },
                {
                    label: 'Services',
                    onClick: () => this.handleTabClick('Services'),
                    tabIcon: "settings-01",
                    iconProps: { variant: "stroke", type: "sharp" }
                },
                {
                    label: 'About Us',
                    onClick: () => this.handleTabClick('About'),
                    tabIcon: "users-01",
                    iconProps: { variant: "stroke", type: "sharp" }
                },
                {
                    label: 'Contact',
                    onClick: () => this.handleTabClick('Contact'),
                    tabIcon: "mail-01",
                    iconProps: { variant: "stroke", type: "sharp" }
                }
            ] })), h("section", { key: 'ac30ffa3237d329d51e97f7dc25869317fe1af15', style: { marginBottom: '40px' } }, h("h2", { key: 'a99e2b72a6acf73c6a578c3852e1beca8da8c479' }, "2. Dashboard Analytics Tabs"), h("p", { key: '97a6c0ae6e4e9b956b9cce289bb3b5ea3829e8df' }, "Analytics dashboard with different data views"), h("dga-tabs", { key: '7451a8657a6c0d2bd8fc48c7ae9a233edaef645c', size: "lg", orientation: "horizontal", flush: false, external: false, divider: true, tabsList: [
                {
                    label: 'Overview',
                    onClick: () => this.handleTabClick('Overview'),
                    tabIcon: "chart-line-01",
                    iconProps: { variant: "stroke", type: "sharp" }
                },
                {
                    label: 'Sales',
                    onClick: () => this.handleTabClick('Sales'),
                    tabIcon: "trending-up-01",
                    iconProps: { variant: "stroke", type: "sharp" }
                },
                {
                    label: 'Users',
                    onClick: () => this.handleTabClick('Users'),
                    tabIcon: "user-group-01",
                    iconProps: { variant: "stroke", type: "sharp" }
                },
                {
                    label: 'Reports',
                    onClick: () => this.handleTabClick('Reports'),
                    tabIcon: "file-chart-01",
                    iconProps: { variant: "stroke", type: "sharp" }
                }
            ] })), h("section", { key: '0689b72936054cb1f2da1930cab43df37e3cc029', style: { marginBottom: '40px' } }, h("h2", { key: 'c57ff77efcbd1064f84821a8b01c07ecb349f26a' }, "3. Settings Panel (Vertical Layout)"), h("p", { key: 'beffd07210f5ab9e61def74a584f6285c5fd4cc8' }, "User settings panel with vertical tab navigation"), h("div", { key: '890a3049f60e0ac77ac81ea500ff098f63608687', style: { display: 'flex', gap: '20px' } }, h("dga-tabs", { key: '1a5d721cfcab7eed640fae78e819874186ed1f59', size: "md", orientation: "vertical", flush: false, external: false, divider: true, tabsList: [
                {
                    label: 'Profile',
                    onClick: () => this.handleTabClick('Profile'),
                    tabIcon: "user-01",
                    iconProps: { variant: "stroke", type: "sharp" }
                },
                {
                    label: 'Security',
                    onClick: () => this.handleTabClick('Security'),
                    tabIcon: "shield-01",
                    iconProps: { variant: "stroke", type: "sharp" }
                },
                {
                    label: 'Notifications',
                    onClick: () => this.handleTabClick('Notifications'),
                    tabIcon: "bell-01",
                    iconProps: { variant: "stroke", type: "sharp" }
                },
                {
                    label: 'Billing',
                    onClick: () => this.handleTabClick('Billing'),
                    tabIcon: "credit-card-01",
                    iconProps: { variant: "stroke", type: "sharp" }
                },
                {
                    label: 'Privacy',
                    onClick: () => this.handleTabClick('Privacy'),
                    tabIcon: "lock-01",
                    iconProps: { variant: "stroke", type: "sharp" }
                }
            ] }), h("div", { key: 'bc2f58630038644b9b1a8a46883e9f00aa7860ba', style: { padding: '20px', border: '1px solid #ddd', borderRadius: '4px', flex: '1 ' } }, h("p", { key: '855fb002401aaeb7652fb15a4a3951650182952e' }, "Content area for the selected tab (", this.activeTab, ") would appear here.")))), h("section", { key: '541ae6735b07a74f9477393771c2edc2f8743843', style: { marginBottom: '40px' } }, h("h2", { key: 'd8a95543c243fee41e07f0bc1be544ee11042f7e' }, "4. E-commerce Product Information"), h("p", { key: '9731d6a371219930aad5a8441dd65dca96ecb64b' }, "Product detail page with different information sections"), h("dga-tabs", { key: 'e900cf3532e98a9a710a87c396a9007725384223', size: "md", orientation: "horizontal", flush: false, external: false, divider: true, tabsList: [
                {
                    label: 'Description',
                    onClick: () => this.handleTabClick('Description'),
                    tabIcon: "file-text-01",
                    iconProps: { variant: "stroke", type: "sharp" }
                },
                {
                    label: 'Specifications',
                    onClick: () => this.handleTabClick('Specifications'),
                    tabIcon: "list-01",
                    iconProps: { variant: "stroke", type: "sharp" }
                },
                {
                    label: 'Reviews',
                    onClick: () => this.handleTabClick('Reviews'),
                    tabIcon: "star-01",
                    iconProps: { variant: "stroke", type: "sharp" }
                },
                {
                    label: 'Shipping',
                    onClick: () => this.handleTabClick('Shipping'),
                    tabIcon: "truck-01",
                    iconProps: { variant: "stroke", type: "sharp" }
                },
                {
                    label: 'Returns',
                    onClick: () => this.handleTabClick('Returns'),
                    tabIcon: "refresh-01",
                    iconProps: { variant: "stroke", type: "sharp" }
                }
            ] })), h("section", { key: '4c0231c06d95ba0b2e329309f9af076d54635eda', style: { marginBottom: '40px' } }, h("h2", { key: 'ad0a701d5425e77b41f6be054ffa2f7c92e45b1a' }, "5. External Navigation Links"), h("p", { key: 'ba28ac4ee0cb9f04a53d5373ab1cf286c2ad988d' }, "Tabs that navigate to external pages or different sections"), h("dga-tabs", { key: '7d862fb413054675125d7d900658c98c635e8193', size: "md", orientation: "horizontal", flush: false, external: true, divider: true, tabsList: [
                {
                    label: 'Documentation',
                    link: '/docs',
                    tabIcon: "book-01",
                    iconProps: { variant: "stroke", type: "sharp" }
                },
                {
                    label: 'API Reference',
                    link: '/api',
                    tabIcon: "code-01",
                    iconProps: { variant: "stroke", type: "sharp" }
                },
                {
                    label: 'Examples',
                    link: '/examples',
                    tabIcon: "folder-01",
                    iconProps: { variant: "stroke", type: "sharp" }
                },
                {
                    label: 'GitHub',
                    link: 'https://github.com',
                    tabIcon: "github-01",
                    iconProps: { variant: "stroke", type: "sharp" }
                }
            ] })), h("section", { key: '82d9d626e00663238a2033f14fed25a26822aaa4', style: { marginBottom: '40px' } }, h("h2", { key: 'b830f3e83310d556bc3225083564d1dd0ed0ca38' }, "6. Many Tabs with Overflow Menu"), h("p", { key: 'dd2b7a07b65e08ee1dc067bb6df3d97fddee41e5' }, "Example with more than 5 tabs to demonstrate overflow behavior"), h("dga-tabs", { key: '58d95d3e46608a9aba9214f2b0116d93f53ed9c8', size: "md", orientation: "horizontal", flush: false, external: false, divider: true, tabsList: [
                {
                    label: 'January',
                    onClick: () => this.handleTabClick('January'),
                    tabIcon: "calendar-01",
                    iconProps: { variant: "stroke", type: "sharp" }
                },
                {
                    label: 'February',
                    onClick: () => this.handleTabClick('February'),
                    tabIcon: "calendar-01",
                    iconProps: { variant: "stroke", type: "sharp" }
                },
                {
                    label: 'March',
                    onClick: () => this.handleTabClick('March'),
                    tabIcon: "calendar-01",
                    iconProps: { variant: "stroke", type: "sharp" }
                },
                {
                    label: 'April',
                    onClick: () => this.handleTabClick('April'),
                    tabIcon: "calendar-01",
                    iconProps: { variant: "stroke", type: "sharp" }
                },
                {
                    label: 'May',
                    onClick: () => this.handleTabClick('May'),
                    tabIcon: "calendar-01",
                    iconProps: { variant: "stroke", type: "sharp" }
                },
                {
                    label: 'June',
                    onClick: () => this.handleTabClick('June'),
                    tabIcon: "calendar-01",
                    iconProps: { variant: "stroke", type: "sharp" }
                },
                {
                    label: 'July',
                    onClick: () => this.handleTabClick('July'),
                    tabIcon: "calendar-01",
                    iconProps: { variant: "stroke", type: "sharp" }
                },
                {
                    label: 'August',
                    onClick: () => this.handleTabClick('August'),
                    tabIcon: "calendar-01",
                    iconProps: { variant: "stroke", type: "sharp" }
                }
            ] })), h("section", { key: '0fb0a2202dacaf3aaf1569a1af05200e1760b349', style: { marginBottom: '40px' } }, h("h2", { key: '3a5b239df81a9d4217cfcda67f314f79faa13d31' }, "7. Different Tab Sizes"), h("h3", { key: 'f4c6e85b444c74ecd150b8daef1ee5abf8f1b4f8' }, "Small Size"), h("dga-tabs", { key: '1d68badd946653fe4ebcd08bfdea682767f9cf59', size: "sm", orientation: "horizontal", flush: false, external: false, divider: true, tabsList: [
                { label: 'Small', onClick: () => this.handleTabClick('Small') },
                { label: 'Compact', onClick: () => this.handleTabClick('Compact') },
                { label: 'Minimal', onClick: () => this.handleTabClick('Minimal') }
            ] }), h("h3", { key: '80b81dc8af0d5f5f838067186147c8cc9af6b7e2' }, "Medium Size (Default)"), h("dga-tabs", { key: 'ff0f41477f3af42127027804d36be2c798d5550a', size: "md", orientation: "horizontal", flush: false, external: false, divider: true, tabsList: [
                { label: 'Medium', onClick: () => this.handleTabClick('Medium') },
                { label: 'Standard', onClick: () => this.handleTabClick('Standard') },
                { label: 'Default', onClick: () => this.handleTabClick('Default') }
            ] }), h("h3", { key: 'e0cb239d28faaadb51d4e6926d51370d369387cf' }, "Large Size"), h("dga-tabs", { key: '030d13de3cf406754cda715d6d3aa4f70ef4054a', size: "lg", orientation: "horizontal", flush: false, external: false, divider: true, tabsList: [
                { label: 'Large', onClick: () => this.handleTabClick('Large') },
                { label: 'Prominent', onClick: () => this.handleTabClick('Prominent') },
                { label: 'Bold', onClick: () => this.handleTabClick('Bold') }
            ] })), h("section", { key: '3d7e70da9172f1fd9bf4cf2007ab754e1cb44630', style: { marginBottom: '40px' } }, h("h2", { key: 'd930af857aca20ad76de9415eb68994bd150cc42' }, "8. Flush Tabs (No Padding)"), h("p", { key: '3263fb9222aebd899501ee82416daffbb33fe9c8' }, "Tabs with flush styling for tight layouts"), h("dga-tabs", { key: '8984f72ed6f38c37929a128cf3001226e5a8f884', size: "md", orientation: "horizontal", flush: true, external: false, divider: false, tabsList: [
                {
                    label: 'Flush Tab 1',
                    onClick: () => this.handleTabClick('Flush1'),
                    tabIcon: "square-01",
                    iconProps: { variant: "stroke", type: "sharp" }
                },
                {
                    label: 'Flush Tab 2',
                    onClick: () => this.handleTabClick('Flush2'),
                    tabIcon: "square-01",
                    iconProps: { variant: "stroke", type: "sharp" }
                },
                {
                    label: 'Flush Tab 3',
                    onClick: () => this.handleTabClick('Flush3'),
                    tabIcon: "square-01",
                    iconProps: { variant: "stroke", type: "sharp" }
                }
            ] })), h("section", { key: 'f0032c6dfe7b5b2446a1406e6f4f8ff6f0098636', style: { marginBottom: '40px' } }, h("h2", { key: 'a2b39e768eed846cf3cd26d1c390a49410569fa8' }, "9. Disabled Tabs"), h("p", { key: '069cae0c877fcf961a332e2f2b0009ade21d7735' }, "Tabs in disabled state for maintenance or restricted access"), h("dga-tabs", { key: '02d54cdda6a0f4b94e34688dba8904a868cb117d', size: "md", orientation: "horizontal", flush: false, external: false, divider: true, disabled: true, tabsList: [
                {
                    label: 'Disabled Tab 1',
                    onClick: () => this.handleTabClick('Disabled1'),
                    tabIcon: "lock-01",
                    iconProps: { variant: "stroke", type: "sharp" }
                },
                {
                    label: 'Disabled Tab 2',
                    onClick: () => this.handleTabClick('Disabled2'),
                    tabIcon: "lock-01",
                    iconProps: { variant: "stroke", type: "sharp" }
                },
                {
                    label: 'Disabled Tab 3',
                    onClick: () => this.handleTabClick('Disabled3'),
                    tabIcon: "lock-01",
                    iconProps: { variant: "stroke", type: "sharp" }
                }
            ] })), h("section", { key: 'c54905109099f55bb9a6f73a1e8ecc8c964eab32', style: { marginBottom: '40px' } }, h("h2", { key: '3d210d801df271d475f2f284f28062e36e6c0917' }, "10. Mobile-Friendly Tabs"), h("p", { key: '35fc6023cdd8914a12bcc45eda5864a9c53ec12d' }, "Compact tabs suitable for mobile interfaces"), h("dga-tabs", { key: 'fc674ca6857d1b070a52b745508ebdc022cdfb67', size: "sm", orientation: "horizontal", flush: true, external: false, divider: false, tabsList: [
                {
                    label: 'Feed',
                    onClick: () => this.handleTabClick('Feed'),
                    tabIcon: "rss-01",
                    iconProps: { variant: "stroke", type: "sharp" }
                },
                {
                    label: 'Messages',
                    onClick: () => this.handleTabClick('Messages'),
                    tabIcon: "message-01",
                    iconProps: { variant: "stroke", type: "sharp" }
                },
                {
                    label: 'Profile',
                    onClick: () => this.handleTabClick('Profile'),
                    tabIcon: "user-01",
                    iconProps: { variant: "stroke", type: "sharp" }
                }
            ] }))));
    }
}, [0, "dga-tabs-usage", {
        "activeTab": [32]
    }]);
function defineCustomElement$1() {
    if (typeof customElements === "undefined") {
        return;
    }
    const components = ["dga-tabs-usage", "dga-button", "dga-icon", "dga-tabs"];
    components.forEach(tagName => { switch (tagName) {
        case "dga-tabs-usage":
            if (!customElements.get(tagName)) {
                customElements.define(tagName, DgaTabsUsage$1);
            }
            break;
        case "dga-button":
            if (!customElements.get(tagName)) {
                defineCustomElement$4();
            }
            break;
        case "dga-icon":
            if (!customElements.get(tagName)) {
                defineCustomElement$3();
            }
            break;
        case "dga-tabs":
            if (!customElements.get(tagName)) {
                defineCustomElement$2();
            }
            break;
    } });
}

const DgaTabsUsage = DgaTabsUsage$1;
const defineCustomElement = defineCustomElement$1;

export { DgaTabsUsage, defineCustomElement };
