// @platformscode/core/dist/components/dga-pagination-usage.js
import { p as proxyCustomElement, H, h } from './p-JCGAt0My.js';
import { d as defineCustomElement$2 } from './p-DDV4TUDe.js';

const DgaPaginationUsage$1 = /*@__PURE__*/ proxyCustomElement(class DgaPaginationUsage extends H {
    constructor() {
        super();
        this.__registerHost();
        this.variant = 'neutral';
    }
    render() {
        return (h("div", { key: '90e571eb4e4c86b1d040563def1e63227f758430' }, h("h1", { key: '4cac213523a1ba23345328988df0b835a9c4cff9' }, "Pagination Example"), h("div", { key: '44c2bcce44cb4e165fbe562f13315c4d4a2ed4f6', style: { padding: '50px', display: 'flex', gap: '20px' } }, h("div", { key: 'e4e544aabf9ab4e57196e4b56805dd6fd8ee7cc9', style: { display: 'flex', flexDirection: 'column', gap: '20px' } }, h("dga-pagination", { key: '1af325367643fb1a86cc0bcb8624e7062a1500a9', totalPageCount: 10, size: 'large', onChange: (page) => console.log(page, "page changed") }), h("dga-pagination", { key: '3d516d50bf028ed7b5042f728f1135d14c073f81', totalPageCount: 10, size: 'medium' }), h("dga-pagination", { key: '25a0ef6f236f662e7a0e17c10d3e4b2f58e26c1d', totalPageCount: 10, size: 'small' })), h("div", { key: 'd84647b1c5e6de3a60510bd24fa7a25d8f1698c0', style: { display: 'flex', flexDirection: 'column', gap: '20px' } }))));
    }
}, [0, "dga-pagination-usage", {
        "variant": [32]
    }]);
function defineCustomElement$1() {
    if (typeof customElements === "undefined") {
        return;
    }
    const components = ["dga-pagination-usage", "dga-pagination"];
    components.forEach(tagName => { switch (tagName) {
        case "dga-pagination-usage":
            if (!customElements.get(tagName)) {
                customElements.define(tagName, DgaPaginationUsage$1);
            }
            break;
        case "dga-pagination":
            if (!customElements.get(tagName)) {
                defineCustomElement$2();
            }
            break;
    } });
}

const DgaPaginationUsage = DgaPaginationUsage$1;
const defineCustomElement = defineCustomElement$1;

export { DgaPaginationUsage, defineCustomElement };
