// @platformscode/core/dist/components/dga-modal-v2-usage.js
import { p as proxyCustomElement, H, h } from './p-JCGAt0My.js';
import { d as defineCustomElement$b } from './p-Bs_sPiZE.js';
import { d as defineCustomElement$a } from './p-CnoQYu5E.js';
import { d as defineCustomElement$9 } from './p-Cg2ruBZP.js';
import { d as defineCustomElement$8 } from './p-CcNS0CYN.js';
import { d as defineCustomElement$7 } from './p-CPioG2Cd.js';
import { d as defineCustomElement$6 } from './p-BGbuMluI.js';
import { d as defineCustomElement$5 } from './p-CzqjlWwh.js';
import { d as defineCustomElement$4 } from './p-Cgi2LtEh.js';
import { d as defineCustomElement$3 } from './p-B_m6I75z.js';
import { d as defineCustomElement$2 } from './p-eJMs31Nm.js';

const DgaModalUsage = /*@__PURE__*/ proxyCustomElement(class DgaModalUsage extends H {
    constructor() {
        super();
        this.__registerHost();
        this.show = false;
    }
    render() {
        return (h("div", { key: '9aaa7ec5d254c4ce0b47fb5acd2fb2101d70525f', style: { position: 'relative' } }, h("h1", { key: 'b166ef41f5d3fceb957ce090d93200b3e7ef2283' }, "Modal Example"), h("button", { key: 'f59e9e27958f573198e1dd71add2805c7572070c', onClick: () => (this.show = true) }, "Open Modal"), h("dga-modal-v2", { key: '2856d8704b25bab43a6c04e2e4ff6f6feb39d974', open: this.show, showCloseButton: true, onClose: () => (this.show = false) }, h("dga-modal-header", { key: 'ac6f5efd55dc5ba4f2dc693123001b6f5eb9ee78' }, h("dga-featured-icon", { key: '0761a03e39fe49a7bde152c548c001ec49995f55', color: "brand", variant: "light", size: "md", icon: {
                name: 'InformationCircle',
                variant: 'stroke',
                type: 'sharp',
                size: 20,
            } }), h("dga-modal-title", { key: 'adc0528a028a8dab0460418457c172e73e0407c7' }, "Title goes here")), h("dga-modal-body", { key: 'ad1fd1b5f3f8817da9f362f0be3b0d56c7964cc4' }, h("dga-flex", { key: 'fe70e22f63e51233fd84a4d2a89764421e0fba41', direction: "column", spacing: 8 }, h("p", { key: '2d2d5837063e0bf23ee5246bcbf1f0de2bf07b74', class: "text-sm-regular" }, "When a Modal needs a further detailed explanation, it goes here."), h("dga-rating", { key: '19e93e78a603fb9106fce68b6e1f9c43309168cb' }))), h("dga-modal-actions", { key: '080daf32c947a3459e1230146ace01599f0211bd' }, h("dga-button-v2", { key: 'e7d63c48c68ab2edb3f08c3ff35e044c4a7e1428', label: "Primary", size: "lg", variant: "primary", onClick: () => console.log('Primary Action') }), h("dga-button-v2", { key: '15e75ede1473436e80efde01608b25540a300248', label: "Secondary", size: "lg", variant: "secondary-outline", onClick: () => console.log('Secondary Action') })))));
    }
}, [0, "dga-modal-v2-usage", {
        "show": [32]
    }]);
function defineCustomElement$1() {
    if (typeof customElements === "undefined") {
        return;
    }
    const components = ["dga-modal-v2-usage", "dga-button-v2", "dga-featured-icon", "dga-flex", "dga-icon", "dga-modal-actions", "dga-modal-body", "dga-modal-header", "dga-modal-title", "dga-modal-v2", "dga-rating"];
    components.forEach(tagName => { switch (tagName) {
        case "dga-modal-v2-usage":
            if (!customElements.get(tagName)) {
                customElements.define(tagName, DgaModalUsage);
            }
            break;
        case "dga-button-v2":
            if (!customElements.get(tagName)) {
                defineCustomElement$b();
            }
            break;
        case "dga-featured-icon":
            if (!customElements.get(tagName)) {
                defineCustomElement$a();
            }
            break;
        case "dga-flex":
            if (!customElements.get(tagName)) {
                defineCustomElement$9();
            }
            break;
        case "dga-icon":
            if (!customElements.get(tagName)) {
                defineCustomElement$8();
            }
            break;
        case "dga-modal-actions":
            if (!customElements.get(tagName)) {
                defineCustomElement$7();
            }
            break;
        case "dga-modal-body":
            if (!customElements.get(tagName)) {
                defineCustomElement$6();
            }
            break;
        case "dga-modal-header":
            if (!customElements.get(tagName)) {
                defineCustomElement$5();
            }
            break;
        case "dga-modal-title":
            if (!customElements.get(tagName)) {
                defineCustomElement$4();
            }
            break;
        case "dga-modal-v2":
            if (!customElements.get(tagName)) {
                defineCustomElement$3();
            }
            break;
        case "dga-rating":
            if (!customElements.get(tagName)) {
                defineCustomElement$2();
            }
            break;
    } });
}

const DgaModalV2Usage = DgaModalUsage;
const defineCustomElement = defineCustomElement$1;

export { DgaModalV2Usage, defineCustomElement };
