// @platformscode/core/dist/components/dga-datepicker-usage.js
import { p as proxyCustomElement, H, h } from './p-JCGAt0My.js';
import { d as defineCustomElement$9 } from './p-Q1ZPiOaS.js';
import { d as defineCustomElement$8 } from './p-Bs_sPiZE.js';
import { d as defineCustomElement$7 } from './p-DzkNYNWQ.js';
import { d as defineCustomElement$6 } from './p-awnvl4MX.js';
import { d as defineCustomElement$5 } from './p-CcNS0CYN.js';
import { d as defineCustomElement$4 } from './p-B8Ehyo7m.js';
import { d as defineCustomElement$3 } from './p-COOZpHM-.js';
import { d as defineCustomElement$2 } from './p-DIBv3sUd.js';

const DgaPickerUsage = /*@__PURE__*/ proxyCustomElement(class DgaPickerUsage extends H {
    constructor() {
        super();
        this.__registerHost();
    }
    render() {
        return (h("div", { key: 'fbcbc060965934ebb20a3a0fdcb67565c24b8e42' }, h("h1", { key: '7ac0043599c31bc9739ea31abb6f3d2eea951923' }, "Datepicker Example"), h("div", { key: 'ec75b642bbc6926ee788dd6b4a523b8ac47ebbfb', style: { padding: '50px', display: 'flex', gap: '20px' } }, h("div", { key: 'c736092459f994c2449d2b9adbcf8aac95c0066e', style: { display: 'flex', flexDirection: 'column', gap: '20px' } }, h("dga-datepicker", { key: 'e31baf3ec5aaa21a3b4d5d5786f76966a7066267', inputField: false, qickOptions: true }), h("dga-datepicker", { key: 'e041ac56fad390dbe8476299ae92eadcc8106f45', RTL: true, inputField: true, range: true, qickOptions: true }), h("dga-datepicker", { key: 'fd911a8f875f789ba134fcba78d752f73000fd24', range: true })), h("div", { key: '57f0c922ab386a5d8ed794bade8f27a73c1d295b', style: { display: 'flex', flexDirection: 'column', gap: '20px' } }))));
    }
}, [0, "dga-datepicker-usage"]);
function defineCustomElement$1() {
    if (typeof customElements === "undefined") {
        return;
    }
    const components = ["dga-datepicker-usage", "date-cell", "dga-button-v2", "dga-datepicker", "dga-helper-text", "dga-icon", "dga-label", "dga-text-input", "month-navigator"];
    components.forEach(tagName => { switch (tagName) {
        case "dga-datepicker-usage":
            if (!customElements.get(tagName)) {
                customElements.define(tagName, DgaPickerUsage);
            }
            break;
        case "date-cell":
            if (!customElements.get(tagName)) {
                defineCustomElement$9();
            }
            break;
        case "dga-button-v2":
            if (!customElements.get(tagName)) {
                defineCustomElement$8();
            }
            break;
        case "dga-datepicker":
            if (!customElements.get(tagName)) {
                defineCustomElement$7();
            }
            break;
        case "dga-helper-text":
            if (!customElements.get(tagName)) {
                defineCustomElement$6();
            }
            break;
        case "dga-icon":
            if (!customElements.get(tagName)) {
                defineCustomElement$5();
            }
            break;
        case "dga-label":
            if (!customElements.get(tagName)) {
                defineCustomElement$4();
            }
            break;
        case "dga-text-input":
            if (!customElements.get(tagName)) {
                defineCustomElement$3();
            }
            break;
        case "month-navigator":
            if (!customElements.get(tagName)) {
                defineCustomElement$2();
            }
            break;
    } });
}

const DgaDatepickerUsage = DgaPickerUsage;
const defineCustomElement = defineCustomElement$1;

export { DgaDatepickerUsage, defineCustomElement };
