// @platformscode/core/dist/components/dga-floating-button-usage.js
import { p as proxyCustomElement, H, h } from './p-JCGAt0My.js';
import { d as defineCustomElement$3 } from './p-CcNS0CYN.js';
import { d as defineCustomElement$2 } from './p-DSD9HLIM.js';

const DgaButtonUsage = /*@__PURE__*/ proxyCustomElement(class DgaButtonUsage extends H {
    constructor() {
        super();
        this.__registerHost();
    }
    render() {
        return (h("div", { key: 'ac667cc05719ace6df84e5a8442522e842c1e925', style: {
                margin: "10px"
            } }, h("floating-button", { key: 'cec0a9b79781fcab05919358ad8ecbab7ee1a09a', variant: "secondary-solid" }), h("div", { key: 'a13b403b58d9a4d5026444702dc9b4a4a55cda59', style: {
                marginTop: "10px",
            } }, h("floating-button", { key: '09f0676171b6ef92dc62d93181ad10a4f2d866e2', variant: "primary-neutral", iconOnly: true }))));
    }
}, [0, "dga-floating-button-usage"]);
function defineCustomElement$1() {
    if (typeof customElements === "undefined") {
        return;
    }
    const components = ["dga-floating-button-usage", "dga-icon", "floating-button"];
    components.forEach(tagName => { switch (tagName) {
        case "dga-floating-button-usage":
            if (!customElements.get(tagName)) {
                customElements.define(tagName, DgaButtonUsage);
            }
            break;
        case "dga-icon":
            if (!customElements.get(tagName)) {
                defineCustomElement$3();
            }
            break;
        case "floating-button":
            if (!customElements.get(tagName)) {
                defineCustomElement$2();
            }
            break;
    } });
}

const DgaFloatingButtonUsage = DgaButtonUsage;
const defineCustomElement = defineCustomElement$1;

export { DgaFloatingButtonUsage, defineCustomElement };
