// @platformscode/core/dist/components/dga-date-field-usage.js
import { p as proxyCustomElement, H, h } from './p-JCGAt0My.js';
import { d as defineCustomElement$a } from './p-Q1ZPiOaS.js';
import { d as defineCustomElement$9 } from './p-Bs_sPiZE.js';
import { d as defineCustomElement$8 } from './p-BM-plR02.js';
import { d as defineCustomElement$7 } from './p-DzkNYNWQ.js';
import { d as defineCustomElement$6 } from './p-awnvl4MX.js';
import { d as defineCustomElement$5 } from './p-CcNS0CYN.js';
import { d as defineCustomElement$4 } from './p-B8Ehyo7m.js';
import { d as defineCustomElement$3 } from './p-COOZpHM-.js';
import { d as defineCustomElement$2 } from './p-DIBv3sUd.js';

const DgaDateFieldUsage$1 = /*@__PURE__*/ proxyCustomElement(class DgaDateFieldUsage extends H {
    constructor() {
        super();
        this.__registerHost();
        this.value = '';
    }
    render() {
        return (h("div", { key: '65d405bc1f0e94a8ba70d819a62ccaf2d53f205d' }, h("h1", { key: 'd9dd3c1908d8b1cd70c0af078a68c71c05a4720a' }, "Date Field Example"), h("div", { key: '9d0aa0de8dc68b7890fbc1a541b8579f2a91242e', style: { padding: '50px' } }, h("div", { key: 'ed0d390b3ee39ec8959ef72391e7c20a29d9cc4d', style: { display: "flex", flexDirection: "column", gap: "16px" } }, h("input", { key: 'fb9574659d66cc7a87cefb85f397eb5a129fab1b', type: "date", value: this.value, onChange: (e) => this.value = e.target.value }), h("dga-date-field", { key: 'a2319a369cbc86d2990c7f256337587bc44bf66d', value: this.value, onChange: (e) => this.value = e.target.value })))));
    }
}, [0, "dga-date-field-usage", {
        "value": [32]
    }]);
function defineCustomElement$1() {
    if (typeof customElements === "undefined") {
        return;
    }
    const components = ["dga-date-field-usage", "date-cell", "dga-button-v2", "dga-date-field", "dga-datepicker", "dga-helper-text", "dga-icon", "dga-label", "dga-text-input", "month-navigator"];
    components.forEach(tagName => { switch (tagName) {
        case "dga-date-field-usage":
            if (!customElements.get(tagName)) {
                customElements.define(tagName, DgaDateFieldUsage$1);
            }
            break;
        case "date-cell":
            if (!customElements.get(tagName)) {
                defineCustomElement$a();
            }
            break;
        case "dga-button-v2":
            if (!customElements.get(tagName)) {
                defineCustomElement$9();
            }
            break;
        case "dga-date-field":
            if (!customElements.get(tagName)) {
                defineCustomElement$8();
            }
            break;
        case "dga-datepicker":
            if (!customElements.get(tagName)) {
                defineCustomElement$7();
            }
            break;
        case "dga-helper-text":
            if (!customElements.get(tagName)) {
                defineCustomElement$6();
            }
            break;
        case "dga-icon":
            if (!customElements.get(tagName)) {
                defineCustomElement$5();
            }
            break;
        case "dga-label":
            if (!customElements.get(tagName)) {
                defineCustomElement$4();
            }
            break;
        case "dga-text-input":
            if (!customElements.get(tagName)) {
                defineCustomElement$3();
            }
            break;
        case "month-navigator":
            if (!customElements.get(tagName)) {
                defineCustomElement$2();
            }
            break;
    } });
}

const DgaDateFieldUsage = DgaDateFieldUsage$1;
const defineCustomElement = defineCustomElement$1;

export { DgaDateFieldUsage, defineCustomElement };
