// @platformscode/core/dist/components/dga-carousel-usage.js
import { p as proxyCustomElement, H, h } from './p-JCGAt0My.js';
import { d as defineCustomElement$4 } from './p-BQEjovbJ.js';
import { d as defineCustomElement$3 } from './p-BXNu0jRu.js';
import { d as defineCustomElement$2 } from './p-CcNS0CYN.js';

const DgaCarouselUsage$1 = /*@__PURE__*/ proxyCustomElement(class DgaCarouselUsage extends H {
    constructor() {
        super();
        this.__registerHost();
        this.variant = 'neutral';
    }
    render() {
        return (h("div", { key: '5c4f191063a3da952e6e42f4cd19103c2d715a3b' }, h("h1", { key: 'e0dbbd2edcfb86c2fbbfe547dcd490f00ac70ca4' }, "Carousel Example"), h("div", { key: '4fa719669714950f9d9fc8beb56bb7009aa215fc', style: { padding: '50px', display: 'flex', gap: '20px' } }, h("div", { key: 'e2af84279a8ce9c95632ff8f7db99f7521996775', style: { display: 'flex', flexDirection: 'column', gap: '20px' } }, h("dga-carousel", { key: '123e46bcbb544322886f75173036e3f73b00d9a6', size: 'lg', arrowsVariant: 'primary' }, h("dga-carousel-item", { key: '6e96cb5449d40b355d628ac69195afc9748d4960' }, h("img", { key: '1cd7447ad28a0c7cb66d2bbe642d33abf5ff86dc', src: "https://cdn.pixabay.com/photo/2019/11/03/01/56/landscape-4597742_1280.jpg", style: {
                height: '300px',
                width: '450px',
            } })), h("dga-carousel-item", { key: 'd41b8aa19885d0b04971dcaef7f3219c3962a6c5' }, h("img", { key: 'b58dbc9370456bf5b10504dcdacab0469bbedc7e', src: "https://cdn.pixabay.com/photo/2020/07/09/10/31/sea-5386810_1280.jpg", style: {
                height: '300px',
                width: '450px',
                objectFit: 'cover',
            } })), h("dga-carousel-item", { key: '949685809aec0a767579989a47a6473e9fd76a8b' }, h("img", { key: 'e39c80ef752e705b2fc6d1d1cc4bf05a467c6a59', src: "https://cdn.pixabay.com/photo/2019/11/03/01/56/landscape-4597742_1280.jpg", style: {
                height: '300px',
                width: '450px',
                objectFit: 'cover',
            } })), h("dga-carousel-item", { key: '24b5db9929035fd46e2ce1cc4c797b168a2ab9a8' }, h("img", { key: '3072a257cd7875ab93177278bf4abff5fe8eb4b1', src: "https://cdn.pixabay.com/photo/2020/07/09/10/31/sea-5386810_1280.jpg", style: {
                height: '300px',
                width: '450px',
                objectFit: 'cover',
            } })), h("dga-carousel-item", { key: 'a442930d2b9a6d82acd007036efad10fe8a67a18' }, h("img", { key: 'a9b413a1ccda7a7a57485622a3655da0ff92b6fb', src: "https://cdn.pixabay.com/photo/2020/07/09/10/31/sea-5386810_1280.jpg", style: {
                height: '300px',
                width: '450px',
                objectFit: 'cover',
            } })), h("dga-carousel-item", { key: '420484e73d6c166aad1b93dd996571854c73e579' }, h("img", { key: '30a1942c7fa903720132a3afa6c4ce6fd2617212', src: "https://cdn.pixabay.com/photo/2020/07/09/10/31/sea-5386810_1280.jpg", style: {
                height: '300px',
                width: '450px',
                objectFit: 'cover',
            } })), h("dga-carousel-item", { key: 'a958e9504a61578fa9281245bc1789a90a52684e' }, h("img", { key: 'd5f6faa2c6c757f31178349e293ad48fba3243e5', src: "https://cdn.pixabay.com/photo/2020/07/09/10/31/sea-5386810_1280.jpg", style: {
                height: '300px',
                width: '450px',
                objectFit: 'cover',
            } }))), h("dga-carousel", { key: 'cb182020cd4ed2306c47b9f4476db03a8613d559', size: 'lg', arrowsVariant: 'subtle' }, h("dga-carousel-item", { key: 'fd47971614a2997dd27590b0085ae683ba9e8bd7' }, h("img", { key: '2558ccb9d09e6127e5ae4bba09cddc771c3ab4ad', src: "https://cdn.pixabay.com/photo/2019/11/03/01/56/landscape-4597742_1280.jpg", style: {
                height: '300px',
                width: '450px',
            } })), h("dga-carousel-item", { key: '0b84913d49afb0680abb4c34ab21e4468957f1fc' }, h("img", { key: 'e6f25499145bcc4a66d78a49d440769bd960bd8e', src: "https://cdn.pixabay.com/photo/2020/07/09/10/31/sea-5386810_1280.jpg", style: {
                height: '300px',
                width: '450px',
                objectFit: 'cover',
            } })), h("dga-carousel-item", { key: '8d842f30488f32405023cdecfb3a060a61e42983' }, h("img", { key: '7aa3b89e397bb0cf819d910fd9edeba6ca66ad75', src: "https://cdn.pixabay.com/photo/2019/11/03/01/56/landscape-4597742_1280.jpg", style: {
                height: '300px',
                width: '450px',
                objectFit: 'cover',
            } })), h("dga-carousel-item", { key: '901089733f955a759488b513c4861e32803d8646' }, h("img", { key: '949d05ea24a20b538a9c23bd08d421b64496947a', src: "https://cdn.pixabay.com/photo/2020/07/09/10/31/sea-5386810_1280.jpg", style: {
                height: '300px',
                width: '450px',
                objectFit: 'cover',
            } })), h("dga-carousel-item", { key: '8c129144b60e0d81b1b1f6f8990cd2ea0afd32b8' }, h("img", { key: 'a1d2aedbee127fb4f1513f9385721cd529bb90dd', src: "https://cdn.pixabay.com/photo/2020/07/09/10/31/sea-5386810_1280.jpg", style: {
                height: '300px',
                width: '450px',
                objectFit: 'cover',
            } })), h("dga-carousel-item", { key: '18d6d6ff1553d34c744b818ea32084bfb2d1ffab' }, h("img", { key: 'ca0a194502888823e851392e29a11083c0f3f73a', src: "https://cdn.pixabay.com/photo/2020/07/09/10/31/sea-5386810_1280.jpg", style: {
                height: '300px',
                width: '450px',
                objectFit: 'cover',
            } })), h("dga-carousel-item", { key: '04251fca381ea93a9111e369da385a2129baab38' }, h("img", { key: 'd2405e0bb9e8210279bb2a434a88687d9e652d4c', src: "https://cdn.pixabay.com/photo/2020/07/09/10/31/sea-5386810_1280.jpg", style: {
                height: '300px',
                width: '450px',
                objectFit: 'cover',
            } }))), h("dga-carousel", { key: '791794e37fdeda3c2e44b9b9c05f70a25b11984c', size: 'md' }, h("dga-carousel-item", { key: '0b1545c5766d9f1aa7f0d7f801e0e8d74a34cfd5' }, h("img", { key: 'b49b679f35e97257d36277fd4628c93f68e47fbb', src: "https://cdn.pixabay.com/photo/2019/11/03/01/56/landscape-4597742_1280.jpg", style: {
                height: '300px',
                width: '450px',
            } })), h("dga-carousel-item", { key: 'd3764b267c528308840c6af974e900ea8446d418' }, h("img", { key: 'c56797b434247b20d096cae54151187e5aed396e', src: "https://cdn.pixabay.com/photo/2020/07/09/10/31/sea-5386810_1280.jpg", style: {
                height: '300px',
                width: '450px',
                objectFit: 'cover',
            } })), h("dga-carousel-item", { key: '412565f32126dfa34bd71b10e0bead1e45b5dc73' }, h("img", { key: '5a142718e6c75312dab32e712d2902dc07d887a6', src: "https://cdn.pixabay.com/photo/2019/11/03/01/56/landscape-4597742_1280.jpg", style: {
                height: '300px',
                width: '450px',
                objectFit: 'cover',
            } })), h("dga-carousel-item", { key: 'a8b08cf2e88df3d08b223593cfc2709192114385' }, h("img", { key: '51e428bb3f6e517723f07aa7f7ab45e73e2e07ef', src: "https://cdn.pixabay.com/photo/2020/07/09/10/31/sea-5386810_1280.jpg", style: {
                height: '300px',
                width: '450px',
                objectFit: 'cover',
            } })), h("dga-carousel-item", { key: '91b82af2f701c26986c39ed67bc507c743b8a1fb' }, h("img", { key: '0aab053ebbdc05b6c860f5e98aaf4185cb3e0084', src: "https://cdn.pixabay.com/photo/2020/07/09/10/31/sea-5386810_1280.jpg", style: {
                height: '300px',
                width: '450px',
                objectFit: 'cover',
            } })), h("dga-carousel-item", { key: 'c5b97df00dfb9feb2cf0b82c4ef297ba4d1d4327' }, h("img", { key: 'ddc507f35476a2e56a81027e0d4f2a2cab8c5873', src: "https://cdn.pixabay.com/photo/2020/07/09/10/31/sea-5386810_1280.jpg", style: {
                height: '300px',
                width: '450px',
                objectFit: 'cover',
            } })), h("dga-carousel-item", { key: 'df85a9251bddd0451d3a0db1d29be1c4a76c45df' }, h("img", { key: 'e9dc5537705fed110ab896b8d36a24c1a09390c1', src: "https://cdn.pixabay.com/photo/2020/07/09/10/31/sea-5386810_1280.jpg", style: {
                height: '300px',
                width: '450px',
                objectFit: 'cover',
            } }))), h("dga-carousel", { key: '396dc965f68ffa84e1552682bfb00ba7785279c5', size: 'sm' }, h("dga-carousel-item", { key: '615c2dbe2d121019feeca8f93dc0ede025fc3f8c' }, h("img", { key: '29467aba91e13287a2ba05e9e97870418a6703cb', src: "https://cdn.pixabay.com/photo/2019/11/03/01/56/landscape-4597742_1280.jpg", style: {
                height: '300px',
                width: '450px',
            } })), h("dga-carousel-item", { key: '4e7dde8d3c30d7bf80456bb70cff555c92921e28' }, h("img", { key: '765363ac7de87c3b00ab9dded6ee99f939ab4765', src: "https://cdn.pixabay.com/photo/2020/07/09/10/31/sea-5386810_1280.jpg", style: {
                height: '300px',
                width: '450px',
                objectFit: 'cover',
            } })), h("dga-carousel-item", { key: '0e0b615c68d944c3a10cc30538e55d3767268a8f' }, h("img", { key: 'bc8ef9fbf8e6d432c3546c13ae147d8a1e7cbc07', src: "https://cdn.pixabay.com/photo/2019/11/03/01/56/landscape-4597742_1280.jpg", style: {
                height: '300px',
                width: '450px',
                objectFit: 'cover',
            } })), h("dga-carousel-item", { key: 'f8f0bbdc6fcd4f5d7e492a036bb9c6ce0077adbc' }, h("img", { key: '2031d46b29a5002cfdc649fcb2254fcb7558859d', src: "https://cdn.pixabay.com/photo/2020/07/09/10/31/sea-5386810_1280.jpg", style: {
                height: '300px',
                width: '450px',
                objectFit: 'cover',
            } })), h("dga-carousel-item", { key: '5a26a3e45ea0305dbe667d888cddd53127361ec6' }, h("img", { key: 'df7ad698fe14cbbf5c2067a84adaf12048daa9ca', src: "https://cdn.pixabay.com/photo/2020/07/09/10/31/sea-5386810_1280.jpg", style: {
                height: '300px',
                width: '450px',
                objectFit: 'cover',
            } })), h("dga-carousel-item", { key: '14d6ffca06e1ba5c334e0ba075c6499e2802eae2' }, h("img", { key: '8ffaaf1cb69a4bf6148099e6b1fc73a948eee06a', src: "https://cdn.pixabay.com/photo/2020/07/09/10/31/sea-5386810_1280.jpg", style: {
                height: '300px',
                width: '450px',
                objectFit: 'cover',
            } })), h("dga-carousel-item", { key: '41243a8f1cd16a4fa3e8983e554fc4bcffe51eb9' }, h("img", { key: '20d11e1c584f72351c7c1f135ea52324f8554347', src: "https://cdn.pixabay.com/photo/2020/07/09/10/31/sea-5386810_1280.jpg", style: {
                height: '300px',
                width: '450px',
                objectFit: 'cover',
            } })))), h("div", { key: '5dc368ef3e35e246d8704849c95c2f09bb1ea61a', style: { display: 'flex', flexDirection: 'column', gap: '20px' } }))));
    }
}, [0, "dga-carousel-usage", {
        "variant": [32]
    }]);
function defineCustomElement$1() {
    if (typeof customElements === "undefined") {
        return;
    }
    const components = ["dga-carousel-usage", "dga-carousel", "dga-carousel-item", "dga-icon"];
    components.forEach(tagName => { switch (tagName) {
        case "dga-carousel-usage":
            if (!customElements.get(tagName)) {
                customElements.define(tagName, DgaCarouselUsage$1);
            }
            break;
        case "dga-carousel":
            if (!customElements.get(tagName)) {
                defineCustomElement$4();
            }
            break;
        case "dga-carousel-item":
            if (!customElements.get(tagName)) {
                defineCustomElement$3();
            }
            break;
        case "dga-icon":
            if (!customElements.get(tagName)) {
                defineCustomElement$2();
            }
            break;
    } });
}

const DgaCarouselUsage = DgaCarouselUsage$1;
const defineCustomElement = defineCustomElement$1;

export { DgaCarouselUsage, defineCustomElement };
