// @platformscode/core/dist/components/dga-modal-usage.js
import { p as proxyCustomElement, H, h } from './p-JCGAt0My.js';
import { d as defineCustomElement$5 } from './p-Bs_sPiZE.js';
import { d as defineCustomElement$4 } from './p-CnoQYu5E.js';
import { d as defineCustomElement$3 } from './p-CcNS0CYN.js';
import { d as defineCustomElement$2 } from './p-gl2x-iRx.js';

const DgaModalUsage$1 = /*@__PURE__*/ proxyCustomElement(class DgaModalUsage extends H {
    constructor() {
        super();
        this.__registerHost();
        this.show = false;
    }
    render() {
        return (h("div", { key: '1d7c4cef1e1f98538810ce4b0eb77166f30cf030', style: { position: "relative" } }, h("h1", { key: 'a7ebea0aa5d67666454bfeecc4306a7ab7a71dfd' }, "Modal Example"), h("button", { key: 'ab9d06082be15e449e14b96eb2eac55b3f65a800', onClick: () => this.show = true }, "Open Modal"), h("dga-modal", { key: '7328e375daaf6abc83cfc846d8458b34f924f94b', buttonsList: [
                {
                    extraClass: '',
                    label: 'Close',
                    onClick: () => { },
                    title: 'Close Modal'
                }
            ], closeButton: true, modalTitle: "Basic Modal", name: "basicModal", onClose: () => { }, open: this.show }, "This is a simple modal with default configuration."), this.show ? "shown" : "hidden"));
    }
}, [0, "dga-modal-usage", {
        "show": [32]
    }]);
function defineCustomElement$1() {
    if (typeof customElements === "undefined") {
        return;
    }
    const components = ["dga-modal-usage", "dga-button-v2", "dga-featured-icon", "dga-icon", "dga-modal"];
    components.forEach(tagName => { switch (tagName) {
        case "dga-modal-usage":
            if (!customElements.get(tagName)) {
                customElements.define(tagName, DgaModalUsage$1);
            }
            break;
        case "dga-button-v2":
            if (!customElements.get(tagName)) {
                defineCustomElement$5();
            }
            break;
        case "dga-featured-icon":
            if (!customElements.get(tagName)) {
                defineCustomElement$4();
            }
            break;
        case "dga-icon":
            if (!customElements.get(tagName)) {
                defineCustomElement$3();
            }
            break;
        case "dga-modal":
            if (!customElements.get(tagName)) {
                defineCustomElement$2();
            }
            break;
    } });
}

const DgaModalUsage = DgaModalUsage$1;
const defineCustomElement = defineCustomElement$1;

export { DgaModalUsage, defineCustomElement };
