// @platformscode/core/dist/components/dga-button-v2-usage.js
import { p as proxyCustomElement, H, h } from './p-JCGAt0My.js';
import { d as defineCustomElement$5 } from './p-CfQ8F8Bh.js';
import { d as defineCustomElement$4 } from './p-Bs_sPiZE.js';
import { d as defineCustomElement$3 } from './p-Cg2ruBZP.js';
import { d as defineCustomElement$2 } from './p-CcNS0CYN.js';

const DgaButtonUsage = /*@__PURE__*/ proxyCustomElement(class DgaButtonUsage extends H {
    constructor() {
        super();
        this.__registerHost();
        this.variant = 'neutral';
    }
    render() {
        return (h("div", { key: '5523fcf839868200e2f885b2f5ffe0a2ac92c38e' }, h("h1", { key: '8c446ebc307e784287a652c0e9183dc4542fabf2' }, "Button Example"), h("div", { key: '51f0dc01fe8b35fd9aebc073a9127f642146fa0a', style: { padding: '50px', display: 'flex', gap: '20px' } }, h("div", { key: '21c12e0579b99fb44164a4755301482b2dac1b41', style: { display: 'flex', flexDirection: 'column', gap: '20px' } }, h("dga-flex", { key: 'f9aaf1cb31dabb25d3b000aa22360ba1efbd7b69', direction: "column", spacing: 16 }, h("dga-button-v2", { key: 'fe90da1f4e5a9c71261f723e52d0f93a12125452', label: "Button", variant: "primary", size: "lg" }), h("dga-button-v2", { key: '25d036902f713e54b77f9a3611f231b47fcfd1eb', label: "Button", variant: "primary", size: "md" }), h("dga-button-v2", { key: '92a8444a7d5dfb0fe08bf52260b85b4fad3c2436', label: "Button", variant: "primary", size: "sm" }), h("dga-button-v2", { key: '3b2fd6212430234b7db9717ffe8c50b2203c5a9b', label: "Button", variant: "neutral", size: "lg" }), h("dga-button-v2", { key: '465460b966b50979670fa85fb894910fdfc7a04e', label: "Button", variant: "neutral", size: "md" }), h("dga-button-v2", { key: '23abf1fc63c47d833e6095d85914356be40b07d7', label: "Button", variant: "neutral", size: "sm" }), h("dga-button-v2", { key: '20290811a07733b77af71467c32445907d272686', label: "Button", variant: "secondary-solid", size: "lg" }), h("dga-button-v2", { key: '79a3bc5e1bd7745a6910f7effb1f92b2c7b6f979', label: "Button", variant: "secondary-solid", size: "md" }), h("dga-button-v2", { key: 'ce0b5540330c1e20c3cf66fdb15efe2ec0830607', label: "Button", variant: "secondary-solid", size: "sm" }), h("dga-button-v2", { key: 'f5214703c87436e7acbd8f26e43f1e4e4fa61b2f', label: "Button", variant: "primary", destructive: true, size: "lg" }), h("dga-button-v2", { key: '1ab821a70059d54343b55951bf05a95645fd87f3', label: "Button", variant: "primary", destructive: true, size: "md" }), h("dga-button-v2", { key: 'b6c5e5d1248f48f0a0a2ba4bccd4f27a542b5083', label: "Button", variant: "primary", destructive: true, size: "sm" })), h("dga-button-v2", { key: 'c21e765eb09b2fb1fcc7c01c7516b24c755d0cad', label: "Button", variant: "primary", onColor: true }, h("dga-button-leading-icon", { key: '6df379b1b089021fa02b94c02e9842e538712883', name: "link-square-01", size: 16, color: "red" })), h("dga-button-v2", { key: 'efd9c98325728d6a9287784b32d8ae07afa52f38', label: "Source code", variant: "secondary-outline", leadIcon: true, leadIconProps: { name: 'source-code', variant: 'stroke', type: 'standard' } }), h("dga-button-v2", { key: 'bcd305f5c8c8256f2c2bc3c2eeb25e5a6d021e5f',
            // className="absolute top-[50%] left-[50%] transform -translate-x-1/2 -translate-y-1/2 opacity-0 group-hover:opacity-100"
            label: "See live preview", size: "lg", variant: "secondary-outline", trailIcon: true, trailIconProps: {
                name: 'link-square-01',
                variant: 'stroke',
                type: 'standard',
            } })), h("div", { key: '68c90d90c6f615c504d7cdeb2394c4341fd0ac6b', style: { display: 'flex', flexDirection: 'column', gap: '20px' } }))));
    }
}, [0, "dga-button-v2-usage", {
        "variant": [32]
    }]);
function defineCustomElement$1() {
    if (typeof customElements === "undefined") {
        return;
    }
    const components = ["dga-button-v2-usage", "dga-button-leading-icon", "dga-button-v2", "dga-flex", "dga-icon"];
    components.forEach(tagName => { switch (tagName) {
        case "dga-button-v2-usage":
            if (!customElements.get(tagName)) {
                customElements.define(tagName, DgaButtonUsage);
            }
            break;
        case "dga-button-leading-icon":
            if (!customElements.get(tagName)) {
                defineCustomElement$5();
            }
            break;
        case "dga-button-v2":
            if (!customElements.get(tagName)) {
                defineCustomElement$4();
            }
            break;
        case "dga-flex":
            if (!customElements.get(tagName)) {
                defineCustomElement$3();
            }
            break;
        case "dga-icon":
            if (!customElements.get(tagName)) {
                defineCustomElement$2();
            }
            break;
    } });
}

const DgaButtonV2Usage = DgaButtonUsage;
const defineCustomElement = defineCustomElement$1;

export { DgaButtonV2Usage, defineCustomElement };
