// @platformscode/core/dist/components/dga-radio-button-usage.js
import { p as proxyCustomElement, H, h } from './p-JCGAt0My.js';
import { d as defineCustomElement$3 } from './p-CcNS0CYN.js';
import { d as defineCustomElement$2 } from './p-Bs1WQBbi.js';

const DgaRadioButtonUsage$1 = /*@__PURE__*/ proxyCustomElement(class DgaRadioButtonUsage extends H {
    constructor() {
        super();
        this.__registerHost();
        this.activeStep = 1;
        this.activeRadio = "radion-1";
    }
    render() {
        return (h("div", { key: '62906a14468b7d4a28d5700a18e317179f575868' }, h("h1", { key: '2e82948b03e30e7d040d414e29789ec4ee5e39c9' }, "RadioButton Example"), h("div", { key: '7597141a32352ef8f384e3438432bc26ff4ab02c', style: { padding: '50px', display: 'flex', gap: '20px' } }, h("div", { key: '37a55e92273fb92763583d9073961e85eae65a53', style: { display: 'flex', flexDirection: 'column', gap: '20px' } }, h("dga-radio-button", { key: '83e8156712c8950be1a07dcc889c13530b7e4c43',
            // id="radio1"
            label: "Radio Label 1", hasHelperText: true, helperText: "When a selection needs a further detailed explanation, it goes here.", alertMessage: true, alertText: "Error/Warning message", name: "radio", value: 'radion-1', checked: this.activeRadio === 'radion-1', onChange: (e) => this.activeRadio = e.target.value }), h("dga-radio-button", { key: '9e8464f3ecbd9bbdb7acd171fb76cbd926f04481',
            // id="radio2"
            label: "Radio Label 2", hasHelperText: true, helperText: "When a selection needs a further detailed explanation, it goes here.", alertMessage: true, alertText: "Error/Warning message", name: "radio", value: 'radion-2', checked: this.activeRadio === 'radion-2', onChange: (e) => this.activeRadio = e.target.value })), h("div", { key: '2c165379c18f2434a6eaea9b6dd372b332a62f21', style: { display: 'flex', flexDirection: 'column', gap: '20px' } }))));
    }
}, [0, "dga-radio-button-usage", {
        "activeStep": [32],
        "activeRadio": [32]
    }]);
function defineCustomElement$1() {
    if (typeof customElements === "undefined") {
        return;
    }
    const components = ["dga-radio-button-usage", "dga-icon", "dga-radio-button"];
    components.forEach(tagName => { switch (tagName) {
        case "dga-radio-button-usage":
            if (!customElements.get(tagName)) {
                customElements.define(tagName, DgaRadioButtonUsage$1);
            }
            break;
        case "dga-icon":
            if (!customElements.get(tagName)) {
                defineCustomElement$3();
            }
            break;
        case "dga-radio-button":
            if (!customElements.get(tagName)) {
                defineCustomElement$2();
            }
            break;
    } });
}

const DgaRadioButtonUsage = DgaRadioButtonUsage$1;
const defineCustomElement = defineCustomElement$1;

export { DgaRadioButtonUsage, defineCustomElement };
