// @platformscode/core/dist/components/dga-tooltip-usage.js
import { p as proxyCustomElement, H, h } from './p-JCGAt0My.js';
import { d as defineCustomElement$3 } from './p-CcNS0CYN.js';
import { d as defineCustomElement$2 } from './p-D9uV831l.js';

const DgaTooltipUsage$1 = /*@__PURE__*/ proxyCustomElement(class DgaTooltipUsage extends H {
    constructor() {
        super();
        this.__registerHost();
        this.activeStep = 1;
    }
    render() {
        return (h("div", { key: '0064e5dff22f0ff8a3b5f569deb22d68e2b2c94c' }, h("h1", { key: '825e0479f5239adfe639b7b66da34de9725d7bc1' }, "Tooltip Example"), h("div", { key: '331afe44f9f531fabf614d567f9353f32ab5c092', style: { padding: "50px", display: "flex", gap: "20px" } }, h("div", { key: 'b3d786abd00af5ddf19ce70c866109a73a60bc4e', style: { display: "flex", flexDirection: "column", gap: "20px" } }, h("dga-tooltip", { key: '29a1d4ffc256c0e497f2b0e6641bc3fb5d992c3d', title: "Tooltip title", helperText: "Max width of tooltips is 240px - text will wrap automatically", direction: 'right', theme: 'dark', icon: true, beakAlignment: 'end', beakPlacement: 'left', feedbackIcon: {
                type: "error-!",
                style: "container",
                size: 24
            } }, "Hello")), h("div", { key: '8a42e1d104123e01fc04f7a1daee7beb20129488', style: { display: "flex", flexDirection: "column", gap: "20px" } }))));
    }
}, [0, "dga-tooltip-usage", {
        "activeStep": [32]
    }]);
function defineCustomElement$1() {
    if (typeof customElements === "undefined") {
        return;
    }
    const components = ["dga-tooltip-usage", "dga-icon", "dga-tooltip"];
    components.forEach(tagName => { switch (tagName) {
        case "dga-tooltip-usage":
            if (!customElements.get(tagName)) {
                customElements.define(tagName, DgaTooltipUsage$1);
            }
            break;
        case "dga-icon":
            if (!customElements.get(tagName)) {
                defineCustomElement$3();
            }
            break;
        case "dga-tooltip":
            if (!customElements.get(tagName)) {
                defineCustomElement$2();
            }
            break;
    } });
}

const DgaTooltipUsage = DgaTooltipUsage$1;
const defineCustomElement = defineCustomElement$1;

export { DgaTooltipUsage, defineCustomElement };
