// @platformscode/core/dist/components/dga-notification-toast-usage.js
import { p as proxyCustomElement, H, h } from './p-JCGAt0My.js';
import { d as defineCustomElement$4 } from './p-CnoQYu5E.js';
import { d as defineCustomElement$3 } from './p-CcNS0CYN.js';
import { d as defineCustomElement$2 } from './p-CYqnvK_U.js';

const DgaNotificationToastUsage$1 = /*@__PURE__*/ proxyCustomElement(class DgaNotificationToastUsage extends H {
    constructor() {
        super();
        this.__registerHost();
        this.type = 'success';
        this.open = false;
    }
    render() {
        return (h("div", { key: '56e0f0bf8e4b0adf120ff0d869cdda544cdef401' }, h("h1", { key: '79317ee62ad38058ca4012aa6ffbec8ddbf2f5d4' }, "Notification Toast Example"), h("div", { key: 'a81c0bcad6dc89f42527b614fea0c780e4a748f3', style: { padding: '50px', display: 'flex', gap: '20px' } }, h("div", { key: '574eb746ddf78d37d760aa8670a8a9eb53334db2', style: { display: 'flex', flexDirection: 'column', gap: '20px' } }, h("div", { key: '2c9d26c302f25bf03f843cda3031dc5e91e62232', style: { display: "flex", justifyContent: "center" } }, h("button", { key: 'b0e38267fa51e8e929642fa7ea1e90ea2c8fe34b', onClick: () => this.open = true }, "open"), h("button", { key: '776a37bfd3d2442059289dc0a16fd0a1942f822b', onClick: () => this.open = false }, "close")), h("dga-notification-toast", { key: '029a110d4a944a92bfeb480a24687211ceeb4129', type: this.type, open: this.open, leadText: "Notification/Alert message title", hasHelperText: true, helperText: "When a Notification/Alert needs a further detailed explanation, it goes here.", action: true, buttonsList: [
                {
                    label: 'Button',
                    onClick: () => { },
                    title: 'Button 2',
                },
                {
                    label: 'Button',
                    onClick: () => { },
                    title: 'Button',
                },
            ], closeButton: true, vPostion: 'top', hPostion: 'left', onClose: () => this.open = false }), h("dga-notification-toast", { key: '23a3e770b81f7c12a448df153ac89599b1d4092b', type: this.type, open: this.open, leadText: "Notification/Alert message title", hasHelperText: true, helperText: "When a Notification/Alert needs a further detailed explanation, it goes here.", action: true, buttonsList: [
                {
                    label: 'Button',
                    onClick: () => { },
                    title: 'Button 2',
                },
                {
                    label: 'Button',
                    onClick: () => { },
                    title: 'Button',
                },
            ], closeButton: true, vPostion: 'bottom', hPostion: 'left', onClose: () => this.open = false })), h("div", { key: '79bbebda27aa2e4e94e34c1989b6179ef0502f3e', style: { display: 'flex', flexDirection: 'column', gap: '20px' } }))));
    }
}, [0, "dga-notification-toast-usage", {
        "type": [32],
        "open": [32]
    }]);
function defineCustomElement$1() {
    if (typeof customElements === "undefined") {
        return;
    }
    const components = ["dga-notification-toast-usage", "dga-featured-icon", "dga-icon", "dga-notification-toast"];
    components.forEach(tagName => { switch (tagName) {
        case "dga-notification-toast-usage":
            if (!customElements.get(tagName)) {
                customElements.define(tagName, DgaNotificationToastUsage$1);
            }
            break;
        case "dga-featured-icon":
            if (!customElements.get(tagName)) {
                defineCustomElement$4();
            }
            break;
        case "dga-icon":
            if (!customElements.get(tagName)) {
                defineCustomElement$3();
            }
            break;
        case "dga-notification-toast":
            if (!customElements.get(tagName)) {
                defineCustomElement$2();
            }
            break;
    } });
}

const DgaNotificationToastUsage = DgaNotificationToastUsage$1;
const defineCustomElement = defineCustomElement$1;

export { DgaNotificationToastUsage, defineCustomElement };
