// @platformscode/core/dist/components/dga-footer-usage.js
import { p as proxyCustomElement, H, h } from './p-JCGAt0My.js';
import { d as defineCustomElement$3 } from './p-BAiv59EH.js';
import { d as defineCustomElement$2 } from './p-CcNS0CYN.js';

const DgaFooterUsage$1 = /*@__PURE__*/ proxyCustomElement(class DgaFooterUsage extends H {
    constructor() {
        super();
        this.__registerHost();
    }
    render() {
        return (h("div", { key: '2f479a75f1c2014311fe835fd5aee75eb5012dda' }, h("h1", { key: 'f9077ae7ca07171c3e9c148d2e9363b20ff8496c' }, "Footer Example Dark Green Mode"), h("dga-footer", { key: '363454b8832fedd26ff69feb925e5dc7672a481c', background: "DarkGreen", NavLinks: true, groupLinks: [
                {
                    title: 'Government Services',
                    links: [
                        { name: 'Digital Identity', target: 'http://localhost:3333/digital-identity' },
                        { name: 'E-Services Portal', target: 'http://localhost:3333/e-services' },
                        { name: 'Government Directory', target: 'http://localhost:3333/directory' },
                        { name: 'Digital Transformation', target: 'http://localhost:3333/transformation' },
                        { name: 'Cybersecurity', target: 'http://localhost:3333/cybersecurity' },
                    ],
                },
                {
                    title: 'About Us',
                    links: [
                        { name: 'About Us', target: 'http://localhost:3333/about-us' },
                        { name: 'Contact Us', target: 'http://localhost:3333/contact-us' },
                        { name: 'News and Updates', target: 'http://localhost:3333/news-updates' },
                        { name: 'Careers', target: 'http://localhost:3333/careers' },
                    ],
                },
                {
                    title: 'Support',
                    links: [
                        { name: 'Help Center', target: 'http://localhost:3333/help-center' },
                        { name: 'Feedback', target: 'http://localhost:3333/feedback' },
                        { name: 'Terms of Service', target: 'http://localhost:3333/terms-of-service' },
                        { name: 'Privacy Policy', target: 'http://localhost:3333/privacy-policy' },
                    ],
                },
            ], socialMediaTitle: "Social Media", accessibilityTitle: "accessibility Tools", socialMediaLinks: [
                {
                    title: 'Twitter',
                    target: 'https://twitter.com/dga_gov',
                    icon: { name: 'TwitterIcon', variant: 'stroke' },
                },
                {
                    title: 'YouTube',
                    target: 'https://youtube.com/dga',
                    icon: { name: 'YoutubeIcon', variant: 'stroke' },
                },
                {
                    title: 'Instagram',
                    target: 'https://instagram.com/dga_gov',
                    icon: { name: 'InstagramIcon', variant: 'stroke' },
                },
            ], accessibilityLinks: [
                {
                    title: 'Screen Reader',
                    target: 'http://localhost:3333/accessibility/screen-reader',
                    icon: { name: "VoiceIcon", variant: "stroke" },
                },
                {
                    title: 'High Contrast',
                    target: 'http://localhost:3333/accessibility/contrast',
                    icon: { name: "VoiceIcon", variant: "stroke" },
                },
                {
                    title: 'Text Size',
                    target: 'http://localhost:3333/accessibility/text-size',
                    icon: { name: "VoiceIcon", variant: "stroke" },
                },
            ], copyright: "\u00A9 2024 Digital Government Authority. All rights reserved.", basicLinks: [
                { name: 'Home', target: '/' },
                { name: 'About DGA', target: '/about' },
                { name: 'News & Events', target: '/news' },
                { name: 'Publications', target: '/publications' },
                { name: 'Careers', target: '/careers' },
                { name: 'Procurement', target: '/procurement' },
                { name: 'Open Data', target: '/data' },
                { name: 'Statistics', target: '/stats' },
            ], extraLinks: [
                { name: 'Terms and Conditions', target: '/terms' },
                { name: 'Privacy Policy', target: '/privacy' },
                { name: 'Cookie Policy', target: '/cookies' },
                { name: 'Accessibility Statement', target: '/accessibility' },
            ], bottomImages: ['https://dga-nds-fbhtx.ondigitalocean.app/mobile-logo.svg', 'https://dga-nds-fbhtx.ondigitalocean.app/mobile-logo.svg'] }), h("h1", { key: '74f97f1cb2857b2909d98a85548c4193c0d63d54', style: { padding: '40px', color: 'red' } }, " Footer Example Light Mode"), h("dga-footer", { key: '0f228d46ae1cf766f6ef3c72b691603f93d23aa2', background: "Light", NavLinks: true, groupLinks: [
                {
                    title: 'Group Label',
                    links: [
                        { name: 'Footer Link', target: '' },
                        { name: 'Footer Link', target: '' },
                        { name: 'Footer Link', target: '' },
                        { name: 'Footer Link', target: '' },
                        { name: 'Footer Link', target: '' },
                    ],
                },
                {
                    title: 'Group Label',
                    links: [
                        { name: 'Footer Link', target: '' },
                        { name: 'Footer Link', target: '' },
                        { name: 'Footer Link', target: '' },
                        { name: 'Footer Link', target: '' },
                        { name: 'Footer Link', target: '' },
                    ],
                },
                {
                    title: 'Group Label',
                    links: [
                        { name: 'Footer Link', target: '' },
                        { name: 'Footer Link', target: '' },
                        { name: 'Footer Link', target: '' },
                        { name: 'Footer Link', target: '' },
                        { name: 'Footer Link', target: '' },
                    ],
                },
            ], socialMediaTitle: "Social Media", accessibilityTitle: "accessibility Tools", socialMediaLinks: [
                {
                    title: 'any',
                    target: '',
                    icon: { name: 'ArrowRight02Icon', variant: 'stroke' },
                },
                {
                    title: 'any',
                    target: '',
                    icon: { name: 'ArrowRight02Icon', variant: 'stroke' },
                },
                {
                    title: 'any',
                    target: '',
                    icon: { name: 'ArrowRight02Icon', variant: 'stroke' },
                },
                {
                    title: 'any',
                    target: '',
                    icon: { name: 'ArrowRight02Icon', variant: 'stroke' },
                },
            ], accessibilityLinks: [
                {
                    title: 'any',
                    target: '',
                    icon: { name: 'ArrowRight02Icon', variant: 'stroke' },
                },
                {
                    title: 'any',
                    target: '',
                    icon: { name: 'ArrowRight02Icon', variant: 'stroke' },
                },
                {
                    title: 'any',
                    target: '',
                    icon: { name: 'ArrowRight02Icon', variant: 'stroke' },
                },
            ], copyright: "All Right Reserved For Digital Government Authority \u00A9 2024", basicLinks: [
                {
                    name: 'Footer Link',
                    target: '#',
                },
                {
                    name: 'Footer Link',
                    target: '#',
                },
                {
                    name: 'Footer Link',
                    target: '#',
                },
                {
                    name: 'Footer Link',
                    target: '#',
                },
                {
                    name: 'Footer Link',
                    target: '#',
                },
                {
                    name: 'Footer Link',
                    target: '#',
                },
                {
                    name: 'Footer Link',
                    target: '#',
                },
                {
                    name: 'Footer Link',
                    target: '#',
                },
            ], extraLinks: [
                {
                    name: 'Terms and Conditions',
                    target: '#',
                },
                {
                    name: 'Privacy Policy',
                    target: '#',
                },
            ] }), h("h1", { key: '5c3971ce4a578c7cbedf7ce8e17637a849f1eaa7', style: { padding: '40px', color: 'red' } }, "End Footer Example Here Zoz")));
    }
}, [0, "dga-footer-usage"]);
function defineCustomElement$1() {
    if (typeof customElements === "undefined") {
        return;
    }
    const components = ["dga-footer-usage", "dga-footer", "dga-icon"];
    components.forEach(tagName => { switch (tagName) {
        case "dga-footer-usage":
            if (!customElements.get(tagName)) {
                customElements.define(tagName, DgaFooterUsage$1);
            }
            break;
        case "dga-footer":
            if (!customElements.get(tagName)) {
                defineCustomElement$3();
            }
            break;
        case "dga-icon":
            if (!customElements.get(tagName)) {
                defineCustomElement$2();
            }
            break;
    } });
}

const DgaFooterUsage = DgaFooterUsage$1;
const defineCustomElement = defineCustomElement$1;

export { DgaFooterUsage, defineCustomElement };
