// @platformscode/core/dist/components/dga-text-input-usage.js
import { p as proxyCustomElement, H, h } from './p-JCGAt0My.js';
import { d as defineCustomElement$6 } from './p-awnvl4MX.js';
import { d as defineCustomElement$5 } from './p-CcNS0CYN.js';
import { d as defineCustomElement$4 } from './p-B8Ehyo7m.js';
import { d as defineCustomElement$3 } from './p-VIZKxB_k.js';
import { d as defineCustomElement$2 } from './p-COOZpHM-.js';

const DgaTextInputUsage$1 = /*@__PURE__*/ proxyCustomElement(class DgaTextInputUsage extends H {
    constructor() {
        super();
        this.__registerHost();
        this.value = 'Text Input Example';
    }
    render() {
        return (h("div", { key: '0c90baa699105bb5d608b1e78216da2f6c083d30' }, h("h1", { key: 'c1ae1ee88e98fed39ebf112dd81a086953ec4e1d' }, "Text Input Example"), h("div", { key: '1ef35c412bc760d61ff8598fc1fbe79285e8ed0e', style: { padding: '50px' } }, h("div", { key: 'b2c9286b887c058e35b246ec3e499f918e23c7e1', style: { display: 'flex', flexDirection: 'column', gap: '16px' } }, h("input", { key: 'b65a0ddea8d13b9aaed9ab1a68e3ed7b863f08a3', type: "text", value: this.value, onChange: (e) => (this.value = e.target.value) }), h("p", { key: '452491e078e77f5749aa3eb510e6b1fc8749e47f' }, this.value), h("dga-text-input", { key: 'ac16179be1e3c95671d24c4215b3b7e5e98b50ba', readonly: false, variant: "darker", label: "Label", value: this.value, required: true, feedbackIcon: true, icon: { name: 'search-01', variant: 'stroke', type: 'standard' }, helperTextIcon: { name: 'alert-circle', variant: 'stroke', type: 'standard' }, helperText: "test", error: true, onInput: (e) => (this.value = e.target.value) }), h("dga-text-input", { key: 'e8bf194137d68c93becf43de1f26889fb1d51ef6', readonly: false, variant: "darker", label: "Label", value: "Entered text", required: true, feedbackIcon: true, icon: { name: 'search-01', variant: 'stroke', type: 'standard' }, helperText: "test", error: true }), h("dga-text-input", { key: '993b6a76f021e94cf19db6b9c2c24e0a443dadb1', readonly: false, variant: "darker", label: "Label", value: "Entered text", required: true, feedbackIcon: true, icon: { name: 'search-01', variant: 'stroke', type: 'standard' }, helperTextIcon: { name: 'alert-circle', variant: 'stroke', type: 'standard' }, helperText: "test", error: true }, h("dga-prefix", { key: '9d75eea2e36727d86d48738ea4188c842815729b', type: "button", variant: "solid",
            // label="Button Label"
            icon: { name: 'search-01' } })), h("dga-label", { key: 'bdf3180d3dfd53e11ff40018dca7c55ae6597879', label: "Mandatory Text Input" }), h("dga-label", { key: '15d6e4191f7b51fc5c232ea3d029f60bbd1e1172', label: "Mandatory Text Input", required: true })))));
    }
}, [0, "dga-text-input-usage", {
        "value": [32]
    }]);
function defineCustomElement$1() {
    if (typeof customElements === "undefined") {
        return;
    }
    const components = ["dga-text-input-usage", "dga-helper-text", "dga-icon", "dga-label", "dga-prefix", "dga-text-input"];
    components.forEach(tagName => { switch (tagName) {
        case "dga-text-input-usage":
            if (!customElements.get(tagName)) {
                customElements.define(tagName, DgaTextInputUsage$1);
            }
            break;
        case "dga-helper-text":
            if (!customElements.get(tagName)) {
                defineCustomElement$6();
            }
            break;
        case "dga-icon":
            if (!customElements.get(tagName)) {
                defineCustomElement$5();
            }
            break;
        case "dga-label":
            if (!customElements.get(tagName)) {
                defineCustomElement$4();
            }
            break;
        case "dga-prefix":
            if (!customElements.get(tagName)) {
                defineCustomElement$3();
            }
            break;
        case "dga-text-input":
            if (!customElements.get(tagName)) {
                defineCustomElement$2();
            }
            break;
    } });
}

const DgaTextInputUsage = DgaTextInputUsage$1;
const defineCustomElement = defineCustomElement$1;

export { DgaTextInputUsage, defineCustomElement };
