// @platformscode/core/dist/components/dga-data-table-usage.js
import { p as proxyCustomElement, H, h } from './p-JCGAt0My.js';
import { d as defineCustomElement$5 } from './p-BLDzt_Uq.js';
import { d as defineCustomElement$4 } from './p-C-62j229.js';
import { d as defineCustomElement$3 } from './p-CKrmM6in.js';
import { d as defineCustomElement$2 } from './p-CcNS0CYN.js';

const DgaDateFieldUsage = /*@__PURE__*/ proxyCustomElement(class DgaDateFieldUsage extends H {
    constructor() {
        super();
        this.__registerHost();
        this.value = '';
    }
    render() {
        return (h("div", { key: 'f5634d14aa0af69dfec196238e9771ea4387afaf' }, h("h1", { key: '10b89f738ada8b9dffa038131f1b922fff9a7d2d' }, "Date Field Example"), h("div", { key: '27d4efdf00c4af04bd276891bed9d79dfc09ee97', style: { padding: '50px' } }, h("div", { key: '3a8e86317ca962d6aff3e87c35b18c638b44e07b', style: { display: 'flex', flexDirection: 'column', gap: '16px' } }, h("dga-data-table", { key: '320dfe63275dc14537c762771e4ce56814a8378a', cells: [
                {
                    colSpan: 1,
                    isFilter: true,
                    isSort: true,
                    propertyName: 'city',
                    sort: 'up',
                    title: 'city',
                    type: 'propertyName',
                },
                {
                    colSpan: 1,
                    isFilter: true,
                    isSort: true,
                    propertyName: 'price',
                    sort: 'down',
                    title: 'price',
                    type: 'propertyName',
                },
                {
                    colSpan: 1,
                    isFilter: true,
                    isSort: true,
                    propertyName: 'state',
                    title: 'state',
                    type: 'element', // <-- change this
                    El: row => h("dga-button", { label: row.state }, console.log(row, 'roooooow')), // <-- use a functional component
                },
            ], contained: true, data: [
                {
                    city: 'Dallas',
                    h_id: '3',
                    price: '162500',
                    selected: false,
                    state: 'TX',
                    zip: '75201',
                },
                {
                    city: 'Bevery Hills',
                    h_id: '4',
                    price: '319250',
                    selected: false,
                    state: 'CA',
                    zip: '90210',
                },
                {
                    city: 'New York',
                    h_id: '5',
                    price: '962500',
                    selected: false,
                    state: 'NY',
                    zip: '00010',
                },
            ], getSelectedRows: function vu() { }, pressOnFilter: function vu() { }, rowDivider: true, showSelectCheckBox: true })))));
    }
}, [0, "dga-data-table-usage", {
        "value": [32]
    }]);
function defineCustomElement$1() {
    if (typeof customElements === "undefined") {
        return;
    }
    const components = ["dga-data-table-usage", "dga-button", "dga-checkbox", "dga-data-table", "dga-icon"];
    components.forEach(tagName => { switch (tagName) {
        case "dga-data-table-usage":
            if (!customElements.get(tagName)) {
                customElements.define(tagName, DgaDateFieldUsage);
            }
            break;
        case "dga-button":
            if (!customElements.get(tagName)) {
                defineCustomElement$5();
            }
            break;
        case "dga-checkbox":
            if (!customElements.get(tagName)) {
                defineCustomElement$4();
            }
            break;
        case "dga-data-table":
            if (!customElements.get(tagName)) {
                defineCustomElement$3();
            }
            break;
        case "dga-icon":
            if (!customElements.get(tagName)) {
                defineCustomElement$2();
            }
            break;
    } });
}

const DgaDataTableUsage = DgaDateFieldUsage;
const defineCustomElement = defineCustomElement$1;

export { DgaDataTableUsage, defineCustomElement };
