// @platformscode/core/dist/components/dga-inline-alert-usage.js
import { p as proxyCustomElement, H, h } from './p-JCGAt0My.js';
import { d as defineCustomElement$3 } from './p-CcNS0CYN.js';
import { d as defineCustomElement$2 } from './p-CgHnB4Ok.js';

const DgaInlineAlertUsage$1 = /*@__PURE__*/ proxyCustomElement(class DgaInlineAlertUsage extends H {
    constructor() {
        super();
        this.__registerHost();
        this.variant = 'neutral';
    }
    render() {
        return (h("div", { key: '8a0d2a49d15fbc55046a4049ae737b7d941210dd', dir: 'rlt' }, h("h1", { key: '853c65729de18bb7b3f168f84936619ed28fc74b' }, "Inline ALert Example"), h("div", { key: '967fa59b4017997f6ce42624a2b2d2de23646cfa', style: { padding: '50px', display: 'flex', gap: '20px' } }, h("div", { key: 'b53fb09470ebe141d1091f5b4ecc72f6ce84209e', style: { display: 'flex', flexDirection: 'column', gap: '20px' } }, h("dga-inline-alert", { key: '06a700bc70698258ec4e855835b1020f52ab59ca', type: "neutral", leadText: "Notification/Alert message title ", helperText: "When a nation, it goes here.", colored: true }), h("dga-inline-alert", { key: '53c419198d672ba4eec3d5d6a298dcde9e76dc02', type: "error", leadText: "Notification/Alert message title ", helperText: "When a Notification/Alert needs a further detailed explanation, it goes here.", colored: true }), h("dga-inline-alert", { key: 'c18b2e050daf4ad6f44f8e92c92d3d2e4012fae3', type: "info", leadText: "Notification/Alert message title ", helperText: "When a Notification/Alert needs a further detailed explanation, it goes here.", colored: true }), h("dga-inline-alert", { key: '0e68e4942078f1960559afc5fb296e3297578c15', type: "success", leadText: "Notification/Alert message title ", helperText: "When a Notification/Alert needs a further detailed explanation, it goes here.", colored: true }), h("dga-inline-alert", { key: '2d4158619b859a0e270918f9713ccb7dac9aef67', type: "warning", leadText: "Notification/Alert message title ", helperText: "When a Notification/Alert needs a further detailed explanation, it goes here.", colored: true })), h("div", { key: 'a1caca30371a1393c7c33df8fc912564380901e6', style: { display: 'flex', flexDirection: 'column', gap: '20px' } }))));
    }
}, [0, "dga-inline-alert-usage", {
        "variant": [32]
    }]);
function defineCustomElement$1() {
    if (typeof customElements === "undefined") {
        return;
    }
    const components = ["dga-inline-alert-usage", "dga-icon", "dga-inline-alert"];
    components.forEach(tagName => { switch (tagName) {
        case "dga-inline-alert-usage":
            if (!customElements.get(tagName)) {
                customElements.define(tagName, DgaInlineAlertUsage$1);
            }
            break;
        case "dga-icon":
            if (!customElements.get(tagName)) {
                defineCustomElement$3();
            }
            break;
        case "dga-inline-alert":
            if (!customElements.get(tagName)) {
                defineCustomElement$2();
            }
            break;
    } });
}

const DgaInlineAlertUsage = DgaInlineAlertUsage$1;
const defineCustomElement = defineCustomElement$1;

export { DgaInlineAlertUsage, defineCustomElement };
