// @platformscode/core/dist/components/dga-structured-list-usage.js
import { p as proxyCustomElement, H, h } from './p-JCGAt0My.js';
import { d as defineCustomElement$2 } from './p-BFy9BbBP.js';

const DgaStructuredListUsage$1 = /*@__PURE__*/ proxyCustomElement(class DgaStructuredListUsage extends H {
    constructor() {
        super();
        this.__registerHost();
    }
    render() {
        return (h("div", { key: '8af8dd832806e8928a197fa1988d1d86d1e30011' }, h("h1", { key: 'caa4445333d278a053ab424bd89ba765a82da8d4' }, "Structured List Example"), h("div", { key: '6089dad32ecf56764f7e23c926066fcd94ad49e8', style: { padding: "50px" } }, h("dga-structured-list", { key: '90b31d161cae14b9f478d5235975e82b06b33fbd', cells: [
                {
                    title: "ID",
                    propertyName: "id",
                    type: "propertyName",
                },
                {
                    title: "Name",
                    propertyName: "name",
                    type: "propertyName",
                },
            ], data: [
                { id: 1, name: "Alice", selected: false },
                { id: 2, name: "Bob", selected: false },
                { id: 3, name: "Charlie", selected: false },
            ], rowDivider: true, colDivider: false, selectable: true, getSelectedRows: selectedRows => console.log(selectedRows) }), h("dga-structured-list", { key: '20d63169ca065c07a6128e680d5bc988d09409a8', compact: true, cells: [
                {
                    title: "ID",
                    propertyName: "id",
                    type: "propertyName",
                },
                {
                    title: "Name",
                    propertyName: "name",
                    type: "propertyName",
                },
            ], data: [
                { id: 1, name: "Alice", selected: false },
                { id: 2, name: "Bob", selected: false },
                { id: 3, name: "Charlie", selected: false },
            ], rowDivider: true, selectable: true, getSelectedRows: selectedRows => console.log(selectedRows) }))));
    }
}, [0, "dga-structured-list-usage"]);
function defineCustomElement$1() {
    if (typeof customElements === "undefined") {
        return;
    }
    const components = ["dga-structured-list-usage", "dga-structured-list"];
    components.forEach(tagName => { switch (tagName) {
        case "dga-structured-list-usage":
            if (!customElements.get(tagName)) {
                customElements.define(tagName, DgaStructuredListUsage$1);
            }
            break;
        case "dga-structured-list":
            if (!customElements.get(tagName)) {
                defineCustomElement$2();
            }
            break;
    } });
}

const DgaStructuredListUsage = DgaStructuredListUsage$1;
const defineCustomElement = defineCustomElement$1;

export { DgaStructuredListUsage, defineCustomElement };
