// @platformscode/core/dist/components/dga-accordion-usage.js
import { p as proxyCustomElement, H, h } from './p-JCGAt0My.js';
import { d as defineCustomElement$3 } from './p-Cp4XrPXy.js';
import { d as defineCustomElement$2 } from './p-CcNS0CYN.js';

const DgaAccordionUsage$1 = /*@__PURE__*/ proxyCustomElement(class DgaAccordionUsage extends H {
    constructor() {
        super();
        this.__registerHost();
    }
    render() {
        return (h("div", { key: '83a34f37d8845bf89576abbea8c91e5be5c7132c' }, h("h1", { key: '823461073067bbeca1049e0d0661f87381943e29' }, "Accordion Example"), h("div", { key: '618ffbc9d482d6e1c7ef2cb29a473e2d93b252cd', style: { padding: '50px', display: 'flex', gap: '20px' } }, h("div", { key: '07d3d0c0e86b6ec73ba4f9f674c116d4727d85eb', style: { display: 'flex', flexDirection: 'column', gap: '20px' } }, h("dga-accordion", { key: '270e32b26ed0a335378dc2736b41b754426170a3', title: "Accordion Title", size: "lg", content: "The accordion component delivers large amounts of content in a small space through progressive disclosure. The user gets key details about the underlying content and can choose to expand that content within the constraints of the accordion." }), h("dga-accordion", { key: '28e37fc984132115feafa2aadb763a16012108f2', title: "Accordion Title", size: "md" }), h("dga-accordion", { key: 'e2de2ce853f0e3440868b8781af12a9b7fafa2cc', title: "Accordion Title", size: "sm" }), h("dga-accordion", { key: 'e29acf09c9035e4864de6bcb3c69e291bfd908f6', defaultExpanded: true, title: "Accordion Title", size: "lg", content: "The accordion component delivers large amounts of content in a small space through progressive disclosure. The user gets key details about the underlying content and can choose to expand that content within the constraints of the accordion.", disabled: true })), h("div", { key: '79c14e4237380037592ee42d4c20da0739f18938', style: { display: 'flex', flexDirection: 'column', gap: '20px' } }))));
    }
}, [0, "dga-accordion-usage"]);
function defineCustomElement$1() {
    if (typeof customElements === "undefined") {
        return;
    }
    const components = ["dga-accordion-usage", "dga-accordion", "dga-icon"];
    components.forEach(tagName => { switch (tagName) {
        case "dga-accordion-usage":
            if (!customElements.get(tagName)) {
                customElements.define(tagName, DgaAccordionUsage$1);
            }
            break;
        case "dga-accordion":
            if (!customElements.get(tagName)) {
                defineCustomElement$3();
            }
            break;
        case "dga-icon":
            if (!customElements.get(tagName)) {
                defineCustomElement$2();
            }
            break;
    } });
}

const DgaAccordionUsage = DgaAccordionUsage$1;
const defineCustomElement = defineCustomElement$1;

export { DgaAccordionUsage, defineCustomElement };
