// @platformscode/core/dist/components/dga-quote-usage.js
import { p as proxyCustomElement, H, h } from './p-JCGAt0My.js';
import { d as defineCustomElement$5 } from './p-DmiaKdLo.js';
import { d as defineCustomElement$4 } from './p-Cg2ruBZP.js';
import { d as defineCustomElement$3 } from './p-CcNS0CYN.js';
import { d as defineCustomElement$2 } from './p-BbTFdVcG.js';

const DgaQuoteUsage$1 = /*@__PURE__*/ proxyCustomElement(class DgaQuoteUsage extends H {
    constructor() {
        super();
        this.__registerHost();
        this.activeStep = 1;
    }
    render() {
        return (h("div", { key: '7b59d5cd7223387ec9ecaf09ae430cd5fa031665' }, h("h1", { key: 'c559eb076b7986f94c1bdbe8813692219cd85206' }, "Quote Example"), h("div", { key: 'fdd12b17fbd6083a322bf11b6b7d591876711099', style: { padding: "50px", display: "flex", gap: "20px" } }, h("div", { key: '33fe8a38ca236161d916a9e334586a752f10e2e6', style: { display: "flex", flexDirection: "column", gap: "20px" } }, h("dga-quote", { key: 'a8997e3d3f0bca7f9529bac03e0e8ed90d682234', quoteTitle: 'Title of quote', quoteDescription: 'The quote is placed here to highlight a specific saying or to present a brief quote that expresses an important idea or concept.', authorName: "Author's name", authorDescription: 'brief or description.' }), h("dga-quote", { key: '9a27f63db1512dc033c546b9e2a5cc86e25e1fc6', quoteTitle: 'Title of quote', quoteDescription: 'The quote is placed here to highlight a specific saying or to present a brief quote that expresses an important idea or concept.', authorName: "Author's name", authorDescription: 'brief or description.', whiteBackground: true })), h("div", { key: '7b7fb07e6b082f02a58d87cdebaf03660d5cbc45', style: { display: "flex", flexDirection: "column", gap: "20px" } }))));
    }
}, [0, "dga-quote-usage", {
        "activeStep": [32]
    }]);
function defineCustomElement$1() {
    if (typeof customElements === "undefined") {
        return;
    }
    const components = ["dga-quote-usage", "dga-avatar", "dga-flex", "dga-icon", "dga-quote"];
    components.forEach(tagName => { switch (tagName) {
        case "dga-quote-usage":
            if (!customElements.get(tagName)) {
                customElements.define(tagName, DgaQuoteUsage$1);
            }
            break;
        case "dga-avatar":
            if (!customElements.get(tagName)) {
                defineCustomElement$5();
            }
            break;
        case "dga-flex":
            if (!customElements.get(tagName)) {
                defineCustomElement$4();
            }
            break;
        case "dga-icon":
            if (!customElements.get(tagName)) {
                defineCustomElement$3();
            }
            break;
        case "dga-quote":
            if (!customElements.get(tagName)) {
                defineCustomElement$2();
            }
            break;
    } });
}

const DgaQuoteUsage = DgaQuoteUsage$1;
const defineCustomElement = defineCustomElement$1;

export { DgaQuoteUsage, defineCustomElement };
