// @platformscode/core/dist/components/dga-notification-usage.js
import { p as proxyCustomElement, H, h } from './p-JCGAt0My.js';
import { d as defineCustomElement$3 } from './p-CcNS0CYN.js';
import { d as defineCustomElement$2 } from './p-C0pZ4KiI.js';

const DgaNotificationUsage$1 = /*@__PURE__*/ proxyCustomElement(class DgaNotificationUsage extends H {
    constructor() {
        super();
        this.__registerHost();
        this.variant = 'neutral';
    }
    render() {
        return (h("div", { key: '061b9fa818dbfecccfa83abd752c050f8bb789c7' }, h("h1", { key: 'a61b841d5bae620fb5d3824637b506bf986688d0' }, "Notification Example"), h("div", { key: 'fc2c00772e8dab3dd1a1f27536126f5c4e472590', style: { padding: '50px', display: 'flex', gap: '20px' } }, h("div", { key: '0525f6745e9b7a7090d32c54d7ff2b7386587d78', style: { display: 'flex', flexDirection: 'column', gap: '20px' } }, h("dga-notification", { key: 'aab7266ac630410ec5a41a091c7fbe25e17085b2', leadText: "This is a very important banner message that requires attention", variant: this.variant, dismissable: true }), h("dga-notification", { key: 'b9b666915fbceac5e9f6208867ba8b570fe10007', leadText: "This is a very important banner message that requires attention", variant: "critical", dismissable: true }), h("dga-notification", { key: '1d8f6208d39bf9a389f14a18c5cc27e927de9ae5', leadText: "This is a very important banner message that requires attention", variant: "info", dismissable: true }), h("dga-notification", { key: 'e3aac38df8a972c5afc470e4bb19c4e7ec3bceaf', leadText: "This is a very important banner message that requires attention", variant: "success", dismissable: true }), h("dga-notification", { key: '9e2cd75f1a7d661da809a22de64733f4d2a98e5f', leadText: "This is a very important banner message that requires attention", variant: "warning", dismissable: true })), h("div", { key: 'a9b26f80c1ca89ab081b077c646884eec6a7023a', style: { display: 'flex', flexDirection: 'column', gap: '20px' } }))));
    }
}, [0, "dga-notification-usage", {
        "variant": [32]
    }]);
function defineCustomElement$1() {
    if (typeof customElements === "undefined") {
        return;
    }
    const components = ["dga-notification-usage", "dga-icon", "dga-notification"];
    components.forEach(tagName => { switch (tagName) {
        case "dga-notification-usage":
            if (!customElements.get(tagName)) {
                customElements.define(tagName, DgaNotificationUsage$1);
            }
            break;
        case "dga-icon":
            if (!customElements.get(tagName)) {
                defineCustomElement$3();
            }
            break;
        case "dga-notification":
            if (!customElements.get(tagName)) {
                defineCustomElement$2();
            }
            break;
    } });
}

const DgaNotificationUsage = DgaNotificationUsage$1;
const defineCustomElement = defineCustomElement$1;

export { DgaNotificationUsage, defineCustomElement };
