// @platformscode/core/dist/components/dga-skeleton-usage.js
import { p as proxyCustomElement, H, h } from './p-JCGAt0My.js';
import { d as defineCustomElement$6 } from './p-CINdaz9C.js';
import { d as defineCustomElement$5 } from './p-dbmT_7aU.js';
import { d as defineCustomElement$4 } from './p-BKOTAIoM.js';
import { d as defineCustomElement$3 } from './p-BqQQvQYh.js';
import { d as defineCustomElement$2 } from './p-CfwGl7Wl.js';

const LineSkeleton = /*@__PURE__*/ proxyCustomElement(class LineSkeleton extends H {
    constructor() {
        super();
        this.__registerHost();
    }
    render() {
        return (h("div", { key: '13689d7473e63220e19a6c8e30d18877db38092a' }, h("dga-line-skeleton", { key: '4374e1af0a74c1ae1fbae198df31d5a51df9c81d' }), h("dga-circle-skeleton", { key: '7ce5c2365e10110f807ccaaf843038a63ec44b0d' }), h("dga-rectangle-skeleton", { key: 'dfe62ba6ecd9e955b024faaeef26028dbdbbe0ec' }), h("dga-square-skeleton", { key: '5ab742d263c93c03ededdf58bd957ae5dea84d75' }), h("dga-skeleton-component", { key: '16b4fcc9bec8047a414938bd1c95f721bb4ad8b7' })));
    }
}, [0, "dga-skeleton-usage"]);
function defineCustomElement$1() {
    if (typeof customElements === "undefined") {
        return;
    }
    const components = ["dga-skeleton-usage", "dga-circle-skeleton", "dga-line-skeleton", "dga-rectangle-skeleton", "dga-skeleton-component", "dga-square-skeleton"];
    components.forEach(tagName => { switch (tagName) {
        case "dga-skeleton-usage":
            if (!customElements.get(tagName)) {
                customElements.define(tagName, LineSkeleton);
            }
            break;
        case "dga-circle-skeleton":
            if (!customElements.get(tagName)) {
                defineCustomElement$6();
            }
            break;
        case "dga-line-skeleton":
            if (!customElements.get(tagName)) {
                defineCustomElement$5();
            }
            break;
        case "dga-rectangle-skeleton":
            if (!customElements.get(tagName)) {
                defineCustomElement$4();
            }
            break;
        case "dga-skeleton-component":
            if (!customElements.get(tagName)) {
                defineCustomElement$3();
            }
            break;
        case "dga-square-skeleton":
            if (!customElements.get(tagName)) {
                defineCustomElement$2();
            }
            break;
    } });
}

const DgaSkeletonUsage = LineSkeleton;
const defineCustomElement = defineCustomElement$1;

export { DgaSkeletonUsage, defineCustomElement };
