// @platformscode/core/dist/components/dga-slider-usage.js
import { p as proxyCustomElement, H, h, F as Fragment } from './p-JCGAt0My.js';
import { d as defineCustomElement$2 } from './p-MftkTgWc.js';

const DgaSliderUsage$1 = /*@__PURE__*/ proxyCustomElement(class DgaSliderUsage extends H {
    constructor() {
        super();
        this.__registerHost();
        this.__attachShadow();
        this.activeStep = 1;
        this.minValue = 10;
        this.maxValue = 100;
    }
    render() {
        return (h(Fragment, { key: '2dd0277992f2ccc09b234b39015acc9d3505d07d' }, h("br", { key: '2f8014793fbb6f09227dd05677935a595d18a401' }), h("dga-slider", { key: '786da1950c5ca8b4e38297c4054eef013b3c91a7', min: this.minValue, max: this.maxValue, step: 1, range: true, size: "medium", helperText: "Helper Text Example", label: "Test", sx: {
                backgroundColor: "yellow",
                marginTop: "200px",
                ".range-slider-inner-rail": {
                    backgroundColor: "red"
                }
            } }), ";", h("br", { key: '1a3ddb903fa675ee0cab4114b608fa2ac8d26488' }), this.minValue, " - ", this.maxValue, h("dga-slider", { key: 'eb9501d6b38fd40671fb138ccf8cfeb4b69a8ddb' })));
    }
}, [1, "dga-slider-usage", {
        "activeStep": [32],
        "minValue": [32],
        "maxValue": [32]
    }]);
function defineCustomElement$1() {
    if (typeof customElements === "undefined") {
        return;
    }
    const components = ["dga-slider-usage", "dga-slider"];
    components.forEach(tagName => { switch (tagName) {
        case "dga-slider-usage":
            if (!customElements.get(tagName)) {
                customElements.define(tagName, DgaSliderUsage$1);
            }
            break;
        case "dga-slider":
            if (!customElements.get(tagName)) {
                defineCustomElement$2();
            }
            break;
    } });
}

const DgaSliderUsage = DgaSliderUsage$1;
const defineCustomElement = defineCustomElement$1;

export { DgaSliderUsage, defineCustomElement };
