// @platformscode/core/dist/components/dga-loading-usage.js
import { p as proxyCustomElement, H, h } from './p-JCGAt0My.js';
import { d as defineCustomElement$2 } from './p-FjQbvIG7.js';

const DgaLoadingUsage$1 = /*@__PURE__*/ proxyCustomElement(class DgaLoadingUsage extends H {
    constructor() {
        super();
        this.__registerHost();
        this.activeStep = 1;
    }
    render() {
        return (h("div", { key: 'bf58955c9fa97077b7e15aea084429b2f711d4f9' }, h("h1", { key: 'eaf082fa93ac33b2e10a6667a09b6a50015edd14' }, "Loading Example"), h("div", { key: '73586bff09c11cf8f921840408bcc75a93afb3a5', style: { padding: "50px", display: "flex", gap: "20px" } }, h("div", { key: 'f4d9cc537d1deede62091c6b1a6c85598d8fcd18', style: { display: "flex", flexDirection: "column", gap: "20px" } }, h("dga-loading", { key: '013fa149d371c83070897c04ffdbd120f0ecf4eb' })), h("div", { key: '769c7a6ab524a4469f98615e9cdaaa01c4c3acf3', style: { display: "flex", flexDirection: "column", gap: "20px" } }))));
    }
}, [0, "dga-loading-usage", {
        "activeStep": [32]
    }]);
function defineCustomElement$1() {
    if (typeof customElements === "undefined") {
        return;
    }
    const components = ["dga-loading-usage", "dga-loading"];
    components.forEach(tagName => { switch (tagName) {
        case "dga-loading-usage":
            if (!customElements.get(tagName)) {
                customElements.define(tagName, DgaLoadingUsage$1);
            }
            break;
        case "dga-loading":
            if (!customElements.get(tagName)) {
                defineCustomElement$2();
            }
            break;
    } });
}

const DgaLoadingUsage = DgaLoadingUsage$1;
const defineCustomElement = defineCustomElement$1;

export { DgaLoadingUsage, defineCustomElement };
