// @platformscode/core/dist/components/nds-button-usage.js
import { p as proxyCustomElement, H, h } from './p-JCGAt0My.js';
import { d as defineCustomElement$6 } from './p-CfQ8F8Bh.js';
import { d as defineCustomElement$5 } from './p-Bs_sPiZE.js';
import { d as defineCustomElement$4 } from './p-Cg2ruBZP.js';
import { d as defineCustomElement$3 } from './p-CcNS0CYN.js';
import { d as defineCustomElement$2 } from './p-Dd1XVYJF.js';

const NdsButtonUsage$1 = /*@__PURE__*/ proxyCustomElement(class NdsButtonUsage extends H {
    constructor() {
        super();
        this.__registerHost();
        this.variant = 'neutral';
    }
    render() {
        return (h("div", { key: '93e8a534cdf0510938fddc6c71b50f9f187e6152' }, h("h1", { key: 'f685f95bf42630668e6bb87a4028fc33454a5b06' }, "Button Example"), h("div", { key: '11541348742b59c5a9f8a5a5bed82ed2ad061396', style: { padding: '50px', display: 'flex', gap: '20px' } }, h("div", { key: '3fc6da5081ee18fdde08258dc9f45bc71c28fbf9', style: { display: 'flex', flexDirection: 'column', gap: '20px' } }, h("dga-flex", { key: '89dd10ed91dcf4fd1ede6f24a04aa509056f104c', direction: "column", spacing: 16 }, h("dga-button-v2", { key: '2d47c266e8ad231d663d87e4227683708954e559', label: "Button", variant: "primary", size: "lg" }), h("dga-button-v2", { key: 'f3baec3e2915dbf0def6d0077875efd3c1d3b2f3', label: "Button", variant: "primary", size: "md" }), h("dga-button-v2", { key: '6048983c1b1957a2fd7a0547cbb78a4440780442', label: "Button", variant: "primary", size: "sm" }), h("dga-button-v2", { key: '4706e798f37607b7415d3fe060a4de13a1c29780', label: "Button", variant: "neutral", size: "lg" }), h("dga-button-v2", { key: 'd509b14e82495c156118cae3e94192ece9cba6ba', label: "Button", variant: "neutral", size: "md" }), h("dga-button-v2", { key: '97f3e4a7d2b68040ba71631bde80ce24030ced5d', label: "Button", variant: "neutral", size: "sm" }), h("dga-button-v2", { key: '90cc0c839970d1eb57aa7ab03b8ef549ae8c87d9', label: "Button", variant: "secondary-solid", size: "lg" }), h("dga-button-v2", { key: 'f6dac77cea26b99aa9c3e529580f2b84e4ea3339', label: "Button", variant: "secondary-solid", size: "md" }), h("dga-button-v2", { key: '9dd7afd56cb8bf3dcbd1eb410930453616d5e17a', label: "Button", variant: "secondary-solid", size: "sm" }), h("dga-button-v2", { key: '1ad41574bbb4620a5e89d42ea910a5e76023039e', label: "Button", variant: "primary", destructive: true, size: "lg" }), h("dga-button-v2", { key: 'e84b533c0c642c43e5a96c4375d815c5c0c579f8', label: "Button", variant: "primary", destructive: true, size: "md" }), h("dga-button-v2", { key: '897cf9772232073d72f3bf968a75ba25655050a0', label: "Button", variant: "primary", destructive: true, size: "sm" })), h("dga-button-v2", { key: 'bfe0dc83e184ee3a96094ae9df700233945a467f', label: "Button", variant: "primary", onColor: true }, h("dga-button-leading-icon", { key: '50081c16bbe5f639683fa5e61bf30d7565834bda', name: "link-square-01", size: 16, color: "red" })), h("dga-button-v2", { key: '4d80da0b9f949940798328a80173b16027712dd8', label: "Source code", variant: "secondary-outline", leadIcon: true, leadIconProps: { name: 'source-code', variant: 'stroke', type: 'standard' } }), h("dga-button-v2", { key: '7dbeee7bc5832202920fd13f68fb9abb215961e7',
            // className="absolute top-[50%] left-[50%] transform -translate-x-1/2 -translate-y-1/2 opacity-0 group-hover:opacity-100"
            label: "See live preview", size: "lg", variant: "secondary-outline", trailIcon: true, trailIconProps: {
                name: 'link-square-01',
                variant: 'stroke',
                type: 'standard',
            } })), h("div", { key: 'dce63e3e728fe8c771efe21b7b2266e2179b34c0', style: { display: 'flex', flexDirection: 'column', gap: '20px', backgroundColor: '#074D31', padding: '20px' } }, h("nds-button", { key: '75b7a2ffbaaa620cef29fd4a715d2bfce1a2a62d', label: "Button", variant: "primary-brand", size: "lg" }), h("nds-button", { key: '6a365df713c9306e537e905b2ac571d8c5066764', label: "Button", variant: "primary-brand", size: "lg", selected: true }), h("nds-button", { key: '8936024c769ef277d82d6f1ec7d79cd086b42dc2', leadingIcon: 'link-square-01', size: 'sm' }, h("dga-icon-v2", { key: '177803e3a86a468faa282a8593bbe289f5b2ea1e', slot: 'leading-icon', name: 'source-code', variant: 'stroke', type: 'rounded', size: 16, color: 'white' }))), h("dga-icon-v2", { key: '038d94fd691a71142218f1d3708621a29fe40c15', name: 'link-square-01', variant: 'stroke', type: 'rounded', color: 'red' }), h("div", { key: '0e36f957af0065debbdcd737c56c77393b42f57b', style: { display: 'flex', flexDirection: 'column', gap: '20px' } }))));
    }
}, [0, "nds-button-usage", {
        "variant": [32]
    }]);
function defineCustomElement$1() {
    if (typeof customElements === "undefined") {
        return;
    }
    const components = ["nds-button-usage", "dga-button-leading-icon", "dga-button-v2", "dga-flex", "dga-icon", "nds-button"];
    components.forEach(tagName => { switch (tagName) {
        case "nds-button-usage":
            if (!customElements.get(tagName)) {
                customElements.define(tagName, NdsButtonUsage$1);
            }
            break;
        case "dga-button-leading-icon":
            if (!customElements.get(tagName)) {
                defineCustomElement$6();
            }
            break;
        case "dga-button-v2":
            if (!customElements.get(tagName)) {
                defineCustomElement$5();
            }
            break;
        case "dga-flex":
            if (!customElements.get(tagName)) {
                defineCustomElement$4();
            }
            break;
        case "dga-icon":
            if (!customElements.get(tagName)) {
                defineCustomElement$3();
            }
            break;
        case "nds-button":
            if (!customElements.get(tagName)) {
                defineCustomElement$2();
            }
            break;
    } });
}

const NdsButtonUsage = NdsButtonUsage$1;
const defineCustomElement = defineCustomElement$1;

export { NdsButtonUsage, defineCustomElement };
