// @platformscode/core/dist/components/nds-accordion-usage.js
import { p as proxyCustomElement, H, h } from './p-JCGAt0My.js';
import { d as defineCustomElement$2 } from './p-SzLnmchK.js';

const NdsAccordionUsage$1 = /*@__PURE__*/ proxyCustomElement(class NdsAccordionUsage extends H {
    constructor() {
        super();
        this.__registerHost();
    }
    render() {
        return (h("div", { key: '9123890eda0c2fc65da22ce4b5f127d0dc4a121c' }, h("h1", { key: '8895f7600d63dcad69de62f266223218e4ff7524' }, "Accordion Example"), h("div", { key: '7c0735f21b91f2dae999defec1eed1286674f4e2', style: { padding: '50px', display: 'flex', gap: '20px' } }, h("div", { key: 'c12c81a421330ebb1742196374dc69ae69159e43', style: { display: 'flex', flexDirection: 'column', gap: '20px' } }, h("nds-accordion", { key: '7cbb4adc70d15301b4f1918bde2aadaf04769236', title: "Accordion Title", size: "lg", content: "The accordion component delivers large amounts of content in a small space through progressive disclosure. The user gets key details about the underlying content and can choose to expand that content within the constraints of the accordion." }), h("nds-accordion", { key: '42c0003deaba848fd3a5c1d3d78b76f605d6845c', title: "Accordion Title", size: "lg", content: "The accordion component delivers large amounts of content in a small space through progressive disclosure. The user gets key details about the underlying content and can choose to expand that content within the constraints of the accordion." }, h("div", { key: '68663eec1c2fb257a57f2801beac7fb35c2a6b0c' }, h("p", { key: 'dd631be5a344ba7e9b508b13c23a6f844850e3e9' }, "This is a ", h("span", { key: '57ac48e29944525e5e1bc12ed3cd7d169ae47214', style: { color: "red" } }, "custom"), " content inside the accordion."), h("p", { key: '0172a5f691529db9ab417917c3d4ece70bfc1937' }, "You can put any React nodes here.")))))));
    }
}, [0, "nds-accordion-usage"]);
function defineCustomElement$1() {
    if (typeof customElements === "undefined") {
        return;
    }
    const components = ["nds-accordion-usage", "nds-accordion"];
    components.forEach(tagName => { switch (tagName) {
        case "nds-accordion-usage":
            if (!customElements.get(tagName)) {
                customElements.define(tagName, NdsAccordionUsage$1);
            }
            break;
        case "nds-accordion":
            if (!customElements.get(tagName)) {
                defineCustomElement$2();
            }
            break;
    } });
}

const NdsAccordionUsage = NdsAccordionUsage$1;
const defineCustomElement = defineCustomElement$1;

export { NdsAccordionUsage, defineCustomElement };
