// @platformscode/core/dist/components/p-awnvl4MX.js
import { p as proxyCustomElement, H, h } from './p-JCGAt0My.js';
import { d as defineCustomElement$1 } from './p-CcNS0CYN.js';

const DgaHelperText = /*@__PURE__*/ proxyCustomElement(class DgaHelperText extends H {
    constructor() {
        super();
        this.__registerHost();
        this.helperText = '';
        this.error = false;
    }
    render() {
        var _a;
        return (h("div", { key: 'f2a73d39268684cec9fba2210c59ebc7f5c5741c', class: `dga-helper-text ${this.error ? "dga-helper-text--error" : ""}` }, h("span", { key: '80f3bfa13dc22335d4fc636a069acbea322d3b3e', class: "dga-helper-text__icon" }, this.helperTextIcon && ((_a = this.helperTextIcon) === null || _a === void 0 ? void 0 : _a.name) && h("dga-icon", { key: '2bd2fe44c408bff2ffc26a71712b9717f94b6579', name: this.helperTextIcon.name, variant: this.helperTextIcon.variant, type: this.helperTextIcon.type, size: 16 })), h("span", { key: 'e886b1b9c9a018aa623df4920d53272615187e66', class: "dga-helper-text__desc" }, this.helperText)));
    }
}, [0, "dga-helper-text", {
        "helperTextIcon": [16, "helper-text-icon"],
        "helperText": [1, "helper-text"],
        "error": [4]
    }]);
function defineCustomElement() {
    if (typeof customElements === "undefined") {
        return;
    }
    const components = ["dga-helper-text", "dga-icon"];
    components.forEach(tagName => { switch (tagName) {
        case "dga-helper-text":
            if (!customElements.get(tagName)) {
                customElements.define(tagName, DgaHelperText);
            }
            break;
        case "dga-icon":
            if (!customElements.get(tagName)) {
                defineCustomElement$1();
            }
            break;
    } });
}

export { DgaHelperText as D, defineCustomElement as d };
