// @platformscode/core/dist/components/p-B0P1tNYn.js
import { p as proxyCustomElement, H, h, d as Host } from './p-JCGAt0My.js';

const DgaSecondNavHeaderItem = /*@__PURE__*/ proxyCustomElement(class DgaSecondNavHeaderItem extends H {
    constructor() {
        super();
        this.__registerHost();
    }
    connectedCallback() {
        this.el.childNodes.forEach((item) => {
            const el = item;
            if (el.tagName === 'DGA-ICON'.toLocaleUpperCase() && el.attributes.getNamedItem('slot') === null) {
                el.setAttribute('slot', 'icon');
            }
        });
    }
    render() {
        return (h(Host, { key: '229c796621d1842e6484a2857130228a6064aae2' }, h("slot", { key: '9b10838752cd3103a736a4bf1931ec58e142ec36', name: 'icon' }), h("slot", { key: '70317608a423de1002f8b762096309aed55b2311', name: 'label' }, this.label)));
    }
    get el() { return this; }
}, [4, "dga-second-nav-header-item", {
        "label": [1]
    }]);
function defineCustomElement() {
    if (typeof customElements === "undefined") {
        return;
    }
    const components = ["dga-second-nav-header-item"];
    components.forEach(tagName => { switch (tagName) {
        case "dga-second-nav-header-item":
            if (!customElements.get(tagName)) {
                customElements.define(tagName, DgaSecondNavHeaderItem);
            }
            break;
    } });
}

export { DgaSecondNavHeaderItem as D, defineCustomElement as d };
