// @platformscode/core/dist/components/p-B8Ehyo7m.js
import { p as proxyCustomElement, H, h } from './p-JCGAt0My.js';

const DgaLabel = /*@__PURE__*/ proxyCustomElement(class DgaLabel extends H {
    constructor() {
        super();
        this.__registerHost();
        this.label = '';
        this.variant = 'default';
        this.size = 'md';
        this.disabled = false;
        this.required = false;
    }
    render() {
        return (h("label", { key: '972ee04f32e79084d35fd5c47f922ca01019c1ec', class: `dga-label dga-label--${this.size} ${this.disabled ? "disabled" : ""}` }, this.required && h("span", { key: '07cb58fcbc3eda4d621bda3a5263d87dc5a1583b', class: "star" }, "*"), this.label));
    }
}, [0, "dga-label", {
        "label": [1],
        "variant": [1],
        "size": [1],
        "disabled": [4],
        "required": [4]
    }]);
function defineCustomElement() {
    if (typeof customElements === "undefined") {
        return;
    }
    const components = ["dga-label"];
    components.forEach(tagName => { switch (tagName) {
        case "dga-label":
            if (!customElements.get(tagName)) {
                customElements.define(tagName, DgaLabel);
            }
            break;
    } });
}

export { DgaLabel as D, defineCustomElement as d };
