// @platformscode/core/dist/components/p-CfQ8F8Bh.js
import { p as proxyCustomElement, H, h } from './p-JCGAt0My.js';
import { d as defineCustomElement$1 } from './p-CcNS0CYN.js';

const DgaButtonLeadingIcon = /*@__PURE__*/ proxyCustomElement(class DgaButtonLeadingIcon extends H {
    constructor() {
        super();
        this.__registerHost();
        this.variant = 'bulk';
        this.type = 'rounded';
    }
    render() {
        // Check if there are any child elements
        return (h("span", { key: 'fceaa3f8e2739d7468bfd21e8efbacd8b32ad0f5', class: "dga-btn-icon dga-btn-icon--leading", "aria-hidden": "true", slot: "leading-icon" }, h("dga-icon", { key: '619ef6a0d019fb39f57a1381fc2f4008374e9bc1', name: this.name, size: this.size, color: this.color, variant: this.variant, type: this.type })));
    }
    get el() { return this; }
}, [0, "dga-button-leading-icon", {
        "name": [1537],
        "size": [2],
        "color": [1],
        "variant": [1],
        "type": [1]
    }]);
function defineCustomElement() {
    if (typeof customElements === "undefined") {
        return;
    }
    const components = ["dga-button-leading-icon", "dga-icon"];
    components.forEach(tagName => { switch (tagName) {
        case "dga-button-leading-icon":
            if (!customElements.get(tagName)) {
                customElements.define(tagName, DgaButtonLeadingIcon);
            }
            break;
        case "dga-icon":
            if (!customElements.get(tagName)) {
                defineCustomElement$1();
            }
            break;
    } });
}

export { DgaButtonLeadingIcon as D, defineCustomElement as d };
