// @platformscode/core/dist/components/p-ChOKtheJ.js
import { p as proxyCustomElement, H, h, d as Host } from './p-JCGAt0My.js';

const DgaSecondNavHeader = /*@__PURE__*/ proxyCustomElement(class DgaSecondNavHeader extends H {
    constructor() {
        super();
        this.__registerHost();
        this.variant = 'primary';
    }
    componentWillLoad() {
        this.el.childNodes.forEach((item) => {
            const el = item;
            if (el.tagName === 'DGA-SECOND-NAV-HEADER-CONTENT' && this.hideContent === true) {
                el.style.display = 'none';
            }
            if (el.tagName === 'DGA-SECOND-NAV-HEADER-ACTIONS' && this.hideActions === true) {
                el.style.display = 'none';
            }
            if (el.tagName === 'DGA-SECOND-NAV-HEADER-CONTENT' && el.attributes.getNamedItem('slot') === null) {
                el.setAttribute('slot', 'content');
            }
            if (el.tagName === 'DGA-SECOND-NAV-HEADER-ACTIONS' && el.attributes.getNamedItem('slot') === null) {
                el.setAttribute('slot', 'actions');
            }
        });
    }
    render() {
        return (h(Host, { key: '2148ff638962c3774f6296bb153ab227b2037242' }, h("div", { key: '1876a750095870936c30c938417dbf8f2edc2e7c', class: `wrapper ${this.variant} `, "aria-label": 'a secondary navigation navbar' }, h("div", { key: '39f09c68a16d833e43610913fb903bcafd8b1f9f', class: `container ${this.hideDivider ? '' : 'border'}` }, h("div", { key: '4185be99ad09528b5ab48dfb2f9f9683c80dcc62', class: `content` }, h("slot", { key: '9c880c19c64d3b5a13fd531e1b4e8f6d88ebf13b', name: "content" })), h("div", { key: '9365fa894dc11cbfbccd783834481fb86fd3d0e6', class: `actions` }, h("slot", { key: '252fa64837f5c111ff0954c921c906aa9bba1053', name: "actions" }))))));
    }
    get el() { return this; }
}, [6, "dga-second-nav-header", {
        "variant": [1],
        "hideDivider": [1028, "hide-divider"],
        "hideActions": [4, "hide-actions"],
        "hideContent": [4, "hide-content"]
    }]);
function defineCustomElement() {
    if (typeof customElements === "undefined") {
        return;
    }
    const components = ["dga-second-nav-header"];
    components.forEach(tagName => { switch (tagName) {
        case "dga-second-nav-header":
            if (!customElements.get(tagName)) {
                customElements.define(tagName, DgaSecondNavHeader);
            }
            break;
    } });
}

export { DgaSecondNavHeader as D, defineCustomElement as d };
