// @platformscode/core/dist/components/p-COOZpHM-.js
import { h, p as proxyCustomElement, H, d as Host } from './p-JCGAt0My.js';
import { g as guid } from './p-DEtoWGOc.js';
import { S as SxStyleManager } from './p-BFEqPAFn.js';
import { c as alert02SolidRoundedIcon, d as alertCircleSolidRoundedIcon, e as checkmarkCircle02SolidRoundedIcon } from './p-YcCmBZW3.js';
import { d as defineCustomElement$3 } from './p-awnvl4MX.js';
import { d as defineCustomElement$2 } from './p-CcNS0CYN.js';
import { d as defineCustomElement$1 } from './p-B8Ehyo7m.js';

const feedbackIconTypes = {
    success: h("dga-icon", { icon: checkmarkCircle02SolidRoundedIcon }),
    error: h("dga-icon", { icon: alertCircleSolidRoundedIcon }),
    warning: h("dga-icon", { icon: alert02SolidRoundedIcon }),
};
const DgaTextInput = /*@__PURE__*/ proxyCustomElement(class DgaTextInput extends H {
    constructor() {
        super();
        this.__registerHost();
        this.size = 'lg';
        this.type = 'text';
        this.variant = 'default';
        this.error = false;
        this.feedbackIcon = false;
        this.feedbackIconType = 'success';
        this.feedbackIconRing = false;
        this.helperText = '';
        this.disabled = false;
        this.readonly = false;
        this.required = false;
        this.fullwidth = true;
        //   @Prop() id: string = guid();
        this.name = guid();
        this.value = '';
        this.isActive = false;
        this.isFocused = false;
        // Generate a unique ID if not provided
        this.componentId = `dga-button-${Math.random().toString(36).substring(2, 9)}`;
        // Computed inline styles
        this.rootStyles = {};
    }
    valueChanged(newValue, oldValue) {
        if (newValue !== oldValue) {
            this.textInputEl['value'] = newValue;
        }
    }
    handleSxChange() {
        this.processStyling();
    }
    componentWillLoad() {
        // Initialize style manager with element reference
        this.sxStyleManager = new SxStyleManager(this.componentId, this.el);
    }
    componentDidLoad() {
        // Process styling after the component is fully loaded and shadow DOM is ready
        this.processStyling();
    }
    componentDidUpdate() {
        // Re-process styling after updates
        this.processStyling();
    }
    processStyling() {
        if (!this.sx)
            return;
        // Handle both string and object formats
        const sxValue = typeof this.sx === 'string' ? JSON.parse(this.sx) : this.sx;
        this.rootStyles = this.sxStyleManager.processSxProp(sxValue);
    }
    disconnectedCallback() {
        var _a;
        // Clean up styles when component is removed
        (_a = this.sxStyleManager) === null || _a === void 0 ? void 0 : _a.cleanup();
    }
    render() {
        return (h(Host, { key: '6a7418f442f89d250b71a49abb24329d63ab6634', id: this.componentId, style: this.rootStyles }, h("div", { key: '111513b59d70e0695cafdd3bad7b1a2cfdb4c471', class: { "dga-form-control": true, "dga-form-control--fullwidth": this.fullwidth } }, this.label && h("dga-label", { key: 'ecfdad326868959f54caa6ae050d893f88dd2891', label: this.label, size: this.size, required: this.required, disabled: this.disabled }), h("div", { key: 'fac3f3bc3c844263f1bc61a8c145a3527ec16bf7', class: `input input--${this.size} ${this.error ? 'input--error' : ''} ${this.variant !== 'default' ? `input--${this.variant}` : ''} ${this.isActive ? 'active' : ''} ${this.isFocused ? 'focus' : ''} ${this.disabled ? 'input--disabled' : ''} ${this.readonly ? 'readOnly' : ''}`, ref: el => (this.inputContainerRef = el) }, h("slot", { key: '7f984cb04e2f0595d9645e7ad75eec7ed7f35372', name: "prefix" }), this.icon && (h("span", { key: 'ff68fe935f634c0fbe46884e6da2d939547a48ac', class: "input__icon" }, h("dga-icon", { key: '0f73e6935cfcf8805153b4beca14be2dfb82daeb', name: this.icon.name, variant: this.icon.variant, type: this.icon.type, size: 20 }))), h("input", { key: '9fc19c8968a223fb7888e3ead0018901044a42c2',
            // id={id}
            ref: (el) => (this.textInputEl = el), value: this.value, name: this.name, type: this.type, onFocus: () => (this.isFocused = true), onBlur: e => {
                this.isFocused = false;
                this.onBlur(e);
            }, class: "input__field", placeholder: this.placeholder, disabled: this.disabled, readOnly: this.readonly, required: this.required, onMouseDown: () => (this.isActive = true), onMouseUp: () => (this.isActive = false), onMouseOut: () => (this.isActive = false), onChange: this.onChange, onInput: this.onInput }), this.error ? (h("span", { class: `input__feedback-icon input__feedback-icon--error ${this.feedbackIconRing ? 'input__feedback-icon--ring' : ''}` })) : this.feedbackIcon ? (h("span", { class: `input__feedback-icon input__feedback-icon--${this.feedbackIconType} ${this.feedbackIconRing ? 'input__feedback-icon--ring' : ''}` }, feedbackIconTypes[this.feedbackIconType])) : null, h("slot", { key: 'ec58f555af3ba84febf9358b96331e072c6616fc', name: "suffix" }), " "), this.helperText && h("dga-helper-text", { key: '10c91a2f4376725600f7b3172be27f57bb455dfc', helperText: this.helperText, helperTextIcon: this.helperTextIcon, error: this.error }))));
    }
    get el() { return this; }
    static get watchers() { return {
        "value": ["valueChanged"],
        "sx": ["handleSxChange"]
    }; }
}, [4, "dga-text-input", {
        "label": [1],
        "placeholder": [1],
        "size": [1],
        "type": [1],
        "variant": [1],
        "error": [4],
        "feedbackIcon": [4, "feedback-icon"],
        "feedbackIconType": [1, "feedback-icon-type"],
        "feedbackIconRing": [4, "feedback-icon-ring"],
        "helperTextIcon": [16, "helper-text-icon"],
        "helperText": [1, "helper-text"],
        "icon": [16],
        "disabled": [4],
        "readonly": [4],
        "required": [4],
        "fullwidth": [4],
        "name": [1],
        "value": [1],
        "onChange": [16, "on-change"],
        "onInput": [16, "on-input"],
        "onBlur": [16, "on-blur"],
        "sx": [1],
        "isActive": [32],
        "isFocused": [32]
    }, undefined, {
        "value": ["valueChanged"],
        "sx": ["handleSxChange"]
    }]);
function defineCustomElement() {
    if (typeof customElements === "undefined") {
        return;
    }
    const components = ["dga-text-input", "dga-helper-text", "dga-icon", "dga-label"];
    components.forEach(tagName => { switch (tagName) {
        case "dga-text-input":
            if (!customElements.get(tagName)) {
                customElements.define(tagName, DgaTextInput);
            }
            break;
        case "dga-helper-text":
            if (!customElements.get(tagName)) {
                defineCustomElement$3();
            }
            break;
        case "dga-icon":
            if (!customElements.get(tagName)) {
                defineCustomElement$2();
            }
            break;
        case "dga-label":
            if (!customElements.get(tagName)) {
                defineCustomElement$1();
            }
            break;
    } });
}

export { DgaTextInput as D, defineCustomElement as d };
