// @platformscode/core/dist/components/p-DgX1GhOA.js
import { p as proxyCustomElement, H, h } from './p-JCGAt0My.js';
import { d as defineCustomElement$1 } from './p-CcNS0CYN.js';

const DgaProgressIndicatorStep = /*@__PURE__*/ proxyCustomElement(class DgaProgressIndicatorStep extends H {
    constructor() {
        super();
        this.__registerHost();
        this.currentStep = 1;
        this.title = '';
        this.description = '';
        this.nextStep = true;
        this.showStepName = true;
        this.showStepDescription = true;
        this.labelStyle = 'circle';
        this.internalActiveStep = 1;
        this.isActive = false;
        this.labelState = 'upcoming';
        this.handleFocus = () => {
            this.stepContentRef.classList.add('focus');
        };
        this.handleFocusOut = () => {
            this.stepContentRef.classList.remove('focus');
        };
        this.handleEscape = (event) => {
            if (event.key === 'Escape') {
                this.handleFocusOut();
                const inputElement = this.el.querySelector('input');
                if (inputElement) {
                    inputElement.blur();
                }
            }
        };
    }
    componentWillLoad() {
        this.updateState();
        window.addEventListener('keydown', this.handleEscape);
        // Get the progress key from the parent
        this.progressKey = this.el.getAttribute('data-progress-key');
    }
    disconnectedCallback() {
        window.removeEventListener('keydown', this.handleEscape);
    }
    stepChangeHandler(event) {
        // Only update if the event is from our parent progress indicator
        if (event.detail.progressKey === this.progressKey) {
            this.internalActiveStep = event.detail.step;
            this.updateState();
        }
    }
    updateState() {
        this.isActive = this.internalActiveStep > this.currentStep;
        if (this.internalActiveStep > this.currentStep) {
            this.labelState = 'completed';
        }
        else if (this.internalActiveStep < this.currentStep) {
            this.labelState = 'upcoming';
        }
        else {
            this.labelState = 'current';
        }
    }
    labelChildren(currentStep) {
        if (this.labelState === 'completed') {
            return h("dga-icon", { name: "tick-02", variant: "stroke", size: 16 });
        }
        return currentStep.toString();
    }
    render() {
        return (h("div", { key: 'aefa8b3de485cd680b113d810ad23efbec59c4d7', class: "progress-indicator__step", onMouseDown: this.handleFocus, onMouseUp: this.handleFocusOut }, h("div", { key: 'bd3f05e617713811c6ad839b0f62f256ed16e399', class: "progress-indicator__step-content dga-flex-column", ref: el => (this.stepContentRef = el) }, h("div", { key: '8e7fd1cacb3ba89ca65b51342ee528976c82f8a3', class: "step-label", "data-state": this.labelState }, h("span", { key: '27ba058dd4ff8fcaccfae89b35e9f9826df2b71b', class: `step-label--${this.labelStyle}` }, this.labelStyle === 'circle' && this.labelChildren(this.currentStep))), h("div", { key: 'ded1e2ca0776961b628b1030a6516019b23cd75e', class: "step-text dga-flex-column" }, this.showStepName && h("span", { key: 'c72973d7ae85a68855bc1359a19db957f73e2906', class: this.isActive ? 'text-md-medium' : 'text-md-regular' }, this.title), this.showStepDescription && h("p", { key: '3a54081c28289a076882b358e26ecb1058057d9c', class: "text-sm-regular" }, this.description))), this.nextStep && h("span", { key: '659ad1c346eda2a2491e693c3e148094d7a6f632', class: `progress-indicator__step-connector ${this.isActive ? 'progress-indicator__step-connector--active' : ''}` })));
    }
    get el() { return this; }
}, [0, "dga-progress-indicator-step", {
        "currentStep": [1538, "current-step"],
        "title": [1],
        "description": [1],
        "nextStep": [4, "next-step"],
        "showStepName": [4, "show-step-name"],
        "showStepDescription": [4, "show-step-description"],
        "labelStyle": [1, "label-style"],
        "internalActiveStep": [32],
        "isActive": [32],
        "labelState": [32]
    }, [[4, "stepChange", "stepChangeHandler"]]]);
function defineCustomElement() {
    if (typeof customElements === "undefined") {
        return;
    }
    const components = ["dga-progress-indicator-step", "dga-icon"];
    components.forEach(tagName => { switch (tagName) {
        case "dga-progress-indicator-step":
            if (!customElements.get(tagName)) {
                customElements.define(tagName, DgaProgressIndicatorStep);
            }
            break;
        case "dga-icon":
            if (!customElements.get(tagName)) {
                defineCustomElement$1();
            }
            break;
    } });
}

export { DgaProgressIndicatorStep as D, defineCustomElement as d };
