// @platformscode/core/dist/components/p-DJjQrzWf.js
import { p as proxyCustomElement, H, h, d as Host } from './p-JCGAt0My.js';

const DgaSecondNavHeaderContent = /*@__PURE__*/ proxyCustomElement(class DgaSecondNavHeaderContent extends H {
    constructor() {
        super();
        this.__registerHost();
    }
    render() {
        return (h(Host, { key: 'd1e640c6c31419544931a943928581ef2ea7eb5c' }, h("div", { key: '6e70bdd808ba728ed6ef56f595cc5bef8001cb59', class: `items-container` }, h("slot", { key: '47451b0f111654c0c6691e28b6f75aff959efa01' }))));
    }
}, [4, "dga-second-nav-header-content"]);
function defineCustomElement() {
    if (typeof customElements === "undefined") {
        return;
    }
    const components = ["dga-second-nav-header-content"];
    components.forEach(tagName => { switch (tagName) {
        case "dga-second-nav-header-content":
            if (!customElements.get(tagName)) {
                customElements.define(tagName, DgaSecondNavHeaderContent);
            }
            break;
    } });
}

export { DgaSecondNavHeaderContent as D, defineCustomElement as d };
