// @platformscode/core/dist/components/p-ybSjjgRd.js
import { p as proxyCustomElement, H, h } from './p-JCGAt0My.js';

const DgaContentSwitcher = /*@__PURE__*/ proxyCustomElement(class DgaContentSwitcher extends H {
    constructor() {
        super();
        this.__registerHost();
        this.value = 0;
        this.onColor = false;
        this.size = 'md';
        this.items = [
            {
                label: 'Item 1',
                content: 'Content 1',
            },
            {
                label: 'Item 2',
                content: 'Content 2',
            },
            {
                label: 'Item 3',
                content: 'Content 3',
            },
        ];
        this.onChange = () => { };
        this.selected = this.value;
        this.handleSelect = (index) => {
            this.selected = index;
            this.onChange(index);
        };
    }
    render() {
        return (h("div", { key: '2466b35db0280a8325df5b74a22c71b192666141', class: "content-switcher" }, h("div", { key: '348081a64adfd115ef5715bb5f05004cf475ea5c', class: "content-switcher__items" }, this.items.map((item, index) => {
            const selected = index === this.selected;
            const itemType = index === 0 ? 'first' : index === this.items.length - 1 ? 'last' : 'mid';
            const className = `content-switcher__item content-switcher__item--${itemType} ${this.onColor ? 'content-switcher__item--on-color' : ''} content-switcher__item--${this.size} ${selected ? 'content-switcher__item--selected' : ''}`;
            return (h("button", { key: index, type: "button", class: className, onClick: () => this.handleSelect(index) }, item.label));
        }))));
    }
}, [0, "dga-content-switcher", {
        "value": [1538],
        "onColor": [4, "on-color"],
        "size": [1],
        "items": [16],
        "onChange": [16, "on-change"],
        "selected": [32]
    }]);
function defineCustomElement() {
    if (typeof customElements === "undefined") {
        return;
    }
    const components = ["dga-content-switcher"];
    components.forEach(tagName => { switch (tagName) {
        case "dga-content-switcher":
            if (!customElements.get(tagName)) {
                customElements.define(tagName, DgaContentSwitcher);
            }
            break;
    } });
}

export { DgaContentSwitcher as D, defineCustomElement as d };
