// @platformscode/core/dist/collection/components/dga-card-v2/dga-card-v2.js
import { h, Host } from "@stencil/core";
import { SxStyleManager } from "../../utils/sx-style-manager";
export class DgaCardV2 {
    constructor() {
        this.type = 'default';
        // @Prop() showTitle?: boolean = true;
        // @Prop() showDescription?: boolean = true;
        // @Prop() showFeaturedIcon?: boolean = true;
        // @Prop() image?: string = '';
        this.disabled = false;
        this.effect = 'with-shadow';
        // @Prop({attribute: "title"}) cardTitle?: string = 'Card Title';
        // @Prop() description?: string = 'Card content placeholder text goes here';
        // @Prop() featuredIcon?: Icon = {
        //   name: 'checkmark-circle-02',
        //   variant: 'stroke',
        //   type: 'sharp',
        //   size: 24
        // }; // Default icon
        // @Prop() showSecondaryAction?: boolean = true;
        this.isSelected = false;
        // @Prop() secondaryActionLabel?: string = 'Secondary Action';
        this.onCheckboxChange = () => { };
        this.width = '360px';
        this.maxWidth = '100%';
        this.minWidth = 'fit-content';
        this.isFocused = false;
        // @State() expanded: boolean = false;
        // toggleExpand() {
        //   this.expanded = !this.expanded;
        // }
        // Generate a unique ID if not provided
        this.componentId = `dga-button-${Math.random().toString(36).substring(2, 9)}`;
        // Computed inline styles
        this.rootStyles = {};
        this.onMouseEnterHandler = () => {
            this.isFocused = true;
        };
        this.onMouseLeaveHandler = () => {
            this.isFocused = false;
        };
        this.cardStyles = {
            width: this.width,
            maxWidth: this.maxWidth,
            minWidth: this.minWidth
        };
    }
    handleSxChange() {
        this.processStyling();
    }
    componentWillLoad() {
        // Initialize style manager with element reference
        this.sxStyleManager = new SxStyleManager(this.componentId, this.el);
        this.el.childNodes.forEach(item => {
            const el = item;
            if (el.tagName === 'DGA-CARD-CONTENT' && el.attributes.getNamedItem('slot') === null) {
                el.setAttribute('slot', 'card-content');
            }
            if (el.tagName === 'DGA-CARD-ACTIONS' && el.attributes.getNamedItem('slot') === null) {
                el.setAttribute('slot', 'card-actions');
            }
            if (el.tagName === 'DGA-CARD-EXPANDABLE-CONTENT' && el.attributes.getNamedItem('slot') === null) {
                el.setAttribute('slot', 'card-expandable-content');
            }
        });
    }
    componentDidLoad() {
        // Process styling after the component is fully loaded and shadow DOM is ready
        this.processStyling();
    }
    componentDidUpdate() {
        // Re-process styling after updates
        this.processStyling();
    }
    processStyling() {
        if (!this.sx)
            return;
        // Handle both string and object formats
        const sxValue = typeof this.sx === 'string'
            ? JSON.parse(this.sx)
            : this.sx;
        this.rootStyles = this.sxStyleManager.processSxProp(sxValue);
    }
    disconnectedCallback() {
        var _a;
        // Clean up styles when component is removed
        (_a = this.sxStyleManager) === null || _a === void 0 ? void 0 : _a.cleanup();
    }
    render() {
        return (h(Host, { key: 'dc0c881744e6458953c9becdbcb2a4fa1556d81f', id: this.componentId, style: this.rootStyles }, h("div", Object.assign({ key: 'd26ff98d8700c7731729bde7281d148963fd0a35', class: `card ${this.disabled ? 'disabled' : ''} ${this.isFocused ? 'focused' : ''} card--${this.effect}`, onMouseEnter: this.onMouseEnterHandler, onMouseLeave: this.onMouseLeaveHandler }, (this.type === 'selectable' ? { onClick: this.onCheckboxChange } : {}), { style: this.cardStyles }), this.type === 'selectable' && (h("div", { key: 'ee47d3c745d0510cbf0a5c7b98b09a5d5e51c75d', class: "card__checkbox-container" }, h("dga-checkbox", { key: '83c21946cf0f86e76dc97abb3e140b732dd11087', checked: this.isSelected, color: "brand", onChange: this.onCheckboxChange }))), h("slot", { key: '350d32fdd7354aed8a86fc93defc0430070637ac', name: 'card-content' }), this.type === 'default' && h("slot", { key: '24d507cc53e8d80a35374a73d6583dca45417050', name: 'card-actions' }))));
    }
    static get is() { return "dga-card-v2"; }
    static get encapsulation() { return "shadow"; }
    static get originalStyleUrls() {
        return {
            "$": ["dga-card.scss"]
        };
    }
    static get styleUrls() {
        return {
            "$": ["dga-card.css"]
        };
    }
    static get properties() {
        return {
            "type": {
                "type": "string",
                "attribute": "type",
                "mutable": false,
                "complexType": {
                    "original": "'default' | 'expanded' | 'selectable'",
                    "resolved": "\"default\" | \"expanded\" | \"selectable\"",
                    "references": {}
                },
                "required": false,
                "optional": true,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false,
                "defaultValue": "'default'"
            },
            "disabled": {
                "type": "boolean",
                "attribute": "disabled",
                "mutable": false,
                "complexType": {
                    "original": "boolean",
                    "resolved": "boolean",
                    "references": {}
                },
                "required": false,
                "optional": true,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false,
                "defaultValue": "false"
            },
            "effect": {
                "type": "string",
                "attribute": "effect",
                "mutable": false,
                "complexType": {
                    "original": "'with-shadow' | 'no-shadow' | 'stroke'",
                    "resolved": "\"no-shadow\" | \"stroke\" | \"with-shadow\"",
                    "references": {}
                },
                "required": false,
                "optional": true,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false,
                "defaultValue": "'with-shadow'"
            },
            "isSelected": {
                "type": "boolean",
                "attribute": "is-selected",
                "mutable": true,
                "complexType": {
                    "original": "boolean",
                    "resolved": "boolean",
                    "references": {}
                },
                "required": false,
                "optional": true,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": true,
                "defaultValue": "false"
            },
            "onCheckboxChange": {
                "type": "unknown",
                "attribute": "on-checkbox-change",
                "mutable": false,
                "complexType": {
                    "original": "() => void",
                    "resolved": "() => void",
                    "references": {}
                },
                "required": false,
                "optional": true,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "defaultValue": "() => {}"
            },
            "sx": {
                "type": "string",
                "attribute": "sx",
                "mutable": false,
                "complexType": {
                    "original": "SxProps | string",
                    "resolved": "string | { [x: string]: any; }",
                    "references": {
                        "SxProps": {
                            "location": "import",
                            "path": "../../utils/sx-style-manager",
                            "id": "src/utils/sx-style-manager.ts::SxProps"
                        }
                    }
                },
                "required": false,
                "optional": true,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false
            },
            "width": {
                "type": "string",
                "attribute": "width",
                "mutable": false,
                "complexType": {
                    "original": "string",
                    "resolved": "string",
                    "references": {}
                },
                "required": false,
                "optional": true,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false,
                "defaultValue": "'360px'"
            },
            "maxWidth": {
                "type": "string",
                "attribute": "max-width",
                "mutable": false,
                "complexType": {
                    "original": "string",
                    "resolved": "string",
                    "references": {}
                },
                "required": false,
                "optional": true,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false,
                "defaultValue": "'100%'"
            },
            "minWidth": {
                "type": "string",
                "attribute": "min-width",
                "mutable": false,
                "complexType": {
                    "original": "string",
                    "resolved": "string",
                    "references": {}
                },
                "required": false,
                "optional": true,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false,
                "defaultValue": "'fit-content'"
            }
        };
    }
    static get states() {
        return {
            "isFocused": {}
        };
    }
    static get elementRef() { return "el"; }
    static get watchers() {
        return [{
                "propName": "sx",
                "methodName": "handleSxChange"
            }];
    }
}
