// @platformscode/core/dist/collection/components/dga-notification/dga-notification.js
import { Host, h } from "@stencil/core";
import { SxStyleManager } from "../../utils/sx-style-manager";
export class DgaNotification {
    constructor() {
        this.icon = true;
        this.link = false;
        this.dismissable = false;
        this.icons = {
            success: {
                name: "checkmark-circle-02",
                variant: "solid",
                type: "sharp",
            },
            info: {
                name: "information-circle",
                variant: "solid",
                type: "sharp",
            },
            neutral: {
                name: "information-circle",
                variant: "solid",
                type: "sharp",
            },
            warning: {
                name: "alert-02",
                variant: "solid",
                type: "sharp",
            },
            critical: {
                name: "alert-circle",
                variant: "solid",
                type: "sharp",
            },
        };
        // Generate a unique ID if not provided
        this.componentId = `dga-notification-${Math.random().toString(36).substring(2, 9)}`;
        // Computed inline styles
        this.rootStyles = {};
    }
    handleSxChange() {
        this.processStyling();
    }
    componentWillLoad() {
        // Initialize style manager with element reference
        this.sxStyleManager = new SxStyleManager(this.componentId, this.el);
    }
    componentDidLoad() {
        // Process styling after the component is fully loaded and shadow DOM is ready
        this.processStyling();
    }
    componentDidUpdate() {
        // Re-process styling after updates
        this.processStyling();
    }
    processStyling() {
        if (!this.sx)
            return;
        // Handle both string and object formats
        const sxValue = typeof this.sx === 'string' ? JSON.parse(this.sx) : this.sx;
        this.rootStyles = this.sxStyleManager.processSxProp(sxValue);
    }
    disconnectedCallback() {
        var _a;
        // Clean up styles when component is removed
        (_a = this.sxStyleManager) === null || _a === void 0 ? void 0 : _a.cleanup();
    }
    render() {
        return (h(Host, { key: '2c3b1752091f30f334fc3e3418961f01598c4613', id: this.componentId, style: this.rootStyles }, h("div", { key: 'eddcf9ee0fc07c8fc7c7b32ba32045001e9d138c', class: `notification notification--${this.variant}`, role: "alert" }, h("div", { key: '7726501fe6832c4124c995c42e3e80c595095e58', class: "notification__content" }, this.icon && h("div", { key: '6c309ac12539be755c3dafd6770bd108de146288', class: "notification__icon" }, h("dga-icon-v2", { key: '50a5232c665386fcefe50cdeff7b8eec4c567446', name: this.icons[this.variant].name, variant: this.icons[this.variant].variant, type: this.icons[this.variant].type, size: 24 })), this.leadText && h("strong", { key: '022e92ada1f26ebde1846fdecde570d4bf381629', class: "notification__lead-text" }, this.leadText, ": "), h("p", { key: '2e9f585f4d43faee6a52e3fe2211c54e0ae84ffa' }, this.content), this.link && (h("a", { key: 'e5d6b05478e4de18189ef5179e98c0da882902f5', onClick: () => {
                this.navigateTo && this.navigateTo(this.setNavigateTo);
            }, class: "notification__link link link--md link--neutral link--inline" }, h("span", { key: 'ffd075f0a3f9e25b0b268538437f8ac7c5d63ce0', class: "link__label" }, this.textLink)))), this.dismissable && (h("div", { key: 'af29afd7d1bb209a94d31d23ca1bebe562fa44b0', class: "notification__dismiss" }, h("button", { key: '610803be38c0e0f0ff96886cf6eae92c7e4ea70a', class: "dga-btn dga-btn--close dga-btn--icon dga-btn--md", onClick: () => {
                this.dismmissed && this.dismmissed();
            } }, h("span", { key: '386acaaa7397e7dfa82601709c4b7d856ab467fa', class: "dga-btn-icon" }, h("dga-icon-v2", { key: '603a3cb477f402bd65d29890f51b63270bd0ad53', name: "multiplication-sign", variant: 'stroke', size: 20 }))))))));
    }
    static get is() { return "dga-notification"; }
    static get encapsulation() { return "shadow"; }
    static get originalStyleUrls() {
        return {
            "$": ["dga-notification.scss"]
        };
    }
    static get styleUrls() {
        return {
            "$": ["dga-notification.css"]
        };
    }
    static get properties() {
        return {
            "icon": {
                "type": "boolean",
                "attribute": "icon",
                "mutable": false,
                "complexType": {
                    "original": "boolean",
                    "resolved": "boolean",
                    "references": {}
                },
                "required": false,
                "optional": true,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false,
                "defaultValue": "true"
            },
            "leadText": {
                "type": "string",
                "attribute": "lead-text",
                "mutable": false,
                "complexType": {
                    "original": "string",
                    "resolved": "string",
                    "references": {}
                },
                "required": false,
                "optional": false,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false
            },
            "content": {
                "type": "string",
                "attribute": "content",
                "mutable": false,
                "complexType": {
                    "original": "string",
                    "resolved": "string",
                    "references": {}
                },
                "required": false,
                "optional": false,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false
            },
            "variant": {
                "type": "string",
                "attribute": "variant",
                "mutable": false,
                "complexType": {
                    "original": "'critical' | 'warning' | 'success' | 'info' | 'neutral'",
                    "resolved": "\"critical\" | \"info\" | \"neutral\" | \"success\" | \"warning\"",
                    "references": {}
                },
                "required": false,
                "optional": false,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false
            },
            "link": {
                "type": "boolean",
                "attribute": "link",
                "mutable": false,
                "complexType": {
                    "original": "boolean",
                    "resolved": "boolean",
                    "references": {}
                },
                "required": false,
                "optional": true,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false,
                "defaultValue": "false"
            },
            "textLink": {
                "type": "string",
                "attribute": "text-link",
                "mutable": false,
                "complexType": {
                    "original": "string",
                    "resolved": "string",
                    "references": {}
                },
                "required": false,
                "optional": true,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false
            },
            "dismissable": {
                "type": "boolean",
                "attribute": "dismissable",
                "mutable": false,
                "complexType": {
                    "original": "boolean",
                    "resolved": "boolean",
                    "references": {}
                },
                "required": false,
                "optional": true,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false,
                "defaultValue": "false"
            },
            "setNavigateTo": {
                "type": "string",
                "attribute": "set-navigate-to",
                "mutable": false,
                "complexType": {
                    "original": "string",
                    "resolved": "string",
                    "references": {}
                },
                "required": false,
                "optional": true,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false
            },
            "navigateTo": {
                "type": "unknown",
                "attribute": "navigate-to",
                "mutable": false,
                "complexType": {
                    "original": "(link?: string) => void",
                    "resolved": "(link?: string) => void",
                    "references": {}
                },
                "required": false,
                "optional": true,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false
            },
            "dismmissed": {
                "type": "unknown",
                "attribute": "dismmissed",
                "mutable": false,
                "complexType": {
                    "original": "() => void",
                    "resolved": "() => void",
                    "references": {}
                },
                "required": false,
                "optional": true,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false
            },
            "sx": {
                "type": "string",
                "attribute": "sx",
                "mutable": false,
                "complexType": {
                    "original": "SxProps | string",
                    "resolved": "string | { [x: string]: any; }",
                    "references": {
                        "SxProps": {
                            "location": "import",
                            "path": "../../utils/sx-style-manager",
                            "id": "src/utils/sx-style-manager.ts::SxProps"
                        }
                    }
                },
                "required": false,
                "optional": true,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false
            }
        };
    }
    static get elementRef() { return "el"; }
    static get watchers() {
        return [{
                "propName": "sx",
                "methodName": "handleSxChange"
            }];
    }
}
