// @platformscode/core/dist/collection/components/dga-tooltip/dga-tooltip.js
import { Host, h } from "@stencil/core";
import { SxHandler } from "../../utils/sx-handler";
export class DgaTooltip {
    constructor() {
        this.beakPlacement = 'none';
        this.beakAlignment = 'center';
        this.direction = 'top';
        this.icon = false;
        this.theme = 'light';
        this.isHovered = false;
        this.icons = {
            'success-check': {
                icon: 'checkmark-circle-02',
                color: '#067647',
            },
            'blue-info': {
                icon: 'information-circle',
                color: '#175CD3',
            },
            'neutral-info': {
                icon: 'information-circle',
                color: '#384250',
            },
            'neutral-?': {
                icon: 'help-circle',
                color: '#384250',
            },
            'warning-!-circle': {
                icon: 'alert-circle',
                color: '#B54708',
            },
            'warning-!-triangle': {
                icon: 'sharp hgi-alert-02',
                color: '#B54708',
            },
            'error-!': {
                icon: 'alert-circle',
                color: '#B42318',
            },
            'error-x': {
                icon: 'cancel-circle',
                color: '#B42318',
            },
        };
        this.handleMouseEnter = () => {
            this.isHovered = true;
        };
        this.handleMouseLeave = () => {
            this.isHovered = false;
        };
    }
    sxChanged() {
        this.sxHandler.processSx(this.sx);
    }
    componentWillLoad() {
        this.sxHandler = new SxHandler(this.el, 'dga-tooltip');
    }
    componentDidLoad() {
        this.sxHandler.processSx(this.sx);
    }
    disconnectedCallback() {
        this.sxHandler.cleanup();
    }
    render() {
        return (h(Host, { key: 'bcd609108df4b22656d60c316d669aaba9a10faf', id: this.sxHandler.getId(), style: this.sxHandler.getStyles() }, h("div", { key: '618f6d8a3b29705e677130afd1c8a176e3d05c4f', class: "dga-tooltip", "data-theme": this.theme, "data-beak-direction": `${this.beakPlacement}-${this.beakAlignment}` }, this.isHovered && (h("div", { key: '43f39e1ef791e5fbddca3f502fe02b5b3fbeefc5', class: "dga-tooltip-main", "data-direction": this.direction }, h("div", { key: 'b1f2a00b81837ffbc393200f608fe254c29163b7', class: "dga-tooltip-main__wrapper" }, this.beakPlacement !== 'none' && h("span", { key: 'bcb3e728fb339542e7b8a0b4c17a48e535c4b50b', class: "dga-tooltip-main__beak", "data-beak-direction": `${this.beakPlacement}-${this.beakAlignment}` }), this.icon && (h("span", { key: '00b2b92a4139ce15dcf49cd979213e5e02840390', class: "dga-tooltip-main__icon" }, h("dga-icon", { key: '08f0b01d402a7bac3144f94ceede1131af468de7', name: this.icons[this.feedbackIcon.type].icon, variant: this.feedbackIcon.style === 'icon-only' ? "stroke" : "solid", size: this.feedbackIcon.size, color: this.icons[this.feedbackIcon.type].color }))), h("div", { key: '180aa44f3dfc1de5fc7f4263a53bf67ae951acc1', class: "dga-tooltip-main__content" }, this.tooltipTitle && h("span", { key: 'ffc976fc3af613f31e94454b378e9a79a59b5f21', class: "dga-tooltip-main__content__title" }, this.tooltipTitle), this.helperText && h("span", { key: 'fe0b179074fd52ff66c4a5c25ce238e302f75548', class: "dga-tooltip-main__content__helper-text" }, this.helperText))))), h("div", { key: '7af00a76eba069e3a8acf90fc5ec6e75b6b5e21e', class: "dga-tooltip-item", onMouseEnter: this.handleMouseEnter, onMouseLeave: this.handleMouseLeave }, h("slot", { key: '2961b12c20242ff839f21c0da1d1e0925c455a69' })))));
    }
    static get is() { return "dga-tooltip"; }
    static get originalStyleUrls() {
        return {
            "$": ["dga-tooltip.scss"]
        };
    }
    static get styleUrls() {
        return {
            "$": ["dga-tooltip.css"]
        };
    }
    static get properties() {
        return {
            "sx": {
                "type": "string",
                "attribute": "sx",
                "mutable": false,
                "complexType": {
                    "original": "SxProps | string | undefined",
                    "resolved": "SxProps | string",
                    "references": {
                        "SxProps": {
                            "location": "import",
                            "path": "../../utils/sx-handler",
                            "id": "src/utils/sx-handler.ts::SxProps"
                        }
                    }
                },
                "required": false,
                "optional": true,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false
            },
            "beakPlacement": {
                "type": "string",
                "attribute": "beak-placement",
                "mutable": false,
                "complexType": {
                    "original": "BeakPlacement",
                    "resolved": "\"bottom\" | \"left\" | \"none\" | \"right\" | \"top\"",
                    "references": {
                        "BeakPlacement": {
                            "location": "local",
                            "path": "/Users/mohamedaboelezz/projects/Azm-X/platform-code-cross-platform-storybook/packages/core/src/components/dga-tooltip/dga-tooltip.tsx",
                            "id": "src/components/dga-tooltip/dga-tooltip.tsx::BeakPlacement"
                        }
                    }
                },
                "required": false,
                "optional": false,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false,
                "defaultValue": "'none'"
            },
            "beakAlignment": {
                "type": "string",
                "attribute": "beak-alignment",
                "mutable": false,
                "complexType": {
                    "original": "BeakAlignment",
                    "resolved": "\"center\" | \"end\" | \"start\"",
                    "references": {
                        "BeakAlignment": {
                            "location": "local",
                            "path": "/Users/mohamedaboelezz/projects/Azm-X/platform-code-cross-platform-storybook/packages/core/src/components/dga-tooltip/dga-tooltip.tsx",
                            "id": "src/components/dga-tooltip/dga-tooltip.tsx::BeakAlignment"
                        }
                    }
                },
                "required": false,
                "optional": false,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false,
                "defaultValue": "'center'"
            },
            "direction": {
                "type": "string",
                "attribute": "direction",
                "mutable": false,
                "complexType": {
                    "original": "Direction",
                    "resolved": "\"bottom\" | \"left\" | \"right\" | \"top\"",
                    "references": {
                        "Direction": {
                            "location": "local",
                            "path": "/Users/mohamedaboelezz/projects/Azm-X/platform-code-cross-platform-storybook/packages/core/src/components/dga-tooltip/dga-tooltip.tsx",
                            "id": "src/components/dga-tooltip/dga-tooltip.tsx::Direction"
                        }
                    }
                },
                "required": false,
                "optional": false,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false,
                "defaultValue": "'top'"
            },
            "tooltipTitle": {
                "type": "string",
                "attribute": "title",
                "mutable": false,
                "complexType": {
                    "original": "string",
                    "resolved": "string",
                    "references": {}
                },
                "required": false,
                "optional": true,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false
            },
            "helperText": {
                "type": "string",
                "attribute": "helper-text",
                "mutable": false,
                "complexType": {
                    "original": "string",
                    "resolved": "string",
                    "references": {}
                },
                "required": false,
                "optional": true,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false
            },
            "icon": {
                "type": "boolean",
                "attribute": "icon",
                "mutable": true,
                "complexType": {
                    "original": "boolean",
                    "resolved": "boolean",
                    "references": {}
                },
                "required": false,
                "optional": false,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": true,
                "defaultValue": "false"
            },
            "feedbackIcon": {
                "type": "unknown",
                "attribute": "feedback-icon",
                "mutable": false,
                "complexType": {
                    "original": "FeedbackIcon",
                    "resolved": "FeedbackIcon",
                    "references": {
                        "FeedbackIcon": {
                            "location": "local",
                            "path": "/Users/mohamedaboelezz/projects/Azm-X/platform-code-cross-platform-storybook/packages/core/src/components/dga-tooltip/dga-tooltip.tsx",
                            "id": "src/components/dga-tooltip/dga-tooltip.tsx::FeedbackIcon"
                        }
                    }
                },
                "required": false,
                "optional": false,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false
            },
            "theme": {
                "type": "string",
                "attribute": "theme",
                "mutable": false,
                "complexType": {
                    "original": "Theme",
                    "resolved": "\"dark\" | \"light\"",
                    "references": {
                        "Theme": {
                            "location": "local",
                            "path": "/Users/mohamedaboelezz/projects/Azm-X/platform-code-cross-platform-storybook/packages/core/src/components/dga-tooltip/dga-tooltip.tsx",
                            "id": "src/components/dga-tooltip/dga-tooltip.tsx::Theme"
                        }
                    }
                },
                "required": false,
                "optional": false,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false,
                "defaultValue": "'light'"
            }
        };
    }
    static get states() {
        return {
            "isHovered": {}
        };
    }
    static get elementRef() { return "el"; }
    static get watchers() {
        return [{
                "propName": "sx",
                "methodName": "sxChanged"
            }];
    }
}
