// @platformscode/core/dist/collection/components/dga-actions/dga-actions.js
import { h, Host } from "@stencil/core";
export class DgaActions {
    constructor() {
        /** Direction of stack items: 'row' (horizontal) or 'column' (vertical) */
        this.direction = 'row';
        /** Spacing between stack items */
        this.gap = '8px';
        /** Horizontal alignment of stack items */
        this.align = 'start';
        /** Vertical alignment of stack items */
        this.justify = 'start';
        /** Controls if items should wrap */
        this.wrap = false;
    }
    render() {
        const styles = {
            display: 'flex',
            flexDirection: this.direction,
            gap: this.gap,
            alignItems: this.align,
            justifyContent: this.justify === 'between' ? 'space-between' : this.justify,
            flexWrap: this.wrap ? 'wrap' : 'nowrap',
        };
        return (h(Host, { key: '6803cc210304fad1552964069fe919e22277291a', style: styles }, h("slot", { key: '10acb9942424bfcf4360c17ee2428c94e5f12c8d' })));
    }
    static get is() { return "dga-actions"; }
    static get encapsulation() { return "shadow"; }
    static get originalStyleUrls() {
        return {
            "$": ["dga-actions.scss"]
        };
    }
    static get styleUrls() {
        return {
            "$": ["dga-actions.css"]
        };
    }
    static get properties() {
        return {
            "direction": {
                "type": "string",
                "attribute": "direction",
                "mutable": false,
                "complexType": {
                    "original": "'row' | 'column'",
                    "resolved": "\"column\" | \"row\"",
                    "references": {}
                },
                "required": false,
                "optional": false,
                "docs": {
                    "tags": [],
                    "text": "Direction of stack items: 'row' (horizontal) or 'column' (vertical)"
                },
                "getter": false,
                "setter": false,
                "reflect": false,
                "defaultValue": "'row'"
            },
            "gap": {
                "type": "string",
                "attribute": "gap",
                "mutable": false,
                "complexType": {
                    "original": "string",
                    "resolved": "string",
                    "references": {}
                },
                "required": false,
                "optional": false,
                "docs": {
                    "tags": [],
                    "text": "Spacing between stack items"
                },
                "getter": false,
                "setter": false,
                "reflect": false,
                "defaultValue": "'8px'"
            },
            "align": {
                "type": "string",
                "attribute": "align",
                "mutable": false,
                "complexType": {
                    "original": "'start' | 'center' | 'end' | 'stretch'",
                    "resolved": "\"center\" | \"end\" | \"start\" | \"stretch\"",
                    "references": {}
                },
                "required": false,
                "optional": false,
                "docs": {
                    "tags": [],
                    "text": "Horizontal alignment of stack items"
                },
                "getter": false,
                "setter": false,
                "reflect": false,
                "defaultValue": "'start'"
            },
            "justify": {
                "type": "string",
                "attribute": "justify",
                "mutable": false,
                "complexType": {
                    "original": "'start' | 'center' | 'end' | 'between'",
                    "resolved": "\"between\" | \"center\" | \"end\" | \"start\"",
                    "references": {}
                },
                "required": false,
                "optional": false,
                "docs": {
                    "tags": [],
                    "text": "Vertical alignment of stack items"
                },
                "getter": false,
                "setter": false,
                "reflect": false,
                "defaultValue": "'start'"
            },
            "wrap": {
                "type": "boolean",
                "attribute": "wrap",
                "mutable": false,
                "complexType": {
                    "original": "boolean",
                    "resolved": "boolean",
                    "references": {}
                },
                "required": false,
                "optional": false,
                "docs": {
                    "tags": [],
                    "text": "Controls if items should wrap"
                },
                "getter": false,
                "setter": false,
                "reflect": false,
                "defaultValue": "false"
            }
        };
    }
}
