// @platformscode/core/dist/collection/components/dga-slideout-menu/dga-slideout-menu.js
import { h } from "@stencil/core";
export class DgaSlideoutMenu {
    constructor() {
        this.backgroundColor = 'gray';
        this.anchor = 'right';
        this.open = false;
        this.scrollbar = true;
    }
    // componentDidRender() {
    //     if (this.el.children.length > 0) {
    //         this.el.children
    //         .namedItem('DgaSlideoutMenuHeader')
    //         .setAttribute('slot', 'sideout-menu-header');
    //       this.el.children
    //         .namedItem('DgaSideoutMenuItems')
    //         .setAttribute('slot', 'sideout-menu-items');
    //       this.el.children
    //         .namedItem('DgaSideoutMenuActions')
    //         .setAttribute('slot', 'sideout-menu-actions');
    //     }
    //   }
    componentWillLoad() {
        this.el.childNodes.forEach(item => {
            const el = item;
            //   if(el.tagName === 'DGA-SECOND-NAV-HEADER-CONTENT' && this.hideContent === true) {
            //     el.style.display = 'none';
            //   }
            //   if(el.tagName === 'DGA-SECOND-NAV-HEADER-ACTIONS' && this.hideActions === true) {
            //     el.style.display = 'none';
            //   }
            if (el.tagName === 'DGA-SLIDEOUT-MENU-HEADER' && el.attributes.getNamedItem('slot') === null) {
                el.setAttribute('slot', 'slideout-menu-header');
                // Pass onClose to header component
                el.setAttribute('on-close', 'true');
                el.addEventListener('closeClick', () => {
                    if (this.onClose) {
                        this.onClose();
                    }
                });
            }
            if (el.tagName === 'DGA-SLIDEOUT-MENU-ITEMS' && el.attributes.getNamedItem('slot') === null) {
                el.setAttribute('slot', 'slideout-menu-items');
            }
            if (el.tagName === 'DGA-SLIDEOUT-MENU-ACTIONS' && el.attributes.getNamedItem('slot') === null) {
                el.setAttribute('slot', 'slideout-menu-actions');
            }
        });
    }
    disconnectedCallback() {
        // Clean up listeners
        this.el.childNodes.forEach(item => {
            const el = item;
            if (el.tagName === 'DGA-SLIDEOUT-MENU-HEADER') {
                el.removeEventListener('closeClick', this.onClose);
            }
        });
    }
    render() {
        return (h("div", { key: '98140cd1dbe34d38162faf7c6dd23fc6a2a3c452', class: `dga-slideout-menu dga-slideout-menu--${this.anchor} dga-slideout-menu--${this.backgroundColor} ${this.open ? "active" : ""}` }, h("slot", { key: 'baee78d547dbd06abcd81684a1f350b790bd4981', name: "slideout-menu-header" }), h("div", { key: '208c2b075354dd700841190ebc705ec0e1109afc', class: "dga-slideout-menu__body" }, h("div", { key: '8fcc135dd6cacafb3462940f3a6b0577b2538944', class: {
                'dga-slideout-menu__items': true,
                'dga-slideout-menu__items--hide-scrollbar': !this.scrollbar,
            } }, h("slot", { key: '4550a438c062d7061ca673affe726d30a0b02b44', name: "slideout-menu-items" })), h("dga-divider", { key: '5631f6f3b94b178394dd33275c7f5c962ee665d6' }), h("slot", { key: '1a19f53ef76f34957e68467b81a13b3fc5dc4681', name: "slideout-menu-actions" }))));
    }
    static get is() { return "dga-slideout-menu"; }
    static get originalStyleUrls() {
        return {
            "$": ["dga-slideout-menu.scss"]
        };
    }
    static get styleUrls() {
        return {
            "$": ["dga-slideout-menu.css"]
        };
    }
    static get properties() {
        return {
            "backgroundColor": {
                "type": "string",
                "attribute": "background-color",
                "mutable": false,
                "complexType": {
                    "original": "'gray' | 'white'",
                    "resolved": "\"gray\" | \"white\"",
                    "references": {}
                },
                "required": false,
                "optional": true,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false,
                "defaultValue": "'gray'"
            },
            "anchor": {
                "type": "string",
                "attribute": "anchor",
                "mutable": false,
                "complexType": {
                    "original": "'left'|'right'",
                    "resolved": "\"left\" | \"right\"",
                    "references": {}
                },
                "required": false,
                "optional": true,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false,
                "defaultValue": "'right'"
            },
            "open": {
                "type": "boolean",
                "attribute": "open",
                "mutable": true,
                "complexType": {
                    "original": "boolean",
                    "resolved": "boolean",
                    "references": {}
                },
                "required": false,
                "optional": true,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": true,
                "defaultValue": "false"
            },
            "scrollbar": {
                "type": "boolean",
                "attribute": "scrollbar",
                "mutable": false,
                "complexType": {
                    "original": "boolean",
                    "resolved": "boolean",
                    "references": {}
                },
                "required": false,
                "optional": true,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false,
                "defaultValue": "true"
            },
            "onClose": {
                "type": "unknown",
                "attribute": "on-close",
                "mutable": false,
                "complexType": {
                    "original": "() => void",
                    "resolved": "() => void",
                    "references": {}
                },
                "required": false,
                "optional": true,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false
            }
        };
    }
    static get elementRef() { return "el"; }
}
