// @platformscode/core/dist/collection/components/dga-slideout-menu/dga-slideout-menu-header.js
import { h } from "@stencil/core";
export class DgaSlideoutMenuHeader {
    constructor() {
        // @Prop() onClose?: () => void;
        this.handleClose = () => {
            // Find parent slideout menu
            const slideoutMenu = this.el.closest('dga-slideout-menu');
            if (slideoutMenu) {
                // Access parent's onClose
                const parentClose = slideoutMenu['onClose'];
                if (typeof parentClose === 'function') {
                    parentClose();
                }
            }
        };
    }
    render() {
        return (
        // <div class="dga-slideout-menu">
        h("div", { key: '644974162d8b708862d95a5de7777aae6b3b8b86', class: "dga-slideout-menu__header" }, h("dga-featured-icon", { key: '0e17b150bed933e50780e4ac92e141126f2b8b60', icon: {
                name: 'add-circle',
                variant: 'stroke',
                type: 'standard',
            }, size: 'sm', color: 'brand', variant: 'light' }), h("dga-flex", { key: 'ae14285fdecd65e9e15b044ded49897c0284c9c3', direction: "column", spacing: 4, class: "dga-slideout-menu__header-content" }, h("h3", { key: '1f17b0aa3e39fd4cb5776853a8ce0603d23547de', class: "dga-slideout-menu__header-title" }, this.title), h("p", { key: 'eaa482126c0d3b51120e99bb80d4754b2836ac0e', class: "dga-slideout-menu__header-description" }, this.description)), h("dga-button-v2", { key: '2565bbfc3ec38da30aa43571eb462a00a12f28cd', iconOnly: true, leadIcon: true, leadIconProps: {
                name: 'multiplication-sign',
                variant: 'stroke',
                type: 'standard',
                color: "#161616"
            }, variant: "transparent", onClick: this.handleClose }))
        // </div>
        );
    }
    static get is() { return "dga-slideout-menu-header"; }
    static get encapsulation() { return "shadow"; }
    static get originalStyleUrls() {
        return {
            "$": ["dga-slideout-menu-header.scss"]
        };
    }
    static get styleUrls() {
        return {
            "$": ["dga-slideout-menu-header.css"]
        };
    }
    static get properties() {
        return {
            "title": {
                "type": "string",
                "attribute": "title",
                "mutable": false,
                "complexType": {
                    "original": "string",
                    "resolved": "string",
                    "references": {}
                },
                "required": false,
                "optional": false,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false
            },
            "description": {
                "type": "string",
                "attribute": "description",
                "mutable": false,
                "complexType": {
                    "original": "string",
                    "resolved": "string",
                    "references": {}
                },
                "required": false,
                "optional": false,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false
            }
        };
    }
    static get elementRef() { return "el"; }
}
