// @platformscode/core/dist/collection/components/dga-data-table/dga-data-table-usage.js
import { h } from "@stencil/core";
export class DgaDateFieldUsage {
    constructor() {
        this.value = '';
    }
    render() {
        return (h("div", { key: '3b496d87fdb6f58e21d63c7bc3670a7905f7caaf' }, h("h1", { key: '914901df4b9cc8d88004029840e7111f32319104' }, "Date Field Example"), h("div", { key: '652ea31f1acdecea7698590cb91df32419546af7', style: { padding: '50px' } }, h("div", { key: '3a004e735cda3eed66b73c888727f7bcfdb0264e', style: { display: 'flex', flexDirection: 'column', gap: '16px' } }, h("dga-data-table", { key: '23161a2440c0483481fc5f6df34a808b43aac2a9', cells: [
                {
                    colSpan: 1,
                    isFilter: true,
                    isSort: true,
                    propertyName: 'city',
                    sort: 'up',
                    title: 'city',
                    type: 'propertyName',
                },
                {
                    colSpan: 1,
                    isFilter: true,
                    isSort: true,
                    propertyName: 'price',
                    sort: 'down',
                    title: 'price',
                    type: 'propertyName',
                },
                {
                    colSpan: 1,
                    isFilter: true,
                    isSort: true,
                    propertyName: 'state',
                    title: 'state',
                    type: 'element', // <-- change this
                    El: row => h("dga-button", { label: row.state }, console.log(row, 'roooooow')), // <-- use a functional component
                },
            ], contained: true, data: [
                {
                    city: 'Dallas',
                    h_id: '3',
                    price: '162500',
                    selected: false,
                    state: 'TX',
                    zip: '75201',
                },
                {
                    city: 'Bevery Hills',
                    h_id: '4',
                    price: '319250',
                    selected: false,
                    state: 'CA',
                    zip: '90210',
                },
                {
                    city: 'New York',
                    h_id: '5',
                    price: '962500',
                    selected: false,
                    state: 'NY',
                    zip: '00010',
                },
            ], getSelectedRows: function vu() { }, pressOnFilter: function vu() { }, rowDivider: true, showSelectCheckBox: true })))));
    }
    static get is() { return "dga-data-table-usage"; }
    static get states() {
        return {
            "value": {}
        };
    }
}
