// @platformscode/core/dist/esm/dga-header-action-btn_11.entry.js
export { a as dga_header_action_btn, f as dga_nav_header, D as dga_nav_header_actions, e as dga_nav_header_link, c as dga_nav_header_logos, b as dga_nav_header_main, d as dga_nav_header_menu } from './dga-header-action-btn-BBEzb24w.js';
import { r as registerInstance, h, H as Host, a as getElement } from './index-D0dWnGvB.js';

const DgaSecondNavHeader = class {
    constructor(hostRef) {
        registerInstance(this, hostRef);
        this.variant = 'primary';
    }
    componentWillLoad() {
        this.el.childNodes.forEach((item) => {
            const el = item;
            if (el.tagName === 'DGA-SECOND-NAV-HEADER-CONTENT' && this.hideContent === true) {
                el.style.display = 'none';
            }
            if (el.tagName === 'DGA-SECOND-NAV-HEADER-ACTIONS' && this.hideActions === true) {
                el.style.display = 'none';
            }
            if (el.tagName === 'DGA-SECOND-NAV-HEADER-CONTENT' && el.attributes.getNamedItem('slot') === null) {
                el.setAttribute('slot', 'content');
            }
            if (el.tagName === 'DGA-SECOND-NAV-HEADER-ACTIONS' && el.attributes.getNamedItem('slot') === null) {
                el.setAttribute('slot', 'actions');
            }
        });
    }
    render() {
        return (h(Host, { key: '4d6a56396f788b3f018b3774c84fb6bc0d0849a9' }, h("div", { key: 'f6c6cf9410c08dfdab2c9889852649634b763dbe', class: `wrapper ${this.variant} `, "aria-label": 'a secondary navigation navbar' }, h("div", { key: '4b851ce02ce31b959365e7a09374097d369973ac', class: `container ${this.hideDivider ? '' : 'border'}` }, h("div", { key: '6214ce6582acd5ac51e252425546acd6a5edad46', class: `content` }, h("slot", { key: '3a70b02c824ba50ca99427169697341dc781c5d5', name: "content" })), h("div", { key: '2193cb09f2bd21e7741460d91ab3ab38c723370b', class: `actions` }, h("slot", { key: 'fa36abcb4f2e101b986b746af0dad2af3fee7fa4', name: "actions" }))))));
    }
    get el() { return getElement(this); }
};

const DgaSecondNavHeaderActions = class {
    constructor(hostRef) {
        registerInstance(this, hostRef);
    }
    render() {
        return (h(Host, { key: 'fec48112353fcd554b608780c7805c11ef383d51' }, h("div", { key: 'bb614b01662d01e7979638c6467014af2f5486e5', class: `items-container` }, h("slot", { key: '34dca111e83fd2489e3cb795dcf0083066a354a4' }))));
    }
};

const DgaSecondNavHeaderContent = class {
    constructor(hostRef) {
        registerInstance(this, hostRef);
    }
    render() {
        return (h(Host, { key: 'd9b732973272bc20a9407bfb291293ff8936a792' }, h("div", { key: 'dce382719b0b96c984dee57dd4b054325ff4834d', class: `items-container` }, h("slot", { key: '851ec4474a4630d627326c6daf4da23f635fd268' }))));
    }
};

const DgaSecondNavHeaderItem = class {
    constructor(hostRef) {
        registerInstance(this, hostRef);
    }
    connectedCallback() {
        this.el.childNodes.forEach((item) => {
            const el = item;
            if (el.tagName === 'DGA-ICON'.toLocaleUpperCase() && el.attributes.getNamedItem('slot') === null) {
                el.setAttribute('slot', 'icon');
            }
        });
    }
    render() {
        return (h(Host, { key: '87572880e04d67c78bbe24dce793d6147d166d37' }, h("slot", { key: 'e478e5be9f8231a25f8f76dc5b44b062552d3305', name: 'icon' }), h("slot", { key: 'bdf901885edd5f6edfe77ae63311748da8f6550e', name: 'label' }, this.label)));
    }
    get el() { return getElement(this); }
};

export { DgaSecondNavHeader as dga_second_nav_header, DgaSecondNavHeaderActions as dga_second_nav_header_actions, DgaSecondNavHeaderContent as dga_second_nav_header_content, DgaSecondNavHeaderItem as dga_second_nav_header_item };
