// @platformscode/core/dist/esm/dga-breadcrumbs-usage.entry.js
import { r as registerInstance, h } from './index-D0dWnGvB.js';

const DgaBreadcrumbsUsage = class {
    constructor(hostRef) {
        registerInstance(this, hostRef);
        this.variant = 'neutral';
    }
    render() {
        return (h("div", { key: '44bda33bc39a374cf77706c91cceefc24fe94c90' }, h("h1", { key: 'd66d37fa7491528933a433002a079eb31c08dfdb' }, "Breadcrumbs Example"), h("div", { key: '733e8c82c23d16d9b40b81e28d4abbaac2b388bd', style: { padding: '50px', display: 'flex', gap: '20px' } }, h("div", { key: '241e9274dfb9e6dc4016b847bdd4074fc128e62d', style: { display: 'flex', flexDirection: 'column', gap: '20px' } }, h("dga-link", { key: '72b9505073ac5998731071f698a0cbc6160d9947', label: "Link", variant: "neutral" }), h("dga-breadcrumbs", { key: 'c157e9c59c857e9bbbc75a7adbea2caa93daabec', items: [
                {
                    label: 'Home',
                    path: '/',
                },
                {
                    label: 'Products',
                    path: '/products',
                },
                {
                    label: 'Computers',
                    path: '/products/computers',
                    disabled: true,
                },
                {
                    label: 'Keyboards',
                    path: '/products/keyboards',
                },
            ], max: 3 })), h("div", { key: 'a5aed8b745081cb100e2766e0a69750f394226cf', style: { display: 'flex', flexDirection: 'column', gap: '20px' } }))));
    }
};

export { DgaBreadcrumbsUsage as dga_breadcrumbs_usage };
