// @platformscode/core/dist/esm/dga-menu.entry.js
import { r as registerInstance, h, F as Fragment } from './index-D0dWnGvB.js';

const DgaMenu = class {
    constructor(hostRef) {
        registerInstance(this, hostRef);
        this.anchorEl = null;
        this.open = false;
        this.onClose = () => { };
        // Method to detect outside click
        this.handleClickOutside = (event) => {
            if (this.anchorEl && !this.anchorEl.contains(event.target)) {
                this.onClose();
            }
        };
    }
    componentDidLoad() {
        // Add event listener to handle clicks outside the menu
        document.addEventListener('click', this.handleClickOutside);
    }
    disconnectedCallback() {
        // Cleanup event listener when the component is destroyed
        document.removeEventListener('click', this.handleClickOutside);
    }
    getMenuStyle() {
        var _a;
        const rect = (_a = this.anchorEl) === null || _a === void 0 ? void 0 : _a.getBoundingClientRect();
        return {
            position: 'absolute',
            top: `${(rect === null || rect === void 0 ? void 0 : rect.bottom) + window.scrollY}px`,
            left: `${(rect === null || rect === void 0 ? void 0 : rect.left) + window.scrollX}px`,
            zIndex: '1000',
        };
    }
    render() {
        return (h(Fragment, { key: '69bdc2e98bd65dfd11de009b1c1b3f514fee4dfa' }, this.open && this.anchorEl && (h("ul", { key: 'fafbdec7bf4ef68b4d79fb54d38883260665e154', class: "menu", style: this.getMenuStyle() }, h("slot", { key: '6fb989d29109ac200f53ff8a2d3ec7069797b702' })))));
    }
};

export { DgaMenu as dga_menu };
