// @platformscode/core/dist/esm/dga-codesnippet-usage.entry.js
import { r as registerInstance, h } from './index-D0dWnGvB.js';

const DgaChipUsage = class {
    constructor(hostRef) {
        registerInstance(this, hostRef);
        this.snippets = {
            JavaScript: `// Imports
import mongoose, { Schema } from 'untitled'

// Collection name
export const collection = 'Design'|

// Schema
const schema = new Schema({
  name: {
    type: String,
    required: true
  },

  description: {
    type: String
  }
}, {timestamps: true})

// Model
export default untitled.model(collection, schema, collection)
        `,
            HTML: `<div>
 <h1>Hello, HTML!</h1>
</div>`,
            CSS: `body {
    background-color: lightblue;
  }
    
h1 {
  color: navy;
  margin-left: 20px;
}`,
        };
        this.snippetLine = {
            JavaScript: 'npm install nds-design-system@^0.0.1',
        };
    }
    render() {
        return (h("div", { key: '8fec3258f454d167e0ff8814303754d0a764885e', style: { width: '743px', display: 'flex', flexDirection: 'column', gap: '123px', margin: "10px" } }, h("dga-codesnippet", { key: '505ca45d20f82e97e3a9e4186f1dd0f3ccd2728d', showButton: true, type: "multi-line", showSidebar: true, showTabList: true, snippets: this.snippets }), h("dga-codesnippet", { key: '72f1a4ab0fc27ed9f8d411b3b1ee0dbc34aa51d4', showButton: false, type: "single", showSidebar: false, showTabList: false, snippets: this.snippetLine })));
    }
};

export { DgaChipUsage as dga_codesnippet_usage };
