// @platformscode/core/dist/esm/dga-datepicker-usage.entry.js
import { r as registerInstance, h } from './index-D0dWnGvB.js';

const DgaPickerUsage = class {
    constructor(hostRef) {
        registerInstance(this, hostRef);
    }
    render() {
        return (h("div", { key: 'b52ca0efce71204aed2adaf89202318f0edcd628' }, h("h1", { key: 'd3bb55b0ed385369e9e4b8c84b5ac32d228274a3' }, "Datepicker Example"), h("div", { key: '9bb4e267a566d78e1233515beae77a9de35aedc6', style: { padding: '50px', display: 'flex', gap: '20px' } }, h("div", { key: '8b01d9914510e772ddea427795d9cb633fd85a5f', style: { display: 'flex', flexDirection: 'column', gap: '20px' } }, h("dga-datepicker", { key: 'ed93d5b6dd71651203cf68015dc6dc6daa5d5bcb', inputField: false, qickOptions: true }), h("dga-datepicker", { key: 'a14cd34017dca39a1ad597f7568df29fe20b08f2', RTL: true, inputField: true, range: true, qickOptions: true }), h("dga-datepicker", { key: '65360941d011ac7b4969a3346bcf3ec3e42f3393', range: true })), h("div", { key: 'b86ca5a1008bbce275ffbb1e2e1751e8c0f75bac', style: { display: 'flex', flexDirection: 'column', gap: '20px' } }))));
    }
};

export { DgaPickerUsage as dga_datepicker_usage };
