// @platformscode/core/dist/esm/dga-content-switcher_3.entry.js
import { r as registerInstance, h } from './index-D0dWnGvB.js';

const DgaContentSwitcher = class {
    constructor(hostRef) {
        registerInstance(this, hostRef);
        this.value = 0;
        this.onColor = false;
        this.size = 'md';
        this.items = [
            {
                label: 'Item 1',
                content: 'Content 1',
            },
            {
                label: 'Item 2',
                content: 'Content 2',
            },
            {
                label: 'Item 3',
                content: 'Content 3',
            },
        ];
        this.onChange = () => { };
        this.selected = this.value;
        this.handleSelect = (index) => {
            this.selected = index;
            this.onChange(index);
        };
    }
    render() {
        return (h("div", { key: 'df812ae9df77430a69ca8a7e75af461d1ef9db32', class: "content-switcher" }, h("div", { key: '4f0ce96404f552c8a70e3298711ecc63bd8a16fb', class: "content-switcher__items" }, this.items.map((item, index) => {
            const selected = index === this.selected;
            const itemType = index === 0 ? 'first' : index === this.items.length - 1 ? 'last' : 'mid';
            const className = `content-switcher__item content-switcher__item--${itemType} ${this.onColor ? 'content-switcher__item--on-color' : ''} content-switcher__item--${this.size} ${selected ? 'content-switcher__item--selected' : ''}`;
            return (h("button", { key: index, type: "button", class: className, onClick: () => this.handleSelect(index) }, item.label));
        }))));
    }
};

const DgaContentSwitcherItem = class {
    constructor(hostRef) {
        registerInstance(this, hostRef);
    }
    render() {
        return h("slot", { key: '419881152c56e0a76e1033a6301cd69a2248d13c' });
    }
};

const DgaContentSwitcherPanel = class {
    constructor(hostRef) {
        registerInstance(this, hostRef);
    }
    render() {
        return (h("div", { key: '5445376097d8e1d28fc5e1d81420b7200e121725', class: "content-switcher-panel", hidden: this.value !== this.index }, h("slot", { key: '2d959f34d533260049284c700a230859d584321a' })));
    }
};

export { DgaContentSwitcher as dga_content_switcher, DgaContentSwitcherItem as dga_content_switcher_item, DgaContentSwitcherPanel as dga_content_switcher_panel };
