// @platformscode/core/dist/esm/dga-button-usage.entry.js
import { r as registerInstance, h } from './index-D0dWnGvB.js';

const DgaButtonUsage = class {
    constructor(hostRef) {
        registerInstance(this, hostRef);
        this.variant = 'neutral';
    }
    render() {
        return (h("div", { key: '19a2433057f8dbc17a692e155d588c8bb24dcff8' }, h("h1", { key: 'e16a447fa9ffd97086aaff516809af8ea730f3be' }, "Button Example"), h("div", { key: 'ef001c1af314ce4875055d6fc1ab55643350ab53', style: { padding: '50px', display: 'flex', gap: '20px' } }, h("div", { key: '736e697fcb99c2251d5f5c5d6a030787e8a522e4', style: { display: 'flex', flexDirection: 'column', gap: '20px' } }, h("dga-button", { key: 'ec8acf0ae303f9b209c32e178d45d14f749e4431', label: "Click me", variant: "primary-brand", sx: {
                '&': {
                    fontSize: '18px',
                    fontWeight: 'bold',
                },
                '.dga-btn': {
                    backgroundColor: { xs: 'red', sm: 'blue', md: 'green' },
                    // backgroundColor: 'yellow',
                    color: { xs: 'black', sm: 'yellow', md: 'white' },
                },
                '.dga-btn:hover': {
                    backgroundColor: 'blue',
                },
            } }), h("dga-button", { key: '122dbe3588a7bdb0b76f4c1a2dd76ed63d28a641', label: "Brown Button", variant: "primary-neutral" }), h("dga-button", { key: '13f2aa7abaeee15f05819f22ba06f11919467874', label: "Button", variant: "des-transparent" }), h("dga-button", { key: '24909b5400736c140977cc90cfcf6e859b8730d5', label: "Source code", variant: "secondary-outline", leadIcon: true, leadIconType: "source-code", leadIconProps: { variant: 'stroke', type: 'standard' } }), h("dga-button", { key: '5a636b64652a28c8c30a50573d26a3f5abee0ba2',
            // className="absolute top-[50%] left-[50%] transform -translate-x-1/2 -translate-y-1/2 opacity-0 group-hover:opacity-100"
            label: "See live preview", size: "lg", variant: "secondary-outline", trailIcon: true, trailIconType: "link-square-01", trailIconProps: {
                variant: 'stroke',
                type: 'standard',
            } })), h("div", { key: '4d55ae90ec8d1e3c3733e481558a43c80e2db70f', style: { display: 'flex', flexDirection: 'column', gap: '20px' } }))));
    }
};

export { DgaButtonUsage as dga_button_usage };
