// @platformscode/core/dist/esm/dga-link-usage.entry.js
import { r as registerInstance, h } from './index-D0dWnGvB.js';

const DgaLinkUsage = class {
    constructor(hostRef) {
        registerInstance(this, hostRef);
        this.variant = 'neutral';
    }
    render() {
        return (h("div", { key: '1898bc03e9d833fc0c4913d365d8af85a2e36c4b' }, h("h1", { key: 'd32db75cb3de7a2d715139bdb09fc93d78ba3b32' }, "Link Example"), h("div", { key: 'ef0c7e0958c7712250d5721ce373bf8ab1b3ad37', style: { padding: '50px', display: 'flex', gap: '20px' } }, h("div", { key: 'd5e97c3a298c1b2c95a9514e0cc2b3864f9d0221', style: { display: 'flex', flexDirection: 'column', gap: '20px' } }, h("dga-link", { key: '597c010fcb51b1763d5dfc908d4a495dd03f8fe4', label: "Link", variant: "neutral" }), h("dga-link", { key: '1555954082c510e2069cf3fca71fe68c12cffc5e', label: "Link", variant: "on-color" }), h("dga-link", { key: 'ae9b97800b1679967b4eb0127eed7487535f4f7f', label: "Link", variant: "primary" })), h("div", { key: '7b114ebb9de94a92a67101ffd2c84afe0e43a607', style: { display: 'flex', flexDirection: 'column', gap: '20px' } }))));
    }
};

export { DgaLinkUsage as dga_link_usage };
