// @platformscode/core/dist/esm/dga-list.entry.js
import { r as registerInstance, h } from './index-D0dWnGvB.js';

const DgaList = class {
    constructor(hostRef) {
        registerInstance(this, hostRef);
        this.type = 'unordered';
        this.color = 'primary';
        //   @Prop() id: string = guid();
        this.title = "";
        this.items = [];
        this.renderIcon = (type) => {
            if (type === "with-icon") {
                return h("span", { class: "list-icon" }, "\u2713"); // Using checkmark icon for "with-icon" type
            }
            return null;
        };
    }
    render() {
        const Tag = this.type === 'ordered' ? 'ol' : 'ul';
        return (h("div", { key: '5082f2e16a2d7ca0c5e032c559bd159e176b33cf', class: `list list--${this.type} list--${this.color}` }, this.title && h("h3", { key: '9cdc61f9060b9c549c777fb260ddb1076d03e8d3', class: "list-title" }, this.title), h(Tag, { key: '45d79b00364e130ce5dc047c23047cb06cd9290d' }, this.items.map((item, index) => (h("li", { key: index, class: "list-item" }, this.type === "with-icon" && this.renderIcon(this.type), item))))));
    }
};

export { DgaList as dga_list };
