// @platformscode/core/dist/esm/dga-helper-text_3.entry.js
import { r as registerInstance, h, H as Host, a as getElement } from './index-D0dWnGvB.js';
import { g as guid } from './index-DEtoWGOc.js';
import { S as SxStyleManager } from './sx-style-manager-BFEqPAFn.js';

const DgaHelperText = class {
    constructor(hostRef) {
        registerInstance(this, hostRef);
        this.helperText = '';
        this.error = false;
    }
    render() {
        var _a;
        return (h("div", { key: '9ce77fa2f456e20064b153aef3c1fa436563eb49', class: `dga-helper-text ${this.error ? "dga-helper-text--error" : ""}` }, h("span", { key: '07990ee96959b34986170620b219194844e30970', class: "dga-helper-text__icon" }, this.helperTextIcon && ((_a = this.helperTextIcon) === null || _a === void 0 ? void 0 : _a.name) && h("dga-icon", { key: '702f2185be7544b679b543a063f0c76049cbe7e3', icon: this.helperTextIcon.name, variant: this.helperTextIcon.variant, type: this.helperTextIcon.type, size: 16 })), h("span", { key: '5fb2b0ceacdda88772a4ade8ac9e94a514e79a22', class: "dga-helper-text__desc" }, this.helperText)));
    }
};

const DgaLabel = class {
    constructor(hostRef) {
        registerInstance(this, hostRef);
        this.label = '';
        this.variant = 'default';
        this.size = 'md';
        this.disabled = false;
        this.required = false;
    }
    render() {
        return (h("label", { key: '4d40b590e3045b65d31807a4a02cbffdd5cb0a76', class: `dga-label dga-label--${this.size} ${this.disabled ? "disabled" : ""}` }, this.required && h("span", { key: '5a5decc5839d7ce035ec25dc84d492cf7e92dc6e', class: "star" }, "*"), this.label));
    }
};

const feedbackIconTypes = {
    success: h("dga-icon", { icon: "CheckmarkCircle02Icon", variant: "solid" }),
    error: h("dga-icon", { icon: "AlertCircleIcon", variant: "solid" }),
    warning: h("dga-icon", { icon: "Alert02Icon", variant: "solid" }),
};
const DgaTextInput = class {
    constructor(hostRef) {
        registerInstance(this, hostRef);
        this.size = 'lg';
        this.type = 'text';
        this.variant = 'default';
        this.error = false;
        this.feedbackIcon = false;
        this.feedbackIconType = 'success';
        this.feedbackIconRing = false;
        this.helperText = '';
        this.disabled = false;
        this.readonly = false;
        this.required = false;
        this.fullwidth = true;
        //   @Prop() id: string = guid();
        this.name = guid();
        this.value = '';
        this.isActive = false;
        this.isFocused = false;
        // Generate a unique ID if not provided
        this.componentId = `dga-button-${Math.random().toString(36).substring(2, 9)}`;
        // Computed inline styles
        this.rootStyles = {};
    }
    valueChanged(newValue, oldValue) {
        if (newValue !== oldValue) {
            this.textInputEl['value'] = newValue;
        }
    }
    handleSxChange() {
        this.processStyling();
    }
    componentWillLoad() {
        // Initialize style manager with element reference
        this.sxStyleManager = new SxStyleManager(this.componentId, this.el);
    }
    componentDidLoad() {
        // Process styling after the component is fully loaded and shadow DOM is ready
        this.processStyling();
    }
    componentDidUpdate() {
        // Re-process styling after updates
        this.processStyling();
    }
    processStyling() {
        if (!this.sx)
            return;
        // Handle both string and object formats
        const sxValue = typeof this.sx === 'string' ? JSON.parse(this.sx) : this.sx;
        this.rootStyles = this.sxStyleManager.processSxProp(sxValue);
    }
    disconnectedCallback() {
        var _a;
        // Clean up styles when component is removed
        (_a = this.sxStyleManager) === null || _a === void 0 ? void 0 : _a.cleanup();
    }
    render() {
        return (h(Host, { key: '0858f48662c1dec58bcf2b35f63501be68c9344a', id: this.componentId, style: this.rootStyles }, h("div", { key: '6e1f1d8717fcc5f8c1ae3da680f21075308d4563', class: { "dga-form-control": true, "dga-form-control--fullwidth": this.fullwidth } }, this.label && h("dga-label", { key: 'd957b562a4e04c7f93be7e2fdf07218e26270b62', label: this.label, size: this.size, required: this.required, disabled: this.disabled }), h("div", { key: '4cff6669f027958bad72f61ea825d3c68d50390b', class: `input input--${this.size} ${this.error ? 'input--error' : ''} ${this.variant !== 'default' ? `input--${this.variant}` : ''} ${this.isActive ? 'active' : ''} ${this.isFocused ? 'focus' : ''} ${this.disabled ? 'input--disabled' : ''} ${this.readonly ? 'readOnly' : ''}`, ref: el => (this.inputContainerRef = el) }, h("slot", { key: '6a72c1f0d8d1c7adf4e1813c0f17d11ee78ba0c7', name: "prefix" }), this.icon && (h("span", { key: '4126803f5157f8baca3ac2b7e31f0cda7434cf06', class: "input__icon" }, h("dga-icon", { key: '0119b859ecf61cb4e8f0a4b4844223646d215607', icon: this.icon.name, variant: this.icon.variant, type: this.icon.type, size: 20 }))), h("input", { key: 'a1011a70fef6d5f31c5e03467f84dbdfb290e34d',
            // id={id}
            ref: (el) => (this.textInputEl = el), value: this.value, name: this.name, type: this.type, onFocus: () => (this.isFocused = true), onBlur: e => {
                this.isFocused = false;
                this.onBlur(e);
            }, class: "input__field", placeholder: this.placeholder, disabled: this.disabled, readOnly: this.readonly, required: this.required, onMouseDown: () => (this.isActive = true), onMouseUp: () => (this.isActive = false), onMouseOut: () => (this.isActive = false), onChange: this.onChange, onInput: this.onInput }), this.error ? (h("span", { class: `input__feedback-icon input__feedback-icon--error ${this.feedbackIconRing ? 'input__feedback-icon--ring' : ''}` })) : this.feedbackIcon ? (h("span", { class: `input__feedback-icon input__feedback-icon--${this.feedbackIconType} ${this.feedbackIconRing ? 'input__feedback-icon--ring' : ''}` }, feedbackIconTypes[this.feedbackIconType])) : null, h("slot", { key: '068446d038a2bf4de9a072898eba807005994cac', name: "suffix" }), " "), this.helperText && h("dga-helper-text", { key: 'cf4e6eb83f1c4534b33b574301f0f4eba46c133b', helperText: this.helperText, helperTextIcon: this.helperTextIcon, error: this.error }))));
    }
    get el() { return getElement(this); }
    static get watchers() { return {
        "value": ["valueChanged"],
        "sx": ["handleSxChange"]
    }; }
};

export { DgaHelperText as dga_helper_text, DgaLabel as dga_label, DgaTextInput as dga_text_input };
