// @platformscode/core/dist/esm/dga-notification-toast-usage.entry.js
import { r as registerInstance, h } from './index-D0dWnGvB.js';

const DgaNotificationToastUsage = class {
    constructor(hostRef) {
        registerInstance(this, hostRef);
        this.type = 'success';
        this.open = false;
    }
    render() {
        return (h("div", { key: '188a74c0fa1d8de5875ea34f8c7367f8d77e21bd' }, h("h1", { key: 'f3b86d0d39b29efb3933f39ec579f1dfd0a48d27' }, "Notification Toast Example"), h("div", { key: '2172f6611d2493fae16852d57fe681fcc2691dac', style: { padding: '50px', display: 'flex', gap: '20px' } }, h("div", { key: '7ba38cdc002db66e44071cf683a48ce89e131ece', style: { display: 'flex', flexDirection: 'column', gap: '20px' } }, h("div", { key: '8f685a4f65db7e3bb6b9fafaa1946fbd1a067999', style: { display: "flex", justifyContent: "center" } }, h("button", { key: '3fbcfcad180ce3f43b52bd1dd7bb6e4807d221b9', onClick: () => this.open = true }, "open"), h("button", { key: '3076b1d47f44ccb30a4ef51a9c0c19c20c91fdb1', onClick: () => this.open = false }, "close")), h("dga-notification-toast", { key: '1d8bfccd839df4eeab7d742f65b04043b435806f', type: this.type, open: this.open, leadText: "Notification/Alert message title", hasHelperText: true, helperText: "When a Notification/Alert needs a further detailed explanation, it goes here.", action: true, buttonsList: [
                {
                    label: 'Button',
                    onClick: () => { },
                    title: 'Button 2',
                },
                {
                    label: 'Button',
                    onClick: () => { },
                    title: 'Button',
                },
            ], closeButton: true, vPostion: 'top', hPostion: 'left', onClose: () => this.open = false }), h("dga-notification-toast", { key: '71672e3d602f6706f5066b1139d4d891739df790', type: this.type, open: this.open, leadText: "Notification/Alert message title", hasHelperText: true, helperText: "When a Notification/Alert needs a further detailed explanation, it goes here.", action: true, buttonsList: [
                {
                    label: 'Button',
                    onClick: () => { },
                    title: 'Button 2',
                },
                {
                    label: 'Button',
                    onClick: () => { },
                    title: 'Button',
                },
            ], closeButton: true, vPostion: 'bottom', hPostion: 'left', onClose: () => this.open = false })), h("div", { key: 'd93ead6559e00717b74bba7fdd73882ba682a715', style: { display: 'flex', flexDirection: 'column', gap: '20px' } }))));
    }
};

export { DgaNotificationToastUsage as dga_notification_toast_usage };
