// @platformscode/core/dist/esm/nds-chip-usage.entry.js
import { r as registerInstance, h } from './index-D0dWnGvB.js';

const NdsChipUsage = class {
    constructor(hostRef) {
        registerInstance(this, hostRef);
        this.isSelected = false;
    }
    render() {
        return (h("div", { key: 'ba5570a6802264a3e6b19c4dd2da122130d15872' }, h("h1", { key: '2dfdad0efc228fa1f5c9ff05447c2e4eb235c34a' }, "Chip Example"), h("div", { key: '6485034d6faf2d57bbf42419ec33915871c248ab', style: { padding: '50px', display: 'flex', gap: '20px' } }, h("div", { key: '5896a0e99e12f8b43f5dc59aaacdcda1a97eb421', style: { display: 'flex', flexDirection: 'column', gap: '20px' } }, h("nds-chip", { key: '7b2add965a905f4dab42157e22ca94b36430c7fc', label: "Label", variant: "neutral",
            //   outlined={false}
            rounded: false, size: "lg" }), h("nds-chip", { key: 'a000edc0619b19cc48e40ae73a5caa8824de2921', label: "Label", variant: "success", rounded: false, size: "lg", onChange: isSelected => {
                console.log(isSelected, 'hi');
                this.isSelected = isSelected;
            }, selected: this.isSelected }), h("nds-chip", { key: '96dd2014d250fb2ca36faec9c0b972cdc0f39902', label: "Label", variant: "success", rounded: false, size: "lg", onChange: isSelected => {
                console.log(isSelected, 'hi');
            }, selected: false }), h("nds-chip", { key: '12c9d4459a6228203dcdc57ae39f8f5aaba353ab', label: "Label", variant: "success", rounded: false, size: "lg", disabled: true }), h("div", { key: '2fb23a19bfe696a13d0c693c164976fcd57eade4', style: { display: 'flex', flexDirection: 'column', gap: '20px', backgroundColor: '#074D31' } }, h("nds-chip", { key: '442c2c705ef14663d2dd9377278aa5502683c2cf', label: "Label", variant: "on-color", rounded: true, size: "lg", disabled: true }))))));
    }
};

export { NdsChipUsage as nds_chip_usage };
