// @platformscode/core/dist/esm/dga-metric-usage.entry.js
import { r as registerInstance, h } from './index-B4CGwdkg.js';

const DgaMetricUsage = class {
    constructor(hostRef) {
        registerInstance(this, hostRef);
    }
    //   @State() variant: 'critical' | 'warning' | 'success' | 'info' | 'neutral' = 'neutral';
    render() {
        return (h("div", { key: '6ad33030f3c6b2bfa0d5a6335dc7da2fb6329344' }, h("h1", { key: '840c70c1184b6cef0ff244625f9d0f81e762b57e' }, "Metric Example"), h("div", { key: '254ba7be1413a89752672db5db2d9bc8513e2ec2', style: { padding: '50px', display: 'flex', gap: '20px' } }, h("div", { key: '37b2eee502974e1baa435385d3493286a2821f24', style: { display: 'flex', flexDirection: 'column', gap: '20px' } }, h("dga-apex-chart", { key: '64d9cec17f91b0caf327bf37e9e23ca73920592a',
            // type="donut" 
            //   series={ [{
            //     name: "Desktops",
            //     data: [10, 41, 35, 51, 49, 62, 69, 91, 148]
            // }]} 
            // labels={['Apple', 'Mango', 'Orange', 'Watermelon']}
            options: {
                series: [{
                        name: 'series1',
                        data: [31, 40, 28, 51, 42, 109, 100]
                    }],
                chart: {
                    height: 350,
                    type: 'area',
                    toolbar: {
                        show: false // Hides the toolbar
                    },
                    zoom: {
                        enabled: false // Disables zooming
                    },
                },
                dataLabels: {
                    enabled: false
                },
                // markers:{
                //   shape:"circle",
                //   size:6,
                //   strokeColors:"#067647",
                //   strokeWidth:2,
                //   fillOpacity: 1, // Simulates inner fill opacity
                //   colors:"#fff",
                // },
                stroke: {
                    curve: 'stepline'
                },
                fill: {
                    type: 'gradient', // Options: 'solid', 'gradient'
                    gradient: {
                        shade: 'light',
                        type: "vertical", // Options: 'horizontal', 'vertical'
                        gradientToColors: ['#fff'], // End color of the gradient
                        stops: [0, 100] // Color distribution
                    },
                    colors: ['#079455'] // Start color
                },
                // xaxis: {
                //   type: 'datetime',
                //   categories: ["2018-09-19T00:00:00.000Z", "2018-09-19T01:30:00.000Z", "2018-09-19T02:30:00.000Z", "2018-09-19T03:30:00.000Z", "2018-09-19T04:30:00.000Z", "2018-09-19T05:30:00.000Z", "2018-09-19T06:30:00.000Z"]
                // },
                // tooltip: {
                //   x: {
                //     format: 'dd/MM/yy HH:mm'
                //   },
                // },
                xaxis: {
                    labels: {
                        show: false // Hides X-axis labels
                    },
                    axisBorder: {
                        show: false // Hides the X-axis border line
                    },
                    axisTicks: {
                        show: false // Hides the X-axis ticks
                    }
                },
                yaxis: {
                    labels: {
                        show: false // Hides X-axis labels
                    },
                    axisBorder: {
                        show: false // Hides the X-axis border line
                    },
                    axisTicks: {
                        show: false // Hides the X-axis ticks
                    }
                },
                // xaxis: {
                //   show: false // Hides the X-axis completely
                // },
                // yaxis: {
                //   show: false // Hides the Y-axis completely
                // },
                grid: {
                    show: false
                },
                // tooltip:{
                //   enabled: false
                // }
                legend: {
                    show: false
                }
            } }), h("dga-apex-chart", { key: '983adc91525b44d33f817c98cb33f1207b47c397',
            // type="donut" 
            //   series={ [{
            //     name: "Desktops",
            //     data: [10, 41, 35, 51, 49, 62, 69, 91, 148]
            // }]} 
            // labels={['Apple', 'Mango', 'Orange', 'Watermelon']}
            options: {
                series: [{
                        name: 'series1',
                        data: [31, 40, 28, 51, 42, 109, 100]
                    }],
                chart: {
                    height: 350,
                    type: 'area',
                    toolbar: {
                        show: false // Hides the toolbar
                    },
                    zoom: {
                        enabled: false // Disables zooming
                    },
                },
                dataLabels: {
                    enabled: false
                },
                // markers:{
                //   shape:"circle",
                //   size:6,
                //   strokeColors:"#067647",
                //   strokeWidth:2,
                //   fillOpacity: 1, // Simulates inner fill opacity
                //   colors:"#fff",
                // },
                stroke: {
                    curve: 'straight'
                },
                fill: {
                    type: 'gradient', // Options: 'solid', 'gradient'
                    gradient: {
                        shade: 'light',
                        type: "vertical", // Options: 'horizontal', 'vertical'
                        gradientToColors: ['#fff'], // End color of the gradient
                        stops: [0, 100] // Color distribution
                    },
                    colors: ['#079455'] // Start color
                },
                // xaxis: {
                //   type: 'datetime',
                //   categories: ["2018-09-19T00:00:00.000Z", "2018-09-19T01:30:00.000Z", "2018-09-19T02:30:00.000Z", "2018-09-19T03:30:00.000Z", "2018-09-19T04:30:00.000Z", "2018-09-19T05:30:00.000Z", "2018-09-19T06:30:00.000Z"]
                // },
                // tooltip: {
                //   x: {
                //     format: 'dd/MM/yy HH:mm'
                //   },
                // },
                xaxis: {
                    labels: {
                        show: false // Hides X-axis labels
                    },
                    axisBorder: {
                        show: false // Hides the X-axis border line
                    },
                    axisTicks: {
                        show: false // Hides the X-axis ticks
                    }
                },
                yaxis: {
                    labels: {
                        show: false // Hides X-axis labels
                    },
                    axisBorder: {
                        show: false // Hides the X-axis border line
                    },
                    axisTicks: {
                        show: false // Hides the X-axis ticks
                    }
                },
                // xaxis: {
                //   show: false // Hides the X-axis completely
                // },
                // yaxis: {
                //   show: false // Hides the Y-axis completely
                // },
                grid: {
                    show: false
                },
                // tooltip:{
                //   enabled: false
                // }
                legend: {
                    show: false
                }
            } }), h("dga-apex-chart", { key: '39b62eccb5959336769307c7c89864e6aaa41cc1',
            // type="donut" 
            //   series={ [{
            //     name: "Desktops",
            //     data: [10, 41, 35, 51, 49, 62, 69, 91, 148]
            // }]} 
            // labels={['Apple', 'Mango', 'Orange', 'Watermelon']}
            options: {
                series: [{
                        name: 'series1',
                        data: [31, 40, 28, 51, 42, 109, 100]
                    }],
                chart: {
                    height: 350,
                    type: 'area',
                    toolbar: {
                        show: false // Hides the toolbar
                    },
                    zoom: {
                        enabled: false // Disables zooming
                    },
                },
                dataLabels: {
                    enabled: false
                },
                // markers:{
                //   shape:"circle",
                //   size:6,
                //   strokeColors:"#067647",
                //   strokeWidth:2,
                //   fillOpacity: 1, // Simulates inner fill opacity
                //   colors:"#fff",
                // },
                stroke: {
                    curve: 'smooth'
                },
                fill: {
                    type: 'gradient', // Options: 'solid', 'gradient'
                    gradient: {
                        shade: 'light',
                        type: "vertical", // Options: 'horizontal', 'vertical'
                        gradientToColors: ['#fff'], // End color of the gradient
                        stops: [0, 100] // Color distribution
                    },
                    colors: ['#079455'] // Start color
                },
                // xaxis: {
                //   type: 'datetime',
                //   categories: ["2018-09-19T00:00:00.000Z", "2018-09-19T01:30:00.000Z", "2018-09-19T02:30:00.000Z", "2018-09-19T03:30:00.000Z", "2018-09-19T04:30:00.000Z", "2018-09-19T05:30:00.000Z", "2018-09-19T06:30:00.000Z"]
                // },
                // tooltip: {
                //   x: {
                //     format: 'dd/MM/yy HH:mm'
                //   },
                // },
                xaxis: {
                    labels: {
                        show: false // Hides X-axis labels
                    },
                    axisBorder: {
                        show: false // Hides the X-axis border line
                    },
                    axisTicks: {
                        show: false // Hides the X-axis ticks
                    }
                },
                yaxis: {
                    labels: {
                        show: false // Hides X-axis labels
                    },
                    axisBorder: {
                        show: false // Hides the X-axis border line
                    },
                    axisTicks: {
                        show: false // Hides the X-axis ticks
                    }
                },
                // xaxis: {
                //   show: false // Hides the X-axis completely
                // },
                // yaxis: {
                //   show: false // Hides the Y-axis completely
                // },
                grid: {
                    show: false
                },
                // tooltip:{
                //   enabled: false
                // }
                legend: {
                    show: false
                }
            } }), h("dga-apex-chart", { key: '3d5839ca4a8020f42bdd8f70af1d26966a31ab3c',
            // type="donut" 
            //   series={ [{
            //     name: "Desktops",
            //     data: [10, 41, 35, 51, 49, 62, 69, 91, 148]
            // }]} 
            // labels={['Apple', 'Mango', 'Orange', 'Watermelon']}
            options: {
                series: [{
                        name: 'series1',
                        data: [31, 40, 28, 51, 42, 109, 100]
                    }, {
                        name: 'series2',
                        data: [11, 32, 45, 32, 34, 52, 41]
                    }],
                chart: {
                    height: 350,
                    type: 'area',
                    toolbar: {
                        show: false // Hides the toolbar
                    },
                    zoom: {
                        enabled: false // Disables zooming
                    },
                },
                dataLabels: {
                    enabled: false
                },
                // markers:{
                //   shape:"circle",
                //   size:6,
                //   strokeColors:"#067647",
                //   strokeWidth:2,
                //   fillOpacity: 1, // Simulates inner fill opacity
                //   colors:"#fff",
                // },
                stroke: {
                    curve: 'smooth'
                },
                fill: {
                    type: 'gradient', // Options: 'solid', 'gradient'
                    gradient: {
                        shade: 'light',
                        type: "vertical", // Options: 'horizontal', 'vertical'
                        gradientToColors: ['#fff'], // End color of the gradient
                        stops: [0, 100] // Color distribution
                    },
                    colors: ['#079455'] // Start color
                },
                // xaxis: {
                //   type: 'datetime',
                //   categories: ["2018-09-19T00:00:00.000Z", "2018-09-19T01:30:00.000Z", "2018-09-19T02:30:00.000Z", "2018-09-19T03:30:00.000Z", "2018-09-19T04:30:00.000Z", "2018-09-19T05:30:00.000Z", "2018-09-19T06:30:00.000Z"]
                // },
                // tooltip: {
                //   x: {
                //     format: 'dd/MM/yy HH:mm'
                //   },
                // },
                xaxis: {
                    labels: {
                        show: false // Hides X-axis labels
                    },
                    axisBorder: {
                        show: false // Hides the X-axis border line
                    },
                    axisTicks: {
                        show: false // Hides the X-axis ticks
                    }
                },
                yaxis: {
                    labels: {
                        show: false // Hides X-axis labels
                    },
                    axisBorder: {
                        show: false // Hides the X-axis border line
                    },
                    axisTicks: {
                        show: false // Hides the X-axis ticks
                    }
                },
                // xaxis: {
                //   show: false // Hides the X-axis completely
                // },
                // yaxis: {
                //   show: false // Hides the Y-axis completely
                // },
                grid: {
                    show: false
                },
                // tooltip:{
                //   enabled: false
                // }
                legend: {
                    show: false
                }
            } })), h("div", { key: '7d01de739fde8ac5ae26cf84af841c01ca1ce01a', style: { display: 'flex', flexDirection: 'column', gap: '20px' } }))));
    }
};

export { DgaMetricUsage as dga_metric_usage };
