// @platformscode/core/dist/esm/dga-header-menu-item-usage.entry.js
import { r as registerInstance, h, F as Fragment } from './index-B4CGwdkg.js';

const SidePanelRoutes = [
    {
        name: 'Link',
        path: '/',
        icon: {
            name: 'home-01',
            variant: 'stroke',
            size: 16,
        },
        disabled: false,
        children: [
            {
                name: 'Link',
                path: '1',
                // icon: <i class='hgi-stroke hgi-arrow-down-01' style={{fontSize:"16px"}} />,
                // badge: 39,
            },
            {
                name: 'Link',
                path: '2',
                // icon: <i class='hgi-stroke hgi-arrow-down-01' style={{fontSize:"16px"}} />,
                // badge: 21,
            },
            {
                name: 'Link',
                path: '3',
                // icon: <i class='hgi-stroke hgi-arrow-down-01' style={{fontSize:"16px"}} />,
                // badge: 2,
            },
            // {
            //   name: "Home 4",
            //   path: "4",
            //   icon: <i class='hgi-stroke hgi-arrow-down-01' style={{fontSize:"16px"}} />,
            //   badge: 20,
            // },
            // {
            //   name: "Home 5",
            //   path: "5",
            //   icon: <i class='hgi-stroke hgi-arrow-down-01' style={{fontSize:"16px"}} />,
            //   badge: 150,
            //   children: [
            //     {
            //       name: "Home 1",
            //       path: "1",
            //       icon: <i class='hgi-stroke hgi-arrow-down-01' style={{fontSize:"16px"}} />,
            //       badge: 39,
            //       level: 2,
            //     },
            //     {
            //       name: "Home 4",
            //       path: "4",
            //       icon: <i class='hgi-stroke hgi-arrow-down-01' style={{fontSize:"16px"}} />,
            //       badge: 20,
            //       level: 2,
            //     },
            //     {
            //       name: "Home 5",
            //       path: "5",
            //       icon: <i class='hgi-stroke hgi-arrow-down-01' style={{fontSize:"16px"}} />,
            //       badge: 150,
            //       level: 2,
            //     },
            //     {
            //       name: "Home 6",
            //       path: "6",
            //       icon: <i class='hgi-stroke hgi-arrow-down-01' style={{fontSize:"16px"}} />,
            //       badge: 200,
            //       level: 2,
            //     },
            //     {
            //       name: "Home 7",
            //       path: "7",
            //       icon: <i class='hgi-stroke hgi-arrow-down-01' style={{fontSize:"16px"}} />,
            //       badge: 24,
            //       level: 2,
            //     },
            //   ],
            // },
            // {
            //   name: "Home 6",
            //   path: "6",
            //   icon: <i class='hgi-stroke hgi-arrow-down-01' style={{fontSize:"16px"}} />,
            //   badge: 200,
            // },
            // {
            //   name: "Home 7",
            //   path: "7",
            //   icon: <i class='hgi-stroke hgi-arrow-down-01' style={{fontSize:"16px"}} />,
            //   badge: 24,
            // },
        ],
    },
    {
        name: 'Components',
        path: '/components',
        icon: {
            name: 'home-01',
            variant: 'stroke',
            size: 16,
        },
        disabled: false,
        children: [
            {
                name: 'Buttons',
                path: '/buttons',
                icon: {
                    name: 'home-01',
                    variant: 'stroke',
                    size: 16,
                },
                badge: 100,
            },
            {
                name: 'Badges',
                path: '/badges',
                icon: {
                    name: 'home-01',
                    variant: 'stroke',
                    size: 16,
                },
                badge: 100,
            },
            {
                name: 'Paths',
                path: '/Paths',
                children: [
                    {
                        name: 'Buttons',
                        path: '/buttons',
                        icon: {
                            name: 'home-01',
                            variant: 'stroke',
                            size: 16,
                        },
                        badge: 100,
                    },
                    {
                        name: 'Badges',
                        path: '/badges',
                        icon: {
                            name: 'home-01',
                            variant: 'stroke',
                            size: 16,
                        },
                        badge: 100,
                    },
                    {
                        name: 'Paths',
                        path: '/Paths',
                        icon: {
                            name: 'home-01',
                            variant: 'stroke',
                            size: 16,
                        },
                        badge: 100,
                    },
                ],
            },
            {
                name: 'Inputs',
                path: '/inputs',
                icon: {
                    name: 'home-01',
                    variant: 'stroke',
                    size: 16,
                },
                badge: 100,
            },
            {
                name: 'Accordions',
                path: '/accordions',
                icon: {
                    name: 'home-01',
                    variant: 'stroke',
                    size: 16,
                },
                badge: 100,
            },
            {
                name: 'Tabs',
                path: '/tabs',
                icon: {
                    name: 'home-01',
                    variant: 'stroke',
                    size: 16,
                },
                badge: 100,
            },
        ],
    },
    {
        name: 'Components 2',
        path: '/components',
        icon: {
            name: 'home-01',
            variant: 'stroke',
            size: 16,
        },
        disabled: false,
        children: [
            {
                name: 'Buttons',
                path: '/buttons',
                icon: {
                    name: 'home-01',
                    variant: 'stroke',
                    size: 16,
                },
                badge: 100,
            },
            {
                name: 'Badges',
                path: '/badges',
                icon: {
                    name: 'home-01',
                    variant: 'stroke',
                    size: 16,
                },
                badge: 100,
            },
            {
                name: 'Paths',
                path: '/Paths',
                children: [
                    {
                        name: 'Buttons',
                        path: '/buttons',
                        badge: 100,
                    },
                    {
                        name: 'Badges',
                        path: '/badges',
                        badge: 100,
                    },
                    {
                        name: 'Paths',
                        path: '/Paths',
                        badge: 100,
                    },
                ],
            },
            {
                name: 'Inputs',
                path: '/inputs',
                badge: 100,
            },
            {
                name: 'Accordions',
                path: '/accordions',
                badge: 100,
            },
            {
                name: 'Tabs',
                path: '/tabs',
                badge: 100,
            },
        ],
    },
];
const DgaHeaderMenuItemUsage = class {
    constructor(hostRef) {
        registerInstance(this, hostRef);
        this.isMenuOpen = false;
        this.collapsed = false;
        // private buttonRef: HTMLElement | null = null;
        this.anchorEl = null;
        // private buttonRef: HTMLElement | null = null; // Reference to the button
        this.toggleMenu = () => {
            this.isMenuOpen = !this.isMenuOpen;
        };
        this.closeMenu = () => {
            this.isMenuOpen = false;
        };
    }
    get open() {
        return Boolean(this.anchorEl);
    }
    handleClick(event) {
        this.anchorEl = event.currentTarget;
    }
    handleClose() {
        this.anchorEl = null;
    }
    render() {
        return (h(Fragment, { key: 'bcf170764049ee8c4062b02950531a7e724906d7' }, h("dga-header-vv2", { key: '4fa6c185279ee23f2bbef96c8a6a90305f728f81' }, h("dga-flex", { key: 'e67dadcf6e3bf63a58b100aad5d865165c7413eb', spacing: 16, direction: "row", align: "center", justify: "between" }, h("dga-button", { key: 'd1d14b45e9c6f7f2f63c6157ce0559339e0ecc24', iconOnly: true, leadIcon: true, leadIconType: this.collapsed ? 'cancel01' : 'menu01', leadIconProps: {
                name: this.collapsed ? 'menu01' : 'cancel01',
                size: 24,
                variant: 'stroke',
                type: 'rounded',
            }, variant: "transparent", onClick: () => (this.collapsed = !this.collapsed), class: "header-menu__btn", sx: {
                display: { xs: 'block', md: 'none' },
            } }), h("dga-logo-placeholder", { key: '80441e7163318ff76cab8c038ff3aea43d43ec59', size: "md" }), h("dga-flex", { key: '900a8a6e87b75aae795a2b8cb68ab68640e45729', direction: "row", align: "center", justify: "start", sx: {
                display: { xs: 'none', md: 'block' },
            } }, h("dga-header-menu-item", { key: '84c692c76a183ce91e923b57c92067b2a314698d', label: "Home" }), h("dga-header-menu-item", { key: 'e942cf832bafcc28354a1bef32364fc1a586118a', label: "About" }), h("dga-header-menu-item", { key: '2e301e9fcd92db7ff1dad7b7e260d8d8e6c742f5', label: "Contact", active: true }, h("dga-header-sub-menu", { key: '2b1d6681c44f7a7ad275ae1de8ada45e4b3a3234', subMenuFullWidth: true }, h("dga-header-sub-menu-column", { key: 'b38de6c90010b1fbfbea7843b72e58b3c247b365', label: "Column Label" }, h("dga-header-sub-menu-item", { key: '524b767000e5a9916eb7787f6c427d7196b6a6dc', label: "Menu Item Label", icon: "checkmark-square-02", helperText: "Menu item helper text" }, h("dga-tag", { key: '0066a09a8942b0624b3a6260a0934ce372e51473', label: "Tag", color: "blue", slot: "tag", rounded: true })), h("dga-header-sub-menu-item", { key: '11b6a4b5391815ee95faffba59d3cb1d7d3e143a', label: "Menu Item Label", icon: "checkmark-square-02", helperText: "Menu item helper text" }, h("dga-tag", { key: '4bda1270fe7262fa670ac6a94a207a66ea79240c', label: "Tag", color: "blue", slot: "tag", rounded: true })), h("dga-header-sub-menu-item", { key: '69bbdd106c60e980f669e77ba397ce7496498450', label: "Menu Item Label", icon: "checkmark-square-02", helperText: "Menu item helper text" }, h("dga-tag", { key: '215f7d26eb8ba506643f161fc8936b9d8d0285a2', label: "Tag", color: "blue", slot: "tag", rounded: true }))), h("dga-header-sub-menu-column", { key: '5d5cbbe340c4674c8a4ec6a0e38c2b04e56258fe', label: "Column Label" }, h("dga-header-sub-menu-item", { key: '5efa6dfceadead0f9d9cf8dd565f38fef8986fdc', label: "Menu Item Label", icon: "checkmark-square-02", helperText: "Menu item helper text" }, h("dga-tag", { key: '881641955a7faf1fedebc7e306cadfe4ae76b7bd', label: "Tag", color: "blue", slot: "tag", rounded: true })), h("dga-header-sub-menu-item", { key: '48984d601245a7f092723b639cecc43660852e80', label: "Menu Item Label", icon: "checkmark-square-02", helperText: "Menu item helper text" }, h("dga-tag", { key: '3518afc69d27929d7a2916380089897ffc1c6805', label: "Tag", color: "blue", slot: "tag", rounded: true })), h("dga-header-sub-menu-item", { key: '568ae98489e14b43cff35119f5d8bf332280ce74', label: "Menu Item Label", icon: "checkmark-square-02", helperText: "Menu item helper text" }, h("dga-tag", { key: '7cb1c58c57f725209d1e91dfbc64ee459554c864', label: "Tag", color: "blue", slot: "tag", rounded: true }))), h("dga-header-sub-menu-column", { key: '155f71080ad23d3b40ad5e262ac668250dd7f4db', label: "Column Label" }, h("dga-header-sub-menu-item", { key: 'fdca8142e41ce0b57fb81792342f71cb0cf2cdb3', label: "Menu Item Label", icon: "checkmark-square-02", helperText: "Menu item helper text" }, h("dga-tag", { key: '253cd2f6eddf0813972a343064691f1d53a438ef', label: "Tag", color: "blue", slot: "tag", rounded: true })), h("dga-header-sub-menu-item", { key: '5dffe1a4d7e40274b804c2d03844f22ed8032d01', label: "Menu Item Label", icon: "checkmark-square-02", helperText: "Menu item helper text" }, h("dga-tag", { key: '852400136f746a264282066324dde66f7374327e', label: "Tag", color: "blue", slot: "tag", rounded: true })), h("dga-header-sub-menu-item", { key: '403a4b0ac5a267590a749c76146b7d6d392a8fdd', label: "Menu Item Label", icon: "checkmark-square-02", helperText: "Menu item helper text" }, h("dga-tag", { key: '14c2817d87abab1a734e731e78a107de1bcce4e4', label: "Tag", color: "blue", slot: "tag", rounded: true }))), h("dga-header-sub-menu-column", { key: '7fef1a3be3cadd3cb8d3d71473645745383240a2', label: "Column Label" }, h("dga-header-sub-menu-item", { key: 'e226fea97561fb90da31d1d98a963aa142969d5d', label: "Menu Item Label", icon: "checkmark-square-02", helperText: "Menu item helper text" }, h("dga-tag", { key: '06f1e9fd81807ed53a11e7347d93759953b01dbe', label: "Tag", color: "blue", slot: "tag", rounded: true })), h("dga-header-sub-menu-item", { key: 'da587e357ffcd1d210adf08e1907c6544dab6eac', label: "Menu Item Label", icon: "checkmark-square-02", helperText: "Menu item helper text" }, h("dga-tag", { key: 'cab88460ba8213327a071a3a27375f1ce1662a49', label: "Tag", color: "blue", slot: "tag", rounded: true })), h("dga-header-sub-menu-item", { key: '507d02f9fd7797eb666879fbee36e24c5a1d151d', label: "Menu Item Label", icon: "checkmark-square-02", helperText: "Menu item helper text" }, h("dga-tag", { key: 'b4367a8e2f12867b093505dc7bcbb9fadbfebd2a', label: "Tag", color: "blue", slot: "tag", rounded: true }))), h("dga-header-sub-menu-column", { key: '2e00f73716599fbd2c486aaa7595930cc0853d47', label: "Column Label" }, h("dga-header-sub-menu-item", { key: '514facfa1c87e213cc8476704f71bcfb26dafbe3', label: "Menu Item Label", icon: "checkmark-square-02", helperText: "Menu item helper text" }, h("dga-tag", { key: '06877e302fa1b593d2cff80a6cc7823cd570b91f', label: "Tag", color: "blue", slot: "tag", rounded: true })), h("dga-header-sub-menu-item", { key: '1c3ad153b4f5b396dc2c7de2e21f284b6cd9f8b2', label: "Menu Item Label", icon: "checkmark-square-02", helperText: "Menu item helper text" }, h("dga-tag", { key: '0b1ddc38fad2998306a8891ae10876396c5f5eef', label: "Tag", color: "blue", slot: "tag", rounded: true })), h("dga-header-sub-menu-item", { key: '6843eaf4c120a56afdb28cdf709d6bcda1e172c4', label: "Menu Item Label", icon: "checkmark-square-02", helperText: "Menu item helper text" }, h("dga-tag", { key: 'd491438bd168907e8da9b32904d7ab80dd6b9adb', label: "Tag", color: "blue", slot: "tag", rounded: true }))))))), h("dga-flex", { key: 'ce256c8f988dbb9be9a7892db3c3edcd686a78d5', direction: "row", align: "center", justify: "end" }, h("dga-header-action", { key: '5ba5e15685cf4ee2edbf1267c0923feb41eb173a', label: "Action", icon: "login-square-01" }), h("dga-header-action", { key: 'e8d6d175e739ddc6ceb475674d76b506a1114663', icon: "login-square-01", iconPosition: "top" }, h("dga-header-sub-menu", { key: '1c201f98ed79afa68c9b21b5444df2347a328c06', subMenuFullWidth: true }, h("dga-search-box", { key: '5b66a6f3b20a2f7b5a7ac74377ee2d37080ebcdc' }))), h("dga-action-menu", { key: '5015c9f5c4d53d685607f0a85ad184d65e64c6e0', open: this.isMenuOpen, onClose: () => (this.isMenuOpen = false), placement: 'bottom-end' }, h("dga-button-v2", { key: '46f25d43a095199c7fc948088bba856c44907a90', slot: "trigger", iconOnly: true, leadIcon: true, leadIconProps: {
                name: 'more-horizontal-circle-01',
                size: 24,
                variant: 'solid',
                type: 'rounded',
            }, variant: "transparent", onClick: () => (this.isMenuOpen = true), class: "header-menu__btn" }), h("dga-menu-list", { key: '72876c3c8714415a23218e68bb7141389868f311' }, h("dga-menu-item", { key: '916fc9be0b119288386b7021c0396f832f855a64', label: "Item Label" }), h("dga-menu-item", { key: '5dbe82ef0984a9eaac164689fb512c91895facc8', label: "Item Label" }), h("dga-menu-item", { key: '781f2e5c7f9f09648cf445f24a6cdc333fe62f0c', label: "Item Label" }))))), h("dga-drawer", { key: 'd8adde1e533a464a18148475a51c8581e6c556e0', routes: SidePanelRoutes, open: this.collapsed, background: 'gray', headerLogo: "https://casamedia.com/wp-content/uploads/2023/04/adidas-1024x683.png", sx: {
                '& > .sidepanel': {
                    top: '72px',
                },
            }, overlay: true }, "efddsfdsfsdfdsfdsf dfdsfdsfdsfdsfdsfdf", h("a", { key: '2b6f1a59b047bb552c4e18fbc3267e297c682f46' }, "fdgdfgfdg"), h("div", { key: '33db7bda83a5c8bc0e3094067e44e0f431e455b8' }, "gfdgfdgfd")), h("dga-logo-placeholder", { key: 'e580d491cb66183c6f193479201e0a25c6e9c847', text: "Platform Logo", size: "md" }), h("dga-logo-placeholder", { key: '11247d359187396ff45548ceb9f1057147b3d4e1', text: "Platform Logo", size: "sm" }), h("dga-action-menu", { key: 'd216f3e6957f74143008ff8e4aa751d0c32f33f3', open: this.isMenuOpen, onClose: () => (this.isMenuOpen = false), placement: 'bottom-end' }, h("dga-button-v2", { key: '6576ffab2d6942add2e26fdb5b89c58714876fab', slot: "trigger", iconOnly: true, leadIcon: true, leadIconProps: {
                name: 'more-horizontal-circle-01',
                size: 24,
                variant: 'solid',
                type: 'rounded',
            }, variant: "transparent", onClick: () => (this.isMenuOpen = true), class: "header-menu__btn" }), h("dga-menu-list", { key: '02b514ed6b424e755a8e74cd63bc20d62ff3f612' }, h("dga-menu-item", { key: '076339e422afef951f54dc33fcdc2484c646b1be', label: "Item Label" }), h("dga-menu-item", { key: 'f61cecc39d8f2aff489484702a845fd8b4c4836d', label: "Item Label" }), h("dga-menu-item", { key: 'e3bd4881454b357266be92f09cf8956385d88d0c', label: "Item Label" })))));
    }
};

export { DgaHeaderMenuItemUsage as dga_header_menu_item_usage };
