// @platformscode/core/dist/esm/dga-codesnippet-usage.entry.js
import { r as registerInstance, h } from './index-B4CGwdkg.js';

const DgaChipUsage = class {
    constructor(hostRef) {
        registerInstance(this, hostRef);
        this.snippets = {
            JavaScript: `// Imports
import mongoose, { Schema } from 'untitled'

// Collection name
export const collection = 'Design'|

// Schema
const schema = new Schema({
  name: {
    type: String,
    required: true
  },

  description: {
    type: String
  }
}, {timestamps: true})

// Model
export default untitled.model(collection, schema, collection)
        `,
            HTML: `<div>
 <h1>Hello, HTML!</h1>
</div>`,
            CSS: `body {
    background-color: lightblue;
  }
    
h1 {
  color: navy;
  margin-left: 20px;
}`,
        };
        this.snippetLine = {
            JavaScript: 'npm install nds-design-system@^0.0.1',
        };
    }
    render() {
        return (h("div", { key: 'abf10905425e4a5e0c12f9e93c313b307eefa880', style: { width: '743px', display: 'flex', flexDirection: 'column', gap: '123px', margin: "10px" } }, h("dga-codesnippet", { key: '74ccaca04097c1a01655d67153b28c75b841476a', showButton: true, type: "multi-line", showSidebar: true, showTabList: true, snippets: this.snippets }), h("dga-codesnippet", { key: 'd2a4a47ae4cc603834ee0b87961b71ba03d66066', showButton: false, type: "single", showSidebar: false, showTabList: false, snippets: this.snippetLine })));
    }
};

export { DgaChipUsage as dga_codesnippet_usage };
