// @platformscode/core/dist/esm/dga-list.entry.js
import { r as registerInstance, h } from './index-B4CGwdkg.js';

const DgaList = class {
    constructor(hostRef) {
        registerInstance(this, hostRef);
        this.type = 'unordered';
        this.color = 'primary';
        //   @Prop() id: string = guid();
        this.title = "";
        this.items = [];
        this.renderIcon = (type) => {
            if (type === "with-icon") {
                return h("span", { class: "list-icon" }, "\u2713"); // Using checkmark icon for "with-icon" type
            }
            return null;
        };
    }
    render() {
        const Tag = this.type === 'ordered' ? 'ol' : 'ul';
        return (h("div", { key: '7d95361dcea178a59da32b4d08bf7f0560eb3d48', class: `list list--${this.type} list--${this.color}` }, this.title && h("h3", { key: '0344d56c05bb2edaee1716489128994fd076a0bf', class: "list-title" }, this.title), h(Tag, { key: 'd7e8b8e2a3f2162f9b0c6b25e9a6025d44f0ea71' }, this.items.map((item, index) => (h("li", { key: index, class: "list-item" }, this.type === "with-icon" && this.renderIcon(this.type), item))))));
    }
};

export { DgaList as dga_list };
