// @platformscode/core/dist/esm/dga-modal-usage.entry.js
import { r as registerInstance, h } from './index-B4CGwdkg.js';

const DgaModalUsage = class {
    constructor(hostRef) {
        registerInstance(this, hostRef);
        this.show = false;
    }
    render() {
        return (h("div", { key: '1d7c4cef1e1f98538810ce4b0eb77166f30cf030', style: { position: "relative" } }, h("h1", { key: 'a7ebea0aa5d67666454bfeecc4306a7ab7a71dfd' }, "Modal Example"), h("button", { key: 'ab9d06082be15e449e14b96eb2eac55b3f65a800', onClick: () => this.show = true }, "Open Modal"), h("dga-modal", { key: '7328e375daaf6abc83cfc846d8458b34f924f94b', buttonsList: [
                {
                    extraClass: '',
                    label: 'Close',
                    onClick: () => { },
                    title: 'Close Modal'
                }
            ], closeButton: true, modalTitle: "Basic Modal", name: "basicModal", onClose: () => { }, open: this.show }, "This is a simple modal with default configuration."), this.show ? "shown" : "hidden"));
    }
};

export { DgaModalUsage as dga_modal_usage };
