// @platformscode/core/dist/esm/dga-breadcrumbs-usage.entry.js
import { r as registerInstance, h } from './index-B4CGwdkg.js';

const DgaBreadcrumbsUsage = class {
    constructor(hostRef) {
        registerInstance(this, hostRef);
        this.variant = 'neutral';
    }
    render() {
        return (h("div", { key: '309588c044157c237b4f1ffbbfda0a523d1c4d73' }, h("h1", { key: 'cb268da1bb2282e12ae123e6a5a3b366c6e49d50' }, "Breadcrumbs Example"), h("div", { key: 'd7f048d41b3785a8aac90e7fa931d2f7e6b0ddb4', style: { padding: '50px', display: 'flex', gap: '20px' } }, h("div", { key: '645ee3c57cbc5f81a88baee57f52da0b04adb890', style: { display: 'flex', flexDirection: 'column', gap: '20px' } }, h("dga-link", { key: '739dfabb573bbc8c08ef7f8f10a5be02f372c121', label: "Link", variant: "neutral" }), h("dga-breadcrumbs", { key: '7acc4f3095ad9a78f93d6a73068b0d222b46e925', items: [
                {
                    label: 'Home',
                    path: '/',
                },
                {
                    label: 'Products',
                    path: '/products',
                },
                {
                    label: 'Computers',
                    path: '/products/computers',
                    disabled: true,
                },
                {
                    label: 'Keyboards',
                    path: '/products/keyboards',
                },
            ], max: 3 })), h("div", { key: '93da9b1847fbfe5253baf25eafbc4eb4206c4649', style: { display: 'flex', flexDirection: 'column', gap: '20px' } }))));
    }
};

export { DgaBreadcrumbsUsage as dga_breadcrumbs_usage };
