// @platformscode/core/dist/esm/dga-header-action-btn_11.entry.js
export { a as dga_header_action_btn, f as dga_nav_header, D as dga_nav_header_actions, e as dga_nav_header_link, c as dga_nav_header_logos, b as dga_nav_header_main, d as dga_nav_header_menu } from './dga-header-action-btn-HtpgbZi0.js';
import { r as registerInstance, h, H as Host, a as getElement } from './index-B4CGwdkg.js';

const DgaSecondNavHeader = class {
    constructor(hostRef) {
        registerInstance(this, hostRef);
        this.variant = 'primary';
    }
    componentWillLoad() {
        this.el.childNodes.forEach((item) => {
            const el = item;
            if (el.tagName === 'DGA-SECOND-NAV-HEADER-CONTENT' && this.hideContent === true) {
                el.style.display = 'none';
            }
            if (el.tagName === 'DGA-SECOND-NAV-HEADER-ACTIONS' && this.hideActions === true) {
                el.style.display = 'none';
            }
            if (el.tagName === 'DGA-SECOND-NAV-HEADER-CONTENT' && el.attributes.getNamedItem('slot') === null) {
                el.setAttribute('slot', 'content');
            }
            if (el.tagName === 'DGA-SECOND-NAV-HEADER-ACTIONS' && el.attributes.getNamedItem('slot') === null) {
                el.setAttribute('slot', 'actions');
            }
        });
    }
    render() {
        return (h(Host, { key: '2148ff638962c3774f6296bb153ab227b2037242' }, h("div", { key: '1876a750095870936c30c938417dbf8f2edc2e7c', class: `wrapper ${this.variant} `, "aria-label": 'a secondary navigation navbar' }, h("div", { key: '39f09c68a16d833e43610913fb903bcafd8b1f9f', class: `container ${this.hideDivider ? '' : 'border'}` }, h("div", { key: '4185be99ad09528b5ab48dfb2f9f9683c80dcc62', class: `content` }, h("slot", { key: '9c880c19c64d3b5a13fd531e1b4e8f6d88ebf13b', name: "content" })), h("div", { key: '9365fa894dc11cbfbccd783834481fb86fd3d0e6', class: `actions` }, h("slot", { key: '252fa64837f5c111ff0954c921c906aa9bba1053', name: "actions" }))))));
    }
    get el() { return getElement(this); }
};

const DgaSecondNavHeaderActions = class {
    constructor(hostRef) {
        registerInstance(this, hostRef);
    }
    render() {
        return (h(Host, { key: '37d52515535536dd4bbec09ff34561c2cf641af7' }, h("div", { key: '70359fb2e401799c98cc206539e5307a44dfc8bb', class: `items-container` }, h("slot", { key: '00f13fc62dfdb9e26dae61b74bfe54bfcaad69cd' }))));
    }
};

const DgaSecondNavHeaderContent = class {
    constructor(hostRef) {
        registerInstance(this, hostRef);
    }
    render() {
        return (h(Host, { key: 'd1e640c6c31419544931a943928581ef2ea7eb5c' }, h("div", { key: '6e70bdd808ba728ed6ef56f595cc5bef8001cb59', class: `items-container` }, h("slot", { key: '47451b0f111654c0c6691e28b6f75aff959efa01' }))));
    }
};

const DgaSecondNavHeaderItem = class {
    constructor(hostRef) {
        registerInstance(this, hostRef);
    }
    connectedCallback() {
        this.el.childNodes.forEach((item) => {
            const el = item;
            if (el.tagName === 'DGA-ICON'.toLocaleUpperCase() && el.attributes.getNamedItem('slot') === null) {
                el.setAttribute('slot', 'icon');
            }
        });
    }
    render() {
        return (h(Host, { key: '229c796621d1842e6484a2857130228a6064aae2' }, h("slot", { key: '9b10838752cd3103a736a4bf1931ec58e142ec36', name: 'icon' }), h("slot", { key: '70317608a423de1002f8b762096309aed55b2311', name: 'label' }, this.label)));
    }
    get el() { return getElement(this); }
};

export { DgaSecondNavHeader as dga_second_nav_header, DgaSecondNavHeaderActions as dga_second_nav_header_actions, DgaSecondNavHeaderContent as dga_second_nav_header_content, DgaSecondNavHeaderItem as dga_second_nav_header_item };
