// @platformscode/core/dist/esm/dga-metric-chart.entry.js
import { r as registerInstance, h } from './index-B4CGwdkg.js';

const DgaMetricChart = class {
    constructor(hostRef) {
        registerInstance(this, hostRef);
        this.type = 'realistic';
        this.trend = 'positive';
        this.size = 'lg';
        this.showMarker = false;
        this.data = [];
        this.lineColor = this.trend === 'positive' ? '#067647' : '#B42318';
        this.fillColor = this.trend === 'positive' ? '#079455' : '#D92D20';
        this.width = this.size === 'lg' ? 328 : 128;
        this.height = 64;
    }
    handleSizeChange() {
        this.width = this.size === 'lg' ? 328 : 128;
    }
    getLinePath() {
        const points = this.getPoints();
        switch (this.type) {
            case 'realistic':
                return this.getSmoothPath(points);
            case 'wavy':
                return this.getWavyPath(points);
            case 'straight':
                return this.getStraightPath(points);
            case 'layers':
                return this.getSmoothPath(points);
            default:
                return '';
        }
    }
    getPoints() {
        return this.data.map((d, i) => ({
            x: (i / (this.data.length - 1)) * this.width,
            y: this.height - (d / Math.max(...this.data)) * this.height
        }));
    }
    getSmoothPath(points) {
        return points.reduce((acc, point, i, arr) => {
            if (i === 0)
                return `M ${point.x},${point.y}`;
            const prev = arr[i - 1];
            const cX = (prev.x + point.x) / 2;
            return `${acc} C ${cX},${prev.y} ${cX},${point.y} ${point.x},${point.y}`;
        }, '');
    }
    getWavyPath(points) {
        return points.reduce((acc, point, i) => i === 0 ? `M ${point.x},${point.y}` : `${acc} Q ${point.x - 10},${point.y + 10} ${point.x},${point.y}`, '');
    }
    getStraightPath(points) {
        return points.reduce((acc, point, i) => i === 0 ? `M ${point.x},${point.y}` : `${acc} L ${point.x},${point.y}`, '');
    }
    getAreaPath() {
        const linePath = this.getLinePath();
        return `${linePath} L ${this.width},${this.height} L 0,${this.height} Z`;
    }
    render() {
        const linePath = this.getLinePath();
        const areaPath = this.getAreaPath();
        return (h("svg", { key: '6e0e53da9e43dfed03b40b987a62f83893ea546a', width: this.width, height: this.height, viewBox: `0 0 ${this.width} ${this.height}` }, h("defs", { key: '7d5cf14f37fd3273a262abc279017136407e7334' }, h("linearGradient", { key: '5e1d5bfe2c22e62b37e6aa76b92b3f17f43c53c3', id: "areaGradient", x1: "0", x2: "0", y1: "0", y2: "1" }, h("stop", { key: 'e8a9ef5cb36775b8e8883941c828ec70090c7d5a', offset: "0%", "stop-color": this.fillColor, "stop-opacity": "0.25" }), h("stop", { key: '76df9718b5edb53c6a24585f0a9b9d101b400579', offset: "100%", "stop-color": this.fillColor, "stop-opacity": "0" }))), h("path", { key: '5f2f736807d130c7fd3b0f7c802d5d1249b5c698', d: areaPath, fill: "url(#areaGradient)" }), h("path", { key: '830d65dc2186a1616f8c05eb2856e80f21814cde', d: linePath, fill: "none", stroke: this.lineColor, "stroke-width": "2" }), this.showMarker && this.getPoints().map((point) => (h("circle", { cx: point.x, cy: point.y, r: "4", fill: "white", stroke: this.lineColor, "stroke-width": "2" })))));
    }
    static get watchers() { return {
        "size": ["handleSizeChange"]
    }; }
};

export { DgaMetricChart as dga_metric_chart };
