// @platformscode/core/dist/esm/dga-progress-indicator-step-v3.entry.js
import { r as registerInstance, h, a as getElement } from './index-B4CGwdkg.js';

const DgaProgressIndicatorStep = class {
    constructor(hostRef) {
        registerInstance(this, hostRef);
        this.currentStep = 1;
        this.title = '';
        this.description = '';
        this.activeStep = 0;
        this.nextStep = true;
        this.showStepName = true;
        this.showStepDescription = true;
        this.internalActiveStep = 1;
        this.isActive = false;
        this.labelState = 'upcoming';
        this.labelStyle = 'circle';
        this.handleFocus = () => {
            this.stepContentRef.classList.add('focus');
        };
        this.handleFocusOut = () => {
            this.stepContentRef.classList.remove('focus');
        };
        this.handleEscape = (event) => {
            if (event.key === 'Escape') {
                this.handleFocusOut();
                const inputElement = this.el.querySelector('input');
                if (inputElement) {
                    inputElement.blur();
                }
            }
        };
    }
    componentWillLoad() {
        this.updateState();
        window.addEventListener('keydown', this.handleEscape);
    }
    disconnectedCallback() {
        window.removeEventListener('keydown', this.handleEscape);
    }
    stepChangeHandler(event) {
        this.internalActiveStep = event.detail;
        console.log('stepChangeHandler', event.detail);
        this.isActive = event.detail > this.currentStep;
        if (event.detail > this.currentStep) {
            this.labelState = 'completed';
        }
        else if (event.detail < this.currentStep) {
            this.labelState = 'upcoming';
        }
        else {
            this.labelState = 'current';
        }
    }
    // @Watch('internalActiveStep')
    updateState() {
        this.isActive = this.internalActiveStep > this.currentStep;
        if (this.internalActiveStep > this.currentStep) {
            this.labelState = 'completed';
        }
        else if (this.internalActiveStep < this.currentStep) {
            this.labelState = 'upcoming';
        }
        else {
            this.labelState = 'current';
        }
    }
    labelChildren(currentStep) {
        if (this.labelState === 'completed') {
            return h("dga-icon", { name: "tick-02", variant: "stroke", size: 16 });
        }
        return currentStep.toString();
    }
    render() {
        return (h("div", { key: '2544d7cf7a9c11194fe596987e1b992b5b9bcba0', class: "progress-indicator__step", onMouseDown: this.handleFocus, onMouseUp: this.handleFocusOut }, h("div", { key: '7981ca45bac9a195f81759fd424d9a62fd695b2f', class: "progress-indicator__step-content dga-flex-column", ref: el => (this.stepContentRef = el) }, h("div", { key: '951a145c98cca8b02351068efd00a4e3875889af', class: "step-label", "data-state": this.labelState }, h("span", { key: '776405bfc8d14e62886a8be93bc63cbfa2bb6dff', class: `step-label--${this.labelStyle}` }, this.labelStyle === 'circle' && this.labelChildren(this.currentStep))), h("div", { key: '63fef306b1e435195e2ae7eb52c1ec2ad864e2fa', class: "step-text dga-flex-column" }, this.showStepName && h("span", { key: '513540640173803976c08f8581c2b110d8a51987', class: this.isActive ? 'text-md-medium' : 'text-md-regular' }, this.title), this.showStepDescription && h("p", { key: 'e7b057d27cbda72fd9e9353a66e43bdc136c5e1a', class: "text-sm-regular" }, this.description))), this.nextStep && h("span", { key: '6cdf0dca2aa4fe31ca4f8529ba1d9d214374af9a', class: `progress-indicator__step-connector ${this.isActive ? 'progress-indicator__step-connector--active' : ''}` })));
    }
    get el() { return getElement(this); }
};

export { DgaProgressIndicatorStep as dga_progress_indicator_step_v3 };
