// @platformscode/core/dist/esm/dga-rating-usage.entry.js
import { r as registerInstance, h } from './index-B4CGwdkg.js';

const DgaRatingUsage = class {
    constructor(hostRef) {
        registerInstance(this, hostRef);
        this.variant = 'neutral';
        this.rating = 2.5;
    }
    render() {
        return (h("div", { key: 'c1387a2f19bb2cfaa4ef4c86fc70abcdf98aa5da', dir: "rtl" }, h("h1", { key: '72aaefac87149c684c7b2b60dab6e14a28025092' }, "Rating Example"), h("div", { key: 'f49ecb0f804fd5441f34da21889749172014c918', style: { padding: '50px', display: 'flex', gap: '20px' } }, h("div", { key: '185994d202c61108e248dcade8b4be7ca2b744ed', style: { display: 'flex', flexDirection: 'column', gap: '20px' } }, h("button", { key: '5135732fd29e9e1a28bd856d6c266b414c4a7cd2', onClick: () => (this.rating += 1) }, "Increase ", this.rating), h("dga-rating", { key: 'a549107cd0023f828106e9bd24887a40a58d40fe', onChange: val => {
                this.rating = val;
            }, value: this.rating, size: "lg" }), h("dga-rating", { key: 'c14ceff74a64cfa7669f2ae804a09b5c79668436', onChange: val => (this.rating = val), value: 2.5, size: "md" }), h("dga-rating", { key: '01b77dad97fb3992f1e5383d760042560886b273', onChange: val => (this.rating = val), value: this.rating, size: "sm" })), h("div", { key: '19ae50c0e073d9a25c7ed112f66af89df062a238', style: { display: 'flex', flexDirection: 'column', gap: '20px' } }))));
    }
};

export { DgaRatingUsage as dga_rating_usage };
