// @platformscode/core/dist/esm/dga-metric-chart-v2.entry.js
import { r as registerInstance, h } from './index-B4CGwdkg.js';

const DgaMetricChartV2 = class {
    constructor(hostRef) {
        registerInstance(this, hostRef);
        this.width = 130;
        this.height = 66;
        this.strokeColor = '#067647';
        this.fillColor = '#079455';
        this.strokeWidth = 2;
        this.data = [];
        this.showMarker = true;
        this.pathD = '';
        this.areaD = '';
        this.currentMarkerPos = { x: 76, y: 5 }; // default position
    }
    dataChangeHandler() {
        this.generatePaths();
        this.updateMarkerPosition();
    }
    componentWillLoad() {
        this.generatePaths();
        this.updateMarkerPosition();
    }
    normalizeData() {
        if (this.data.length === 0)
            return [];
        const xValues = this.data.map(d => d.x);
        const yValues = this.data.map(d => d.y);
        const minX = Math.min(...xValues);
        const maxX = Math.max(...xValues);
        const minY = Math.min(...yValues);
        const maxY = Math.max(...yValues);
        const margin = 10;
        return this.data.map(point => ({
            x: ((point.x - minX) / (maxX - minX)) * (128 - 2 * margin) + margin,
            y: 64 - (((point.y - minY) / (maxY - minY)) * (64 - 2 * margin) + margin)
        }));
    }
    updateMarkerPosition() {
        if (this.markerPosition) {
            this.currentMarkerPos = this.markerPosition;
        }
        else if (this.data.length > 0) {
            const normalizedData = this.normalizeData();
            const lastPoint = normalizedData[normalizedData.length - 1];
            // Calculate marker position based on the last data point
            // Adjust these offsets to position the marker correctly relative to the point
            this.currentMarkerPos = {
                x: lastPoint.x - 10, // Adjust these offsets as needed
                y: lastPoint.y - 15 // Adjust these offsets as needed
            };
        }
    }
    generatePaths() {
        const normalizedData = this.normalizeData();
        if (normalizedData.length === 0)
            return;
        this.pathD = normalizedData.reduce((path, point, i) => {
            return path + `${i === 0 ? 'M' : 'L'} ${point.x} ${point.y} `;
        }, '');
        this.areaD = `${this.pathD} L ${normalizedData[normalizedData.length - 1].x} 65 L 1 65 Z`;
    }
    render() {
        const { x: markerX, y: markerY } = this.currentMarkerPos;
        return (h("svg", { key: '8441c2e5f5e05b71aa9219772be966ae538abb99', width: this.width, height: this.height, viewBox: "0 0 130 66", fill: "none", xmlns: "http://www.w3.org/2000/svg" }, h("defs", { key: '2ef4cf4b02a690980a548735a632eee5d317f113' }, h("linearGradient", { key: '3cf5eb7dedd346de4b6663f0f34a871fd31a66b8', id: "gradientMask", x1: "64", y1: "0", x2: "64", y2: "64", gradientUnits: "userSpaceOnUse" }, h("stop", { key: '89d831bf8353aeb7fc62ec8735226779853d541c' }), h("stop", { key: 'bd072fe5eddf721c9c543fcdcb8b9c5afe5b8c05', offset: "1", "stop-opacity": "0" })), h("mask", { key: '73f36217b608723995fd651589dbfb66f1d90f89', id: "chartMask", style: { maskType: 'alpha' }, maskUnits: "userSpaceOnUse", x: "1", y: "1", width: "128", height: "64" }, h("rect", { key: '0984f7e8941af7ed10fe674cddfea16b0494ed4c', width: "128", height: "64", transform: "translate(1 1)", fill: "url(#gradientMask)" }))), h("g", { key: '7e7ccdb3cb39833368570c48bd52643f0f6d3a51', mask: "url(#chartMask)" }, h("path", { key: 'c85d9a84c13fa6b0e07e17ed7b00521dfc7e2010', opacity: "0.2", d: this.areaD, fill: this.fillColor })), h("path", { key: '78c1fee3e019258c5774c83f8d9e076534af7cd4', d: this.pathD, stroke: this.strokeColor, "stroke-width": this.strokeWidth, "stroke-linecap": "round", "stroke-linejoin": "round", fill: "none" }), this.showMarker && (h("g", { key: '7f5dfcd69728523b3abcec8fc3d847f5acddc95c', transform: `translate(${markerX}, ${markerY})` }, h("g", { key: 'ce18730ff77a6e5dc1c7fc8d8572fd7a87e1a449', opacity: "0.2" }, h("rect", { key: '938728b792c35b22a127080a94593669d063b1d1', width: "20", height: "20", rx: "10", stroke: this.strokeColor, "stroke-width": this.strokeWidth })), h("rect", { key: 'd8a9dc11b706ff001739a4950874e02db13f14a3', x: "5", y: "5", width: "10", height: "10", rx: "5", fill: "white" }), h("rect", { key: 'f3ea78180ae863f4108cab1f9e70571b8d3d0208', x: "5", y: "5", width: "10", height: "10", rx: "5", stroke: this.strokeColor, "stroke-width": this.strokeWidth })))));
    }
    static get watchers() { return {
        "data": ["dataChangeHandler"]
    }; }
};

export { DgaMetricChartV2 as dga_metric_chart_v2 };
