// @platformscode/core/dist/esm/nds-chip-usage.entry.js
import { r as registerInstance, h } from './index-B4CGwdkg.js';

const NdsChipUsage = class {
    constructor(hostRef) {
        registerInstance(this, hostRef);
        this.isSelected = false;
    }
    render() {
        return (h("div", { key: '860e518b969456a06f966cdc761e6bf2d5964994' }, h("h1", { key: '3bbc6620975ef82775b294ab0762860538160ac3' }, "Chip Example"), h("div", { key: '9c243a58a360b5c79a175bc9ef0e770a520f900e', style: { padding: '50px', display: 'flex', gap: '20px' } }, h("div", { key: '4e6f4aa140cf41554ebf3db18d7886dd9eab161f', style: { display: 'flex', flexDirection: 'column', gap: '20px' } }, h("nds-chip", { key: '2be97d733b5e63de2853901dce34cc86d1993624', label: "Label", variant: "neutral",
            //   outlined={false}
            rounded: false, size: "lg" }), h("nds-chip", { key: '4c4b89ef047ef7931b03889a5485ba86596cfc44', label: "Label", variant: "success", rounded: false, size: "lg", onChange: isSelected => {
                console.log(isSelected, 'hi');
                this.isSelected = isSelected;
            }, selected: this.isSelected }), h("nds-chip", { key: '87d39a4044d00baf1d899057aa926692c5f52c0f', label: "Label", variant: "success", rounded: false, size: "lg", onChange: isSelected => {
                console.log(isSelected, 'hi');
            }, selected: false }), h("nds-chip", { key: '85ab156161f0c208bea3a2a02e20713f6a516341', label: "Label", variant: "success", rounded: false, size: "lg", disabled: true }), h("div", { key: 'ecdee48745e665a86c60c00a38b4bd6c425be3e3', style: { display: 'flex', flexDirection: 'column', gap: '20px', backgroundColor: '#074D31' } }, h("nds-chip", { key: 'e7827fb541f18725b15daae7ece8da332190f533', label: "Label", variant: "on-color", rounded: true, size: "lg", disabled: true }))))));
    }
};

export { NdsChipUsage as nds_chip_usage };
