// @platformscode/core/dist/collection/components/dga-text-input/dga-helper-text.js
import { h } from "@stencil/core";
export class DgaHelperText {
    constructor() {
        this.helperText = '';
        this.error = false;
    }
    render() {
        var _a;
        return (h("div", { key: 'f2a73d39268684cec9fba2210c59ebc7f5c5741c', class: `dga-helper-text ${this.error ? "dga-helper-text--error" : ""}` }, h("span", { key: '80f3bfa13dc22335d4fc636a069acbea322d3b3e', class: "dga-helper-text__icon" }, this.helperTextIcon && ((_a = this.helperTextIcon) === null || _a === void 0 ? void 0 : _a.name) && h("dga-icon", { key: '2bd2fe44c408bff2ffc26a71712b9717f94b6579', name: this.helperTextIcon.name, variant: this.helperTextIcon.variant, type: this.helperTextIcon.type, size: 16 })), h("span", { key: 'e886b1b9c9a018aa623df4920d53272615187e66', class: "dga-helper-text__desc" }, this.helperText)));
    }
    static get is() { return "dga-helper-text"; }
    static get properties() {
        return {
            "helperTextIcon": {
                "type": "unknown",
                "attribute": "helper-text-icon",
                "mutable": false,
                "complexType": {
                    "original": "Icon",
                    "resolved": "Icon",
                    "references": {
                        "Icon": {
                            "location": "import",
                            "path": "../dga-featured-icon/dga-featured-icon",
                            "id": "src/components/dga-featured-icon/dga-featured-icon.tsx::Icon"
                        }
                    }
                },
                "required": false,
                "optional": true,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false
            },
            "helperText": {
                "type": "string",
                "attribute": "helper-text",
                "mutable": false,
                "complexType": {
                    "original": "string",
                    "resolved": "string",
                    "references": {}
                },
                "required": false,
                "optional": true,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false,
                "defaultValue": "''"
            },
            "error": {
                "type": "boolean",
                "attribute": "error",
                "mutable": false,
                "complexType": {
                    "original": "boolean",
                    "resolved": "boolean",
                    "references": {}
                },
                "required": false,
                "optional": true,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false,
                "defaultValue": "false"
            }
        };
    }
}
