// @platformscode/core/dist/collection/components/dga-metric/dga-metric-chart.js
import { h } from "@stencil/core";
export class DgaMetricChart {
    constructor() {
        this.type = 'realistic';
        this.trend = 'positive';
        this.size = 'lg';
        this.showMarker = false;
        this.data = [];
        this.lineColor = this.trend === 'positive' ? '#067647' : '#B42318';
        this.fillColor = this.trend === 'positive' ? '#079455' : '#D92D20';
        this.width = this.size === 'lg' ? 328 : 128;
        this.height = 64;
    }
    handleSizeChange() {
        this.width = this.size === 'lg' ? 328 : 128;
    }
    getLinePath() {
        const points = this.getPoints();
        switch (this.type) {
            case 'realistic':
                return this.getSmoothPath(points);
            case 'wavy':
                return this.getWavyPath(points);
            case 'straight':
                return this.getStraightPath(points);
            case 'layers':
                return this.getSmoothPath(points);
            default:
                return '';
        }
    }
    getPoints() {
        return this.data.map((d, i) => ({
            x: (i / (this.data.length - 1)) * this.width,
            y: this.height - (d / Math.max(...this.data)) * this.height
        }));
    }
    getSmoothPath(points) {
        return points.reduce((acc, point, i, arr) => {
            if (i === 0)
                return `M ${point.x},${point.y}`;
            const prev = arr[i - 1];
            const cX = (prev.x + point.x) / 2;
            return `${acc} C ${cX},${prev.y} ${cX},${point.y} ${point.x},${point.y}`;
        }, '');
    }
    getWavyPath(points) {
        return points.reduce((acc, point, i) => i === 0 ? `M ${point.x},${point.y}` : `${acc} Q ${point.x - 10},${point.y + 10} ${point.x},${point.y}`, '');
    }
    getStraightPath(points) {
        return points.reduce((acc, point, i) => i === 0 ? `M ${point.x},${point.y}` : `${acc} L ${point.x},${point.y}`, '');
    }
    getAreaPath() {
        const linePath = this.getLinePath();
        return `${linePath} L ${this.width},${this.height} L 0,${this.height} Z`;
    }
    render() {
        const linePath = this.getLinePath();
        const areaPath = this.getAreaPath();
        return (h("svg", { key: '2ad9c2c0939a122d3e3c1e727e9f33fe52ecd376', width: this.width, height: this.height, viewBox: `0 0 ${this.width} ${this.height}` }, h("defs", { key: '1dcfbe63f895bf76d6a8e8e4e54c9b2d6fa9fafb' }, h("linearGradient", { key: 'c7461f4ddf4587d347f132523448a025f9b1a017', id: "areaGradient", x1: "0", x2: "0", y1: "0", y2: "1" }, h("stop", { key: 'fa5fc33dbe033170dc884c122e403ab4ec316093', offset: "0%", "stop-color": this.fillColor, "stop-opacity": "0.25" }), h("stop", { key: 'c234bf4bedd32b97edef1bb0e69d7b8ca7b3fd7f', offset: "100%", "stop-color": this.fillColor, "stop-opacity": "0" }))), h("path", { key: '85aca411c2c387929308fa2814f47b4ab5b02271', d: areaPath, fill: "url(#areaGradient)" }), h("path", { key: '28018df6b9951eef64277075d0686bfe51176f07', d: linePath, fill: "none", stroke: this.lineColor, "stroke-width": "2" }), this.showMarker && this.getPoints().map((point) => (h("circle", { cx: point.x, cy: point.y, r: "4", fill: "white", stroke: this.lineColor, "stroke-width": "2" })))));
    }
    static get is() { return "dga-metric-chart"; }
    static get properties() {
        return {
            "type": {
                "type": "string",
                "attribute": "type",
                "mutable": false,
                "complexType": {
                    "original": "'realistic' | 'wavy' | 'straight' | 'layers'",
                    "resolved": "\"layers\" | \"realistic\" | \"straight\" | \"wavy\"",
                    "references": {}
                },
                "required": false,
                "optional": true,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false,
                "defaultValue": "'realistic'"
            },
            "trend": {
                "type": "string",
                "attribute": "trend",
                "mutable": false,
                "complexType": {
                    "original": "'positive' | 'negative'",
                    "resolved": "\"negative\" | \"positive\"",
                    "references": {}
                },
                "required": false,
                "optional": false,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false,
                "defaultValue": "'positive'"
            },
            "size": {
                "type": "string",
                "attribute": "size",
                "mutable": false,
                "complexType": {
                    "original": "'sm' | 'lg'",
                    "resolved": "\"lg\" | \"sm\"",
                    "references": {}
                },
                "required": false,
                "optional": true,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false,
                "defaultValue": "'lg'"
            },
            "showMarker": {
                "type": "boolean",
                "attribute": "show-marker",
                "mutable": false,
                "complexType": {
                    "original": "boolean",
                    "resolved": "boolean",
                    "references": {}
                },
                "required": false,
                "optional": false,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false,
                "defaultValue": "false"
            },
            "data": {
                "type": "unknown",
                "attribute": "data",
                "mutable": false,
                "complexType": {
                    "original": "number[]",
                    "resolved": "number[]",
                    "references": {}
                },
                "required": false,
                "optional": false,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "defaultValue": "[]"
            }
        };
    }
    static get states() {
        return {
            "lineColor": {},
            "fillColor": {},
            "width": {},
            "height": {}
        };
    }
    static get watchers() {
        return [{
                "propName": "size",
                "methodName": "handleSizeChange"
            }];
    }
}
