// @platformscode/core/dist/collection/components/dga-metric/dga-metric-chart-v2.js
// import { Component, h, Prop } from '@stencil/core';
// @Component({
//   tag: 'reusable-chart',
//   // styleUrl: 'reusable-chart.css',
//   // shadow: true,
// })
// export class ReusableChart {
//   // Props to make the chart reusable
//   @Prop() width: number = 330;
//   @Prop() height: number = 66;
//   @Prop() strokeColor: string = '#067647';
//   @Prop() fillColor: string = '#079455';
//   @Prop() strokeWidth: number = 2;
//   @Prop() opacity: number = 0.2;
//   // Data prop for passing an array of data points
//   @Prop() data: number[] = [];
//   // A method to generate a path based on the data
//   generatePath(): string {
//     if (!this.data || this.data.length === 0) {
//       return '';
//     }
//     const maxValue = Math.max(...this.data);
//     const minValue = Math.min(...this.data);
//     const verticalScale = this.height / (maxValue - minValue);
//     const horizontalScale = this.width / (this.data.length - 1);
//     return this.data
//       .map((value, index) => {
//         const x = index * horizontalScale;
//         const y = this.height - (value - minValue) * verticalScale;
//         return `${index === 0 ? 'M' : 'L'} ${x} ${y}`;
//       })
//       .join(' ');
//   }
//   render() {
//     const pathD = this.generatePath();
//     return (
//       <svg
//         width={this.width}
//         height={this.height}
//         viewBox={`0 0 ${this.width} ${this.height}`}
//         fill="none"
//         xmlns="http://www.w3.org/2000/svg"
//       >
//         {/* Mask for clipping */}
//         <mask
//           id="mask0"
//           style={{ maskType: 'alpha' }}
//           maskUnits="userSpaceOnUse"
//           x="1"
//           y="1"
//           width={this.width - 2}
//           height={this.height - 2}
//         >
//           <rect
//             width={this.width - 2}
//             height={this.height - 2}
//             transform="translate(1 1)"
//             fill="url(#paint0_linear)"
//           />
//         </mask>
//         <g mask="url(#mask0)">
//           {/* Dynamic path generated from data */}
//           <path
//             d={pathD}
//             fill={this.fillColor}
//             opacity={this.opacity}
//           />
//         </g>
//         {/* Border or outline path */}
//         <path
//           d={pathD}
//           stroke={this.strokeColor}
//           stroke-width={this.strokeWidth}
//           stroke-linecap="round"
//           stroke-linejoin="round"
//         />
//         <defs>
//           <linearGradient
//             id="paint0_linear"
//             x1="164"
//             y1="0"
//             x2="164"
//             y2="64"
//             gradientUnits="userSpaceOnUse"
//           >
//             <stop />
//             <stop offset="1" stop-opacity="0" />
//           </linearGradient>
//         </defs>
//       </svg>
//     );
//   }
// }
// import { Component, Prop, h } from '@stencil/core';
// @Component({
//   tag: 'dga-realistic-chart',
//   // styleUrl: 'dga-chart.css',
//   shadow: true,
// })
// export class DgaRealisticChart {
//   @Prop() data: number[] = [];
//   @Prop() width: number = 330;
//   @Prop() height: number = 66;
//   private calculatePath(): string {
//     // Function to generate SVG path based on data
//     const maxData = Math.max(...this.data);
//     const minData = Math.min(...this.data);
//     const range = maxData - minData || 1;
//     const scaleX = this.width / (this.data.length - 1);
//     const scaleY = this.height / range;
//     return this.data
//       .map((point, i) => {
//         const x = i * scaleX;
//         const y = this.height - (point - minData) * scaleY;
//         return `${i === 0 ? 'M' : 'L'} ${x} ${y}`;
//       })
//       .join(' ');
//   }
//   render() {
//     const pathData = this.calculatePath();
//     return (
//       <svg width={this.width} height={this.height} viewBox={`0 0 ${this.width} ${this.height}`} xmlns="http://www.w3.org/2000/svg">
//         <mask id="mask0" maskUnits="userSpaceOnUse" x="1" y="1" width={this.width - 2} height={this.height - 2}>
//           <rect width={this.width - 2} height={this.height - 2} fill="url(#paint0_linear)" />
//         </mask>
//         <g mask="url(#mask0)">
//           <path
//             d={pathData}
//             opacity="0.2"
//             fill="#079455"
//           />
//         </g>
//         <path
//           d={pathData}
//           stroke="#067647"
//           stroke-width="2"
//           stroke-linecap="round"
//           stroke-linejoin="round"
//         />
//         {/* Optional marker */}
//         <g opacity="0.2">
//           <rect x="210.786" y="3.78571" width="20.4286" height="20.4286" rx="10.2143" stroke="#067647" stroke-width="1.57143" />
//         </g>
//         <rect x="215.786" y="8.78571" width="10.4286" height="10.4286" rx="5.21429" fill="white" />
//         <rect x="215.786" y="8.78571" width="10.4286" height="10.4286" rx="5.21429" stroke="#067647" stroke-width="1.57143" />
//         <defs>
//           <linearGradient id="paint0_linear" x1="164" y1="0" x2="164" y2="64" gradientUnits="userSpaceOnUse">
//             <stop />
//             <stop offset="1" stop-opacity="0" />
//           </linearGradient>
//         </defs>
//       </svg>
//     );
//   }
// }
import { h } from "@stencil/core";
export class DgaMetricChartV2 {
    constructor() {
        this.width = 130;
        this.height = 66;
        this.strokeColor = '#067647';
        this.fillColor = '#079455';
        this.strokeWidth = 2;
        this.data = [];
        this.showMarker = true;
        this.pathD = '';
        this.areaD = '';
        this.currentMarkerPos = { x: 76, y: 5 }; // default position
    }
    dataChangeHandler() {
        this.generatePaths();
        this.updateMarkerPosition();
    }
    componentWillLoad() {
        this.generatePaths();
        this.updateMarkerPosition();
    }
    normalizeData() {
        if (this.data.length === 0)
            return [];
        const xValues = this.data.map(d => d.x);
        const yValues = this.data.map(d => d.y);
        const minX = Math.min(...xValues);
        const maxX = Math.max(...xValues);
        const minY = Math.min(...yValues);
        const maxY = Math.max(...yValues);
        const margin = 10;
        return this.data.map(point => ({
            x: ((point.x - minX) / (maxX - minX)) * (128 - 2 * margin) + margin,
            y: 64 - (((point.y - minY) / (maxY - minY)) * (64 - 2 * margin) + margin)
        }));
    }
    updateMarkerPosition() {
        if (this.markerPosition) {
            this.currentMarkerPos = this.markerPosition;
        }
        else if (this.data.length > 0) {
            const normalizedData = this.normalizeData();
            const lastPoint = normalizedData[normalizedData.length - 1];
            // Calculate marker position based on the last data point
            // Adjust these offsets to position the marker correctly relative to the point
            this.currentMarkerPos = {
                x: lastPoint.x - 10, // Adjust these offsets as needed
                y: lastPoint.y - 15 // Adjust these offsets as needed
            };
        }
    }
    generatePaths() {
        const normalizedData = this.normalizeData();
        if (normalizedData.length === 0)
            return;
        this.pathD = normalizedData.reduce((path, point, i) => {
            return path + `${i === 0 ? 'M' : 'L'} ${point.x} ${point.y} `;
        }, '');
        this.areaD = `${this.pathD} L ${normalizedData[normalizedData.length - 1].x} 65 L 1 65 Z`;
    }
    render() {
        const { x: markerX, y: markerY } = this.currentMarkerPos;
        return (h("svg", { key: 'ae4f6c6ce47f55d7be763bc451a82c6296c8871e', width: this.width, height: this.height, viewBox: "0 0 130 66", fill: "none", xmlns: "http://www.w3.org/2000/svg" }, h("defs", { key: '373160ff8e80c0771458d48eb57da931b33815c5' }, h("linearGradient", { key: 'c5b44dcaefc231d0f57c2b075f4c1cfb299fdfe3', id: "gradientMask", x1: "64", y1: "0", x2: "64", y2: "64", gradientUnits: "userSpaceOnUse" }, h("stop", { key: '93e39951f14f45d6e549820999d09e5757fd4f6f' }), h("stop", { key: '4168b4b2042bdd02c37dc0b513eef83f29e79611', offset: "1", "stop-opacity": "0" })), h("mask", { key: '935ec0798a2cc114e46df0a066fc1fe10478f816', id: "chartMask", style: { maskType: 'alpha' }, maskUnits: "userSpaceOnUse", x: "1", y: "1", width: "128", height: "64" }, h("rect", { key: '44a4f224977ec36e70dd7c7b99597015da8b6086', width: "128", height: "64", transform: "translate(1 1)", fill: "url(#gradientMask)" }))), h("g", { key: 'db2f234e332c89a7979bd8cc2ce6fbb61c84f7fe', mask: "url(#chartMask)" }, h("path", { key: '294437166a6343bb153a92ebd9cfdf51bb7de505', opacity: "0.2", d: this.areaD, fill: this.fillColor })), h("path", { key: '57d7bc4921af8228960ce546b72f1f827225ea72', d: this.pathD, stroke: this.strokeColor, "stroke-width": this.strokeWidth, "stroke-linecap": "round", "stroke-linejoin": "round", fill: "none" }), this.showMarker && (h("g", { key: '682ceb408c8a4e3ce22b38a7fe12b093801a8ee2', transform: `translate(${markerX}, ${markerY})` }, h("g", { key: '81446c80be31b90e89538e4546bd98462f17721d', opacity: "0.2" }, h("rect", { key: '1ba0df8a0ec569126b421f81c1edafd4199c3c48', width: "20", height: "20", rx: "10", stroke: this.strokeColor, "stroke-width": this.strokeWidth })), h("rect", { key: '8fdb7874d2a8c679584266173041026d55e2af80', x: "5", y: "5", width: "10", height: "10", rx: "5", fill: "white" }), h("rect", { key: 'd97a882fea1dccce6a7a5f6a6f0e1af5aeab8568', x: "5", y: "5", width: "10", height: "10", rx: "5", stroke: this.strokeColor, "stroke-width": this.strokeWidth })))));
    }
    static get is() { return "dga-metric-chart-v2"; }
    static get encapsulation() { return "shadow"; }
    static get properties() {
        return {
            "width": {
                "type": "number",
                "attribute": "width",
                "mutable": false,
                "complexType": {
                    "original": "number",
                    "resolved": "number",
                    "references": {}
                },
                "required": false,
                "optional": false,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false,
                "defaultValue": "130"
            },
            "height": {
                "type": "number",
                "attribute": "height",
                "mutable": false,
                "complexType": {
                    "original": "number",
                    "resolved": "number",
                    "references": {}
                },
                "required": false,
                "optional": false,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false,
                "defaultValue": "66"
            },
            "strokeColor": {
                "type": "string",
                "attribute": "stroke-color",
                "mutable": false,
                "complexType": {
                    "original": "string",
                    "resolved": "string",
                    "references": {}
                },
                "required": false,
                "optional": false,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false,
                "defaultValue": "'#067647'"
            },
            "fillColor": {
                "type": "string",
                "attribute": "fill-color",
                "mutable": false,
                "complexType": {
                    "original": "string",
                    "resolved": "string",
                    "references": {}
                },
                "required": false,
                "optional": false,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false,
                "defaultValue": "'#079455'"
            },
            "strokeWidth": {
                "type": "number",
                "attribute": "stroke-width",
                "mutable": false,
                "complexType": {
                    "original": "number",
                    "resolved": "number",
                    "references": {}
                },
                "required": false,
                "optional": false,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false,
                "defaultValue": "2"
            },
            "data": {
                "type": "unknown",
                "attribute": "data",
                "mutable": false,
                "complexType": {
                    "original": "DataPoint[]",
                    "resolved": "DataPoint[]",
                    "references": {
                        "DataPoint": {
                            "location": "local",
                            "path": "/home/mamoud/Documents/temp/components/platform-code-cross-platform-storybook/packages/core/src/components/dga-metric/dga-metric-chart-v2.tsx",
                            "id": "src/components/dga-metric/dga-metric-chart-v2.tsx::DataPoint"
                        }
                    }
                },
                "required": false,
                "optional": false,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "defaultValue": "[]"
            },
            "showMarker": {
                "type": "boolean",
                "attribute": "show-marker",
                "mutable": false,
                "complexType": {
                    "original": "boolean",
                    "resolved": "boolean",
                    "references": {}
                },
                "required": false,
                "optional": false,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false,
                "defaultValue": "true"
            },
            "markerPosition": {
                "type": "unknown",
                "attribute": "marker-position",
                "mutable": false,
                "complexType": {
                    "original": "MarkerPosition",
                    "resolved": "MarkerPosition",
                    "references": {
                        "MarkerPosition": {
                            "location": "local",
                            "path": "/home/mamoud/Documents/temp/components/platform-code-cross-platform-storybook/packages/core/src/components/dga-metric/dga-metric-chart-v2.tsx",
                            "id": "src/components/dga-metric/dga-metric-chart-v2.tsx::MarkerPosition"
                        }
                    }
                },
                "required": false,
                "optional": true,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false
            }
        };
    }
    static get watchers() {
        return [{
                "propName": "data",
                "methodName": "dataChangeHandler"
            }];
    }
}
