// @platformscode/core/dist/collection/components/dga-notification/dga-notification.js
import { Host, h } from "@stencil/core";
import { SxStyleManager } from "../../utils/sx-style-manager";
export class DgaNotification {
    constructor() {
        this.icon = true;
        this.link = false;
        this.dismissable = false;
        this.icons = {
            success: {
                name: "checkmark-circle-02",
                variant: "solid",
                type: "sharp",
            },
            info: {
                name: "information-circle",
                variant: "solid",
                type: "sharp",
            },
            neutral: {
                name: "information-circle",
                variant: "solid",
                type: "sharp",
            },
            warning: {
                name: "alert-02",
                variant: "solid",
                type: "sharp",
            },
            critical: {
                name: "alert-circle",
                variant: "solid",
                type: "sharp",
            },
        };
        // Generate a unique ID if not provided
        this.componentId = `dga-notification-${Math.random().toString(36).substring(2, 9)}`;
        // Computed inline styles
        this.rootStyles = {};
    }
    handleSxChange() {
        this.processStyling();
    }
    componentWillLoad() {
        // Initialize style manager with element reference
        this.sxStyleManager = new SxStyleManager(this.componentId, this.el);
    }
    componentDidLoad() {
        // Process styling after the component is fully loaded and shadow DOM is ready
        this.processStyling();
    }
    componentDidUpdate() {
        // Re-process styling after updates
        this.processStyling();
    }
    processStyling() {
        if (!this.sx)
            return;
        // Handle both string and object formats
        const sxValue = typeof this.sx === 'string' ? JSON.parse(this.sx) : this.sx;
        this.rootStyles = this.sxStyleManager.processSxProp(sxValue);
    }
    disconnectedCallback() {
        var _a;
        // Clean up styles when component is removed
        (_a = this.sxStyleManager) === null || _a === void 0 ? void 0 : _a.cleanup();
    }
    render() {
        return (h(Host, { key: '579d2614e65103439af62032332c6a2e208f03c6', id: this.componentId, style: this.rootStyles }, h("div", { key: '9aacdd487c273ccf9acb7bf8f9b20ad32dad0636', class: `notification notification--${this.variant}`, role: "alert" }, h("div", { key: '2fb5e89d0abebe8028233bd0bd1a6d60a2c0b37b', class: "notification__content" }, this.icon && h("div", { key: 'a3e728c59d97c219cd3fbd5315bc80db953d6071', class: "notification__icon" }, h("dga-icon", { key: '1fde62052ad50ccb8d29062791c9b41b9e00ecc5', name: this.icons[this.variant].name, variant: this.icons[this.variant].variant, type: this.icons[this.variant].type, size: 24 })), this.leadText && h("strong", { key: '99bf3bcc33b5467941c42ed7bd6d54b8be329b65', class: "notification__lead-text" }, this.leadText, ": "), h("p", { key: 'a5ccee40878e673a863dab578484ec1554caf4dd' }, this.content), this.link && (h("a", { key: '2a99612f46242e3b34ea5b4d61a7e8f5818318e0', onClick: () => {
                this.navigateTo && this.navigateTo(this.setNavigateTo);
            }, class: "notification__link link link--md link--neutral link--inline" }, h("span", { key: '8106eae1476e9912aabdb45f773c195e642d5e86', class: "link__label" }, this.textLink)))), this.dismissable && (h("div", { key: 'ddd12f7c22c4079add4f1b744f102f2c4938c055', class: "notification__dismiss" }, h("button", { key: '4ce90062a91a8103cbaf573a6dc464ef6df9aa79', class: "dga-btn dga-btn--close dga-btn--icon dga-btn--md", onClick: () => {
                this.dismmissed && this.dismmissed();
            } }, h("span", { key: '2ed65757bae2ea164ac58aa61dd1038c78797f2a', class: "dga-btn-icon" }, h("dga-icon", { key: '37247fcf010d8f4f876b4e4f8d55ca4402576625', name: "multiplication-sign", variant: 'stroke', size: 20 }))))))));
    }
    static get is() { return "dga-notification"; }
    static get encapsulation() { return "shadow"; }
    static get originalStyleUrls() {
        return {
            "$": ["dga-notification.scss"]
        };
    }
    static get styleUrls() {
        return {
            "$": ["dga-notification.css"]
        };
    }
    static get properties() {
        return {
            "icon": {
                "type": "boolean",
                "attribute": "icon",
                "mutable": false,
                "complexType": {
                    "original": "boolean",
                    "resolved": "boolean",
                    "references": {}
                },
                "required": false,
                "optional": true,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false,
                "defaultValue": "true"
            },
            "leadText": {
                "type": "string",
                "attribute": "lead-text",
                "mutable": false,
                "complexType": {
                    "original": "string",
                    "resolved": "string",
                    "references": {}
                },
                "required": false,
                "optional": false,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false
            },
            "content": {
                "type": "string",
                "attribute": "content",
                "mutable": false,
                "complexType": {
                    "original": "string",
                    "resolved": "string",
                    "references": {}
                },
                "required": false,
                "optional": false,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false
            },
            "variant": {
                "type": "string",
                "attribute": "variant",
                "mutable": false,
                "complexType": {
                    "original": "'critical' | 'warning' | 'success' | 'info' | 'neutral'",
                    "resolved": "\"critical\" | \"info\" | \"neutral\" | \"success\" | \"warning\"",
                    "references": {}
                },
                "required": false,
                "optional": false,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false
            },
            "link": {
                "type": "boolean",
                "attribute": "link",
                "mutable": false,
                "complexType": {
                    "original": "boolean",
                    "resolved": "boolean",
                    "references": {}
                },
                "required": false,
                "optional": true,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false,
                "defaultValue": "false"
            },
            "textLink": {
                "type": "string",
                "attribute": "text-link",
                "mutable": false,
                "complexType": {
                    "original": "string",
                    "resolved": "string",
                    "references": {}
                },
                "required": false,
                "optional": true,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false
            },
            "dismissable": {
                "type": "boolean",
                "attribute": "dismissable",
                "mutable": false,
                "complexType": {
                    "original": "boolean",
                    "resolved": "boolean",
                    "references": {}
                },
                "required": false,
                "optional": true,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false,
                "defaultValue": "false"
            },
            "setNavigateTo": {
                "type": "string",
                "attribute": "set-navigate-to",
                "mutable": false,
                "complexType": {
                    "original": "string",
                    "resolved": "string",
                    "references": {}
                },
                "required": false,
                "optional": true,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false
            },
            "navigateTo": {
                "type": "unknown",
                "attribute": "navigate-to",
                "mutable": false,
                "complexType": {
                    "original": "(link?: string) => void",
                    "resolved": "(link?: string) => void",
                    "references": {}
                },
                "required": false,
                "optional": true,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false
            },
            "dismmissed": {
                "type": "unknown",
                "attribute": "dismmissed",
                "mutable": false,
                "complexType": {
                    "original": "() => void",
                    "resolved": "() => void",
                    "references": {}
                },
                "required": false,
                "optional": true,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false
            },
            "sx": {
                "type": "string",
                "attribute": "sx",
                "mutable": false,
                "complexType": {
                    "original": "SxProps | string",
                    "resolved": "string | { [x: string]: any; }",
                    "references": {
                        "SxProps": {
                            "location": "import",
                            "path": "../../utils/sx-style-manager",
                            "id": "src/utils/sx-style-manager.ts::SxProps"
                        }
                    }
                },
                "required": false,
                "optional": true,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false
            }
        };
    }
    static get elementRef() { return "el"; }
    static get watchers() {
        return [{
                "propName": "sx",
                "methodName": "handleSxChange"
            }];
    }
}
