// @platformscode/core/dist/collection/components/dga-card/dga-card.js
import { h, Host } from "@stencil/core";
import { SxStyleManager } from "../../utils/sx-style-manager";
export class DgaCard {
    constructor() {
        this.type = 'default';
        this.showTitle = true;
        this.showDescription = true;
        this.showFeaturedIcon = true;
        this.image = '';
        this.disabled = false;
        this.effect = 'with-shadow';
        this.cardTitle = 'Card Title';
        this.description = 'Card content placeholder text goes here';
        this.featuredIcon = {
            name: 'checkmark-circle-02',
            variant: 'stroke',
            type: 'sharp',
            size: 24,
        }; // Default icon
        this.showPrimaryAction = true;
        this.primaryActionLabel = 'Primary Action';
        this.showSecondaryAction = true;
        this.isSelected = false;
        this.secondaryActionLabel = 'Secondary Action';
        this.onCheckboxChange = () => { };
        this.onPrimaryAction = () => { };
        this.onSecondaryAction = () => { };
        this.onExpandAction = () => { };
        this.isFocused = false;
        this.expanded = false;
        // Generate a unique ID if not provided
        this.componentId = `dga-button-${Math.random().toString(36).substring(2, 9)}`;
        // Computed inline styles
        this.rootStyles = {};
    }
    toggleExpand() {
        this.expanded = !this.expanded;
    }
    handleSxChange() {
        this.processStyling();
    }
    componentWillLoad() {
        // Initialize style manager with element reference
        this.sxStyleManager = new SxStyleManager(this.componentId, this.el);
    }
    componentDidLoad() {
        // Process styling after the component is fully loaded and shadow DOM is ready
        this.processStyling();
    }
    componentDidUpdate() {
        // Re-process styling after updates
        this.processStyling();
    }
    processStyling() {
        if (!this.sx)
            return;
        // Handle both string and object formats
        const sxValue = typeof this.sx === 'string'
            ? JSON.parse(this.sx)
            : this.sx;
        this.rootStyles = this.sxStyleManager.processSxProp(sxValue);
    }
    disconnectedCallback() {
        var _a;
        // Clean up styles when component is removed
        (_a = this.sxStyleManager) === null || _a === void 0 ? void 0 : _a.cleanup();
    }
    render() {
        return (h(Host, { key: '8261ba2914cdf2a5f75807c02cb6aa8a14a360db', id: this.componentId, style: this.rootStyles }, h("div", Object.assign({ key: '05520314bec93fd15322af30672a6e0e3b749d5b', class: `card dga-flex-column ${this.disabled ? 'disabled' : ''} ${this.isFocused ? 'focused' : ''} card--${this.effect}` }, (this.type === 'selectable' ? { onClick: this.onCheckboxChange } : {}), { tabIndex: 0 }), this.image && (h("div", { key: '2a636e776587b74671b8b0e66a07d70a2f477372', class: "card__image-container" }, h("img", { key: 'fec683586ffcad88bae4462c6c8feef916aceebd', src: this.image, alt: "Card" }))), this.showFeaturedIcon && (h("dga-featured-icon", { key: '5a6770039768c3e6f5f31366e3eabf1e9d48c3c3', icon: this.featuredIcon, size: 'lg' })), this.type === 'selectable' && (h("div", { key: 'eb0da9002baac0cef488cd86cf79bf5b14eb7684', class: "card__checkbox-container" }, h("dga-checkbox", { key: '1688b5c48ffe235a66b84c1b78f6d105e9bd79b9', checked: this.isSelected, color: "brand", onChange: this.onCheckboxChange }))), (this.showTitle || this.showDescription) && (h("div", { key: '53a67e1e526fcc9e00076bfebb77e2e39fbd07ef', class: "card__content-text dga-flex-column" }, this.showTitle && h("h4", { key: '99884fc68c04fac6b999a5487d4d955567e2c1f3', class: "text-lg-bold" }, this.cardTitle, " "), this.showDescription && h("p", { key: '181870477d9cb6fc68e7bfdadb3ab86c354224c0', class: "text-md-regular" }, this.description))), h("slot", { key: '53d408370d8a6b3ef08ac78c81acdbc0f0981857' }), this.type === 'default' && (this.showPrimaryAction || this.showSecondaryAction) && (h("div", { key: 'bbb70fa1fb8e4fe32eef311d5b610d93a457eb64', class: "card__action card__action--default" }, this.showSecondaryAction && h("dga-button", { key: 'b1d11e34257876fb1b3860a58853966dffe0a86f', label: this.secondaryActionLabel, onClick: this.onSecondaryAction, size: "lg", variant: "secondary-outline" }), this.showPrimaryAction && h("dga-button-v2", { key: '6784cd356f4de682fdb9113ad1f81fde11a959d9', label: this.primaryActionLabel, onClick: this.onPrimaryAction, size: "lg", variant: "primary" }))), this.type === 'expanded' && (h("div", { key: '57d9735c731eb75d904e23a3cc26235b8cc42019', class: "card__action card__action--expanded" }, h("dga-button-v2", { key: this.expanded ? 'expanded' : 'collapsed', iconOnly: true, onClick: () => this.toggleExpand(), size: "lg", variant: "subtle", leadIcon: true, leadIconProps: {
                name: this.expanded ? 'arrow-up-01' : 'arrow-down-01',
                variant: 'stroke',
                type: 'rounded',
            } }))), h("div", { key: '2ec837d8fc00d69d134e480a4ba38d583e3564b6', class: `collapse-content ${this.expanded ? 'expanded' : ''}` }, h("slot", { key: '1b5f1d9a5dd28ed4e86781a044336b20d1c033e9', name: "collapse-content" })))));
    }
    static get is() { return "dga-card"; }
    static get encapsulation() { return "shadow"; }
    static get originalStyleUrls() {
        return {
            "$": ["dga-card.scss"]
        };
    }
    static get styleUrls() {
        return {
            "$": ["dga-card.css"]
        };
    }
    static get properties() {
        return {
            "sx": {
                "type": "string",
                "attribute": "sx",
                "mutable": false,
                "complexType": {
                    "original": "SxProps | string",
                    "resolved": "string | { [x: string]: any; }",
                    "references": {
                        "SxProps": {
                            "location": "import",
                            "path": "../../utils/sx-style-manager",
                            "id": "src/utils/sx-style-manager.ts::SxProps"
                        }
                    }
                },
                "required": false,
                "optional": true,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false
            },
            "type": {
                "type": "string",
                "attribute": "type",
                "mutable": false,
                "complexType": {
                    "original": "'default' | 'expanded' | 'selectable'",
                    "resolved": "\"default\" | \"expanded\" | \"selectable\"",
                    "references": {}
                },
                "required": false,
                "optional": true,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false,
                "defaultValue": "'default'"
            },
            "showTitle": {
                "type": "boolean",
                "attribute": "show-title",
                "mutable": false,
                "complexType": {
                    "original": "boolean",
                    "resolved": "boolean",
                    "references": {}
                },
                "required": false,
                "optional": true,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false,
                "defaultValue": "true"
            },
            "showDescription": {
                "type": "boolean",
                "attribute": "show-description",
                "mutable": false,
                "complexType": {
                    "original": "boolean",
                    "resolved": "boolean",
                    "references": {}
                },
                "required": false,
                "optional": true,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false,
                "defaultValue": "true"
            },
            "showFeaturedIcon": {
                "type": "boolean",
                "attribute": "show-featured-icon",
                "mutable": false,
                "complexType": {
                    "original": "boolean",
                    "resolved": "boolean",
                    "references": {}
                },
                "required": false,
                "optional": true,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false,
                "defaultValue": "true"
            },
            "image": {
                "type": "string",
                "attribute": "image",
                "mutable": false,
                "complexType": {
                    "original": "string",
                    "resolved": "string",
                    "references": {}
                },
                "required": false,
                "optional": true,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false,
                "defaultValue": "''"
            },
            "disabled": {
                "type": "boolean",
                "attribute": "disabled",
                "mutable": false,
                "complexType": {
                    "original": "boolean",
                    "resolved": "boolean",
                    "references": {}
                },
                "required": false,
                "optional": true,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false,
                "defaultValue": "false"
            },
            "effect": {
                "type": "string",
                "attribute": "effect",
                "mutable": false,
                "complexType": {
                    "original": "'with-shadow' | 'no-shadow' | 'stroke'",
                    "resolved": "\"no-shadow\" | \"stroke\" | \"with-shadow\"",
                    "references": {}
                },
                "required": false,
                "optional": true,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false,
                "defaultValue": "'with-shadow'"
            },
            "cardTitle": {
                "type": "string",
                "attribute": "card-title",
                "mutable": false,
                "complexType": {
                    "original": "string",
                    "resolved": "string",
                    "references": {}
                },
                "required": false,
                "optional": true,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false,
                "defaultValue": "'Card Title'"
            },
            "description": {
                "type": "string",
                "attribute": "description",
                "mutable": false,
                "complexType": {
                    "original": "string",
                    "resolved": "string",
                    "references": {}
                },
                "required": false,
                "optional": true,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false,
                "defaultValue": "'Card content placeholder text goes here'"
            },
            "featuredIcon": {
                "type": "unknown",
                "attribute": "featured-icon",
                "mutable": false,
                "complexType": {
                    "original": "Icon",
                    "resolved": "Icon",
                    "references": {
                        "Icon": {
                            "location": "import",
                            "path": "../dga-featured-icon/dga-featured-icon",
                            "id": "src/components/dga-featured-icon/dga-featured-icon.tsx::Icon"
                        }
                    }
                },
                "required": false,
                "optional": true,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "defaultValue": "{\n    name: 'checkmark-circle-02',\n    variant: 'stroke',\n    type: 'sharp',\n    size: 24,\n  }"
            },
            "showPrimaryAction": {
                "type": "boolean",
                "attribute": "show-primary-action",
                "mutable": false,
                "complexType": {
                    "original": "boolean",
                    "resolved": "boolean",
                    "references": {}
                },
                "required": false,
                "optional": true,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false,
                "defaultValue": "true"
            },
            "primaryActionLabel": {
                "type": "string",
                "attribute": "primary-action-label",
                "mutable": false,
                "complexType": {
                    "original": "string",
                    "resolved": "string",
                    "references": {}
                },
                "required": false,
                "optional": true,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false,
                "defaultValue": "'Primary Action'"
            },
            "showSecondaryAction": {
                "type": "boolean",
                "attribute": "show-secondary-action",
                "mutable": false,
                "complexType": {
                    "original": "boolean",
                    "resolved": "boolean",
                    "references": {}
                },
                "required": false,
                "optional": true,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false,
                "defaultValue": "true"
            },
            "isSelected": {
                "type": "boolean",
                "attribute": "is-selected",
                "mutable": true,
                "complexType": {
                    "original": "boolean",
                    "resolved": "boolean",
                    "references": {}
                },
                "required": false,
                "optional": true,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": true,
                "defaultValue": "false"
            },
            "secondaryActionLabel": {
                "type": "string",
                "attribute": "secondary-action-label",
                "mutable": false,
                "complexType": {
                    "original": "string",
                    "resolved": "string",
                    "references": {}
                },
                "required": false,
                "optional": true,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false,
                "defaultValue": "'Secondary Action'"
            },
            "onCheckboxChange": {
                "type": "unknown",
                "attribute": "on-checkbox-change",
                "mutable": false,
                "complexType": {
                    "original": "() => void",
                    "resolved": "() => void",
                    "references": {}
                },
                "required": false,
                "optional": true,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "defaultValue": "() => {}"
            },
            "onPrimaryAction": {
                "type": "unknown",
                "attribute": "on-primary-action",
                "mutable": false,
                "complexType": {
                    "original": "() => void",
                    "resolved": "() => void",
                    "references": {}
                },
                "required": false,
                "optional": true,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "defaultValue": "() => {}"
            },
            "onSecondaryAction": {
                "type": "unknown",
                "attribute": "on-secondary-action",
                "mutable": false,
                "complexType": {
                    "original": "() => void",
                    "resolved": "() => void",
                    "references": {}
                },
                "required": false,
                "optional": true,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "defaultValue": "() => {}"
            },
            "onExpandAction": {
                "type": "unknown",
                "attribute": "on-expand-action",
                "mutable": false,
                "complexType": {
                    "original": "() => void",
                    "resolved": "() => void",
                    "references": {}
                },
                "required": false,
                "optional": true,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "defaultValue": "() => {}"
            }
        };
    }
    static get states() {
        return {
            "isFocused": {},
            "expanded": {}
        };
    }
    static get elementRef() { return "el"; }
    static get watchers() {
        return [{
                "propName": "sx",
                "methodName": "handleSxChange"
            }];
    }
}
