// platformscode-new-react/dist/components/stencil-generated/components.js
'use client';
import { DateCell as DateCellElement, defineCustomElement as defineDateCell } from "@platformscode/core/dist/components/date-cell.js";
import { DgaAccordion as DgaAccordionElement, defineCustomElement as defineDgaAccordion } from "@platformscode/core/dist/components/dga-accordion.js";
import { DgaActionMenu as DgaActionMenuElement, defineCustomElement as defineDgaActionMenu } from "@platformscode/core/dist/components/dga-action-menu.js";
import { DgaActions as DgaActionsElement, defineCustomElement as defineDgaActions } from "@platformscode/core/dist/components/dga-actions.js";
import { DgaApexChart as DgaApexChartElement, defineCustomElement as defineDgaApexChart } from "@platformscode/core/dist/components/dga-apex-chart.js";
import { DgaAvatarGroup as DgaAvatarGroupElement, defineCustomElement as defineDgaAvatarGroup } from "@platformscode/core/dist/components/dga-avatar-group.js";
import { DgaAvatar as DgaAvatarElement, defineCustomElement as defineDgaAvatar } from "@platformscode/core/dist/components/dga-avatar.js";
import { DgaBreadcrumbs as DgaBreadcrumbsElement, defineCustomElement as defineDgaBreadcrumbs } from "@platformscode/core/dist/components/dga-breadcrumbs.js";
import { DgaButtonLeadingIcon as DgaButtonLeadingIconElement, defineCustomElement as defineDgaButtonLeadingIcon } from "@platformscode/core/dist/components/dga-button-leading-icon.js";
import { DgaButtonV2Usage as DgaButtonV2UsageElement, defineCustomElement as defineDgaButtonV2Usage } from "@platformscode/core/dist/components/dga-button-v2-usage.js";
import { DgaButtonV2 as DgaButtonV2Element, defineCustomElement as defineDgaButtonV2 } from "@platformscode/core/dist/components/dga-button-v2.js";
import { DgaButton as DgaButtonElement, defineCustomElement as defineDgaButton } from "@platformscode/core/dist/components/dga-button.js";
import { DgaCardActions as DgaCardActionsElement, defineCustomElement as defineDgaCardActions } from "@platformscode/core/dist/components/dga-card-actions.js";
import { DgaCardContent as DgaCardContentElement, defineCustomElement as defineDgaCardContent } from "@platformscode/core/dist/components/dga-card-content.js";
import { DgaCardExpandableContent as DgaCardExpandableContentElement, defineCustomElement as defineDgaCardExpandableContent } from "@platformscode/core/dist/components/dga-card-expandable-content.js";
import { DgaCardImage as DgaCardImageElement, defineCustomElement as defineDgaCardImage } from "@platformscode/core/dist/components/dga-card-image.js";
import { DgaCardTitle as DgaCardTitleElement, defineCustomElement as defineDgaCardTitle } from "@platformscode/core/dist/components/dga-card-title.js";
import { DgaCardV2 as DgaCardV2Element, defineCustomElement as defineDgaCardV2 } from "@platformscode/core/dist/components/dga-card-v2.js";
import { DgaCard as DgaCardElement, defineCustomElement as defineDgaCard } from "@platformscode/core/dist/components/dga-card.js";
import { DgaCarouselItem as DgaCarouselItemElement, defineCustomElement as defineDgaCarouselItem } from "@platformscode/core/dist/components/dga-carousel-item.js";
import { DgaCarouselV2 as DgaCarouselV2Element, defineCustomElement as defineDgaCarouselV2 } from "@platformscode/core/dist/components/dga-carousel-v2.js";
import { DgaCarousel as DgaCarouselElement, defineCustomElement as defineDgaCarousel } from "@platformscode/core/dist/components/dga-carousel.js";
import { DgaChart as DgaChartElement, defineCustomElement as defineDgaChart } from "@platformscode/core/dist/components/dga-chart.js";
import { DgaCheckbox as DgaCheckboxElement, defineCustomElement as defineDgaCheckbox } from "@platformscode/core/dist/components/dga-checkbox.js";
import { DgaChip as DgaChipElement, defineCustomElement as defineDgaChip } from "@platformscode/core/dist/components/dga-chip.js";
import { DgaCircleSkeleton as DgaCircleSkeletonElement, defineCustomElement as defineDgaCircleSkeleton } from "@platformscode/core/dist/components/dga-circle-skeleton.js";
import { DgaCircularProgressBar as DgaCircularProgressBarElement, defineCustomElement as defineDgaCircularProgressBar } from "@platformscode/core/dist/components/dga-circular-progress-bar.js";
import { DgaCodesnippet as DgaCodesnippetElement, defineCustomElement as defineDgaCodesnippet } from "@platformscode/core/dist/components/dga-codesnippet.js";
import { DgaCollapseUsage as DgaCollapseUsageElement, defineCustomElement as defineDgaCollapseUsage } from "@platformscode/core/dist/components/dga-collapse-usage.js";
import { DgaCollapse as DgaCollapseElement, defineCustomElement as defineDgaCollapse } from "@platformscode/core/dist/components/dga-collapse.js";
import { DgaContentSwitcherItem as DgaContentSwitcherItemElement, defineCustomElement as defineDgaContentSwitcherItem } from "@platformscode/core/dist/components/dga-content-switcher-item.js";
import { DgaContentSwitcherPanel as DgaContentSwitcherPanelElement, defineCustomElement as defineDgaContentSwitcherPanel } from "@platformscode/core/dist/components/dga-content-switcher-panel.js";
import { DgaContentSwitcher as DgaContentSwitcherElement, defineCustomElement as defineDgaContentSwitcher } from "@platformscode/core/dist/components/dga-content-switcher.js";
import { DgaDataTableUsage as DgaDataTableUsageElement, defineCustomElement as defineDgaDataTableUsage } from "@platformscode/core/dist/components/dga-data-table-usage.js";
import { DgaDataTable as DgaDataTableElement, defineCustomElement as defineDgaDataTable } from "@platformscode/core/dist/components/dga-data-table.js";
import { DgaDateFieldUsage as DgaDateFieldUsageElement, defineCustomElement as defineDgaDateFieldUsage } from "@platformscode/core/dist/components/dga-date-field-usage.js";
import { DgaDateField as DgaDateFieldElement, defineCustomElement as defineDgaDateField } from "@platformscode/core/dist/components/dga-date-field.js";
import { DgaDatepicker as DgaDatepickerElement, defineCustomElement as defineDgaDatepicker } from "@platformscode/core/dist/components/dga-datepicker.js";
import { DgaDigitalSignature as DgaDigitalSignatureElement, defineCustomElement as defineDgaDigitalSignature } from "@platformscode/core/dist/components/dga-digital-signature.js";
import { DgaDivider as DgaDividerElement, defineCustomElement as defineDgaDivider } from "@platformscode/core/dist/components/dga-divider.js";
import { DgaDrawer as DgaDrawerElement, defineCustomElement as defineDgaDrawer } from "@platformscode/core/dist/components/dga-drawer.js";
import { DgaDropdown as DgaDropdownElement, defineCustomElement as defineDgaDropdown } from "@platformscode/core/dist/components/dga-dropdown.js";
import { DgaFeaturedIcon as DgaFeaturedIconElement, defineCustomElement as defineDgaFeaturedIcon } from "@platformscode/core/dist/components/dga-featured-icon.js";
import { DgaFeedbackIconUsage as DgaFeedbackIconUsageElement, defineCustomElement as defineDgaFeedbackIconUsage } from "@platformscode/core/dist/components/dga-feedback-icon-usage.js";
import { DgaFeedbackIcon as DgaFeedbackIconElement, defineCustomElement as defineDgaFeedbackIcon } from "@platformscode/core/dist/components/dga-feedback-icon.js";
import { DgaFileUpload as DgaFileUploadElement, defineCustomElement as defineDgaFileUpload } from "@platformscode/core/dist/components/dga-file-upload.js";
import { DgaFilteration as DgaFilterationElement, defineCustomElement as defineDgaFilteration } from "@platformscode/core/dist/components/dga-filteration.js";
import { DgaFlex as DgaFlexElement, defineCustomElement as defineDgaFlex } from "@platformscode/core/dist/components/dga-flex.js";
import { DgaFooter as DgaFooterElement, defineCustomElement as defineDgaFooter } from "@platformscode/core/dist/components/dga-footer.js";
import { DgaGridContainer as DgaGridContainerElement, defineCustomElement as defineDgaGridContainer } from "@platformscode/core/dist/components/dga-grid-container.js";
import { DgaGridItem as DgaGridItemElement, defineCustomElement as defineDgaGridItem } from "@platformscode/core/dist/components/dga-grid-item.js";
import { DgaGridUsage as DgaGridUsageElement, defineCustomElement as defineDgaGridUsage } from "@platformscode/core/dist/components/dga-grid-usage.js";
import { DgaHeaderActionBtn as DgaHeaderActionBtnElement, defineCustomElement as defineDgaHeaderActionBtn } from "@platformscode/core/dist/components/dga-header-action-btn.js";
import { DgaHeaderAction as DgaHeaderActionElement, defineCustomElement as defineDgaHeaderAction } from "@platformscode/core/dist/components/dga-header-action.js";
import { DgaHeaderMenuItemUsage as DgaHeaderMenuItemUsageElement, defineCustomElement as defineDgaHeaderMenuItemUsage } from "@platformscode/core/dist/components/dga-header-menu-item-usage.js";
import { DgaHeaderMenuItem as DgaHeaderMenuItemElement, defineCustomElement as defineDgaHeaderMenuItem } from "@platformscode/core/dist/components/dga-header-menu-item.js";
import { DgaHeaderMenuVv2 as DgaHeaderMenuVv2Element, defineCustomElement as defineDgaHeaderMenuVv2 } from "@platformscode/core/dist/components/dga-header-menu-vv2.js";
import { DgaHeaderMenu as DgaHeaderMenuElement, defineCustomElement as defineDgaHeaderMenu } from "@platformscode/core/dist/components/dga-header-menu.js";
import { DgaHeaderMobileActions as DgaHeaderMobileActionsElement, defineCustomElement as defineDgaHeaderMobileActions } from "@platformscode/core/dist/components/dga-header-mobile-actions.js";
import { DgaHeaderSubMenuColumn as DgaHeaderSubMenuColumnElement, defineCustomElement as defineDgaHeaderSubMenuColumn } from "@platformscode/core/dist/components/dga-header-sub-menu-column.js";
import { DgaHeaderSubMenuItem as DgaHeaderSubMenuItemElement, defineCustomElement as defineDgaHeaderSubMenuItem } from "@platformscode/core/dist/components/dga-header-sub-menu-item.js";
import { DgaHeaderSubMenu as DgaHeaderSubMenuElement, defineCustomElement as defineDgaHeaderSubMenu } from "@platformscode/core/dist/components/dga-header-sub-menu.js";
import { DgaHeaderVv2 as DgaHeaderVv2Element, defineCustomElement as defineDgaHeaderVv2 } from "@platformscode/core/dist/components/dga-header-vv2.js";
import { DgaHeader as DgaHeaderElement, defineCustomElement as defineDgaHeader } from "@platformscode/core/dist/components/dga-header.js";
import { DgaHelperText as DgaHelperTextElement, defineCustomElement as defineDgaHelperText } from "@platformscode/core/dist/components/dga-helper-text.js";
import { DgaHugeIconUsage as DgaHugeIconUsageElement, defineCustomElement as defineDgaHugeIconUsage } from "@platformscode/core/dist/components/dga-huge-icon-usage.js";
import { DgaIcon as DgaIconElement, defineCustomElement as defineDgaIcon } from "@platformscode/core/dist/components/dga-icon.js";
import { DgaInlineAlert as DgaInlineAlertElement, defineCustomElement as defineDgaInlineAlert } from "@platformscode/core/dist/components/dga-inline-alert.js";
import { DgaLabel as DgaLabelElement, defineCustomElement as defineDgaLabel } from "@platformscode/core/dist/components/dga-label.js";
import { DgaLineSkeleton as DgaLineSkeletonElement, defineCustomElement as defineDgaLineSkeleton } from "@platformscode/core/dist/components/dga-line-skeleton.js";
import { DgaLinearProgressBar as DgaLinearProgressBarElement, defineCustomElement as defineDgaLinearProgressBar } from "@platformscode/core/dist/components/dga-linear-progress-bar.js";
import { DgaLink as DgaLinkElement, defineCustomElement as defineDgaLink } from "@platformscode/core/dist/components/dga-link.js";
import { DgaListItem as DgaListItemElement, defineCustomElement as defineDgaListItem } from "@platformscode/core/dist/components/dga-list-item.js";
import { DgaListV2 as DgaListV2Element, defineCustomElement as defineDgaListV2 } from "@platformscode/core/dist/components/dga-list-v2.js";
import { DgaList as DgaListElement, defineCustomElement as defineDgaList } from "@platformscode/core/dist/components/dga-list.js";
import { DgaLoading as DgaLoadingElement, defineCustomElement as defineDgaLoading } from "@platformscode/core/dist/components/dga-loading.js";
import { DgaLogoPlaceholder as DgaLogoPlaceholderElement, defineCustomElement as defineDgaLogoPlaceholder } from "@platformscode/core/dist/components/dga-logo-placeholder.js";
import { DgaMenuGroup as DgaMenuGroupElement, defineCustomElement as defineDgaMenuGroup } from "@platformscode/core/dist/components/dga-menu-group.js";
import { DgaMenuItem as DgaMenuItemElement, defineCustomElement as defineDgaMenuItem } from "@platformscode/core/dist/components/dga-menu-item.js";
import { DgaMenuList as DgaMenuListElement, defineCustomElement as defineDgaMenuList } from "@platformscode/core/dist/components/dga-menu-list.js";
import { DgaMenuV2 as DgaMenuV2Element, defineCustomElement as defineDgaMenuV2 } from "@platformscode/core/dist/components/dga-menu-v2.js";
import { DgaMenu as DgaMenuElement, defineCustomElement as defineDgaMenu } from "@platformscode/core/dist/components/dga-menu.js";
import { DgaMetricChartV2 as DgaMetricChartV2Element, defineCustomElement as defineDgaMetricChartV2 } from "@platformscode/core/dist/components/dga-metric-chart-v2.js";
import { DgaMetricChart as DgaMetricChartElement, defineCustomElement as defineDgaMetricChart } from "@platformscode/core/dist/components/dga-metric-chart.js";
import { DgaMetricV2 as DgaMetricV2Element, defineCustomElement as defineDgaMetricV2 } from "@platformscode/core/dist/components/dga-metric-v2.js";
import { DgaMetric as DgaMetricElement, defineCustomElement as defineDgaMetric } from "@platformscode/core/dist/components/dga-metric.js";
import { DgaModalActions as DgaModalActionsElement, defineCustomElement as defineDgaModalActions } from "@platformscode/core/dist/components/dga-modal-actions.js";
import { DgaModalBody as DgaModalBodyElement, defineCustomElement as defineDgaModalBody } from "@platformscode/core/dist/components/dga-modal-body.js";
import { DgaModalHeader as DgaModalHeaderElement, defineCustomElement as defineDgaModalHeader } from "@platformscode/core/dist/components/dga-modal-header.js";
import { DgaModalTitle as DgaModalTitleElement, defineCustomElement as defineDgaModalTitle } from "@platformscode/core/dist/components/dga-modal-title.js";
import { DgaModalV2Usage as DgaModalV2UsageElement, defineCustomElement as defineDgaModalV2Usage } from "@platformscode/core/dist/components/dga-modal-v2-usage.js";
import { DgaModalV2 as DgaModalV2Element, defineCustomElement as defineDgaModalV2 } from "@platformscode/core/dist/components/dga-modal-v2.js";
import { DgaModal as DgaModalElement, defineCustomElement as defineDgaModal } from "@platformscode/core/dist/components/dga-modal.js";
import { DgaNavHeaderActions as DgaNavHeaderActionsElement, defineCustomElement as defineDgaNavHeaderActions } from "@platformscode/core/dist/components/dga-nav-header-actions.js";
import { DgaNavHeaderLink as DgaNavHeaderLinkElement, defineCustomElement as defineDgaNavHeaderLink } from "@platformscode/core/dist/components/dga-nav-header-link.js";
import { DgaNavHeaderLogos as DgaNavHeaderLogosElement, defineCustomElement as defineDgaNavHeaderLogos } from "@platformscode/core/dist/components/dga-nav-header-logos.js";
import { DgaNavHeaderMain as DgaNavHeaderMainElement, defineCustomElement as defineDgaNavHeaderMain } from "@platformscode/core/dist/components/dga-nav-header-main.js";
import { DgaNavHeaderMenu as DgaNavHeaderMenuElement, defineCustomElement as defineDgaNavHeaderMenu } from "@platformscode/core/dist/components/dga-nav-header-menu.js";
import { DgaNavHeaderSubMenuColumn as DgaNavHeaderSubMenuColumnElement, defineCustomElement as defineDgaNavHeaderSubMenuColumn } from "@platformscode/core/dist/components/dga-nav-header-sub-menu-column.js";
import { DgaNavHeaderSubMenuLink as DgaNavHeaderSubMenuLinkElement, defineCustomElement as defineDgaNavHeaderSubMenuLink } from "@platformscode/core/dist/components/dga-nav-header-sub-menu-link.js";
import { DgaNavHeaderSubMenu as DgaNavHeaderSubMenuElement, defineCustomElement as defineDgaNavHeaderSubMenu } from "@platformscode/core/dist/components/dga-nav-header-sub-menu.js";
import { DgaNavHeader as DgaNavHeaderElement, defineCustomElement as defineDgaNavHeader } from "@platformscode/core/dist/components/dga-nav-header.js";
import { DgaNotificationToast as DgaNotificationToastElement, defineCustomElement as defineDgaNotificationToast } from "@platformscode/core/dist/components/dga-notification-toast.js";
import { DgaNotification as DgaNotificationElement, defineCustomElement as defineDgaNotification } from "@platformscode/core/dist/components/dga-notification.js";
import { DgaNumberInput as DgaNumberInputElement, defineCustomElement as defineDgaNumberInput } from "@platformscode/core/dist/components/dga-number-input.js";
import { DgaPagination as DgaPaginationElement, defineCustomElement as defineDgaPagination } from "@platformscode/core/dist/components/dga-pagination.js";
import { DgaPrefix as DgaPrefixElement, defineCustomElement as defineDgaPrefix } from "@platformscode/core/dist/components/dga-prefix.js";
import { DgaProgressIndicatorStepV3 as DgaProgressIndicatorStepV3Element, defineCustomElement as defineDgaProgressIndicatorStepV3 } from "@platformscode/core/dist/components/dga-progress-indicator-step-v3.js";
import { DgaProgressIndicatorStep as DgaProgressIndicatorStepElement, defineCustomElement as defineDgaProgressIndicatorStep } from "@platformscode/core/dist/components/dga-progress-indicator-step.js";
import { DgaProgressIndicatorV3Usage as DgaProgressIndicatorV3UsageElement, defineCustomElement as defineDgaProgressIndicatorV3Usage } from "@platformscode/core/dist/components/dga-progress-indicator-v3-usage.js";
import { DgaProgressIndicatorV3 as DgaProgressIndicatorV3Element, defineCustomElement as defineDgaProgressIndicatorV3 } from "@platformscode/core/dist/components/dga-progress-indicator-v3.js";
import { DgaProgressIndicator as DgaProgressIndicatorElement, defineCustomElement as defineDgaProgressIndicator } from "@platformscode/core/dist/components/dga-progress-indicator.js";
import { DgaQuote as DgaQuoteElement, defineCustomElement as defineDgaQuote } from "@platformscode/core/dist/components/dga-quote.js";
import { DgaRadialStepper as DgaRadialStepperElement, defineCustomElement as defineDgaRadialStepper } from "@platformscode/core/dist/components/dga-radial-stepper.js";
import { DgaRadioButton as DgaRadioButtonElement, defineCustomElement as defineDgaRadioButton } from "@platformscode/core/dist/components/dga-radio-button.js";
import { DgaRating as DgaRatingElement, defineCustomElement as defineDgaRating } from "@platformscode/core/dist/components/dga-rating.js";
import { DgaRectangleSkeleton as DgaRectangleSkeletonElement, defineCustomElement as defineDgaRectangleSkeleton } from "@platformscode/core/dist/components/dga-rectangle-skeleton.js";
import { DgaSearchBox as DgaSearchBoxElement, defineCustomElement as defineDgaSearchBox } from "@platformscode/core/dist/components/dga-search-box.js";
import { DgaSecondNavHeaderActions as DgaSecondNavHeaderActionsElement, defineCustomElement as defineDgaSecondNavHeaderActions } from "@platformscode/core/dist/components/dga-second-nav-header-actions.js";
import { DgaSecondNavHeaderContent as DgaSecondNavHeaderContentElement, defineCustomElement as defineDgaSecondNavHeaderContent } from "@platformscode/core/dist/components/dga-second-nav-header-content.js";
import { DgaSecondNavHeaderItem as DgaSecondNavHeaderItemElement, defineCustomElement as defineDgaSecondNavHeaderItem } from "@platformscode/core/dist/components/dga-second-nav-header-item.js";
import { DgaSecondNavHeader as DgaSecondNavHeaderElement, defineCustomElement as defineDgaSecondNavHeader } from "@platformscode/core/dist/components/dga-second-nav-header.js";
import { DgaSkeletonComponent as DgaSkeletonComponentElement, defineCustomElement as defineDgaSkeletonComponent } from "@platformscode/core/dist/components/dga-skeleton-component.js";
import { DgaSlideoutMenuActions as DgaSlideoutMenuActionsElement, defineCustomElement as defineDgaSlideoutMenuActions } from "@platformscode/core/dist/components/dga-slideout-menu-actions.js";
import { DgaSlideoutMenuHeader as DgaSlideoutMenuHeaderElement, defineCustomElement as defineDgaSlideoutMenuHeader } from "@platformscode/core/dist/components/dga-slideout-menu-header.js";
import { DgaSlideoutMenuItems as DgaSlideoutMenuItemsElement, defineCustomElement as defineDgaSlideoutMenuItems } from "@platformscode/core/dist/components/dga-slideout-menu-items.js";
import { DgaSlideoutMenu as DgaSlideoutMenuElement, defineCustomElement as defineDgaSlideoutMenu } from "@platformscode/core/dist/components/dga-slideout-menu.js";
import { DgaSlider as DgaSliderElement, defineCustomElement as defineDgaSlider } from "@platformscode/core/dist/components/dga-slider.js";
import { DgaSquareSkeleton as DgaSquareSkeletonElement, defineCustomElement as defineDgaSquareSkeleton } from "@platformscode/core/dist/components/dga-square-skeleton.js";
import { DgaStatusTag as DgaStatusTagElement, defineCustomElement as defineDgaStatusTag } from "@platformscode/core/dist/components/dga-status-tag.js";
import { DgaStructuredList as DgaStructuredListElement, defineCustomElement as defineDgaStructuredList } from "@platformscode/core/dist/components/dga-structured-list.js";
import { DgaSuffix as DgaSuffixElement, defineCustomElement as defineDgaSuffix } from "@platformscode/core/dist/components/dga-suffix.js";
import { DgaSwitch as DgaSwitchElement, defineCustomElement as defineDgaSwitch } from "@platformscode/core/dist/components/dga-switch.js";
import { DgaTableBody as DgaTableBodyElement, defineCustomElement as defineDgaTableBody } from "@platformscode/core/dist/components/dga-table-body.js";
import { DgaTableOfContent as DgaTableOfContentElement, defineCustomElement as defineDgaTableOfContent } from "@platformscode/core/dist/components/dga-table-of-content.js";
import { DgaTable as DgaTableElement, defineCustomElement as defineDgaTable } from "@platformscode/core/dist/components/dga-table.js";
import { DgaTabs as DgaTabsElement, defineCustomElement as defineDgaTabs } from "@platformscode/core/dist/components/dga-tabs.js";
import { DgaTag as DgaTagElement, defineCustomElement as defineDgaTag } from "@platformscode/core/dist/components/dga-tag.js";
import { DgaTextInput as DgaTextInputElement, defineCustomElement as defineDgaTextInput } from "@platformscode/core/dist/components/dga-text-input.js";
import { DgaTextarea as DgaTextareaElement, defineCustomElement as defineDgaTextarea } from "@platformscode/core/dist/components/dga-textarea.js";
import { DgaTooltip as DgaTooltipElement, defineCustomElement as defineDgaTooltip } from "@platformscode/core/dist/components/dga-tooltip.js";
import { FloatingButton as FloatingButtonElement, defineCustomElement as defineFloatingButton } from "@platformscode/core/dist/components/floating-button.js";
import { MonthNavigator as MonthNavigatorElement, defineCustomElement as defineMonthNavigator } from "@platformscode/core/dist/components/month-navigator.js";
import { NdsAccordionUsage as NdsAccordionUsageElement, defineCustomElement as defineNdsAccordionUsage } from "@platformscode/core/dist/components/nds-accordion-usage.js";
import { NdsAccordion as NdsAccordionElement, defineCustomElement as defineNdsAccordion } from "@platformscode/core/dist/components/nds-accordion.js";
import { NdsAvatarUsage as NdsAvatarUsageElement, defineCustomElement as defineNdsAvatarUsage } from "@platformscode/core/dist/components/nds-avatar-usage.js";
import { NdsAvatar as NdsAvatarElement, defineCustomElement as defineNdsAvatar } from "@platformscode/core/dist/components/nds-avatar.js";
import { NdsButtonUsage as NdsButtonUsageElement, defineCustomElement as defineNdsButtonUsage } from "@platformscode/core/dist/components/nds-button-usage.js";
import { NdsButton as NdsButtonElement, defineCustomElement as defineNdsButton } from "@platformscode/core/dist/components/nds-button.js";
import { NdsChipUsage as NdsChipUsageElement, defineCustomElement as defineNdsChipUsage } from "@platformscode/core/dist/components/nds-chip-usage.js";
import { NdsChip as NdsChipElement, defineCustomElement as defineNdsChip } from "@platformscode/core/dist/components/nds-chip.js";
import { NdsFloatingButtonUsage as NdsFloatingButtonUsageElement, defineCustomElement as defineNdsFloatingButtonUsage } from "@platformscode/core/dist/components/nds-floating-button-usage.js";
import { NdsFloatingButton as NdsFloatingButtonElement, defineCustomElement as defineNdsFloatingButton } from "@platformscode/core/dist/components/nds-floating-button.js";
import { NdsLinkUsage as NdsLinkUsageElement, defineCustomElement as defineNdsLinkUsage } from "@platformscode/core/dist/components/nds-link-usage.js";
import { NdsLink as NdsLinkElement, defineCustomElement as defineNdsLink } from "@platformscode/core/dist/components/nds-link.js";
import { NdsTableOfContentUsage as NdsTableOfContentUsageElement, defineCustomElement as defineNdsTableOfContentUsage } from "@platformscode/core/dist/components/nds-table-of-content-usage.js";
import { NdsTableOfContent as NdsTableOfContentElement, defineCustomElement as defineNdsTableOfContent } from "@platformscode/core/dist/components/nds-table-of-content.js";
import { SidePanelMenu as SidePanelMenuElement, defineCustomElement as defineSidePanelMenu } from "@platformscode/core/dist/components/side-panel-menu.js";
import { SubmenuComponent as SubmenuComponentElement, defineCustomElement as defineSubmenuComponent } from "@platformscode/core/dist/components/submenu-component.js";
import { TabComponent as TabComponentElement, defineCustomElement as defineTabComponent } from "@platformscode/core/dist/components/tab-component.js";
import { TabLink as TabLinkElement, defineCustomElement as defineTabLink } from "@platformscode/core/dist/components/tab-link.js";
import { createComponent } from '@stencil/react-output-target/runtime';
import React from 'react';
export const DateCell = createComponent({
    tagName: 'date-cell',
    elementClass: DateCellElement,
    react: React,
    events: {},
    defineCustomElement: defineDateCell
});
export const DgaAccordion = createComponent({
    tagName: 'dga-accordion',
    elementClass: DgaAccordionElement,
    react: React,
    events: {},
    defineCustomElement: defineDgaAccordion
});
export const DgaActionMenu = createComponent({
    tagName: 'dga-action-menu',
    elementClass: DgaActionMenuElement,
    react: React,
    events: {},
    defineCustomElement: defineDgaActionMenu
});
export const DgaActions = createComponent({
    tagName: 'dga-actions',
    elementClass: DgaActionsElement,
    react: React,
    events: {},
    defineCustomElement: defineDgaActions
});
export const DgaApexChart = createComponent({
    tagName: 'dga-apex-chart',
    elementClass: DgaApexChartElement,
    react: React,
    events: {},
    defineCustomElement: defineDgaApexChart
});
export const DgaAvatar = createComponent({
    tagName: 'dga-avatar',
    elementClass: DgaAvatarElement,
    react: React,
    events: {},
    defineCustomElement: defineDgaAvatar
});
export const DgaAvatarGroup = createComponent({
    tagName: 'dga-avatar-group',
    elementClass: DgaAvatarGroupElement,
    react: React,
    events: {},
    defineCustomElement: defineDgaAvatarGroup
});
export const DgaBreadcrumbs = createComponent({
    tagName: 'dga-breadcrumbs',
    elementClass: DgaBreadcrumbsElement,
    react: React,
    events: { onBreadcrumbClick: 'breadcrumbClick' },
    defineCustomElement: defineDgaBreadcrumbs
});
export const DgaButton = createComponent({
    tagName: 'dga-button',
    elementClass: DgaButtonElement,
    react: React,
    events: { onOnClick: 'onClick' },
    defineCustomElement: defineDgaButton
});
export const DgaButtonLeadingIcon = createComponent({
    tagName: 'dga-button-leading-icon',
    elementClass: DgaButtonLeadingIconElement,
    react: React,
    events: {},
    defineCustomElement: defineDgaButtonLeadingIcon
});
export const DgaButtonV2 = createComponent({
    tagName: 'dga-button-v2',
    elementClass: DgaButtonV2Element,
    react: React,
    events: { onOnClick: 'onClick' },
    defineCustomElement: defineDgaButtonV2
});
export const DgaButtonV2Usage = createComponent({
    tagName: 'dga-button-v2-usage',
    elementClass: DgaButtonV2UsageElement,
    react: React,
    events: {},
    defineCustomElement: defineDgaButtonV2Usage
});
export const DgaCard = createComponent({
    tagName: 'dga-card',
    elementClass: DgaCardElement,
    react: React,
    events: {},
    defineCustomElement: defineDgaCard
});
export const DgaCardActions = createComponent({
    tagName: 'dga-card-actions',
    elementClass: DgaCardActionsElement,
    react: React,
    events: {},
    defineCustomElement: defineDgaCardActions
});
export const DgaCardContent = createComponent({
    tagName: 'dga-card-content',
    elementClass: DgaCardContentElement,
    react: React,
    events: {},
    defineCustomElement: defineDgaCardContent
});
export const DgaCardExpandableContent = createComponent({
    tagName: 'dga-card-expandable-content',
    elementClass: DgaCardExpandableContentElement,
    react: React,
    events: {},
    defineCustomElement: defineDgaCardExpandableContent
});
export const DgaCardImage = createComponent({
    tagName: 'dga-card-image',
    elementClass: DgaCardImageElement,
    react: React,
    events: {},
    defineCustomElement: defineDgaCardImage
});
export const DgaCardTitle = createComponent({
    tagName: 'dga-card-title',
    elementClass: DgaCardTitleElement,
    react: React,
    events: {},
    defineCustomElement: defineDgaCardTitle
});
export const DgaCardV2 = createComponent({
    tagName: 'dga-card-v2',
    elementClass: DgaCardV2Element,
    react: React,
    events: {},
    defineCustomElement: defineDgaCardV2
});
export const DgaCarousel = createComponent({
    tagName: 'dga-carousel',
    elementClass: DgaCarouselElement,
    react: React,
    events: {},
    defineCustomElement: defineDgaCarousel
});
export const DgaCarouselItem = createComponent({
    tagName: 'dga-carousel-item',
    elementClass: DgaCarouselItemElement,
    react: React,
    events: {},
    defineCustomElement: defineDgaCarouselItem
});
export const DgaCarouselV2 = createComponent({
    tagName: 'dga-carousel-v2',
    elementClass: DgaCarouselV2Element,
    react: React,
    events: {},
    defineCustomElement: defineDgaCarouselV2
});
export const DgaChart = createComponent({
    tagName: 'dga-chart',
    elementClass: DgaChartElement,
    react: React,
    events: {},
    defineCustomElement: defineDgaChart
});
export const DgaCheckbox = createComponent({
    tagName: 'dga-checkbox',
    elementClass: DgaCheckboxElement,
    react: React,
    events: {},
    defineCustomElement: defineDgaCheckbox
});
export const DgaChip = createComponent({
    tagName: 'dga-chip',
    elementClass: DgaChipElement,
    react: React,
    events: {},
    defineCustomElement: defineDgaChip
});
export const DgaCircleSkeleton = createComponent({
    tagName: 'dga-circle-skeleton',
    elementClass: DgaCircleSkeletonElement,
    react: React,
    events: {},
    defineCustomElement: defineDgaCircleSkeleton
});
export const DgaCircularProgressBar = createComponent({
    tagName: 'dga-circular-progress-bar',
    elementClass: DgaCircularProgressBarElement,
    react: React,
    events: {},
    defineCustomElement: defineDgaCircularProgressBar
});
export const DgaCodesnippet = createComponent({
    tagName: 'dga-codesnippet',
    elementClass: DgaCodesnippetElement,
    react: React,
    events: {},
    defineCustomElement: defineDgaCodesnippet
});
export const DgaCollapse = createComponent({
    tagName: 'dga-collapse',
    elementClass: DgaCollapseElement,
    react: React,
    events: {},
    defineCustomElement: defineDgaCollapse
});
export const DgaCollapseUsage = createComponent({
    tagName: 'dga-collapse-usage',
    elementClass: DgaCollapseUsageElement,
    react: React,
    events: {},
    defineCustomElement: defineDgaCollapseUsage
});
export const DgaContentSwitcher = createComponent({
    tagName: 'dga-content-switcher',
    elementClass: DgaContentSwitcherElement,
    react: React,
    events: {},
    defineCustomElement: defineDgaContentSwitcher
});
export const DgaContentSwitcherItem = createComponent({
    tagName: 'dga-content-switcher-item',
    elementClass: DgaContentSwitcherItemElement,
    react: React,
    events: {},
    defineCustomElement: defineDgaContentSwitcherItem
});
export const DgaContentSwitcherPanel = createComponent({
    tagName: 'dga-content-switcher-panel',
    elementClass: DgaContentSwitcherPanelElement,
    react: React,
    events: {},
    defineCustomElement: defineDgaContentSwitcherPanel
});
export const DgaDataTable = createComponent({
    tagName: 'dga-data-table',
    elementClass: DgaDataTableElement,
    react: React,
    events: {},
    defineCustomElement: defineDgaDataTable
});
export const DgaDataTableUsage = createComponent({
    tagName: 'dga-data-table-usage',
    elementClass: DgaDataTableUsageElement,
    react: React,
    events: {},
    defineCustomElement: defineDgaDataTableUsage
});
export const DgaDateField = createComponent({
    tagName: 'dga-date-field',
    elementClass: DgaDateFieldElement,
    react: React,
    events: {},
    defineCustomElement: defineDgaDateField
});
export const DgaDateFieldUsage = createComponent({
    tagName: 'dga-date-field-usage',
    elementClass: DgaDateFieldUsageElement,
    react: React,
    events: {},
    defineCustomElement: defineDgaDateFieldUsage
});
export const DgaDatepicker = createComponent({
    tagName: 'dga-datepicker',
    elementClass: DgaDatepickerElement,
    react: React,
    events: { onChange: 'change' },
    defineCustomElement: defineDgaDatepicker
});
export const DgaDigitalSignature = createComponent({
    tagName: 'dga-digital-signature',
    elementClass: DgaDigitalSignatureElement,
    react: React,
    events: {},
    defineCustomElement: defineDgaDigitalSignature
});
export const DgaDivider = createComponent({
    tagName: 'dga-divider',
    elementClass: DgaDividerElement,
    react: React,
    events: {},
    defineCustomElement: defineDgaDivider
});
export const DgaDrawer = createComponent({
    tagName: 'dga-drawer',
    elementClass: DgaDrawerElement,
    react: React,
    events: {},
    defineCustomElement: defineDgaDrawer
});
export const DgaDropdown = createComponent({
    tagName: 'dga-dropdown',
    elementClass: DgaDropdownElement,
    react: React,
    events: {},
    defineCustomElement: defineDgaDropdown
});
export const DgaFeaturedIcon = createComponent({
    tagName: 'dga-featured-icon',
    elementClass: DgaFeaturedIconElement,
    react: React,
    events: {},
    defineCustomElement: defineDgaFeaturedIcon
});
export const DgaFeedbackIcon = createComponent({
    tagName: 'dga-feedback-icon',
    elementClass: DgaFeedbackIconElement,
    react: React,
    events: {},
    defineCustomElement: defineDgaFeedbackIcon
});
export const DgaFeedbackIconUsage = createComponent({
    tagName: 'dga-feedback-icon-usage',
    elementClass: DgaFeedbackIconUsageElement,
    react: React,
    events: {},
    defineCustomElement: defineDgaFeedbackIconUsage
});
export const DgaFileUpload = createComponent({
    tagName: 'dga-file-upload',
    elementClass: DgaFileUploadElement,
    react: React,
    events: {},
    defineCustomElement: defineDgaFileUpload
});
export const DgaFilteration = createComponent({
    tagName: 'dga-filteration',
    elementClass: DgaFilterationElement,
    react: React,
    events: { onOnFilterChange: 'onFilterChange' },
    defineCustomElement: defineDgaFilteration
});
export const DgaFlex = createComponent({
    tagName: 'dga-flex',
    elementClass: DgaFlexElement,
    react: React,
    events: {},
    defineCustomElement: defineDgaFlex
});
export const DgaFooter = createComponent({
    tagName: 'dga-footer',
    elementClass: DgaFooterElement,
    react: React,
    events: {},
    defineCustomElement: defineDgaFooter
});
export const DgaGridContainer = createComponent({
    tagName: 'dga-grid-container',
    elementClass: DgaGridContainerElement,
    react: React,
    events: {},
    defineCustomElement: defineDgaGridContainer
});
export const DgaGridItem = createComponent({
    tagName: 'dga-grid-item',
    elementClass: DgaGridItemElement,
    react: React,
    events: {},
    defineCustomElement: defineDgaGridItem
});
export const DgaGridUsage = createComponent({
    tagName: 'dga-grid-usage',
    elementClass: DgaGridUsageElement,
    react: React,
    events: {},
    defineCustomElement: defineDgaGridUsage
});
export const DgaHeader = createComponent({
    tagName: 'dga-header',
    elementClass: DgaHeaderElement,
    react: React,
    events: {},
    defineCustomElement: defineDgaHeader
});
export const DgaHeaderAction = createComponent({
    tagName: 'dga-header-action',
    elementClass: DgaHeaderActionElement,
    react: React,
    events: {},
    defineCustomElement: defineDgaHeaderAction
});
export const DgaHeaderActionBtn = createComponent({
    tagName: 'dga-header-action-btn',
    elementClass: DgaHeaderActionBtnElement,
    react: React,
    events: {},
    defineCustomElement: defineDgaHeaderActionBtn
});
export const DgaHeaderMenu = createComponent({
    tagName: 'dga-header-menu',
    elementClass: DgaHeaderMenuElement,
    react: React,
    events: {},
    defineCustomElement: defineDgaHeaderMenu
});
export const DgaHeaderMenuItem = createComponent({
    tagName: 'dga-header-menu-item',
    elementClass: DgaHeaderMenuItemElement,
    react: React,
    events: { onLinkClicked: 'linkClicked' },
    defineCustomElement: defineDgaHeaderMenuItem
});
export const DgaHeaderMenuItemUsage = createComponent({
    tagName: 'dga-header-menu-item-usage',
    elementClass: DgaHeaderMenuItemUsageElement,
    react: React,
    events: {},
    defineCustomElement: defineDgaHeaderMenuItemUsage
});
export const DgaHeaderMenuVv2 = createComponent({
    tagName: 'dga-header-menu-vv2',
    elementClass: DgaHeaderMenuVv2Element,
    react: React,
    events: {},
    defineCustomElement: defineDgaHeaderMenuVv2
});
export const DgaHeaderMobileActions = createComponent({
    tagName: 'dga-header-mobile-actions',
    elementClass: DgaHeaderMobileActionsElement,
    react: React,
    events: {},
    defineCustomElement: defineDgaHeaderMobileActions
});
export const DgaHeaderSubMenu = createComponent({
    tagName: 'dga-header-sub-menu',
    elementClass: DgaHeaderSubMenuElement,
    react: React,
    events: {},
    defineCustomElement: defineDgaHeaderSubMenu
});
export const DgaHeaderSubMenuColumn = createComponent({
    tagName: 'dga-header-sub-menu-column',
    elementClass: DgaHeaderSubMenuColumnElement,
    react: React,
    events: {},
    defineCustomElement: defineDgaHeaderSubMenuColumn
});
export const DgaHeaderSubMenuItem = createComponent({
    tagName: 'dga-header-sub-menu-item',
    elementClass: DgaHeaderSubMenuItemElement,
    react: React,
    events: {},
    defineCustomElement: defineDgaHeaderSubMenuItem
});
export const DgaHeaderVv2 = createComponent({
    tagName: 'dga-header-vv2',
    elementClass: DgaHeaderVv2Element,
    react: React,
    events: {},
    defineCustomElement: defineDgaHeaderVv2
});
export const DgaHelperText = createComponent({
    tagName: 'dga-helper-text',
    elementClass: DgaHelperTextElement,
    react: React,
    events: {},
    defineCustomElement: defineDgaHelperText
});
export const DgaHugeIconUsage = createComponent({
    tagName: 'dga-huge-icon-usage',
    elementClass: DgaHugeIconUsageElement,
    react: React,
    events: {},
    defineCustomElement: defineDgaHugeIconUsage
});
export const DgaIcon = createComponent({
    tagName: 'dga-icon',
    elementClass: DgaIconElement,
    react: React,
    events: {},
    defineCustomElement: defineDgaIcon
});
export const DgaInlineAlert = createComponent({
    tagName: 'dga-inline-alert',
    elementClass: DgaInlineAlertElement,
    react: React,
    events: {},
    defineCustomElement: defineDgaInlineAlert
});
export const DgaLabel = createComponent({
    tagName: 'dga-label',
    elementClass: DgaLabelElement,
    react: React,
    events: {},
    defineCustomElement: defineDgaLabel
});
export const DgaLineSkeleton = createComponent({
    tagName: 'dga-line-skeleton',
    elementClass: DgaLineSkeletonElement,
    react: React,
    events: {},
    defineCustomElement: defineDgaLineSkeleton
});
export const DgaLinearProgressBar = createComponent({
    tagName: 'dga-linear-progress-bar',
    elementClass: DgaLinearProgressBarElement,
    react: React,
    events: {},
    defineCustomElement: defineDgaLinearProgressBar
});
export const DgaLink = createComponent({
    tagName: 'dga-link',
    elementClass: DgaLinkElement,
    react: React,
    events: { onOnClick: 'onClick' },
    defineCustomElement: defineDgaLink
});
export const DgaList = createComponent({
    tagName: 'dga-list',
    elementClass: DgaListElement,
    react: React,
    events: {},
    defineCustomElement: defineDgaList
});
export const DgaListItem = createComponent({
    tagName: 'dga-list-item',
    elementClass: DgaListItemElement,
    react: React,
    events: {},
    defineCustomElement: defineDgaListItem
});
export const DgaListV2 = createComponent({
    tagName: 'dga-list-v2',
    elementClass: DgaListV2Element,
    react: React,
    events: {},
    defineCustomElement: defineDgaListV2
});
export const DgaLoading = createComponent({
    tagName: 'dga-loading',
    elementClass: DgaLoadingElement,
    react: React,
    events: {},
    defineCustomElement: defineDgaLoading
});
export const DgaLogoPlaceholder = createComponent({
    tagName: 'dga-logo-placeholder',
    elementClass: DgaLogoPlaceholderElement,
    react: React,
    events: {},
    defineCustomElement: defineDgaLogoPlaceholder
});
export const DgaMenu = createComponent({
    tagName: 'dga-menu',
    elementClass: DgaMenuElement,
    react: React,
    events: {},
    defineCustomElement: defineDgaMenu
});
export const DgaMenuGroup = createComponent({
    tagName: 'dga-menu-group',
    elementClass: DgaMenuGroupElement,
    react: React,
    events: {},
    defineCustomElement: defineDgaMenuGroup
});
export const DgaMenuItem = createComponent({
    tagName: 'dga-menu-item',
    elementClass: DgaMenuItemElement,
    react: React,
    events: { onOnClick: 'onClick' },
    defineCustomElement: defineDgaMenuItem
});
export const DgaMenuList = createComponent({
    tagName: 'dga-menu-list',
    elementClass: DgaMenuListElement,
    react: React,
    events: {},
    defineCustomElement: defineDgaMenuList
});
export const DgaMenuV2 = createComponent({
    tagName: 'dga-menu-v2',
    elementClass: DgaMenuV2Element,
    react: React,
    events: {},
    defineCustomElement: defineDgaMenuV2
});
export const DgaMetric = createComponent({
    tagName: 'dga-metric',
    elementClass: DgaMetricElement,
    react: React,
    events: {},
    defineCustomElement: defineDgaMetric
});
export const DgaMetricChart = createComponent({
    tagName: 'dga-metric-chart',
    elementClass: DgaMetricChartElement,
    react: React,
    events: {},
    defineCustomElement: defineDgaMetricChart
});
export const DgaMetricChartV2 = createComponent({
    tagName: 'dga-metric-chart-v2',
    elementClass: DgaMetricChartV2Element,
    react: React,
    events: {},
    defineCustomElement: defineDgaMetricChartV2
});
export const DgaMetricV2 = createComponent({
    tagName: 'dga-metric-v2',
    elementClass: DgaMetricV2Element,
    react: React,
    events: {},
    defineCustomElement: defineDgaMetricV2
});
export const DgaModal = createComponent({
    tagName: 'dga-modal',
    elementClass: DgaModalElement,
    react: React,
    events: {},
    defineCustomElement: defineDgaModal
});
export const DgaModalActions = createComponent({
    tagName: 'dga-modal-actions',
    elementClass: DgaModalActionsElement,
    react: React,
    events: {},
    defineCustomElement: defineDgaModalActions
});
export const DgaModalBody = createComponent({
    tagName: 'dga-modal-body',
    elementClass: DgaModalBodyElement,
    react: React,
    events: {},
    defineCustomElement: defineDgaModalBody
});
export const DgaModalHeader = createComponent({
    tagName: 'dga-modal-header',
    elementClass: DgaModalHeaderElement,
    react: React,
    events: {},
    defineCustomElement: defineDgaModalHeader
});
export const DgaModalTitle = createComponent({
    tagName: 'dga-modal-title',
    elementClass: DgaModalTitleElement,
    react: React,
    events: {},
    defineCustomElement: defineDgaModalTitle
});
export const DgaModalV2 = createComponent({
    tagName: 'dga-modal-v2',
    elementClass: DgaModalV2Element,
    react: React,
    events: {},
    defineCustomElement: defineDgaModalV2
});
export const DgaModalV2Usage = createComponent({
    tagName: 'dga-modal-v2-usage',
    elementClass: DgaModalV2UsageElement,
    react: React,
    events: {},
    defineCustomElement: defineDgaModalV2Usage
});
export const DgaNavHeader = createComponent({
    tagName: 'dga-nav-header',
    elementClass: DgaNavHeaderElement,
    react: React,
    events: {},
    defineCustomElement: defineDgaNavHeader
});
export const DgaNavHeaderActions = createComponent({
    tagName: 'dga-nav-header-actions',
    elementClass: DgaNavHeaderActionsElement,
    react: React,
    events: {},
    defineCustomElement: defineDgaNavHeaderActions
});
export const DgaNavHeaderLink = createComponent({
    tagName: 'dga-nav-header-link',
    elementClass: DgaNavHeaderLinkElement,
    react: React,
    events: { onLinkClicked: 'linkClicked' },
    defineCustomElement: defineDgaNavHeaderLink
});
export const DgaNavHeaderLogos = createComponent({
    tagName: 'dga-nav-header-logos',
    elementClass: DgaNavHeaderLogosElement,
    react: React,
    events: {},
    defineCustomElement: defineDgaNavHeaderLogos
});
export const DgaNavHeaderMain = createComponent({
    tagName: 'dga-nav-header-main',
    elementClass: DgaNavHeaderMainElement,
    react: React,
    events: { onToggleCollapsed: 'toggleCollapsed' },
    defineCustomElement: defineDgaNavHeaderMain
});
export const DgaNavHeaderMenu = createComponent({
    tagName: 'dga-nav-header-menu',
    elementClass: DgaNavHeaderMenuElement,
    react: React,
    events: {},
    defineCustomElement: defineDgaNavHeaderMenu
});
export const DgaNavHeaderSubMenu = createComponent({
    tagName: 'dga-nav-header-sub-menu',
    elementClass: DgaNavHeaderSubMenuElement,
    react: React,
    events: {},
    defineCustomElement: defineDgaNavHeaderSubMenu
});
export const DgaNavHeaderSubMenuColumn = createComponent({
    tagName: 'dga-nav-header-sub-menu-column',
    elementClass: DgaNavHeaderSubMenuColumnElement,
    react: React,
    events: {},
    defineCustomElement: defineDgaNavHeaderSubMenuColumn
});
export const DgaNavHeaderSubMenuLink = createComponent({
    tagName: 'dga-nav-header-sub-menu-link',
    elementClass: DgaNavHeaderSubMenuLinkElement,
    react: React,
    events: {},
    defineCustomElement: defineDgaNavHeaderSubMenuLink
});
export const DgaNotification = createComponent({
    tagName: 'dga-notification',
    elementClass: DgaNotificationElement,
    react: React,
    events: {},
    defineCustomElement: defineDgaNotification
});
export const DgaNotificationToast = createComponent({
    tagName: 'dga-notification-toast',
    elementClass: DgaNotificationToastElement,
    react: React,
    events: {},
    defineCustomElement: defineDgaNotificationToast
});
export const DgaNumberInput = createComponent({
    tagName: 'dga-number-input',
    elementClass: DgaNumberInputElement,
    react: React,
    events: {},
    defineCustomElement: defineDgaNumberInput
});
export const DgaPagination = createComponent({
    tagName: 'dga-pagination',
    elementClass: DgaPaginationElement,
    react: React,
    events: {},
    defineCustomElement: defineDgaPagination
});
export const DgaPrefix = createComponent({
    tagName: 'dga-prefix',
    elementClass: DgaPrefixElement,
    react: React,
    events: { onOnSelect: 'onSelect' },
    defineCustomElement: defineDgaPrefix
});
export const DgaProgressIndicator = createComponent({
    tagName: 'dga-progress-indicator',
    elementClass: DgaProgressIndicatorElement,
    react: React,
    events: { onStepChange: 'stepChange' },
    defineCustomElement: defineDgaProgressIndicator
});
export const DgaProgressIndicatorStep = createComponent({
    tagName: 'dga-progress-indicator-step',
    elementClass: DgaProgressIndicatorStepElement,
    react: React,
    events: {},
    defineCustomElement: defineDgaProgressIndicatorStep
});
export const DgaProgressIndicatorStepV3 = createComponent({
    tagName: 'dga-progress-indicator-step-v3',
    elementClass: DgaProgressIndicatorStepV3Element,
    react: React,
    events: {},
    defineCustomElement: defineDgaProgressIndicatorStepV3
});
export const DgaProgressIndicatorV3 = createComponent({
    tagName: 'dga-progress-indicator-v3',
    elementClass: DgaProgressIndicatorV3Element,
    react: React,
    events: { onStepChange: 'stepChange' },
    defineCustomElement: defineDgaProgressIndicatorV3
});
export const DgaProgressIndicatorV3Usage = createComponent({
    tagName: 'dga-progress-indicator-v3-usage',
    elementClass: DgaProgressIndicatorV3UsageElement,
    react: React,
    events: {},
    defineCustomElement: defineDgaProgressIndicatorV3Usage
});
export const DgaQuote = createComponent({
    tagName: 'dga-quote',
    elementClass: DgaQuoteElement,
    react: React,
    events: {},
    defineCustomElement: defineDgaQuote
});
export const DgaRadialStepper = createComponent({
    tagName: 'dga-radial-stepper',
    elementClass: DgaRadialStepperElement,
    react: React,
    events: {},
    defineCustomElement: defineDgaRadialStepper
});
export const DgaRadioButton = createComponent({
    tagName: 'dga-radio-button',
    elementClass: DgaRadioButtonElement,
    react: React,
    events: {},
    defineCustomElement: defineDgaRadioButton
});
export const DgaRating = createComponent({
    tagName: 'dga-rating',
    elementClass: DgaRatingElement,
    react: React,
    events: { onRatingChange: 'ratingChange' },
    defineCustomElement: defineDgaRating
});
export const DgaRectangleSkeleton = createComponent({
    tagName: 'dga-rectangle-skeleton',
    elementClass: DgaRectangleSkeletonElement,
    react: React,
    events: {},
    defineCustomElement: defineDgaRectangleSkeleton
});
export const DgaSearchBox = createComponent({
    tagName: 'dga-search-box',
    elementClass: DgaSearchBoxElement,
    react: React,
    events: {
        onOnChange: 'onChange',
        onOnBlur: 'onBlur'
    },
    defineCustomElement: defineDgaSearchBox
});
export const DgaSecondNavHeader = createComponent({
    tagName: 'dga-second-nav-header',
    elementClass: DgaSecondNavHeaderElement,
    react: React,
    events: {},
    defineCustomElement: defineDgaSecondNavHeader
});
export const DgaSecondNavHeaderActions = createComponent({
    tagName: 'dga-second-nav-header-actions',
    elementClass: DgaSecondNavHeaderActionsElement,
    react: React,
    events: {},
    defineCustomElement: defineDgaSecondNavHeaderActions
});
export const DgaSecondNavHeaderContent = createComponent({
    tagName: 'dga-second-nav-header-content',
    elementClass: DgaSecondNavHeaderContentElement,
    react: React,
    events: {},
    defineCustomElement: defineDgaSecondNavHeaderContent
});
export const DgaSecondNavHeaderItem = createComponent({
    tagName: 'dga-second-nav-header-item',
    elementClass: DgaSecondNavHeaderItemElement,
    react: React,
    events: {},
    defineCustomElement: defineDgaSecondNavHeaderItem
});
export const DgaSkeletonComponent = createComponent({
    tagName: 'dga-skeleton-component',
    elementClass: DgaSkeletonComponentElement,
    react: React,
    events: {},
    defineCustomElement: defineDgaSkeletonComponent
});
export const DgaSlideoutMenu = createComponent({
    tagName: 'dga-slideout-menu',
    elementClass: DgaSlideoutMenuElement,
    react: React,
    events: {},
    defineCustomElement: defineDgaSlideoutMenu
});
export const DgaSlideoutMenuActions = createComponent({
    tagName: 'dga-slideout-menu-actions',
    elementClass: DgaSlideoutMenuActionsElement,
    react: React,
    events: {},
    defineCustomElement: defineDgaSlideoutMenuActions
});
export const DgaSlideoutMenuHeader = createComponent({
    tagName: 'dga-slideout-menu-header',
    elementClass: DgaSlideoutMenuHeaderElement,
    react: React,
    events: {},
    defineCustomElement: defineDgaSlideoutMenuHeader
});
export const DgaSlideoutMenuItems = createComponent({
    tagName: 'dga-slideout-menu-items',
    elementClass: DgaSlideoutMenuItemsElement,
    react: React,
    events: {},
    defineCustomElement: defineDgaSlideoutMenuItems
});
export const DgaSlider = createComponent({
    tagName: 'dga-slider',
    elementClass: DgaSliderElement,
    react: React,
    events: {},
    defineCustomElement: defineDgaSlider
});
export const DgaSquareSkeleton = createComponent({
    tagName: 'dga-square-skeleton',
    elementClass: DgaSquareSkeletonElement,
    react: React,
    events: {},
    defineCustomElement: defineDgaSquareSkeleton
});
export const DgaStatusTag = createComponent({
    tagName: 'dga-status-tag',
    elementClass: DgaStatusTagElement,
    react: React,
    events: {},
    defineCustomElement: defineDgaStatusTag
});
export const DgaStructuredList = createComponent({
    tagName: 'dga-structured-list',
    elementClass: DgaStructuredListElement,
    react: React,
    events: {},
    defineCustomElement: defineDgaStructuredList
});
export const DgaSuffix = createComponent({
    tagName: 'dga-suffix',
    elementClass: DgaSuffixElement,
    react: React,
    events: { onOnSelect: 'onSelect' },
    defineCustomElement: defineDgaSuffix
});
export const DgaSwitch = createComponent({
    tagName: 'dga-switch',
    elementClass: DgaSwitchElement,
    react: React,
    events: {},
    defineCustomElement: defineDgaSwitch
});
export const DgaTable = createComponent({
    tagName: 'dga-table',
    elementClass: DgaTableElement,
    react: React,
    events: {},
    defineCustomElement: defineDgaTable
});
export const DgaTableBody = createComponent({
    tagName: 'dga-table-body',
    elementClass: DgaTableBodyElement,
    react: React,
    events: {},
    defineCustomElement: defineDgaTableBody
});
export const DgaTableOfContent = createComponent({
    tagName: 'dga-table-of-content',
    elementClass: DgaTableOfContentElement,
    react: React,
    events: {},
    defineCustomElement: defineDgaTableOfContent
});
export const DgaTabs = createComponent({
    tagName: 'dga-tabs',
    elementClass: DgaTabsElement,
    react: React,
    events: {},
    defineCustomElement: defineDgaTabs
});
export const DgaTag = createComponent({
    tagName: 'dga-tag',
    elementClass: DgaTagElement,
    react: React,
    events: {},
    defineCustomElement: defineDgaTag
});
export const DgaTextInput = createComponent({
    tagName: 'dga-text-input',
    elementClass: DgaTextInputElement,
    react: React,
    events: {},
    defineCustomElement: defineDgaTextInput
});
export const DgaTextarea = createComponent({
    tagName: 'dga-textarea',
    elementClass: DgaTextareaElement,
    react: React,
    events: {},
    defineCustomElement: defineDgaTextarea
});
export const DgaTooltip = createComponent({
    tagName: 'dga-tooltip',
    elementClass: DgaTooltipElement,
    react: React,
    events: {},
    defineCustomElement: defineDgaTooltip
});
export const FloatingButton = createComponent({
    tagName: 'floating-button',
    elementClass: FloatingButtonElement,
    react: React,
    events: { onButtonClick: 'buttonClick' },
    defineCustomElement: defineFloatingButton
});
export const MonthNavigator = createComponent({
    tagName: 'month-navigator',
    elementClass: MonthNavigatorElement,
    react: React,
    events: {},
    defineCustomElement: defineMonthNavigator
});
export const NdsAccordion = createComponent({
    tagName: 'nds-accordion',
    elementClass: NdsAccordionElement,
    react: React,
    events: {},
    defineCustomElement: defineNdsAccordion
});
export const NdsAccordionUsage = createComponent({
    tagName: 'nds-accordion-usage',
    elementClass: NdsAccordionUsageElement,
    react: React,
    events: {},
    defineCustomElement: defineNdsAccordionUsage
});
export const NdsAvatar = createComponent({
    tagName: 'nds-avatar',
    elementClass: NdsAvatarElement,
    react: React,
    events: {},
    defineCustomElement: defineNdsAvatar
});
export const NdsAvatarUsage = createComponent({
    tagName: 'nds-avatar-usage',
    elementClass: NdsAvatarUsageElement,
    react: React,
    events: {},
    defineCustomElement: defineNdsAvatarUsage
});
export const NdsButton = createComponent({
    tagName: 'nds-button',
    elementClass: NdsButtonElement,
    react: React,
    events: { onOnClick: 'onClick' },
    defineCustomElement: defineNdsButton
});
export const NdsButtonUsage = createComponent({
    tagName: 'nds-button-usage',
    elementClass: NdsButtonUsageElement,
    react: React,
    events: {},
    defineCustomElement: defineNdsButtonUsage
});
export const NdsChip = createComponent({
    tagName: 'nds-chip',
    elementClass: NdsChipElement,
    react: React,
    events: {},
    defineCustomElement: defineNdsChip
});
export const NdsChipUsage = createComponent({
    tagName: 'nds-chip-usage',
    elementClass: NdsChipUsageElement,
    react: React,
    events: {},
    defineCustomElement: defineNdsChipUsage
});
export const NdsFloatingButton = createComponent({
    tagName: 'nds-floating-button',
    elementClass: NdsFloatingButtonElement,
    react: React,
    events: { onButtonClick: 'buttonClick' },
    defineCustomElement: defineNdsFloatingButton
});
export const NdsFloatingButtonUsage = createComponent({
    tagName: 'nds-floating-button-usage',
    elementClass: NdsFloatingButtonUsageElement,
    react: React,
    events: {},
    defineCustomElement: defineNdsFloatingButtonUsage
});
export const NdsLink = createComponent({
    tagName: 'nds-link',
    elementClass: NdsLinkElement,
    react: React,
    events: { onOnClick: 'onClick' },
    defineCustomElement: defineNdsLink
});
export const NdsLinkUsage = createComponent({
    tagName: 'nds-link-usage',
    elementClass: NdsLinkUsageElement,
    react: React,
    events: {},
    defineCustomElement: defineNdsLinkUsage
});
export const NdsTableOfContent = createComponent({
    tagName: 'nds-table-of-content',
    elementClass: NdsTableOfContentElement,
    react: React,
    events: {},
    defineCustomElement: defineNdsTableOfContent
});
export const NdsTableOfContentUsage = createComponent({
    tagName: 'nds-table-of-content-usage',
    elementClass: NdsTableOfContentUsageElement,
    react: React,
    events: {},
    defineCustomElement: defineNdsTableOfContentUsage
});
export const SidePanelMenu = createComponent({
    tagName: 'side-panel-menu',
    elementClass: SidePanelMenuElement,
    react: React,
    events: {},
    defineCustomElement: defineSidePanelMenu
});
export const SubmenuComponent = createComponent({
    tagName: 'submenu-component',
    elementClass: SubmenuComponentElement,
    react: React,
    events: {},
    defineCustomElement: defineSubmenuComponent
});
export const TabComponent = createComponent({
    tagName: 'tab-component',
    elementClass: TabComponentElement,
    react: React,
    events: {},
    defineCustomElement: defineTabComponent
});
export const TabLink = createComponent({
    tagName: 'tab-link',
    elementClass: TabLinkElement,
    react: React,
    events: {},
    defineCustomElement: defineTabLink
});
