// @platformscode/core/dist/collection/usage/dga-pagination-usage.js
import { h } from "@stencil/core";
export class DgaPaginationUsage {
    constructor() {
        this.variant = 'neutral';
    }
    render() {
        return (h("div", { key: '9f625c29d368b4e0411eaea590656a31fa6a63fc' }, h("h1", { key: '9366abd95d516a8ae9e582bd1bf1444a894a1783' }, "Pagination Example"), h("div", { key: '0ed92ae7626fcf7480a3b29d40fa3d04ac653c67', style: { padding: '50px', display: 'flex', gap: '20px' } }, h("div", { key: '33bb5d88aef980ac3d22a3cf5b36d7a9324e174b', style: { display: 'flex', flexDirection: 'column', gap: '20px' } }, h("dga-pagination", { key: '4ba100561b9fb4eab364c869b404800228361213', totalPageCount: 10, size: 'large', onChange: (page) => console.log(page, "page changed") }), h("dga-pagination", { key: '820eb9c46cafd1cc23508cf87dfdfc58abbcd3f9', totalPageCount: 10, size: 'medium' }), h("dga-pagination", { key: 'bf95e07d94ba006af7b8f50367cf79f0affc544b', totalPageCount: 10, size: 'small' })), h("div", { key: 'b83a6896afac005d53719bcdbe1ec1e8d9b98e90', style: { display: 'flex', flexDirection: 'column', gap: '20px' } }))));
    }
    static get is() { return "dga-pagination-usage"; }
    static get states() {
        return {
            "variant": {}
        };
    }
}
