// @platformscode/core/dist/collection/usage/dga-number-input-usage.js
import { h } from "@stencil/core";
export class DgaNumberInputUsage {
    constructor() {
        this.value = '';
    }
    render() {
        return (h("div", { key: '3f7083c0a957a30aa2190fbe04ded363d8d66be1' }, h("h1", { key: '4e143a435c8c0bbefec7f9e13d000d5aadbd8d8a' }, "Number Input Example"), h("div", { key: '7ecd8da3895351cdbf03b9696796ef8c4aa3312a', style: { padding: '50px' } }, h("div", { key: '9769176f7a55b154264d891943c08925022fdfb1', style: { display: 'flex', flexDirection: 'column', gap: '16px' } }, h("dga-number-input", { key: 'e5deea9b7982a644cd712085f346e4eddcc2f0a8', size: "lg", variant: "default", label: "Label", value: 0, required: true, helperText: "test" }), h("dga-number-input", { key: '1e72d6160cd2ef916ee6f5f335ef3856e4694b48', size: "lg", variant: "darker", label: "Label", value: 0, required: true,
            // feedbackIcon
            helperText: "test" }), h("dga-number-input", { key: '197b2b1023695075eae9d33705e71fbe1f8ffa3c', size: "lg", variant: "lighter", label: "Label", value: 0, required: true,
            // feedbackIcon
            helperText: "test" }), h("dga-number-input", { key: 'cdeb37daa1ac55924bcfaf090f49c314f15c00eb', size: "lg", variant: "darker", label: "Label", value: 0, required: true,
            // feedbackIcon
            helperText: "test" }), h("dga-label", { key: '0c1a6e92835a6ebbdc6121067682aedf9f44386c', label: "Mandatory Text Input" }), h("dga-label", { key: '492b44db0efb5337f0d1aff9bf182763eb278390', label: "Mandatory Text Input", required: true })))));
    }
    static get is() { return "dga-number-input-usage"; }
    static get states() {
        return {
            "value": {}
        };
    }
}
