// @platformscode/core/dist/collection/usage/dga-structured-list-usage.js
import { h } from "@stencil/core";
export class DgaStructuredListUsage {
    render() {
        return (h("div", { key: '835af6a2a1227b77efda3f2fd0a2a6b6c8ff0f7d' }, h("h1", { key: '3c3964ae8b15757863ef74072f2ace681c0290be' }, "Structured List Example"), h("div", { key: 'b33121fb66a4504642d873ee63963dabe44beae4', style: { padding: "50px" } }, h("dga-structured-list", { key: '3f6a8c489f0b8ba6ee2aed588372e8ca2ab76c79', cells: [
                {
                    title: "ID",
                    propertyName: "id",
                    type: "propertyName",
                },
                {
                    title: "Name",
                    propertyName: "name",
                    type: "propertyName",
                },
            ], data: [
                { id: 1, name: "Alice", selected: false },
                { id: 2, name: "Bob", selected: false },
                { id: 3, name: "Charlie", selected: false },
            ], rowDivider: true, colDivider: false, selectable: true, getSelectedRows: selectedRows => console.log(selectedRows) }), h("dga-structured-list", { key: 'a689aee40cd55c027f051f78675c3e3e27a60275', compact: true, cells: [
                {
                    title: "ID",
                    propertyName: "id",
                    type: "propertyName",
                },
                {
                    title: "Name",
                    propertyName: "name",
                    type: "propertyName",
                },
            ], data: [
                { id: 1, name: "Alice", selected: false },
                { id: 2, name: "Bob", selected: false },
                { id: 3, name: "Charlie", selected: false },
            ], rowDivider: true, selectable: true, getSelectedRows: selectedRows => console.log(selectedRows) }))));
    }
    static get is() { return "dga-structured-list-usage"; }
}
