// @platformscode/core/dist/collection/usage/dga-grid-usage.js
// import { Component, h } from '@stencil/core';
// @Component({
//   tag: 'dga-grid-usage',
//   styleUrl: 'dga-grid-usage.scss',
// })
// export class DgaGridUsage {
//   render() {
//     return (
//       <div class="filteration-usage" style={{
//         display: 'flex',
//         flexDirection: 'column',
//         gap: '20px',
//       }}>
//         <h2>Filteration Component Examples</h2>
//         {/* <!-- Basic Grid Layout Example --> */}
//         <dga-grid-container spacing={2}>
//           <dga-grid-item xs={12} sm={6} md={4}>
//             <div class="demo-content">Item 1</div>
//           </dga-grid-item>
//           <dga-grid-item xs={12} sm={6} md={4}>
//             <div class="demo-content">Item 2</div>
//           </dga-grid-item>
//           <dga-grid-item xs={12} sm={6} md={4}>
//             <div class="demo-content">Item 3</div>
//           </dga-grid-item>
//         </dga-grid-container>
//         {/* <!-- Responsive Grid with Different Column Widths --> */}
//         <dga-grid-container spacing={3}>
//           <dga-grid-item xs={12} sm={6} md={4} lg={3}>
//             <div class="demo-content">Full width on mobile, half width on tablet, 1/3 on laptop, 1/4 on desktop</div>
//           </dga-grid-item>
//           <dga-grid-item xs={12} sm={6} md={4} lg={3}>
//             <div class="demo-content">Responsive Item 2</div>
//           </dga-grid-item>
//           <dga-grid-item xs={12} sm={6} md={4} lg={3}>
//             <div class="demo-content">Responsive Item 3</div>
//           </dga-grid-item>
//           <dga-grid-item xs={12} sm={6} md={12} lg={3}>
//             <div class="demo-content">Responsive Item 4</div>
//           </dga-grid-item>
//         </dga-grid-container>
//         {/* <!-- Different Spacing at Different Breakpoints --> */}
//         <dga-grid-container spacingXs={1} spacingSm={2} spacingMd={3} spacingLg={4}>
//           <dga-grid-item xs={6}>
//             <div class="demo-content">Different spacing at different screen sizes</div>
//           </dga-grid-item>
//           <dga-grid-item xs={6}>
//             <div class="demo-content">Different spacing at different screen sizes</div>
//           </dga-grid-item>
//         </dga-grid-container>
//         {/* <!-- Auto-layout Grid --> */}
//         <dga-grid-container spacing={2}>
//           <dga-grid-item xs>
//             <div class="demo-content">Auto width 1</div>
//           </dga-grid-item>
//           <dga-grid-item xs>
//             <div class="demo-content">Auto width 2</div>
//           </dga-grid-item>
//           <dga-grid-item xs>
//             <div class="demo-content">Auto width 3</div>
//           </dga-grid-item>
//         </dga-grid-container>
//         {/* <!-- Nested Grids --> */}
//         <dga-grid-container spacing={2}>
//           <dga-grid-item xs={12} sm={6}>
//             <div class="demo-content">
//               <h3>Nested Grid</h3>
//               <dga-grid-container spacing={1}>
//                 <dga-grid-item xs={6}>
//                   <div class="demo-content nested">Nested item 1</div>
//                 </dga-grid-item>
//                 <dga-grid-item xs={6}>
//                   <div class="demo-content nested">Nested item 2</div>
//                 </dga-grid-item>
//               </dga-grid-container>
//             </div>
//           </dga-grid-item>
//           <dga-grid-item xs={12} sm={6}>
//             <div class="demo-content">Regular item</div>
//           </dga-grid-item>
//         </dga-grid-container>
//         {/* <!-- Item Alignment Example --> */}
//         <dga-grid-container spacing={2} style={{height: "200px"}}>
//           <dga-grid-item xs={4} alignSelf="flex-start">
//             <div class="demo-content">Align start</div>
//           </dga-grid-item>
//           <dga-grid-item xs={4} alignSelf="center">
//             <div class="demo-content">Align center</div>
//           </dga-grid-item>
//           <dga-grid-item xs={4} alignSelf="flex-end">
//             <div class="demo-content">Align end</div>
//           </dga-grid-item>
//         </dga-grid-container>
//         {/* <!-- Growing and Shrinking Items --> */}
//         <dga-grid-container spacing={2}>
//           <dga-grid-item xs={3}>
//             <div class="demo-content">Fixed width</div>
//           </dga-grid-item>
//           <dga-grid-item xs grow>
//             <div class="demo-content">Grow to fill space</div>
//           </dga-grid-item>
//           <dga-grid-item xs={3}>
//             <div class="demo-content">Fixed width</div>
//           </dga-grid-item>
//         </dga-grid-container>
//         {/* <!-- Item Ordering Example --> */}
//         <dga-grid-container spacing={2}>
//           <dga-grid-item xs={4} orderXs={3} orderMd={1}>
//             <div class="demo-content">First on desktop, last on mobile</div>
//           </dga-grid-item>
//           <dga-grid-item xs={4} orderXs={1} orderMd={2}>
//             <div class="demo-content">Second on desktop, first on mobile</div>
//           </dga-grid-item>
//           <dga-grid-item xs={4} orderXs={2} orderMd={3}>
//             <div class="demo-content">Third on desktop, second on mobile</div>
//           </dga-grid-item>
//         </dga-grid-container>
//         {/* <!-- Fluid Container with No Gutters --> */}
//         <dga-grid-container fluid disableGutters spacing={0}>
//           <dga-grid-item xs={12} md={6}>
//             <div class="demo-content">Full-width fluid container with no gutters</div>
//           </dga-grid-item>
//           <dga-grid-item xs={12} md={6}>
//             <div class="demo-content">Full-width fluid container with no gutters</div>
//           </dga-grid-item>
//         </dga-grid-container>
//       </div>
//     );
//   }
// }
import { h } from "@stencil/core";
export class DgaGridUsage {
    render() {
        return (h("div", { key: '6f451f67fb4853d2c8785c09f34668e7a5280f98', class: "flexible-gap-usage", style: {
                display: 'flex',
                flexDirection: 'column',
                gap: '20px',
            } }, h("h2", { key: 'ca14e1fac659cd63223d2533af5e45c67a32a71a' }, "Grid Component with Flexible Gap Examples"), h("section", { key: '06ab77105c63da0422c0fca8313149d1d68e6773' }, h("h3", { key: '9622f7a9d5a37b1a7d2106d68368847b3b6f8ac1' }, "Simple String Gap"), h("dga-grid-container", { key: 'd9a1717a94a4523cd88a9d8b8289cdaf66a8ca23', gap: "16px" }, h("dga-grid-item", { key: 'd3d2081186a3d3809fcf210c5c0d5606f7f5d93d', xs: 12, sm: 6, md: 4 }, h("div", { key: '8dabea1575d4fb6ef4be5e33ce030132316739a6', class: "demo-content" }, "16px gap between all items")), h("dga-grid-item", { key: 'e3497bf25aa52b003ca6ae0d7309d71420e94b8d', xs: 12, sm: 6, md: 4 }, h("div", { key: 'b843487ccbe7e2cdc82ee61ce776a2715b60dd03', class: "demo-content" }, "16px gap between all items")), h("dga-grid-item", { key: '91bbbe993b7f8b482b8ed13dc779079fb9ffb49f', xs: 12, sm: 6, md: 4 }, h("div", { key: '937dd468e17f2e36da18768613199d4ea05bb083', class: "demo-content" }, "16px gap between all items")), h("dga-grid-item", { key: 'd51f3c30b52505b27be7dd59dd8210485a41e63d', xs: 12, sm: 6 }, h("div", { key: 'aea366da1bbb4eadd84e6dfca0f04c7b7b54bba2', class: "demo-content" }, "16px gap between all items")), h("dga-grid-item", { key: '331681e46175617c89adbcabf252bb29aefd8f35', xs: 12, sm: 6 }, h("div", { key: '7d5518f17374f7080070665b56593f9446376fd3', class: "demo-content" }, "16px gap between all items")), h("dga-grid-item", { key: '26672161c75e8bf12c70b7b5c65ae335d1659c74', xs: 12 }, h("div", { key: '7b22c99ff38e46449f28d3e60fee89263b6ea5e7', class: "demo-content" }, "16px gap between all items"))))));
    }
    static get is() { return "dga-grid-usage"; }
    static get originalStyleUrls() {
        return {
            "$": ["dga-grid-usage.scss"]
        };
    }
    static get styleUrls() {
        return {
            "$": ["dga-grid-usage.css"]
        };
    }
}
