// @platformscode/core/dist/collection/usage/dga-modal-usage.js
import { h } from "@stencil/core";
export class DgaModalUsage {
    constructor() {
        this.show = false;
    }
    render() {
        return (h("div", { key: '8cde72199e094c6b7dff6f5cd34ba6aa6a6f081f', style: { position: "relative" } }, h("h1", { key: 'ab21096bfdaac655e70beb5f1f9657b943f10940' }, "Modal Example"), h("button", { key: '7c9e08e528699b0770816eb7a585aa79269e012a', onClick: () => this.show = true }, "Open Modal"), h("dga-modal", { key: '2e4e878d1e859e8017c369d66aec7b92e77f1ac9', buttonsList: [
                {
                    extraClass: '',
                    label: 'Close',
                    onClick: () => { },
                    title: 'Close Modal'
                }
            ], closeButton: true, modalTitle: "Basic Modal", name: "basicModal", onClose: () => { }, open: this.show }, "This is a simple modal with default configuration."), this.show ? "shown" : "hidden"));
    }
    static get is() { return "dga-modal-usage"; }
    static get states() {
        return {
            "show": {}
        };
    }
}
