// @platformscode/core/dist/collection/usage/dga-collapse-usage.js
import { h } from "@stencil/core";
export class DgaCollapseUsage {
    constructor() {
        this.variant = 'neutral';
        this.open = false;
    }
    render() {
        return (h("div", { key: 'ce45988cc189e31973289c2785bd351adcbcd027' }, h("h1", { key: 'd73ae20e4ac38dbb1ddd6276ba1a3aa1c5328174' }, "Collapse Example"), h("div", { key: '9a1b43f1f384c22d3428aadf3d19af7dbe343ed5', style: { padding: '50px', display: 'flex', gap: '20px' } }, h("div", { key: 'aa3d3742b330872605c3c7ec42a5073eb97e2b0a', style: { display: 'flex', flexDirection: 'column', gap: '20px' } }, h("div", { key: 'a3381846bea7990d01306834ffa3e07f896f0853', style: { display: "flex", justifyContent: "center" } }, h("button", { key: '08e7ae751bbf3866a45a37f241c9d8e748b36bb9', onClick: () => this.open = true }, "open"), h("button", { key: '367a2f8354b616d07ce6f8f0c64f57d8adfc0361', onClick: () => this.open = false }, "close"))), h("dga-collapse", { key: '7b5f2514c6f258b1a5b21f286d399114abe59b07', open: this.open, "transition-duration": "0.1s" }, h("div", { key: 'f358334a58500c364213da1aff61cdd8925fbef9' }, "Your content here")), h("div", { key: '666f605c04a91b66ac0013b3b3696447c16e4116', style: { display: 'flex', flexDirection: 'column', gap: '20px' } }))));
    }
    static get is() { return "dga-collapse-usage"; }
    static get states() {
        return {
            "variant": {},
            "open": {}
        };
    }
}
