// @platformscode/core/dist/collection/usage/dga-divider-usage.js
import { h } from "@stencil/core";
export class DgaDividerUsage {
    render() {
        return (h("div", { key: 'bc740b93b419ee3b10a3bb617b05bba4cd399d2e' }, h("h1", { key: 'aabadc49976d7eddd277aaa8ed59c4070ce37ef8' }, "Divider Example"), h("div", { key: 'e3d21269f94c77f1c628a8514c717badb474bafe', style: { padding: '50px', display: 'flex', gap: '20px' } }, h("div", { key: '8bce98d3be0f9b24fa777cdc29754115af9ee6e4', style: { display: 'flex', flexDirection: 'column', gap: '20px' } }, h("div", { key: 'acfad7ddae093ebfcd94caf5fd15654b003901f8', style: {
                borderRadius: '20px',
                background: 'gray',
            } }, h("div", { key: '5e85859f43a96529629bd1e8be57733859f953bf', style: {
                width: '224px',
                height: '82px',
                background: '#fff',
                padding: '16px',
                display: 'flex',
                flexDirection: 'column',
                gap: '40px',
            } }, h("dga-divider", { key: 'a5c3a613dc4e3c7a293ae42cabb67698318c2d27', color: "neutral" }), h("dga-divider", { key: 'c06959904187c8df1c9f5b01772792d432e1de30', color: "primary" })), h("div", { key: '4dd3cfc1a7f41458cadad9ba4a4a8a518ba4053b', style: {
                width: '224px',
                height: '82px',
                background: '#fff',
                padding: '16px',
                display: 'flex',
                // flexDirection: 'column',
                gap: '40px',
            } }, h("dga-divider", { key: '05e4a992683c9adf6beb1dc85c33eec9d79ce81f', color: "neutral", orientation: 'vertical' }), h("dga-divider", { key: '2cb98622ae4a1d8a81087e5ffd979e3327388442', color: "primary", orientation: 'vertical' })), h("div", { key: '3d12e3c11e790acb1938666330cf5fdec78be8e2', style: {
                width: '224px',
                height: '82px',
                background: '#1B8354',
                padding: '16px',
                display: 'flex',
                flexDirection: 'column',
                gap: '40px',
            } }, h("dga-divider", { key: '7f3bb282c92259b29bbce48a1a77156d6bbee752', color: "alpha-white" }), h("dga-divider", { key: '612c0f8bf659725d8a6a4893703bbccd3618b530', color: "white" })), h("div", { key: '16baa94bcec06536c8b3699e06c70f226d72dccb', style: {
                width: '224px',
                height: '82px',
                background: '#1B8354',
                padding: '16px',
                display: 'flex',
                // flexDirection: 'column',
                gap: '40px',
            } }, h("dga-divider", { key: 'a2ddca2b5403fbf24f9f08e75f7aa38ed2528d1c', color: "alpha-white", orientation: 'vertical' }), h("dga-divider", { key: '1f928598988acdc734fed4da4cb808a857d1258f', color: "white", orientation: 'vertical' })))), h("div", { key: '895f06dc283a53010fc59b00158585062aca0b82', style: { display: 'flex', flexDirection: 'column', gap: '20px' } }))));
    }
    static get is() { return "dga-divider-usage"; }
}
