// @platformscode/core/dist/collection/usage/dga-rating-usage.js
import { h } from "@stencil/core";
export class DgaRatingUsage {
    constructor() {
        this.variant = 'neutral';
        this.rating = 2.5;
    }
    render() {
        return (h("div", { key: '6e1fe64caf919b6520502799ae5996930f0b4566', dir: "rtl" }, h("h1", { key: '0fa34bce08e4610fe7248829eb098c90e010f7a8' }, "Rating Example"), h("div", { key: '16ca0bf248dbafa474f6abe91eec6dbfa7323ce9', style: { padding: '50px', display: 'flex', gap: '20px' } }, h("div", { key: '23b6938615085c1d6c326a99701f3369a1fcda94', style: { display: 'flex', flexDirection: 'column', gap: '20px' } }, h("button", { key: 'f8edf82671bfb75b2375a91f8e7590cb837486ec', onClick: () => (this.rating += 1) }, "Increase ", this.rating), h("dga-rating", { key: '0d4529e87932497f453f940011b1bf4d7612261a', onChange: val => {
                this.rating = val;
            }, value: this.rating, size: "lg" }), h("dga-rating", { key: '2697792a3066e03d43cc2870034f7efc39a75fe1', onChange: val => (this.rating = val), value: 2.5, size: "md" }), h("dga-rating", { key: '89773275538a75b4406ae2de9c628f7c3750cf20', onChange: val => (this.rating = val), value: this.rating, size: "sm" })), h("div", { key: '3808488af034c0f7fa56223b40cd8205cf367d82', style: { display: 'flex', flexDirection: 'column', gap: '20px' } }))));
    }
    static get is() { return "dga-rating-usage"; }
    static get states() {
        return {
            "variant": {},
            "rating": {}
        };
    }
}
