// @platformscode/core/dist/collection/usage/dga-chip-usage.js
import { h } from "@stencil/core";
export class DgaChipUsage {
    constructor() {
        this.isSelected = false;
    }
    render() {
        return (h("div", { key: '0178f4d6fd4b0719a2d3434d39f86bb66ad142f1' }, h("h1", { key: '907c68073c0b297355aecaacde5e247ec09f480d' }, "Chip Example"), h("div", { key: '372fb45c5eeda1ee383972ad6d2dab7a031daf42', style: { padding: '50px', display: 'flex', gap: '20px' } }, h("div", { key: 'cb2aa1e9c18f64a1b681d5b3ae50262208a120db', style: { display: 'flex', flexDirection: 'column', gap: '20px' } }, h("dga-chip", { key: '2e5a611ef8ba489f242e09562a73054dde3ed496', label: "Label", variant: "neutral",
            //   outlined={false}
            rounded: false, size: "lg" }), h("dga-chip", { key: '0b476dc34c94286ce0bf263c86a0bd26840d0d64', label: "Label", variant: "success", rounded: false, size: "lg", onChange: (isSelected) => {
                console.log(isSelected, "hi");
                this.isSelected = isSelected;
            }, isSelected: this.isSelected }), h("dga-chip", { key: '023fe5cefaed3d957d37e4d5311ec5d23c89b858', label: "Label", variant: "success", rounded: false, size: "lg", onChange: (isSelected) => {
                console.log(isSelected, "hi");
            }, isSelected: false }), h("dga-chip", { key: 'f514a151ac14828bcbeafebbdd631c91db0c2192', label: "Label", variant: "success", rounded: false, size: "lg", disabled: true }), h("div", { key: '720b5c589d97d5c0ab0471e4997a6f0354204f9a', style: { display: 'flex', flexDirection: 'column', gap: '20px', backgroundColor: '#074D31' } }, h("dga-chip", { key: '5b6fbb30727bca96ea55fb9179ad031373699ce7', label: "Label", variant: "on-color", rounded: true, size: "lg", disabled: true }))))));
    }
    static get is() { return "dga-chip-usage"; }
    static get states() {
        return {
            "isSelected": {}
        };
    }
}
