// @platformscode/core/dist/collection/usage/dga-footer-usage.js
import { h } from "@stencil/core";
export class DgaFooterUsage {
    render() {
        return (h("div", { key: '4312d5f97243062de9ac182e538ea708709e321f' }, h("h1", { key: 'ba5babe6cbe22b435140fab5cfe838c632e5026b' }, "Footer Example Dark Green Mode"), h("dga-footer", { key: 'b92abb588121ba6b4e0119863098ecc1d64d6945', background: "DarkGreen", NavLinks: true, groupLinks: [
                {
                    title: 'Government Services',
                    links: [
                        { name: 'Digital Identity', target: 'http://localhost:3333/digital-identity' },
                        { name: 'E-Services Portal', target: 'http://localhost:3333/e-services' },
                        { name: 'Government Directory', target: 'http://localhost:3333/directory' },
                        { name: 'Digital Transformation', target: 'http://localhost:3333/transformation' },
                        { name: 'Cybersecurity', target: 'http://localhost:3333/cybersecurity' },
                    ],
                },
                {
                    title: 'About Us',
                    links: [
                        { name: 'About Us', target: 'http://localhost:3333/about-us' },
                        { name: 'Contact Us', target: 'http://localhost:3333/contact-us' },
                        { name: 'News and Updates', target: 'http://localhost:3333/news-updates' },
                        { name: 'Careers', target: 'http://localhost:3333/careers' },
                    ],
                },
                {
                    title: 'Support',
                    links: [
                        { name: 'Help Center', target: 'http://localhost:3333/help-center' },
                        { name: 'Feedback', target: 'http://localhost:3333/feedback' },
                        { name: 'Terms of Service', target: 'http://localhost:3333/terms-of-service' },
                        { name: 'Privacy Policy', target: 'http://localhost:3333/privacy-policy' },
                    ],
                },
            ], socialMediaTitle: "Social Media", accessibilityTitle: "accessibility Tools", socialMediaLinks: [
                {
                    title: 'Twitter',
                    target: 'https://twitter.com/dga_gov',
                    icon: { name: 'TwitterIcon', variant: 'stroke' },
                },
                {
                    title: 'YouTube',
                    target: 'https://youtube.com/dga',
                    icon: { name: 'YoutubeIcon', variant: 'stroke' },
                },
                {
                    title: 'Instagram',
                    target: 'https://instagram.com/dga_gov',
                    icon: { name: 'InstagramIcon', variant: 'stroke' },
                },
            ], accessibilityLinks: [
                {
                    title: 'Screen Reader',
                    target: 'http://localhost:3333/accessibility/screen-reader',
                    icon: { name: "VoiceIcon", variant: "stroke" },
                },
                {
                    title: 'High Contrast',
                    target: 'http://localhost:3333/accessibility/contrast',
                    icon: { name: "VoiceIcon", variant: "stroke" },
                },
                {
                    title: 'Text Size',
                    target: 'http://localhost:3333/accessibility/text-size',
                    icon: { name: "VoiceIcon", variant: "stroke" },
                },
            ], copyright: "\u00A9 2024 Digital Government Authority. All rights reserved.", basicLinks: [
                { name: 'Home', target: '/' },
                { name: 'About DGA', target: '/about' },
                { name: 'News & Events', target: '/news' },
                { name: 'Publications', target: '/publications' },
                { name: 'Careers', target: '/careers' },
                { name: 'Procurement', target: '/procurement' },
                { name: 'Open Data', target: '/data' },
                { name: 'Statistics', target: '/stats' },
            ], extraLinks: [
                { name: 'Terms and Conditions', target: '/terms' },
                { name: 'Privacy Policy', target: '/privacy' },
                { name: 'Cookie Policy', target: '/cookies' },
                { name: 'Accessibility Statement', target: '/accessibility' },
            ], bottomImages: ['https://dga-nds-fbhtx.ondigitalocean.app/mobile-logo.svg', 'https://dga-nds-fbhtx.ondigitalocean.app/mobile-logo.svg'] }), h("h1", { key: 'b259cb5692c841d539b5d3869e4d917ddcb77c87', style: { padding: '40px', color: 'red' } }, " Footer Example Light Mode"), h("dga-footer", { key: 'cefe330943ea59479a8d49992c41db7b1b0d4e21', background: "Light", NavLinks: true, groupLinks: [
                {
                    title: 'Group Label',
                    links: [
                        { name: 'Footer Link', target: '' },
                        { name: 'Footer Link', target: '' },
                        { name: 'Footer Link', target: '' },
                        { name: 'Footer Link', target: '' },
                        { name: 'Footer Link', target: '' },
                    ],
                },
                {
                    title: 'Group Label',
                    links: [
                        { name: 'Footer Link', target: '' },
                        { name: 'Footer Link', target: '' },
                        { name: 'Footer Link', target: '' },
                        { name: 'Footer Link', target: '' },
                        { name: 'Footer Link', target: '' },
                    ],
                },
                {
                    title: 'Group Label',
                    links: [
                        { name: 'Footer Link', target: '' },
                        { name: 'Footer Link', target: '' },
                        { name: 'Footer Link', target: '' },
                        { name: 'Footer Link', target: '' },
                        { name: 'Footer Link', target: '' },
                    ],
                },
            ], socialMediaTitle: "Social Media", accessibilityTitle: "accessibility Tools", socialMediaLinks: [
                {
                    title: 'any',
                    target: '',
                    icon: { name: 'ArrowRight02Icon', variant: 'stroke' },
                },
                {
                    title: 'any',
                    target: '',
                    icon: { name: 'ArrowRight02Icon', variant: 'stroke' },
                },
                {
                    title: 'any',
                    target: '',
                    icon: { name: 'ArrowRight02Icon', variant: 'stroke' },
                },
                {
                    title: 'any',
                    target: '',
                    icon: { name: 'ArrowRight02Icon', variant: 'stroke' },
                },
            ], accessibilityLinks: [
                {
                    title: 'any',
                    target: '',
                    icon: { name: 'ArrowRight02Icon', variant: 'stroke' },
                },
                {
                    title: 'any',
                    target: '',
                    icon: { name: 'ArrowRight02Icon', variant: 'stroke' },
                },
                {
                    title: 'any',
                    target: '',
                    icon: { name: 'ArrowRight02Icon', variant: 'stroke' },
                },
            ], copyright: "All Right Reserved For Digital Government Authority \u00A9 2024", basicLinks: [
                {
                    name: 'Footer Link',
                    target: '#',
                },
                {
                    name: 'Footer Link',
                    target: '#',
                },
                {
                    name: 'Footer Link',
                    target: '#',
                },
                {
                    name: 'Footer Link',
                    target: '#',
                },
                {
                    name: 'Footer Link',
                    target: '#',
                },
                {
                    name: 'Footer Link',
                    target: '#',
                },
                {
                    name: 'Footer Link',
                    target: '#',
                },
                {
                    name: 'Footer Link',
                    target: '#',
                },
            ], extraLinks: [
                {
                    name: 'Terms and Conditions',
                    target: '#',
                },
                {
                    name: 'Privacy Policy',
                    target: '#',
                },
            ] }), h("h1", { key: '4aeb75cbbfdcaf70f373fc7e5fa341473cbfd364', style: { padding: '40px', color: 'red' } }, "End Footer Example Here Zoz")));
    }
    static get is() { return "dga-footer-usage"; }
}
