// @platformscode/core/dist/collection/usage/dga-avatar-usage.js
import { h } from "@stencil/core";
export class DgaAvatarUsage {
    constructor() {
        this.variant = 'neutral';
    }
    render() {
        return (h("div", { key: '3a583b627573a5d3092e833bf2c9b74d901f5fed' }, h("h1", { key: '06e9492e0c7cb5a2812a775168aea3e6345bd9cb' }, "Avatar Example"), h("div", { key: 'ba81aad310ca183f3a17acca96512ae9712c5020', style: { padding: '50px', display: 'flex', gap: '20px' } }, h("div", { key: '5905c6fea7b56a33717873b1a24a1bc1ec10b1f9', style: { display: 'flex', flexDirection: 'column', gap: '20px' } }, h("div", { key: '3e5103b26f65bb048d3498874ed4750e6f6d9422', style: { padding: "10px", display: 'flex', flexDirection: 'column', gap: '20px', backgroundColor: "black" } }, h("dga-avatar", { key: '0523ab6d65cde4dbbccc2df1e0c617acd5066818', size: 120, border: false, square: true, sx: {
                "& .dga-avatar": {
                    borderRadius: "100px"
                }
            } }), h("dga-avatar", { key: 'f5c18465900032bca70edd3881e51ccdcf6f9403', size: 120, border: true, square: true })), h("dga-avatar", { key: '3f9bea90d36869ee4b91b4f99a09d1079ab5ebc5', type: "icon", size: 120, icon: {
                name: 'user',
                variant: "stroke",
                type: "rounded"
            } }), h("dga-avatar", { key: '29656a76abc53a7d9db5bb208bb216af15701b5f', type: "icon", size: 80, icon: {
                name: 'user',
                variant: "stroke",
                type: "rounded"
            } }), h("dga-avatar", { key: 'b231be665cdf003ad0709e6243e082767b288445', type: "icon", size: 68, icon: {
                name: 'user',
                variant: "stroke",
                type: "rounded"
            } }), h("dga-avatar", { key: '22cd1631de536d7da98cd94c6e5e12246e7fb8c0', type: "icon", size: 48, icon: {
                name: 'user',
                variant: "stroke",
                type: "rounded"
            } }), h("dga-avatar", { key: '0579fae37b7b8b5c423ffcafa544e87d106628fe', type: "icon", size: 40, icon: {
                name: 'user',
                variant: "stroke",
                type: "rounded"
            } }), h("dga-avatar", { key: 'aa4b17fc3c1bdb1078fb9443da294fe2662b1495', type: "icon", size: 32, icon: {
                name: 'user',
                variant: "stroke",
                type: "rounded"
            } }), h("dga-avatar", { key: '2ccd27f61b7ef5152c5c5cc1eb8d257ab1d59323', type: "icon", size: 24, icon: {
                name: 'user',
                variant: "stroke",
                type: "rounded"
            } })), h("dga-avatar-group", { key: 'f7893d2a922d7a3320f4ca2d686fc6b02164075d', stacked: true }, h("dga-avatar", { key: '7da6aa774b8191340ba710dac81ddbf8331269f8', type: "initials", text: 'AB', size: 40 }), h("dga-avatar", { key: 'b8dbd22ef151463fc68f2ee492903008ff49015c', type: "initials", text: 'CD', size: 40 }), h("dga-avatar", { key: 'fb1665fd2dee927dd00acf1576715bd0b94efa67', type: "initials", text: 'SB', size: 40 }), h("dga-avatar", { key: 'b02fe1efd279b74290399c312c800df8dd20586b', type: "initials", text: 'WK', size: 40 }), h("dga-avatar", { key: 'cb1a83124d4b36d699063e5367e2d1e9d47a130d', type: 'initials', text: `+4`, size: 40 })), h("div", { key: '94e1aefc5a074dd4c6975f53423b96637d93b250', style: { display: 'flex', flexDirection: 'column', gap: '20px' } }))));
    }
    static get is() { return "dga-avatar-usage"; }
    static get states() {
        return {
            "variant": {}
        };
    }
}
