// @platformscode/core/dist/collection/components/dga-second-nav-header/dga-second-nav-header-item.js
import { Host, h } from "@stencil/core";
export class DgaSecondNavHeaderItem {
    connectedCallback() {
        this.el.childNodes.forEach((item) => {
            const el = item;
            if (el.tagName === 'DGA-ICON'.toLocaleUpperCase() && el.attributes.getNamedItem('slot') === null) {
                el.setAttribute('slot', 'icon');
            }
        });
    }
    render() {
        return (h(Host, { key: '87572880e04d67c78bbe24dce793d6147d166d37' }, h("slot", { key: 'e478e5be9f8231a25f8f76dc5b44b062552d3305', name: 'icon' }), h("slot", { key: 'bdf901885edd5f6edfe77ae63311748da8f6550e', name: 'label' }, this.label)));
    }
    static get is() { return "dga-second-nav-header-item"; }
    static get properties() {
        return {
            "label": {
                "type": "string",
                "attribute": "label",
                "mutable": false,
                "complexType": {
                    "original": "string",
                    "resolved": "string",
                    "references": {}
                },
                "required": true,
                "optional": false,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false
            }
        };
    }
    static get elementRef() { return "el"; }
}
