// @platformscode/core/dist/collection/components/dga-second-nav-header/dga-second-nav-header.js
import { Host, h } from "@stencil/core";
export class DgaSecondNavHeader {
    constructor() {
        this.variant = 'primary';
    }
    componentWillLoad() {
        this.el.childNodes.forEach((item) => {
            const el = item;
            if (el.tagName === 'DGA-SECOND-NAV-HEADER-CONTENT' && this.hideContent === true) {
                el.style.display = 'none';
            }
            if (el.tagName === 'DGA-SECOND-NAV-HEADER-ACTIONS' && this.hideActions === true) {
                el.style.display = 'none';
            }
            if (el.tagName === 'DGA-SECOND-NAV-HEADER-CONTENT' && el.attributes.getNamedItem('slot') === null) {
                el.setAttribute('slot', 'content');
            }
            if (el.tagName === 'DGA-SECOND-NAV-HEADER-ACTIONS' && el.attributes.getNamedItem('slot') === null) {
                el.setAttribute('slot', 'actions');
            }
        });
    }
    render() {
        return (h(Host, { key: '4d6a56396f788b3f018b3774c84fb6bc0d0849a9' }, h("div", { key: 'f6c6cf9410c08dfdab2c9889852649634b763dbe', class: `wrapper ${this.variant} `, "aria-label": 'a secondary navigation navbar' }, h("div", { key: '4b851ce02ce31b959365e7a09374097d369973ac', class: `container ${this.hideDivider ? '' : 'border'}` }, h("div", { key: '6214ce6582acd5ac51e252425546acd6a5edad46', class: `content` }, h("slot", { key: '3a70b02c824ba50ca99427169697341dc781c5d5', name: "content" })), h("div", { key: '2193cb09f2bd21e7741460d91ab3ab38c723370b', class: `actions` }, h("slot", { key: 'fa36abcb4f2e101b986b746af0dad2af3fee7fa4', name: "actions" }))))));
    }
    static get is() { return "dga-second-nav-header"; }
    static get encapsulation() { return "scoped"; }
    static get properties() {
        return {
            "variant": {
                "type": "string",
                "attribute": "variant",
                "mutable": false,
                "complexType": {
                    "original": "'primary' | 'gray'",
                    "resolved": "\"gray\" | \"primary\"",
                    "references": {}
                },
                "required": false,
                "optional": false,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false,
                "defaultValue": "'primary'"
            },
            "hideDivider": {
                "type": "boolean",
                "attribute": "hide-divider",
                "mutable": true,
                "complexType": {
                    "original": "boolean",
                    "resolved": "boolean",
                    "references": {}
                },
                "required": false,
                "optional": false,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false
            },
            "hideActions": {
                "type": "boolean",
                "attribute": "hide-actions",
                "mutable": false,
                "complexType": {
                    "original": "boolean",
                    "resolved": "boolean",
                    "references": {}
                },
                "required": false,
                "optional": false,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false
            },
            "hideContent": {
                "type": "boolean",
                "attribute": "hide-content",
                "mutable": false,
                "complexType": {
                    "original": "boolean",
                    "resolved": "boolean",
                    "references": {}
                },
                "required": false,
                "optional": false,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false
            }
        };
    }
    static get elementRef() { return "el"; }
}
