// @platformscode/core/dist/collection/components/dga-switch/dga-switch.js
import { h, Host } from "@stencil/core";
import { SxHandler } from "../../utils/sx-handler";
export class DgaSwitch {
    constructor() {
        this.color = 'brand';
        this.checked = false;
        this.disabled = false;
        this.id = this.generateGuid();
        this.name = this.generateGuid();
        this.value = true;
        this.isChecked = this.checked;
        // Handle active class on mouse down
        this.handleActive = () => {
            if (!this.disabled) {
                this.labelRef.classList.add('active');
            }
        };
        // Toggle switch on change event
        this.handleRemoveActive = () => {
            this.labelRef.classList.remove('active');
            this.isChecked = !this.isChecked;
        };
        // Handle focus
        this.handleFocus = () => {
            this.labelRef.classList.add('focus');
        };
        // Handle focus out
        this.handleFocusOut = () => {
            this.labelRef.classList.remove('focus');
        };
        // Handle Escape key event to remove focus
        this.handleEscape = (event) => {
            if (event.key === 'Escape') {
                this.handleFocusOut();
                const inputElement = this.el.querySelector('input');
                inputElement === null || inputElement === void 0 ? void 0 : inputElement.blur();
            }
        };
    }
    // Helper to generate a unique ID similar to React's `guid()`
    generateGuid() {
        return Math.random().toString(36).substr(2, 9);
    }
    componentDidLoad() {
        window.addEventListener('keydown', this.handleEscape);
        this.sxHandler.processSx(this.sx);
    }
    disconnectedCallback() {
        window.removeEventListener('keydown', this.handleEscape);
        this.sxHandler.cleanup();
    }
    sxChanged() {
        this.sxHandler.processSx(this.sx);
    }
    componentWillLoad() {
        this.sxHandler = new SxHandler(this.el, 'dga-switch');
    }
    render() {
        return (h(Host, { key: 'ba9b37f9628f33216c19e0ab0fd6297fd47a4af8', id: this.sxHandler.getId(), style: this.sxHandler.getStyles() }, h("label", { key: '0e13d24d07611003b8dda659130d6b1cb83406c6', ref: el => (this.labelRef = el), class: `switch ${this.color ? `switch--${this.color}` : ''}`, onMouseDown: this.handleActive, onChange: this.handleRemoveActive }, h("div", { key: '9038e7905715e003019ae410d8bdd4f48d7c3ea8', class: "switch-main" }, h("div", { key: 'e1975b977c606e54ecd6204e5d6277cf101623c5', class: `switch-main__content ${this.disabled ? 'disabled' : ''}` }, h("div", { key: '37c479b73870964b5e3b87db8780f4353a97f48d', class: "switch-wrapper" }, h("input", { key: '89b9844185c385ad13ed813c1df1a60257e6bf1a', type: "checkbox", name: this.name, value: this.value, id: this.id, disabled: this.disabled, onFocus: this.handleFocus, onBlur: this.handleFocusOut, onChange: this.onChange, onInput: this.onInput, checked: this.isChecked }), h("span", { key: 'd3375662f4318707490e83d3ecad6e5652236c78', class: `slider round ${this.isChecked ? 'slider--active' : 'slider--inactive'}` })), !this.hasHelperText && !this.helperText ? (this.label && h("span", { class: "switch__label" }, this.label)) : (h("div", { class: "dga-flex-column" }, h("span", { class: "switch__label" }, this.label), h("span", { class: "switch__helper-text" }, this.helperText)))), this.alertMessage && (h("div", { key: '9234fe3bfa83102d071f31e84c6eab540bd40d22', class: "switch__warning" }, h("span", { key: '599e8c28199149f632e9dfc5fe2675a4ba2d2bca', class: "switch__warning-icon" }, h("svg", { key: '5fb755717ff5ecef1ba0eee73392d3db7c8a27b9', width: "16", height: "16", viewBox: "0 0 16 16", fill: "none" }, h("path", { key: '166c967368936015cd5e965f71f1460cec795c95', d: "M8 1.33333C4.324 1.33333 1.33333 4.324 1.33333 8C1.33333 11.676 4.324 14.6667 8 14.6667C11.676 14.6667 14.6667 11.676 14.6667 8C14.6667 4.324 11.676 1.33333 8 1.33333ZM8 13.3333C5.05467 13.3333 2.66667 10.9453 2.66667 8C2.66667 5.05467 5.05467 2.66667 8 2.66667C10.9453 2.66667 13.3333 5.05467 13.3333 8C13.3333 10.9453 10.9453 13.3333 8 13.3333ZM7.33333 4H8.66667V8H7.33333V4ZM7.33333 9.33333H8.66667V10.6667H7.33333V9.33333Z", fill: "currentColor" }))), h("span", { key: '399df61257674a35ce0541d7ccc04114bf79981a', class: "switch__warning-text" }, this.alertText)))))));
    }
    static get is() { return "dga-switch"; }
    static get encapsulation() { return "shadow"; }
    static get originalStyleUrls() {
        return {
            "$": ["dga-switch.scss"]
        };
    }
    static get styleUrls() {
        return {
            "$": ["dga-switch.css"]
        };
    }
    static get properties() {
        return {
            "sx": {
                "type": "string",
                "attribute": "sx",
                "mutable": false,
                "complexType": {
                    "original": "SxProps | string | undefined",
                    "resolved": "SxProps | string",
                    "references": {
                        "SxProps": {
                            "location": "import",
                            "path": "../../utils/sx-handler",
                            "id": "src/utils/sx-handler.ts::SxProps"
                        }
                    }
                },
                "required": false,
                "optional": true,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false
            },
            "color": {
                "type": "string",
                "attribute": "color",
                "mutable": false,
                "complexType": {
                    "original": "string",
                    "resolved": "string",
                    "references": {}
                },
                "required": false,
                "optional": false,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false,
                "defaultValue": "'brand'"
            },
            "label": {
                "type": "string",
                "attribute": "label",
                "mutable": false,
                "complexType": {
                    "original": "string",
                    "resolved": "string",
                    "references": {}
                },
                "required": false,
                "optional": true,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false
            },
            "checked": {
                "type": "boolean",
                "attribute": "checked",
                "mutable": true,
                "complexType": {
                    "original": "boolean",
                    "resolved": "boolean",
                    "references": {}
                },
                "required": false,
                "optional": false,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false,
                "defaultValue": "false"
            },
            "hasHelperText": {
                "type": "boolean",
                "attribute": "has-helper-text",
                "mutable": false,
                "complexType": {
                    "original": "boolean",
                    "resolved": "boolean",
                    "references": {}
                },
                "required": false,
                "optional": true,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false
            },
            "helperText": {
                "type": "string",
                "attribute": "helper-text",
                "mutable": false,
                "complexType": {
                    "original": "string",
                    "resolved": "string",
                    "references": {}
                },
                "required": false,
                "optional": true,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false
            },
            "alertMessage": {
                "type": "boolean",
                "attribute": "alert-message",
                "mutable": false,
                "complexType": {
                    "original": "boolean",
                    "resolved": "boolean",
                    "references": {}
                },
                "required": false,
                "optional": true,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false
            },
            "alertText": {
                "type": "string",
                "attribute": "alert-text",
                "mutable": false,
                "complexType": {
                    "original": "string",
                    "resolved": "string",
                    "references": {}
                },
                "required": false,
                "optional": true,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false
            },
            "disabled": {
                "type": "boolean",
                "attribute": "disabled",
                "mutable": false,
                "complexType": {
                    "original": "boolean",
                    "resolved": "boolean",
                    "references": {}
                },
                "required": false,
                "optional": false,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false,
                "defaultValue": "false"
            },
            "id": {
                "type": "string",
                "attribute": "id",
                "mutable": false,
                "complexType": {
                    "original": "string",
                    "resolved": "string",
                    "references": {}
                },
                "required": false,
                "optional": false,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false,
                "defaultValue": "this.generateGuid()"
            },
            "name": {
                "type": "string",
                "attribute": "name",
                "mutable": false,
                "complexType": {
                    "original": "string",
                    "resolved": "string",
                    "references": {}
                },
                "required": false,
                "optional": false,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false,
                "defaultValue": "this.generateGuid()"
            },
            "value": {
                "type": "any",
                "attribute": "value",
                "mutable": false,
                "complexType": {
                    "original": "boolean | any",
                    "resolved": "any",
                    "references": {}
                },
                "required": false,
                "optional": false,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false,
                "defaultValue": "true"
            },
            "onChange": {
                "type": "unknown",
                "attribute": "on-change",
                "mutable": false,
                "complexType": {
                    "original": "(event?: any) => void",
                    "resolved": "(event?: any) => void",
                    "references": {}
                },
                "required": false,
                "optional": true,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false
            },
            "onInput": {
                "type": "unknown",
                "attribute": "on-input",
                "mutable": false,
                "complexType": {
                    "original": "(event?: any) => void",
                    "resolved": "(event?: any) => void",
                    "references": {}
                },
                "required": false,
                "optional": true,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false
            }
        };
    }
    static get states() {
        return {
            "isChecked": {}
        };
    }
    static get elementRef() { return "el"; }
    static get watchers() {
        return [{
                "propName": "sx",
                "methodName": "sxChanged"
            }];
    }
}
