// @platformscode/core/dist/collection/nds-components/nds-table-of-content/nds-table-of-content-usage.js
import { h } from "@stencil/core";
export class NdsTableOfContentUsage {
    render() {
        return (h("div", { key: '819bd10e95e3edba4d0b6ea01eb1aed583dfce14', style: {
                position: 'relative',
                width: '100%',
                maxWidth: '1200px',
                margin: '0 auto',
                padding: '20px'
            } }, h("h1", { key: '91cdaf8b92dbc751441a84e6f094a006d997f632' }, "Table of Content Example"), h("section", { key: '3f4c366c5afe758b33c9ffc2be44ed355ca945ae' }, h("div", { key: '1f6144981097182371904b57900c0fab9ad1878f', id: "section_1", style: {
                minHeight: "100vh",
                backgroundColor: "gray",
                padding: "20px",
                margin: "20px 0 200px 0"
            } }, h("h2", { key: '55ac3d3c19bc52df21534a3aa5c45df41494db64' }, "Section 1"), h("div", { key: '80cbe9adf44c56ec310bd7822adf91407d23bacd', id: "section_1_child_1", style: {
                minHeight: "100vh",
                backgroundColor: "rgba(255,0,0,0.3)",
                margin: "20px 0",
                padding: "20px"
            } }, h("h3", { key: '47c02a0cf9e96dc2c76c38069c2947b485bf2020' }, "Section 1 Child 1")), h("div", { key: '67a035faada4a72443202f3773b7b6852a548078', id: "section_1_child_2", style: {
                minHeight: "100vh",
                backgroundColor: "rgba(0,0,255,0.3)",
                margin: "20px 0",
                padding: "20px"
            } }, h("h3", { key: '13e167c6bb796405f61f4d7e2b07eb2295951537' }, "Section 1 Child 2")), h("div", { key: '19302f93f150f04498c76e7e3981cc9a937c460c', id: "section_1_child_3", style: {
                minHeight: "100vh",
                backgroundColor: "rgba(0,255,0,0.3)",
                margin: "20px 0",
                padding: "20px"
            } }, h("h3", { key: 'd14b0a6089b3cc61386e677022b33760446c94d3' }, "Section 1 Child 3"), h("div", { key: '16dfb4f0205e93efa321965488cc42936ae19520', id: "section_1_sub_child_1", style: {
                minHeight: "50vh",
                backgroundColor: "rgba(255,255,255,0.3)",
                margin: "20px 0",
                padding: "20px"
            } }, h("h4", { key: '55465560552ae3de409af8fbcd2b0ee88daf216b' }, "Section 1 Sub Child 1")), h("div", { key: '560c02814d745d8028077449150586bca956ea68', id: "section_1_sub_child_2", style: {
                minHeight: "50vh",
                backgroundColor: "rgba(255,255,255,0.3)",
                margin: "20px 0",
                padding: "20px"
            } }, h("h4", { key: '3198413f9f5e3c0c864cf12f4567375d079d645e' }, "Section 1 Sub Child 2")), h("div", { key: 'ce2a0fe28bd218334d403ba97739773aa987a22b', id: "section_1_sub_child_3", style: {
                minHeight: "50vh",
                backgroundColor: "rgba(255,255,255,0.3)",
                margin: "20px 0",
                padding: "20px"
            } }, h("h4", { key: 'b60f9c9ca3b70638d971c6911f5d065e7b16459d' }, "Section 1 Sub Child 3")))), h("div", { key: '83ca6530654fbb0565e11955bda0f902d4aa7bbc', id: "section_2", style: {
                minHeight: "100vh",
                backgroundColor: "rgba(0,0,0,0.8)",
                color: "white",
                padding: "20px",
                margin: "20px 0"
            } }, h("h2", { key: '41b89f0c025644330e13495d8a231db3bc0766f0' }, "Section 2")), h("div", { key: '085a8c521c17fcace5ccce494f02df0b91f26ca8', id: "section_3", style: {
                minHeight: "100vh",
                backgroundColor: "brown",
                padding: "20px",
                margin: "20px 0"
            } }, h("h2", { key: '0af0a0ef370ed82a5675fe43af2a49ca65d8385a' }, "Section 3"), h("div", { key: '1432fbdbad8487bd6d55a21d56646ca7fe36d005', id: "section_3_child_1", style: {
                minHeight: "50vh",
                backgroundColor: "rgba(255,255,255,0.3)",
                margin: "20px 0",
                padding: "20px"
            } }, h("h3", { key: '1b1ac49d5652dbbbd2beba407ce1679ce3ae500c' }, "Section 3 Child 1")), h("div", { key: '1bdc723b43b40dcf4d10a2aaf6afaabf54f19749', id: "section_3_child_2", style: {
                minHeight: "50vh",
                backgroundColor: "rgba(255,255,255,0.3)",
                margin: "20px 0",
                padding: "20px"
            } }, h("h3", { key: 'e1b0af152d3540d79a27396090fec1ee058b1a85' }, "Section 3 Child 2"), h("div", { key: '254db3888856828d40551f8f31c2e2a890f2353d', id: "section_3_child_2_1", style: {
                minHeight: "30vh",
                backgroundColor: "rgba(0,0,0,0.1)",
                margin: "20px 0",
                padding: "20px"
            } }, h("h4", { key: '9b1eba61d3a94f3cf883e09e1953008902e171f1' }, "Section 3 Child 2.1")), h("div", { key: '5e2935f570c90f13334b4b9909e44041a9b94dd5', id: "section_3_child_2_2", style: {
                minHeight: "30vh",
                backgroundColor: "rgba(0,0,0,0.1)",
                margin: "20px 0",
                padding: "20px"
            } }, h("h4", { key: '04e16bd451630e9b2518447899d73ee62cfd7aff' }, "Section 3 Child 2.2")), h("div", { key: '54fd805685bd48de1c96383a8ff65ea1dad19b10', id: "section_3_child_2_3", style: {
                minHeight: "30vh",
                backgroundColor: "rgba(0,0,0,0.1)",
                margin: "20px 0",
                padding: "20px"
            } }, h("h4", { key: '62d77b202b63f6b747d46df504562d52064516f9' }, "Section 3 Child 2.3"))))), h("div", { key: '143f4ef993450120c9aae58ee235d3ab0e6e2c9b', style: {
                position: "fixed",
                top: "100px",
                left: "40px",
                zIndex: '100',
                background: "white",
                borderRadius: "8px",
                boxShadow: "0 2px 8px rgba(0, 0, 0, 0.1)",
            } }, h("dga-table-of-content", { key: '68b905bdc574a0c36de16d2d52cc43f8af3f126a', title: "On this Page", subTitle: "Page Title", sections: [
                {
                    Name: 'Section 1',
                    Target: 'section_1',
                    Children: [
                        { Name: 'Section 1 Child 1', Target: 'section_1_child_1' },
                        { Name: 'Section 1 Child 2', Target: 'section_1_child_2' },
                        {
                            Name: 'Section 1 Child 3',
                            Target: 'section_1_child_3',
                            Children: [
                                { Name: 'Sub Child 1', Target: 'section_1_sub_child_1' },
                                { Name: 'Sub Child 2', Target: 'section_1_sub_child_2' },
                                { Name: 'Sub Child 3', Target: 'section_1_sub_child_3' }
                            ]
                        }
                    ]
                },
                { Name: 'Section 2', Target: 'section_2' },
                {
                    Name: 'Section 3',
                    Target: 'section_3',
                    Children: [
                        { Name: 'Section 3 Child 1', Target: 'section_3_child_1' },
                        {
                            Name: 'Section 3 Child 2',
                            Target: 'section_3_child_2',
                            Children: [
                                { Name: 'Child 2.1', Target: 'section_3_child_2_1' },
                                { Name: 'Child 2.2', Target: 'section_3_child_2_2' },
                                { Name: 'Child 2.3', Target: 'section_3_child_2_3' }
                            ]
                        }
                    ]
                }
            ] })), h("div", { key: '53fbcabe9af9e2a9a9a27ca7a6896aa036a3bb92', style: {
                position: "fixed",
                top: "100px",
                right: "40px",
                zIndex: '100',
                background: "white",
                borderRadius: "8px",
                boxShadow: "0 2px 8px rgba(0, 0, 0, 0.1)",
            } }, h("dga-table-of-content", { key: '54559ebc83088265ce72d225d2ed5ff08816e5c0', title: "\u0641\u064A \u0647\u0630\u0647 \u0627\u0644\u0635\u0641\u062D\u0629", subTitle: "[\u0627\u0633\u0645 \u0627\u0644\u0635\u0641\u062D\u0629]", rtl: true, sections: [
                {
                    Name: 'قسم صفحة',
                    Target: 'section_1',
                    Children: [
                        { Name: 'قسم صفحة', Target: 'section_1_child_1' },
                        { Name: 'قسم صفحة', Target: 'section_1_child_2' },
                        {
                            Name: 'قسم صفحة',
                            Target: 'section_1_child_3',
                            Children: [
                                { Name: 'قسم صفحة', Target: 'section_1_sub_child_1' },
                                { Name: 'قسم صفحة', Target: 'section_1_sub_child_2' },
                                { Name: 'قسم صفحة', Target: 'section_1_sub_child_3' }
                            ]
                        }
                    ]
                },
                { Name: 'Section 2', Target: 'section_2' },
                {
                    Name: 'Section 3',
                    Target: 'section_3',
                    Children: [
                        { Name: 'Section 3 Child 1', Target: 'section_3_child_1' },
                        {
                            Name: 'Section 3 Child 2',
                            Target: 'section_3_child_2',
                            Children: [
                                { Name: 'Child 2.1', Target: 'section_3_child_2_1' },
                                { Name: 'Child 2.2', Target: 'section_3_child_2_2' },
                                { Name: 'Child 2.3', Target: 'section_3_child_2_3' }
                            ]
                        }
                    ]
                }
            ] }))));
    }
    static get is() { return "nds-table-of-content-usage"; }
}
