// @platformscode/core/dist/collection/components/dga-radio-button/dga-radio-button.js
import { h } from "@stencil/core";
import { guid } from "../../utils";
import { isRTL } from "../../utils/rtl";
import { alertCircleStrokeRoundedIcon } from "../../assets/icons/index";
export class RadioButton {
    constructor() {
        this.color = 'brand';
        this.hasHelperText = false;
        this.disabled = false;
        this.id = guid();
        this.value = '';
        this.checked = false;
        // New SX prop for custom styling
        this.sx = {};
        // Store computed styles
        this.customStyleSheet = null;
        this.styleId = '';
        this.handleActive = () => {
            var _a;
            (_a = this.radioLabelRef) === null || _a === void 0 ? void 0 : _a.classList.add('active');
        };
        this.handleRemoveActive = () => {
            var _a;
            (_a = this.radioLabelRef) === null || _a === void 0 ? void 0 : _a.classList.remove('active');
        };
        this.handleFocus = () => {
            var _a;
            (_a = this.radioLabelRef) === null || _a === void 0 ? void 0 : _a.classList.add('focus');
        };
        this.handleFocusOut = () => {
            var _a;
            (_a = this.radioLabelRef) === null || _a === void 0 ? void 0 : _a.classList.remove('focus');
        };
        this.handleEscape = (event) => {
            var _a, _b;
            if (event.key === 'Escape') {
                (_b = (_a = this.radioLabelRef) === null || _a === void 0 ? void 0 : _a.querySelector('input')) === null || _b === void 0 ? void 0 : _b.blur();
                this.handleFocusOut();
            }
        };
    }
    handleSxChange() {
        this.processStyles();
    }
    componentWillLoad() {
        // Generate a unique ID for this component instance
        this.styleId = `dga-radio-${this.id}-styles`;
        this.processStyles();
    }
    componentDidLoad() {
        window.addEventListener('keydown', this.handleEscape);
        // Apply styles after the component is loaded
        this.applyCustomStyles();
    }
    disconnectedCallback() {
        window.removeEventListener('keydown', this.handleEscape);
        // Remove the style element when component is destroyed
        if (this.customStyleSheet && this.customStyleSheet.parentNode) {
            this.customStyleSheet.parentNode.removeChild(this.customStyleSheet);
        }
    }
    applyCustomStyles() {
        // Remove existing style element if it exists
        if (this.customStyleSheet && this.customStyleSheet.parentNode) {
            this.customStyleSheet.parentNode.removeChild(this.customStyleSheet);
        }
        // Create a new style element
        this.customStyleSheet = document.createElement('style');
        this.customStyleSheet.id = this.styleId;
        document.head.appendChild(this.customStyleSheet);
    }
    processStyles() {
        if (!this.sx || Object.keys(this.sx).length === 0) {
            return;
        }
        // Convert the sx prop to CSS
        let cssText = '';
        // Track if we need to add a style element
        let hasCustomSelectors = false;
        // Process each property in sx
        for (const [key, value] of Object.entries(this.sx)) {
            // Handle CSS-like selectors (e.g., "& .radio__label")
            if (key.includes('&')) {
                hasCustomSelectors = true;
                const selector = key.replace('&', `#${this.id}`);
                cssText += `${selector} { `;
                // Handle objects with multiple CSS properties
                if (typeof value === 'object') {
                    for (const [propName, propValue] of Object.entries(value)) {
                        cssText += `${this.camelToDash(propName)}: ${propValue}; `;
                    }
                }
                else {
                    // Handle single property value
                    cssText += `${value}; `;
                }
                cssText += `}\n`;
            }
        }
        // If we have custom selectors, apply them as a stylesheet
        if (hasCustomSelectors) {
            // Wait for component to be in the DOM
            setTimeout(() => {
                if (this.customStyleSheet) {
                    this.customStyleSheet.textContent = cssText;
                }
            }, 0);
        }
    }
    // Helper to convert camelCase to dash-case for CSS properties
    camelToDash(str) {
        return str.replace(/([A-Z])/g, (g) => `-${g[0].toLowerCase()}`);
    }
    render() {
        const { el, disabled } = this;
        const dir = isRTL(el) ? 'rtl' : 'ltr';
        console.log(dir);
        // Extract direct style properties (non-selector styles)
        const inlineStyles = {};
        for (const [key, value] of Object.entries(this.sx || {})) {
            if (!key.includes('&')) {
                inlineStyles[key] = value;
            }
        }
        return (h("label", { key: '6a9c0349e4d34e040f7645d8007c60c3b66c8a15', ref: el => (this.radioLabelRef = el), id: this.id, class: `radio ${this.color ? 'radio--' + this.color : ''}`, style: inlineStyles, onMouseDown: this.handleActive, onMouseUp: this.handleRemoveActive, onMouseOut: this.handleRemoveActive, onChange: this.handleRemoveActive, tabindex: disabled ? undefined : 0 }, h("div", { key: 'c4b994a5b408d5e2c2b8d72f3baab984fd13a503', class: "radio-main" }, h("div", { key: '4a7b104f2b557faec2961f489f65f9b277b364a7', class: "radio-main__content" }, h("input", { key: '482b511b78780f626b53c672c2d83704a5b20266', type: "radio", name: this.name, value: this.value,
            // id={this.id}
            onChange: this.onChange, onInput: this.onInput, disabled: this.disabled,
            // onFocus={this.handleFocus}
            // onBlur={this.handleFocusOut}
            checked: this.checked }), h("div", { key: 'b840492ef7978645ee95bec33327219d7998bc64', class: "radio-container" }), (this.label || this.hasHelperText) && h("div", { key: 'e4024ea33fc2b921497d99f05370592c8211d530', class: "dga-flex-column" }, this.label && h("span", { key: 'dd166381716e6dbc03d443e7ac398d2c21c5ce0f', class: "radio__label" }, this.label), this.hasHelperText && this.helperText && h("span", { key: '90ae3e5e34c0743c72f039e29b4b7cf112ee697d', class: "radio__helper-text" }, this.helperText))), this.alertMessage && this.alertText && (h("div", { key: 'cecd80a9dacadce4cb3e379fab7baf1647c688ae', class: "radio__warning" }, h("span", { key: '5643b60b29ea7250b54239bdf8fd2ef9f252db14', class: "radio__warning-icon" }, h("dga-icon", { key: 'e884727842963526ea74ec210b84a0c6c2ab9969', icon: alertCircleStrokeRoundedIcon, size: 16 })), h("span", { key: 'bb2659c7af946977ffbc4984a9ad6b0145a1e007', class: "radio__warning-text" }, this.alertText))))));
    }
    static get is() { return "dga-radio-button"; }
    static get originalStyleUrls() {
        return {
            "$": ["dga-radio-button.scss"]
        };
    }
    static get styleUrls() {
        return {
            "$": ["dga-radio-button.css"]
        };
    }
    static get properties() {
        return {
            "color": {
                "type": "string",
                "attribute": "color",
                "mutable": false,
                "complexType": {
                    "original": "'neutral' | 'brand'",
                    "resolved": "\"brand\" | \"neutral\"",
                    "references": {}
                },
                "required": false,
                "optional": false,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false,
                "defaultValue": "'brand'"
            },
            "label": {
                "type": "string",
                "attribute": "label",
                "mutable": false,
                "complexType": {
                    "original": "string",
                    "resolved": "string",
                    "references": {}
                },
                "required": false,
                "optional": true,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false
            },
            "hasHelperText": {
                "type": "boolean",
                "attribute": "has-helper-text",
                "mutable": false,
                "complexType": {
                    "original": "boolean",
                    "resolved": "boolean",
                    "references": {}
                },
                "required": false,
                "optional": true,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false,
                "defaultValue": "false"
            },
            "helperText": {
                "type": "string",
                "attribute": "helper-text",
                "mutable": false,
                "complexType": {
                    "original": "string",
                    "resolved": "string",
                    "references": {}
                },
                "required": false,
                "optional": true,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false
            },
            "alertMessage": {
                "type": "boolean",
                "attribute": "alert-message",
                "mutable": false,
                "complexType": {
                    "original": "boolean",
                    "resolved": "boolean",
                    "references": {}
                },
                "required": false,
                "optional": true,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false
            },
            "alertText": {
                "type": "string",
                "attribute": "alert-text",
                "mutable": false,
                "complexType": {
                    "original": "string",
                    "resolved": "string",
                    "references": {}
                },
                "required": false,
                "optional": true,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false
            },
            "disabled": {
                "type": "boolean",
                "attribute": "disabled",
                "mutable": false,
                "complexType": {
                    "original": "boolean",
                    "resolved": "boolean",
                    "references": {}
                },
                "required": false,
                "optional": false,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false,
                "defaultValue": "false"
            },
            "id": {
                "type": "string",
                "attribute": "id",
                "mutable": false,
                "complexType": {
                    "original": "string",
                    "resolved": "string",
                    "references": {}
                },
                "required": false,
                "optional": false,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false,
                "defaultValue": "guid()"
            },
            "name": {
                "type": "string",
                "attribute": "name",
                "mutable": false,
                "complexType": {
                    "original": "string",
                    "resolved": "string",
                    "references": {}
                },
                "required": false,
                "optional": false,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": true
            },
            "value": {
                "type": "string",
                "attribute": "value",
                "mutable": false,
                "complexType": {
                    "original": "string",
                    "resolved": "string",
                    "references": {}
                },
                "required": false,
                "optional": false,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false,
                "defaultValue": "''"
            },
            "checked": {
                "type": "boolean",
                "attribute": "checked",
                "mutable": false,
                "complexType": {
                    "original": "boolean",
                    "resolved": "boolean",
                    "references": {}
                },
                "required": false,
                "optional": true,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false,
                "defaultValue": "false"
            },
            "onChange": {
                "type": "unknown",
                "attribute": "on-change",
                "mutable": false,
                "complexType": {
                    "original": "(event: any) => void",
                    "resolved": "(event: any) => void",
                    "references": {}
                },
                "required": false,
                "optional": true,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false
            },
            "onInput": {
                "type": "unknown",
                "attribute": "on-input",
                "mutable": false,
                "complexType": {
                    "original": "(event: any) => void",
                    "resolved": "(event: any) => void",
                    "references": {}
                },
                "required": false,
                "optional": true,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false
            },
            "sx": {
                "type": "unknown",
                "attribute": "sx",
                "mutable": false,
                "complexType": {
                    "original": "SxProps",
                    "resolved": "SxProps",
                    "references": {
                        "SxProps": {
                            "location": "local",
                            "path": "/Users/mohamedaboelezz/projects/Azm-X/platform-code-cross-platform-storybook/packages/core/src/components/dga-radio-button/dga-radio-button.tsx",
                            "id": "src/components/dga-radio-button/dga-radio-button.tsx::SxProps"
                        }
                    }
                },
                "required": false,
                "optional": true,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "defaultValue": "{}"
            }
        };
    }
    static get elementRef() { return "el"; }
    static get watchers() {
        return [{
                "propName": "sx",
                "methodName": "handleSxChange"
            }];
    }
}
