// @platformscode/core/dist/collection/nds-components/nds-button/nds-button.js
import { h, Fragment, Host } from "@stencil/core";
export class NdsButton {
    constructor() {
        this.variant = 'primary-brand';
        this.size = 'md';
        this.label = '';
        this.type = 'button';
        this.inverted = false;
        this.destructive = false;
        this.selected = false;
    }
    handleClick(event) {
        if (!this.disabled) {
            this.onClick.emit(event);
        }
    }
    // componentWillLoad() {
    //   this.el.childNodes.forEach(item => {
    //     const el = item as HTMLElement;
    //     if (el.tagName === 'DGA-BUTTON-LEADING-ICON' && el.attributes.getNamedItem('slot') === null) {
    //       el.setAttribute('slot', 'button-leading-icon');
    //     }
    //   });
    // }
    render() {
        const classList = [
            'dga-btn',
            `dga-btn--${this.size}`,
            `dga-btn--${this.destructive ? 'des-' : ''}${this.variant}`,
            this.destructive ? `$dga-btn--${this.destructive}` : '',
            this.inverted ? `dga-btn--${this.variant}--on-color` : '',
            this.iconOnly ? 'dga-btn--icon' : '',
            this.selected ? `dga-btn--${this.variant}--selected` : '',
        ]
            .filter(Boolean)
            .join(' ');
        return (h(Host, { key: 'b4e653297ac3e2157b346edfc16dc4bf0d4f498c' }, h("button", { key: 'a0c3b1f19aae0847e3769aabde60e77ef9913cd0', type: this.type, class: classList, disabled: this.disabled, "aria-label": this.iconOnly && !this.label ? 'Button with icon' : this.label, role: "button", onClick: (event) => this.handleClick(event) }, h("slot", { key: 'ade659e73531be00e5dbd1313165585909fd07d4', name: "leading-icon" }, this.leadingIcon && (h("span", { key: 'aa6dd4837d8c6f3b50e1d9700cbb00aed34132b5', class: "dga-btn-icon", "aria-hidden": "true" }, h("dga-icon-v2", Object.assign({ key: '24fdc869fc91eca7dcea0b052c75ebf9ce1f6b27', name: this.leadingIcon }, this.leadingIconProps))))), !this.iconOnly && this.label && h(Fragment, { key: '8f90526c9d374388dea3090a494c5d0375aec565' }, this.label), !this.iconOnly && (h("slot", { key: '7feda037dddcc9f0b4c475202724bf3a102937f7', name: 'trailing-icon' }, this.trailingIcon && (h("span", { key: 'ba81f7e1291391fbfa9677e558770bcb840d894b', class: "dga-btn-icon", "aria-hidden": "true" }, h("dga-icon-v2", Object.assign({ key: 'fa8143b47012eb87ec810610b4449babd5668371', name: this.trailingIcon }, this.trailingIconProps)))))))));
    }
    static get is() { return "nds-button"; }
    static get encapsulation() { return "shadow"; }
    static get originalStyleUrls() {
        return {
            "$": ["nds-button.scss"]
        };
    }
    static get styleUrls() {
        return {
            "$": ["nds-button.css"]
        };
    }
    static get properties() {
        return {
            "variant": {
                "type": "string",
                "attribute": "variant",
                "mutable": false,
                "complexType": {
                    "original": "ButtonStyle",
                    "resolved": "\"primary-brand\" | \"primary-neutral\" | \"secondary-outline\" | \"secondary-solid\" | \"subtle\" | \"transparent\"",
                    "references": {
                        "ButtonStyle": {
                            "location": "local",
                            "path": "/Users/mohamedaboelezz/projects/Azm-X/platform-code-cross-platform-storybook/packages/core/src/nds-components/nds-button/nds-button.tsx",
                            "id": "src/nds-components/nds-button/nds-button.tsx::ButtonStyle"
                        }
                    }
                },
                "required": false,
                "optional": true,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false,
                "defaultValue": "'primary-brand'"
            },
            "size": {
                "type": "string",
                "attribute": "size",
                "mutable": false,
                "complexType": {
                    "original": "ButtonSize",
                    "resolved": "\"lg\" | \"md\" | \"sm\"",
                    "references": {
                        "ButtonSize": {
                            "location": "local",
                            "path": "/Users/mohamedaboelezz/projects/Azm-X/platform-code-cross-platform-storybook/packages/core/src/nds-components/nds-button/nds-button.tsx",
                            "id": "src/nds-components/nds-button/nds-button.tsx::ButtonSize"
                        }
                    }
                },
                "required": false,
                "optional": true,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false,
                "defaultValue": "'md'"
            },
            "label": {
                "type": "string",
                "attribute": "label",
                "mutable": false,
                "complexType": {
                    "original": "string",
                    "resolved": "string",
                    "references": {}
                },
                "required": false,
                "optional": true,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false,
                "defaultValue": "''"
            },
            "iconOnly": {
                "type": "boolean",
                "attribute": "icon-only",
                "mutable": false,
                "complexType": {
                    "original": "boolean",
                    "resolved": "boolean",
                    "references": {}
                },
                "required": false,
                "optional": true,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false
            },
            "leadingIcon": {
                "type": "string",
                "attribute": "leading-icon",
                "mutable": false,
                "complexType": {
                    "original": "string",
                    "resolved": "string",
                    "references": {}
                },
                "required": false,
                "optional": true,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false
            },
            "leadingIconProps": {
                "type": "unknown",
                "attribute": "leading-icon-props",
                "mutable": false,
                "complexType": {
                    "original": "Omit<Icon , 'name'>",
                    "resolved": "{ color?: string; type?: \"rounded\" | \"sharp\" | \"standard\"; size?: number; variant?: \"stroke\" | \"solid\" | \"duotone\" | \"twotone\" | \"bulk\"; }",
                    "references": {
                        "Omit": {
                            "location": "global",
                            "id": "global::Omit"
                        },
                        "Icon": {
                            "location": "import",
                            "path": "../../components",
                            "id": "src/components.d.ts::Icon"
                        }
                    }
                },
                "required": false,
                "optional": true,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false
            },
            "trailingIcon": {
                "type": "string",
                "attribute": "trailing-icon",
                "mutable": false,
                "complexType": {
                    "original": "string",
                    "resolved": "string",
                    "references": {}
                },
                "required": false,
                "optional": true,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false
            },
            "trailingIconProps": {
                "type": "unknown",
                "attribute": "trailing-icon-props",
                "mutable": false,
                "complexType": {
                    "original": "Omit<Icon , 'name'>",
                    "resolved": "{ color?: string; type?: \"rounded\" | \"sharp\" | \"standard\"; size?: number; variant?: \"stroke\" | \"solid\" | \"duotone\" | \"twotone\" | \"bulk\"; }",
                    "references": {
                        "Omit": {
                            "location": "global",
                            "id": "global::Omit"
                        },
                        "Icon": {
                            "location": "import",
                            "path": "../../components",
                            "id": "src/components.d.ts::Icon"
                        }
                    }
                },
                "required": false,
                "optional": true,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false
            },
            "disabled": {
                "type": "boolean",
                "attribute": "disabled",
                "mutable": false,
                "complexType": {
                    "original": "boolean",
                    "resolved": "boolean",
                    "references": {}
                },
                "required": false,
                "optional": true,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false
            },
            "type": {
                "type": "string",
                "attribute": "type",
                "mutable": false,
                "complexType": {
                    "original": "'button' | 'submit' | 'reset'",
                    "resolved": "\"button\" | \"reset\" | \"submit\"",
                    "references": {}
                },
                "required": false,
                "optional": true,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false,
                "defaultValue": "'button'"
            },
            "inverted": {
                "type": "boolean",
                "attribute": "inverted",
                "mutable": false,
                "complexType": {
                    "original": "boolean",
                    "resolved": "boolean",
                    "references": {}
                },
                "required": false,
                "optional": true,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false,
                "defaultValue": "false"
            },
            "destructive": {
                "type": "boolean",
                "attribute": "destructive",
                "mutable": false,
                "complexType": {
                    "original": "boolean",
                    "resolved": "boolean",
                    "references": {}
                },
                "required": false,
                "optional": true,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false,
                "defaultValue": "false"
            },
            "selected": {
                "type": "boolean",
                "attribute": "selected",
                "mutable": false,
                "complexType": {
                    "original": "boolean",
                    "resolved": "boolean",
                    "references": {}
                },
                "required": false,
                "optional": true,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false,
                "defaultValue": "false"
            }
        };
    }
    static get events() {
        return [{
                "method": "onClick",
                "name": "onClick",
                "bubbles": true,
                "cancelable": true,
                "composed": true,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "complexType": {
                    "original": "MouseEvent",
                    "resolved": "MouseEvent",
                    "references": {
                        "MouseEvent": {
                            "location": "global",
                            "id": "global::MouseEvent"
                        }
                    }
                }
            }];
    }
    static get elementRef() { return "el"; }
}
