// @platformscode/core/dist/collection/usage/dga-modal-usage.js
import { h } from "@stencil/core";
export class DgaModalUsage {
    constructor() {
        this.show = false;
    }
    render() {
        return (h("div", { key: '1d7c4cef1e1f98538810ce4b0eb77166f30cf030', style: { position: "relative" } }, h("h1", { key: 'a7ebea0aa5d67666454bfeecc4306a7ab7a71dfd' }, "Modal Example"), h("button", { key: 'ab9d06082be15e449e14b96eb2eac55b3f65a800', onClick: () => this.show = true }, "Open Modal"), h("dga-modal", { key: '7328e375daaf6abc83cfc846d8458b34f924f94b', buttonsList: [
                {
                    extraClass: '',
                    label: 'Close',
                    onClick: () => { },
                    title: 'Close Modal'
                }
            ], closeButton: true, modalTitle: "Basic Modal", name: "basicModal", onClose: () => { }, open: this.show }, "This is a simple modal with default configuration."), this.show ? "shown" : "hidden"));
    }
    static get is() { return "dga-modal-usage"; }
    static get states() {
        return {
            "show": {}
        };
    }
}
