// @platformscode/core/dist/collection/usage/dga-collapse-usage.js
import { h } from "@stencil/core";
export class DgaCollapseUsage {
    constructor() {
        this.variant = 'neutral';
        this.open = false;
    }
    render() {
        return (h("div", { key: 'e3bee751f056ea024d448e6d3d6de84648ec0f87' }, h("h1", { key: 'aabf8052cb17f6933e3e4ba6f90bf0f9aaddb718' }, "Collapse Example"), h("div", { key: '4befebde255ad9a58e6cbc63b02876b1e778b26e', style: { padding: '50px', display: 'flex', gap: '20px' } }, h("div", { key: 'd83ee2ed4d85b6febc0a8f0bc8979df674c7909c', style: { display: 'flex', flexDirection: 'column', gap: '20px' } }, h("div", { key: 'e718b88c384153ab0f5affe656e442677a12e4f6', style: { display: "flex", justifyContent: "center" } }, h("button", { key: '8f1ed1190335950e939a6b1dfb3a8c114d5c2ab2', onClick: () => this.open = true }, "open"), h("button", { key: '8ddc2e84fc3ffe2ce1c07da88edf4adee22d215f', onClick: () => this.open = false }, "close"))), h("dga-collapse", { key: 'a48c933c61c2ceb142ffd6ad7c36e229f14825cb', open: this.open, "transition-duration": "0.1s" }, h("div", { key: '49ec3f0b4def0c80d3668e04d2ed35c53a73ca92' }, "Your content here")), h("div", { key: '18da42c79de9fa051dcfef9796358796c10c31f3', style: { display: 'flex', flexDirection: 'column', gap: '20px' } }))));
    }
    static get is() { return "dga-collapse-usage"; }
    static get states() {
        return {
            "variant": {},
            "open": {}
        };
    }
}
