// @platformscode/core/dist/collection/usage/dga-list-usage.js
import { h } from "@stencil/core";
// const orderedItems = [
//   {
//     type: 'ordered' as const,
//     level: 'one' as const,
//     label: 'First main point',
//     itemNumber: '1-'
//   },
//   {
//     type: 'ordered' as const,
//     level: 'two' as const,
//     label: 'Sub point A',
//     itemLetter: 'a-'
//   },
//   {
//     type: 'ordered' as const,
//     level: 'two' as const,
//     label: 'Sub point B',
//     itemLetter: 'b-'
//   },
//   {
//     type: 'ordered' as const,
//     level: 'one' as const,
//     label: 'Second main point',
//     itemNumber: '2-'
//   }
// ];
// const orderedItems = [
//   {
//     type: 'ordered' as const,
//     level: 'one' as const,
//     label: 'First main point',
//     itemNumber: '1-'
//   },
//   {
//     type: 'ordered' as const,
//     level: 'two' as const,
//     label: 'Sub point A',
//     itemLetter: 'a-'
//   },
//   {
//     type: 'ordered' as const,
//     level: 'two' as const,
//     label: 'Sub point B',
//     itemLetter: 'b-'
//   },
//   {
//     type: 'ordered' as const,
//     level: 'one' as const,
//     label: 'Second main point',
//     itemNumber: '2-'
//   }
// ];
// Example 2: Unordered List with Multiple Levels
// const unorderedItems = [
//   {
//     type: 'unordered' as const,
//     level: 'one' as const,
//     label: 'Frontend Technologies'
//   },
//   {
//     type: 'unordered' as const,
//     level: 'two' as const,
//     label: 'React'
//   },
//   {
//     type: 'unordered' as const,
//     level: 'two' as const,
//     label: 'Angular'
//   },
//   {
//     type: 'unordered' as const,
//     level: 'one' as const,
//     label: 'Backend Technologies'
//   },
//   {
//     type: 'unordered' as const,
//     level: 'two' as const,
//     label: 'Node.js'
//   },
//   {
//     type: 'unordered' as const,
//     level: 'two' as const,
//     label: 'Python'
//   }
// ];
// const iconItems = [
//   {
//     type: 'with-icon' as const,
//     level: 'one' as const,
//     label: 'Critical Issue',
//     icon: '<i class="hgi-stroke hgi-checkmark-circle-02"></i>'
//   },
//   {
//     type: 'with-icon' as const,
//     level: 'two' as const,
//     label: 'Database connection failed',
//     icon: '<i class="hgi-stroke hgi-checkmark-circle-02"></i>'
//   },
//   {
//     type: 'with-icon' as const,
//     level: 'one' as const,
//     label: 'Success',
//     icon: '<i class="hgi-stroke hgi-checkmark-circle-02"></i>'
//   },
//   {
//     type: 'with-icon' as const,
//     level: 'two' as const,
//     label: 'Backup completed',
//     icon: '<i class="hgi-stroke hgi-checkmark-circle-02"></i>'
//   }
// ];
export class DgaListUsage {
    render() {
        return (h("div", { key: '333055f1445a491712e754528544eefcb1124cca' }, h("h1", { key: '55fc6ed6c9bfc52d31292754caa835704f7ae84e' }, "List Example"), h("div", { key: '200da3bea75948a48774fc267f0a9abd8c5519e1', style: { padding: '50px', display: 'flex', gap: '20px' } }, h("div", { key: 'def847a3f277b0d375731322d37e157ebf45baa2', style: { display: 'flex', flexDirection: 'column', gap: '20px' } }, h("dga-list-v2", { key: '5abfa0eecf96cbc497d6cb8f65c3bfb75e71cc4c', type: 'ordered', color: 'neutral' }, h("dga-list-item", { key: '881d17816f0f7c5702620d6780770615e0e891b8', type: "ordered", level: 'one', itemNumber: '1-', itemText: 'List item' }), h("dga-list-item", { key: 'b27d47bef1319b58ba0efeae7db2dc67b1dd13ec', type: "ordered", level: 'two', itemLetter: 'a-', itemText: 'List item' }), h("dga-list-item", { key: '1af4e569e6cc8cba15279e9a563033b89cd726a1', type: "ordered", level: 'two', itemLetter: 'b-', itemText: 'List item' }), h("dga-list-item", { key: 'c22fc6afc19adf3196946afa84b53b94cbe70679', type: "ordered", level: 'one', itemNumber: '2-', itemText: 'List item' }), h("dga-list-item", { key: '2622b17ef4d6b551e69fd553ed04ee20d5341f19', type: "ordered", level: 'two', itemLetter: 'a-', itemText: 'List item' })), h("dga-list-v2", { key: '6589813c6ef152ac4d2bbbeebfca593635f049d4', type: 'unordered', color: 'neutral' }, h("dga-list-item", { key: 'ff1fc6b39742f15904194497804eca755a974ea4', type: "unordered", level: 'one', itemText: 'List item' }), h("dga-list-item", { key: 'a3216f0c92a10572435b79bad45c92a2b5370cb4', type: "unordered", level: 'two', itemText: 'List item' }), h("dga-list-item", { key: '8eed6a32e3ba3ed88c794d5a2219cecf1c39ab2f', type: "unordered", level: 'two', itemText: 'List item' })), h("dga-list-v2", { key: '9e22e64ae91a952dba0cf8c6906f38bf4e54d104', type: 'with-icon', color: 'neutral' }, h("dga-list-item", { key: 'fbb8df316254deb99fbe877c2e9f58e21d45d71c', type: "with-icon", level: 'one', icon: '<i  class="hgi-stroke hgi-checkmark-circle-02"></i>', itemText: 'List item' }), h("dga-list-item", { key: '8320d065049e7849bf3ee7570fc3edee693fddeb', type: "with-icon", level: 'two', icon: '<i  class="hgi-stroke hgi-checkmark-circle-02"></i>', itemText: 'List item' }), h("dga-list-item", { key: '56fc2bbf243d3ea3e69650eed85d503b681b365f', type: "with-icon", level: 'two', icon: '<i  class="hgi-stroke hgi-checkmark-circle-02"></i>', itemText: 'List item' }))))));
    }
    static get is() { return "dga-list-usage"; }
}
