// @platformscode/core/dist/collection/usage/dga-table-of-content-usage.js
import { h } from "@stencil/core";
export class DgaTableOfContentUsage {
    render() {
        return (h("div", { key: 'f365703736fe68f2bf1b268031025c6637222337', style: {
                position: 'relative',
                width: '100%',
                maxWidth: '1200px',
                margin: '0 auto',
                padding: '20px'
            } }, h("h1", { key: '19a7fc78fd270765feef57f18625fc4fcab9683e' }, "Table of Content Example"), h("section", { key: '238c99fd01706a9e87bfb09aa829030e662051a9' }, h("div", { key: 'b1aa7090a7d4272a9ae718fe8eb631adf2b7a7ed', id: "section_1", style: {
                minHeight: "100vh",
                backgroundColor: "gray",
                padding: "20px",
                margin: "20px 0 200px 0"
            } }, h("h2", { key: 'e0e3d8ee6aad27c62d86115316b8d9a5a2172db2' }, "Section 1"), h("div", { key: '2c5f6932b71c4944dd1ba964dcd388c7f35d330f', id: "section_1_child_1", style: {
                minHeight: "100vh",
                backgroundColor: "rgba(255,0,0,0.3)",
                margin: "20px 0",
                padding: "20px"
            } }, h("h3", { key: '868615fb583491d2164a0758631222aa2a6a7b75' }, "Section 1 Child 1")), h("div", { key: '877c8c2eac6126113b5fed21a8c4ffe51cae11ed', id: "section_1_child_2", style: {
                minHeight: "100vh",
                backgroundColor: "rgba(0,0,255,0.3)",
                margin: "20px 0",
                padding: "20px"
            } }, h("h3", { key: 'd01ec3b25413f687495a59eee46e0f95808a6e52' }, "Section 1 Child 2")), h("div", { key: '860c1415bd996da440ba88ed771eb199b1aefc82', id: "section_1_child_3", style: {
                minHeight: "100vh",
                backgroundColor: "rgba(0,255,0,0.3)",
                margin: "20px 0",
                padding: "20px"
            } }, h("h3", { key: '94720f9a03a3ea2901ddf1e85b988c43b5a9b0f5' }, "Section 1 Child 3"), h("div", { key: '6fcfa0fdd98cc018dafc83732ee0202f923bb340', id: "section_1_sub_child_1", style: {
                minHeight: "50vh",
                backgroundColor: "rgba(255,255,255,0.3)",
                margin: "20px 0",
                padding: "20px"
            } }, h("h4", { key: '29667e61b8ae34643536129b4942ca9f7945e229' }, "Section 1 Sub Child 1")), h("div", { key: '3e7ad4dc25169e094c8270a6a06d31b772fcc722', id: "section_1_sub_child_2", style: {
                minHeight: "50vh",
                backgroundColor: "rgba(255,255,255,0.3)",
                margin: "20px 0",
                padding: "20px"
            } }, h("h4", { key: '01ceae69186064eef3576f4e21da317b00ae5049' }, "Section 1 Sub Child 2")), h("div", { key: '6dc61bd8ef6e6390f0633dd5213246b3173447c7', id: "section_1_sub_child_3", style: {
                minHeight: "50vh",
                backgroundColor: "rgba(255,255,255,0.3)",
                margin: "20px 0",
                padding: "20px"
            } }, h("h4", { key: '8a84db32e43d8db3e560fff49e0b09543eb10fed' }, "Section 1 Sub Child 3")))), h("div", { key: 'cdbe8e6375363b6db0b6ef27572aad7d58e823a4', id: "section_2", style: {
                minHeight: "100vh",
                backgroundColor: "rgba(0,0,0,0.8)",
                color: "white",
                padding: "20px",
                margin: "20px 0"
            } }, h("h2", { key: 'd5be890457494b8825a99a60a1e9a4bce7d107be' }, "Section 2")), h("div", { key: 'fb89e6916322e1bfaf3e66e9cbb1bee602062240', id: "section_3", style: {
                minHeight: "100vh",
                backgroundColor: "brown",
                padding: "20px",
                margin: "20px 0"
            } }, h("h2", { key: '6a0474acb65b42530844d3d68aac768827f20311' }, "Section 3"), h("div", { key: 'f12ae40d737214b6f8b6cd35311fccbad2f29065', id: "section_3_child_1", style: {
                minHeight: "50vh",
                backgroundColor: "rgba(255,255,255,0.3)",
                margin: "20px 0",
                padding: "20px"
            } }, h("h3", { key: 'f6a9812f6bd2225f6dd82d6f5ddb7d92263c40e5' }, "Section 3 Child 1")), h("div", { key: '8cf241d306e8d1880778ff3b27ceeb0ea0569af6', id: "section_3_child_2", style: {
                minHeight: "50vh",
                backgroundColor: "rgba(255,255,255,0.3)",
                margin: "20px 0",
                padding: "20px"
            } }, h("h3", { key: 'a2071569efb86b10e6b9f18607f1afb8cadcf043' }, "Section 3 Child 2"), h("div", { key: '6fe13f4a533a82eaa94e34ce20b6a86556558b3e', id: "section_3_child_2_1", style: {
                minHeight: "30vh",
                backgroundColor: "rgba(0,0,0,0.1)",
                margin: "20px 0",
                padding: "20px"
            } }, h("h4", { key: 'ccef8abc44c5317d996d2f1230bdb97f8c68945f' }, "Section 3 Child 2.1")), h("div", { key: '2defa94be8314486a7f133137b23908cd8fd0e1e', id: "section_3_child_2_2", style: {
                minHeight: "30vh",
                backgroundColor: "rgba(0,0,0,0.1)",
                margin: "20px 0",
                padding: "20px"
            } }, h("h4", { key: 'b03745ce0d88eefb92286039f3c15c7c017ee656' }, "Section 3 Child 2.2")), h("div", { key: '2f29a9acd13c7839f0141901190a2fa2ade9f39b', id: "section_3_child_2_3", style: {
                minHeight: "30vh",
                backgroundColor: "rgba(0,0,0,0.1)",
                margin: "20px 0",
                padding: "20px"
            } }, h("h4", { key: 'f9d473305d4595d25e0687b03f0cd45b5fcea0fe' }, "Section 3 Child 2.3"))))), h("div", { key: '9be479c676d7c9cb272ac15c00daf3007da967bd', style: {
                position: "fixed",
                top: "100px",
                left: "40px",
                zIndex: '100',
                background: "white",
                borderRadius: "8px",
                boxShadow: "0 2px 8px rgba(0, 0, 0, 0.1)",
            } }, h("dga-table-of-content", { key: '006c81d35ca96f58494982d7e008c3bcc22fe792', title: "On this Page", subTitle: "Page Title", sections: [
                {
                    Name: 'Section 1',
                    Target: 'section_1',
                    Children: [
                        { Name: 'Section 1 Child 1', Target: 'section_1_child_1' },
                        { Name: 'Section 1 Child 2', Target: 'section_1_child_2' },
                        {
                            Name: 'Section 1 Child 3',
                            Target: 'section_1_child_3',
                            Children: [
                                { Name: 'Sub Child 1', Target: 'section_1_sub_child_1' },
                                { Name: 'Sub Child 2', Target: 'section_1_sub_child_2' },
                                { Name: 'Sub Child 3', Target: 'section_1_sub_child_3' }
                            ]
                        }
                    ]
                },
                { Name: 'Section 2', Target: 'section_2' },
                {
                    Name: 'Section 3',
                    Target: 'section_3',
                    Children: [
                        { Name: 'Section 3 Child 1', Target: 'section_3_child_1' },
                        {
                            Name: 'Section 3 Child 2',
                            Target: 'section_3_child_2',
                            Children: [
                                { Name: 'Child 2.1', Target: 'section_3_child_2_1' },
                                { Name: 'Child 2.2', Target: 'section_3_child_2_2' },
                                { Name: 'Child 2.3', Target: 'section_3_child_2_3' }
                            ]
                        }
                    ]
                }
            ] })), h("div", { key: 'dea17559fcb60eb120f6b6ab835b8e2f101a5f23', style: {
                position: "fixed",
                top: "100px",
                right: "40px",
                zIndex: '100',
                background: "white",
                borderRadius: "8px",
                boxShadow: "0 2px 8px rgba(0, 0, 0, 0.1)",
            } }, h("dga-table-of-content", { key: '3ecb4eb4ec7ae3bf767df4cadeb20edc43277635', title: "\u0641\u064A \u0647\u0630\u0647 \u0627\u0644\u0635\u0641\u062D\u0629", subTitle: "[\u0627\u0633\u0645 \u0627\u0644\u0635\u0641\u062D\u0629]", rtl: true, sections: [
                {
                    Name: 'قسم صفحة',
                    Target: 'section_1',
                    Children: [
                        { Name: 'قسم صفحة', Target: 'section_1_child_1' },
                        { Name: 'قسم صفحة', Target: 'section_1_child_2' },
                        {
                            Name: 'قسم صفحة',
                            Target: 'section_1_child_3',
                            Children: [
                                { Name: 'قسم صفحة', Target: 'section_1_sub_child_1' },
                                { Name: 'قسم صفحة', Target: 'section_1_sub_child_2' },
                                { Name: 'قسم صفحة', Target: 'section_1_sub_child_3' }
                            ]
                        }
                    ]
                },
                { Name: 'Section 2', Target: 'section_2' },
                {
                    Name: 'Section 3',
                    Target: 'section_3',
                    Children: [
                        { Name: 'Section 3 Child 1', Target: 'section_3_child_1' },
                        {
                            Name: 'Section 3 Child 2',
                            Target: 'section_3_child_2',
                            Children: [
                                { Name: 'Child 2.1', Target: 'section_3_child_2_1' },
                                { Name: 'Child 2.2', Target: 'section_3_child_2_2' },
                                { Name: 'Child 2.3', Target: 'section_3_child_2_3' }
                            ]
                        }
                    ]
                }
            ] }))));
    }
    static get is() { return "dga-table-of-content-usage"; }
}
