// @platformscode/core/dist/collection/usage/dga-grid-usage.js
// import { Component, h } from '@stencil/core';
// @Component({
//   tag: 'dga-grid-usage',
//   styleUrl: 'dga-grid-usage.scss',
// })
// export class DgaGridUsage {
//   render() {
//     return (
//       <div class="filteration-usage" style={{
//         display: 'flex',
//         flexDirection: 'column',
//         gap: '20px',
//       }}>
//         <h2>Filteration Component Examples</h2>
//         {/* <!-- Basic Grid Layout Example --> */}
//         <dga-grid-container spacing={2}>
//           <dga-grid-item xs={12} sm={6} md={4}>
//             <div class="demo-content">Item 1</div>
//           </dga-grid-item>
//           <dga-grid-item xs={12} sm={6} md={4}>
//             <div class="demo-content">Item 2</div>
//           </dga-grid-item>
//           <dga-grid-item xs={12} sm={6} md={4}>
//             <div class="demo-content">Item 3</div>
//           </dga-grid-item>
//         </dga-grid-container>
//         {/* <!-- Responsive Grid with Different Column Widths --> */}
//         <dga-grid-container spacing={3}>
//           <dga-grid-item xs={12} sm={6} md={4} lg={3}>
//             <div class="demo-content">Full width on mobile, half width on tablet, 1/3 on laptop, 1/4 on desktop</div>
//           </dga-grid-item>
//           <dga-grid-item xs={12} sm={6} md={4} lg={3}>
//             <div class="demo-content">Responsive Item 2</div>
//           </dga-grid-item>
//           <dga-grid-item xs={12} sm={6} md={4} lg={3}>
//             <div class="demo-content">Responsive Item 3</div>
//           </dga-grid-item>
//           <dga-grid-item xs={12} sm={6} md={12} lg={3}>
//             <div class="demo-content">Responsive Item 4</div>
//           </dga-grid-item>
//         </dga-grid-container>
//         {/* <!-- Different Spacing at Different Breakpoints --> */}
//         <dga-grid-container spacingXs={1} spacingSm={2} spacingMd={3} spacingLg={4}>
//           <dga-grid-item xs={6}>
//             <div class="demo-content">Different spacing at different screen sizes</div>
//           </dga-grid-item>
//           <dga-grid-item xs={6}>
//             <div class="demo-content">Different spacing at different screen sizes</div>
//           </dga-grid-item>
//         </dga-grid-container>
//         {/* <!-- Auto-layout Grid --> */}
//         <dga-grid-container spacing={2}>
//           <dga-grid-item xs>
//             <div class="demo-content">Auto width 1</div>
//           </dga-grid-item>
//           <dga-grid-item xs>
//             <div class="demo-content">Auto width 2</div>
//           </dga-grid-item>
//           <dga-grid-item xs>
//             <div class="demo-content">Auto width 3</div>
//           </dga-grid-item>
//         </dga-grid-container>
//         {/* <!-- Nested Grids --> */}
//         <dga-grid-container spacing={2}>
//           <dga-grid-item xs={12} sm={6}>
//             <div class="demo-content">
//               <h3>Nested Grid</h3>
//               <dga-grid-container spacing={1}>
//                 <dga-grid-item xs={6}>
//                   <div class="demo-content nested">Nested item 1</div>
//                 </dga-grid-item>
//                 <dga-grid-item xs={6}>
//                   <div class="demo-content nested">Nested item 2</div>
//                 </dga-grid-item>
//               </dga-grid-container>
//             </div>
//           </dga-grid-item>
//           <dga-grid-item xs={12} sm={6}>
//             <div class="demo-content">Regular item</div>
//           </dga-grid-item>
//         </dga-grid-container>
//         {/* <!-- Item Alignment Example --> */}
//         <dga-grid-container spacing={2} style={{height: "200px"}}>
//           <dga-grid-item xs={4} alignSelf="flex-start">
//             <div class="demo-content">Align start</div>
//           </dga-grid-item>
//           <dga-grid-item xs={4} alignSelf="center">
//             <div class="demo-content">Align center</div>
//           </dga-grid-item>
//           <dga-grid-item xs={4} alignSelf="flex-end">
//             <div class="demo-content">Align end</div>
//           </dga-grid-item>
//         </dga-grid-container>
//         {/* <!-- Growing and Shrinking Items --> */}
//         <dga-grid-container spacing={2}>
//           <dga-grid-item xs={3}>
//             <div class="demo-content">Fixed width</div>
//           </dga-grid-item>
//           <dga-grid-item xs grow>
//             <div class="demo-content">Grow to fill space</div>
//           </dga-grid-item>
//           <dga-grid-item xs={3}>
//             <div class="demo-content">Fixed width</div>
//           </dga-grid-item>
//         </dga-grid-container>
//         {/* <!-- Item Ordering Example --> */}
//         <dga-grid-container spacing={2}>
//           <dga-grid-item xs={4} orderXs={3} orderMd={1}>
//             <div class="demo-content">First on desktop, last on mobile</div>
//           </dga-grid-item>
//           <dga-grid-item xs={4} orderXs={1} orderMd={2}>
//             <div class="demo-content">Second on desktop, first on mobile</div>
//           </dga-grid-item>
//           <dga-grid-item xs={4} orderXs={2} orderMd={3}>
//             <div class="demo-content">Third on desktop, second on mobile</div>
//           </dga-grid-item>
//         </dga-grid-container>
//         {/* <!-- Fluid Container with No Gutters --> */}
//         <dga-grid-container fluid disableGutters spacing={0}>
//           <dga-grid-item xs={12} md={6}>
//             <div class="demo-content">Full-width fluid container with no gutters</div>
//           </dga-grid-item>
//           <dga-grid-item xs={12} md={6}>
//             <div class="demo-content">Full-width fluid container with no gutters</div>
//           </dga-grid-item>
//         </dga-grid-container>
//       </div>
//     );
//   }
// }
import { h } from "@stencil/core";
export class DgaGridUsage {
    render() {
        return (h("div", { key: 'db2939f8ea1477199e484555884da44f90d37bcf', class: "flexible-gap-usage", style: {
                display: 'flex',
                flexDirection: 'column',
                gap: '20px',
            } }, h("h2", { key: 'a024312c7347789f310c5b05453e2d72fb1d030e' }, "Grid Component with Flexible Gap Examples"), h("section", { key: 'fea4b42f867ebdf28297075ad81e0c37a73f35cc' }, h("h3", { key: 'e4506b676a25588988df8afdfbca81b3556c77b3' }, "Simple String Gap"), h("dga-grid-container", { key: '0087631dbceb9fd6f967d33e39edba5fc4a1f07e', gap: "16px" }, h("dga-grid-item", { key: 'e90f5ae2f065163b94b53f61abbfffad3c734bc7', xs: 12, sm: 6, md: 4 }, h("div", { key: '555a1c9e2b37671a8ea0b3bdc1dcec403a2bbdda', class: "demo-content" }, "16px gap between all items")), h("dga-grid-item", { key: 'c4df5d9b52f7853ef8b81532c34282bd3bd9e068', xs: 12, sm: 6, md: 4 }, h("div", { key: '6fd7595ab49001ca82784ecb64ac36dc154975d2', class: "demo-content" }, "16px gap between all items")), h("dga-grid-item", { key: '134aec9d2128c77e47a4dedf6347be1c91880798', xs: 12, sm: 6, md: 4 }, h("div", { key: 'e32c42e516896165fbfe60ef7df4aab3630c0788', class: "demo-content" }, "16px gap between all items")), h("dga-grid-item", { key: 'fd24328a33b4c930389dd099c53be213593a2462', xs: 12, sm: 6 }, h("div", { key: '612f89984c1c9e8d4ccafddfc4b5717f95ea5d73', class: "demo-content" }, "16px gap between all items")), h("dga-grid-item", { key: 'd2555b16e49ad61a02ab2008265d108b1b2be698', xs: 12, sm: 6 }, h("div", { key: 'd74ed851afec55d37329c44c8886eba45634a50a', class: "demo-content" }, "16px gap between all items")), h("dga-grid-item", { key: 'd067ba3d86e9d2580214427642d34a7e6a2ee547', xs: 12 }, h("div", { key: 'b4ee61c26875861e6281cc1d2b429bcb2a2d6bce', class: "demo-content" }, "16px gap between all items"))))));
    }
    static get is() { return "dga-grid-usage"; }
    static get originalStyleUrls() {
        return {
            "$": ["dga-grid-usage.scss"]
        };
    }
    static get styleUrls() {
        return {
            "$": ["dga-grid-usage.css"]
        };
    }
}
