// @platformscode/core/dist/collection/usage/dga-card-usage.js
import { h } from "@stencil/core";
export class DgaCardUsage {
    constructor() {
        this.variant = 'neutral';
        this.selected = false;
    }
    render() {
        return (h("div", { key: '6a7a1b6e441772c76742ea6b571f95b088d5f03f' }, h("h1", { key: '3caae8983f0a589d090eb3847aa8deee721bc748' }, "Card Example"), h("div", { key: '63e25e6cd62416548d5e110b67c3fe4633194470', style: { padding: '50px', display: 'flex', gap: '20px' } }, h("div", { key: '5bd873a8dac0c0a9f80ac70fe7021fe7ce5748db', style: { display: 'flex', flexDirection: 'column', gap: '20px' } }, h("dga-card", { key: '5289afbd10c9734bab1658dd24ded2df90312cbf', type: "expanded", cardTitle: "Card Title", description: "Card content placeholder text goes here", showFeaturedIcon: true }, h("div", { key: 'e415bef8be7707b2a269e9e87b8ab6fa00a1b2ad', slot: "collapse-content" }, "dgdsgdfgdfg")), h("dga-card", { key: '8c55e101dd0a4279c2d7582021a519483c2b133d', type: "selectable", cardTitle: "Card Title", description: "Card content placeholder text goes here", isSelected: this.selected, onCheckboxChange: () => (this.selected = !this.selected) }), h("dga-card", { key: '730606135cce7a68f305d62365e6ff41288f97a2', cardTitle: "Card Title", sx: {
                '& > .card': {
                    backgroundColor: 'red',
                    height: '400px',
                },
            } }), h("dga-card", { key: 'c75b0d0d15ad21eb5dccd5784a5ee67ed0604706', cardTitle: "No Shadow", effect: "no-shadow" }), h("dga-card", { key: 'a0d79d3f779669bd7417e196f02ac7f20b116fc1', cardTitle: "Stroke", effect: "stroke" }), h("dga-card", { key: 'd451d77e46ee7a513f6cbfd8a5b40315383493aa', cardTitle: "With Shadow", effect: "with-shadow" })), h("div", { key: '960fa71feb652f88ba897de219a664b16cbacd32', style: { display: 'flex', flexDirection: 'column', gap: '20px' } }))));
    }
    static get is() { return "dga-card-usage"; }
    static get states() {
        return {
            "variant": {},
            "selected": {}
        };
    }
}
