// @platformscode/core/dist/collection/usage/dga-notification-toast-usage.js
import { h } from "@stencil/core";
export class DgaNotificationToastUsage {
    constructor() {
        this.type = 'success';
        this.open = false;
    }
    render() {
        return (h("div", { key: '56e0f0bf8e4b0adf120ff0d869cdda544cdef401' }, h("h1", { key: '79317ee62ad38058ca4012aa6ffbec8ddbf2f5d4' }, "Notification Toast Example"), h("div", { key: 'a81c0bcad6dc89f42527b614fea0c780e4a748f3', style: { padding: '50px', display: 'flex', gap: '20px' } }, h("div", { key: '574eb746ddf78d37d760aa8670a8a9eb53334db2', style: { display: 'flex', flexDirection: 'column', gap: '20px' } }, h("div", { key: '2c9d26c302f25bf03f843cda3031dc5e91e62232', style: { display: "flex", justifyContent: "center" } }, h("button", { key: 'b0e38267fa51e8e929642fa7ea1e90ea2c8fe34b', onClick: () => this.open = true }, "open"), h("button", { key: '776a37bfd3d2442059289dc0a16fd0a1942f822b', onClick: () => this.open = false }, "close")), h("dga-notification-toast", { key: '029a110d4a944a92bfeb480a24687211ceeb4129', type: this.type, open: this.open, leadText: "Notification/Alert message title", hasHelperText: true, helperText: "When a Notification/Alert needs a further detailed explanation, it goes here.", action: true, buttonsList: [
                {
                    label: 'Button',
                    onClick: () => { },
                    title: 'Button 2',
                },
                {
                    label: 'Button',
                    onClick: () => { },
                    title: 'Button',
                },
            ], closeButton: true, vPostion: 'top', hPostion: 'left', onClose: () => this.open = false }), h("dga-notification-toast", { key: '23a3e770b81f7c12a448df153ac89599b1d4092b', type: this.type, open: this.open, leadText: "Notification/Alert message title", hasHelperText: true, helperText: "When a Notification/Alert needs a further detailed explanation, it goes here.", action: true, buttonsList: [
                {
                    label: 'Button',
                    onClick: () => { },
                    title: 'Button 2',
                },
                {
                    label: 'Button',
                    onClick: () => { },
                    title: 'Button',
                },
            ], closeButton: true, vPostion: 'bottom', hPostion: 'left', onClose: () => this.open = false })), h("div", { key: '79bbebda27aa2e4e94e34c1989b6179ef0502f3e', style: { display: 'flex', flexDirection: 'column', gap: '20px' } }))));
    }
    static get is() { return "dga-notification-toast-usage"; }
    static get states() {
        return {
            "type": {},
            "open": {}
        };
    }
}
