// @platformscode/core/dist/collection/usage/dga-checkbox-usage.js
import { h } from "@stencil/core";
export class DgaCheckboxUsage {
    constructor() {
        this.variant = 'neutral';
    }
    render() {
        return (h("div", { key: 'c4e540214198e65b5f432db3c5bbc8158c81dd06' }, h("h1", { key: '338bb151722f1f32b745782ccf3ae74af30fe79a' }, "Checkbox Example"), h("div", { key: 'cda273202f63609b6ba5309fc72eec20ab2ca924', style: { padding: '50px', display: 'flex', gap: '20px' } }, h("div", { key: 'a860342b975c3efe2ad9bb82351160baf52ae1ba', style: { display: 'flex', flexDirection: 'column', gap: '20px' } }, h("h1", { key: '6986b01315fefac2b69e9decc528ee6ca1182e6c' }, "test"), h("dga-checkbox", { key: '7de3fccd35c2d7c14243ae82adb8458c6d3832d8', label: 'Checkbox Label Checkbox Label Checkbox Label Checkbox Label', Indeterminate: false, alertMessage: true, alertText: "Error/Warning message" }), h("dga-checkbox", { key: 'da2dbc3ced0bd9991370a40d80b589d14cc9a80f', size: 'lg' }), h("dga-checkbox", { key: 'a71a5d2e0c963996cfbd4a3d74e66a131651af4a', size: 'md' }), h("dga-checkbox", { key: 'e360f109e741920d24ceffdb8e939877cc4f0523', size: 'sm' })), h("div", { key: 'ad09ab4003870d5b6af75e4c3bf8868bd8748d1f', style: { display: 'flex', flexDirection: 'column', gap: '20px' } }))));
    }
    static get is() { return "dga-checkbox-usage"; }
    static get states() {
        return {
            "variant": {}
        };
    }
}
