// @platformscode/core/dist/collection/usage/dga-dropdown-usage.js
import { h } from "@stencil/core";
export class DgaDropdownUsage {
    constructor() {
        this.activeStep = 1;
        this.selectedValue = 'Option-1';
    }
    render() {
        return (h("div", { key: '1882688c6b9c91983903879d826e3358a219e5a4' }, h("h1", { key: 'd9d2931815da214c1caa5a476c72220abcda12c6' }, "Dropdown Example"), h("div", { key: 'a35f90af0490a27a3b7303bcfdb9b0711c5b2008', style: { padding: '50px', display: 'flex', gap: '20px' } }, h("div", { key: '126c0ba46a33cb8e927abf88724ef95ac00063c5', style: { display: 'flex', flexDirection: 'column', gap: '20px' } }, h("dga-dropdown", { key: '26952570c0256eb9c030b85f8c8893755e8a213b', placeholder: "Placeholder text", variant: "default", optionLabel: "label", trackBy: "value", value: this.selectedValue, options: [
                {
                    label: 'Netherlands Antiles',
                    value: 'Option-1',
                },
                {
                    label: 'Option 2',
                    value: 'Option-2',
                },
                {
                    label: 'Option 3',
                    value: 'Option-3',
                },
            ], onChange: (value) => this.selectedValue = value }), this.selectedValue, h("dga-dropdown", { key: '597849e9b5a665d3b6c3ecbd2c210d59d689826d', placeholder: "Placeholder text", variant: "darker", optionLabel: "label", trackBy: "value", options: [
                {
                    label: 'Option',
                    value: 'Option',
                },
                {
                    label: 'Option',
                    value: 'Option',
                },
                {
                    label: 'Option Optionbbb',
                    value: 'Option',
                },
                {
                    label: 'Option',
                    value: 'Option',
                },
            ] }), h("dga-dropdown", { key: 'a4923bbb563a37f562a6814fd373a4d51851e057', placeholder: "Placeholder text", variant: "lighter", optionLabel: "label", trackBy: "value", options: [
                {
                    label: 'Option',
                    value: 'Option',
                },
                {
                    label: 'Option',
                    value: 'Option',
                },
                {
                    label: 'Option',
                    value: 'Option',
                },
                {
                    label: 'Option',
                    value: 'Option',
                },
            ] }), h("dga-dropdown", { key: 'fb13b75be207248ea647f5abc65462a006355905', error: true, placeholder: "Placeholder text", variant: "lighter", optionLabel: "label", trackBy: "value", options: [
                {
                    label: 'Option',
                    value: 'Option',
                },
                {
                    label: 'Option',
                    value: 'Option',
                },
                {
                    label: 'Option',
                    value: 'Option',
                },
                {
                    label: 'Option',
                    value: 'Option',
                },
            ] }), ['lg', 'md'].map((size) => (h("dga-dropdown", { key: size, size: size, placeholder: "Placeholder text", variant: "default", optionLabel: "label", trackBy: "value", options: [
                {
                    label: 'Option',
                    value: 'Option',
                },
                {
                    label: 'Option',
                    value: 'Option',
                },
                {
                    label: 'Option',
                    value: 'Option',
                },
                {
                    label: 'Option',
                    value: 'Option',
                },
            ] }))), h("dga-dropdown", { key: 'fc62c523c87dc2ace95f9059c6eacac9eb42a823', multiSelect: true, placeholder: "Placeholder text", variant: "lighter", optionLabel: "label", trackBy: "value", options: [
                {
                    label: 'Option',
                    value: 'Option',
                },
                {
                    label: 'Option',
                    value: 'Option',
                },
                {
                    label: 'Option',
                    value: 'Option',
                },
                {
                    label: 'Option',
                    value: 'Option',
                },
            ] })), h("div", { key: '2a3284bbeacb949680d4971d49916ef5327c6974', style: { display: 'flex', flexDirection: 'column', gap: '20px' } }))));
    }
    static get is() { return "dga-dropdown-usage"; }
    static get states() {
        return {
            "activeStep": {},
            "selectedValue": {}
        };
    }
}
