// @platformscode/core/dist/cjs/dga-dropdown-usage.cjs.entry.js
'use strict';

var index = require('./index-DSiE2De_.js');

const DgaDropdownUsage = class {
    constructor(hostRef) {
        index.registerInstance(this, hostRef);
        this.activeStep = 1;
        this.selectedValue = 'Option-1';
    }
    render() {
        return (index.h("div", { key: '0a6b57e348ddc9ce5a0964865d5d2af5603f0cb8' }, index.h("h1", { key: 'df207a773226becdc93f531f2b1499fa61566af4' }, "Dropdown Example"), index.h("div", { key: '00d703c374e5b3727dc5cfaa984984d3f23917b5', style: { padding: '50px', display: 'flex', gap: '20px' } }, index.h("div", { key: '8e257d0f1e1ee8b6522d231f54a9b6abf9151931', style: { display: 'flex', flexDirection: 'column', gap: '20px' } }, index.h("dga-dropdown", { key: '53fc7829cf7bccc4013cf8110ba6a9c2441ff848', placeholder: "Placeholder text", variant: "default", optionLabel: "label", trackBy: "value", value: this.selectedValue, options: [
                {
                    label: 'Netherlands Antiles',
                    value: 'Option-1',
                },
                {
                    label: 'Option 2',
                    value: 'Option-2',
                },
                {
                    label: 'Option 3',
                    value: 'Option-3',
                },
            ], onChange: (value) => this.selectedValue = value }), this.selectedValue, index.h("dga-dropdown", { key: 'e12ad71e46991a1adc6bc48c9f7e9016f9802dba', placeholder: "Placeholder text", variant: "darker", optionLabel: "label", trackBy: "value", options: [
                {
                    label: 'Option',
                    value: 'Option',
                },
                {
                    label: 'Option',
                    value: 'Option',
                },
                {
                    label: 'Option Optionbbb',
                    value: 'Option',
                },
                {
                    label: 'Option',
                    value: 'Option',
                },
            ] }), index.h("dga-dropdown", { key: '6b25aa37571846e3a9eab024d8210714908c3a77', placeholder: "Placeholder text", variant: "lighter", optionLabel: "label", trackBy: "value", options: [
                {
                    label: 'Option',
                    value: 'Option',
                },
                {
                    label: 'Option',
                    value: 'Option',
                },
                {
                    label: 'Option',
                    value: 'Option',
                },
                {
                    label: 'Option',
                    value: 'Option',
                },
            ] }), index.h("dga-dropdown", { key: '1504cdf3f7aeb48c285c00c9515c1881f3f42bfe', error: true, placeholder: "Placeholder text", variant: "lighter", optionLabel: "label", trackBy: "value", options: [
                {
                    label: 'Option',
                    value: 'Option',
                },
                {
                    label: 'Option',
                    value: 'Option',
                },
                {
                    label: 'Option',
                    value: 'Option',
                },
                {
                    label: 'Option',
                    value: 'Option',
                },
            ] }), ['lg', 'md'].map((size) => (index.h("dga-dropdown", { key: size, size: size, placeholder: "Placeholder text", variant: "default", optionLabel: "label", trackBy: "value", options: [
                {
                    label: 'Option',
                    value: 'Option',
                },
                {
                    label: 'Option',
                    value: 'Option',
                },
                {
                    label: 'Option',
                    value: 'Option',
                },
                {
                    label: 'Option',
                    value: 'Option',
                },
            ] }))), index.h("dga-dropdown", { key: '9eec2e8183875e424ebf805cd27b675ecbb6d99b', multiSelect: true, placeholder: "Placeholder text", variant: "lighter", optionLabel: "label", trackBy: "value", options: [
                {
                    label: 'Option',
                    value: 'Option',
                },
                {
                    label: 'Option',
                    value: 'Option',
                },
                {
                    label: 'Option',
                    value: 'Option',
                },
                {
                    label: 'Option',
                    value: 'Option',
                },
            ] })), index.h("div", { key: '7447713413d02ad8b39ed86c303c59ed5def5648', style: { display: 'flex', flexDirection: 'column', gap: '20px' } }))));
    }
};

exports.dga_dropdown_usage = DgaDropdownUsage;
