// @platformscode/core/dist/cjs/dga-carousel-usage.cjs.entry.js
'use strict';

var index = require('./index-DSiE2De_.js');

const DgaCarouselUsage = class {
    constructor(hostRef) {
        index.registerInstance(this, hostRef);
        this.variant = 'neutral';
    }
    render() {
        return (index.h("div", { key: '135bc619c23070b1e730b23411c9699060c30142' }, index.h("h1", { key: '19361805583065becf477b9dde8b4279113d6c63' }, "Carousel Example"), index.h("div", { key: 'fb1053e8e18da8351fa757cdaf9df34a32157241', style: { padding: '50px', display: 'flex', gap: '20px' } }, index.h("div", { key: '988c510cddb672792d3b7ee78ed5f64d69c2073a', style: { display: 'flex', flexDirection: 'column', gap: '20px' } }, index.h("dga-carousel", { key: 'fce2953ef3bb991ac8adfcfbaa25b6ba6864f438', size: 'lg', arrowsVariant: 'primary' }, index.h("dga-carousel-item", { key: 'd30c755eb3dda44425862f75124feccba440698f' }, index.h("img", { key: '4c952b404c49224ecae9fba743a46dcfe81984b9', src: "https://cdn.pixabay.com/photo/2019/11/03/01/56/landscape-4597742_1280.jpg", style: {
                height: '300px',
                width: '450px',
            } })), index.h("dga-carousel-item", { key: '86891484bd67a22ade9c5af424e82ea0b2226507' }, index.h("img", { key: '20f413fe7d84ff0eaffee44636629e0a514df3d5', src: "https://cdn.pixabay.com/photo/2020/07/09/10/31/sea-5386810_1280.jpg", style: {
                height: '300px',
                width: '450px',
                objectFit: 'cover',
            } })), index.h("dga-carousel-item", { key: 'fdc402ece21ffaf9ded84e57f9fe18878b9fc886' }, index.h("img", { key: 'ac9d20b1cd927078ec3341bc890bc5637815d95d', src: "https://cdn.pixabay.com/photo/2019/11/03/01/56/landscape-4597742_1280.jpg", style: {
                height: '300px',
                width: '450px',
                objectFit: 'cover',
            } })), index.h("dga-carousel-item", { key: 'c59ac0f2a901971dcbd5d0fa469699278d9a87f1' }, index.h("img", { key: 'eceb26c5e4222f3e1ee8f99e94cd36fa1e9cf7a5', src: "https://cdn.pixabay.com/photo/2020/07/09/10/31/sea-5386810_1280.jpg", style: {
                height: '300px',
                width: '450px',
                objectFit: 'cover',
            } })), index.h("dga-carousel-item", { key: '66334d303ec952bda189d55d97ab0e89a0b23e04' }, index.h("img", { key: '7dc3ef5181ae62b28886bfd49598a59471a0c4d6', src: "https://cdn.pixabay.com/photo/2020/07/09/10/31/sea-5386810_1280.jpg", style: {
                height: '300px',
                width: '450px',
                objectFit: 'cover',
            } })), index.h("dga-carousel-item", { key: '901f34d0accac9a4cf4d39a02933f119b01d818f' }, index.h("img", { key: '47205fe157c913823a8f6696c3bca233fe3e4b10', src: "https://cdn.pixabay.com/photo/2020/07/09/10/31/sea-5386810_1280.jpg", style: {
                height: '300px',
                width: '450px',
                objectFit: 'cover',
            } })), index.h("dga-carousel-item", { key: '4317213ac905bfeeb1fa056160949d046babe9d3' }, index.h("img", { key: '5f9f8b54bb78473b7823aa53d05b42c8c6704d71', src: "https://cdn.pixabay.com/photo/2020/07/09/10/31/sea-5386810_1280.jpg", style: {
                height: '300px',
                width: '450px',
                objectFit: 'cover',
            } }))), index.h("dga-carousel", { key: '93107d2f0979c2ea3c2bed33c59711b55c2621f4', size: 'lg', arrowsVariant: 'subtle' }, index.h("dga-carousel-item", { key: '6d93f75f1a0c4e99c9737f2c56b42a4c10693393' }, index.h("img", { key: '9519a6e58f6250e54b6d2ddefb9ecc83935138bc', src: "https://cdn.pixabay.com/photo/2019/11/03/01/56/landscape-4597742_1280.jpg", style: {
                height: '300px',
                width: '450px',
            } })), index.h("dga-carousel-item", { key: '7f7cd20624cf002137cd3178618650087eeb7b86' }, index.h("img", { key: '1b74ee12ca218cad33a6dfef691c05a38b9c7ce7', src: "https://cdn.pixabay.com/photo/2020/07/09/10/31/sea-5386810_1280.jpg", style: {
                height: '300px',
                width: '450px',
                objectFit: 'cover',
            } })), index.h("dga-carousel-item", { key: '9afcbedf396c5116805768a13736eac12669e2e7' }, index.h("img", { key: 'cfcd3e82c3aad3bd0389afdd22d2321928c9e30d', src: "https://cdn.pixabay.com/photo/2019/11/03/01/56/landscape-4597742_1280.jpg", style: {
                height: '300px',
                width: '450px',
                objectFit: 'cover',
            } })), index.h("dga-carousel-item", { key: 'ecfe8c991251ff3951b258476a14664e2721fa66' }, index.h("img", { key: 'ffe03e6dc7024dd3750294cf88ab9778946ef591', src: "https://cdn.pixabay.com/photo/2020/07/09/10/31/sea-5386810_1280.jpg", style: {
                height: '300px',
                width: '450px',
                objectFit: 'cover',
            } })), index.h("dga-carousel-item", { key: 'cf0970da4964c3143d26e80df9044cbf6723558e' }, index.h("img", { key: '6fd2e64708c63edf9900625b18296232d35df568', src: "https://cdn.pixabay.com/photo/2020/07/09/10/31/sea-5386810_1280.jpg", style: {
                height: '300px',
                width: '450px',
                objectFit: 'cover',
            } })), index.h("dga-carousel-item", { key: '43bb1988072b7b7f611357966a1da757f5d1a81c' }, index.h("img", { key: '796c455a7182934856a49e48d9b9ca1faaa361e2', src: "https://cdn.pixabay.com/photo/2020/07/09/10/31/sea-5386810_1280.jpg", style: {
                height: '300px',
                width: '450px',
                objectFit: 'cover',
            } })), index.h("dga-carousel-item", { key: '223ad22d488905bf52524e90c19140af1b86b164' }, index.h("img", { key: '1c2118b1f2088a8b9dbc4912599c6e5644482ecf', src: "https://cdn.pixabay.com/photo/2020/07/09/10/31/sea-5386810_1280.jpg", style: {
                height: '300px',
                width: '450px',
                objectFit: 'cover',
            } }))), index.h("dga-carousel", { key: 'a2dde1ddf904e8d8b0250b830c144a9a374bce7f', size: 'md' }, index.h("dga-carousel-item", { key: '28f4cc663a739e6e39d082142809794298027e98' }, index.h("img", { key: '2b9d648247efead8ad1d99ad214b798280c6b6e6', src: "https://cdn.pixabay.com/photo/2019/11/03/01/56/landscape-4597742_1280.jpg", style: {
                height: '300px',
                width: '450px',
            } })), index.h("dga-carousel-item", { key: 'a5e70ccfe76774b30e3be16470795a0bda76b7d8' }, index.h("img", { key: '2558d5172b31f428cb2b282a57b930bce454f62b', src: "https://cdn.pixabay.com/photo/2020/07/09/10/31/sea-5386810_1280.jpg", style: {
                height: '300px',
                width: '450px',
                objectFit: 'cover',
            } })), index.h("dga-carousel-item", { key: '9dbde8591e321f65bd7eb16630e2f93cfe55f270' }, index.h("img", { key: '4a723c5da6fa7e3d441c5f6083075e700f1e80bd', src: "https://cdn.pixabay.com/photo/2019/11/03/01/56/landscape-4597742_1280.jpg", style: {
                height: '300px',
                width: '450px',
                objectFit: 'cover',
            } })), index.h("dga-carousel-item", { key: '108ed117d304ec9054931c6bc567bd77490224a9' }, index.h("img", { key: 'cf03b3bc02387607e3995e6d139d565decfc7110', src: "https://cdn.pixabay.com/photo/2020/07/09/10/31/sea-5386810_1280.jpg", style: {
                height: '300px',
                width: '450px',
                objectFit: 'cover',
            } })), index.h("dga-carousel-item", { key: '970168634238327a3665be7a73073ce063b8261f' }, index.h("img", { key: '8e5551a1f7ef62152e6bb0b9c13da2c2c764d3f2', src: "https://cdn.pixabay.com/photo/2020/07/09/10/31/sea-5386810_1280.jpg", style: {
                height: '300px',
                width: '450px',
                objectFit: 'cover',
            } })), index.h("dga-carousel-item", { key: 'f6ba2fa8e0fe1f1387672fe1a2562dda91f117b8' }, index.h("img", { key: '9d08bb3353f83853dca40395b3064c7f3f10c857', src: "https://cdn.pixabay.com/photo/2020/07/09/10/31/sea-5386810_1280.jpg", style: {
                height: '300px',
                width: '450px',
                objectFit: 'cover',
            } })), index.h("dga-carousel-item", { key: '51f29f40d938cec59f550cf8307cb82b833bbb03' }, index.h("img", { key: 'fb774efe7edc5322e2ebd3abb2e1d729a4f2f230', src: "https://cdn.pixabay.com/photo/2020/07/09/10/31/sea-5386810_1280.jpg", style: {
                height: '300px',
                width: '450px',
                objectFit: 'cover',
            } }))), index.h("dga-carousel", { key: '1bccdd98ffdb8dd69571654f5705099556843d83', size: 'sm' }, index.h("dga-carousel-item", { key: 'fc95894fad57cb3b4f6c6feeb2c283b977f616c8' }, index.h("img", { key: '8b01e3aad270bea54f294b68688ccc532ce43da5', src: "https://cdn.pixabay.com/photo/2019/11/03/01/56/landscape-4597742_1280.jpg", style: {
                height: '300px',
                width: '450px',
            } })), index.h("dga-carousel-item", { key: 'df33685f5e6be331df84098f4603b2c022580af8' }, index.h("img", { key: '1d215f8c243ead5aaff39fed933accccbb443391', src: "https://cdn.pixabay.com/photo/2020/07/09/10/31/sea-5386810_1280.jpg", style: {
                height: '300px',
                width: '450px',
                objectFit: 'cover',
            } })), index.h("dga-carousel-item", { key: '151443d185460fa50c5622234e80be7fd65a5030' }, index.h("img", { key: 'b718c70a445ff8b32086990d717a161a18dd192d', src: "https://cdn.pixabay.com/photo/2019/11/03/01/56/landscape-4597742_1280.jpg", style: {
                height: '300px',
                width: '450px',
                objectFit: 'cover',
            } })), index.h("dga-carousel-item", { key: 'b0ed03b6585ce3761841243e66ca4aee823d153a' }, index.h("img", { key: '8b24e50c71c59f5af62b3b455477fbe306c5de83', src: "https://cdn.pixabay.com/photo/2020/07/09/10/31/sea-5386810_1280.jpg", style: {
                height: '300px',
                width: '450px',
                objectFit: 'cover',
            } })), index.h("dga-carousel-item", { key: '1a8820d54d0247d376a6d65a06f691785fb50fec' }, index.h("img", { key: 'cda02f0945307a6636adfd8ff0c6d64cd199f1f5', src: "https://cdn.pixabay.com/photo/2020/07/09/10/31/sea-5386810_1280.jpg", style: {
                height: '300px',
                width: '450px',
                objectFit: 'cover',
            } })), index.h("dga-carousel-item", { key: '1b69effb44e5ec0d2fab3844745fc901faee9b4f' }, index.h("img", { key: 'c7ac460167c559ef50bfec9d7eb1e546a38e68a4', src: "https://cdn.pixabay.com/photo/2020/07/09/10/31/sea-5386810_1280.jpg", style: {
                height: '300px',
                width: '450px',
                objectFit: 'cover',
            } })), index.h("dga-carousel-item", { key: 'e004bb8b6ae2bde6bc3d3256f3de6902ea53c97c' }, index.h("img", { key: '98cdf9c3f6dc73e76ea68f1bfe52ae261afbea09', src: "https://cdn.pixabay.com/photo/2020/07/09/10/31/sea-5386810_1280.jpg", style: {
                height: '300px',
                width: '450px',
                objectFit: 'cover',
            } })))), index.h("div", { key: 'c1d2b3a303033b6b0e3b98157270254e02c4d194', style: { display: 'flex', flexDirection: 'column', gap: '20px' } }))));
    }
};

exports.dga_carousel_usage = DgaCarouselUsage;
