// @platformscode/core/dist/cjs/dga-slider-usage.cjs.entry.js
'use strict';

var index = require('./index-DSiE2De_.js');

const DgaSliderUsage = class {
    constructor(hostRef) {
        index.registerInstance(this, hostRef);
        this.activeStep = 1;
        this.minValue = 10;
        this.maxValue = 100;
    }
    render() {
        return (index.h(index.Fragment, { key: '79432bef1a05778ebf774597630f1c632a96a905' }, index.h("br", { key: '31dcfd1804fa1a215fadb0c3f73cd02661a5df0f' }), index.h("dga-slider", { key: 'cc3df3d76a5f8390f9d0a52146cf04e9bfa9d3ef', min: this.minValue, max: this.maxValue, step: 1, range: true, size: "medium", helperText: "Helper Text Example", label: "Test", sx: {
                backgroundColor: "yellow",
                marginTop: "200px",
                ".range-slider-inner-rail": {
                    backgroundColor: "red"
                }
            } }), ";", index.h("br", { key: 'bb12eea2854eee3bc8d2522af5b379b11334bfbb' }), this.minValue, " - ", this.maxValue, index.h("dga-slider", { key: 'e6a2902e2f0ee0ec13073498389bc98c4d6da869' })));
    }
};

exports.dga_slider_usage = DgaSliderUsage;
