// @platformscode/core/dist/cjs/dga-header-action-btn_11.cjs.entry.js
'use strict';

var dgaHeaderActionBtn = require('./dga-header-action-btn-C9zCNIGD.js');
var index = require('./index-DSiE2De_.js');

const DgaSecondNavHeader = class {
    constructor(hostRef) {
        index.registerInstance(this, hostRef);
        this.variant = 'primary';
    }
    componentWillLoad() {
        this.el.childNodes.forEach((item) => {
            const el = item;
            if (el.tagName === 'DGA-SECOND-NAV-HEADER-CONTENT' && this.hideContent === true) {
                el.style.display = 'none';
            }
            if (el.tagName === 'DGA-SECOND-NAV-HEADER-ACTIONS' && this.hideActions === true) {
                el.style.display = 'none';
            }
            if (el.tagName === 'DGA-SECOND-NAV-HEADER-CONTENT' && el.attributes.getNamedItem('slot') === null) {
                el.setAttribute('slot', 'content');
            }
            if (el.tagName === 'DGA-SECOND-NAV-HEADER-ACTIONS' && el.attributes.getNamedItem('slot') === null) {
                el.setAttribute('slot', 'actions');
            }
        });
    }
    render() {
        return (index.h(index.Host, { key: '4d6a56396f788b3f018b3774c84fb6bc0d0849a9' }, index.h("div", { key: 'f6c6cf9410c08dfdab2c9889852649634b763dbe', class: `wrapper ${this.variant} `, "aria-label": 'a secondary navigation navbar' }, index.h("div", { key: '4b851ce02ce31b959365e7a09374097d369973ac', class: `container ${this.hideDivider ? '' : 'border'}` }, index.h("div", { key: '6214ce6582acd5ac51e252425546acd6a5edad46', class: `content` }, index.h("slot", { key: '3a70b02c824ba50ca99427169697341dc781c5d5', name: "content" })), index.h("div", { key: '2193cb09f2bd21e7741460d91ab3ab38c723370b', class: `actions` }, index.h("slot", { key: 'fa36abcb4f2e101b986b746af0dad2af3fee7fa4', name: "actions" }))))));
    }
    get el() { return index.getElement(this); }
};

const DgaSecondNavHeaderActions = class {
    constructor(hostRef) {
        index.registerInstance(this, hostRef);
    }
    render() {
        return (index.h(index.Host, { key: 'fec48112353fcd554b608780c7805c11ef383d51' }, index.h("div", { key: 'bb614b01662d01e7979638c6467014af2f5486e5', class: `items-container` }, index.h("slot", { key: '34dca111e83fd2489e3cb795dcf0083066a354a4' }))));
    }
};

const DgaSecondNavHeaderContent = class {
    constructor(hostRef) {
        index.registerInstance(this, hostRef);
    }
    render() {
        return (index.h(index.Host, { key: 'd9b732973272bc20a9407bfb291293ff8936a792' }, index.h("div", { key: 'dce382719b0b96c984dee57dd4b054325ff4834d', class: `items-container` }, index.h("slot", { key: '851ec4474a4630d627326c6daf4da23f635fd268' }))));
    }
};

const DgaSecondNavHeaderItem = class {
    constructor(hostRef) {
        index.registerInstance(this, hostRef);
    }
    connectedCallback() {
        this.el.childNodes.forEach((item) => {
            const el = item;
            if (el.tagName === 'DGA-ICON'.toLocaleUpperCase() && el.attributes.getNamedItem('slot') === null) {
                el.setAttribute('slot', 'icon');
            }
        });
    }
    render() {
        return (index.h(index.Host, { key: '87572880e04d67c78bbe24dce793d6147d166d37' }, index.h("slot", { key: 'e478e5be9f8231a25f8f76dc5b44b062552d3305', name: 'icon' }), index.h("slot", { key: 'bdf901885edd5f6edfe77ae63311748da8f6550e', name: 'label' }, this.label)));
    }
    get el() { return index.getElement(this); }
};

exports.dga_header_action_btn = dgaHeaderActionBtn.DgaHeaderActionBtn;
exports.dga_nav_header = dgaHeaderActionBtn.DgaNavHeader;
exports.dga_nav_header_actions = dgaHeaderActionBtn.DgaNavHeaderActions;
exports.dga_nav_header_link = dgaHeaderActionBtn.DgaNavHeaderLink;
exports.dga_nav_header_logos = dgaHeaderActionBtn.DgaNavHeaderLogos;
exports.dga_nav_header_main = dgaHeaderActionBtn.DgaNavHeaderMain;
exports.dga_nav_header_menu = dgaHeaderActionBtn.DgaNavHeaderMenu;
exports.dga_second_nav_header = DgaSecondNavHeader;
exports.dga_second_nav_header_actions = DgaSecondNavHeaderActions;
exports.dga_second_nav_header_content = DgaSecondNavHeaderContent;
exports.dga_second_nav_header_item = DgaSecondNavHeaderItem;
