// @platformscode/core/dist/cjs/dga-radio-button-usage.cjs.entry.js
'use strict';

var index = require('./index-DSiE2De_.js');

const DgaRadioButtonUsage = class {
    constructor(hostRef) {
        index.registerInstance(this, hostRef);
        this.activeStep = 1;
        this.activeRadio = "radion-1";
    }
    render() {
        return (index.h("div", { key: '57e500e777bc5d9fa8a5bf19ca3a8f47365890e2' }, index.h("h1", { key: 'a1ef1a55da1556f40f9d00f96eb63d16e8cd3a7e' }, "RadioButton Example"), index.h("div", { key: '6c81a5127513f739a5521df126a04b3f44363e9a', style: { padding: '50px', display: 'flex', gap: '20px' } }, index.h("div", { key: '4d131c121ec48210245f4f18267f0de4bcc2f6ae', style: { display: 'flex', flexDirection: 'column', gap: '20px' } }, index.h("dga-radio-button", { key: 'c40df4a50fe9093aa4d5713c4765de7c9ba3f708',
            // id="radio1"
            label: "Radio Label 1", hasHelperText: true, helperText: "When a selection needs a further detailed explanation, it goes here.", alertMessage: true, alertText: "Error/Warning message", name: "radio", value: 'radion-1', checked: this.activeRadio === 'radion-1', onChange: (e) => this.activeRadio = e.target.value }), index.h("dga-radio-button", { key: '8889959103d845e2f1d6086fa49856bb044b2eed',
            // id="radio2"
            label: "Radio Label 2", hasHelperText: true, helperText: "When a selection needs a further detailed explanation, it goes here.", alertMessage: true, alertText: "Error/Warning message", name: "radio", value: 'radion-2', checked: this.activeRadio === 'radion-2', onChange: (e) => this.activeRadio = e.target.value })), index.h("div", { key: 'ec48c19904dd96a468e185ae789689a0cc95d376', style: { display: 'flex', flexDirection: 'column', gap: '20px' } }))));
    }
};

exports.dga_radio_button_usage = DgaRadioButtonUsage;
