// @platformscode/core/dist/cjs/dga-list.cjs.entry.js
'use strict';

var index = require('./index-DSiE2De_.js');

const DgaList = class {
    constructor(hostRef) {
        index.registerInstance(this, hostRef);
        this.type = 'unordered';
        this.color = 'primary';
        //   @Prop() id: string = guid();
        this.title = "";
        this.items = [];
        this.renderIcon = (type) => {
            if (type === "with-icon") {
                return index.h("span", { class: "list-icon" }, "\u2713"); // Using checkmark icon for "with-icon" type
            }
            return null;
        };
    }
    render() {
        const Tag = this.type === 'ordered' ? 'ol' : 'ul';
        return (index.h("div", { key: '5082f2e16a2d7ca0c5e032c559bd159e176b33cf', class: `list list--${this.type} list--${this.color}` }, this.title && index.h("h3", { key: '9cdc61f9060b9c549c777fb260ddb1076d03e8d3', class: "list-title" }, this.title), index.h(Tag, { key: '45d79b00364e130ce5dc047c23047cb06cd9290d' }, this.items.map((item, index$1) => (index.h("li", { key: index$1, class: "list-item" }, this.type === "with-icon" && this.renderIcon(this.type), item))))));
    }
};

exports.dga_list = DgaList;
