// @platformscode/core/dist/cjs/dga-collapse-usage.cjs.entry.js
'use strict';

var index = require('./index-DSiE2De_.js');

const DgaCollapseUsage = class {
    constructor(hostRef) {
        index.registerInstance(this, hostRef);
        this.variant = 'neutral';
        this.open = false;
    }
    render() {
        return (index.h("div", { key: 'ce45988cc189e31973289c2785bd351adcbcd027' }, index.h("h1", { key: 'd73ae20e4ac38dbb1ddd6276ba1a3aa1c5328174' }, "Collapse Example"), index.h("div", { key: '9a1b43f1f384c22d3428aadf3d19af7dbe343ed5', style: { padding: '50px', display: 'flex', gap: '20px' } }, index.h("div", { key: 'aa3d3742b330872605c3c7ec42a5073eb97e2b0a', style: { display: 'flex', flexDirection: 'column', gap: '20px' } }, index.h("div", { key: 'a3381846bea7990d01306834ffa3e07f896f0853', style: { display: "flex", justifyContent: "center" } }, index.h("button", { key: '08e7ae751bbf3866a45a37f241c9d8e748b36bb9', onClick: () => this.open = true }, "open"), index.h("button", { key: '367a2f8354b616d07ce6f8f0c64f57d8adfc0361', onClick: () => this.open = false }, "close"))), index.h("dga-collapse", { key: '7b5f2514c6f258b1a5b21f286d399114abe59b07', open: this.open, "transition-duration": "0.1s" }, index.h("div", { key: 'f358334a58500c364213da1aff61cdd8925fbef9' }, "Your content here")), index.h("div", { key: '666f605c04a91b66ac0013b3b3696447c16e4116', style: { display: 'flex', flexDirection: 'column', gap: '20px' } }))));
    }
};

exports.dga_collapse_usage = DgaCollapseUsage;
