// @platformscode/core/dist/cjs/dga-table-of-content-usage.cjs.entry.js
'use strict';

var index = require('./index-DSiE2De_.js');

const DgaTableOfContentUsage = class {
    constructor(hostRef) {
        index.registerInstance(this, hostRef);
    }
    render() {
        return (index.h("div", { key: 'c983d0464d8d6ee6379fb06f8f6f1eb6210b4646', style: {
                position: 'relative',
                width: '100%',
                maxWidth: '1200px',
                margin: '0 auto',
                padding: '20px'
            } }, index.h("h1", { key: '2e1022ebb6fda513c57eaad9037dc5d5c0fee5e8' }, "Table of Content Example"), index.h("section", { key: 'c28ead61eabd589da3f75665164c59ba1d803f5f' }, index.h("div", { key: '5f42a6876b4b9889cc655ae5fa7d855f88268191', id: "section_1", style: {
                minHeight: "100vh",
                backgroundColor: "gray",
                padding: "20px",
                margin: "20px 0 200px 0"
            } }, index.h("h2", { key: '37a62dab4bb30cece993961811e805279c16b011' }, "Section 1"), index.h("div", { key: 'cf46ed4064d0ffd3f6a5f2780d4d4f24f10b716d', id: "section_1_child_1", style: {
                minHeight: "100vh",
                backgroundColor: "rgba(255,0,0,0.3)",
                margin: "20px 0",
                padding: "20px"
            } }, index.h("h3", { key: '3974a5bb5c74da1bacae8e8cc03131cbac64a54d' }, "Section 1 Child 1")), index.h("div", { key: 'bb0e0788d13e0cd04e867e35ad8329c676fb592d', id: "section_1_child_2", style: {
                minHeight: "100vh",
                backgroundColor: "rgba(0,0,255,0.3)",
                margin: "20px 0",
                padding: "20px"
            } }, index.h("h3", { key: '3d47a9f6f1daba571abc709ce05c1bb32d34c5da' }, "Section 1 Child 2")), index.h("div", { key: '1056ca8ccb04e711920a8c6a20faa667c3482259', id: "section_1_child_3", style: {
                minHeight: "100vh",
                backgroundColor: "rgba(0,255,0,0.3)",
                margin: "20px 0",
                padding: "20px"
            } }, index.h("h3", { key: 'd573b416d26e7930993bcb8af5404aa4365dd6e4' }, "Section 1 Child 3"), index.h("div", { key: 'e5b5e11e7a65b2eb89176963386105e60a2fcd06', id: "section_1_sub_child_1", style: {
                minHeight: "50vh",
                backgroundColor: "rgba(255,255,255,0.3)",
                margin: "20px 0",
                padding: "20px"
            } }, index.h("h4", { key: '21dcd99711f531e3d26feee68169550a503b31e3' }, "Section 1 Sub Child 1")), index.h("div", { key: '14f0fc69662547fa46ab56fc249ab53638330494', id: "section_1_sub_child_2", style: {
                minHeight: "50vh",
                backgroundColor: "rgba(255,255,255,0.3)",
                margin: "20px 0",
                padding: "20px"
            } }, index.h("h4", { key: 'c249e18bb7050e8ce10c55833dc732c3fc813a40' }, "Section 1 Sub Child 2")), index.h("div", { key: '23f7f8474e9ad8237ebc1759dd3650d24c9abd91', id: "section_1_sub_child_3", style: {
                minHeight: "50vh",
                backgroundColor: "rgba(255,255,255,0.3)",
                margin: "20px 0",
                padding: "20px"
            } }, index.h("h4", { key: '0e4282f9e1402adaf72cf0584ae5b7a52b3d4fd6' }, "Section 1 Sub Child 3")))), index.h("div", { key: '210eed05020f92b17d0d7d9f1468b1d575269bbd', id: "section_2", style: {
                minHeight: "100vh",
                backgroundColor: "rgba(0,0,0,0.8)",
                color: "white",
                padding: "20px",
                margin: "20px 0"
            } }, index.h("h2", { key: '85956f31bd566bf4b0fb4b7396fd0f66c9a013a2' }, "Section 2")), index.h("div", { key: '8398df7b6d92fd08b9e6b91110b96e29cd360d11', id: "section_3", style: {
                minHeight: "100vh",
                backgroundColor: "brown",
                padding: "20px",
                margin: "20px 0"
            } }, index.h("h2", { key: 'c04051a357d5389102f2b94593428d3e019eb1f6' }, "Section 3"), index.h("div", { key: '30ecc6f5e16af9622434511562d8d7dad6367eeb', id: "section_3_child_1", style: {
                minHeight: "50vh",
                backgroundColor: "rgba(255,255,255,0.3)",
                margin: "20px 0",
                padding: "20px"
            } }, index.h("h3", { key: '1d5366a8c60c322fee2a9902d103f432c4c107ab' }, "Section 3 Child 1")), index.h("div", { key: '6034b390801e6a803ac2bfc3db839bc61d5840e5', id: "section_3_child_2", style: {
                minHeight: "50vh",
                backgroundColor: "rgba(255,255,255,0.3)",
                margin: "20px 0",
                padding: "20px"
            } }, index.h("h3", { key: 'c277dc393cc632222d35b5a8d156f9cb69a0a3ad' }, "Section 3 Child 2"), index.h("div", { key: '680a8723b12ae8ec9ccc866c53c050f093581d1f', id: "section_3_child_2_1", style: {
                minHeight: "30vh",
                backgroundColor: "rgba(0,0,0,0.1)",
                margin: "20px 0",
                padding: "20px"
            } }, index.h("h4", { key: '12a7d0cad49ccd91a7b1db2582fe267dd61fa84e' }, "Section 3 Child 2.1")), index.h("div", { key: '9fb82323c2e689e157d1180a0da90c078b0b125a', id: "section_3_child_2_2", style: {
                minHeight: "30vh",
                backgroundColor: "rgba(0,0,0,0.1)",
                margin: "20px 0",
                padding: "20px"
            } }, index.h("h4", { key: 'cc64b76269fbf93e0c5aec932bb022a34724f14e' }, "Section 3 Child 2.2")), index.h("div", { key: 'c2e4b63399bb73bc67f15f344e0afb7db700677a', id: "section_3_child_2_3", style: {
                minHeight: "30vh",
                backgroundColor: "rgba(0,0,0,0.1)",
                margin: "20px 0",
                padding: "20px"
            } }, index.h("h4", { key: 'a7bee4e488a06732f97ce5850db3be04b86a0d4c' }, "Section 3 Child 2.3"))))), index.h("div", { key: '2725488cbd1f19c6a48aad41e79f0f524a4ca332', style: {
                position: "fixed",
                top: "100px",
                left: "40px",
                zIndex: '100',
                background: "white",
                borderRadius: "8px",
                boxShadow: "0 2px 8px rgba(0, 0, 0, 0.1)",
            } }, index.h("dga-table-of-content", { key: '112972cdd00c4036880244f4c9a17a72c7ab7ae6', title: "On this Page", subTitle: "Page Title", sections: [
                {
                    Name: 'Section 1',
                    Target: 'section_1',
                    Children: [
                        { Name: 'Section 1 Child 1', Target: 'section_1_child_1' },
                        { Name: 'Section 1 Child 2', Target: 'section_1_child_2' },
                        {
                            Name: 'Section 1 Child 3',
                            Target: 'section_1_child_3',
                            Children: [
                                { Name: 'Sub Child 1', Target: 'section_1_sub_child_1' },
                                { Name: 'Sub Child 2', Target: 'section_1_sub_child_2' },
                                { Name: 'Sub Child 3', Target: 'section_1_sub_child_3' }
                            ]
                        }
                    ]
                },
                { Name: 'Section 2', Target: 'section_2' },
                {
                    Name: 'Section 3',
                    Target: 'section_3',
                    Children: [
                        { Name: 'Section 3 Child 1', Target: 'section_3_child_1' },
                        {
                            Name: 'Section 3 Child 2',
                            Target: 'section_3_child_2',
                            Children: [
                                { Name: 'Child 2.1', Target: 'section_3_child_2_1' },
                                { Name: 'Child 2.2', Target: 'section_3_child_2_2' },
                                { Name: 'Child 2.3', Target: 'section_3_child_2_3' }
                            ]
                        }
                    ]
                }
            ] })), index.h("div", { key: '1189116fa038429ebe293b239fb30ab7d3d788da', style: {
                position: "fixed",
                top: "100px",
                right: "40px",
                zIndex: '100',
                background: "white",
                borderRadius: "8px",
                boxShadow: "0 2px 8px rgba(0, 0, 0, 0.1)",
            } }, index.h("dga-table-of-content", { key: '27fc846547d8a3e877f77e1d1c55fa0f548b3baf', title: "\u0641\u064A \u0647\u0630\u0647 \u0627\u0644\u0635\u0641\u062D\u0629", subTitle: "[\u0627\u0633\u0645 \u0627\u0644\u0635\u0641\u062D\u0629]", rtl: true, sections: [
                {
                    Name: 'قسم صفحة',
                    Target: 'section_1',
                    Children: [
                        { Name: 'قسم صفحة', Target: 'section_1_child_1' },
                        { Name: 'قسم صفحة', Target: 'section_1_child_2' },
                        {
                            Name: 'قسم صفحة',
                            Target: 'section_1_child_3',
                            Children: [
                                { Name: 'قسم صفحة', Target: 'section_1_sub_child_1' },
                                { Name: 'قسم صفحة', Target: 'section_1_sub_child_2' },
                                { Name: 'قسم صفحة', Target: 'section_1_sub_child_3' }
                            ]
                        }
                    ]
                },
                { Name: 'Section 2', Target: 'section_2' },
                {
                    Name: 'Section 3',
                    Target: 'section_3',
                    Children: [
                        { Name: 'Section 3 Child 1', Target: 'section_3_child_1' },
                        {
                            Name: 'Section 3 Child 2',
                            Target: 'section_3_child_2',
                            Children: [
                                { Name: 'Child 2.1', Target: 'section_3_child_2_1' },
                                { Name: 'Child 2.2', Target: 'section_3_child_2_2' },
                                { Name: 'Child 2.3', Target: 'section_3_child_2_3' }
                            ]
                        }
                    ]
                }
            ] }))));
    }
};

exports.dga_table_of_content_usage = DgaTableOfContentUsage;
