// @platformscode/core/dist/cjs/dga-card-usage.cjs.entry.js
'use strict';

var index = require('./index-DSiE2De_.js');

const DgaCardUsage = class {
    constructor(hostRef) {
        index.registerInstance(this, hostRef);
        this.variant = 'neutral';
        this.selected = false;
    }
    render() {
        return (index.h("div", { key: 'a5891b4a7e1c7c6f40534dba563a0868c030b6be' }, index.h("h1", { key: 'f4bdc496f1b640e4ca9debae252aebae0408cccc' }, "Card Example"), index.h("div", { key: '3852ccdd17c58d4369377947b5eb2aefd553afe5', style: { padding: '50px', display: 'flex', gap: '20px' } }, index.h("div", { key: 'a24d0337bd66b59374ee5c651bd1e2104cac991f', style: { display: 'flex', flexDirection: 'column', gap: '20px' } }, index.h("dga-card", { key: '9ec040fdee43c39087ed6ea1e80f1e1907f1794d', type: "expanded", cardTitle: "Card Title", description: "Card content placeholder text goes here", showFeaturedIcon: true }, index.h("div", { key: 'cfa10cad35b0f4c53ea5a0c16ade5e01d179372d', slot: "collapse-content" }, "dgdsgdfgdfg")), index.h("dga-card", { key: '7fca6697abb3c4218c8eb6ebe0ea026a2d6dce90', type: "selectable", cardTitle: "Card Title", description: "Card content placeholder text goes here", isSelected: this.selected, onCheckboxChange: () => (this.selected = !this.selected) }), index.h("dga-card", { key: '9d31f50eaeb197f8cbce1467aeb573b78b59dd70', cardTitle: "Card Title", sx: {
                '& > .card': {
                    backgroundColor: 'red',
                    height: '400px',
                },
            } }), index.h("dga-card", { key: 'ec89d1a7e46c0b89e5097f4d734796f57916293f', cardTitle: "No Shadow", effect: "no-shadow" }), index.h("dga-card", { key: 'd14ef0aa42b8d53b305fd4fdf0d1abd06d6cc812', cardTitle: "Stroke", effect: "stroke" }), index.h("dga-card", { key: 'f8a9b640ab7b7fd169422c61ca285dfb41358c82', cardTitle: "With Shadow", effect: "with-shadow" })), index.h("div", { key: '26a0b295fc2397a857048501d22fc1d8a0914ada', style: { display: 'flex', flexDirection: 'column', gap: '20px' } }))));
    }
};

exports.dga_card_usage = DgaCardUsage;
