// @platformscode/core/dist/cjs/dga-table-usage.cjs.entry.js
'use strict';

var index = require('./index-DSiE2De_.js');

const DgaTableUsage = class {
    constructor(hostRef) {
        index.registerInstance(this, hostRef);
    }
    render() {
        return (index.h("div", { key: 'b5054a6d0fd3e7dae1e12c3b4cecea2ae94e16a6' }, index.h("h1", { key: 'ee2a65396fbbf71469deabe03fbfdd9ec16d80d2' }, "Table Example"), index.h("div", { key: '108b830e26a740d0d73e8a2ba0206f9848d92e63', style: { padding: '50px', display: 'flex', gap: '20px' } }, index.h("div", { key: 'b452499d51152fe62c37b6cb9c2ab539444451db', style: { display: 'flex', flexDirection: 'column', gap: '20px' } }, index.h("dga-data-table", { key: '8cfded33a0e06c92fb33cea795a028d1ceb30349', columns: [
                {
                    label: 'Header',
                },
            ], cells: [
                {
                    title: 'city',
                    sort: 'up',
                    isSort: true,
                    isFilter: true,
                    type: 'propertyName',
                    propertyName: 'city',
                    colSpan: 1, // Set colSpan to 1 (default is 1)
                },
                {
                    title: 'price',
                    sort: 'down',
                    isSort: true,
                    isFilter: true,
                    type: 'propertyName',
                    propertyName: 'price',
                    colSpan: 1,
                },
                {
                    title: 'state',
                    isSort: true,
                    isFilter: true,
                    type: 'propertyName',
                    propertyName: 'state',
                    colSpan: 1,
                },
            ], data: [
                {
                    h_id: '3',
                    city: 'Dallas',
                    state: 'TX',
                    zip: '75201',
                    price: '162500',
                    selected: false, // Add selected property for checkbox functionality
                },
                {
                    h_id: '4',
                    city: 'Bevery Hills',
                    state: 'CA',
                    zip: '90210',
                    price: '319250',
                    selected: false,
                },
                {
                    h_id: '5',
                    city: 'New York',
                    state: 'NY',
                    zip: '00010',
                    price: '962500',
                    selected: false,
                },
            ] })), index.h("div", { key: '441a9cf2d8ad1afa33d6f857da4dba1e3e3e4b79', style: { display: 'flex', flexDirection: 'column', gap: '20px' } }))));
    }
};

exports.dga_table_usage = DgaTableUsage;
