// @platformscode/core/dist/cjs/dga-checkbox-usage.cjs.entry.js
'use strict';

var index = require('./index-DSiE2De_.js');

const DgaCheckboxUsage = class {
    constructor(hostRef) {
        index.registerInstance(this, hostRef);
        this.variant = 'neutral';
    }
    render() {
        return (index.h("div", { key: '657e02ca429e61dc3c69f03ca4713f7f490acff4' }, index.h("h1", { key: '009b790c4ebd2dfd57f9417644167e4a12731811' }, "Checkbox Example"), index.h("div", { key: 'a22382e8088dd1af150d283c0dd7fae316249b5d', style: { padding: '50px', display: 'flex', gap: '20px' } }, index.h("div", { key: 'c26ee773857a7c6d50b8606147b57b363c8f0cc7', style: { display: 'flex', flexDirection: 'column', gap: '20px' } }, index.h("h1", { key: 'ecef81a3af106be510bd07c6c6b146a8956f6c44' }, "test"), index.h("dga-checkbox", { key: '305ae9eaa7437668131dac41c92bd25f33776099', label: 'Checkbox Label Checkbox Label Checkbox Label Checkbox Label', Indeterminate: false, alertMessage: true, alertText: "Error/Warning message" }), index.h("dga-checkbox", { key: 'd0f555bad89339e225972cfa8114faea6fcff33c', size: 'lg' }), index.h("dga-checkbox", { key: '1495f1ee81452c4497650f29ed6f0b2e1900ed7e', size: 'md' }), index.h("dga-checkbox", { key: '4c79ac1b264bf1645d534644b35c15c22e72fdb6', size: 'sm' })), index.h("div", { key: '38f92ac5f0d0a78b5d6cb880c5a2e01efd750360', style: { display: 'flex', flexDirection: 'column', gap: '20px' } }))));
    }
};

exports.dga_checkbox_usage = DgaCheckboxUsage;
