// @platformscode/core/dist/cjs/nds-avatar-usage.cjs.entry.js
'use strict';

var index = require('./index-DSiE2De_.js');

const NdsAvatarUsage = class {
    constructor(hostRef) {
        index.registerInstance(this, hostRef);
        this.variant = 'neutral';
    }
    render() {
        return (index.h("div", { key: '8bd634bed9b9e92a0275b357f099e15393b2d60c' }, index.h("h1", { key: '1b35a22ac8ab73a01da969531f106fd5ddf8468f' }, "Avatar Example"), index.h("div", { key: '030907c7c20f3d9f9544ea06c6b524cae9c72404', style: { padding: '50px', display: 'flex', gap: '20px' } }, index.h("div", { key: '6f830b08409293cdcf8ed1956554c31e9912cabb', style: { display: 'flex', flexDirection: 'column', gap: '20px' } }, index.h("div", { key: '8b516d35b9cd1b4586e9b99ed6ac92cbd11b76c5', style: { padding: "10px", display: 'flex', flexDirection: 'column', gap: '20px', backgroundColor: "black" } }, index.h("dga-avatar", { key: '360e6274833e963e43f05e4d75a60c2bd077eb34', size: 120, border: false, square: true, sx: {
                "& .dga-avatar": {
                    borderRadius: "100px"
                }
            } }), index.h("dga-avatar", { key: 'c5b6b73b2c84a90afa813cbd8eb830aec8a78072', size: 120, border: true, square: true })), index.h("dga-avatar", { key: 'afffc1aeff60b4a61a05dfb5d609b7c7f158a615', type: "icon", size: 120, icon: {
                name: 'user',
                variant: "stroke",
                type: "rounded"
            } }), index.h("dga-avatar", { key: 'b9fe7852b1286d7453adf4e76b1d318c31f8bfc3', type: "icon", size: 80, icon: {
                name: 'user',
                variant: "stroke",
                type: "rounded"
            } }), index.h("dga-avatar", { key: '4ccec05c82edf0db15a969cc884abe4855211192', type: "icon", size: 68, icon: {
                name: 'user',
                variant: "stroke",
                type: "rounded"
            } }), index.h("dga-avatar", { key: 'db05536bdf27c69c26dee95198e07fe50968cea8', type: "icon", size: 48, icon: {
                name: 'user',
                variant: "stroke",
                type: "rounded"
            } }), index.h("dga-avatar", { key: '218b07ae12667a148a940de7e3102b626869ec37', type: "icon", size: 40, icon: {
                name: 'user',
                variant: "stroke",
                type: "rounded"
            } }), index.h("dga-avatar", { key: 'e0a44837e32b3106892d5f9b47919da919ac4f3e', type: "icon", size: 32, icon: {
                name: 'user',
                variant: "stroke",
                type: "rounded"
            } }), index.h("dga-avatar", { key: 'c7e8adfdf788cd982d6901f8e3efb26d203c7906', type: "icon", size: 24, icon: {
                name: 'user',
                variant: "stroke",
                type: "rounded"
            } })), index.h("dga-avatar-group", { key: '47c372bf8721299f219ff74e840a285eff79b699', stacked: true }, index.h("dga-avatar", { key: '07aa8287abd4af2a6221293ec35808e5702b0d5e', type: "initials", text: 'AB', size: 40 }), index.h("dga-avatar", { key: '96867377c6fd55c43046e3e0efcaa949b286323a', type: "initials", text: 'CD', size: 40 }), index.h("dga-avatar", { key: '809288ee6ed709606007149384f2279e6a844585', type: "initials", text: 'SB', size: 40 }), index.h("dga-avatar", { key: 'f7061655286136967511e50fe486a1d8f1aa0e87', type: "initials", text: 'WK', size: 40 }), index.h("dga-avatar", { key: '4acb2642529d776ee500c4a2b09f3f54e9b735a9', type: 'initials', text: `+4`, size: 40 })), index.h("div", { key: '4251378603fbdf673a652be95b6cbc6363f102fb', style: { display: 'flex', flexDirection: 'column', gap: '20px' } }))));
    }
};

exports.nds_avatar_usage = NdsAvatarUsage;
