// @platformscode/core/dist/collection/components/dga-inline-alert/dga-inline-alert.js
import { h, Fragment } from "@stencil/core";
import { guid } from "../../utils";
export class DgaInlineAlert {
    constructor() {
        this.type = 'neutral';
        this.isactionButtons = true;
        this.isCloseButton = true;
        this.colored = false;
        this.buttonsList = [];
        this.icons = {
            error: 'hgi-stroke hgi-help-circle',
            warning: 'hgi-stroke hgi-alert-02',
            success: 'hgi-stroke hgi-tick-02',
            info: 'hgi-stroke hgi-help-circle',
            neutral: 'hgi-stroke hgi-help-circle',
        };
    }
    render() {
        return (h("div", { key: '53e0f8ad88dd392f570c18c5ebc9f6067c2d1548', class: `alert alert--${this.type} ${this.colored ? `alert--${this.type}-bg` : ''}` }, h("div", { key: '5c57b6068d508dc30360ff2be6e3bc8d2d023d90', class: `alert__header ${this.helperText ? '' : 'dga-flex-center-y'}` }, h("div", { key: '64dffbbe9247e26c84e397bcf3ce687b63512bc2', class: `featured-icon featured-icon--circle featured-icon--md featured-icon--light-${this.type === 'neutral' ? 'gray' : this.type}` }, h("i", { key: '80db45390465e4d4fb757126e2d8f3fe59052fbc', class: this.icons[this.type], style: { fontSize: '20px' } })), h("div", { key: '1a3905e6bdb440cda59718bf3d2409be2c5ba11e', class: "alert__body" }, h("h4", { key: 'e4461712015adf96e0ff9ec0cc3b04890af7f915', class: "alert__title text-md-semibold" }, this.leadText), this.helperText && h("p", { key: 'f8d8f19514c759a71b5f6534ad2c6b09490c9cd2', class: "alert__content text-sm-regular" }, this.helperText)), this.isCloseButton && (h("div", { key: '725d08616f1db395035fdadf2d1451e1b09119b7', class: "alert__close-btn" }, h("button", { key: 'e409b36188db32e9949763c28f1fb48c803c1a70', class: "dga-btn dga-btn--close dga-btn--icon dga-btn--md", onClick: e => this.onClose && this.onClose(e) }, h("span", { key: '44a37ad0ce5716d52fa859a3bf80a294458fad0e', class: "dga-btn-icon" }, h("i", { key: 'fdda958b88150f39f891224f2cb5ff6ed84117ec', class: "hgi-stroke hgi-multiplication-sign", style: { fontSize: '20px' } })))))), this.isactionButtons && this.buttonsList.length > 0 && (h(Fragment, { key: '0c22cee0731db028b82b633cf0fa5cd064b2ba58' }, h("div", { key: 'e1b1961c526d02ee59f3f2e8d7658f8fd86a3d88', class: "alert__action" }, this.buttonsList.map(({ id, onClick, exraClasses, title, label }) => {
            return (h("button", Object.assign({ key: id ? id : guid(), id: id ? id : guid(), onClick: e => onClick(e), class: `dga-btn dga-btn--subtle dga-btn--md ${exraClasses}` }, (title && { title: title })), label));
        })), h("div", { key: 'd1efc7a34248bfa339f67526383a7dc213b9d25a', class: "alert__action-mobile" }, this.buttonsList.map(({ id, onClick, exraClasses, title, label }) => {
            return (h("button", Object.assign({ key: id ? id : guid(), id: id ? id : guid(), onClick: e => onClick(e), class: `btn btn--secondary btn--lg ${exraClasses}` }, (title && { title: title })), label));
        }))))));
    }
    static get is() { return "dga-inline-alert"; }
    static get originalStyleUrls() {
        return {
            "$": ["dga-inline-alert.scss"]
        };
    }
    static get styleUrls() {
        return {
            "$": ["dga-inline-alert.css"]
        };
    }
    static get properties() {
        return {
            "type": {
                "type": "string",
                "attribute": "type",
                "mutable": false,
                "complexType": {
                    "original": "'neutral' | 'info' | 'error' | 'success' | 'warning'",
                    "resolved": "\"error\" | \"info\" | \"neutral\" | \"success\" | \"warning\"",
                    "references": {}
                },
                "required": false,
                "optional": false,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false,
                "defaultValue": "'neutral'"
            },
            "leadText": {
                "type": "string",
                "attribute": "lead-text",
                "mutable": false,
                "complexType": {
                    "original": "string",
                    "resolved": "string",
                    "references": {}
                },
                "required": false,
                "optional": false,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false
            },
            "helperText": {
                "type": "string",
                "attribute": "helper-text",
                "mutable": false,
                "complexType": {
                    "original": "string",
                    "resolved": "string",
                    "references": {}
                },
                "required": false,
                "optional": true,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false
            },
            "isactionButtons": {
                "type": "boolean",
                "attribute": "isaction-buttons",
                "mutable": false,
                "complexType": {
                    "original": "boolean",
                    "resolved": "boolean",
                    "references": {}
                },
                "required": false,
                "optional": false,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false,
                "defaultValue": "true"
            },
            "isCloseButton": {
                "type": "boolean",
                "attribute": "is-close-button",
                "mutable": false,
                "complexType": {
                    "original": "boolean",
                    "resolved": "boolean",
                    "references": {}
                },
                "required": false,
                "optional": false,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false,
                "defaultValue": "true"
            },
            "onClose": {
                "type": "unknown",
                "attribute": "on-close",
                "mutable": false,
                "complexType": {
                    "original": "(event: any) => void",
                    "resolved": "(event: any) => void",
                    "references": {}
                },
                "required": false,
                "optional": true,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false
            },
            "colored": {
                "type": "boolean",
                "attribute": "colored",
                "mutable": false,
                "complexType": {
                    "original": "boolean",
                    "resolved": "boolean",
                    "references": {}
                },
                "required": false,
                "optional": false,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false,
                "defaultValue": "false"
            },
            "buttonsList": {
                "type": "unknown",
                "attribute": "buttons-list",
                "mutable": false,
                "complexType": {
                    "original": "{\n    id?: string;\n    label: string;\n    onClick: (event?: any) => void;\n    exraClasses?: string;\n    title?: string;\n  }[]",
                    "resolved": "{ id?: string; label: string; onClick: (event?: any) => void; exraClasses?: string; title?: string; }[]",
                    "references": {}
                },
                "required": false,
                "optional": false,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "defaultValue": "[]"
            }
        };
    }
}
