// @platformscode/core/dist/collection/components/dga-list/dga-list-item.js
import { h } from "@stencil/core";
export class DgaListItem {
    componentWillLoad() {
        const parentElement = this.el.closest('[type]');
        if (parentElement) {
            this.inheritedType = parentElement.getAttribute('type');
        }
    }
    // componentWillLoad() {
    //   if (this.type === 'with-icon' && !this.icon) {
    //     throw new Error('Icon prop is required when type is "with-icon"');
    //   }
    // }
    getListClass() {
        return {
            'list-item': true,
            [`list-item--level-${this.level}`]: true,
            [`list-item--${this.type}`]: true
        };
    }
    renderMarker() {
        if (this.type === 'ordered') {
            return (h("span", { class: "list-item__marker" }, this.level === 'one' ? this.itemNumber : this.itemLetter));
        }
        return null;
    }
    render() {
        return (h("li", { key: 'ea563aa2b52b600800ce2991e055b2db73344cf4', class: this.getListClass() }, this.type === 'with-icon' && (h("span", { key: '23548209d319ae5791f14da2b56068a3775b137f', class: "list-item__icon" }, h("dga-icon", { key: 'ecf93dfc8a40471abe01c4e3f78229ea09dd8a82', icon: this.icon, size: 16 }))), this.renderMarker(), h("span", { key: '8ef34fbd75174cb7ba82c84835cb3441b14d33b6', class: "list-item__text" }, this.itemText)));
    }
    static get is() { return "dga-list-item"; }
    static get originalStyleUrls() {
        return {
            "$": ["dga-list-item.scss"]
        };
    }
    static get styleUrls() {
        return {
            "$": ["dga-list-item.css"]
        };
    }
    static get properties() {
        return {
            "type": {
                "type": "string",
                "attribute": "type",
                "mutable": false,
                "complexType": {
                    "original": "ListItemProps['type']",
                    "resolved": "\"ordered\" | \"unordered\" | \"with-icon\"",
                    "references": {
                        "ListItemProps": {
                            "location": "local",
                            "path": "/Users/mohamedaboelezz/projects/Azm-X/platform-code-cross-platform-storybook/packages/core/src/components/dga-list/dga-list-item.tsx",
                            "id": "src/components/dga-list/dga-list-item.tsx::ListItemProps"
                        }
                    }
                },
                "required": true,
                "optional": false,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false
            },
            "level": {
                "type": "string",
                "attribute": "level",
                "mutable": false,
                "complexType": {
                    "original": "ListItemProps['level']",
                    "resolved": "\"one\" | \"two\"",
                    "references": {
                        "ListItemProps": {
                            "location": "local",
                            "path": "/Users/mohamedaboelezz/projects/Azm-X/platform-code-cross-platform-storybook/packages/core/src/components/dga-list/dga-list-item.tsx",
                            "id": "src/components/dga-list/dga-list-item.tsx::ListItemProps"
                        }
                    }
                },
                "required": true,
                "optional": false,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false
            },
            "itemText": {
                "type": "string",
                "attribute": "item-text",
                "mutable": false,
                "complexType": {
                    "original": "string",
                    "resolved": "string",
                    "references": {}
                },
                "required": true,
                "optional": false,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false
            },
            "itemNumber": {
                "type": "string",
                "attribute": "item-number",
                "mutable": false,
                "complexType": {
                    "original": "string",
                    "resolved": "string",
                    "references": {}
                },
                "required": false,
                "optional": true,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false
            },
            "itemLetter": {
                "type": "string",
                "attribute": "item-letter",
                "mutable": false,
                "complexType": {
                    "original": "string",
                    "resolved": "string",
                    "references": {}
                },
                "required": false,
                "optional": true,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false
            },
            "icon": {
                "type": "string",
                "attribute": "icon",
                "mutable": false,
                "complexType": {
                    "original": "string",
                    "resolved": "string",
                    "references": {}
                },
                "required": false,
                "optional": true,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false
            }
        };
    }
    static get states() {
        return {
            "inheritedType": {}
        };
    }
    static get elementRef() { return "el"; }
}
