// @platformscode/core/dist/collection/components/dga-file-upload/dga-file-upload.js
import { h } from "@stencil/core";
import { guid } from "../../utils";
export class DgaFileUpload {
    constructor() {
        this.id = guid();
        this.name = guid();
        this.title = 'Drag and drop files here to upload';
        this.fileTypesText = 'Maximum file size allowed is 2MB, supported file formats include .jpg, .png, and .pdf.';
        this.actionName = 'Browse Files';
        this.disabled = false;
        this.files = [];
        this.dragState = false;
        this.handleFileSelection = (event) => {
            const input = event.target;
            this.onChange && this.onChange(event);
            const selectedFiles = input.files;
            if (selectedFiles) {
                if (this.maximumFilesSize) {
                    const isFileOverSize = Array.from(selectedFiles).some(file => {
                        return file.size > this.maximumFilesSize;
                    });
                    if (isFileOverSize) {
                        this.uploadFileOverSize && this.uploadFileOverSize();
                        return;
                    }
                }
                const fileItems = Array.from(selectedFiles).map(file => ({
                    name: file.name,
                    uploadStatus: 'pending', // initial status
                }));
                this.files = [...this.files, ...fileItems];
                this.getUploadedFile && this.getUploadedFile([...this.files]);
            }
        };
        this.handleDrop = (event) => {
            event.preventDefault();
            event.stopPropagation();
            const droppedFiles = event.dataTransfer.files; // Accessing the dropped files
            if (droppedFiles) {
                const fileItems = Array.from(droppedFiles).map(file => ({
                    name: file.name,
                    uploadStatus: 'pending',
                }));
                this.files = [...this.files, ...fileItems]; // Update state with new files
            }
            this.dragState = false;
        };
        this.addDragOverClass = (event) => {
            event.preventDefault();
            event.stopPropagation();
            this.dragState = true;
        };
        this.removeDragOverClass = (event) => {
            event.preventDefault();
            event.stopPropagation();
            this.dragState = false;
        };
    }
    componentDidLoad() {
        const dropArea = document.querySelector('.file-upload__drop-area');
        dropArea.addEventListener('dragover', this.addDragOverClass);
        dropArea.addEventListener('dragleave', this.removeDragOverClass);
        dropArea.addEventListener('drop', this.handleDrop);
    }
    disconnectedCallback() {
        const dropArea = document.querySelector('.file-upload__drop-area');
        dropArea.removeEventListener('dragover', this.addDragOverClass);
        dropArea.removeEventListener('dragleave', this.removeDragOverClass);
        dropArea.removeEventListener('drop', this.handleDrop);
    }
    handleDeleteFile(index, file) {
        this.files = [...this.files.slice(0, index), ...this.files.slice(index + 1)];
        this.emitDeleteFile && this.emitDeleteFile(file, index);
    }
    renderFile(file, index) {
        if (file.uploadStatus === 'error') {
            return (h("div", { class: "file-upload__file-item--error" }, h("div", { class: "file-upload__file-item" }, h("span", { class: "file-upload__file-status file-upload__file-status--error" }, h("i", { class: "hgi-solid hgi-alert-circle", style: { fontSize: "20px" } })), h("span", { class: "file-upload__file-name" }, file.name), h("button", { class: "btn btn--close btn--icon btn--sm", onClick: () => this.handleDeleteFile(index, file) }, h("span", { class: "btn-icon" }, h("i", { class: "hgi-stroke hgi-multiplication-sign", style: { fontSize: "20px" } })))), h("div", { class: "file-upload__file-item-msg" }, file.errorMessage ? file.errorMessage : 'Here goes the helper text.')));
        }
        return (h("div", { class: "file-upload__file-item" }, h("span", { class: `file-upload__file-status file-upload__file-status--${file.uploadStatus}` }, file.uploadStatus === 'uploading' && file.progress && h("dga-loading", { size: "tiny", variant: "neutral" }), file.uploadStatus === 'complete' && h("i", { class: 'hgi-solid hgi-checkmark-circle-02' })), h("span", { class: "file-upload__file-name" }, file.name), h("button", { class: "btn btn--close btn--icon btn--sm", onClick: () => this.handleDeleteFile(index, file) }, h("span", { class: "btn-icon" }, h("i", { class: "hgi-stroke hgi-multiplication-sign", style: { fontSize: "20px" } })))));
    }
    render() {
        return (h("div", { key: '354db699e2ca9a983d346ba99a4d3f62ffb1ffb9', class: `file-upload ${this.dragState ? 'file-upload--drag-over' : ''} ${this.disabled ? 'file-upload--disabled' : ''}` }, h("input", { key: 'bd0f9af9418fc728ce3e24664b5d9fb72baed9d9', type: "file", id: this.id, name: this.name, class: "file-upload__input", accept: this.accept, multiple: this.multiple, "aria-hidden": "true", onChange: this.handleFileSelection, ref: el => (this.fileInputRef = el) }), h("label", { key: '8d44991a59f4e4ca4f4f35721abf9a0289bf475d', htmlFor: this.id, class: "file-upload__drop-area" }, h("span", { key: '5b1c048c681bed8aa96c3b78fa36a068ce563fd0', class: "file-upload__icon" }, h("i", { key: '8db9cfe179a6f07a2488a97b773a8326943f6f27', class: "hgi-stroke hgi-file-upload", style: { fontSize: "20px" } })), h("div", { key: 'bd1584e0eaa583f9b65f1543c3788a4b19c231b1', class: "file-upload__content" }, h("p", { key: '0f63c39033091ffc4b4b76084b0a5cf857aa3387', class: "file-upload__instructions" }, this.title), h("p", { key: 'b9c895fd5c4295384a7b2dee8a676bea19ae4d9f', class: "file-upload__formats" }, this.fileTypesText)), h("button", { key: 'b23ed88d60bf16cf05393e55779e568ffb486bc5', class: "btn btn--md btn--secondary", "aria-label": "Browse Files", disabled: this.dragState || this.disabled, type: "button", onClick: () => { var _a; return (_a = this.fileInputRef) === null || _a === void 0 ? void 0 : _a.click(); } }, h("span", { key: '9839293615be1e0059def33533f540c3b5f06d8e' }, this.actionName))), h("div", { key: '856088f7656a35c90b4652cba90c67ecd26a2680', class: "file-upload__files-list" }, this.files.map((file, index) => this.renderFile(file, index)))));
    }
    static get is() { return "dga-file-upload"; }
    static get encapsulation() { return "shadow"; }
    static get originalStyleUrls() {
        return {
            "$": ["dga-file-upload.scss"]
        };
    }
    static get styleUrls() {
        return {
            "$": ["dga-file-upload.css"]
        };
    }
    static get properties() {
        return {
            "id": {
                "type": "string",
                "attribute": "id",
                "mutable": false,
                "complexType": {
                    "original": "string",
                    "resolved": "string",
                    "references": {}
                },
                "required": false,
                "optional": false,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false,
                "defaultValue": "guid()"
            },
            "name": {
                "type": "string",
                "attribute": "name",
                "mutable": false,
                "complexType": {
                    "original": "string",
                    "resolved": "string",
                    "references": {}
                },
                "required": false,
                "optional": false,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false,
                "defaultValue": "guid()"
            },
            "accept": {
                "type": "string",
                "attribute": "accept",
                "mutable": false,
                "complexType": {
                    "original": "string",
                    "resolved": "string",
                    "references": {}
                },
                "required": false,
                "optional": false,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false
            },
            "multiple": {
                "type": "boolean",
                "attribute": "multiple",
                "mutable": false,
                "complexType": {
                    "original": "boolean",
                    "resolved": "boolean",
                    "references": {}
                },
                "required": false,
                "optional": false,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false
            },
            "title": {
                "type": "string",
                "attribute": "title",
                "mutable": false,
                "complexType": {
                    "original": "string",
                    "resolved": "string",
                    "references": {}
                },
                "required": false,
                "optional": false,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false,
                "defaultValue": "'Drag and drop files here to upload'"
            },
            "fileTypesText": {
                "type": "string",
                "attribute": "file-types-text",
                "mutable": false,
                "complexType": {
                    "original": "string",
                    "resolved": "string",
                    "references": {}
                },
                "required": false,
                "optional": false,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false,
                "defaultValue": "'Maximum file size allowed is 2MB, supported file formats include .jpg, .png, and .pdf.'"
            },
            "actionName": {
                "type": "string",
                "attribute": "action-name",
                "mutable": false,
                "complexType": {
                    "original": "string",
                    "resolved": "string",
                    "references": {}
                },
                "required": false,
                "optional": false,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false,
                "defaultValue": "'Browse Files'"
            },
            "maximumFilesSize": {
                "type": "number",
                "attribute": "maximum-files-size",
                "mutable": false,
                "complexType": {
                    "original": "number",
                    "resolved": "number",
                    "references": {}
                },
                "required": false,
                "optional": true,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false
            },
            "disabled": {
                "type": "boolean",
                "attribute": "disabled",
                "mutable": false,
                "complexType": {
                    "original": "boolean",
                    "resolved": "boolean",
                    "references": {}
                },
                "required": false,
                "optional": true,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false,
                "defaultValue": "false"
            },
            "getUploadedFile": {
                "type": "unknown",
                "attribute": "get-uploaded-file",
                "mutable": false,
                "complexType": {
                    "original": "(\n    files: {\n      name: string;\n      uploadStatus: 'pending' | 'uploading' | 'complete' | 'error';\n      progress?: number; // from 0 to 100, if uploading\n      errorMessage?: string;\n    }[],\n  ) => void",
                    "resolved": "(files: { name: string; uploadStatus: \"error\" | \"pending\" | \"uploading\" | \"complete\"; progress?: number; errorMessage?: string; }[]) => void",
                    "references": {}
                },
                "required": false,
                "optional": true,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false
            },
            "uploadFileOverSize": {
                "type": "unknown",
                "attribute": "upload-file-over-size",
                "mutable": false,
                "complexType": {
                    "original": "() => void",
                    "resolved": "() => void",
                    "references": {}
                },
                "required": false,
                "optional": true,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false
            },
            "onChange": {
                "type": "unknown",
                "attribute": "on-change",
                "mutable": false,
                "complexType": {
                    "original": "(event: Event) => void",
                    "resolved": "(event: Event) => void",
                    "references": {
                        "Event": {
                            "location": "global",
                            "id": "global::Event"
                        }
                    }
                },
                "required": false,
                "optional": true,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false
            },
            "emitDeleteFile": {
                "type": "unknown",
                "attribute": "emit-delete-file",
                "mutable": false,
                "complexType": {
                    "original": "(\n    file: {\n      name: string;\n      uploadStatus: 'pending' | 'uploading' | 'complete' | 'error';\n      progress?: number; // from 0 to 100, if uploading\n      errorMessage?: string;\n    },\n    index: number,\n  ) => void",
                    "resolved": "(file: { name: string; uploadStatus: \"error\" | \"pending\" | \"uploading\" | \"complete\"; progress?: number; errorMessage?: string; }, index: number) => void",
                    "references": {}
                },
                "required": false,
                "optional": true,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false
            }
        };
    }
    static get states() {
        return {
            "files": {},
            "dragState": {}
        };
    }
}
