// @platformscode/core/dist/collection/components/dga-tabs/dga-tabs.js
import { h } from "@stencil/core";
import { SxStyleManager } from "../../utils/sx-style-manager";
import { Host } from "@stencil/core";
export class DgaTabs {
    constructor() {
        this.size = 'md';
        this.orientation = 'horizontal';
        this.external = false;
        this.flush = false;
        this.divider = false;
        this.tabsList = [];
        this.disabled = false;
        this.activeTab = 0;
        this.isMenuOpen = false; // State to control menu visibility
        this.toggleMenu = () => {
            this.isMenuOpen = !this.isMenuOpen;
            if (this.isMenuOpen) {
                document.addEventListener('click', this.handleOutsideClick);
            }
            else {
                document.removeEventListener('click', this.handleOutsideClick);
            }
        };
        this.closeMenu = () => {
            this.isMenuOpen = false;
            document.removeEventListener('click', this.handleOutsideClick);
        };
        this.handleOutsideClick = (event) => {
            const target = event.target;
            if (!this.el.contains(target)) {
                this.closeMenu();
            }
        };
        // Generate a unique ID if not provided
        this.componentId = `dga-button-${Math.random().toString(36).substring(2, 9)}`;
        // Computed inline styles
        this.rootStyles = {};
    }
    setActiveTab(index) {
        this.activeTab = index;
    }
    renderIcon(iconName, iconProps) {
        return h("dga-icon", Object.assign({ name: iconName }, iconProps));
    }
    handleSxChange() {
        this.processStyling();
    }
    componentWillLoad() {
        // Initialize style manager with element reference
        this.sxStyleManager = new SxStyleManager(this.componentId, this.el);
    }
    componentDidLoad() {
        // Process styling after the component is fully loaded and shadow DOM is ready
        this.processStyling();
    }
    componentDidUpdate() {
        // Re-process styling after updates
        this.processStyling();
    }
    processStyling() {
        if (!this.sx)
            return;
        // Handle both string and object formats
        const sxValue = typeof this.sx === 'string' ? JSON.parse(this.sx) : this.sx;
        this.rootStyles = this.sxStyleManager.processSxProp(sxValue);
    }
    disconnectedCallback() {
        var _a;
        // Clean up styles when component is removed
        (_a = this.sxStyleManager) === null || _a === void 0 ? void 0 : _a.cleanup();
    }
    render() {
        var _a;
        return (h(Host, { key: 'cf608207eb889e64245950d2cc1a19bd8c7b7ab8', id: this.componentId, style: this.rootStyles }, h("ul", { key: '3a04ac3e5a467cf891b25698dcc5b42979331cf9', class: `dga-tabs-list dga-tabs-list--${this.orientation} ${this.divider ? 'dga-tabs-list--divider' : ''} ${this.flush ? 'flush' : ''} ${this.extraClass}` }, (_a = this.tabsList) === null || _a === void 0 ? void 0 :
            _a.slice(0, 5).map((tab, index) => (h("li", { key: index }, this.external ? (h("a", { href: tab.link, class: `dga-tabs-list__item dga-tabs-list__item--${this.size} ${this.activeTab === index ? 'dga-tabs-list__item--active' : ''} ${this.disabled ? 'disabled' : ''}`, onClick: e => {
                    e.preventDefault();
                    !this.disabled && this.setActiveTab(index);
                } }, (tab === null || tab === void 0 ? void 0 : tab.tabIcon) && h("div", { class: "dga-tabs-list__icon" }, this.renderIcon(tab === null || tab === void 0 ? void 0 : tab.tabIcon, tab === null || tab === void 0 ? void 0 : tab.iconProps)), h("div", { class: "dga-tabs-list__label" }, tab === null || tab === void 0 ? void 0 : tab.label))) : (h("a", { href: "#", class: `dga-tabs-list__item dga-tabs-list__item--${this.size} ${this.activeTab === index ? 'dga-tabs-list__item--active' : ''} ${this.disabled ? 'disabled' : ''}`, onClick: e => {
                    e.preventDefault();
                    !this.disabled && this.setActiveTab(index);
                    (tab === null || tab === void 0 ? void 0 : tab.onClick) && (tab === null || tab === void 0 ? void 0 : tab.onClick(e));
                } }, (tab === null || tab === void 0 ? void 0 : tab.tabIcon) && h("div", { class: "dga-tabs-list__icon" }, this.renderIcon(tab === null || tab === void 0 ? void 0 : tab.tabIcon, tab === null || tab === void 0 ? void 0 : tab.iconProps)), h("div", { class: "dga-tabs-list__label" }, tab === null || tab === void 0 ? void 0 : tab.label)))))), this.orientation === 'horizontal' && this.tabsList.length > 5 && (h("li", { key: 'c22c5f055ae2324e10ed7d28c5473dc654c6de13', class: "breadcrumb-item ellipsis" }, h("dga-button", { key: '7663fda3c9ed1bb9871f7d6f434612af0a0dc36f', iconOnly: true, leadIcon: true, leadIconProps: {
                name: 'MoreHorizontalCircle01Icon',
                size: 17,
                variant: 'solid',
            }, label: "Button", onClick: () => {
                this.toggleMenu();
            }, size: "md", variant: "subtle", disabled: this.disabled, "aria-haspopup": "true", "aria-expanded": this.isMenuOpen ? 'true' : 'false' }), this.isMenuOpen && (h("div", { key: 'f7ddf8175c459667931837257d5e33c69d9bf7f8', class: "breadcrumb-dropdown" }, h("dga-tabs", { key: '5d2a089302384ef642128d223e96f8d2cf9350f7', size: "md", orientation: "vertical",
            // tabIcons: true,
            flush: false, external: true, divider: true, tabsList: this.tabsList.slice(5) }))))))));
    }
    static get is() { return "dga-tabs"; }
    static get originalStyleUrls() {
        return {
            "$": ["dga-tabs.scss"]
        };
    }
    static get styleUrls() {
        return {
            "$": ["dga-tabs.css"]
        };
    }
    static get properties() {
        return {
            "size": {
                "type": "string",
                "attribute": "size",
                "mutable": false,
                "complexType": {
                    "original": "'sm' | 'md' | 'lg'",
                    "resolved": "\"lg\" | \"md\" | \"sm\"",
                    "references": {}
                },
                "required": false,
                "optional": false,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false,
                "defaultValue": "'md'"
            },
            "orientation": {
                "type": "string",
                "attribute": "orientation",
                "mutable": false,
                "complexType": {
                    "original": "'horizontal' | 'vertical'",
                    "resolved": "\"horizontal\" | \"vertical\"",
                    "references": {}
                },
                "required": false,
                "optional": false,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false,
                "defaultValue": "'horizontal'"
            },
            "external": {
                "type": "boolean",
                "attribute": "external",
                "mutable": false,
                "complexType": {
                    "original": "boolean",
                    "resolved": "boolean",
                    "references": {}
                },
                "required": false,
                "optional": false,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false,
                "defaultValue": "false"
            },
            "flush": {
                "type": "boolean",
                "attribute": "flush",
                "mutable": false,
                "complexType": {
                    "original": "boolean",
                    "resolved": "boolean",
                    "references": {}
                },
                "required": false,
                "optional": false,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false,
                "defaultValue": "false"
            },
            "divider": {
                "type": "boolean",
                "attribute": "divider",
                "mutable": false,
                "complexType": {
                    "original": "boolean",
                    "resolved": "boolean",
                    "references": {}
                },
                "required": false,
                "optional": false,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false,
                "defaultValue": "false"
            },
            "tabsList": {
                "type": "unknown",
                "attribute": "tabs-list",
                "mutable": false,
                "complexType": {
                    "original": "ITab[]",
                    "resolved": "ITab[]",
                    "references": {
                        "ITab": {
                            "location": "local",
                            "path": "/Users/mohamedaboelezz/projects/Azm-X/platform-code-cross-platform-storybook/packages/core/src/components/dga-tabs/dga-tabs.tsx",
                            "id": "src/components/dga-tabs/dga-tabs.tsx::ITab"
                        }
                    }
                },
                "required": false,
                "optional": false,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "defaultValue": "[]"
            },
            "extraClass": {
                "type": "string",
                "attribute": "extra-class",
                "mutable": false,
                "complexType": {
                    "original": "string",
                    "resolved": "string",
                    "references": {}
                },
                "required": false,
                "optional": true,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false
            },
            "disabled": {
                "type": "boolean",
                "attribute": "disabled",
                "mutable": false,
                "complexType": {
                    "original": "boolean",
                    "resolved": "boolean",
                    "references": {}
                },
                "required": false,
                "optional": false,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false,
                "defaultValue": "false"
            },
            "sx": {
                "type": "string",
                "attribute": "sx",
                "mutable": false,
                "complexType": {
                    "original": "SxProps | string",
                    "resolved": "CSSProperties & NestedSelector | string",
                    "references": {
                        "SxProps": {
                            "location": "import",
                            "path": "../../utils/sx-system",
                            "id": "src/utils/sx-system.ts::SxProps"
                        }
                    }
                },
                "required": false,
                "optional": true,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false
            }
        };
    }
    static get states() {
        return {
            "activeTab": {},
            "isMenuOpen": {}
        };
    }
    static get elementRef() { return "el"; }
    static get watchers() {
        return [{
                "propName": "sx",
                "methodName": "handleSxChange"
            }];
    }
}
