// @platformscode/core/dist/collection/components/dga-codeSnippet/dga-codeSnippet.js
import { h } from "@stencil/core";
export class CodeSnippet {
    constructor() {
        var _a;
        this.snippets = {};
        this.showButton = false;
        this.type = 'multi-line';
        this.showSidebar = false;
        this.showTabList = false;
        this.showMore = false;
        this.activeTab = ((_a = Object.keys(this.snippets || {})) === null || _a === void 0 ? void 0 : _a[0]) || '';
    }
    componentDidLoad() {
        var _a;
        this.activeTab = ((_a = Object.keys(this.snippets || {})) === null || _a === void 0 ? void 0 : _a[0]) || '';
    }
    handleTabClick(tab) {
        this.activeTab = tab;
        this.showMore = false;
    }
    handleCopy() {
        navigator.clipboard.writeText(this.snippets[this.activeTab]);
    }
    renderWithLineNumbers(code) {
        var _a;
        const isMultiLine = this.type === 'multi-line';
        if (isMultiLine) {
            return (_a = code === null || code === void 0 ? void 0 : code.split('\n')) === null || _a === void 0 ? void 0 : _a.map((line, index) => (h("div", { key: index, class: "code-snippet-code-line" }, this.showSidebar && (h("div", { class: "code-snippet-number_container" }, h("span", { class: "code-snippet-line-number" }, index + 1))), h("span", { class: "code-snippet-line-content" }, line))));
        }
        else {
            return code
                .split('\n')
                .slice(0, 1)
                .map((line, index) => (h("div", { key: index, class: "code-snippet-code-line" }, h("span", { class: "code-snippet-line-content green" }, line))));
        }
    }
    render() {
        return (h("div", { key: 'f78045c8331a251ef51fcca43ebf25a3ef016d0d', class: "code-snippet-container" }, this.showTabList && (h("div", { key: 'd1def53ad4fd933e4c8efef340d4d4acf7c3e5b7', class: "code-snippet-tabs" }, h("dga-tabs", { key: '0c4e63d9d7d9ef131513d31eec449bc1bef843c7', divider: true, orientation: "horizontal", size: "lg", tabsList: Object.keys(this.snippets).map(tab => ({
                label: tab,
                onClick: () => this.handleTabClick(tab),
            })) }))), h("pre", { key: '5f8478b994d4c0a8b38fb0672a42d632c401acdb', class: "code-snippet-code-display" }, h("dga-button-v2", { key: 'b13f8f1dbe0b7d8ca3899dcf29f62489e05b3ead', size: "lg", iconOnly: true, variant: 'subtle', leadIcon: true, leadIconProps: {
                name: 'copy-01',
                size: 24,
                color: '#161616',
                variant: 'stroke',
            }, class: `code-snippet-icon ${this.type}`, onClick: () => this.handleCopy(), onColor: false }), h("code", { key: '8f50ebfb4710153e320f37a5a3cbea26ae0ddeea', style: {
                height: this.showButton && this.type === 'multi-line' && !this.showMore ? '150px' : 'auto',
            } }, this.renderWithLineNumbers(this.snippets[this.activeTab])), this.showButton && this.type === 'multi-line' && (h("div", { key: '97d1dc497f1b9e6bc13ff74dc3b6dffe169d6f85', class: "code-snippet-wrapper" }, h("dga-button-v2", { key: '6e843a04be23922f952f64b5f910d66c02699db1', label: this.showMore ? 'Show Less' : 'Show More', size: "md", trailIcon: true, variant: 'subtle', trailIconProps: { name: !this.showMore ? 'arrow-down-01' : 'arrow-up-01', size: 20, color: '#161616', variant: 'stroke' }, onClick: () => (this.showMore = !this.showMore), onColor: false }))))));
    }
    static get is() { return "dga-codesnippet"; }
    static get originalStyleUrls() {
        return {
            "$": ["./style.scss"]
        };
    }
    static get styleUrls() {
        return {
            "$": ["style.css"]
        };
    }
    static get properties() {
        return {
            "snippets": {
                "type": "unknown",
                "attribute": "snippets",
                "mutable": false,
                "complexType": {
                    "original": "{ [key: string]: string }",
                    "resolved": "{ [key: string]: string; }",
                    "references": {}
                },
                "required": false,
                "optional": false,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "defaultValue": "{}"
            },
            "showButton": {
                "type": "boolean",
                "attribute": "show-button",
                "mutable": false,
                "complexType": {
                    "original": "boolean",
                    "resolved": "boolean",
                    "references": {}
                },
                "required": false,
                "optional": false,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false,
                "defaultValue": "false"
            },
            "type": {
                "type": "string",
                "attribute": "type",
                "mutable": false,
                "complexType": {
                    "original": "'multi-line' | 'single'",
                    "resolved": "\"multi-line\" | \"single\"",
                    "references": {}
                },
                "required": false,
                "optional": false,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false,
                "defaultValue": "'multi-line'"
            },
            "showSidebar": {
                "type": "boolean",
                "attribute": "show-sidebar",
                "mutable": false,
                "complexType": {
                    "original": "boolean",
                    "resolved": "boolean",
                    "references": {}
                },
                "required": false,
                "optional": false,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false,
                "defaultValue": "false"
            },
            "showTabList": {
                "type": "boolean",
                "attribute": "show-tab-list",
                "mutable": false,
                "complexType": {
                    "original": "boolean",
                    "resolved": "boolean",
                    "references": {}
                },
                "required": false,
                "optional": false,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false,
                "defaultValue": "false"
            }
        };
    }
    static get states() {
        return {
            "showMore": {},
            "activeTab": {}
        };
    }
}
