// @platformscode/core/dist/collection/nds-components/dga-header/dga-header-action.js
import { Host } from "@stencil/core";
import { h } from "@stencil/core";
export class DgaHeaderAction {
    constructor() {
        this.iconPosition = 'inline';
        this.iconProps = {
            type: 'rounded',
            variant: 'stroke',
        };
        this.active = false;
        this.extraClasses = '';
        this.openSubMenu = false;
    }
    handleClick(event) {
        event.preventDefault();
        if (this.onClick) {
            this.onClick();
        }
    }
    componentWillLoad() {
        this.el.childNodes.forEach(item => {
            const el = item;
            if (el.tagName === 'DGA-HEADER-SUB-MENU' && el.attributes.getNamedItem('slot') === null) {
                el.setAttribute('slot', 'dga-header-sub-menu');
            }
        });
    }
    render() {
        return (h(Host, { key: '571a0a3a153d70f6c4d5ce3da0742018a66ca60e' }, h("li", { key: 'f244ff01420a7de99d38170d594d30b6b4438f30', onMouseEnter: () => {
                this.openSubMenu = true;
                //   this.active = true;
            }, onMouseLeave: () => {
                this.openSubMenu = false;
                //   this.active = false;
            } }, h("a", { key: '83581052a1523838823c7b010c8768cdccabf365', href: "#", class: `header-menu__item ${this.active ? 'header-menu__item--active' : ''} 
            
            ${this.iconPosition === 'inline' ? 'header-menu__item--inline' : 'header-menu__item--top'}
            
            ${this.extraClasses}`, onClick: this.handleClick.bind(this) }, h("span", { key: '485f04d8bc1eae8d187369df7aa59b47950fc500', class: "header-menu__item-arrow" }, h("dga-icon-v2", Object.assign({ key: '0f7a1b077f05674ba2b1e5167a52cab21a41f33f', name: this.icon }, this.iconProps))), h("span", { key: 'ebd2aa5f1b1a212145d46a0c28bf54c3f1e9d209', class: "header-menu__item-label" }, this.label)), this.openSubMenu && h("slot", { key: '948b45b2648ac72781b5e53836ed8fb47d7e59a1', name: "dga-header-sub-menu" }))));
    }
    static get is() { return "dga-header-action"; }
    static get encapsulation() { return "shadow"; }
    static get originalStyleUrls() {
        return {
            "$": ["dga-header-action.scss"]
        };
    }
    static get styleUrls() {
        return {
            "$": ["dga-header-action.css"]
        };
    }
    static get properties() {
        return {
            "label": {
                "type": "string",
                "attribute": "label",
                "mutable": false,
                "complexType": {
                    "original": "string",
                    "resolved": "string",
                    "references": {}
                },
                "required": false,
                "optional": true,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false
            },
            "iconPosition": {
                "type": "string",
                "attribute": "icon-position",
                "mutable": false,
                "complexType": {
                    "original": "'inline' | 'top'",
                    "resolved": "\"inline\" | \"top\"",
                    "references": {}
                },
                "required": false,
                "optional": true,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false,
                "defaultValue": "'inline'"
            },
            "icon": {
                "type": "string",
                "attribute": "icon",
                "mutable": false,
                "complexType": {
                    "original": "string",
                    "resolved": "string",
                    "references": {}
                },
                "required": false,
                "optional": true,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false
            },
            "iconProps": {
                "type": "unknown",
                "attribute": "icon-props",
                "mutable": false,
                "complexType": {
                    "original": "{\n    size?: number;\n    color?: string;\n    variant?: 'stroke' | 'solid' | 'duotone' | 'twotone' | 'bulk';\n    type?: 'rounded' | 'sharp' | 'standard';\n  }",
                    "resolved": "{ size?: number; color?: string; variant?: \"stroke\" | \"solid\" | \"duotone\" | \"twotone\" | \"bulk\"; type?: \"rounded\" | \"sharp\" | \"standard\"; }",
                    "references": {}
                },
                "required": false,
                "optional": false,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "defaultValue": "{\n    type: 'rounded',\n    variant: 'stroke',\n  }"
            },
            "active": {
                "type": "boolean",
                "attribute": "active",
                "mutable": false,
                "complexType": {
                    "original": "boolean",
                    "resolved": "boolean",
                    "references": {}
                },
                "required": false,
                "optional": true,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false,
                "defaultValue": "false"
            },
            "onClick": {
                "type": "unknown",
                "attribute": "on-click",
                "mutable": false,
                "complexType": {
                    "original": "() => void",
                    "resolved": "() => void",
                    "references": {}
                },
                "required": false,
                "optional": false,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false
            },
            "extraClasses": {
                "type": "string",
                "attribute": "extra-classes",
                "mutable": false,
                "complexType": {
                    "original": "string",
                    "resolved": "string",
                    "references": {}
                },
                "required": false,
                "optional": true,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false,
                "defaultValue": "''"
            }
        };
    }
    static get states() {
        return {
            "openSubMenu": {}
        };
    }
    static get elementRef() { return "el"; }
}
