// @platformscode/core/dist/collection/nds-components/dga-header/dga-header-sub-menu-item.js
import { h } from "@stencil/core";
export class DgaHeaderSubMenuItem {
    constructor() {
        this.linkStyle = 2;
        this.iconProps = {
            size: 24,
            type: 'rounded',
            variant: 'stroke',
        };
    }
    render() {
        const backgroundClass = this.background === 'Brand' ? 'sub-menu__link--on-color' : '';
        const linkStyleClass = this.linkStyle && this.linkStyle > 0 ? (this.linkStyle === 2 ? 'sub-menu__link-icon--bg' : 'sub-menu__link-icon') : '';
        return (h("a", { key: '1a0f4096a3c293309f46c5cceb5ae75910cfbf70', href: this.link, class: `sub-menu__link ${backgroundClass}` }, this.linkStyle && this.linkStyle > 0 ? h("div", { class: linkStyleClass }, this.icon && h("dga-icon-v2", Object.assign({ name: this.icon }, this.iconProps))) : null, h("dga-flex", { key: '2f29c50c537df8be70a327b5d3c33e4a72a33087', direction: 'column', spacing: 4 }, h("dga-flex", { key: 'c9ebad7beb3543bf6764c0d34483cd321025a70e', direction: 'row', align: 'center', spacing: 8 }, h("span", { key: 'e95730327156f7b8edd03fd8f58d1d3dfad88e4d', class: "sub-menu__link-label" }, this.label), h("slot", { key: 'a9fa479dde28ca909fb69d4a2a9076fcbb3707f0', name: "tag" })), this.helperText && h("span", { key: '49560e47ae2f90a8b58e8a3af62d858ff7081583', class: "sub-menu__link-helper" }, this.helperText))));
    }
    static get is() { return "dga-header-sub-menu-item"; }
    static get encapsulation() { return "shadow"; }
    static get originalStyleUrls() {
        return {
            "$": ["dga-header-sub-menu-item.scss"]
        };
    }
    static get styleUrls() {
        return {
            "$": ["dga-header-sub-menu-item.css"]
        };
    }
    static get properties() {
        return {
            "background": {
                "type": "string",
                "attribute": "background",
                "mutable": false,
                "complexType": {
                    "original": "string",
                    "resolved": "string",
                    "references": {}
                },
                "required": false,
                "optional": true,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false
            },
            "link": {
                "type": "string",
                "attribute": "link",
                "mutable": false,
                "complexType": {
                    "original": "string",
                    "resolved": "string",
                    "references": {}
                },
                "required": false,
                "optional": true,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false
            },
            "linkStyle": {
                "type": "number",
                "attribute": "link-style",
                "mutable": false,
                "complexType": {
                    "original": "number",
                    "resolved": "number",
                    "references": {}
                },
                "required": false,
                "optional": true,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false,
                "defaultValue": "2"
            },
            "label": {
                "type": "string",
                "attribute": "label",
                "mutable": false,
                "complexType": {
                    "original": "string",
                    "resolved": "string",
                    "references": {}
                },
                "required": false,
                "optional": true,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false
            },
            "helperText": {
                "type": "string",
                "attribute": "helper-text",
                "mutable": false,
                "complexType": {
                    "original": "string",
                    "resolved": "string",
                    "references": {}
                },
                "required": false,
                "optional": true,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false
            },
            "icon": {
                "type": "string",
                "attribute": "icon",
                "mutable": false,
                "complexType": {
                    "original": "string",
                    "resolved": "string",
                    "references": {}
                },
                "required": false,
                "optional": true,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false
            },
            "iconProps": {
                "type": "unknown",
                "attribute": "icon-props",
                "mutable": false,
                "complexType": {
                    "original": "{\n      size?: number;\n      color?: string;\n      variant?: 'stroke' | 'solid' | 'duotone' | 'twotone' | 'bulk';\n      type?: 'rounded' | 'sharp' | 'standard';\n    }",
                    "resolved": "{ size?: number; color?: string; variant?: \"stroke\" | \"solid\" | \"duotone\" | \"twotone\" | \"bulk\"; type?: \"rounded\" | \"sharp\" | \"standard\"; }",
                    "references": {}
                },
                "required": false,
                "optional": false,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "defaultValue": "{\n    size: 24,\n      type: 'rounded',\n      variant: 'stroke',\n    }"
            }
        };
    }
}
