// @platformscode/core/dist/collection/components/dga-slider/dga-slider.js
import { h, Host } from "@stencil/core";
import { SxHandler } from "../../utils/sx-handler";
export class DgaSlider {
    constructor() {
        this.min = 0;
        this.max = 100;
        this.step = 1;
        this.range = true;
        this.showLabel = true;
        this.showHelperText = true;
        // @Prop() rangeValues?: [number,number]=[this.min,this.max];
        this.size = 'small';
    }
    // private minPos: number;
    // private maxPos: number;
    // private railColorStyle: { left: string; width: string };
    calculatePositions() {
        const minPos = ((this.minValue - this.min) / (this.max - this.min)) * 100;
        const maxPos = ((this.maxValue - this.min) / (this.max - this.min)) * 100;
        this.minPos = minPos;
        this.maxPos = maxPos;
        this.railColorStyle = !this.range ? { left: '0%', width: `${minPos}%` } : { left: `${minPos}%`, width: `${maxPos - minPos}%` };
    }
    componentWillLoad() {
        this.minValue = this.initialMin !== undefined ? this.initialMin : this.min;
        this.maxValue = this.initialMax !== undefined ? this.initialMax : this.max;
        this.calculatePositions();
        this.sxHandler = new SxHandler(this.el, 'dga-slider');
    }
    watchMinValue(newValue) {
        this.minValue = newValue;
        this.calculatePositions();
    }
    watchMaxValue(newValue) {
        this.maxValue = newValue;
        this.calculatePositions();
    }
    watchRangeValue() {
        this.calculatePositions();
    }
    watchInitialMin() {
        this.minValue = this.initialMin !== undefined ? this.initialMin : this.min;
        this.calculatePositions();
    }
    watchInitialMax() {
        this.maxValue = this.initialMax !== undefined ? this.initialMax : this.max;
        this.calculatePositions();
    }
    handleMinChange(e) {
        const target = e.target;
        const newMinVal = this.range ? Math.min(+target.value, this.maxValue - this.step) : Math.min(+target.value);
        this.minValue = newMinVal;
        if (this.onChange) {
            this.onChange({ max: this.maxValue, min: this.minValue });
        }
        this.calculatePositions();
    }
    handleMaxChange(e) {
        const target = e.target;
        const newMaxVal = Math.max(+target.value, this.minValue + this.step);
        this.maxValue = newMaxVal;
        if (this.onChange) {
            this.onChange({ max: this.maxValue, min: this.minValue });
        }
        this.calculatePositions();
    }
    componentDidLoad() {
        this.sxHandler.processSx(this.sx);
    }
    sxChanged() {
        this.sxHandler.processSx(this.sx);
    }
    disconnectedCallback() {
        this.sxHandler.cleanup();
    }
    render() {
        return (h(Host, { key: 'ff9dc523cd6181d337141fd333aeb1df945799a1', id: this.sxHandler.getId(), style: this.sxHandler.getStyles() }, h("div", { key: '3b4dbc21a48b6a76be9c186f977bcdf8b2781a2b', class: `range-slider-wrapper ${this.size}` }, h("div", { key: '0efbaf84a93f8f5f2c468f8ec5e0c6a022f1bf91', class: "range-slider-input-wrapper" }, h("input", { key: '04cfbd25703697683cadaf8f807b3115a64fd9ef', class: "range-slider-input", type: "range", value: this.minValue, min: this.min, max: this.max, step: this.step, onInput: this.handleMinChange.bind(this) }), this.range && (h("input", { key: '048fc56c9402d23483569733ce606adcd97fe5c9', class: "range-slider-input", type: "range", value: this.maxValue, min: this.min, max: this.max, step: this.step, onInput: this.handleMaxChange.bind(this) }))), h("div", { key: '64e1b7f65d4cae5f2eb6967500dfb12ed55399cb', class: "range-slider-container" }, this.showLabel && h("h3", { key: '2e710990519a3de60980bfacddc9af5aabbb7183' }, this.label), h("div", { key: 'b423c915987b8ea99685a3407580401797da92f7', class: "range-slider-control-wrapper" }, h("div", { key: '25baaa3f5abfc0ffbc09b10d38b5bf354ee24a85', class: "range-slider-control", style: { left: `${this.minPos}%` } }), h("div", { key: 'c733c9b76fdcf42fcc9596611e9c57b562c33c97', class: "range-slider-rail" }, this.range && h("div", { key: '9b4b12d4e8466e28a87d65c98b5a360a25fc3ed7', class: "range-slider-number_left" }, this.minValue), h("div", { key: 'b99e23abc452218f1efd84948abf4bf3651b90d5', class: "range-slider-inner-rail", style: this.railColorStyle }), this.range && h("div", { key: 'f4f6264662658d9b601b4a9382842f83dd19f964', class: "range-slider-number_right" }, this.maxValue), !this.range && h("div", { key: '8a8c2f7fd644f1b6664ffc826ebabfb367e18197', class: "range-slider-number_right" }, this.minValue)), this.range && h("div", { key: 'af9b65fe0501390798a471029b00cd0a3556e445', class: "range-slider-control", style: { left: `${this.maxPos}%` } }), this.showHelperText && this.helperText && (h("div", { key: 'e90ebcb2cc1cc9f09b62754b36c97229d3cf39aa', class: "range-slider-note" }, h("div", { key: '627e81a62ec40e1eb37351d231f49b000951cac4', class: "range-slider-note-icon" }, h("svg", { key: '980debb7f227d1fab87558d4af0bcc7587c0f147', width: "16", height: "16", viewBox: "0 0 16 16", fill: "none", xmlns: "http://www.w3.org/2000/svg" }, h("g", { key: '5e36241d637e91275f5433211993ece5a60907b8', "clip-path": "url(#clip0_17869_405)" }, h("path", { key: '63e0df5676fe1d5ca2ffbc1366fe238d5055515b', d: "M6.06004 6.00001C6.21678 5.55446 6.52614 5.17875 6.93334 4.93943C7.34055 4.70012 7.8193 4.61264 8.28483 4.69248C8.75035 4.77233 9.17259 5.01436 9.47676 5.3757C9.78093 5.73703 9.94741 6.19436 9.94671 6.66668C9.94671 8.00001 7.94671 8.66668 7.94671 8.66668M8.00004 11.3333H8.00671M14.6667 8.00001C14.6667 11.6819 11.6819 14.6667 8.00004 14.6667C4.31814 14.6667 1.33337 11.6819 1.33337 8.00001C1.33337 4.31811 4.31814 1.33334 8.00004 1.33334C11.6819 1.33334 14.6667 4.31811 14.6667 8.00001Z", stroke: "#384250", "stroke-width": "1.5", "stroke-linecap": "round", "stroke-linejoin": "round" })), h("defs", { key: 'f2c26b39e97fbd08326ee2bbbb365268ea830bd7' }, h("clipPath", { key: '37db6c22b87ec78ee796d7aa41873a1db2db8ca5', id: "clip0_17869_405" }, h("rect", { key: '3123c60bf33523eccad40ac0246f384b7f2d6186', width: "16", height: "16", fill: "white" }))))), h("div", { key: 'b6a58b1cf00d05d80dfaf654cf5ce1f9662587f1', class: "range-slider-note-content" }, h("p", { key: '5882496ea9e07a5ef38121732bf01883aa47d1b8' }, this.helperText)))))))));
    }
    static get is() { return "dga-slider"; }
    static get encapsulation() { return "shadow"; }
    static get originalStyleUrls() {
        return {
            "$": ["dga-slider.scss"]
        };
    }
    static get styleUrls() {
        return {
            "$": ["dga-slider.css"]
        };
    }
    static get properties() {
        return {
            "min": {
                "type": "number",
                "attribute": "min",
                "mutable": false,
                "complexType": {
                    "original": "number",
                    "resolved": "number",
                    "references": {}
                },
                "required": false,
                "optional": false,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false,
                "defaultValue": "0"
            },
            "max": {
                "type": "number",
                "attribute": "max",
                "mutable": false,
                "complexType": {
                    "original": "number",
                    "resolved": "number",
                    "references": {}
                },
                "required": false,
                "optional": false,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false,
                "defaultValue": "100"
            },
            "step": {
                "type": "number",
                "attribute": "step",
                "mutable": false,
                "complexType": {
                    "original": "number",
                    "resolved": "number",
                    "references": {}
                },
                "required": false,
                "optional": false,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false,
                "defaultValue": "1"
            },
            "range": {
                "type": "boolean",
                "attribute": "range",
                "mutable": false,
                "complexType": {
                    "original": "boolean",
                    "resolved": "boolean",
                    "references": {}
                },
                "required": false,
                "optional": false,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false,
                "defaultValue": "true"
            },
            "label": {
                "type": "string",
                "attribute": "label",
                "mutable": false,
                "complexType": {
                    "original": "string",
                    "resolved": "string",
                    "references": {}
                },
                "required": false,
                "optional": false,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false
            },
            "showLabel": {
                "type": "boolean",
                "attribute": "show-label",
                "mutable": false,
                "complexType": {
                    "original": "boolean",
                    "resolved": "boolean",
                    "references": {}
                },
                "required": false,
                "optional": false,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false,
                "defaultValue": "true"
            },
            "showHelperText": {
                "type": "boolean",
                "attribute": "show-helper-text",
                "mutable": false,
                "complexType": {
                    "original": "boolean",
                    "resolved": "boolean",
                    "references": {}
                },
                "required": false,
                "optional": false,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false,
                "defaultValue": "true"
            },
            "helperText": {
                "type": "string",
                "attribute": "helper-text",
                "mutable": false,
                "complexType": {
                    "original": "string",
                    "resolved": "string",
                    "references": {}
                },
                "required": false,
                "optional": false,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false
            },
            "size": {
                "type": "string",
                "attribute": "size",
                "mutable": false,
                "complexType": {
                    "original": "'small' | 'medium'",
                    "resolved": "\"medium\" | \"small\"",
                    "references": {}
                },
                "required": false,
                "optional": false,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false,
                "defaultValue": "'small'"
            },
            "initialMin": {
                "type": "number",
                "attribute": "initial-min",
                "mutable": false,
                "complexType": {
                    "original": "number",
                    "resolved": "number",
                    "references": {}
                },
                "required": false,
                "optional": false,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false
            },
            "initialMax": {
                "type": "number",
                "attribute": "initial-max",
                "mutable": false,
                "complexType": {
                    "original": "number",
                    "resolved": "number",
                    "references": {}
                },
                "required": false,
                "optional": false,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false
            },
            "onChange": {
                "type": "unknown",
                "attribute": "on-change",
                "mutable": false,
                "complexType": {
                    "original": "(props: { min: number; max: number }) => void",
                    "resolved": "(props: { min: number; max: number; }) => void",
                    "references": {}
                },
                "required": false,
                "optional": true,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false
            },
            "sx": {
                "type": "string",
                "attribute": "sx",
                "mutable": false,
                "complexType": {
                    "original": "SxProps | string | undefined",
                    "resolved": "SxProps | string",
                    "references": {
                        "SxProps": {
                            "location": "import",
                            "path": "../../utils/sx-handler",
                            "id": "src/utils/sx-handler.ts::SxProps"
                        }
                    }
                },
                "required": false,
                "optional": true,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false
            }
        };
    }
    static get states() {
        return {
            "minValue": {},
            "maxValue": {},
            "minPos": {},
            "maxPos": {},
            "railColorStyle": {}
        };
    }
    static get elementRef() { return "el"; }
    static get watchers() {
        return [{
                "propName": "min",
                "methodName": "watchMinValue"
            }, {
                "propName": "max",
                "methodName": "watchMaxValue"
            }, {
                "propName": "range",
                "methodName": "watchRangeValue"
            }, {
                "propName": "initialMin",
                "methodName": "watchInitialMin"
            }, {
                "propName": "initialMax",
                "methodName": "watchInitialMax"
            }, {
                "propName": "sx",
                "methodName": "sxChanged"
            }];
    }
}
