// @platformscode/core/dist/collection/components/dga-skeleton/dga-square-skeleton.js
import { h } from "@stencil/core";
export class SquareSkeleton {
    constructor() {
        this.type = '1';
        this.width = '80px';
    }
    getBorderRadius() {
        const borderRadiusMapper = {
            '24px': 'rounded-[2px]',
            '64px': 'rounded-[4px]',
            '48px': 'rounded-[2px]',
            '80px': 'rounded-[4px]',
            '120px': 'rounded-[8px]',
            '170px': 'rounded-[8px]',
            '240px': 'rounded-[16px]',
        };
        return borderRadiusMapper[this.width] || '';
    }
    render() {
        return (h("div", { key: '7b34898c6abef2291a6c6d3ef5348a79fa9f7a9d' }, h("div", { key: 'b014a4229f90fe9646b4f6a0bcdde7502bdc4609', style: { width: this.width, height: this.width }, class: `square_skeleton type${this.type} ${this.getBorderRadius()}` })));
    }
    static get is() { return "dga-square-skeleton"; }
    static get encapsulation() { return "shadow"; }
    static get originalStyleUrls() {
        return {
            "$": ["./style.scss"]
        };
    }
    static get styleUrls() {
        return {
            "$": ["style.css"]
        };
    }
    static get properties() {
        return {
            "type": {
                "type": "string",
                "attribute": "type",
                "mutable": false,
                "complexType": {
                    "original": "'1' | '2'",
                    "resolved": "\"1\" | \"2\"",
                    "references": {}
                },
                "required": false,
                "optional": false,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false,
                "defaultValue": "'1'"
            },
            "width": {
                "type": "string",
                "attribute": "width",
                "mutable": false,
                "complexType": {
                    "original": "'24px' | '64px' | '48px' | '80px' | '120px' | '170px' | '240px'",
                    "resolved": "\"120px\" | \"170px\" | \"240px\" | \"24px\" | \"48px\" | \"64px\" | \"80px\"",
                    "references": {}
                },
                "required": false,
                "optional": false,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false,
                "defaultValue": "'80px'"
            }
        };
    }
}
