// @platformscode/core/dist/collection/components/dga-skeleton/dga-line-skeleton.js
import { h } from "@stencil/core";
export class LineSkeleton {
    constructor() {
        this.size = 'lg';
        this.type = '1';
    }
    render() {
        return (h("div", { key: '3387a0e823a08659b101c7a00656711e3898a954', class: `line_text ${this.size} type${this.type}` }));
    }
    static get is() { return "dga-line-skeleton"; }
    static get encapsulation() { return "shadow"; }
    static get originalStyleUrls() {
        return {
            "$": ["./style.scss"]
        };
    }
    static get styleUrls() {
        return {
            "$": ["style.css"]
        };
    }
    static get properties() {
        return {
            "size": {
                "type": "string",
                "attribute": "size",
                "mutable": false,
                "complexType": {
                    "original": "'lg' | 'sm'",
                    "resolved": "\"lg\" | \"sm\"",
                    "references": {}
                },
                "required": false,
                "optional": false,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false,
                "defaultValue": "'lg'"
            },
            "type": {
                "type": "string",
                "attribute": "type",
                "mutable": false,
                "complexType": {
                    "original": "'1' | '2'",
                    "resolved": "\"1\" | \"2\"",
                    "references": {}
                },
                "required": false,
                "optional": false,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false,
                "defaultValue": "'1'"
            }
        };
    }
}
