// @platformscode/core/dist/collection/nds-components/dga-header/dga-header.js
import { h } from "@stencil/core";
export class DgaHeader {
    constructor() {
        this.fullWidth = true;
        this.divider = true;
        this.sticky = false;
    }
    render() {
        return (h("header", { key: '012d2f08f9a693323ea77fe8a8a6ae97d38669b8', class: `header ${this.divider ? 'header--divider' : ''} ${this.sticky ? 'header--sticky' : ''}` }, h("nav", { key: 'b41738066ed14b8b3803c1bd09c4369020354d44', class: `header-nav--${this.fullWidth ? 'full' : 'fixed'}` }, h("slot", { key: '065a6b9877d0c54a4c925f12b68ff8ab199f1a60' }))));
    }
    static get is() { return "dga-header-vv2"; }
    static get encapsulation() { return "shadow"; }
    static get originalStyleUrls() {
        return {
            "$": ["dga-header.scss"]
        };
    }
    static get styleUrls() {
        return {
            "$": ["dga-header.css"]
        };
    }
    static get properties() {
        return {
            "fullWidth": {
                "type": "boolean",
                "attribute": "full-width",
                "mutable": false,
                "complexType": {
                    "original": "boolean",
                    "resolved": "boolean",
                    "references": {}
                },
                "required": false,
                "optional": false,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false,
                "defaultValue": "true"
            },
            "divider": {
                "type": "boolean",
                "attribute": "divider",
                "mutable": false,
                "complexType": {
                    "original": "boolean",
                    "resolved": "boolean",
                    "references": {}
                },
                "required": false,
                "optional": false,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false,
                "defaultValue": "true"
            },
            "sticky": {
                "type": "boolean",
                "attribute": "sticky",
                "mutable": false,
                "complexType": {
                    "original": "boolean",
                    "resolved": "boolean",
                    "references": {}
                },
                "required": false,
                "optional": false,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false,
                "defaultValue": "false"
            }
        };
    }
}
