// @platformscode/core/dist/collection/nds-components/dga-header/dga-logo-placeholder.js
import { h, Host } from "@stencil/core";
export class DgaLogoPlaceholder {
    constructor() {
        this.src = "https://dga-nds-fbhtx.ondigitalocean.app/mobile-logo.svg";
        /** Layout direction (LTR or RTL) */
        this.layout = 'side';
        /** Logo size (sm, md, lg) */
        this.size = 'md';
    }
    // private getSize(): string {
    //   const sizes = {
    //     sm: '60px',
    //     md: '80px',
    //     lg: '100px',
    //   };
    //   return sizes[this.size] || '80px';
    // }
    render() {
        // const logoSize = this.getSize();
        return (h(Host, { key: 'fcc36b4e9d5cb8ce754cbe88293269e2d71ffbfe' }, h("div", { key: '9d7110482df3debe19d3e1120b199b1b365ccfb2', class: `logo-wrapper logo-placeholder--${this.size} logo-placeholder--${this.layout}` }, h("img", { key: '173bf3b69c456e97c59d725209dedeac438bd362', src: this.src, alt: "Platform Logo",
            // style={{ width: logoSize, height: logoSize }}
            class: "logo-image" }), h("span", { key: '0451724ba37be0fb485ab10f56776b467f12b76e', class: "logo-text" }, this.text))));
    }
    static get is() { return "dga-logo-placeholder"; }
    static get encapsulation() { return "shadow"; }
    static get originalStyleUrls() {
        return {
            "$": ["dga-logo-placeholder.scss"]
        };
    }
    static get styleUrls() {
        return {
            "$": ["dga-logo-placeholder.css"]
        };
    }
    static get properties() {
        return {
            "src": {
                "type": "string",
                "attribute": "src",
                "mutable": false,
                "complexType": {
                    "original": "string",
                    "resolved": "string",
                    "references": {}
                },
                "required": false,
                "optional": false,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false,
                "defaultValue": "\"https://dga-nds-fbhtx.ondigitalocean.app/mobile-logo.svg\""
            },
            "text": {
                "type": "string",
                "attribute": "text",
                "mutable": false,
                "complexType": {
                    "original": "string",
                    "resolved": "string",
                    "references": {}
                },
                "required": false,
                "optional": false,
                "docs": {
                    "tags": [],
                    "text": "Text to display next to the logo"
                },
                "getter": false,
                "setter": false,
                "reflect": false
            },
            "layout": {
                "type": "string",
                "attribute": "layout",
                "mutable": false,
                "complexType": {
                    "original": "'top' | 'side'",
                    "resolved": "\"side\" | \"top\"",
                    "references": {}
                },
                "required": false,
                "optional": false,
                "docs": {
                    "tags": [],
                    "text": "Layout direction (LTR or RTL)"
                },
                "getter": false,
                "setter": false,
                "reflect": false,
                "defaultValue": "'side'"
            },
            "size": {
                "type": "string",
                "attribute": "size",
                "mutable": false,
                "complexType": {
                    "original": "'sm' | 'md' | 'lg'",
                    "resolved": "\"lg\" | \"md\" | \"sm\"",
                    "references": {}
                },
                "required": false,
                "optional": false,
                "docs": {
                    "tags": [],
                    "text": "Logo size (sm, md, lg)"
                },
                "getter": false,
                "setter": false,
                "reflect": false,
                "defaultValue": "'md'"
            }
        };
    }
}
