// @platformscode/core/dist/collection/nds-components/dga-header/dga-header-sub-menu-item.js
import { h } from "@stencil/core";
export class DgaHeaderSubMenuItem {
    constructor() {
        this.linkStyle = 2;
        this.iconProps = {
            size: 24,
            type: 'rounded',
            variant: 'stroke',
        };
    }
    render() {
        const backgroundClass = this.background === 'Brand' ? 'sub-menu__link--on-color' : '';
        const linkStyleClass = this.linkStyle && this.linkStyle > 0 ? (this.linkStyle === 2 ? 'sub-menu__link-icon--bg' : 'sub-menu__link-icon') : '';
        return (h("a", { key: 'c840b0e0c1e273268859cbc2a390d79fefdb6d9f', href: this.link, class: `sub-menu__link ${backgroundClass}` }, this.linkStyle && this.linkStyle > 0 ? h("div", { class: linkStyleClass }, this.icon && h("dga-icon-v2", Object.assign({ name: this.icon }, this.iconProps))) : null, h("dga-flex", { key: '249974c764ab2560fc8ecb466eb7603a90fec463', direction: 'column', spacing: 4 }, h("dga-flex", { key: '4c09e995518c9559619ec73e5edd45407b58b2ea', direction: 'row', align: 'center', spacing: 8 }, h("span", { key: '1d1f53055212affac702feb8751f8f0256234942', class: "sub-menu__link-label" }, this.label), h("slot", { key: '862b66a27ebbf37dcba96d5cbe13b677d6391837', name: "tag" })), this.helperText && h("span", { key: '7e65f6603c17310a00ba6b5c774a15bb0203606d', class: "sub-menu__link-helper" }, this.helperText))));
    }
    static get is() { return "dga-header-sub-menu-item"; }
    static get encapsulation() { return "shadow"; }
    static get originalStyleUrls() {
        return {
            "$": ["dga-header-sub-menu-item.scss"]
        };
    }
    static get styleUrls() {
        return {
            "$": ["dga-header-sub-menu-item.css"]
        };
    }
    static get properties() {
        return {
            "background": {
                "type": "string",
                "attribute": "background",
                "mutable": false,
                "complexType": {
                    "original": "string",
                    "resolved": "string",
                    "references": {}
                },
                "required": false,
                "optional": true,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false
            },
            "link": {
                "type": "string",
                "attribute": "link",
                "mutable": false,
                "complexType": {
                    "original": "string",
                    "resolved": "string",
                    "references": {}
                },
                "required": false,
                "optional": true,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false
            },
            "linkStyle": {
                "type": "number",
                "attribute": "link-style",
                "mutable": false,
                "complexType": {
                    "original": "number",
                    "resolved": "number",
                    "references": {}
                },
                "required": false,
                "optional": true,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false,
                "defaultValue": "2"
            },
            "label": {
                "type": "string",
                "attribute": "label",
                "mutable": false,
                "complexType": {
                    "original": "string",
                    "resolved": "string",
                    "references": {}
                },
                "required": false,
                "optional": true,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false
            },
            "helperText": {
                "type": "string",
                "attribute": "helper-text",
                "mutable": false,
                "complexType": {
                    "original": "string",
                    "resolved": "string",
                    "references": {}
                },
                "required": false,
                "optional": true,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false
            },
            "icon": {
                "type": "string",
                "attribute": "icon",
                "mutable": false,
                "complexType": {
                    "original": "string",
                    "resolved": "string",
                    "references": {}
                },
                "required": false,
                "optional": true,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false
            },
            "iconProps": {
                "type": "unknown",
                "attribute": "icon-props",
                "mutable": false,
                "complexType": {
                    "original": "{\n      size?: number;\n      color?: string;\n      variant?: 'stroke' | 'solid' | 'duotone' | 'twotone' | 'bulk';\n      type?: 'rounded' | 'sharp' | 'standard';\n    }",
                    "resolved": "{ size?: number; color?: string; variant?: \"stroke\" | \"solid\" | \"duotone\" | \"twotone\" | \"bulk\"; type?: \"rounded\" | \"sharp\" | \"standard\"; }",
                    "references": {}
                },
                "required": false,
                "optional": false,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "defaultValue": "{\n    size: 24,\n      type: 'rounded',\n      variant: 'stroke',\n    }"
            }
        };
    }
}
