// @platformscode/core/dist/collection/nds-components/dga-header/dga-header-action.js
import { Host } from "@stencil/core";
import { h } from "@stencil/core";
export class DgaHeaderAction {
    constructor() {
        this.iconPosition = 'inline';
        this.iconProps = {
            type: 'rounded',
            variant: 'stroke',
        };
        this.active = false;
        this.extraClasses = '';
        this.openSubMenu = false;
    }
    handleClick(event) {
        event.preventDefault();
        if (this.onClick) {
            this.onClick();
        }
    }
    componentWillLoad() {
        this.el.childNodes.forEach(item => {
            const el = item;
            if (el.tagName === 'DGA-HEADER-SUB-MENU' && el.attributes.getNamedItem('slot') === null) {
                el.setAttribute('slot', 'dga-header-sub-menu');
            }
        });
    }
    render() {
        return (h(Host, { key: 'f5112ca33fbbc173767780449e1457e373100594' }, h("li", { key: 'c09017454334bfe998c1c768cd2b12d428207983', onMouseEnter: () => {
                this.openSubMenu = true;
                //   this.active = true;
            }, onMouseLeave: () => {
                this.openSubMenu = false;
                //   this.active = false;
            } }, h("a", { key: '2aba9c368d5f057b16ac2a48d448a4f9fede2cc7', href: "#", class: `header-menu__item ${this.active ? 'header-menu__item--active' : ''} 
            
            ${this.iconPosition === 'inline' ? 'header-menu__item--inline' : 'header-menu__item--top'}
            
            ${this.extraClasses}`, onClick: this.handleClick.bind(this) }, h("span", { key: 'b34fe765105b174ea958745c2153fb5dd7aa8895', class: "header-menu__item-arrow" }, h("dga-icon-v2", Object.assign({ key: '7db7ec8a19cb51cb57ddfbc7848a1fec8e1da1ac', name: this.icon }, this.iconProps))), h("span", { key: 'd562fd27b0f9d020b1dd7548fc2ff0951d283972', class: "header-menu__item-label" }, this.label)), this.openSubMenu && h("slot", { key: '3e61bed3f1f5d4ab6158e8a12d1fbc2a1a556ee0', name: "dga-header-sub-menu" }))));
    }
    static get is() { return "dga-header-action"; }
    static get encapsulation() { return "shadow"; }
    static get originalStyleUrls() {
        return {
            "$": ["dga-header-action.scss"]
        };
    }
    static get styleUrls() {
        return {
            "$": ["dga-header-action.css"]
        };
    }
    static get properties() {
        return {
            "label": {
                "type": "string",
                "attribute": "label",
                "mutable": false,
                "complexType": {
                    "original": "string",
                    "resolved": "string",
                    "references": {}
                },
                "required": false,
                "optional": true,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false
            },
            "iconPosition": {
                "type": "string",
                "attribute": "icon-position",
                "mutable": false,
                "complexType": {
                    "original": "'inline' | 'top'",
                    "resolved": "\"inline\" | \"top\"",
                    "references": {}
                },
                "required": false,
                "optional": true,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false,
                "defaultValue": "'inline'"
            },
            "icon": {
                "type": "string",
                "attribute": "icon",
                "mutable": false,
                "complexType": {
                    "original": "string",
                    "resolved": "string",
                    "references": {}
                },
                "required": false,
                "optional": true,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false
            },
            "iconProps": {
                "type": "unknown",
                "attribute": "icon-props",
                "mutable": false,
                "complexType": {
                    "original": "{\n    size?: number;\n    color?: string;\n    variant?: 'stroke' | 'solid' | 'duotone' | 'twotone' | 'bulk';\n    type?: 'rounded' | 'sharp' | 'standard';\n  }",
                    "resolved": "{ size?: number; color?: string; variant?: \"stroke\" | \"solid\" | \"duotone\" | \"twotone\" | \"bulk\"; type?: \"rounded\" | \"sharp\" | \"standard\"; }",
                    "references": {}
                },
                "required": false,
                "optional": false,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "defaultValue": "{\n    type: 'rounded',\n    variant: 'stroke',\n  }"
            },
            "active": {
                "type": "boolean",
                "attribute": "active",
                "mutable": false,
                "complexType": {
                    "original": "boolean",
                    "resolved": "boolean",
                    "references": {}
                },
                "required": false,
                "optional": true,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false,
                "defaultValue": "false"
            },
            "onClick": {
                "type": "unknown",
                "attribute": "on-click",
                "mutable": false,
                "complexType": {
                    "original": "() => void",
                    "resolved": "() => void",
                    "references": {}
                },
                "required": false,
                "optional": false,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false
            },
            "extraClasses": {
                "type": "string",
                "attribute": "extra-classes",
                "mutable": false,
                "complexType": {
                    "original": "string",
                    "resolved": "string",
                    "references": {}
                },
                "required": false,
                "optional": true,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false,
                "defaultValue": "''"
            }
        };
    }
    static get states() {
        return {
            "openSubMenu": {}
        };
    }
    static get elementRef() { return "el"; }
}
