// @platformscode/core/dist/collection/nds-components/dga-header/dga-header-menu-item-usage.js
import { Fragment, h } from "@stencil/core";
const SidePanelRoutes = [
    {
        name: 'Link',
        path: '/',
        icon: {
            name: 'home-01',
            variant: 'stroke',
            size: 16,
        },
        disabled: false,
        children: [
            {
                name: 'Link',
                path: '1',
                // icon: <i class='hgi-stroke hgi-arrow-down-01' style={{fontSize:"16px"}} />,
                // badge: 39,
            },
            {
                name: 'Link',
                path: '2',
                // icon: <i class='hgi-stroke hgi-arrow-down-01' style={{fontSize:"16px"}} />,
                // badge: 21,
            },
            {
                name: 'Link',
                path: '3',
                // icon: <i class='hgi-stroke hgi-arrow-down-01' style={{fontSize:"16px"}} />,
                // badge: 2,
            },
            // {
            //   name: "Home 4",
            //   path: "4",
            //   icon: <i class='hgi-stroke hgi-arrow-down-01' style={{fontSize:"16px"}} />,
            //   badge: 20,
            // },
            // {
            //   name: "Home 5",
            //   path: "5",
            //   icon: <i class='hgi-stroke hgi-arrow-down-01' style={{fontSize:"16px"}} />,
            //   badge: 150,
            //   children: [
            //     {
            //       name: "Home 1",
            //       path: "1",
            //       icon: <i class='hgi-stroke hgi-arrow-down-01' style={{fontSize:"16px"}} />,
            //       badge: 39,
            //       level: 2,
            //     },
            //     {
            //       name: "Home 4",
            //       path: "4",
            //       icon: <i class='hgi-stroke hgi-arrow-down-01' style={{fontSize:"16px"}} />,
            //       badge: 20,
            //       level: 2,
            //     },
            //     {
            //       name: "Home 5",
            //       path: "5",
            //       icon: <i class='hgi-stroke hgi-arrow-down-01' style={{fontSize:"16px"}} />,
            //       badge: 150,
            //       level: 2,
            //     },
            //     {
            //       name: "Home 6",
            //       path: "6",
            //       icon: <i class='hgi-stroke hgi-arrow-down-01' style={{fontSize:"16px"}} />,
            //       badge: 200,
            //       level: 2,
            //     },
            //     {
            //       name: "Home 7",
            //       path: "7",
            //       icon: <i class='hgi-stroke hgi-arrow-down-01' style={{fontSize:"16px"}} />,
            //       badge: 24,
            //       level: 2,
            //     },
            //   ],
            // },
            // {
            //   name: "Home 6",
            //   path: "6",
            //   icon: <i class='hgi-stroke hgi-arrow-down-01' style={{fontSize:"16px"}} />,
            //   badge: 200,
            // },
            // {
            //   name: "Home 7",
            //   path: "7",
            //   icon: <i class='hgi-stroke hgi-arrow-down-01' style={{fontSize:"16px"}} />,
            //   badge: 24,
            // },
        ],
    },
    {
        name: 'Components',
        path: '/components',
        icon: {
            name: 'home-01',
            variant: 'stroke',
            size: 16,
        },
        disabled: false,
        children: [
            {
                name: 'Buttons',
                path: '/buttons',
                icon: {
                    name: 'home-01',
                    variant: 'stroke',
                    size: 16,
                },
                badge: 100,
            },
            {
                name: 'Badges',
                path: '/badges',
                icon: {
                    name: 'home-01',
                    variant: 'stroke',
                    size: 16,
                },
                badge: 100,
            },
            {
                name: 'Paths',
                path: '/Paths',
                children: [
                    {
                        name: 'Buttons',
                        path: '/buttons',
                        icon: {
                            name: 'home-01',
                            variant: 'stroke',
                            size: 16,
                        },
                        badge: 100,
                    },
                    {
                        name: 'Badges',
                        path: '/badges',
                        icon: {
                            name: 'home-01',
                            variant: 'stroke',
                            size: 16,
                        },
                        badge: 100,
                    },
                    {
                        name: 'Paths',
                        path: '/Paths',
                        icon: {
                            name: 'home-01',
                            variant: 'stroke',
                            size: 16,
                        },
                        badge: 100,
                    },
                ],
            },
            {
                name: 'Inputs',
                path: '/inputs',
                icon: {
                    name: 'home-01',
                    variant: 'stroke',
                    size: 16,
                },
                badge: 100,
            },
            {
                name: 'Accordions',
                path: '/accordions',
                icon: {
                    name: 'home-01',
                    variant: 'stroke',
                    size: 16,
                },
                badge: 100,
            },
            {
                name: 'Tabs',
                path: '/tabs',
                icon: {
                    name: 'home-01',
                    variant: 'stroke',
                    size: 16,
                },
                badge: 100,
            },
        ],
    },
    {
        name: 'Components 2',
        path: '/components',
        icon: {
            name: 'home-01',
            variant: 'stroke',
            size: 16,
        },
        disabled: false,
        children: [
            {
                name: 'Buttons',
                path: '/buttons',
                icon: {
                    name: 'home-01',
                    variant: 'stroke',
                    size: 16,
                },
                badge: 100,
            },
            {
                name: 'Badges',
                path: '/badges',
                icon: {
                    name: 'home-01',
                    variant: 'stroke',
                    size: 16,
                },
                badge: 100,
            },
            {
                name: 'Paths',
                path: '/Paths',
                children: [
                    {
                        name: 'Buttons',
                        path: '/buttons',
                        badge: 100,
                    },
                    {
                        name: 'Badges',
                        path: '/badges',
                        badge: 100,
                    },
                    {
                        name: 'Paths',
                        path: '/Paths',
                        badge: 100,
                    },
                ],
            },
            {
                name: 'Inputs',
                path: '/inputs',
                badge: 100,
            },
            {
                name: 'Accordions',
                path: '/accordions',
                badge: 100,
            },
            {
                name: 'Tabs',
                path: '/tabs',
                badge: 100,
            },
        ],
    },
];
export class DgaHeaderMenuItemUsage {
    constructor() {
        this.isMenuOpen = false;
        this.collapsed = false;
        // private buttonRef: HTMLElement | null = null;
        this.anchorEl = null;
        // private buttonRef: HTMLElement | null = null; // Reference to the button
        this.toggleMenu = () => {
            this.isMenuOpen = !this.isMenuOpen;
        };
        this.closeMenu = () => {
            this.isMenuOpen = false;
        };
    }
    get open() {
        return Boolean(this.anchorEl);
    }
    handleClick(event) {
        this.anchorEl = event.currentTarget;
    }
    handleClose() {
        this.anchorEl = null;
    }
    render() {
        return (h(Fragment, { key: '9b17b5e0c676aceaa3d9a712557eb2d032726921' }, h("dga-header-vv2", { key: '4fd73912c0fc04e0e478af9df1bcf2445bc6b785' }, h("dga-flex", { key: '2e46938121f7af5e60c808fae31baea6da4118af', spacing: 16, direction: "row", align: "center", justify: "between" }, h("dga-button", { key: '23a3de82afbfb4e806f478231d8dcb3899d95081', iconOnly: true, leadIcon: true, leadIconType: this.collapsed ? 'cancel01' : 'menu01', leadIconProps: {
                name: this.collapsed ? 'menu01' : 'cancel01',
                size: 24,
                variant: 'stroke',
                type: 'rounded',
            }, variant: "transparent", onClick: () => (this.collapsed = !this.collapsed), class: "header-menu__btn", sx: {
                display: { xs: 'block', md: 'none' },
            } }), h("dga-logo-placeholder", { key: '5a24170fd0ba84e4aacb2764d78ead33206b8130', size: "md" }), h("dga-flex", { key: '16c39a9f47c46d8e9e3b4e11df0dae0890803699', direction: "row", align: "center", justify: "start", sx: {
                display: { xs: 'none', md: 'block' },
            } }, h("dga-header-menu-item", { key: 'd1a44b1a3fa06390329fd259aa606a8b9d500a69', label: "Home" }), h("dga-header-menu-item", { key: '048903fdc5e3dd523e43b9182b4ec57f5ff2dc0a', label: "About" }), h("dga-header-menu-item", { key: 'd8d0eac0a1563dd672ce224963b9579fc24d4163', label: "Contact", active: true }, h("dga-header-sub-menu", { key: 'ba0cbc1f04096df838d6b5ebddc327da6778156e', subMenuFullWidth: true }, h("dga-header-sub-menu-column", { key: '6f039181ad680015597cf315df3c632dbd367d3e', label: "Column Label" }, h("dga-header-sub-menu-item", { key: '45d6854b456c7e41a6198ac8e791c35ca7c9340b', label: "Menu Item Label", icon: "checkmark-square-02", helperText: "Menu item helper text" }, h("dga-tag", { key: '661a15ac87eb6adbfbe60ae60c0427f7198a7578', label: "Tag", color: "blue", slot: "tag", rounded: true })), h("dga-header-sub-menu-item", { key: '0236cd032274ef8db62534309518fd89da5e3e5c', label: "Menu Item Label", icon: "checkmark-square-02", helperText: "Menu item helper text" }, h("dga-tag", { key: '377e829213c78cf3d9d00849288950a1a7b95be0', label: "Tag", color: "blue", slot: "tag", rounded: true })), h("dga-header-sub-menu-item", { key: '3b1e48b6b62b3ab81d45b5bea2ef7aed20785d99', label: "Menu Item Label", icon: "checkmark-square-02", helperText: "Menu item helper text" }, h("dga-tag", { key: '3fb80fe721c0a6c4908428ac9d10201ca2afdff9', label: "Tag", color: "blue", slot: "tag", rounded: true }))), h("dga-header-sub-menu-column", { key: '05d89ff5b8ca02600bded15ef67f9005d45905c4', label: "Column Label" }, h("dga-header-sub-menu-item", { key: 'f1e547be000ab6f7d6deb69ee709283d0dbd47f0', label: "Menu Item Label", icon: "checkmark-square-02", helperText: "Menu item helper text" }, h("dga-tag", { key: '8f6d7de0170b7df0671d64aa97dd9dcd23871d5c', label: "Tag", color: "blue", slot: "tag", rounded: true })), h("dga-header-sub-menu-item", { key: '4bd3b3a871eaeaef51f66fae5b984ba5b8f58adf', label: "Menu Item Label", icon: "checkmark-square-02", helperText: "Menu item helper text" }, h("dga-tag", { key: '07d1b64303eb18279be110cc0020e1c0d7898eb1', label: "Tag", color: "blue", slot: "tag", rounded: true })), h("dga-header-sub-menu-item", { key: '0cb5588868b61f99a544de82919029a08958f64e', label: "Menu Item Label", icon: "checkmark-square-02", helperText: "Menu item helper text" }, h("dga-tag", { key: 'be0e1a8b21bdcddcd045ab0570ab81fba3092bdc', label: "Tag", color: "blue", slot: "tag", rounded: true }))), h("dga-header-sub-menu-column", { key: 'c02cc839d61e1fb7300ba0aa5d6ae0e3c2e41b89', label: "Column Label" }, h("dga-header-sub-menu-item", { key: '73f50238371eb39d543baa038386e12c97fa879c', label: "Menu Item Label", icon: "checkmark-square-02", helperText: "Menu item helper text" }, h("dga-tag", { key: 'a18d4b162eee50f14c9f209d6bbf6983cb24d1f0', label: "Tag", color: "blue", slot: "tag", rounded: true })), h("dga-header-sub-menu-item", { key: '0c293de4bb1d03746288efd4590d9637673505ff', label: "Menu Item Label", icon: "checkmark-square-02", helperText: "Menu item helper text" }, h("dga-tag", { key: 'ae5c7b591bb2d2419b03b8f0d765c9d46e09c926', label: "Tag", color: "blue", slot: "tag", rounded: true })), h("dga-header-sub-menu-item", { key: '1bf88714d60959e2991e74a633c55bb5da985c0b', label: "Menu Item Label", icon: "checkmark-square-02", helperText: "Menu item helper text" }, h("dga-tag", { key: '8ae16d8325dc88fb022d97a87f7a9352c07c3de8', label: "Tag", color: "blue", slot: "tag", rounded: true }))), h("dga-header-sub-menu-column", { key: '42c7201e0b3ab28f95453feaeb9cda28187d6bab', label: "Column Label" }, h("dga-header-sub-menu-item", { key: '92396024309bca2b462f3dcae0b779d7bdcc465c', label: "Menu Item Label", icon: "checkmark-square-02", helperText: "Menu item helper text" }, h("dga-tag", { key: '9b3301804159c47dfef025adc9a966f7a2d852cc', label: "Tag", color: "blue", slot: "tag", rounded: true })), h("dga-header-sub-menu-item", { key: 'b030ea4d6145b073f9b7b13c5d6678871de25e99', label: "Menu Item Label", icon: "checkmark-square-02", helperText: "Menu item helper text" }, h("dga-tag", { key: '2da197bd1354de85f7bdf96e437b010e0b090650', label: "Tag", color: "blue", slot: "tag", rounded: true })), h("dga-header-sub-menu-item", { key: 'f9d78d18f323dc8f38f31cbcbcd1d1635d03a84a', label: "Menu Item Label", icon: "checkmark-square-02", helperText: "Menu item helper text" }, h("dga-tag", { key: '4e553cdeb5acce7a40229e7b4f57e9c209d3b1dd', label: "Tag", color: "blue", slot: "tag", rounded: true }))), h("dga-header-sub-menu-column", { key: '9e6029a3960a0bbe238c3ac8a909de557168df71', label: "Column Label" }, h("dga-header-sub-menu-item", { key: '8e9b6fdebcea479ffefc01899e5af93bd425be7c', label: "Menu Item Label", icon: "checkmark-square-02", helperText: "Menu item helper text" }, h("dga-tag", { key: '18f015ec79cd0585fee082da52323d0783848c09', label: "Tag", color: "blue", slot: "tag", rounded: true })), h("dga-header-sub-menu-item", { key: '3cf09d00a77dab63a2ed69c4a59e6e64ddc16f99', label: "Menu Item Label", icon: "checkmark-square-02", helperText: "Menu item helper text" }, h("dga-tag", { key: '9e294a795ab459bb37ef60584404afa61bbfd0a5', label: "Tag", color: "blue", slot: "tag", rounded: true })), h("dga-header-sub-menu-item", { key: '64dd9564da06d400b240b93ab33b102ee24d2efe', label: "Menu Item Label", icon: "checkmark-square-02", helperText: "Menu item helper text" }, h("dga-tag", { key: 'f191ffd9ba20208c40148c119934993aef406301', label: "Tag", color: "blue", slot: "tag", rounded: true }))))))), h("dga-flex", { key: 'fb52421880899609b2f124865f958850a0c849b6', direction: "row", align: "center", justify: "end" }, h("dga-header-action", { key: 'f6b4d359614b72302122d0527cdadafbc08c56d5', label: "Action", icon: "login-square-01" }), h("dga-header-action", { key: 'f046712790f652d6334659acf72986ee1b59bc97', icon: "login-square-01", iconPosition: "top" }, h("dga-header-sub-menu", { key: '46163112ce84032f35b6dfa82477a30007f5f622', subMenuFullWidth: true }, h("dga-search-box", { key: '407bdff92515fbffcf7c5c266b6a9717d3d803da' }))), h("dga-action-menu", { key: '87bc8780d9ff26e91d8ed69dd022c20c46ed4391', open: this.isMenuOpen, onClose: () => (this.isMenuOpen = false), placement: 'bottom-end' }, h("dga-button-v2", { key: '4a621bd6d6f69a1f0ad8b144894aa0f1566ad50e', slot: "trigger", iconOnly: true, leadIcon: true, leadIconProps: {
                name: 'more-horizontal-circle-01',
                size: 24,
                variant: 'solid',
                type: 'rounded',
            }, variant: "transparent", onClick: () => (this.isMenuOpen = true), class: "header-menu__btn" }), h("dga-menu-list", { key: '56ab7289c58d0fdcecaa94b4fbd0e771de7733e6' }, h("dga-menu-item", { key: 'ae62f8334b2263477abd732b5b33bce8780a2706', label: "Item Label" }), h("dga-menu-item", { key: '6577a31bb95ca903369313edeb66c295fb86b17d', label: "Item Label" }), h("dga-menu-item", { key: '302b90827d7c4aa80c09f40069c32685f605ad87', label: "Item Label" }))))), h("dga-drawer", { key: 'e7ffc3013d9336e07d50aeb4072ac99f1e7c2fe6', routes: SidePanelRoutes, open: this.collapsed, background: 'gray', headerLogo: "https://casamedia.com/wp-content/uploads/2023/04/adidas-1024x683.png", sx: {
                '& > .sidepanel': {
                    top: '72px',
                },
            }, overlay: true }, "efddsfdsfsdfdsfdsf dfdsfdsfdsfdsfdsfdf", h("a", { key: 'f4e29e5c7a7641520e323c05260987542c6b2279' }, "fdgdfgfdg"), h("div", { key: '340fc9d899ac7ef883f0943350ebe5422e5374a2' }, "gfdgfdgfd")), h("dga-logo-placeholder", { key: 'c924df6afe4639703277f0561e1baffea7e33bd7', text: "Platform Logo", size: "md" }), h("dga-logo-placeholder", { key: '25bef7a087c2a9db44d285a42474edbda2072d84', text: "Platform Logo", size: "sm" }), h("dga-action-menu", { key: 'aadc9212502beed9390fcff12aedeb93f2ca7865', open: this.isMenuOpen, onClose: () => (this.isMenuOpen = false), placement: 'bottom-end' }, h("dga-button-v2", { key: '0ee976d3cf17c6777ce60967b90704d1c0dffa3a', slot: "trigger", iconOnly: true, leadIcon: true, leadIconProps: {
                name: 'more-horizontal-circle-01',
                size: 24,
                variant: 'solid',
                type: 'rounded',
            }, variant: "transparent", onClick: () => (this.isMenuOpen = true), class: "header-menu__btn" }), h("dga-menu-list", { key: 'e36a7fb35adb9745d8f818df671f8c373348f0bc' }, h("dga-menu-item", { key: 'a1a176b8359a7639b61b40f0932e6a67a42f0e37', label: "Item Label" }), h("dga-menu-item", { key: '0fa44facdd01fd8037a94bc1c8a0a865f7926967', label: "Item Label" }), h("dga-menu-item", { key: 'a7795b05c5a6b1e64027132d4ea09d4bb75bca01', label: "Item Label" })))));
    }
    static get is() { return "dga-header-menu-item-usage"; }
    static get states() {
        return {
            "isMenuOpen": {},
            "collapsed": {},
            "anchorEl": {}
        };
    }
}
