// @platformscode/core/dist/collection/nds-components/dga-header/dga-header-sub-menu-column.js
import { h } from "@stencil/core";
export class DgaHeaderSubMenuColumn {
    constructor() {
        this.label = '';
    }
    render() {
        return (h("div", { key: '47fd03418cbc0be7ae13fff47a09661d0bc4b8cc', class: "sub-menu__column" }, this.label && h("h2", { key: '7f8e941e88f1b87f1b1b48e55043c80b9e9ccc9d', class: "sub-menu__column__label" }, this.label), h("slot", { key: 'd2d7d30e6f0d0840c868afe88281ba915e6ae350' })));
    }
    static get is() { return "dga-header-sub-menu-column"; }
    static get encapsulation() { return "shadow"; }
    static get originalStyleUrls() {
        return {
            "$": ["dga-header-sub-menu-column.scss"]
        };
    }
    static get styleUrls() {
        return {
            "$": ["dga-header-sub-menu-column.css"]
        };
    }
    static get properties() {
        return {
            "label": {
                "type": "string",
                "attribute": "label",
                "mutable": false,
                "complexType": {
                    "original": "string",
                    "resolved": "string",
                    "references": {}
                },
                "required": false,
                "optional": true,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false,
                "defaultValue": "''"
            }
        };
    }
}
