// @platformscode/core/dist/collection/components/dga-header/dga-header-mobile-actions.js
import { h, Fragment } from "@stencil/core";
export class DgaHeaderMobileActions {
    constructor() {
        this.actions = [];
        this.isMenuOpen = false; // State to control menu visibility
        this.anchorEl = null;
        // private buttonRef: HTMLElement | null = null; // Reference to the button
        this.toggleMenu = () => {
            this.isMenuOpen = !this.isMenuOpen;
            if (this.isMenuOpen) {
                document.addEventListener('click', this.handleOutsideClick);
            }
            else {
                document.removeEventListener('click', this.handleOutsideClick);
            }
        };
        this.closeMenu = () => {
            this.isMenuOpen = false;
            document.removeEventListener('click', this.handleOutsideClick);
        };
        this.handleOutsideClick = (event) => {
            const target = event.target;
            if (!this.el.contains(target)) {
                this.closeMenu();
            }
        };
    }
    render() {
        return (h(Fragment, { key: 'fa8e28741295c36920b1ded056bba092e825268f' }, h("div", { key: 'd29bb008ca535d1bf5a8d6389a256d52b994cff5', style: { position: 'relative' } }), h("ul", { key: 'b97047cd1e61ed3f909b35897a4ed3aefe9cc6b5', class: "header-nav__mobile-actions" }, h("slot", { key: '659c87efda0230729d8753e52689e90ce5dc1975' }))));
    }
    static get is() { return "dga-header-mobile-actions"; }
    static get encapsulation() { return "shadow"; }
    static get originalStyleUrls() {
        return {
            "$": ["dga-nav-header-sub-menu.scss"]
        };
    }
    static get styleUrls() {
        return {
            "$": ["dga-nav-header-sub-menu.css"]
        };
    }
    static get properties() {
        return {
            "actions": {
                "type": "unknown",
                "attribute": "actions",
                "mutable": false,
                "complexType": {
                    "original": "Action[]",
                    "resolved": "Action[]",
                    "references": {
                        "Action": {
                            "location": "local",
                            "path": "/Users/mohamedaboelezz/projects/Azm-X/platform-code-cross-platform-storybook/packages/core/src/components/dga-header/dga-header-mobile-actions.tsx",
                            "id": "src/components/dga-header/dga-header-mobile-actions.tsx::Action"
                        }
                    }
                },
                "required": false,
                "optional": true,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "defaultValue": "[]"
            }
        };
    }
    static get states() {
        return {
            "isMenuOpen": {},
            "anchorEl": {}
        };
    }
    static get elementRef() { return "el"; }
}
