// @platformscode/core/dist/collection/components/dga-pagination/dga-pagination.js
import { h } from "@stencil/core";
import { guid } from "../../utils";
import { SxHandler } from "../../utils/sx-handler";
import { isRTL } from "../../utils/rtl";
const DOTS = '...';
const range = (start, end) => {
    let length = end - start + 1;
    return Array.from({ length }, (_, idx) => idx + start);
};
export class DgaPagination {
    constructor() {
        this.size = 'large';
        this.totalPageCount = 10;
        this.currentPage = 1;
        this.siblingCount = 1;
        this.onChange = () => { };
        this.onNext = () => {
            if (this.currentPage == this.totalPageCount)
                return;
            this.currentPage = this.currentPage + 1;
            this.onChange && this.onChange(this.currentPage);
        };
        this.onPrevious = () => {
            if (this.currentPage == 1)
                return;
            this.currentPage = this.currentPage - 1;
            this.onChange && this.onChange(this.currentPage);
        };
        this.handelPageChange = (activePage) => {
            if (activePage == this.currentPage)
                return;
            this.currentPage = activePage;
            this.onChange(this.currentPage);
        };
    }
    // @State() currentPage = 1;
    get paginationRange() {
        const { totalPageCount, siblingCount, currentPage } = this;
        const totalPageNumbers = siblingCount + 5;
        // If the number of pages is less than the page numbers we want to show in our pagination component, we return the range [1..totalPageCount]
        if (totalPageNumbers >= totalPageCount) {
            return range(1, totalPageCount);
        }
        const leftSiblingIndex = Math.max(currentPage - siblingCount, 1);
        const rightSiblingIndex = Math.min(currentPage + siblingCount, totalPageCount);
        const shouldShowLeftDots = leftSiblingIndex > 2;
        const shouldShowRightDots = rightSiblingIndex < totalPageCount - 2;
        const firstPageIndex = 1;
        const lastPageIndex = totalPageCount;
        if (!shouldShowLeftDots && shouldShowRightDots) {
            let leftItemCount = 3 + 2 * siblingCount;
            let leftRange = range(1, leftItemCount);
            return [...leftRange, DOTS, totalPageCount];
        }
        if (shouldShowLeftDots && !shouldShowRightDots) {
            let rightItemCount = 3 + 2 * siblingCount;
            let rightRange = range(totalPageCount - rightItemCount + 1, totalPageCount);
            return [firstPageIndex, DOTS, ...rightRange];
        }
        if (shouldShowLeftDots && shouldShowRightDots) {
            let middleRange = range(leftSiblingIndex, rightSiblingIndex);
            return [firstPageIndex, DOTS, ...middleRange, DOTS, lastPageIndex];
        }
        return [];
    }
    componentWillLoad() {
        this.sxHandler = new SxHandler(this.el, 'dga-slider');
    }
    componentDidLoad() {
        this.sxHandler.processSx(this.sx);
    }
    sxChanged() {
        this.sxHandler.processSx(this.sx);
    }
    disconnectedCallback() {
        this.sxHandler.cleanup();
    }
    render() {
        const { el } = this;
        const dir = isRTL(el) ? 'rtl' : 'ltr';
        return (h("ul", { key: '89574d2284bb22b95fd7c3cbf670e2160a2893bc', dir: dir, class: `pagination pagination--${this.size}` }, h("li", { key: '8cd804b8bb3ae0f5ba4afd21842a6015d49cc3f6', class: "pagination__item pagination__arrow" }, h("button", { key: '21e29706de888e40cdf6930ceb963b843766bf85', onClick: this.onPrevious, disabled: this.currentPage === 1 }, h("svg", { key: '3effbba35d2225a953be58de3344b66f27ccbafa', width: "17", height: "16", viewBox: "0 0 17 16", fill: "none", xmlns: "http://www.w3.org/2000/svg" }, h("g", { key: 'be59b6288445aaae431e22c18f9bff9b0a4a4b29', id: "Leading Icon" }, h("path", { key: 'ac3eb19633e2db8952f3e2e03683c38709c39204', id: "Icon", d: "M10.5 12L6.5 8L10.5 4", stroke: "#161616", "stroke-width": "1.5", "stroke-linecap": "round", "stroke-linejoin": "round" }))))), this.paginationRange &&
            this.paginationRange.map(pageNumber => {
                if (pageNumber === DOTS) {
                    return (h("li", { key: guid(), class: "pagination__item" }, "\u2026"));
                }
                return (h("li", { key: guid(), class: `pagination__item ${this.currentPage === pageNumber ? 'pagination__item--active' : ''}` }, h("button", { onClick: () => {
                        this.handelPageChange(pageNumber);
                    } }, pageNumber)));
            }), h("li", { key: '46b18473db22872b05dab57b90712693e1a491a1', class: "pagination__item pagination__arrow" }, h("button", { key: 'd9e8459064fb22cdb834c22eca83b10791f9150a', onClick: this.onNext, disabled: this.currentPage === this.totalPageCount }, h("svg", { key: '6f5cda029a252cc73f9c4507ff6be2f57a26a712', width: "17", height: "16", viewBox: "0 0 17 16", fill: "none", xmlns: "http://www.w3.org/2000/svg" }, h("g", { key: '19e46f0b48a6649d39cf95409e3765b1c7fd7e13', id: "Leading Icon" }, h("path", { key: '5c3f05a30bd4d48195b50d8ba88b02ef3dd101d1', id: "Icon", d: "M6.5 12L10.5 8L6.5 4", stroke: "#161616", "stroke-width": "1.5", "stroke-linecap": "round", "stroke-linejoin": "round" })))))));
    }
    static get is() { return "dga-pagination"; }
    static get encapsulation() { return "shadow"; }
    static get originalStyleUrls() {
        return {
            "$": ["dga-pagination.scss"]
        };
    }
    static get styleUrls() {
        return {
            "$": ["dga-pagination.css"]
        };
    }
    static get properties() {
        return {
            "size": {
                "type": "string",
                "attribute": "size",
                "mutable": false,
                "complexType": {
                    "original": "'small' | 'medium' | 'large'",
                    "resolved": "\"large\" | \"medium\" | \"small\"",
                    "references": {}
                },
                "required": false,
                "optional": true,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false,
                "defaultValue": "'large'"
            },
            "totalPageCount": {
                "type": "number",
                "attribute": "total-page-count",
                "mutable": false,
                "complexType": {
                    "original": "number",
                    "resolved": "number",
                    "references": {}
                },
                "required": false,
                "optional": false,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false,
                "defaultValue": "10"
            },
            "currentPage": {
                "type": "number",
                "attribute": "current-page",
                "mutable": false,
                "complexType": {
                    "original": "number",
                    "resolved": "number",
                    "references": {}
                },
                "required": false,
                "optional": false,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false,
                "defaultValue": "1"
            },
            "siblingCount": {
                "type": "number",
                "attribute": "sibling-count",
                "mutable": false,
                "complexType": {
                    "original": "number",
                    "resolved": "number",
                    "references": {}
                },
                "required": false,
                "optional": true,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false,
                "defaultValue": "1"
            },
            "onChange": {
                "type": "unknown",
                "attribute": "on-change",
                "mutable": false,
                "complexType": {
                    "original": "(value: number) => void",
                    "resolved": "(value: number) => void",
                    "references": {}
                },
                "required": false,
                "optional": true,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "defaultValue": "() => {}"
            },
            "sx": {
                "type": "string",
                "attribute": "sx",
                "mutable": false,
                "complexType": {
                    "original": "SxProps | string",
                    "resolved": "SxProps | string",
                    "references": {
                        "SxProps": {
                            "location": "import",
                            "path": "../../utils/sx-handler",
                            "id": "src/utils/sx-handler.ts::SxProps"
                        }
                    }
                },
                "required": false,
                "optional": true,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false
            }
        };
    }
    static get elementRef() { return "el"; }
    static get watchers() {
        return [{
                "propName": "sx",
                "methodName": "sxChanged"
            }];
    }
}
