// @platformscode/core/dist/components/dga-metric-usage.js
import { p as proxyCustomElement, H, h } from './p-BhqVmlhr.js';
import { d as defineCustomElement$2 } from './p-BxGSt2i7.js';

const DgaMetricUsage$1 = /*@__PURE__*/ proxyCustomElement(class DgaMetricUsage extends H {
    constructor() {
        super();
        this.__registerHost();
    }
    //   @State() variant: 'critical' | 'warning' | 'success' | 'info' | 'neutral' = 'neutral';
    render() {
        return (h("div", { key: '1a07df4f182c0401af09a5aef3a8b8bd062cdebb' }, h("h1", { key: '3cf8047b1d5e85fb19c854eb0336358a1f2690c7' }, "Metric Example"), h("div", { key: '62100d47056658878c8299c613ca2bea8da78aa6', style: { padding: '50px', display: 'flex', gap: '20px' } }, h("div", { key: '397b8e44ac6f9c100239ac0284fe64ad2156b6fa', style: { display: 'flex', flexDirection: 'column', gap: '20px' } }, h("dga-apex-chart", { key: '2f60285003d4d93554f6cdaaebf612c6475b4cbd',
            // type="donut" 
            //   series={ [{
            //     name: "Desktops",
            //     data: [10, 41, 35, 51, 49, 62, 69, 91, 148]
            // }]} 
            // labels={['Apple', 'Mango', 'Orange', 'Watermelon']}
            options: {
                series: [{
                        name: 'series1',
                        data: [31, 40, 28, 51, 42, 109, 100]
                    }],
                chart: {
                    height: 350,
                    type: 'area',
                    toolbar: {
                        show: false // Hides the toolbar
                    },
                    zoom: {
                        enabled: false // Disables zooming
                    },
                },
                dataLabels: {
                    enabled: false
                },
                // markers:{
                //   shape:"circle",
                //   size:6,
                //   strokeColors:"#067647",
                //   strokeWidth:2,
                //   fillOpacity: 1, // Simulates inner fill opacity
                //   colors:"#fff",
                // },
                stroke: {
                    curve: 'stepline'
                },
                fill: {
                    type: 'gradient', // Options: 'solid', 'gradient'
                    gradient: {
                        shade: 'light',
                        type: "vertical", // Options: 'horizontal', 'vertical'
                        gradientToColors: ['#fff'], // End color of the gradient
                        stops: [0, 100] // Color distribution
                    },
                    colors: ['#079455'] // Start color
                },
                // xaxis: {
                //   type: 'datetime',
                //   categories: ["2018-09-19T00:00:00.000Z", "2018-09-19T01:30:00.000Z", "2018-09-19T02:30:00.000Z", "2018-09-19T03:30:00.000Z", "2018-09-19T04:30:00.000Z", "2018-09-19T05:30:00.000Z", "2018-09-19T06:30:00.000Z"]
                // },
                // tooltip: {
                //   x: {
                //     format: 'dd/MM/yy HH:mm'
                //   },
                // },
                xaxis: {
                    labels: {
                        show: false // Hides X-axis labels
                    },
                    axisBorder: {
                        show: false // Hides the X-axis border line
                    },
                    axisTicks: {
                        show: false // Hides the X-axis ticks
                    }
                },
                yaxis: {
                    labels: {
                        show: false // Hides X-axis labels
                    },
                    axisBorder: {
                        show: false // Hides the X-axis border line
                    },
                    axisTicks: {
                        show: false // Hides the X-axis ticks
                    }
                },
                // xaxis: {
                //   show: false // Hides the X-axis completely
                // },
                // yaxis: {
                //   show: false // Hides the Y-axis completely
                // },
                grid: {
                    show: false
                },
                // tooltip:{
                //   enabled: false
                // }
                legend: {
                    show: false
                }
            } }), h("dga-apex-chart", { key: 'dc1de635a28c5571e214ebb5b57bb519ef840b43',
            // type="donut" 
            //   series={ [{
            //     name: "Desktops",
            //     data: [10, 41, 35, 51, 49, 62, 69, 91, 148]
            // }]} 
            // labels={['Apple', 'Mango', 'Orange', 'Watermelon']}
            options: {
                series: [{
                        name: 'series1',
                        data: [31, 40, 28, 51, 42, 109, 100]
                    }],
                chart: {
                    height: 350,
                    type: 'area',
                    toolbar: {
                        show: false // Hides the toolbar
                    },
                    zoom: {
                        enabled: false // Disables zooming
                    },
                },
                dataLabels: {
                    enabled: false
                },
                // markers:{
                //   shape:"circle",
                //   size:6,
                //   strokeColors:"#067647",
                //   strokeWidth:2,
                //   fillOpacity: 1, // Simulates inner fill opacity
                //   colors:"#fff",
                // },
                stroke: {
                    curve: 'straight'
                },
                fill: {
                    type: 'gradient', // Options: 'solid', 'gradient'
                    gradient: {
                        shade: 'light',
                        type: "vertical", // Options: 'horizontal', 'vertical'
                        gradientToColors: ['#fff'], // End color of the gradient
                        stops: [0, 100] // Color distribution
                    },
                    colors: ['#079455'] // Start color
                },
                // xaxis: {
                //   type: 'datetime',
                //   categories: ["2018-09-19T00:00:00.000Z", "2018-09-19T01:30:00.000Z", "2018-09-19T02:30:00.000Z", "2018-09-19T03:30:00.000Z", "2018-09-19T04:30:00.000Z", "2018-09-19T05:30:00.000Z", "2018-09-19T06:30:00.000Z"]
                // },
                // tooltip: {
                //   x: {
                //     format: 'dd/MM/yy HH:mm'
                //   },
                // },
                xaxis: {
                    labels: {
                        show: false // Hides X-axis labels
                    },
                    axisBorder: {
                        show: false // Hides the X-axis border line
                    },
                    axisTicks: {
                        show: false // Hides the X-axis ticks
                    }
                },
                yaxis: {
                    labels: {
                        show: false // Hides X-axis labels
                    },
                    axisBorder: {
                        show: false // Hides the X-axis border line
                    },
                    axisTicks: {
                        show: false // Hides the X-axis ticks
                    }
                },
                // xaxis: {
                //   show: false // Hides the X-axis completely
                // },
                // yaxis: {
                //   show: false // Hides the Y-axis completely
                // },
                grid: {
                    show: false
                },
                // tooltip:{
                //   enabled: false
                // }
                legend: {
                    show: false
                }
            } }), h("dga-apex-chart", { key: 'c263be3a032352f77b4e6e735f03a02ec694f26a',
            // type="donut" 
            //   series={ [{
            //     name: "Desktops",
            //     data: [10, 41, 35, 51, 49, 62, 69, 91, 148]
            // }]} 
            // labels={['Apple', 'Mango', 'Orange', 'Watermelon']}
            options: {
                series: [{
                        name: 'series1',
                        data: [31, 40, 28, 51, 42, 109, 100]
                    }],
                chart: {
                    height: 350,
                    type: 'area',
                    toolbar: {
                        show: false // Hides the toolbar
                    },
                    zoom: {
                        enabled: false // Disables zooming
                    },
                },
                dataLabels: {
                    enabled: false
                },
                // markers:{
                //   shape:"circle",
                //   size:6,
                //   strokeColors:"#067647",
                //   strokeWidth:2,
                //   fillOpacity: 1, // Simulates inner fill opacity
                //   colors:"#fff",
                // },
                stroke: {
                    curve: 'smooth'
                },
                fill: {
                    type: 'gradient', // Options: 'solid', 'gradient'
                    gradient: {
                        shade: 'light',
                        type: "vertical", // Options: 'horizontal', 'vertical'
                        gradientToColors: ['#fff'], // End color of the gradient
                        stops: [0, 100] // Color distribution
                    },
                    colors: ['#079455'] // Start color
                },
                // xaxis: {
                //   type: 'datetime',
                //   categories: ["2018-09-19T00:00:00.000Z", "2018-09-19T01:30:00.000Z", "2018-09-19T02:30:00.000Z", "2018-09-19T03:30:00.000Z", "2018-09-19T04:30:00.000Z", "2018-09-19T05:30:00.000Z", "2018-09-19T06:30:00.000Z"]
                // },
                // tooltip: {
                //   x: {
                //     format: 'dd/MM/yy HH:mm'
                //   },
                // },
                xaxis: {
                    labels: {
                        show: false // Hides X-axis labels
                    },
                    axisBorder: {
                        show: false // Hides the X-axis border line
                    },
                    axisTicks: {
                        show: false // Hides the X-axis ticks
                    }
                },
                yaxis: {
                    labels: {
                        show: false // Hides X-axis labels
                    },
                    axisBorder: {
                        show: false // Hides the X-axis border line
                    },
                    axisTicks: {
                        show: false // Hides the X-axis ticks
                    }
                },
                // xaxis: {
                //   show: false // Hides the X-axis completely
                // },
                // yaxis: {
                //   show: false // Hides the Y-axis completely
                // },
                grid: {
                    show: false
                },
                // tooltip:{
                //   enabled: false
                // }
                legend: {
                    show: false
                }
            } }), h("dga-apex-chart", { key: '048ee888f1cc575728e10476cb43afdcb2defa6b',
            // type="donut" 
            //   series={ [{
            //     name: "Desktops",
            //     data: [10, 41, 35, 51, 49, 62, 69, 91, 148]
            // }]} 
            // labels={['Apple', 'Mango', 'Orange', 'Watermelon']}
            options: {
                series: [{
                        name: 'series1',
                        data: [31, 40, 28, 51, 42, 109, 100]
                    }, {
                        name: 'series2',
                        data: [11, 32, 45, 32, 34, 52, 41]
                    }],
                chart: {
                    height: 350,
                    type: 'area',
                    toolbar: {
                        show: false // Hides the toolbar
                    },
                    zoom: {
                        enabled: false // Disables zooming
                    },
                },
                dataLabels: {
                    enabled: false
                },
                // markers:{
                //   shape:"circle",
                //   size:6,
                //   strokeColors:"#067647",
                //   strokeWidth:2,
                //   fillOpacity: 1, // Simulates inner fill opacity
                //   colors:"#fff",
                // },
                stroke: {
                    curve: 'smooth'
                },
                fill: {
                    type: 'gradient', // Options: 'solid', 'gradient'
                    gradient: {
                        shade: 'light',
                        type: "vertical", // Options: 'horizontal', 'vertical'
                        gradientToColors: ['#fff'], // End color of the gradient
                        stops: [0, 100] // Color distribution
                    },
                    colors: ['#079455'] // Start color
                },
                // xaxis: {
                //   type: 'datetime',
                //   categories: ["2018-09-19T00:00:00.000Z", "2018-09-19T01:30:00.000Z", "2018-09-19T02:30:00.000Z", "2018-09-19T03:30:00.000Z", "2018-09-19T04:30:00.000Z", "2018-09-19T05:30:00.000Z", "2018-09-19T06:30:00.000Z"]
                // },
                // tooltip: {
                //   x: {
                //     format: 'dd/MM/yy HH:mm'
                //   },
                // },
                xaxis: {
                    labels: {
                        show: false // Hides X-axis labels
                    },
                    axisBorder: {
                        show: false // Hides the X-axis border line
                    },
                    axisTicks: {
                        show: false // Hides the X-axis ticks
                    }
                },
                yaxis: {
                    labels: {
                        show: false // Hides X-axis labels
                    },
                    axisBorder: {
                        show: false // Hides the X-axis border line
                    },
                    axisTicks: {
                        show: false // Hides the X-axis ticks
                    }
                },
                // xaxis: {
                //   show: false // Hides the X-axis completely
                // },
                // yaxis: {
                //   show: false // Hides the Y-axis completely
                // },
                grid: {
                    show: false
                },
                // tooltip:{
                //   enabled: false
                // }
                legend: {
                    show: false
                }
            } })), h("div", { key: 'c6133b5383b20d980e00abeec668d5fb3a9617c8', style: { display: 'flex', flexDirection: 'column', gap: '20px' } }))));
    }
}, [0, "dga-metric-usage"]);
function defineCustomElement$1() {
    if (typeof customElements === "undefined") {
        return;
    }
    const components = ["dga-metric-usage", "dga-apex-chart"];
    components.forEach(tagName => { switch (tagName) {
        case "dga-metric-usage":
            if (!customElements.get(tagName)) {
                customElements.define(tagName, DgaMetricUsage$1);
            }
            break;
        case "dga-apex-chart":
            if (!customElements.get(tagName)) {
                defineCustomElement$2();
            }
            break;
    } });
}

const DgaMetricUsage = DgaMetricUsage$1;
const defineCustomElement = defineCustomElement$1;

export { DgaMetricUsage, defineCustomElement };
