// @platformscode/core/dist/components/dga-button-v2-usage.js
import { p as proxyCustomElement, H, h } from './p-BhqVmlhr.js';
import { d as defineCustomElement$5 } from './p-t2gkYg4G.js';
import { d as defineCustomElement$4 } from './p-B4U0lY4y.js';
import { d as defineCustomElement$3 } from './p-CCimZGO2.js';
import { d as defineCustomElement$2 } from './p-DBTOrdV2.js';

const DgaButtonUsage = /*@__PURE__*/ proxyCustomElement(class DgaButtonUsage extends H {
    constructor() {
        super();
        this.__registerHost();
        this.variant = 'neutral';
    }
    render() {
        return (h("div", { key: '62912bbc6e65a9492a73328564668d34aa9abdcd' }, h("h1", { key: '6c221cdd9c033bd17ee93029d394127e03357994' }, "Button Example"), h("div", { key: 'ddea013f3090c7e7e7455f0046dabd53de96cd0a', style: { padding: '50px', display: 'flex', gap: '20px' } }, h("div", { key: '33330576f9eeca9d0801031e69b9e545e6b25c64', style: { display: 'flex', flexDirection: 'column', gap: '20px' } }, h("dga-flex", { key: '66b5618032bdbde6d8ef3a3ca34ef3e6372ad9be', direction: "column", spacing: 16 }, h("dga-button-v2", { key: '4a0fa7e2d7e1dfc7a3bdf5dd7c599a08d22ea1ff', label: "Button", variant: "primary", size: "lg" }), h("dga-button-v2", { key: '182463c6fccf4d1f3bcd6dd9f5099549c797909e', label: "Button", variant: "primary", size: "md" }), h("dga-button-v2", { key: 'b40fcfa4cd327e3ce84273f5a80d22d7fc0be7f0', label: "Button", variant: "primary", size: "sm" }), h("dga-button-v2", { key: '79b20c61a62868b5b62cdbe756233bf157ed1ab9', label: "Button", variant: "neutral", size: "lg" }), h("dga-button-v2", { key: '71a71102e49f1c73561bb11646f0408a3b43f251', label: "Button", variant: "neutral", size: "md" }), h("dga-button-v2", { key: '3f518b4640c32f7b841fb724d396e91742d123e3', label: "Button", variant: "neutral", size: "sm" }), h("dga-button-v2", { key: 'f0d3309ae06a32963848b1106057ec6e7ffc52e8', label: "Button", variant: "secondary-solid", size: "lg" }), h("dga-button-v2", { key: '0a79a0f545508c9ab8e042416fe04919365e5633', label: "Button", variant: "secondary-solid", size: "md" }), h("dga-button-v2", { key: '92d018c2097b64dd20e1296b136b4ade81ca7146', label: "Button", variant: "secondary-solid", size: "sm" }), h("dga-button-v2", { key: 'beef54cd29768553d739da27ae7fd3c5d7399e84', label: "Button", variant: "primary", destructive: true, size: "lg" }), h("dga-button-v2", { key: '7ec7f514e470dabeeb4bc7390b5709fb9297537a', label: "Button", variant: "primary", destructive: true, size: "md" }), h("dga-button-v2", { key: '6cefb8a9c8407a625b03311ffa80ae7cda1fd425', label: "Button", variant: "primary", destructive: true, size: "sm" })), h("dga-button-v2", { key: '0617cd7f9cc299cba7b9d2cf2999bfeaad715c4d', label: "Button", variant: "primary", onColor: true }, h("dga-button-leading-icon", { key: 'a4ec1c158f6f7c051dceea240d86feb81ccc9b21', name: "link-square-01", size: 16, color: "red" })), h("dga-button-v2", { key: '03604467cf29802efd3df308e5eb9cb44171a44a', label: "Source code", variant: "secondary-outline", leadIcon: true, leadIconProps: { name: 'source-code', variant: 'stroke', type: 'standard' } }), h("dga-button-v2", { key: '18420f649770f00b6745e330d6e4efc49927fda2',
            // className="absolute top-[50%] left-[50%] transform -translate-x-1/2 -translate-y-1/2 opacity-0 group-hover:opacity-100"
            label: "See live preview", size: "lg", variant: "secondary-outline", trailIcon: true, trailIconProps: {
                name: 'link-square-01',
                variant: 'stroke',
                type: 'standard',
            } })), h("div", { key: '8617633fffcb3ce70ff2990dafe16aa05bb3f4fc', style: { display: 'flex', flexDirection: 'column', gap: '20px' } }))));
    }
}, [0, "dga-button-v2-usage", {
        "variant": [32]
    }]);
function defineCustomElement$1() {
    if (typeof customElements === "undefined") {
        return;
    }
    const components = ["dga-button-v2-usage", "dga-button-leading-icon", "dga-button-v2", "dga-flex", "dga-icon-v2"];
    components.forEach(tagName => { switch (tagName) {
        case "dga-button-v2-usage":
            if (!customElements.get(tagName)) {
                customElements.define(tagName, DgaButtonUsage);
            }
            break;
        case "dga-button-leading-icon":
            if (!customElements.get(tagName)) {
                defineCustomElement$5();
            }
            break;
        case "dga-button-v2":
            if (!customElements.get(tagName)) {
                defineCustomElement$4();
            }
            break;
        case "dga-flex":
            if (!customElements.get(tagName)) {
                defineCustomElement$3();
            }
            break;
        case "dga-icon-v2":
            if (!customElements.get(tagName)) {
                defineCustomElement$2();
            }
            break;
    } });
}

const DgaButtonV2Usage = DgaButtonUsage;
const defineCustomElement = defineCustomElement$1;

export { DgaButtonV2Usage, defineCustomElement };
