// @platformscode/core/dist/components/dga-metric-chart-v2.js
import { p as proxyCustomElement, H, h } from './p-BhqVmlhr.js';

const DgaMetricChartV2$1 = /*@__PURE__*/ proxyCustomElement(class DgaMetricChartV2 extends H {
    constructor() {
        super();
        this.__registerHost();
        this.__attachShadow();
        this.width = 130;
        this.height = 66;
        this.strokeColor = '#067647';
        this.fillColor = '#079455';
        this.strokeWidth = 2;
        this.data = [];
        this.showMarker = true;
        this.pathD = '';
        this.areaD = '';
        this.currentMarkerPos = { x: 76, y: 5 }; // default position
    }
    dataChangeHandler() {
        this.generatePaths();
        this.updateMarkerPosition();
    }
    componentWillLoad() {
        this.generatePaths();
        this.updateMarkerPosition();
    }
    normalizeData() {
        if (this.data.length === 0)
            return [];
        const xValues = this.data.map(d => d.x);
        const yValues = this.data.map(d => d.y);
        const minX = Math.min(...xValues);
        const maxX = Math.max(...xValues);
        const minY = Math.min(...yValues);
        const maxY = Math.max(...yValues);
        const margin = 10;
        return this.data.map(point => ({
            x: ((point.x - minX) / (maxX - minX)) * (128 - 2 * margin) + margin,
            y: 64 - (((point.y - minY) / (maxY - minY)) * (64 - 2 * margin) + margin)
        }));
    }
    updateMarkerPosition() {
        if (this.markerPosition) {
            this.currentMarkerPos = this.markerPosition;
        }
        else if (this.data.length > 0) {
            const normalizedData = this.normalizeData();
            const lastPoint = normalizedData[normalizedData.length - 1];
            // Calculate marker position based on the last data point
            // Adjust these offsets to position the marker correctly relative to the point
            this.currentMarkerPos = {
                x: lastPoint.x - 10, // Adjust these offsets as needed
                y: lastPoint.y - 15 // Adjust these offsets as needed
            };
        }
    }
    generatePaths() {
        const normalizedData = this.normalizeData();
        if (normalizedData.length === 0)
            return;
        this.pathD = normalizedData.reduce((path, point, i) => {
            return path + `${i === 0 ? 'M' : 'L'} ${point.x} ${point.y} `;
        }, '');
        this.areaD = `${this.pathD} L ${normalizedData[normalizedData.length - 1].x} 65 L 1 65 Z`;
    }
    render() {
        const { x: markerX, y: markerY } = this.currentMarkerPos;
        return (h("svg", { key: 'ae4f6c6ce47f55d7be763bc451a82c6296c8871e', width: this.width, height: this.height, viewBox: "0 0 130 66", fill: "none", xmlns: "http://www.w3.org/2000/svg" }, h("defs", { key: '373160ff8e80c0771458d48eb57da931b33815c5' }, h("linearGradient", { key: 'c5b44dcaefc231d0f57c2b075f4c1cfb299fdfe3', id: "gradientMask", x1: "64", y1: "0", x2: "64", y2: "64", gradientUnits: "userSpaceOnUse" }, h("stop", { key: '93e39951f14f45d6e549820999d09e5757fd4f6f' }), h("stop", { key: '4168b4b2042bdd02c37dc0b513eef83f29e79611', offset: "1", "stop-opacity": "0" })), h("mask", { key: '935ec0798a2cc114e46df0a066fc1fe10478f816', id: "chartMask", style: { maskType: 'alpha' }, maskUnits: "userSpaceOnUse", x: "1", y: "1", width: "128", height: "64" }, h("rect", { key: '44a4f224977ec36e70dd7c7b99597015da8b6086', width: "128", height: "64", transform: "translate(1 1)", fill: "url(#gradientMask)" }))), h("g", { key: 'db2f234e332c89a7979bd8cc2ce6fbb61c84f7fe', mask: "url(#chartMask)" }, h("path", { key: '294437166a6343bb153a92ebd9cfdf51bb7de505', opacity: "0.2", d: this.areaD, fill: this.fillColor })), h("path", { key: '57d7bc4921af8228960ce546b72f1f827225ea72', d: this.pathD, stroke: this.strokeColor, "stroke-width": this.strokeWidth, "stroke-linecap": "round", "stroke-linejoin": "round", fill: "none" }), this.showMarker && (h("g", { key: '682ceb408c8a4e3ce22b38a7fe12b093801a8ee2', transform: `translate(${markerX}, ${markerY})` }, h("g", { key: '81446c80be31b90e89538e4546bd98462f17721d', opacity: "0.2" }, h("rect", { key: '1ba0df8a0ec569126b421f81c1edafd4199c3c48', width: "20", height: "20", rx: "10", stroke: this.strokeColor, "stroke-width": this.strokeWidth })), h("rect", { key: '8fdb7874d2a8c679584266173041026d55e2af80', x: "5", y: "5", width: "10", height: "10", rx: "5", fill: "white" }), h("rect", { key: 'd97a882fea1dccce6a7a5f6a6f0e1af5aeab8568', x: "5", y: "5", width: "10", height: "10", rx: "5", stroke: this.strokeColor, "stroke-width": this.strokeWidth })))));
    }
    static get watchers() { return {
        "data": ["dataChangeHandler"]
    }; }
}, [1, "dga-metric-chart-v2", {
        "width": [2],
        "height": [2],
        "strokeColor": [1, "stroke-color"],
        "fillColor": [1, "fill-color"],
        "strokeWidth": [2, "stroke-width"],
        "data": [16],
        "showMarker": [4, "show-marker"],
        "markerPosition": [16, "marker-position"]
    }, undefined, {
        "data": ["dataChangeHandler"]
    }]);
function defineCustomElement$1() {
    if (typeof customElements === "undefined") {
        return;
    }
    const components = ["dga-metric-chart-v2"];
    components.forEach(tagName => { switch (tagName) {
        case "dga-metric-chart-v2":
            if (!customElements.get(tagName)) {
                customElements.define(tagName, DgaMetricChartV2$1);
            }
            break;
    } });
}

const DgaMetricChartV2 = DgaMetricChartV2$1;
const defineCustomElement = defineCustomElement$1;

export { DgaMetricChartV2, defineCustomElement };
