// @platformscode/core/dist/components/dga-loading-usage.js
import { p as proxyCustomElement, H, h } from './p-BhqVmlhr.js';
import { d as defineCustomElement$2 } from './p-D3vljtDG.js';

const DgaLoadingUsage$1 = /*@__PURE__*/ proxyCustomElement(class DgaLoadingUsage extends H {
    constructor() {
        super();
        this.__registerHost();
        this.activeStep = 1;
    }
    render() {
        return (h("div", { key: '64474af61f174994d963aa6557b10dc00d761310' }, h("h1", { key: '805a9ac2cb77eacae5e5409ea0aab1ae0e8cd967' }, "Loading Example"), h("div", { key: '834250a21bfd4fbdec13cfccf26ea1f02ea8e5a3', style: { padding: "50px", display: "flex", gap: "20px" } }, h("div", { key: 'b074bfde05c2f7aa5f6df13882d92145193f6f10', style: { display: "flex", flexDirection: "column", gap: "20px" } }, h("dga-loading", { key: 'b305a40be50630b0ee1cd7ea53ff55ff535ffe48' })), h("div", { key: 'abd79345ad1690fa7b54751587d20bc2be13db59', style: { display: "flex", flexDirection: "column", gap: "20px" } }))));
    }
}, [0, "dga-loading-usage", {
        "activeStep": [32]
    }]);
function defineCustomElement$1() {
    if (typeof customElements === "undefined") {
        return;
    }
    const components = ["dga-loading-usage", "dga-loading"];
    components.forEach(tagName => { switch (tagName) {
        case "dga-loading-usage":
            if (!customElements.get(tagName)) {
                customElements.define(tagName, DgaLoadingUsage$1);
            }
            break;
        case "dga-loading":
            if (!customElements.get(tagName)) {
                defineCustomElement$2();
            }
            break;
    } });
}

const DgaLoadingUsage = DgaLoadingUsage$1;
const defineCustomElement = defineCustomElement$1;

export { DgaLoadingUsage, defineCustomElement };
