// @platformscode/core/dist/components/dga-file-upload-usage.js
import { p as proxyCustomElement, H, h } from './p-BhqVmlhr.js';
import { d as defineCustomElement$3 } from './p-Bty5NdFA.js';
import { d as defineCustomElement$2 } from './p-D3vljtDG.js';

const DgaFileUploadUsage$1 = /*@__PURE__*/ proxyCustomElement(class DgaFileUploadUsage extends H {
    constructor() {
        super();
        this.__registerHost();
    }
    render() {
        return (h("div", { key: '8d844034a8a1119e086c1c79917dee196a05c9de' }, h("h1", { key: 'f04893452fcbb4301ae1692d12aa249aec01fa8f' }, "File Upload Example"), h("div", { key: '0e2546b42b8abe6698a778904106fc9344a4c499', style: { padding: "50px", display: "flex", gap: "20px" } }, h("div", { key: 'f0555b930f9541243c0dc06947ca648543420b28', style: { display: "flex", flexDirection: "column", gap: "20px" } }, h("dga-file-upload", { key: '60de6291a200bef5ef1c1024c12560efe113f7cf' })), h("div", { key: 'd5eff3e21f23dd1afeadae2bdcf5f0babe3548c7', style: { display: "flex", flexDirection: "column", gap: "20px" } }))));
    }
}, [0, "dga-file-upload-usage"]);
function defineCustomElement$1() {
    if (typeof customElements === "undefined") {
        return;
    }
    const components = ["dga-file-upload-usage", "dga-file-upload", "dga-loading"];
    components.forEach(tagName => { switch (tagName) {
        case "dga-file-upload-usage":
            if (!customElements.get(tagName)) {
                customElements.define(tagName, DgaFileUploadUsage$1);
            }
            break;
        case "dga-file-upload":
            if (!customElements.get(tagName)) {
                defineCustomElement$3();
            }
            break;
        case "dga-loading":
            if (!customElements.get(tagName)) {
                defineCustomElement$2();
            }
            break;
    } });
}

const DgaFileUploadUsage = DgaFileUploadUsage$1;
const defineCustomElement = defineCustomElement$1;

export { DgaFileUploadUsage, defineCustomElement };
