// @platformscode/core/dist/components/dga-tabs-usage.js
import { p as proxyCustomElement, H, h } from './p-BhqVmlhr.js';
import { d as defineCustomElement$4 } from './p-CClx_RDs.js';
import { d as defineCustomElement$3 } from './p-DN3DJd5E.js';
import { d as defineCustomElement$2 } from './p-CzuS0Atf.js';

const DgaTabsUsage$1 = /*@__PURE__*/ proxyCustomElement(class DgaTabsUsage extends H {
    constructor() {
        super();
        this.__registerHost();
    }
    render() {
        return (h("div", { key: '46b388e665d5eefa731433b3777327dfdd5d4236' }, h("h1", { key: '81c3932a8e580600bd32586bfdc8aad52ad4e702' }, "Tabs Example"), h("h2", { key: 'c329290427c9e2defbbea4c9d95edfdf68b15ee6' }, "Horizontal Tabs"), h("dga-tabs", { key: 'ee9623580bd4895ba43deb794a11b95b2e8c0b61', size: "md", orientation: "vertical",
            // tabIcons: true,
            flush: false, external: true, divider: true, tabsList: [
                {
                    label: 'First Tab',
                    onClick: () => { },
                    tabIcon: "home-01",
                    iconProps: {
                        variant: "stroke",
                        type: "sharp"
                    }
                },
                {
                    label: 'Second Tab',
                    link: '#',
                    onClick: () => { },
                    tabIcon: `<i class="hgi-stroke hgi-sharp hgi-home-01" style={{ fontSize: '16px' }} />`,
                },
                {
                    label: 'Third Tab',
                    link: '#',
                    onClick: () => { },
                    tabIcon: `<i class="hgi-stroke hgi-sharp hgi-home-01" style={{ fontSize: '16px' }} />`,
                },
                {
                    label: 'Fourth Tab',
                    link: '#',
                    onClick: () => { },
                    tabIcon: `<i class="hgi-stroke hgi-sharp hgi-home-01" style={{ fontSize: '16px' }} />`,
                },
            ] }), h("dga-tabs", { key: '3b5d016cb5d461d06799cec8e2928e9d0a7a93c0', size: "md", orientation: "horizontal",
            // tabIcons: true,
            flush: false, external: true, divider: true, tabsList: [
                {
                    label: 'First Tab',
                    onClick: () => { },
                    tabIcon: "home-01",
                    iconProps: {
                        variant: "stroke",
                        type: "sharp"
                    }
                },
                {
                    label: 'Second Tab',
                    link: '#',
                    onClick: () => { },
                    tabIcon: `<i class="hgi-stroke hgi-sharp hgi-home-01" style={{ fontSize: '16px' }} />`,
                },
                {
                    label: 'Third Tab',
                    link: '#',
                    onClick: () => { },
                    tabIcon: `<i class="hgi-stroke hgi-sharp hgi-home-01" style={{ fontSize: '16px' }} />`,
                },
                {
                    label: 'Fourth Tab',
                    link: '#',
                    onClick: () => { },
                    tabIcon: `<i class="hgi-stroke hgi-sharp hgi-home-01" style={{ fontSize: '16px' }} />`,
                },
                {
                    label: 'Fifth Tab',
                    onClick: () => { },
                    tabIcon: "home-01",
                    iconProps: {
                        variant: "stroke",
                        type: "sharp"
                    }
                },
                {
                    label: 'Sixth Tab',
                    onClick: () => { },
                    tabIcon: "home-01",
                    iconProps: {
                        variant: "stroke",
                        type: "sharp"
                    }
                },
            ] }), h("h2", { key: 'c9903dff8692e05a79b2439a4c7ccb34d09a0773' }, "Disabled Tabs"), h("dga-tabs", { key: '6ce853a7ac35a1fe861f5f7639f9f6cdaaf27916', size: "md", orientation: "horizontal",
            // tabIcons={true}
            flush: false, external: true, divider: true, disabled: true, tabsList: [
                { label: 'First Tab', link: '#', tabIcon: `<i class="hgi-stroke hgi-sharp hgi-home-01" style={{ fontSize: '16px' }} />` },
                { label: 'Second Tab', link: '#', tabIcon: `<i class="hgi-stroke hgi-sharp hgi-home-01" style={{ fontSize: '16px' }} />` },
                { label: 'Third Tab', link: '#', tabIcon: `<i class="hgi-stroke hgi-sharp hgi-home-01" style={{ fontSize: '16px' }} />` },
                { label: 'Fourth Tab', link: '#', tabIcon: `<i class="hgi-stroke hgi-sharp hgi-home-01" style={{ fontSize: '16px' }} />` },
            ] })));
    }
}, [0, "dga-tabs-usage"]);
function defineCustomElement$1() {
    if (typeof customElements === "undefined") {
        return;
    }
    const components = ["dga-tabs-usage", "dga-button", "dga-icon", "dga-tabs"];
    components.forEach(tagName => { switch (tagName) {
        case "dga-tabs-usage":
            if (!customElements.get(tagName)) {
                customElements.define(tagName, DgaTabsUsage$1);
            }
            break;
        case "dga-button":
            if (!customElements.get(tagName)) {
                defineCustomElement$4();
            }
            break;
        case "dga-icon":
            if (!customElements.get(tagName)) {
                defineCustomElement$3();
            }
            break;
        case "dga-tabs":
            if (!customElements.get(tagName)) {
                defineCustomElement$2();
            }
            break;
    } });
}

const DgaTabsUsage = DgaTabsUsage$1;
const defineCustomElement = defineCustomElement$1;

export { DgaTabsUsage, defineCustomElement };
