// @platformscode/core/dist/components/dga-floating-button-usage.js
import { p as proxyCustomElement, H, h } from './p-BhqVmlhr.js';
import { d as defineCustomElement$3 } from './p-DN3DJd5E.js';
import { d as defineCustomElement$2 } from './p-TGFiIztK.js';

const DgaButtonUsage = /*@__PURE__*/ proxyCustomElement(class DgaButtonUsage extends H {
    constructor() {
        super();
        this.__registerHost();
    }
    render() {
        return (h("div", { key: '2e3f91ead6798c6f522b85585026ebe30260a477', style: {
                margin: "10px"
            } }, h("floating-button", { key: '0617cf8adc46c27663dd5ad0564f285b0b92b4ae', variant: "secondary-solid" }), h("div", { key: '42979b16af8b296262f40198e7960e6e05172333', style: {
                marginTop: "10px",
            } }, h("floating-button", { key: 'ec8ac68dfc1707e29a05928c5fba231b377f9b28', variant: "primary-neutral", iconOnly: true }))));
    }
}, [0, "dga-floating-button-usage"]);
function defineCustomElement$1() {
    if (typeof customElements === "undefined") {
        return;
    }
    const components = ["dga-floating-button-usage", "dga-icon", "floating-button"];
    components.forEach(tagName => { switch (tagName) {
        case "dga-floating-button-usage":
            if (!customElements.get(tagName)) {
                customElements.define(tagName, DgaButtonUsage);
            }
            break;
        case "dga-icon":
            if (!customElements.get(tagName)) {
                defineCustomElement$3();
            }
            break;
        case "floating-button":
            if (!customElements.get(tagName)) {
                defineCustomElement$2();
            }
            break;
    } });
}

const DgaFloatingButtonUsage = DgaButtonUsage;
const defineCustomElement = defineCustomElement$1;

export { DgaFloatingButtonUsage, defineCustomElement };
