// @platformscode/core/dist/components/dga-radial-stepper-usage.js
import { p as proxyCustomElement, H, h } from './p-BhqVmlhr.js';
import { d as defineCustomElement$3 } from './p-CCimZGO2.js';
import { d as defineCustomElement$2 } from './p-O09aJ_YO.js';

const DgaRadialStepperUsage$1 = /*@__PURE__*/ proxyCustomElement(class DgaRadialStepperUsage extends H {
    constructor() {
        super();
        this.__registerHost();
        this.percentage = 0;
        this.activeStep = 0;
    }
    handleInputChange(event) {
        const target = event.target;
        this.percentage = Number(target.value); // Convert the string value to a number
    }
    handleStepChange(event) {
        const target = event.target;
        this.activeStep = Number(target.value); // Convert the string value to a number
    }
    render() {
        return (h("div", { key: '9c47326563e04d03b5422f50a428cb623cce6d65' }, h("h1", { key: '97cddb162e9bc7d8c83025204612a6d4de56c038' }, "Radial Stepper Example"), h("div", { key: 'cadbed3beda2ac862eb34ce14692bf9a7e34ef2a', style: { padding: '50px', gap: '20px' } }, h("div", { key: 'af0b6c808347210dcc234c9f4b692618dd53b58a', style: { display: 'flex', flexDirection: 'column', gap: '20px' } }, h("dga-radial-stepper", { key: '7bdd6ecd7c6d4294f8fe8bd5647730adceb9aa62', size: "120px", activeStep: this.activeStep, steps: [
                {
                    stepName: "Step One",
                    stepDescription: "this is step one description",
                    showPreStep: true,
                    showNextStep: true
                },
                {
                    stepName: "Step Two",
                    stepDescription: "this is step two description",
                    showPreStep: true,
                    showNextStep: true
                },
                {
                    stepName: "Step Three",
                    stepDescription: "this is step three description",
                    showPreStep: true,
                    showNextStep: true
                },
                {
                    stepName: "Step Four",
                    stepDescription: "this is step four description",
                    showPreStep: true,
                    showNextStep: true
                }
            ] }), h("dga-radial-stepper", { key: '27992509962a448e713582ae0bdd9228cc508e0a', size: "80px", color: "success", activeStep: this.activeStep, steps: [
                {
                    stepName: "Step One",
                    stepDescription: "this is step one description",
                    showPreStep: true,
                    showNextStep: true
                },
                {
                    stepName: "Step Two",
                    stepDescription: "this is step two description",
                    showPreStep: true,
                    showNextStep: true
                },
                {
                    stepName: "Step Three",
                    stepDescription: "this is step three description",
                    showPreStep: true,
                    showNextStep: true
                },
                {
                    stepName: "Step Four",
                    stepDescription: "this is step four description",
                    showPreStep: true,
                    showNextStep: true
                }
            ] }), h("dga-radial-stepper", { key: '9d42d0c4b192722c16e0309a7fd48f37f2207c5c', variant: 'neutral', size: "64px", activeStep: this.activeStep, steps: [
                {
                    stepName: "Step One",
                    stepDescription: "this is step one description",
                    showPreStep: true,
                    showNextStep: true
                },
                {
                    stepName: "Step Two",
                    stepDescription: "this is step two description",
                    showPreStep: true,
                    showNextStep: true
                },
                {
                    stepName: "Step Three",
                    stepDescription: "this is step three description",
                    showPreStep: true,
                    showNextStep: true
                },
                {
                    stepName: "Step Four",
                    stepDescription: "this is step four description",
                    showPreStep: true,
                    showNextStep: true
                }
            ] }), h("dga-radial-stepper", { key: 'bb8494cec27c5057efa6ac98ac8310a28974885d', size: "48px", activeStep: this.activeStep, steps: [
                {
                    stepName: "Step One",
                    stepDescription: "this is step one description",
                    showPreStep: true,
                    showNextStep: true
                },
                {
                    stepName: "Step Two",
                    stepDescription: "this is step two description",
                    showPreStep: true,
                    showNextStep: true
                },
                {
                    stepName: "Step Three",
                    stepDescription: "this is step three description",
                    showPreStep: true,
                    showNextStep: true
                },
                {
                    stepName: "Step Four",
                    stepDescription: "this is step four description",
                    showPreStep: true,
                    showNextStep: true
                }
            ] }), h("dga-radial-stepper", { key: 'c268850e80ea824df23f96261c3459c21f816e45', size: "40px", activeStep: this.activeStep, steps: [
                {
                    stepName: "Step One",
                    stepDescription: "this is step one description",
                    showPreStep: true,
                    showNextStep: true
                },
                {
                    stepName: "Step Two",
                    stepDescription: "this is step two description",
                    showPreStep: true,
                    showNextStep: true
                },
                {
                    stepName: "Step Three",
                    stepDescription: "this is step three description",
                    showPreStep: true,
                    showNextStep: true
                },
                {
                    stepName: "Step Four",
                    stepDescription: "this is step four description",
                    showPreStep: true,
                    showNextStep: true
                }
            ] }), h("div", { key: '111539331b3adfd028ef4386e6427f8f7f772327', style: {
                backgroundColor: "#1B8354",
                padding: "40px"
            } }, h("dga-radial-stepper", { key: 'ff0107615055b8fa120de77d0fb9c7802db933b0', variant: "on-color", size: "120px", activeStep: this.activeStep, steps: [
                {
                    stepName: "Step One",
                    stepDescription: "this is step one description",
                    showPreStep: true,
                    showNextStep: true
                },
                {
                    stepName: "Step Two",
                    stepDescription: "this is step two description",
                    showPreStep: true,
                    showNextStep: true
                },
                {
                    stepName: "Step Three",
                    stepDescription: "this is step three description",
                    showPreStep: true,
                    showNextStep: true
                },
                {
                    stepName: "Step Four",
                    stepDescription: "this is step four description",
                    showPreStep: true,
                    showNextStep: true
                },
                {
                    stepName: "Step Five",
                    stepDescription: "this is step four description",
                    showPreStep: true,
                    showNextStep: true
                },
                {
                    stepName: "Step Six",
                    stepDescription: "this is step four description",
                    showPreStep: true,
                    showNextStep: true
                }
            ], RTL: true })), h("button", { key: '20fa6550f6df7d0daeed82e39ae4b307c81ee2a9', onClick: () => (this.activeStep += 1) }, "Next"), h("button", { key: '8c380c9a63df6dc7db9708d5d1ce7a7c8d63b00f', onClick: () => (this.activeStep -= 1) }, "Prev"), h("input", { key: '77b2171cc10e239f9fd915451b78dc2f4c47cb56', type: "range", min: "0", max: "100", value: this.percentage, onInput: e => this.handleInputChange(e) })))));
    }
}, [0, "dga-radial-stepper-usage", {
        "percentage": [32],
        "activeStep": [32]
    }]);
function defineCustomElement$1() {
    if (typeof customElements === "undefined") {
        return;
    }
    const components = ["dga-radial-stepper-usage", "dga-flex", "dga-radial-stepper"];
    components.forEach(tagName => { switch (tagName) {
        case "dga-radial-stepper-usage":
            if (!customElements.get(tagName)) {
                customElements.define(tagName, DgaRadialStepperUsage$1);
            }
            break;
        case "dga-flex":
            if (!customElements.get(tagName)) {
                defineCustomElement$3();
            }
            break;
        case "dga-radial-stepper":
            if (!customElements.get(tagName)) {
                defineCustomElement$2();
            }
            break;
    } });
}

const DgaRadialStepperUsage = DgaRadialStepperUsage$1;
const defineCustomElement = defineCustomElement$1;

export { DgaRadialStepperUsage, defineCustomElement };
