// @platformscode/core/dist/components/dga-textarea-usage.js
import { p as proxyCustomElement, H, h } from './p-BhqVmlhr.js';
import { d as defineCustomElement$3 } from './p-DN3DJd5E.js';
import { d as defineCustomElement$2 } from './p-CrDgQbGi.js';

const DgaTextareaUsage$1 = /*@__PURE__*/ proxyCustomElement(class DgaTextareaUsage extends H {
    constructor() {
        super();
        this.__registerHost();
    }
    render() {
        return (h("div", { key: '748625590a5502649b3f9702bcc3194685650481' }, h("h1", { key: 'fcce936a711f31ed2398006f0c9a7f4b51693de3' }, "Textarea Example"), h("div", { key: '0f5ad2ee8329cd185749f2d25988a8b7a0edec1a', style: { padding: '50px', display: 'flex', gap: '20px' } }, h("div", { key: 'a80a3c37b040b684d958479a671b6f4389939f11', style: { display: 'flex', flexDirection: 'column', gap: '20px' } }, h("dga-textarea", { key: '9b30321514486269bbbd7158ca91904403ee9363', label: "Label", placeholder: "Placeholder...", scrollbar: true, readonly: true }), h("dga-textarea", { key: '2af70954e218755f015486f0c9295fd5678302d1', placeholder: "Placeholder...", scrollbar: true })), h("div", { key: '4a905c203029c8ee581cf35d7fa85b2eef05fc7a', style: { display: 'flex', flexDirection: 'column', gap: '20px' } }))));
    }
}, [0, "dga-textarea-usage"]);
function defineCustomElement$1() {
    if (typeof customElements === "undefined") {
        return;
    }
    const components = ["dga-textarea-usage", "dga-icon", "dga-textarea"];
    components.forEach(tagName => { switch (tagName) {
        case "dga-textarea-usage":
            if (!customElements.get(tagName)) {
                customElements.define(tagName, DgaTextareaUsage$1);
            }
            break;
        case "dga-icon":
            if (!customElements.get(tagName)) {
                defineCustomElement$3();
            }
            break;
        case "dga-textarea":
            if (!customElements.get(tagName)) {
                defineCustomElement$2();
            }
            break;
    } });
}

const DgaTextareaUsage = DgaTextareaUsage$1;
const defineCustomElement = defineCustomElement$1;

export { DgaTextareaUsage, defineCustomElement };
