// @platformscode/core/dist/components/dga-notification-usage.js
import { p as proxyCustomElement, H, h } from './p-BhqVmlhr.js';
import { d as defineCustomElement$3 } from './p-DBTOrdV2.js';
import { d as defineCustomElement$2 } from './p-aug08tA7.js';

const DgaNotificationUsage$1 = /*@__PURE__*/ proxyCustomElement(class DgaNotificationUsage extends H {
    constructor() {
        super();
        this.__registerHost();
        this.variant = 'neutral';
    }
    render() {
        return (h("div", { key: '57471f201e667cb1810076e695d89558e5a1b234' }, h("h1", { key: '417634a288b5bdb76abd26a44b78e0fba6a2c7b0' }, "Notification Example"), h("div", { key: 'cfea8bf7ea9aa2fb1b18fbb66d3586c06df8ca4b', style: { padding: '50px', display: 'flex', gap: '20px' } }, h("div", { key: '6b784b2a1c2b5165aebee483e7f2ff5fac630902', style: { display: 'flex', flexDirection: 'column', gap: '20px' } }, h("dga-notification", { key: '32ae958959fed4b91eaf026201d96ef8c3014eef', leadText: "This is a very important banner message that requires attention", variant: this.variant, dismissable: true }), h("dga-notification", { key: '97bee463828d792dea10b5386a9fd8bbfbeab35c', leadText: "This is a very important banner message that requires attention", variant: "critical", dismissable: true }), h("dga-notification", { key: 'b6b3aa27eebe463f58a771fac6cdbcec8906803d', leadText: "This is a very important banner message that requires attention", variant: "info", dismissable: true }), h("dga-notification", { key: '7a23725c11f7b14b4112d3283996485cea34fd6d', leadText: "This is a very important banner message that requires attention", variant: "success", dismissable: true }), h("dga-notification", { key: '44591f2e2da8d9e5611930d14fb612e2af025327', leadText: "This is a very important banner message that requires attention", variant: "warning", dismissable: true })), h("div", { key: '9363f097d2bc685bcdc0a7076ae6347ee3fa749d', style: { display: 'flex', flexDirection: 'column', gap: '20px' } }))));
    }
}, [0, "dga-notification-usage", {
        "variant": [32]
    }]);
function defineCustomElement$1() {
    if (typeof customElements === "undefined") {
        return;
    }
    const components = ["dga-notification-usage", "dga-icon-v2", "dga-notification"];
    components.forEach(tagName => { switch (tagName) {
        case "dga-notification-usage":
            if (!customElements.get(tagName)) {
                customElements.define(tagName, DgaNotificationUsage$1);
            }
            break;
        case "dga-icon-v2":
            if (!customElements.get(tagName)) {
                defineCustomElement$3();
            }
            break;
        case "dga-notification":
            if (!customElements.get(tagName)) {
                defineCustomElement$2();
            }
            break;
    } });
}

const DgaNotificationUsage = DgaNotificationUsage$1;
const defineCustomElement = defineCustomElement$1;

export { DgaNotificationUsage, defineCustomElement };
