// @platformscode/core/dist/components/dga-skeleton-usage.js
import { p as proxyCustomElement, H, h } from './p-BhqVmlhr.js';
import { d as defineCustomElement$6 } from './p-CUHFypnM.js';
import { d as defineCustomElement$5 } from './p-BBXZs54p.js';
import { d as defineCustomElement$4 } from './p-CdliX_b4.js';
import { d as defineCustomElement$3 } from './p-usZ1DZDb.js';
import { d as defineCustomElement$2 } from './p-BrEdmVoK.js';

const LineSkeleton = /*@__PURE__*/ proxyCustomElement(class LineSkeleton extends H {
    constructor() {
        super();
        this.__registerHost();
    }
    render() {
        return (h("div", { key: '02c9f389c2764d140a242f0bccaf8148912e1cb1' }, h("dga-line-skeleton", { key: 'eb5f67fac73d05a80598830978f5512824082fa5' }), h("dga-circle-skeleton", { key: 'ae8e96903d12cf1d7bb587f891ed850d870efd78' }), h("dga-rectangle-skeleton", { key: '2db923e5932658a0ffb97deb26dd4fea6ad3646e' }), h("dga-square-skeleton", { key: '6197130455b87510adfb1a3ed3c44aa819c85410' }), h("dga-skeleton-component", { key: '6a6c434e1dcc65a8fa93c0c8089bc4dd53f3a290' })));
    }
}, [0, "dga-skeleton-usage"]);
function defineCustomElement$1() {
    if (typeof customElements === "undefined") {
        return;
    }
    const components = ["dga-skeleton-usage", "dga-circle-skeleton", "dga-line-skeleton", "dga-rectangle-skeleton", "dga-skeleton-component", "dga-square-skeleton"];
    components.forEach(tagName => { switch (tagName) {
        case "dga-skeleton-usage":
            if (!customElements.get(tagName)) {
                customElements.define(tagName, LineSkeleton);
            }
            break;
        case "dga-circle-skeleton":
            if (!customElements.get(tagName)) {
                defineCustomElement$6();
            }
            break;
        case "dga-line-skeleton":
            if (!customElements.get(tagName)) {
                defineCustomElement$5();
            }
            break;
        case "dga-rectangle-skeleton":
            if (!customElements.get(tagName)) {
                defineCustomElement$4();
            }
            break;
        case "dga-skeleton-component":
            if (!customElements.get(tagName)) {
                defineCustomElement$3();
            }
            break;
        case "dga-square-skeleton":
            if (!customElements.get(tagName)) {
                defineCustomElement$2();
            }
            break;
    } });
}

const DgaSkeletonUsage = LineSkeleton;
const defineCustomElement = defineCustomElement$1;

export { DgaSkeletonUsage, defineCustomElement };
