// @platformscode/core/dist/components/dga-date-field-usage.js
import { p as proxyCustomElement, H, h } from './p-BhqVmlhr.js';
import { d as defineCustomElement$b } from './p-BUKkxyIA.js';
import { d as defineCustomElement$a } from './p-B4U0lY4y.js';
import { d as defineCustomElement$9 } from './p-DLdb5jCw.js';
import { d as defineCustomElement$8 } from './p--saaGsBa.js';
import { d as defineCustomElement$7 } from './p-BDegpq_a.js';
import { d as defineCustomElement$6 } from './p-DN3DJd5E.js';
import { d as defineCustomElement$5 } from './p-DBTOrdV2.js';
import { d as defineCustomElement$4 } from './p-Cxhq_uxn.js';
import { d as defineCustomElement$3 } from './p-C99Dr3jt.js';
import { d as defineCustomElement$2 } from './p-C8cyYIhA.js';

const DgaDateFieldUsage$1 = /*@__PURE__*/ proxyCustomElement(class DgaDateFieldUsage extends H {
    constructor() {
        super();
        this.__registerHost();
        this.value = '';
    }
    render() {
        return (h("div", { key: 'e47493fd39924f450f92f9e8f68c043c12e4df5c' }, h("h1", { key: 'd4cf69d43f38b808dbb9830708c17d7912c840e6' }, "Date Field Example"), h("div", { key: '551a9f94a3071badcf01243d8b5ee3184756bd8f', style: { padding: '50px' } }, h("div", { key: 'c3a10cfbadd77ff7ba5a2e41ba4752dffad47b6f', style: { display: "flex", flexDirection: "column", gap: "16px" } }, h("input", { key: '07c26c08b8c643dcbe889f40c088354996e46493', type: "date", value: this.value, onChange: (e) => this.value = e.target.value }), h("dga-date-field", { key: '8696032ed67ff748798102720aabcd78952818b7', value: this.value, onChange: (e) => this.value = e.target.value })))));
    }
}, [0, "dga-date-field-usage", {
        "value": [32]
    }]);
function defineCustomElement$1() {
    if (typeof customElements === "undefined") {
        return;
    }
    const components = ["dga-date-field-usage", "date-cell", "dga-button-v2", "dga-date-field", "dga-datepicker", "dga-helper-text", "dga-icon", "dga-icon-v2", "dga-label", "dga-text-input", "month-navigator"];
    components.forEach(tagName => { switch (tagName) {
        case "dga-date-field-usage":
            if (!customElements.get(tagName)) {
                customElements.define(tagName, DgaDateFieldUsage$1);
            }
            break;
        case "date-cell":
            if (!customElements.get(tagName)) {
                defineCustomElement$b();
            }
            break;
        case "dga-button-v2":
            if (!customElements.get(tagName)) {
                defineCustomElement$a();
            }
            break;
        case "dga-date-field":
            if (!customElements.get(tagName)) {
                defineCustomElement$9();
            }
            break;
        case "dga-datepicker":
            if (!customElements.get(tagName)) {
                defineCustomElement$8();
            }
            break;
        case "dga-helper-text":
            if (!customElements.get(tagName)) {
                defineCustomElement$7();
            }
            break;
        case "dga-icon":
            if (!customElements.get(tagName)) {
                defineCustomElement$6();
            }
            break;
        case "dga-icon-v2":
            if (!customElements.get(tagName)) {
                defineCustomElement$5();
            }
            break;
        case "dga-label":
            if (!customElements.get(tagName)) {
                defineCustomElement$4();
            }
            break;
        case "dga-text-input":
            if (!customElements.get(tagName)) {
                defineCustomElement$3();
            }
            break;
        case "month-navigator":
            if (!customElements.get(tagName)) {
                defineCustomElement$2();
            }
            break;
    } });
}

const DgaDateFieldUsage = DgaDateFieldUsage$1;
const defineCustomElement = defineCustomElement$1;

export { DgaDateFieldUsage, defineCustomElement };
