// @platformscode/core/dist/components/dga-progress-indicator-v3-usage.js
import { p as proxyCustomElement, H, h } from './p-BhqVmlhr.js';
import { d as defineCustomElement$3 } from './p-CBFY-ks-.js';
import { d as defineCustomElement$2 } from './p-pwESyQzO.js';

const DgaProgressIndicatorUsage = /*@__PURE__*/ proxyCustomElement(class DgaProgressIndicatorUsage extends H {
    constructor() {
        super();
        this.__registerHost();
        this.activeStep = 1;
        this.activeStep2 = 1;
    }
    render() {
        console.log(this.activeStep);
        return (h("div", { key: '2e67332be3cbc09aa1f52c603b38b40a425b4ab5' }, h("h1", { key: 'c36873fdbb35ed744706d71832e111e6516dc7bc' }, "Progress Indicator Example"), h("div", { key: '9012f1073f3b65dbc117d6ef397cdbc742c2b45b', style: { padding: '50px', display: 'flex', gap: '20px' } }, h("div", { key: '2a8bd76a479b6a898cca37827871299289ff7ef8', style: { display: 'flex', flexDirection: 'column', gap: '20px' } }, h("button", { key: 'a0cbba9d112124f8046c9681d69d96c084c26c55', onClick: () => (this.activeStep = this.activeStep - 1) }, "prev"), h("button", { key: '03b46ad5b402620bff27e75975261e312081aeb7', onClick: () => (this.activeStep = this.activeStep + 1) }, "next"), h("button", { key: 'ac53fb0d1725e6fc4cfbcf2853e0cacf4f5f3a13', onClick: () => (this.activeStep2 = this.activeStep2 - 1) }, "prev 2"), h("button", { key: 'd7b0f137c48c1598797b493a11d6689b07ab273f', onClick: () => (this.activeStep2 = this.activeStep2 + 1) }, "next 2"), "activeStep: ", this.activeStep, h("dga-progress-indicator", { key: 'ec91d9c12847d8171ab3889546bd02041e50336b', activeStep: this.activeStep, alignment: 'vertical' }, h("dga-progress-indicator-step", { key: 'e15acb2f02f8f0307700f0348d596c167db2fd91', title: "Step One", description: "this is step description", currentStep: 1 }), h("dga-progress-indicator-step", { key: '09ac63b35b0e1b6e8b9f6d2a6cdc8e3f1aa53611', title: "Step Two", description: "this is step description", currentStep: 2 }), h("dga-progress-indicator-step", { key: 'c6d55428303b4844bd4f6886bbb0359efc279c94', title: "Step Three", description: "this is step description", nextStep: false, currentStep: 3 })), h("dga-progress-indicator", { key: '3da8bf9e08d55d927a156c7327a8e4b70a2eb35f', activeStep: this.activeStep2, alignment: 'vertical' }, h("dga-progress-indicator-step", { key: '923dd3327a310769f5771acd1f7efeea38e32355', title: "Step One", description: "this is step description", currentStep: 1 }), h("dga-progress-indicator-step", { key: '80634516df4b935ee481db880bdfb35935a09391', title: "Step Two", description: "this is step description", currentStep: 2 }), h("dga-progress-indicator-step", { key: 'fb028923ac03aa0a1846b12b625db1b39277b6db', title: "Step Three", description: "this is step description", nextStep: false, currentStep: 3 }))), h("div", { key: 'c7e55f39f75b551d218b58788d832115c15fcee2', style: { display: 'flex', flexDirection: 'column', gap: '20px' } }))));
    }
}, [0, "dga-progress-indicator-v3-usage", {
        "activeStep": [32],
        "activeStep2": [32]
    }]);
function defineCustomElement$1() {
    if (typeof customElements === "undefined") {
        return;
    }
    const components = ["dga-progress-indicator-v3-usage", "dga-progress-indicator", "dga-progress-indicator-step"];
    components.forEach(tagName => { switch (tagName) {
        case "dga-progress-indicator-v3-usage":
            if (!customElements.get(tagName)) {
                customElements.define(tagName, DgaProgressIndicatorUsage);
            }
            break;
        case "dga-progress-indicator":
            if (!customElements.get(tagName)) {
                defineCustomElement$3();
            }
            break;
        case "dga-progress-indicator-step":
            if (!customElements.get(tagName)) {
                defineCustomElement$2();
            }
            break;
    } });
}

const DgaProgressIndicatorV3Usage = DgaProgressIndicatorUsage;
const defineCustomElement = defineCustomElement$1;

export { DgaProgressIndicatorV3Usage, defineCustomElement };
