// @platformscode/core/dist/components/dga-filteration-usage.js
import { p as proxyCustomElement, H, h } from './p-BhqVmlhr.js';
import { d as defineCustomElement$c } from './p-B4U0lY4y.js';
import { d as defineCustomElement$b } from './p-ChhNw26D.js';
import { d as defineCustomElement$a } from './p-DTQNJ3_g.js';
import { d as defineCustomElement$9 } from './p-Ds6Hp4Ow.js';
import { d as defineCustomElement$8 } from './p-BDegpq_a.js';
import { d as defineCustomElement$7 } from './p-DN3DJd5E.js';
import { d as defineCustomElement$6 } from './p-DBTOrdV2.js';
import { d as defineCustomElement$5 } from './p-Cxhq_uxn.js';
import { d as defineCustomElement$4 } from './p-BldfN9G_.js';
import { d as defineCustomElement$3 } from './p-BE95gJkd.js';
import { d as defineCustomElement$2 } from './p-C99Dr3jt.js';

const DgaFilterationUsage$1 = /*@__PURE__*/ proxyCustomElement(class DgaFilterationUsage extends H {
    constructor() {
        super();
        this.__registerHost();
        this.filterBlocks = [
            {
                id: 'status',
                label: 'Status',
                type: 'checkbox',
                options: [
                    { id: 'active', label: 'Active', value: 'active', count: 22 },
                    { id: 'inactive', label: 'Inactive', value: 'inactive', count: 22 },
                    { id: 'pending', label: 'Pending', value: 'pending', count: 22 },
                    { id: 'archived', label: 'Archived', value: 'archived', count: 22 },
                    { id: 'draft', label: 'Draft', value: 'draft', count: 22 },
                    { id: 'published', label: 'Published', value: 'published', count: 22 },
                    { id: 'unpublished', label: 'Unpublished', value: 'unpublished', count: 22 },
                    { id: 'deleted', label: 'Deleted', value: 'deleted', count: 22 },
                    { id: 'restored', label: 'Restored', value: 'restored', count: 22 },
                ],
            },
            {
                id: 'select',
                label: 'Variant',
                type: 'select',
                options: [
                    { id: 'option1', label: 'Option 1', value: 'option1' },
                    { id: 'option2', label: 'Option 2', value: 'option2' },
                    { id: 'option3', label: 'Option 3', value: 'option3' },
                    { id: 'option4', label: 'Option 4', value: 'option4' },
                ],
            },
            {
                id: 'userType',
                label: 'User Type',
                type: 'radio',
                options: [
                    { id: 'admin', label: 'Administrator', value: 'admin' },
                    { id: 'editor', label: 'Editor', value: 'editor' },
                    { id: 'viewer', label: 'Viewer', value: 'viewer' },
                    { id: 'contributor', label: 'Contributor', value: 'contributor' },
                ],
            },
            {
                id: 'categories',
                label: 'Categories',
                type: 'chips',
                options: [
                    { id: 'tech', label: 'Technology', value: 'tech' },
                    { id: 'design', label: 'Design', value: 'design' },
                    { id: 'marketing', label: 'Marketing', value: 'marketing' },
                    { id: 'sales', label: 'Sales', value: 'sales' },
                    { id: 'support', label: 'Support', value: 'support' },
                    { id: 'hr', label: 'HR', value: 'hr' },
                    { id: 'finance', label: 'Finance', value: 'finance' },
                ],
            },
            {
                id: 'rating',
                label: 'Rating',
                type: 'slider',
                min: 20,
                max: 70,
            },
            {
                id: 'price',
                label: 'Price Range',
                type: 'range',
            },
            {
                id: 'dateCreated',
                label: 'Created Date',
                type: 'date',
            },
        ];
        this.handleFilterChange = (event) => {
            const selectedFilters = event.detail;
            console.log('Applied Filters:', selectedFilters);
            selectedFilters.forEach(filter => {
                console.log('Filter:', filter);
            });
        };
        this.initialValue = {
            checkbox: {
                blockId: "status",
                options: [{
                        id: 'active', label: 'Active', value: 'active'
                    }]
            },
            chips: {
                blockId: "categories",
                options: [{ id: 'tech', label: 'Technology', value: 'tech' }, { id: 'design', label: 'Design', value: 'design' },]
            },
            // date:{
            //   blockId:"dateCreated",
            //   options:[{
            //     id: 'active', label: 'Active', value: 'active'
            //   }]
            // },
            radio: {
                blockId: "userType",
                options: [{ id: 'admin', label: 'Administrator', value: 'admin' }]
            },
            range: {
                blockId: "price",
                options: { min: 10, max: 20 }
            },
            select: {
                blockId: "select",
                options: [{ id: 'option1', label: 'Option 1', value: 'option1' }]
            },
            slider: {
                blockId: "rating",
                options: { min: 30, max: 50 }
            },
        };
    }
    render() {
        return (h("div", { key: '1be9e286df8f39bb67ccd56c69df6513ae612080', class: "filteration-usage" }, h("h2", { key: '207eb4860764573e884d6fde030ba3a5dd1f54e7' }, "Filteration Component Examples"), h("div", { key: 'aa1ddd3513f7dd50435de12b5d7d6a0f6b414189', class: "filteration-container", style: { margin: '20px 0' } }, h("h3", { key: '27149ea4108ec601711af43a5c5b573807e30f26' }, "Default Filteration"), h("dga-filteration", { key: 'abbe37f861ae3d44ccca01470d2d11980d444c6d', buttonLabel: "Filter Items", filterBlocks: this.filterBlocks, onOnFilterChange: this.handleFilterChange, initvalue: this.initialValue }))));
    }
}, [0, "dga-filteration-usage", {
        "filterBlocks": [32],
        "initialValue": [32]
    }]);
function defineCustomElement$1() {
    if (typeof customElements === "undefined") {
        return;
    }
    const components = ["dga-filteration-usage", "dga-button-v2", "dga-checkbox", "dga-chip", "dga-filteration", "dga-helper-text", "dga-icon", "dga-icon-v2", "dga-label", "dga-radio-button", "dga-slider", "dga-text-input"];
    components.forEach(tagName => { switch (tagName) {
        case "dga-filteration-usage":
            if (!customElements.get(tagName)) {
                customElements.define(tagName, DgaFilterationUsage$1);
            }
            break;
        case "dga-button-v2":
            if (!customElements.get(tagName)) {
                defineCustomElement$c();
            }
            break;
        case "dga-checkbox":
            if (!customElements.get(tagName)) {
                defineCustomElement$b();
            }
            break;
        case "dga-chip":
            if (!customElements.get(tagName)) {
                defineCustomElement$a();
            }
            break;
        case "dga-filteration":
            if (!customElements.get(tagName)) {
                defineCustomElement$9();
            }
            break;
        case "dga-helper-text":
            if (!customElements.get(tagName)) {
                defineCustomElement$8();
            }
            break;
        case "dga-icon":
            if (!customElements.get(tagName)) {
                defineCustomElement$7();
            }
            break;
        case "dga-icon-v2":
            if (!customElements.get(tagName)) {
                defineCustomElement$6();
            }
            break;
        case "dga-label":
            if (!customElements.get(tagName)) {
                defineCustomElement$5();
            }
            break;
        case "dga-radio-button":
            if (!customElements.get(tagName)) {
                defineCustomElement$4();
            }
            break;
        case "dga-slider":
            if (!customElements.get(tagName)) {
                defineCustomElement$3();
            }
            break;
        case "dga-text-input":
            if (!customElements.get(tagName)) {
                defineCustomElement$2();
            }
            break;
    } });
}

const DgaFilterationUsage = DgaFilterationUsage$1;
const defineCustomElement = defineCustomElement$1;

export { DgaFilterationUsage, defineCustomElement };
