// @platformscode/core/dist/components/dga-slider-usage.js
import { p as proxyCustomElement, H, h, F as Fragment } from './p-BhqVmlhr.js';
import { d as defineCustomElement$2 } from './p-BE95gJkd.js';

const DgaSliderUsage$1 = /*@__PURE__*/ proxyCustomElement(class DgaSliderUsage extends H {
    constructor() {
        super();
        this.__registerHost();
        this.__attachShadow();
        this.activeStep = 1;
        this.minValue = 10;
        this.maxValue = 100;
    }
    render() {
        return (h(Fragment, { key: '79432bef1a05778ebf774597630f1c632a96a905' }, h("br", { key: '31dcfd1804fa1a215fadb0c3f73cd02661a5df0f' }), h("dga-slider", { key: 'cc3df3d76a5f8390f9d0a52146cf04e9bfa9d3ef', min: this.minValue, max: this.maxValue, step: 1, range: true, size: "medium", helperText: "Helper Text Example", label: "Test", sx: {
                backgroundColor: "yellow",
                marginTop: "200px",
                ".range-slider-inner-rail": {
                    backgroundColor: "red"
                }
            } }), ";", h("br", { key: 'bb12eea2854eee3bc8d2522af5b379b11334bfbb' }), this.minValue, " - ", this.maxValue, h("dga-slider", { key: 'e6a2902e2f0ee0ec13073498389bc98c4d6da869' })));
    }
}, [1, "dga-slider-usage", {
        "activeStep": [32],
        "minValue": [32],
        "maxValue": [32]
    }]);
function defineCustomElement$1() {
    if (typeof customElements === "undefined") {
        return;
    }
    const components = ["dga-slider-usage", "dga-slider"];
    components.forEach(tagName => { switch (tagName) {
        case "dga-slider-usage":
            if (!customElements.get(tagName)) {
                customElements.define(tagName, DgaSliderUsage$1);
            }
            break;
        case "dga-slider":
            if (!customElements.get(tagName)) {
                defineCustomElement$2();
            }
            break;
    } });
}

const DgaSliderUsage = DgaSliderUsage$1;
const defineCustomElement = defineCustomElement$1;

export { DgaSliderUsage, defineCustomElement };
