// @platformscode/core/dist/components/dga-card-v2-usage.js
import { p as proxyCustomElement, H, h } from './p-BhqVmlhr.js';
import { d as defineCustomElement$c } from './p-B4U0lY4y.js';
import { d as defineCustomElement$b } from './p-CQbQdVrC.js';
import { d as defineCustomElement$a } from './p-y4YLJS_V.js';
import { d as defineCustomElement$9 } from './p-DBpsmom7.js';
import { d as defineCustomElement$8 } from './p-F86Y19t4.js';
import { d as defineCustomElement$7 } from './p-ChhNw26D.js';
import { d as defineCustomElement$6 } from './p-ByzFFJgL.js';
import { d as defineCustomElement$5 } from './p-CCimZGO2.js';
import { d as defineCustomElement$4 } from './p-DN3DJd5E.js';
import { d as defineCustomElement$3 } from './p-DBTOrdV2.js';
import { d as defineCustomElement$2 } from './p-CR6GP5Pr.js';

const DgaCardV2Usage$1 = /*@__PURE__*/ proxyCustomElement(class DgaCardV2Usage extends H {
    constructor() {
        super();
        this.__registerHost();
        this.variant = 'neutral';
        this.selected = false;
    }
    render() {
        return (h("div", { key: '9d09f81e8993d2c618bff18548e7665ef6adfbc2' }, h("h1", { key: '07e861f6bace8351ef3526d0edf0b76760e8341d' }, "Card Example"), h("div", { key: '47b5bd30e5c9640412aa980be6e6a1cc035feaf4', style: { padding: '50px', display: 'flex', gap: '20px' } }, h("div", { key: '17e83040b30370ea0ec928742d3df207270bda5c', style: { display: 'flex', flexDirection: 'column', gap: '20px' } }, h("dga-card-v2", { key: 'b24eae2b52b5dd17a6cbf7efe72cbb67fbfe658d', type: "default", sx: {
                "& > .card": {
                    backgroundColor: 'red',
                    height: '400px',
                }
            } }, h("dga-card-content", { key: 'ec0cf34d45a63283a9909c6be97ddae58ae34932' }, h("dga-featured-icon", { key: 'b7523c04bc1083ffe11ec3524a5604c6c2d7b0e7', color: 'brand', variant: 'light', size: 'lg', icon: {
                name: 'checkmark-circle-02',
                variant: 'stroke',
                type: 'sharp',
                size: 24
            } }), h("dga-flex", { key: '63b901eb2e014bbfc7859f5d9d00330e6c55f92b', direction: 'column', spacing: 8 }, h("h4", { key: '25b5608b06ffda21a96baed68243878825c3bca2', class: "text-lg-bold" }, "Card Title test"), h("p", { key: 'ec5e72e6d8b3267ed73bc074c4f2d6ef5b14fc08', class: "text-md-regular" }, "Card content placeholder text goes here")), h("dga-flex", { key: 'c3748bc0048fae6457f03ff6901e96f812676814', direction: 'column', spacing: 4 }, h("dga-rating", { key: '3466fe198f330160f4bb2da98721efad2832ae8f', value: 4, size: 'sm' }), h("span", { key: '520052c4de814de8a2e97aa56c67b63b328a1b92', class: "text-xs-regular" }, "12 reviews"))), h("dga-card-actions", { key: '2e83da345a9e77678e8b61d919ba2ec0402fa8d2' }, h("dga-button-v2", { key: 'a67d2424ef17f32cebecb2b8051495cee0dddc69', label: "Primary", size: "lg", variant: "primary", onClick: () => console.log('Primary Action') }), h("dga-button-v2", { key: '87b273074ee779041e2ee464d6ac191c22a4da02', label: "Secondary", size: "lg", variant: "secondary-outline", onClick: () => console.log('Secondary Action') })), h("dga-card-expandable-content", { key: '00d02ddbaac56ca49486f08a331c8e5b14c3b2af' }, h("div", { key: 'c1f4d61693b9de39e710314baf560b57799d4344' }, "Expanded Content"), h("p", { key: '49b51037005f1f881e33c52bf1a9841287ecb41b' }, "Lorem ipsum dolor sit amet consectetur, adipisicing elit. Earum voluptatum, facere mollitia quas natus nihil, doloribus repudiandae, quasi vitae nobis deserunt similique magni explicabo. Molestiae unde dolorem expedita aut recusandae.")))), h("div", { key: 'debe7e6bbe87b8b8232265ba4131537f86ac90c7', style: { display: 'flex', flexDirection: 'column', gap: '20px' } }))));
    }
}, [0, "dga-card-v2-usage", {
        "variant": [32],
        "selected": [32]
    }]);
function defineCustomElement$1() {
    if (typeof customElements === "undefined") {
        return;
    }
    const components = ["dga-card-v2-usage", "dga-button-v2", "dga-card-actions", "dga-card-content", "dga-card-expandable-content", "dga-card-v2", "dga-checkbox", "dga-featured-icon", "dga-flex", "dga-icon", "dga-icon-v2", "dga-rating"];
    components.forEach(tagName => { switch (tagName) {
        case "dga-card-v2-usage":
            if (!customElements.get(tagName)) {
                customElements.define(tagName, DgaCardV2Usage$1);
            }
            break;
        case "dga-button-v2":
            if (!customElements.get(tagName)) {
                defineCustomElement$c();
            }
            break;
        case "dga-card-actions":
            if (!customElements.get(tagName)) {
                defineCustomElement$b();
            }
            break;
        case "dga-card-content":
            if (!customElements.get(tagName)) {
                defineCustomElement$a();
            }
            break;
        case "dga-card-expandable-content":
            if (!customElements.get(tagName)) {
                defineCustomElement$9();
            }
            break;
        case "dga-card-v2":
            if (!customElements.get(tagName)) {
                defineCustomElement$8();
            }
            break;
        case "dga-checkbox":
            if (!customElements.get(tagName)) {
                defineCustomElement$7();
            }
            break;
        case "dga-featured-icon":
            if (!customElements.get(tagName)) {
                defineCustomElement$6();
            }
            break;
        case "dga-flex":
            if (!customElements.get(tagName)) {
                defineCustomElement$5();
            }
            break;
        case "dga-icon":
            if (!customElements.get(tagName)) {
                defineCustomElement$4();
            }
            break;
        case "dga-icon-v2":
            if (!customElements.get(tagName)) {
                defineCustomElement$3();
            }
            break;
        case "dga-rating":
            if (!customElements.get(tagName)) {
                defineCustomElement$2();
            }
            break;
    } });
}

const DgaCardV2Usage = DgaCardV2Usage$1;
const defineCustomElement = defineCustomElement$1;

export { DgaCardV2Usage, defineCustomElement };
