// @platformscode/core/dist/components/dga-datepicker-usage.js
import { p as proxyCustomElement, H, h } from './p-BhqVmlhr.js';
import { d as defineCustomElement$a } from './p-BUKkxyIA.js';
import { d as defineCustomElement$9 } from './p-B4U0lY4y.js';
import { d as defineCustomElement$8 } from './p--saaGsBa.js';
import { d as defineCustomElement$7 } from './p-BDegpq_a.js';
import { d as defineCustomElement$6 } from './p-DN3DJd5E.js';
import { d as defineCustomElement$5 } from './p-DBTOrdV2.js';
import { d as defineCustomElement$4 } from './p-Cxhq_uxn.js';
import { d as defineCustomElement$3 } from './p-C99Dr3jt.js';
import { d as defineCustomElement$2 } from './p-C8cyYIhA.js';

const DgaPickerUsage = /*@__PURE__*/ proxyCustomElement(class DgaPickerUsage extends H {
    constructor() {
        super();
        this.__registerHost();
    }
    render() {
        return (h("div", { key: 'b52ca0efce71204aed2adaf89202318f0edcd628' }, h("h1", { key: 'd3bb55b0ed385369e9e4b8c84b5ac32d228274a3' }, "Datepicker Example"), h("div", { key: '9bb4e267a566d78e1233515beae77a9de35aedc6', style: { padding: '50px', display: 'flex', gap: '20px' } }, h("div", { key: '8b01d9914510e772ddea427795d9cb633fd85a5f', style: { display: 'flex', flexDirection: 'column', gap: '20px' } }, h("dga-datepicker", { key: 'ed93d5b6dd71651203cf68015dc6dc6daa5d5bcb', inputField: false, qickOptions: true }), h("dga-datepicker", { key: 'a14cd34017dca39a1ad597f7568df29fe20b08f2', RTL: true, inputField: true, range: true, qickOptions: true }), h("dga-datepicker", { key: '65360941d011ac7b4969a3346bcf3ec3e42f3393', range: true })), h("div", { key: 'b86ca5a1008bbce275ffbb1e2e1751e8c0f75bac', style: { display: 'flex', flexDirection: 'column', gap: '20px' } }))));
    }
}, [0, "dga-datepicker-usage"]);
function defineCustomElement$1() {
    if (typeof customElements === "undefined") {
        return;
    }
    const components = ["dga-datepicker-usage", "date-cell", "dga-button-v2", "dga-datepicker", "dga-helper-text", "dga-icon", "dga-icon-v2", "dga-label", "dga-text-input", "month-navigator"];
    components.forEach(tagName => { switch (tagName) {
        case "dga-datepicker-usage":
            if (!customElements.get(tagName)) {
                customElements.define(tagName, DgaPickerUsage);
            }
            break;
        case "date-cell":
            if (!customElements.get(tagName)) {
                defineCustomElement$a();
            }
            break;
        case "dga-button-v2":
            if (!customElements.get(tagName)) {
                defineCustomElement$9();
            }
            break;
        case "dga-datepicker":
            if (!customElements.get(tagName)) {
                defineCustomElement$8();
            }
            break;
        case "dga-helper-text":
            if (!customElements.get(tagName)) {
                defineCustomElement$7();
            }
            break;
        case "dga-icon":
            if (!customElements.get(tagName)) {
                defineCustomElement$6();
            }
            break;
        case "dga-icon-v2":
            if (!customElements.get(tagName)) {
                defineCustomElement$5();
            }
            break;
        case "dga-label":
            if (!customElements.get(tagName)) {
                defineCustomElement$4();
            }
            break;
        case "dga-text-input":
            if (!customElements.get(tagName)) {
                defineCustomElement$3();
            }
            break;
        case "month-navigator":
            if (!customElements.get(tagName)) {
                defineCustomElement$2();
            }
            break;
    } });
}

const DgaDatepickerUsage = DgaPickerUsage;
const defineCustomElement = defineCustomElement$1;

export { DgaDatepickerUsage, defineCustomElement };
