// @platformscode/core/dist/components/dga-text-input-usage.js
import { p as proxyCustomElement, H, h } from './p-BhqVmlhr.js';
import { d as defineCustomElement$7 } from './p-BDegpq_a.js';
import { d as defineCustomElement$6 } from './p-DN3DJd5E.js';
import { d as defineCustomElement$5 } from './p-DBTOrdV2.js';
import { d as defineCustomElement$4 } from './p-Cxhq_uxn.js';
import { d as defineCustomElement$3 } from './p-C7b9Kw1u.js';
import { d as defineCustomElement$2 } from './p-C99Dr3jt.js';

const DgaTextInputUsage$1 = /*@__PURE__*/ proxyCustomElement(class DgaTextInputUsage extends H {
    constructor() {
        super();
        this.__registerHost();
        this.value = 'Text Input Example';
    }
    render() {
        return (h("div", { key: '9452e3df29178ef2d51b08138e730dca152b6840' }, h("h1", { key: '6ef0a9a5fa65153bba5ee70fe9f613daa106dbb2' }, "Text Input Example"), h("div", { key: '47c3a94a22f059292ea8cb008a9c6e171dffff4e', style: { padding: '50px' } }, h("div", { key: 'a9154d1b74b7823b512a8f5bbe595a9872437e84', style: { display: 'flex', flexDirection: 'column', gap: '16px' } }, h("input", { key: 'd53ca1134429e6ce27c78762acc344b822717d2f', type: "text", value: this.value, onChange: (e) => (this.value = e.target.value) }), h("p", { key: 'ca4ee73291da6dfe391d9b8e43587210c90af583' }, this.value), h("dga-text-input", { key: '27c4d29f81df1d6d7391e65c65f96632221c99a3', readonly: false, variant: "darker", label: "Label", value: this.value, required: true, feedbackIcon: true, icon: { name: 'search-01', variant: 'stroke', type: 'standard' }, helperTextIcon: { name: 'alert-circle', variant: 'stroke', type: 'standard' }, helperText: "test", error: true, onInput: (e) => (this.value = e.target.value) }), h("dga-text-input", { key: 'd14f329253581f1e1ccbd70f30ec3afbdf5dde4b', readonly: false, variant: "darker", label: "Label", value: "Entered text", required: true, feedbackIcon: true, icon: { name: 'search-01', variant: 'stroke', type: 'standard' }, helperText: "test", error: true }), h("dga-text-input", { key: '4092fb8baa54e0ff72cf1821ebe644c7106bffc7', readonly: false, variant: "darker", label: "Label", value: "Entered text", required: true, feedbackIcon: true, icon: { name: 'search-01', variant: 'stroke', type: 'standard' }, helperTextIcon: { name: 'alert-circle', variant: 'stroke', type: 'standard' }, helperText: "test", error: true }, h("dga-prefix", { key: '8fb4cd53adb593c9814c853e047f9094afaf1de0', type: "button", variant: "solid",
            // label="Button Label"
            icon: { name: 'search-01' } })), h("dga-label", { key: '914ddc97e223acf89f27f8f149e45b4d995f0042', label: "Mandatory Text Input" }), h("dga-label", { key: 'cb911b5e5bd049565544b34afbad5154e1f55873', label: "Mandatory Text Input", required: true })))));
    }
}, [0, "dga-text-input-usage", {
        "value": [32]
    }]);
function defineCustomElement$1() {
    if (typeof customElements === "undefined") {
        return;
    }
    const components = ["dga-text-input-usage", "dga-helper-text", "dga-icon", "dga-icon-v2", "dga-label", "dga-prefix", "dga-text-input"];
    components.forEach(tagName => { switch (tagName) {
        case "dga-text-input-usage":
            if (!customElements.get(tagName)) {
                customElements.define(tagName, DgaTextInputUsage$1);
            }
            break;
        case "dga-helper-text":
            if (!customElements.get(tagName)) {
                defineCustomElement$7();
            }
            break;
        case "dga-icon":
            if (!customElements.get(tagName)) {
                defineCustomElement$6();
            }
            break;
        case "dga-icon-v2":
            if (!customElements.get(tagName)) {
                defineCustomElement$5();
            }
            break;
        case "dga-label":
            if (!customElements.get(tagName)) {
                defineCustomElement$4();
            }
            break;
        case "dga-prefix":
            if (!customElements.get(tagName)) {
                defineCustomElement$3();
            }
            break;
        case "dga-text-input":
            if (!customElements.get(tagName)) {
                defineCustomElement$2();
            }
            break;
    } });
}

const DgaTextInputUsage = DgaTextInputUsage$1;
const defineCustomElement = defineCustomElement$1;

export { DgaTextInputUsage, defineCustomElement };
