// @platformscode/core/dist/components/dga-tooltip-usage.js
import { p as proxyCustomElement, H, h } from './p-BhqVmlhr.js';
import { d as defineCustomElement$2 } from './p-HtJMEkmy.js';

const DgaTooltipUsage$1 = /*@__PURE__*/ proxyCustomElement(class DgaTooltipUsage extends H {
    constructor() {
        super();
        this.__registerHost();
        this.activeStep = 1;
    }
    render() {
        return (h("div", { key: '5fa97a5a1ec0ca3445d91d32eff49ba7c559f9cc' }, h("h1", { key: '50a4a4e2090f531eb677944a6fbf0e6cc82f0059' }, "Tooltip Example"), h("div", { key: 'a3dcb05555faf75a32173849cdd2139f318e7559', style: { padding: "50px", display: "flex", gap: "20px" } }, h("div", { key: '7359e9b5472e95bae89508060a00019d8d9457ab', style: { display: "flex", flexDirection: "column", gap: "20px" } }, h("dga-tooltip", { key: '2f45d12a246c9afcb666c2a1e96a0d872a2cba73', title: "Tooltip title", helperText: "Max width of tooltips is 240px - text will wrap automatically", direction: 'right', theme: 'dark', icon: true, beakAlignment: 'end', beakPlacement: 'left', feedbackIcon: {
                type: "error-!",
                style: "container",
                size: 24
            } }, "Hello")), h("div", { key: '73e0f4506321c5fce60f0c9c46cc02057b2de39b', style: { display: "flex", flexDirection: "column", gap: "20px" } }))));
    }
}, [0, "dga-tooltip-usage", {
        "activeStep": [32]
    }]);
function defineCustomElement$1() {
    if (typeof customElements === "undefined") {
        return;
    }
    const components = ["dga-tooltip-usage", "dga-tooltip"];
    components.forEach(tagName => { switch (tagName) {
        case "dga-tooltip-usage":
            if (!customElements.get(tagName)) {
                customElements.define(tagName, DgaTooltipUsage$1);
            }
            break;
        case "dga-tooltip":
            if (!customElements.get(tagName)) {
                defineCustomElement$2();
            }
            break;
    } });
}

const DgaTooltipUsage = DgaTooltipUsage$1;
const defineCustomElement = defineCustomElement$1;

export { DgaTooltipUsage, defineCustomElement };
