// @platformscode/core/dist/components/dga-quote-usage.js
import { p as proxyCustomElement, H, h } from './p-BhqVmlhr.js';
import { d as defineCustomElement$5 } from './p-DVUb8Tt4.js';
import { d as defineCustomElement$4 } from './p-CCimZGO2.js';
import { d as defineCustomElement$3 } from './p-DN3DJd5E.js';
import { d as defineCustomElement$2 } from './p-DCwT1t7T.js';

const DgaQuoteUsage$1 = /*@__PURE__*/ proxyCustomElement(class DgaQuoteUsage extends H {
    constructor() {
        super();
        this.__registerHost();
        this.activeStep = 1;
    }
    render() {
        return (h("div", { key: 'cb447e17357b90a14014fe55a845807dfe95f9c8' }, h("h1", { key: 'a2287112a494870bc1b70d1fe106bd6da74f5086' }, "Quote Example"), h("div", { key: '86982734cee28192a6a898e8ba5de1f4646d9c1b', style: { padding: "50px", display: "flex", gap: "20px" } }, h("div", { key: 'cbb219394b34a855a2f40c3ea523c5151e40f7db', style: { display: "flex", flexDirection: "column", gap: "20px" } }, h("dga-quote", { key: '3614c152ed03747f8858dc0307dd2b72e3dc0d55', quoteTitle: 'Title of quote', quoteDescription: 'The quote is placed here to highlight a specific saying or to present a brief quote that expresses an important idea or concept.', authorName: "Author's name", authorDescription: 'brief or description.' }), h("dga-quote", { key: '1a78531153367a71175e855428ce645b20c7ce0a', quoteTitle: 'Title of quote', quoteDescription: 'The quote is placed here to highlight a specific saying or to present a brief quote that expresses an important idea or concept.', authorName: "Author's name", authorDescription: 'brief or description.', whiteBackground: true })), h("div", { key: '6fdf557138daa3ac1dbc46265776b3e71982ffae', style: { display: "flex", flexDirection: "column", gap: "20px" } }))));
    }
}, [0, "dga-quote-usage", {
        "activeStep": [32]
    }]);
function defineCustomElement$1() {
    if (typeof customElements === "undefined") {
        return;
    }
    const components = ["dga-quote-usage", "dga-avatar", "dga-flex", "dga-icon", "dga-quote"];
    components.forEach(tagName => { switch (tagName) {
        case "dga-quote-usage":
            if (!customElements.get(tagName)) {
                customElements.define(tagName, DgaQuoteUsage$1);
            }
            break;
        case "dga-avatar":
            if (!customElements.get(tagName)) {
                defineCustomElement$5();
            }
            break;
        case "dga-flex":
            if (!customElements.get(tagName)) {
                defineCustomElement$4();
            }
            break;
        case "dga-icon":
            if (!customElements.get(tagName)) {
                defineCustomElement$3();
            }
            break;
        case "dga-quote":
            if (!customElements.get(tagName)) {
                defineCustomElement$2();
            }
            break;
    } });
}

const DgaQuoteUsage = DgaQuoteUsage$1;
const defineCustomElement = defineCustomElement$1;

export { DgaQuoteUsage, defineCustomElement };
