// @platformscode/core/dist/components/dga-second-nav-header-usage.js
import { p as proxyCustomElement, H, h } from './p-BhqVmlhr.js';
import { d as defineCustomElement$f } from './p-B4U0lY4y.js';
import { d as defineCustomElement$e } from './p-cDUWOFIj.js';
import { d as defineCustomElement$d } from './p-DN3DJd5E.js';
import { d as defineCustomElement$c } from './p-DBTOrdV2.js';
import { d as defineCustomElement$b } from './p-DXKT25WK.js';
import { d as defineCustomElement$a } from './p-Bt-84KPV.js';
import { d as defineCustomElement$9 } from './p-C7z9rLjZ.js';
import { d as defineCustomElement$8 } from './p-hAJIuLq_.js';
import { d as defineCustomElement$7 } from './p-D3dLYGkN.js';
import { d as defineCustomElement$6 } from './p-rgcxXsU5.js';
import { d as defineCustomElement$5 } from './p-BRtz_FTw.js';
import { d as defineCustomElement$4 } from './p-OGfNpKA1.js';
import { d as defineCustomElement$3 } from './p-BRUw79oz.js';
import { d as defineCustomElement$2 } from './p-DX_C6M6_.js';

const DgaSecondNavHeaderUsage$1 = /*@__PURE__*/ proxyCustomElement(class DgaSecondNavHeaderUsage extends H {
    constructor() {
        super();
        this.__registerHost();
        this.activeStep = 1;
    }
    render() {
        return (h("div", { key: '9dca7605f8a1cec6ca5bd8ab960340eae19847a7', style: {
                display: "flex",
                flexDirection: "column",
                gap: "20px"
            } }, h("dga-second-nav-header", { key: '92cd04e45b74177514a159d5fa51ddd02427fefb' }, h("dga-second-nav-header-actions", { key: '86f7e2cfd04c22c5ce44c56fb6edbdef779669f3' }, h("dga-button-v2", { key: '9b9ba4b3ab1786df76b151311d6ec0cf6e8c7ce5', label: 'Button', variant: "transparent", size: 'sm', iconOnly: true, leadIcon: true, leadIconProps: {
                name: 'mic-01',
                type: 'rounded',
                variant: 'stroke',
                color: '#FFFFFF'
            } }), h("dga-button-v2", { key: 'b17e8b86d7dea23fa78db772e3d29bed1da0d6e2', label: 'Button', variant: "transparent", size: 'sm', iconOnly: true, leadIcon: true, leadIconProps: {
                name: 'zoom-in-area',
                type: 'rounded',
                variant: 'stroke',
                color: '#FFFFFF'
            } }), h("dga-button-v2", { key: '09cfcedad0e89561722d279834a2679f57224977', label: 'Button', variant: "transparent", size: 'sm', iconOnly: true, leadIcon: true, leadIconProps: {
                name: 'zoom-out-area',
                type: 'rounded',
                variant: 'stroke',
                color: '#FFFFFF'
            } }), h("dga-button-v2", { key: 'a267cfb6cd38a27bc555ef38e34618c60ad805ae', label: 'Button', variant: "transparent", size: 'sm', iconOnly: true, leadIcon: true, leadIconProps: {
                name: 'view',
                type: 'rounded',
                variant: 'stroke',
                color: '#FFFFFF'
            } })), h("dga-second-nav-header-content", { key: 'bfc3f89678cd2c7396d1910a710bfb2e964fb643' }, h("dga-second-nav-header-item", { key: '7a20155a19f87fb5bcef1ea339f15857950d6352', label: 'Cloudy' }, h("dga-icon", { key: '92bbcadaf79e4046a983fb743b4248cd6ae82735', icon: "cloud", variant: "stroke", type: "rounded" })), h("dga-second-nav-header-item", { key: '10400f15c620a9f271131d6c971cb6c85bee74cf', label: '21 Jan 2024' }, h("dga-icon", { key: '1ef502cbddf1f664e74fc4b731306c30555edc21', icon: "calendar-04", variant: "stroke", type: "rounded" })))), h("dga-second-nav-header", { key: '2ba1abeeccc9b7327ee71131ec9887e837c6599c', variant: "gray" }, h("dga-second-nav-header-actions", { key: '59d8997550798aed1a7117c0b6292bb12512727b' }, h("dga-button-v2", { key: '02f32493b1d3cca814aceb2bd6b5dfe5fcc3f15f', label: 'Button', variant: "transparent", leadIcon: true, leadIconProps: {
                name: 'mic-01',
                type: 'rounded',
                variant: 'stroke',
                color: '#FFFFFF'
            }, size: 'sm', iconOnly: true })), h("dga-second-nav-header-content", { key: 'a444e7c6eaf1260ad7697d8b373a1e3ac284e524' }, h("dga-second-nav-header-item", { key: 'b32b46deb055cb8610438d9e435e85e42e3fa596', label: 'Cloudy' }, h("dga-icon", { key: '590289a061098b548fda642da69df845813f5a65', icon: "cloud", variant: "stroke", type: "rounded" })), h("dga-second-nav-header-item", { key: 'a82b63d01362c1a6244918e42780ca273fad336c', label: '21 Jan 2024' }, h("dga-icon", { key: 'e8982a232198d6dd40200bb07ffb96bd6fac41a6', icon: "calendar-04", variant: "stroke", type: "rounded" })))), h("dga-nav-header", { key: '3f26be32247ed10273de3de853d023c590546623', fullWidth: true, divider: true }, h("dga-nav-header-main", { key: 'd2db4d3a79530818885dd82b439bdc17ebfebd15', collapsed: true }, h("dga-nav-header-logos", { key: 'bc1a620385d629610d70e17fe153bec003a54e1a', logoSrc: "https://dga-nds-fbhtx.ondigitalocean.app/mobile-logo.svg", govSrc: "https://dga-nds-fbhtx.ondigitalocean.app/mobile-logo.svg", logoLink: "#", govLink: "#" }), h("dga-nav-header-menu", { key: '5193637ce0993d8e31b03d1d8975cb32fcb3d676' }, h("dga-nav-header-link", { key: '1a20f75c24a496a155c845bbb90080a64cd89706', label: "Item  1", icon: "arrow-down-01", subMenuBackground: "brand", subMenuFullWidth: true }, " "), h("dga-nav-header-link", { key: '770f539617aa164b3bc6600bb6face29eb36b11c', label: "Item  2", icon: "arrow-down-01", subMenuBackground: "brand", subMenuFullWidth: true }, " "), h("dga-nav-header-link", { key: '980457f30b9a1dfe08718b0c521ec55bcabab8ef', label: "Item  3", icon: "arrow-down-01", subMenuBackground: "brand", subMenuFullWidth: true }, " "), h("dga-nav-header-link", { key: '17ce7bd9840fcf5e84884581353cb494d4e865ea', label: "Item  4", icon: "arrow-down-01", subMenuBackground: "brand", subMenuFullWidth: true }, " "))), h("dga-nav-header-actions", { key: 'a21ef6f93b8532fd9cd52e451f04fdeadc346e6a' }, h("dga-header-action-btn", { key: 'c4d97f8b404361d392edf5b4f9b1143f7c959fb3', label: "Search", icon: "search-01" }), h("dga-header-action-btn", { key: '1d05dd3dd49ae3ddab4aa6892be3a832f0877625', label: "\u0639\u0631\u0628\u064A", icon: "translation" }), h("dga-header-action-btn", { key: '1a36bb0abcd04b085f0872ecda555b0edf8a49e3', label: "Login", icon: "user" })))));
    }
}, [0, "dga-second-nav-header-usage", {
        "activeStep": [32]
    }]);
function defineCustomElement$1() {
    if (typeof customElements === "undefined") {
        return;
    }
    const components = ["dga-second-nav-header-usage", "dga-button-v2", "dga-header-action-btn", "dga-icon", "dga-icon-v2", "dga-nav-header", "dga-nav-header-actions", "dga-nav-header-link", "dga-nav-header-logos", "dga-nav-header-main", "dga-nav-header-menu", "dga-second-nav-header", "dga-second-nav-header-actions", "dga-second-nav-header-content", "dga-second-nav-header-item"];
    components.forEach(tagName => { switch (tagName) {
        case "dga-second-nav-header-usage":
            if (!customElements.get(tagName)) {
                customElements.define(tagName, DgaSecondNavHeaderUsage$1);
            }
            break;
        case "dga-button-v2":
            if (!customElements.get(tagName)) {
                defineCustomElement$f();
            }
            break;
        case "dga-header-action-btn":
            if (!customElements.get(tagName)) {
                defineCustomElement$e();
            }
            break;
        case "dga-icon":
            if (!customElements.get(tagName)) {
                defineCustomElement$d();
            }
            break;
        case "dga-icon-v2":
            if (!customElements.get(tagName)) {
                defineCustomElement$c();
            }
            break;
        case "dga-nav-header":
            if (!customElements.get(tagName)) {
                defineCustomElement$b();
            }
            break;
        case "dga-nav-header-actions":
            if (!customElements.get(tagName)) {
                defineCustomElement$a();
            }
            break;
        case "dga-nav-header-link":
            if (!customElements.get(tagName)) {
                defineCustomElement$9();
            }
            break;
        case "dga-nav-header-logos":
            if (!customElements.get(tagName)) {
                defineCustomElement$8();
            }
            break;
        case "dga-nav-header-main":
            if (!customElements.get(tagName)) {
                defineCustomElement$7();
            }
            break;
        case "dga-nav-header-menu":
            if (!customElements.get(tagName)) {
                defineCustomElement$6();
            }
            break;
        case "dga-second-nav-header":
            if (!customElements.get(tagName)) {
                defineCustomElement$5();
            }
            break;
        case "dga-second-nav-header-actions":
            if (!customElements.get(tagName)) {
                defineCustomElement$4();
            }
            break;
        case "dga-second-nav-header-content":
            if (!customElements.get(tagName)) {
                defineCustomElement$3();
            }
            break;
        case "dga-second-nav-header-item":
            if (!customElements.get(tagName)) {
                defineCustomElement$2();
            }
            break;
    } });
}

const DgaSecondNavHeaderUsage = DgaSecondNavHeaderUsage$1;
const defineCustomElement = defineCustomElement$1;

export { DgaSecondNavHeaderUsage, defineCustomElement };
