// @platformscode/core/dist/components/dga-accordion-usage.js
import { p as proxyCustomElement, H, h } from './p-BhqVmlhr.js';
import { d as defineCustomElement$3 } from './p-Dx89BkDF.js';
import { d as defineCustomElement$2 } from './p-DN3DJd5E.js';

const DgaAccordionUsage$1 = /*@__PURE__*/ proxyCustomElement(class DgaAccordionUsage extends H {
    constructor() {
        super();
        this.__registerHost();
    }
    render() {
        return (h("div", { key: 'deb171b69692cb63f0c3d4b134799d654076d050' }, h("h1", { key: '6c7bcc0a7205367a4995d8d66ccf057f8f735d39' }, "Accordion Example"), h("div", { key: '695d6710d2146a4d276567c5a30846209b64a3e0', style: { padding: '50px', display: 'flex', gap: '20px' } }, h("div", { key: '52fdc14e2fd9825785a55cfe1e715e825d2d0e02', style: { display: 'flex', flexDirection: 'column', gap: '20px' } }, h("dga-accordion", { key: 'ec562f14fd77382fd02e662a6fbbee2df0052899', title: "Accordion Title", size: "lg", content: "The accordion component delivers large amounts of content in a small space through progressive disclosure. The user gets key details about the underlying content and can choose to expand that content within the constraints of the accordion." }), h("dga-accordion", { key: '17c72b6fc9e9babbbfa5b553a5b55c1e5c1e2bfc', title: "Accordion Title", size: "md" }), h("dga-accordion", { key: '5c2b25315b4acb9fe7d65345e7136b0c5cc68366', title: "Accordion Title", size: "sm" }), h("dga-accordion", { key: '462694366330e684801e5270227c4c6e64b2527b', defaultExpanded: true, title: "Accordion Title", size: "lg", content: "The accordion component delivers large amounts of content in a small space through progressive disclosure. The user gets key details about the underlying content and can choose to expand that content within the constraints of the accordion.", disabled: true })), h("div", { key: 'd35b5b87d06feeb98bd806c33b615bc291f54092', style: { display: 'flex', flexDirection: 'column', gap: '20px' } }))));
    }
}, [0, "dga-accordion-usage"]);
function defineCustomElement$1() {
    if (typeof customElements === "undefined") {
        return;
    }
    const components = ["dga-accordion-usage", "dga-accordion", "dga-icon"];
    components.forEach(tagName => { switch (tagName) {
        case "dga-accordion-usage":
            if (!customElements.get(tagName)) {
                customElements.define(tagName, DgaAccordionUsage$1);
            }
            break;
        case "dga-accordion":
            if (!customElements.get(tagName)) {
                defineCustomElement$3();
            }
            break;
        case "dga-icon":
            if (!customElements.get(tagName)) {
                defineCustomElement$2();
            }
            break;
    } });
}

const DgaAccordionUsage = DgaAccordionUsage$1;
const defineCustomElement = defineCustomElement$1;

export { DgaAccordionUsage, defineCustomElement };
