// @platformscode/core/dist/components/dga-header-usage.js
import { p as proxyCustomElement, H, h, F as Fragment } from './p-BhqVmlhr.js';
import { d as defineCustomElement$u } from './p-CClx_RDs.js';
import { d as defineCustomElement$t } from './p-B4U0lY4y.js';
import { d as defineCustomElement$s } from './p-CJjTzkLW.js';
import { d as defineCustomElement$r } from './p-CCimZGO2.js';
import { d as defineCustomElement$q } from './p-cDUWOFIj.js';
import { d as defineCustomElement$p } from './p-C-NyQ87i.js';
import { d as defineCustomElement$o } from './p-CtYxM-JA.js';
import { d as defineCustomElement$n } from './p-DN3DJd5E.js';
import { d as defineCustomElement$m } from './p-DBTOrdV2.js';
import { d as defineCustomElement$l } from './p-DXKT25WK.js';
import { d as defineCustomElement$k } from './p-Bt-84KPV.js';
import { d as defineCustomElement$j } from './p-C7z9rLjZ.js';
import { d as defineCustomElement$i } from './p-hAJIuLq_.js';
import { d as defineCustomElement$h } from './p-D3dLYGkN.js';
import { d as defineCustomElement$g } from './p-rgcxXsU5.js';
import { d as defineCustomElement$f } from './p-DgY1cArs.js';
import { d as defineCustomElement$e } from './p-By3ydSQh.js';
import { d as defineCustomElement$d } from './p-B2OxYpgO.js';
import { d as defineCustomElement$c } from './p-oE4xypdy.js';
import { d as defineCustomElement$b } from './p-BRtz_FTw.js';
import { d as defineCustomElement$a } from './p-OGfNpKA1.js';
import { d as defineCustomElement$9 } from './p-BRUw79oz.js';
import { d as defineCustomElement$8 } from './p-DX_C6M6_.js';
import { d as defineCustomElement$7 } from './p-XcGOP6S8.js';
import { d as defineCustomElement$6 } from './p-HtJMEkmy.js';
import { d as defineCustomElement$5 } from './p-CD9sOu_w.js';
import { d as defineCustomElement$4 } from './p-mAJwGPJG.js';
import { d as defineCustomElement$3 } from './p-BVvoCy50.js';
import { d as defineCustomElement$2 } from './p-Dua-gbUp.js';

// import { IRoute } from '../dga-drawer/dga-drawer';
const SidePanelRoutes = [
    {
        name: 'Link',
        path: '/',
        icon: {
            name: 'home-01',
            variant: 'stroke',
            size: 16,
        },
        disabled: false,
        children: [
            {
                name: 'Link',
                path: '1',
                // icon: <i class='hgi-stroke hgi-arrow-down-01' style={{fontSize:"16px"}} />,
                // badge: 39,
            },
            {
                name: 'Link',
                path: '2',
                // icon: <i class='hgi-stroke hgi-arrow-down-01' style={{fontSize:"16px"}} />,
                // badge: 21,
            },
            {
                name: 'Link',
                path: '3',
                // icon: <i class='hgi-stroke hgi-arrow-down-01' style={{fontSize:"16px"}} />,
                // badge: 2,
            },
            // {
            //   name: "Home 4",
            //   path: "4",
            //   icon: <i class='hgi-stroke hgi-arrow-down-01' style={{fontSize:"16px"}} />,
            //   badge: 20,
            // },
            // {
            //   name: "Home 5",
            //   path: "5",
            //   icon: <i class='hgi-stroke hgi-arrow-down-01' style={{fontSize:"16px"}} />,
            //   badge: 150,
            //   children: [
            //     {
            //       name: "Home 1",
            //       path: "1",
            //       icon: <i class='hgi-stroke hgi-arrow-down-01' style={{fontSize:"16px"}} />,
            //       badge: 39,
            //       level: 2,
            //     },
            //     {
            //       name: "Home 4",
            //       path: "4",
            //       icon: <i class='hgi-stroke hgi-arrow-down-01' style={{fontSize:"16px"}} />,
            //       badge: 20,
            //       level: 2,
            //     },
            //     {
            //       name: "Home 5",
            //       path: "5",
            //       icon: <i class='hgi-stroke hgi-arrow-down-01' style={{fontSize:"16px"}} />,
            //       badge: 150,
            //       level: 2,
            //     },
            //     {
            //       name: "Home 6",
            //       path: "6",
            //       icon: <i class='hgi-stroke hgi-arrow-down-01' style={{fontSize:"16px"}} />,
            //       badge: 200,
            //       level: 2,
            //     },
            //     {
            //       name: "Home 7",
            //       path: "7",
            //       icon: <i class='hgi-stroke hgi-arrow-down-01' style={{fontSize:"16px"}} />,
            //       badge: 24,
            //       level: 2,
            //     },
            //   ],
            // },
            // {
            //   name: "Home 6",
            //   path: "6",
            //   icon: <i class='hgi-stroke hgi-arrow-down-01' style={{fontSize:"16px"}} />,
            //   badge: 200,
            // },
            // {
            //   name: "Home 7",
            //   path: "7",
            //   icon: <i class='hgi-stroke hgi-arrow-down-01' style={{fontSize:"16px"}} />,
            //   badge: 24,
            // },
        ],
    },
    {
        name: 'Components',
        path: '/components',
        icon: {
            name: 'home-01',
            variant: 'stroke',
            size: 16,
        },
        disabled: false,
        children: [
            {
                name: 'Buttons',
                path: '/buttons',
                icon: {
                    name: 'home-01',
                    variant: 'stroke',
                    size: 16,
                },
                badge: 100,
            },
            {
                name: 'Badges',
                path: '/badges',
                icon: {
                    name: 'home-01',
                    variant: 'stroke',
                    size: 16,
                },
                badge: 100,
            },
            {
                name: 'Paths',
                path: '/Paths',
                children: [
                    {
                        name: 'Buttons',
                        path: '/buttons',
                        icon: {
                            name: 'home-01',
                            variant: 'stroke',
                            size: 16,
                        },
                        badge: 100,
                    },
                    {
                        name: 'Badges',
                        path: '/badges',
                        icon: {
                            name: 'home-01',
                            variant: 'stroke',
                            size: 16,
                        },
                        badge: 100,
                    },
                    {
                        name: 'Paths',
                        path: '/Paths',
                        icon: {
                            name: 'home-01',
                            variant: 'stroke',
                            size: 16,
                        },
                        badge: 100,
                    },
                ],
            },
            {
                name: 'Inputs',
                path: '/inputs',
                icon: {
                    name: 'home-01',
                    variant: 'stroke',
                    size: 16,
                },
                badge: 100,
            },
            {
                name: 'Accordions',
                path: '/accordions',
                icon: {
                    name: 'home-01',
                    variant: 'stroke',
                    size: 16,
                },
                badge: 100,
            },
            {
                name: 'Tabs',
                path: '/tabs',
                icon: {
                    name: 'home-01',
                    variant: 'stroke',
                    size: 16,
                },
                badge: 100,
            },
        ],
    },
    {
        name: 'Components 2',
        path: '/components',
        icon: {
            name: 'home-01',
            variant: 'stroke',
            size: 16,
        },
        disabled: false,
        children: [
            {
                name: 'Buttons',
                path: '/buttons',
                icon: {
                    name: 'home-01',
                    variant: 'stroke',
                    size: 16,
                },
                badge: 100,
            },
            {
                name: 'Badges',
                path: '/badges',
                icon: {
                    name: 'home-01',
                    variant: 'stroke',
                    size: 16,
                },
                badge: 100,
            },
            {
                name: 'Paths',
                path: '/Paths',
                children: [
                    {
                        name: 'Buttons',
                        path: '/buttons',
                        badge: 100,
                    },
                    {
                        name: 'Badges',
                        path: '/badges',
                        badge: 100,
                    },
                    {
                        name: 'Paths',
                        path: '/Paths',
                        badge: 100,
                    },
                ],
            },
            {
                name: 'Inputs',
                path: '/inputs',
                badge: 100,
            },
            {
                name: 'Accordions',
                path: '/accordions',
                badge: 100,
            },
            {
                name: 'Tabs',
                path: '/tabs',
                badge: 100,
            },
        ],
    },
];
const DgaHeaderUsage$1 = /*@__PURE__*/ proxyCustomElement(class DgaHeaderUsage extends H {
    constructor() {
        super();
        this.__registerHost();
        this.activeLink = 'link-1';
        this.collapsed = false;
        this.open = false;
    }
    render() {
        return (h(Fragment, { key: 'e3385db4cc7c904776685fcfab96e91c156e3a1e' }, h("dga-second-nav-header", { key: '0598782007ec678bb08d9036837935bfcc42105d', hideDivider: true }, h("dga-second-nav-header-actions", { key: '51aae57faf7e9ec1ad21cbb4647223cdb0fd1e52' }, h("dga-button-v2", { key: '7ae90fb38f4ef09c3b447527b85022e1fe3367dc', label: "Button", variant: "transparent", size: "sm", iconOnly: true, leadIcon: true, leadIconProps: {
                name: 'mic-01',
                type: 'rounded',
                variant: 'stroke',
                color: '#FFFFFF',
            } }), h("dga-button-v2", { key: '0efab41a2ba396c1aad28422af66724e1f1056ee', label: "Button", variant: "transparent", size: "sm", iconOnly: true, leadIcon: true, leadIconProps: {
                name: 'zoom-in-area',
                type: 'rounded',
                variant: 'stroke',
                color: '#FFFFFF',
            } }), h("dga-button-v2", { key: '6b6a520dbbf662ce64488e6a84c600cd162d63b2', label: "Button", variant: "transparent", size: "sm", iconOnly: true, leadIcon: true, leadIconProps: {
                name: 'zoom-out-area',
                type: 'rounded',
                variant: 'stroke',
                color: '#FFFFFF',
            } }), h("dga-button-v2", { key: '5708bd6889c7235a0acdde0eb5a6b01731fd8d5b', label: "Button", variant: "transparent", size: "sm", iconOnly: true, leadIcon: true, leadIconProps: {
                name: 'view',
                type: 'rounded',
                variant: 'stroke',
                color: '#FFFFFF',
            } })), h("dga-second-nav-header-content", { key: '650f1a363989398ff319599e8988fc219964ffc6' }, h("dga-second-nav-header-item", { key: '5100b1b03956721fc95c0bbac5875a075bd8f3f1', label: "Cloudy" }, h("dga-icon", { key: 'b01c16c4ad9d22e80c7f867cd4cd71d3cf722273', icon: "cloud", variant: "stroke", type: "rounded" })), h("dga-second-nav-header-item", { key: '3f21cf7ca74150e32cbe47a6e5277656c062e351', label: "21 Jan 2024" }, h("dga-icon", { key: '2638a776bce033e70173d61d7ffc33ae9b29dfcf', icon: "calendar-04", variant: "stroke", type: "rounded" })))), h("dga-nav-header", { key: 'd54c9f5a169b99a4a1b5e62c02b87580380375e1', fullWidth: true, divider: true, sticky: true }, h("dga-nav-header-main", { key: 'd9d15d943203013173ebfc015d01aa2fd68f6df3', collapsed: this.collapsed, onToggleCollapsed: e => {
                this.collapsed = e.detail;
                console.log('collapsed', e.detail);
            } }, h("dga-button", { key: '8382f833a92acb7055fca65144bb89149cd54b23', iconOnly: true, leadIcon: true, leadIconType: this.collapsed ? 'cancel01' : 'menu01', leadIconProps: {
                name: this.collapsed ? 'menu01' : 'cancel01',
                size: 24,
                variant: 'stroke',
                type: 'rounded',
            }, variant: "transparent", onClick: () => (this.collapsed = !this.collapsed), class: "header-menu__btn", sx: {
                display: { xs: 'block', md: 'none' },
            } }), h("dga-nav-header-logos", { key: '4e1d81eef2746d1c1a3f55fcb72d1dc145b9ef44', logoSrc: "https://dga-nds-fbhtx.ondigitalocean.app/mobile-logo.svg", govSrc: "https://dga-nds-fbhtx.ondigitalocean.app/mobile-logo.svg", logoLink: "#", govLink: "#" }), h("dga-nav-header-menu", { key: 'cf21218b457653a86f50cf68540667b135edf1aa' }, h("dga-nav-header-link", { key: '4d41605ab1a72d636110a86715655a17fc5768b7', label: "Item 1", icon: "arrow-down-01", active: this.activeLink === 'link-1', subMenuBackground: "white", onClick: () => {
                this.activeLink = 'link-1';
            } }, h("dga-nav-header-sub-menu", { key: 'f5bfdd619a8c37e5a62c5d3c18459e60fbc92ee3', "sub-menu-full-width": true, background: 'white' }, h("dga-nav-header-sub-menu-column", { key: '0e6510cf1164c6e075c83416f201080de54467fc', label: "Group Label" }, h("dga-nav-header-sub-menu-link", { key: 'fac5b53ebee03ba20ba7fe8413c56b9080e51d4d', label: "Submenu Link", link: "#", icon: "document", helperText: "This is a description" }), h("dga-nav-header-sub-menu-link", { key: 'f090b6d14d69d2e7a6875934ccbae22847fdd975', label: "Submenu Link", link: "#", icon: "document", helperText: "This is a description" })), h("dga-nav-header-sub-menu-column", { key: 'fff415016384a2f393345be222b9e302635a9384', label: "Group Label" }, h("dga-nav-header-sub-menu-link", { key: '3109a500a78f7e23f60dec269e4565058b93d7be', label: "Submenu Link", link: "#", icon: "document", helperText: "This is a description" }), h("dga-nav-header-sub-menu-link", { key: 'dd8707d4183a2fd6011d0c7d8d5981337b60e443', label: "Submenu Link", link: "#", icon: "document", helperText: "This is a description" })), h("dga-nav-header-sub-menu-column", { key: 'd890390ba6e09789f4b67acbc3124a8af7a65fb7', label: "Group Label" }, h("dga-nav-header-sub-menu-link", { key: 'c7f9afa1d6928484841f5f6916196370dfd346de', label: "Submenu Link", link: "#", icon: "document", helperText: "This is a description" }), h("dga-nav-header-sub-menu-link", { key: 'c69f4df3fc6f80c814519a4c496587e9e0628cc2', label: "Submenu Link", link: "#", icon: "document", helperText: "This is a description" })), h("dga-nav-header-sub-menu-column", { key: '0c132794b61081832d3ad63926ef544bc9e47847', label: "Group Label" }, h("dga-nav-header-sub-menu-link", { key: 'ec115f26bfaa3e834a567b88befa169a22437192', label: "Submenu Link", link: "#", icon: "document", helperText: "This is a description" }), h("dga-nav-header-sub-menu-link", { key: '7d0ea811f004361fd2d9af63fc6954b852a00e1c', label: "Submenu Link", link: "#", icon: "document", helperText: "This is a description" })))), h("dga-nav-header-link", { key: '89fcb9b199aa0794760ac49dbc4f917cad298a01', label: "Item  2", icon: "arrow-down-01", active: this.activeLink === 'link-2', subMenuBackground: "white", subMenuFullWidth: true, onClick: () => {
                this.activeLink = 'link-2';
            } }, h("dga-nav-header-sub-menu", { key: '3cc24da9456a9d37c805036e8411c08a9a4f9a9b', "sub-menu-full-width": true, background: 'white' }, h("dga-nav-header-sub-menu-column", { key: 'a7677e76f9685333c86fef2e861e0c8a2d509af2', label: "Group Label" }, h("dga-nav-header-sub-menu-link", { key: '0431f2d3072bf58d330e3893705fab20c9681edb', label: "Submenu Link", link: "#", icon: "document", helperText: "This is a description" }), h("dga-nav-header-sub-menu-link", { key: 'a7b34df862e633420b5de229988f6b05f03912e8', label: "Submenu Link", link: "#", icon: "document", helperText: "This is a description" })), h("dga-nav-header-sub-menu-column", { key: '73d8c75b2473a22aefa946a0845e19915a843b7f', label: "Group Label" }, h("dga-nav-header-sub-menu-link", { key: 'f615e9dbbf06b4ec27aef011e2bea1399769cd05', label: "Submenu Link", link: "#", icon: "document", helperText: "This is a description" }), h("dga-nav-header-sub-menu-link", { key: 'e137066a8af0a584883ed75ddc3719e072215d43', label: "Submenu Link", link: "#", icon: "document", helperText: "This is a description" })))), h("dga-nav-header-link", { key: '51d898d51e795f5b7cba88d109d117c7d63b3767', label: "Item  3", icon: "arrow-down-01", active: this.activeLink === 'link-3', subMenuBackground: "white", subMenuFullWidth: true, onClick: () => {
                this.activeLink = 'link-3';
            } }), h("dga-nav-header-link", { key: '6d9fbf3f411afe0d918856da9bd066f75befc066', label: "Item  4", icon: "arrow-down-01", active: this.activeLink === 'link-4', subMenuBackground: "white", subMenuFullWidth: true, onClick: () => {
                this.activeLink = 'link-4';
            } }, h("dga-nav-header-sub-menu", { key: '65a994159637d3719c5c2c34a788322f9863b952', "sub-menu-full-width": true, background: 'white' }, h("dga-nav-header-sub-menu-column", { key: '0dd0c7fae18dc1ffd52d5b922e991b6a91cdff7a', label: "Group Label" }, h("dga-nav-header-sub-menu-link", { key: '94ce61962c68c80f5d97f7686a3752e35efdc311', label: "Submenu Link", link: "#", icon: "document", helperText: "This is a description" }), h("dga-nav-header-sub-menu-link", { key: '9d0c77fe34dfb9224cd52fe309c18b778341c72b', label: "Submenu Link", link: "#", icon: "document", helperText: "This is a description" })), h("dga-nav-header-sub-menu-column", { key: 'ea5e8dc656f00a6f664899e206ff247f15929472', label: "Group Label" }, h("dga-nav-header-sub-menu-link", { key: 'd7b350f040eb0d3111e57d675d6f5c65a34e5c1c', label: "Submenu Link", link: "#", icon: "document", helperText: "This is a description" }), h("dga-nav-header-sub-menu-link", { key: 'd0c92b607cccf0fbedc82a3a492d8eda1a07a6f4', label: "Submenu Link", link: "#", icon: "document", helperText: "This is a description" })), h("dga-nav-header-sub-menu-column", { key: '130d522dbe4b31b8f24ef29d2d01aff0e2f25165', label: "Group Label" }, h("dga-nav-header-sub-menu-link", { key: '618c6ff3c5e3af8377dc488f74409f46fefe90ad', label: "Submenu Link", link: "#", icon: "document", helperText: "This is a description" }), h("dga-nav-header-sub-menu-link", { key: 'a4f333b748d52a1348dc7913643f8c3118735de0', label: "Submenu Link", link: "#", icon: "document", helperText: "This is a description" })), h("dga-nav-header-sub-menu-column", { key: '08426c03a85805f1c968260f330cb6e44e5892b5', label: "Group Label" }, h("dga-nav-header-sub-menu-link", { key: 'a984c6cae3fbd074cce06ed626efc210ad35c049', label: "Submenu Link", link: "#", icon: "document", helperText: "This is a description" }), h("dga-nav-header-sub-menu-link", { key: 'f5b303af0098a57a29a1940d6c2be0521a8e82c5', label: "Submenu Link", link: "#", icon: "document", helperText: "This is a description" })), h("dga-nav-header-sub-menu-column", { key: '98f9c343c219114fbfb5c032383477517711bf30', label: "Group Label" }, h("dga-nav-header-sub-menu-link", { key: 'fae11923ab1e75e444bff600c1218982fa62ee41', label: "Submenu Link", link: "#", icon: "document", helperText: "This is a description" }), h("dga-nav-header-sub-menu-link", { key: '1df0536a0640fc6c16e0d0baa532d67ee74da40f', label: "Submenu Link", link: "#", icon: "document", helperText: "This is a description" })), h("dga-nav-header-sub-menu-column", { key: '7e7e4b5e47751843adb4c80fac6ad407749f014d', label: "Group Label" }, h("dga-nav-header-sub-menu-link", { key: '7d176c853bb98664e057ae6f080a4f9ff0823c72', label: "Submenu Link", link: "#", icon: "document", helperText: "This is a description" }), h("dga-nav-header-sub-menu-link", { key: '4bc10fd0f5683892591022c95299c484aa1e4d38', label: "Submenu Link", link: "#", icon: "document", helperText: "This is a description" })))))), h("dga-nav-header-actions", { key: '4948ab5bee3565e9708bb2314e2f624ff055c45d' }, h("dga-header-action-btn", { key: 'c64d9f873f2aebb813a47d57f0837b2196bbcbd2', label: "Search", icon: "search-01", subMenuBackground: "white" }, h("dga-nav-header-sub-menu", { key: 'b4e9cff787ae906200dcb7e5c7100be757cc1220', "sub-menu-full-width": true, background: 'white' }, h("dga-flex", { key: '68800e98ab349aecf45fa13f9bce9da8116c5c9a', spacing: 16 }, h("dga-search-box", { key: '59a3e892f806207c614257b798262f0ae0dde7c5', placeholder: "Search" }), h("dga-button", { key: '006f5d39a082ed236eb530ee35869de1b1e56ef6', label: "Search" })))), h("dga-header-action-btn", { key: '230014b636e4170f43095db7773a49d61f87a4e8', label: "\u0639\u0631\u0628\u064A", icon: "translation" }), h("dga-header-action-btn", { key: '701ae73c75a5a47549d91bb085b0a48a5746f733', label: "Login", icon: "user" })), h("dga-header-mobile-actions", { key: '6bc464a9b18d1bb0709b41bbb333401baff41ed5' }, h("dga-header-menu", { key: '6a3a3cdb937005747606defaaba4498f25ade27c', actions: [
                {
                    label: 'Search',
                    icon: 'search-01',
                    onClick: () => {
                        console.log('clicked');
                    },
                },
                {
                    label: 'عربي',
                    icon: 'translation',
                    onClick: () => {
                        console.log('clicked');
                    },
                },
                {
                    label: 'Login',
                    icon: 'user',
                    onClick: () => {
                        console.log('clicked');
                    },
                }
            ] }))), h("dga-drawer", { key: 'bdac38ea781c432b5fd7090618c49cf66f030f6c', routes: SidePanelRoutes, open: this.collapsed, background: 'gray', headerLogo: "https://casamedia.com/wp-content/uploads/2023/04/adidas-1024x683.png", sx: {
                '& > .sidepanel': {
                    top: '112px',
                },
            } }), h("div", { key: '778edde457382b0cce07b9ee15153c65f825792b' }, "Test")));
    }
}, [0, "dga-header-usage", {
        "activeLink": [32],
        "collapsed": [32],
        "open": [32]
    }]);
function defineCustomElement$1() {
    if (typeof customElements === "undefined") {
        return;
    }
    const components = ["dga-header-usage", "dga-button", "dga-button-v2", "dga-drawer", "dga-flex", "dga-header-action-btn", "dga-header-menu", "dga-header-mobile-actions", "dga-icon", "dga-icon-v2", "dga-nav-header", "dga-nav-header-actions", "dga-nav-header-link", "dga-nav-header-logos", "dga-nav-header-main", "dga-nav-header-menu", "dga-nav-header-sub-menu", "dga-nav-header-sub-menu-column", "dga-nav-header-sub-menu-link", "dga-search-box", "dga-second-nav-header", "dga-second-nav-header-actions", "dga-second-nav-header-content", "dga-second-nav-header-item", "dga-tag", "dga-tooltip", "side-panel-menu", "submenu-component", "tab-component", "tab-link"];
    components.forEach(tagName => { switch (tagName) {
        case "dga-header-usage":
            if (!customElements.get(tagName)) {
                customElements.define(tagName, DgaHeaderUsage$1);
            }
            break;
        case "dga-button":
            if (!customElements.get(tagName)) {
                defineCustomElement$u();
            }
            break;
        case "dga-button-v2":
            if (!customElements.get(tagName)) {
                defineCustomElement$t();
            }
            break;
        case "dga-drawer":
            if (!customElements.get(tagName)) {
                defineCustomElement$s();
            }
            break;
        case "dga-flex":
            if (!customElements.get(tagName)) {
                defineCustomElement$r();
            }
            break;
        case "dga-header-action-btn":
            if (!customElements.get(tagName)) {
                defineCustomElement$q();
            }
            break;
        case "dga-header-menu":
            if (!customElements.get(tagName)) {
                defineCustomElement$p();
            }
            break;
        case "dga-header-mobile-actions":
            if (!customElements.get(tagName)) {
                defineCustomElement$o();
            }
            break;
        case "dga-icon":
            if (!customElements.get(tagName)) {
                defineCustomElement$n();
            }
            break;
        case "dga-icon-v2":
            if (!customElements.get(tagName)) {
                defineCustomElement$m();
            }
            break;
        case "dga-nav-header":
            if (!customElements.get(tagName)) {
                defineCustomElement$l();
            }
            break;
        case "dga-nav-header-actions":
            if (!customElements.get(tagName)) {
                defineCustomElement$k();
            }
            break;
        case "dga-nav-header-link":
            if (!customElements.get(tagName)) {
                defineCustomElement$j();
            }
            break;
        case "dga-nav-header-logos":
            if (!customElements.get(tagName)) {
                defineCustomElement$i();
            }
            break;
        case "dga-nav-header-main":
            if (!customElements.get(tagName)) {
                defineCustomElement$h();
            }
            break;
        case "dga-nav-header-menu":
            if (!customElements.get(tagName)) {
                defineCustomElement$g();
            }
            break;
        case "dga-nav-header-sub-menu":
            if (!customElements.get(tagName)) {
                defineCustomElement$f();
            }
            break;
        case "dga-nav-header-sub-menu-column":
            if (!customElements.get(tagName)) {
                defineCustomElement$e();
            }
            break;
        case "dga-nav-header-sub-menu-link":
            if (!customElements.get(tagName)) {
                defineCustomElement$d();
            }
            break;
        case "dga-search-box":
            if (!customElements.get(tagName)) {
                defineCustomElement$c();
            }
            break;
        case "dga-second-nav-header":
            if (!customElements.get(tagName)) {
                defineCustomElement$b();
            }
            break;
        case "dga-second-nav-header-actions":
            if (!customElements.get(tagName)) {
                defineCustomElement$a();
            }
            break;
        case "dga-second-nav-header-content":
            if (!customElements.get(tagName)) {
                defineCustomElement$9();
            }
            break;
        case "dga-second-nav-header-item":
            if (!customElements.get(tagName)) {
                defineCustomElement$8();
            }
            break;
        case "dga-tag":
            if (!customElements.get(tagName)) {
                defineCustomElement$7();
            }
            break;
        case "dga-tooltip":
            if (!customElements.get(tagName)) {
                defineCustomElement$6();
            }
            break;
        case "side-panel-menu":
            if (!customElements.get(tagName)) {
                defineCustomElement$5();
            }
            break;
        case "submenu-component":
            if (!customElements.get(tagName)) {
                defineCustomElement$4();
            }
            break;
        case "tab-component":
            if (!customElements.get(tagName)) {
                defineCustomElement$3();
            }
            break;
        case "tab-link":
            if (!customElements.get(tagName)) {
                defineCustomElement$2();
            }
            break;
    } });
}

const DgaHeaderUsage = DgaHeaderUsage$1;
const defineCustomElement = defineCustomElement$1;

export { DgaHeaderUsage, defineCustomElement };
