// @platformscode/core/dist/components/dga-modal-usage.js
import { p as proxyCustomElement, H, h } from './p-BhqVmlhr.js';
import { d as defineCustomElement$6 } from './p-B4U0lY4y.js';
import { d as defineCustomElement$5 } from './p-ByzFFJgL.js';
import { d as defineCustomElement$4 } from './p-DN3DJd5E.js';
import { d as defineCustomElement$3 } from './p-DBTOrdV2.js';
import { d as defineCustomElement$2 } from './p-bb4bmTVr.js';

const DgaModalUsage$1 = /*@__PURE__*/ proxyCustomElement(class DgaModalUsage extends H {
    constructor() {
        super();
        this.__registerHost();
        this.show = false;
    }
    render() {
        return (h("div", { key: '8cde72199e094c6b7dff6f5cd34ba6aa6a6f081f', style: { position: "relative" } }, h("h1", { key: 'ab21096bfdaac655e70beb5f1f9657b943f10940' }, "Modal Example"), h("button", { key: '7c9e08e528699b0770816eb7a585aa79269e012a', onClick: () => this.show = true }, "Open Modal"), h("dga-modal", { key: '2e4e878d1e859e8017c369d66aec7b92e77f1ac9', buttonsList: [
                {
                    extraClass: '',
                    label: 'Close',
                    onClick: () => { },
                    title: 'Close Modal'
                }
            ], closeButton: true, modalTitle: "Basic Modal", name: "basicModal", onClose: () => { }, open: this.show }, "This is a simple modal with default configuration."), this.show ? "shown" : "hidden"));
    }
}, [0, "dga-modal-usage", {
        "show": [32]
    }]);
function defineCustomElement$1() {
    if (typeof customElements === "undefined") {
        return;
    }
    const components = ["dga-modal-usage", "dga-button-v2", "dga-featured-icon", "dga-icon", "dga-icon-v2", "dga-modal"];
    components.forEach(tagName => { switch (tagName) {
        case "dga-modal-usage":
            if (!customElements.get(tagName)) {
                customElements.define(tagName, DgaModalUsage$1);
            }
            break;
        case "dga-button-v2":
            if (!customElements.get(tagName)) {
                defineCustomElement$6();
            }
            break;
        case "dga-featured-icon":
            if (!customElements.get(tagName)) {
                defineCustomElement$5();
            }
            break;
        case "dga-icon":
            if (!customElements.get(tagName)) {
                defineCustomElement$4();
            }
            break;
        case "dga-icon-v2":
            if (!customElements.get(tagName)) {
                defineCustomElement$3();
            }
            break;
        case "dga-modal":
            if (!customElements.get(tagName)) {
                defineCustomElement$2();
            }
            break;
    } });
}

const DgaModalUsage = DgaModalUsage$1;
const defineCustomElement = defineCustomElement$1;

export { DgaModalUsage, defineCustomElement };
