// @platformscode/core/dist/components/dga-card-usage.js
import { p as proxyCustomElement, H, h } from './p-BhqVmlhr.js';
import { d as defineCustomElement$8 } from './p-CClx_RDs.js';
import { d as defineCustomElement$7 } from './p-B4U0lY4y.js';
import { d as defineCustomElement$6 } from './p-CG_XWnd8.js';
import { d as defineCustomElement$5 } from './p-ChhNw26D.js';
import { d as defineCustomElement$4 } from './p-ByzFFJgL.js';
import { d as defineCustomElement$3 } from './p-DN3DJd5E.js';
import { d as defineCustomElement$2 } from './p-DBTOrdV2.js';

const DgaCardUsage$1 = /*@__PURE__*/ proxyCustomElement(class DgaCardUsage extends H {
    constructor() {
        super();
        this.__registerHost();
        this.variant = 'neutral';
        this.selected = false;
    }
    render() {
        return (h("div", { key: 'a5891b4a7e1c7c6f40534dba563a0868c030b6be' }, h("h1", { key: 'f4bdc496f1b640e4ca9debae252aebae0408cccc' }, "Card Example"), h("div", { key: '3852ccdd17c58d4369377947b5eb2aefd553afe5', style: { padding: '50px', display: 'flex', gap: '20px' } }, h("div", { key: 'a24d0337bd66b59374ee5c651bd1e2104cac991f', style: { display: 'flex', flexDirection: 'column', gap: '20px' } }, h("dga-card", { key: '9ec040fdee43c39087ed6ea1e80f1e1907f1794d', type: "expanded", cardTitle: "Card Title", description: "Card content placeholder text goes here", showFeaturedIcon: true }, h("div", { key: 'cfa10cad35b0f4c53ea5a0c16ade5e01d179372d', slot: "collapse-content" }, "dgdsgdfgdfg")), h("dga-card", { key: '7fca6697abb3c4218c8eb6ebe0ea026a2d6dce90', type: "selectable", cardTitle: "Card Title", description: "Card content placeholder text goes here", isSelected: this.selected, onCheckboxChange: () => (this.selected = !this.selected) }), h("dga-card", { key: '9d31f50eaeb197f8cbce1467aeb573b78b59dd70', cardTitle: "Card Title", sx: {
                '& > .card': {
                    backgroundColor: 'red',
                    height: '400px',
                },
            } }), h("dga-card", { key: 'ec89d1a7e46c0b89e5097f4d734796f57916293f', cardTitle: "No Shadow", effect: "no-shadow" }), h("dga-card", { key: 'd14ef0aa42b8d53b305fd4fdf0d1abd06d6cc812', cardTitle: "Stroke", effect: "stroke" }), h("dga-card", { key: 'f8a9b640ab7b7fd169422c61ca285dfb41358c82', cardTitle: "With Shadow", effect: "with-shadow" })), h("div", { key: '26a0b295fc2397a857048501d22fc1d8a0914ada', style: { display: 'flex', flexDirection: 'column', gap: '20px' } }))));
    }
}, [0, "dga-card-usage", {
        "variant": [32],
        "selected": [32]
    }]);
function defineCustomElement$1() {
    if (typeof customElements === "undefined") {
        return;
    }
    const components = ["dga-card-usage", "dga-button", "dga-button-v2", "dga-card", "dga-checkbox", "dga-featured-icon", "dga-icon", "dga-icon-v2"];
    components.forEach(tagName => { switch (tagName) {
        case "dga-card-usage":
            if (!customElements.get(tagName)) {
                customElements.define(tagName, DgaCardUsage$1);
            }
            break;
        case "dga-button":
            if (!customElements.get(tagName)) {
                defineCustomElement$8();
            }
            break;
        case "dga-button-v2":
            if (!customElements.get(tagName)) {
                defineCustomElement$7();
            }
            break;
        case "dga-card":
            if (!customElements.get(tagName)) {
                defineCustomElement$6();
            }
            break;
        case "dga-checkbox":
            if (!customElements.get(tagName)) {
                defineCustomElement$5();
            }
            break;
        case "dga-featured-icon":
            if (!customElements.get(tagName)) {
                defineCustomElement$4();
            }
            break;
        case "dga-icon":
            if (!customElements.get(tagName)) {
                defineCustomElement$3();
            }
            break;
        case "dga-icon-v2":
            if (!customElements.get(tagName)) {
                defineCustomElement$2();
            }
            break;
    } });
}

const DgaCardUsage = DgaCardUsage$1;
const defineCustomElement = defineCustomElement$1;

export { DgaCardUsage, defineCustomElement };
