// @platformscode/core/dist/components/dga-dropdown-usage.js
import { p as proxyCustomElement, H, h } from './p-BhqVmlhr.js';
import { d as defineCustomElement$3 } from './p-CnmVLwSO.js';
import { d as defineCustomElement$2 } from './p-DN3DJd5E.js';

const DgaDropdownUsage$1 = /*@__PURE__*/ proxyCustomElement(class DgaDropdownUsage extends H {
    constructor() {
        super();
        this.__registerHost();
        this.activeStep = 1;
        this.selectedValue = 'Option-1';
    }
    render() {
        return (h("div", { key: '0a6b57e348ddc9ce5a0964865d5d2af5603f0cb8' }, h("h1", { key: 'df207a773226becdc93f531f2b1499fa61566af4' }, "Dropdown Example"), h("div", { key: '00d703c374e5b3727dc5cfaa984984d3f23917b5', style: { padding: '50px', display: 'flex', gap: '20px' } }, h("div", { key: '8e257d0f1e1ee8b6522d231f54a9b6abf9151931', style: { display: 'flex', flexDirection: 'column', gap: '20px' } }, h("dga-dropdown", { key: '53fc7829cf7bccc4013cf8110ba6a9c2441ff848', placeholder: "Placeholder text", variant: "default", optionLabel: "label", trackBy: "value", value: this.selectedValue, options: [
                {
                    label: 'Netherlands Antiles',
                    value: 'Option-1',
                },
                {
                    label: 'Option 2',
                    value: 'Option-2',
                },
                {
                    label: 'Option 3',
                    value: 'Option-3',
                },
            ], onChange: (value) => this.selectedValue = value }), this.selectedValue, h("dga-dropdown", { key: 'e12ad71e46991a1adc6bc48c9f7e9016f9802dba', placeholder: "Placeholder text", variant: "darker", optionLabel: "label", trackBy: "value", options: [
                {
                    label: 'Option',
                    value: 'Option',
                },
                {
                    label: 'Option',
                    value: 'Option',
                },
                {
                    label: 'Option Optionbbb',
                    value: 'Option',
                },
                {
                    label: 'Option',
                    value: 'Option',
                },
            ] }), h("dga-dropdown", { key: '6b25aa37571846e3a9eab024d8210714908c3a77', placeholder: "Placeholder text", variant: "lighter", optionLabel: "label", trackBy: "value", options: [
                {
                    label: 'Option',
                    value: 'Option',
                },
                {
                    label: 'Option',
                    value: 'Option',
                },
                {
                    label: 'Option',
                    value: 'Option',
                },
                {
                    label: 'Option',
                    value: 'Option',
                },
            ] }), h("dga-dropdown", { key: '1504cdf3f7aeb48c285c00c9515c1881f3f42bfe', error: true, placeholder: "Placeholder text", variant: "lighter", optionLabel: "label", trackBy: "value", options: [
                {
                    label: 'Option',
                    value: 'Option',
                },
                {
                    label: 'Option',
                    value: 'Option',
                },
                {
                    label: 'Option',
                    value: 'Option',
                },
                {
                    label: 'Option',
                    value: 'Option',
                },
            ] }), ['lg', 'md'].map((size) => (h("dga-dropdown", { key: size, size: size, placeholder: "Placeholder text", variant: "default", optionLabel: "label", trackBy: "value", options: [
                {
                    label: 'Option',
                    value: 'Option',
                },
                {
                    label: 'Option',
                    value: 'Option',
                },
                {
                    label: 'Option',
                    value: 'Option',
                },
                {
                    label: 'Option',
                    value: 'Option',
                },
            ] }))), h("dga-dropdown", { key: '9eec2e8183875e424ebf805cd27b675ecbb6d99b', multiSelect: true, placeholder: "Placeholder text", variant: "lighter", optionLabel: "label", trackBy: "value", options: [
                {
                    label: 'Option',
                    value: 'Option',
                },
                {
                    label: 'Option',
                    value: 'Option',
                },
                {
                    label: 'Option',
                    value: 'Option',
                },
                {
                    label: 'Option',
                    value: 'Option',
                },
            ] })), h("div", { key: '7447713413d02ad8b39ed86c303c59ed5def5648', style: { display: 'flex', flexDirection: 'column', gap: '20px' } }))));
    }
}, [0, "dga-dropdown-usage", {
        "activeStep": [32],
        "selectedValue": [32]
    }]);
function defineCustomElement$1() {
    if (typeof customElements === "undefined") {
        return;
    }
    const components = ["dga-dropdown-usage", "dga-dropdown", "dga-icon"];
    components.forEach(tagName => { switch (tagName) {
        case "dga-dropdown-usage":
            if (!customElements.get(tagName)) {
                customElements.define(tagName, DgaDropdownUsage$1);
            }
            break;
        case "dga-dropdown":
            if (!customElements.get(tagName)) {
                defineCustomElement$3();
            }
            break;
        case "dga-icon":
            if (!customElements.get(tagName)) {
                defineCustomElement$2();
            }
            break;
    } });
}

const DgaDropdownUsage = DgaDropdownUsage$1;
const defineCustomElement = defineCustomElement$1;

export { DgaDropdownUsage, defineCustomElement };
