// @platformscode/core/dist/components/dga-content-switcher-usage.js
import { p as proxyCustomElement, H, h } from './p-BhqVmlhr.js';
import { d as defineCustomElement$4 } from './p-DU0OBmMd.js';
import { d as defineCustomElement$3 } from './p-oMkoCN12.js';
import { d as defineCustomElement$2 } from './p-DVhw1b1n.js';

const DgaContentSwitcherUsage$1 = /*@__PURE__*/ proxyCustomElement(class DgaContentSwitcherUsage extends H {
    constructor() {
        super();
        this.__registerHost();
        this.value = 0;
    }
    render() {
        return (h("div", { key: '6903e09d27c812323da710c32b30d7a7c67e2aa6' }, h("h1", { key: '2dac404327f2f2259513f4902a0af1dfd8303856' }, "Content Switcher Example"), h("dga-content-switcher", { key: '3e77ad327e45df9d3090d66dd2d307d54d238f1a', size: 'lg' }, h("dga-content-switcher-item", { key: '45428c19a30312c5171e7668553537999f55677c', label: "Option 1", content: "This is Option 1 content." }), h("dga-content-switcher-item", { key: 'ce8d129c1dc1fda71c1d4327e0c68f09cb1db245', label: "Option 2", content: "This is Option 2 content." }), h("dga-content-switcher-item", { key: '0367aae3341396a297c46ed57a679bebb67c5707', label: "Option 3", content: "This is Option 3 content." })), h("dga-content-switcher", { key: 'a63ee1fc31a1929be258d2d30614dddc545e819d', size: 'md' }, h("dga-content-switcher-item", { key: 'e108e0f2a04360ecfb9f716d74333cd4c10fa937', label: "Option 1", content: "This is Option 1 content." }), h("dga-content-switcher-item", { key: '9d20eb19271d3240bd015dfa5d745091d519e203', label: "Option 2", content: "This is Option 2 content." }), h("dga-content-switcher-item", { key: '375268ab4e6f1c490f897d488cd890e6bff01a93', label: "Option 3", content: "This is Option 3 content." })), h("dga-content-switcher", { key: '4a625a1fa721c34925814a1f19c1e86643f8da13', size: 'sm', onColor: true }, h("dga-content-switcher-item", { key: 'e18724ed936f1ddab4f9f4ce6acb9acb984f7043', label: "Option 1", content: "This is Option 1 content." }), h("dga-content-switcher-item", { key: 'b779698cc4246daa9a73125203fd8df79608be4b', label: "Option 2", content: "This is Option 2 content." }), h("dga-content-switcher-item", { key: 'a9a5e220e10ded846d3cff85927f4e0d4465cd64', label: "Option 3", content: "This is Option 3 content." })), h("h1", { key: 'f0155b208cedb99ff722d3b115cac0fd9c837202' }, this.value), h("dga-content-switcher", { key: '341e37fb473c68ec39c1dd11b69cc6378eb6b9f6', size: 'lg', value: this.value, onChange: (val) => this.value = val }), h("dga-content-switcher-panel", { key: '309a4263c65ebe2445cd4ce6235d947702565991', value: this.value, index: 0 }, h("div", { key: '0c0f27a7290b9496197ee93925ae58d3f9400386' }, "Content 1")), h("dga-content-switcher-panel", { key: '4b4509182481333595a01465fb7c6f106626cc5f', value: this.value, index: 1 }, h("div", { key: '19b0ddfc985c2a4ad078fb42b3f1143611c7f644' }, "Content 2")), h("dga-content-switcher-panel", { key: '7472c85660a3299300af5d02ae880c61080ef243', value: this.value, index: 2 }, h("div", { key: 'f63b0da8787366b4cb646010a23f866ab6b43951' }, "Content 3"))));
    }
}, [0, "dga-content-switcher-usage", {
        "value": [2]
    }]);
function defineCustomElement$1() {
    if (typeof customElements === "undefined") {
        return;
    }
    const components = ["dga-content-switcher-usage", "dga-content-switcher", "dga-content-switcher-item", "dga-content-switcher-panel"];
    components.forEach(tagName => { switch (tagName) {
        case "dga-content-switcher-usage":
            if (!customElements.get(tagName)) {
                customElements.define(tagName, DgaContentSwitcherUsage$1);
            }
            break;
        case "dga-content-switcher":
            if (!customElements.get(tagName)) {
                defineCustomElement$4();
            }
            break;
        case "dga-content-switcher-item":
            if (!customElements.get(tagName)) {
                defineCustomElement$3();
            }
            break;
        case "dga-content-switcher-panel":
            if (!customElements.get(tagName)) {
                defineCustomElement$2();
            }
            break;
    } });
}

const DgaContentSwitcherUsage = DgaContentSwitcherUsage$1;
const defineCustomElement = defineCustomElement$1;

export { DgaContentSwitcherUsage, defineCustomElement };
