// @platformscode/core/dist/components/dga-modal-v2-usage.js
import { p as proxyCustomElement, H, h } from './p-BhqVmlhr.js';
import { d as defineCustomElement$c } from './p-B4U0lY4y.js';
import { d as defineCustomElement$b } from './p-ByzFFJgL.js';
import { d as defineCustomElement$a } from './p-CCimZGO2.js';
import { d as defineCustomElement$9 } from './p-DN3DJd5E.js';
import { d as defineCustomElement$8 } from './p-DBTOrdV2.js';
import { d as defineCustomElement$7 } from './p-Bp8hOr9W.js';
import { d as defineCustomElement$6 } from './p-DVuEJ0rH.js';
import { d as defineCustomElement$5 } from './p-BsTqr68_.js';
import { d as defineCustomElement$4 } from './p-DZ-vlBcF.js';
import { d as defineCustomElement$3 } from './p-CbwJ6GMU.js';
import { d as defineCustomElement$2 } from './p-CR6GP5Pr.js';

const DgaModalUsage = /*@__PURE__*/ proxyCustomElement(class DgaModalUsage extends H {
    constructor() {
        super();
        this.__registerHost();
        this.show = false;
    }
    render() {
        return (h("div", { key: '3a615d3543f9a2f78d36d703e7d09486e61deb66', style: { position: 'relative' } }, h("h1", { key: '52d6b0f5327d4b43c6f2ba441269f18ed2add35b' }, "Modal Example"), h("button", { key: 'f662fa7a71219988266afcea5fd853c346a87ac6', onClick: () => (this.show = true) }, "Open Modal"), h("dga-modal-v2", { key: '031406f95a3da5a3abb7985c8bff2a46a05e27f8', open: this.show, showCloseButton: true, onClose: () => (this.show = false) }, h("dga-modal-header", { key: '646b81b064fa72328ef891433306f76043edbd55' }, h("dga-featured-icon", { key: 'c841091c7904b9a91c530930c199a86c99bbcb5e', color: "brand", variant: "light", size: "md", icon: {
                name: 'InformationCircle',
                variant: 'stroke',
                type: 'sharp',
                size: 20,
            } }), h("dga-modal-title", { key: 'a64a87932ce67def4699a340f6954d001edfd943' }, "Title goes here")), h("dga-modal-body", { key: '6adf7f8b3a3ac14fb5077a31515536bb1cd0eae3' }, h("dga-flex", { key: '6282de196bc18217211be3b049f47f93ed312fd0', direction: "column", spacing: 8 }, h("p", { key: '13304ae7000943f0e539e2eb5a971d7668c10261', class: "text-sm-regular" }, "When a Modal needs a further detailed explanation, it goes here."), h("dga-rating", { key: 'f51de3838de5837834753e0553aea9bceb9a9697' }))), h("dga-modal-actions", { key: '418241a19ea03507e02c85c6725d5ab42af70953' }, h("dga-button-v2", { key: 'bcfadb76b0f68d9d005bad61178eb7cfa0dac077', label: "Primary", size: "lg", variant: "primary", onClick: () => console.log('Primary Action') }), h("dga-button-v2", { key: '721df5c8993773ea7857702bfbb98f448b6e4505', label: "Secondary", size: "lg", variant: "secondary-outline", onClick: () => console.log('Secondary Action') })))));
    }
}, [0, "dga-modal-v2-usage", {
        "show": [32]
    }]);
function defineCustomElement$1() {
    if (typeof customElements === "undefined") {
        return;
    }
    const components = ["dga-modal-v2-usage", "dga-button-v2", "dga-featured-icon", "dga-flex", "dga-icon", "dga-icon-v2", "dga-modal-actions", "dga-modal-body", "dga-modal-header", "dga-modal-title", "dga-modal-v2", "dga-rating"];
    components.forEach(tagName => { switch (tagName) {
        case "dga-modal-v2-usage":
            if (!customElements.get(tagName)) {
                customElements.define(tagName, DgaModalUsage);
            }
            break;
        case "dga-button-v2":
            if (!customElements.get(tagName)) {
                defineCustomElement$c();
            }
            break;
        case "dga-featured-icon":
            if (!customElements.get(tagName)) {
                defineCustomElement$b();
            }
            break;
        case "dga-flex":
            if (!customElements.get(tagName)) {
                defineCustomElement$a();
            }
            break;
        case "dga-icon":
            if (!customElements.get(tagName)) {
                defineCustomElement$9();
            }
            break;
        case "dga-icon-v2":
            if (!customElements.get(tagName)) {
                defineCustomElement$8();
            }
            break;
        case "dga-modal-actions":
            if (!customElements.get(tagName)) {
                defineCustomElement$7();
            }
            break;
        case "dga-modal-body":
            if (!customElements.get(tagName)) {
                defineCustomElement$6();
            }
            break;
        case "dga-modal-header":
            if (!customElements.get(tagName)) {
                defineCustomElement$5();
            }
            break;
        case "dga-modal-title":
            if (!customElements.get(tagName)) {
                defineCustomElement$4();
            }
            break;
        case "dga-modal-v2":
            if (!customElements.get(tagName)) {
                defineCustomElement$3();
            }
            break;
        case "dga-rating":
            if (!customElements.get(tagName)) {
                defineCustomElement$2();
            }
            break;
    } });
}

const DgaModalV2Usage = DgaModalUsage;
const defineCustomElement = defineCustomElement$1;

export { DgaModalV2Usage, defineCustomElement };
