// @platformscode/core/dist/components/dga-codesnippet-usage.js
import { p as proxyCustomElement, H, h } from './p-BhqVmlhr.js';
import { d as defineCustomElement$7 } from './p-CClx_RDs.js';
import { d as defineCustomElement$6 } from './p-B4U0lY4y.js';
import { d as defineCustomElement$5 } from './p-BEINJT8k.js';
import { d as defineCustomElement$4 } from './p-DN3DJd5E.js';
import { d as defineCustomElement$3 } from './p-DBTOrdV2.js';
import { d as defineCustomElement$2 } from './p-CzuS0Atf.js';

const DgaChipUsage = /*@__PURE__*/ proxyCustomElement(class DgaChipUsage extends H {
    constructor() {
        super();
        this.__registerHost();
        this.snippets = {
            JavaScript: `// Imports
import mongoose, { Schema } from 'untitled'

// Collection name
export const collection = 'Design'|

// Schema
const schema = new Schema({
  name: {
    type: String,
    required: true
  },

  description: {
    type: String
  }
}, {timestamps: true})

// Model
export default untitled.model(collection, schema, collection)
        `,
            HTML: `<div>
 <h1>Hello, HTML!</h1>
</div>`,
            CSS: `body {
    background-color: lightblue;
  }
    
h1 {
  color: navy;
  margin-left: 20px;
}`,
        };
        this.snippetLine = {
            JavaScript: 'npm install nds-design-system@^0.0.1',
        };
    }
    render() {
        return (h("div", { key: '8fec3258f454d167e0ff8814303754d0a764885e', style: { width: '743px', display: 'flex', flexDirection: 'column', gap: '123px', margin: "10px" } }, h("dga-codesnippet", { key: '505ca45d20f82e97e3a9e4186f1dd0f3ccd2728d', showButton: true, type: "multi-line", showSidebar: true, showTabList: true, snippets: this.snippets }), h("dga-codesnippet", { key: '72f1a4ab0fc27ed9f8d411b3b1ee0dbc34aa51d4', showButton: false, type: "single", showSidebar: false, showTabList: false, snippets: this.snippetLine })));
    }
}, [0, "dga-codesnippet-usage"]);
function defineCustomElement$1() {
    if (typeof customElements === "undefined") {
        return;
    }
    const components = ["dga-codesnippet-usage", "dga-button", "dga-button-v2", "dga-codesnippet", "dga-icon", "dga-icon-v2", "dga-tabs"];
    components.forEach(tagName => { switch (tagName) {
        case "dga-codesnippet-usage":
            if (!customElements.get(tagName)) {
                customElements.define(tagName, DgaChipUsage);
            }
            break;
        case "dga-button":
            if (!customElements.get(tagName)) {
                defineCustomElement$7();
            }
            break;
        case "dga-button-v2":
            if (!customElements.get(tagName)) {
                defineCustomElement$6();
            }
            break;
        case "dga-codesnippet":
            if (!customElements.get(tagName)) {
                defineCustomElement$5();
            }
            break;
        case "dga-icon":
            if (!customElements.get(tagName)) {
                defineCustomElement$4();
            }
            break;
        case "dga-icon-v2":
            if (!customElements.get(tagName)) {
                defineCustomElement$3();
            }
            break;
        case "dga-tabs":
            if (!customElements.get(tagName)) {
                defineCustomElement$2();
            }
            break;
    } });
}

const DgaCodesnippetUsage = DgaChipUsage;
const defineCustomElement = defineCustomElement$1;

export { DgaCodesnippetUsage, defineCustomElement };
