// @platformscode/core/dist/components/dga-progress-indicator-usage.js
import { p as proxyCustomElement, H, h } from './p-BhqVmlhr.js';
import { d as defineCustomElement$3 } from './p-CBFY-ks-.js';
import { d as defineCustomElement$2 } from './p-pwESyQzO.js';

const DgaProgressIndicatorUsage$1 = /*@__PURE__*/ proxyCustomElement(class DgaProgressIndicatorUsage extends H {
    constructor() {
        super();
        this.__registerHost();
        this.activeStep = 1;
        this.activeStep2 = 1;
    }
    render() {
        console.log(this.activeStep);
        return (h("div", { key: '106fccf5a9376be9fb0b47692373af84ce09cdea' }, h("h1", { key: '44dbe154c248ec8c3a35ee7f60c31b9665159a19' }, "Progress Indicator Example"), h("div", { key: '3ac549dc760eddaaf216e399dc32a068d2ae6bfc', style: { padding: '50px', display: 'flex', gap: '20px' } }, h("div", { key: '15aec93f75ac5fdeae35c1fc60b764c08d36e9e5', style: { display: 'flex', flexDirection: 'column', gap: '20px' } }, h("button", { key: '4aeb40f7c59c99fdef2356aebc37bd0ba8bdb3b8', onClick: () => (this.activeStep = this.activeStep - 1) }, "prev"), h("button", { key: '2f69413030457e7c9c192efb310ee0fc8b278934', onClick: () => (this.activeStep = this.activeStep + 1) }, "next"), h("button", { key: 'bd617089fd8e529ec72832ba1bc9083a15f3bc6e', onClick: () => (this.activeStep2 = this.activeStep2 - 1) }, "prev 2"), h("button", { key: '732d540938f55034f8a48086ddd61437ef4b18c5', onClick: () => (this.activeStep2 = this.activeStep2 + 1) }, "next 2"), "activeStep: ", this.activeStep, h("dga-progress-indicator", { key: 'f3d8312f2d033eb01bd7d03803e481aa4c6703d9', activeStep: this.activeStep, alignment: 'vertical' }, h("dga-progress-indicator-step", { key: '8317a36870b00c58373111bb039702b4a18d6e58', title: "Step One", description: "this is step description", currentStep: 1 }), h("dga-progress-indicator-step", { key: 'a568405aab73427ee03b42e582d575dc65c377bb', title: "Step Two", description: "this is step description", currentStep: 2 }), h("dga-progress-indicator-step", { key: '9221f4bb855dfcc8282144b5cf1e41115562505d', title: "Step Three", description: "this is step description", nextStep: false, currentStep: 3 })), h("dga-progress-indicator", { key: '2c69b186ecac0feb2f7ff0073b7b44d43ef48dbc', activeStep: this.activeStep2, alignment: 'vertical' }, h("dga-progress-indicator-step", { key: 'b423c50761562581d6cdde4b091af150a6458652', title: "Step One", description: "this is step description", currentStep: 1 }), h("dga-progress-indicator-step", { key: '679018080064e782a8647ea5281bffa547b84e3a', title: "Step Two", description: "this is step description", currentStep: 2 }), h("dga-progress-indicator-step", { key: 'c1745bbe4133c654a38936f510701262620db106', title: "Step Three", description: "this is step description", nextStep: false, currentStep: 3 }))), h("div", { key: '0ae88c79be9361d7ed9b72dc1a31a339e6f9834f', style: { display: 'flex', flexDirection: 'column', gap: '20px' } }))));
    }
}, [0, "dga-progress-indicator-usage", {
        "activeStep": [32],
        "activeStep2": [32]
    }]);
function defineCustomElement$1() {
    if (typeof customElements === "undefined") {
        return;
    }
    const components = ["dga-progress-indicator-usage", "dga-progress-indicator", "dga-progress-indicator-step"];
    components.forEach(tagName => { switch (tagName) {
        case "dga-progress-indicator-usage":
            if (!customElements.get(tagName)) {
                customElements.define(tagName, DgaProgressIndicatorUsage$1);
            }
            break;
        case "dga-progress-indicator":
            if (!customElements.get(tagName)) {
                defineCustomElement$3();
            }
            break;
        case "dga-progress-indicator-step":
            if (!customElements.get(tagName)) {
                defineCustomElement$2();
            }
            break;
    } });
}

const DgaProgressIndicatorUsage = DgaProgressIndicatorUsage$1;
const defineCustomElement = defineCustomElement$1;

export { DgaProgressIndicatorUsage, defineCustomElement };
