// @platformscode/core/dist/components/dga-table-usage.js
import { p as proxyCustomElement, H, h } from './p-BhqVmlhr.js';
import { d as defineCustomElement$5 } from './p-ChhNw26D.js';
import { d as defineCustomElement$4 } from './p-CHdhCdsH.js';
import { d as defineCustomElement$3 } from './p-DN3DJd5E.js';
import { d as defineCustomElement$2 } from './p-DBTOrdV2.js';

const DgaTableUsage$1 = /*@__PURE__*/ proxyCustomElement(class DgaTableUsage extends H {
    constructor() {
        super();
        this.__registerHost();
    }
    render() {
        return (h("div", { key: 'b5054a6d0fd3e7dae1e12c3b4cecea2ae94e16a6' }, h("h1", { key: 'ee2a65396fbbf71469deabe03fbfdd9ec16d80d2' }, "Table Example"), h("div", { key: '108b830e26a740d0d73e8a2ba0206f9848d92e63', style: { padding: '50px', display: 'flex', gap: '20px' } }, h("div", { key: 'b452499d51152fe62c37b6cb9c2ab539444451db', style: { display: 'flex', flexDirection: 'column', gap: '20px' } }, h("dga-data-table", { key: '8cfded33a0e06c92fb33cea795a028d1ceb30349', columns: [
                {
                    label: 'Header',
                },
            ], cells: [
                {
                    title: 'city',
                    sort: 'up',
                    isSort: true,
                    isFilter: true,
                    type: 'propertyName',
                    propertyName: 'city',
                    colSpan: 1, // Set colSpan to 1 (default is 1)
                },
                {
                    title: 'price',
                    sort: 'down',
                    isSort: true,
                    isFilter: true,
                    type: 'propertyName',
                    propertyName: 'price',
                    colSpan: 1,
                },
                {
                    title: 'state',
                    isSort: true,
                    isFilter: true,
                    type: 'propertyName',
                    propertyName: 'state',
                    colSpan: 1,
                },
            ], data: [
                {
                    h_id: '3',
                    city: 'Dallas',
                    state: 'TX',
                    zip: '75201',
                    price: '162500',
                    selected: false, // Add selected property for checkbox functionality
                },
                {
                    h_id: '4',
                    city: 'Bevery Hills',
                    state: 'CA',
                    zip: '90210',
                    price: '319250',
                    selected: false,
                },
                {
                    h_id: '5',
                    city: 'New York',
                    state: 'NY',
                    zip: '00010',
                    price: '962500',
                    selected: false,
                },
            ] })), h("div", { key: '441a9cf2d8ad1afa33d6f857da4dba1e3e3e4b79', style: { display: 'flex', flexDirection: 'column', gap: '20px' } }))));
    }
}, [0, "dga-table-usage"]);
function defineCustomElement$1() {
    if (typeof customElements === "undefined") {
        return;
    }
    const components = ["dga-table-usage", "dga-checkbox", "dga-data-table", "dga-icon", "dga-icon-v2"];
    components.forEach(tagName => { switch (tagName) {
        case "dga-table-usage":
            if (!customElements.get(tagName)) {
                customElements.define(tagName, DgaTableUsage$1);
            }
            break;
        case "dga-checkbox":
            if (!customElements.get(tagName)) {
                defineCustomElement$5();
            }
            break;
        case "dga-data-table":
            if (!customElements.get(tagName)) {
                defineCustomElement$4();
            }
            break;
        case "dga-icon":
            if (!customElements.get(tagName)) {
                defineCustomElement$3();
            }
            break;
        case "dga-icon-v2":
            if (!customElements.get(tagName)) {
                defineCustomElement$2();
            }
            break;
    } });
}

const DgaTableUsage = DgaTableUsage$1;
const defineCustomElement = defineCustomElement$1;

export { DgaTableUsage, defineCustomElement };
