// @platformscode/core/dist/components/dga-checkbox-usage.js
import { p as proxyCustomElement, H, h } from './p-BhqVmlhr.js';
import { d as defineCustomElement$4 } from './p-ChhNw26D.js';
import { d as defineCustomElement$3 } from './p-DN3DJd5E.js';
import { d as defineCustomElement$2 } from './p-DBTOrdV2.js';

const DgaCheckboxUsage$1 = /*@__PURE__*/ proxyCustomElement(class DgaCheckboxUsage extends H {
    constructor() {
        super();
        this.__registerHost();
        this.variant = 'neutral';
    }
    render() {
        return (h("div", { key: '657e02ca429e61dc3c69f03ca4713f7f490acff4' }, h("h1", { key: '009b790c4ebd2dfd57f9417644167e4a12731811' }, "Checkbox Example"), h("div", { key: 'a22382e8088dd1af150d283c0dd7fae316249b5d', style: { padding: '50px', display: 'flex', gap: '20px' } }, h("div", { key: 'c26ee773857a7c6d50b8606147b57b363c8f0cc7', style: { display: 'flex', flexDirection: 'column', gap: '20px' } }, h("h1", { key: 'ecef81a3af106be510bd07c6c6b146a8956f6c44' }, "test"), h("dga-checkbox", { key: '305ae9eaa7437668131dac41c92bd25f33776099', label: 'Checkbox Label Checkbox Label Checkbox Label Checkbox Label', Indeterminate: false, alertMessage: true, alertText: "Error/Warning message" }), h("dga-checkbox", { key: 'd0f555bad89339e225972cfa8114faea6fcff33c', size: 'lg' }), h("dga-checkbox", { key: '1495f1ee81452c4497650f29ed6f0b2e1900ed7e', size: 'md' }), h("dga-checkbox", { key: '4c79ac1b264bf1645d534644b35c15c22e72fdb6', size: 'sm' })), h("div", { key: '38f92ac5f0d0a78b5d6cb880c5a2e01efd750360', style: { display: 'flex', flexDirection: 'column', gap: '20px' } }))));
    }
}, [0, "dga-checkbox-usage", {
        "variant": [32]
    }]);
function defineCustomElement$1() {
    if (typeof customElements === "undefined") {
        return;
    }
    const components = ["dga-checkbox-usage", "dga-checkbox", "dga-icon", "dga-icon-v2"];
    components.forEach(tagName => { switch (tagName) {
        case "dga-checkbox-usage":
            if (!customElements.get(tagName)) {
                customElements.define(tagName, DgaCheckboxUsage$1);
            }
            break;
        case "dga-checkbox":
            if (!customElements.get(tagName)) {
                defineCustomElement$4();
            }
            break;
        case "dga-icon":
            if (!customElements.get(tagName)) {
                defineCustomElement$3();
            }
            break;
        case "dga-icon-v2":
            if (!customElements.get(tagName)) {
                defineCustomElement$2();
            }
            break;
    } });
}

const DgaCheckboxUsage = DgaCheckboxUsage$1;
const defineCustomElement = defineCustomElement$1;

export { DgaCheckboxUsage, defineCustomElement };
