// @platformscode/core/dist/components/dga-progress-indicator-step-v3.js
import { p as proxyCustomElement, H, h } from './p-BhqVmlhr.js';

const DgaProgressIndicatorStep = /*@__PURE__*/ proxyCustomElement(class DgaProgressIndicatorStep extends H {
    constructor() {
        super();
        this.__registerHost();
        this.currentStep = 1;
        this.title = '';
        this.description = '';
        this.activeStep = 0;
        this.nextStep = true;
        this.showStepName = true;
        this.showStepDescription = true;
        this.internalActiveStep = 1;
        this.isActive = false;
        this.labelState = 'upcoming';
        this.labelStyle = 'circle';
        this.handleFocus = () => {
            this.stepContentRef.classList.add('focus');
        };
        this.handleFocusOut = () => {
            this.stepContentRef.classList.remove('focus');
        };
        this.handleEscape = (event) => {
            if (event.key === 'Escape') {
                this.handleFocusOut();
                const inputElement = this.el.querySelector('input');
                if (inputElement) {
                    inputElement.blur();
                }
            }
        };
    }
    componentWillLoad() {
        this.updateState();
        window.addEventListener('keydown', this.handleEscape);
    }
    disconnectedCallback() {
        window.removeEventListener('keydown', this.handleEscape);
    }
    stepChangeHandler(event) {
        this.internalActiveStep = event.detail;
        console.log('stepChangeHandler', event.detail);
        this.isActive = event.detail > this.currentStep;
        if (event.detail > this.currentStep) {
            this.labelState = 'completed';
        }
        else if (event.detail < this.currentStep) {
            this.labelState = 'upcoming';
        }
        else {
            this.labelState = 'current';
        }
    }
    // @Watch('internalActiveStep')
    updateState() {
        this.isActive = this.internalActiveStep > this.currentStep;
        if (this.internalActiveStep > this.currentStep) {
            this.labelState = 'completed';
        }
        else if (this.internalActiveStep < this.currentStep) {
            this.labelState = 'upcoming';
        }
        else {
            this.labelState = 'current';
        }
    }
    labelChildren(currentStep) {
        if (this.labelState === 'completed') {
            return h("i", { class: "hgi-stroke hgi-tick-02", style: { fontSize: "16px" } });
        }
        return currentStep.toString();
    }
    render() {
        return (h("div", { key: '0b3880284791a4b0714512be41dd0259a40ac9b8', class: "progress-indicator__step", onMouseDown: this.handleFocus, onMouseUp: this.handleFocusOut }, h("div", { key: '5e0eff774e639788c688b796a2248e86c5ca3ad9', class: "progress-indicator__step-content dga-flex-column", ref: el => (this.stepContentRef = el) }, h("div", { key: '3df2da922c9d20601b45047f1a0e981498d07e7d', class: "step-label", "data-state": this.labelState }, h("span", { key: '73bdf3313205c8c9728e326cdefb96f754f80ca4', class: `step-label--${this.labelStyle}` }, this.labelStyle === 'circle' && this.labelChildren(this.currentStep))), h("div", { key: 'eabd41e9d4e8874d1ea652e81b1d6d6957a9128c', class: "step-text dga-flex-column" }, this.showStepName && h("span", { key: 'ff2b1ff6ee3b552ca9015425b47a617f72322d4b', class: this.isActive ? 'text-md-medium' : 'text-md-regular' }, this.title), this.showStepDescription && h("p", { key: 'ce360370fa4906872f30367120692dd18254506c', class: "text-sm-regular" }, this.description))), this.nextStep && h("span", { key: 'c1b14f4e03fb9279a1d45153ae678d126a0a188c', class: `progress-indicator__step-connector ${this.isActive ? 'progress-indicator__step-connector--active' : ''}` })));
    }
    get el() { return this; }
}, [0, "dga-progress-indicator-step-v3", {
        "currentStep": [1538, "current-step"],
        "title": [1],
        "description": [1],
        "activeStep": [2, "active-step"],
        "nextStep": [4, "next-step"],
        "showStepName": [4, "show-step-name"],
        "showStepDescription": [4, "show-step-description"],
        "labelStyle": [1, "label-style"],
        "internalActiveStep": [32],
        "isActive": [32],
        "labelState": [32]
    }, [[4, "stepChange", "stepChangeHandler"]]]);
function defineCustomElement$1() {
    if (typeof customElements === "undefined") {
        return;
    }
    const components = ["dga-progress-indicator-step-v3"];
    components.forEach(tagName => { switch (tagName) {
        case "dga-progress-indicator-step-v3":
            if (!customElements.get(tagName)) {
                customElements.define(tagName, DgaProgressIndicatorStep);
            }
            break;
    } });
}

const DgaProgressIndicatorStepV3 = DgaProgressIndicatorStep;
const defineCustomElement = defineCustomElement$1;

export { DgaProgressIndicatorStepV3, defineCustomElement };
