// @platformscode/core/dist/components/nds-floating-button-usage.js
import { p as proxyCustomElement, H, h } from './p-BhqVmlhr.js';
import { d as defineCustomElement$4 } from './p-CCimZGO2.js';
import { d as defineCustomElement$3 } from './p-DBTOrdV2.js';
import { d as defineCustomElement$2 } from './p-BGoCiut_.js';

const NdsFloatingButtonUsage$1 = /*@__PURE__*/ proxyCustomElement(class NdsFloatingButtonUsage extends H {
    constructor() {
        super();
        this.__registerHost();
    }
    render() {
        return (h("div", { key: '1077ca0b8a1c641f129a560f2fab077eb071e526' }, h("dga-flex", { key: '0e4fc0f68a48a29cb51304cc7c335c081dd5f8b5', direction: "column", spacing: 16 }, h("nds-floating-button", { key: '51eaecdc1b26b083b1c31e0315fc69a9f420c987', variant: "primary-neutral", label: 'Button', iconOnly: true, icon: 'plus-sign' }), h("nds-floating-button", { key: '95c96244a8e4d6c18bb8df8b34d4b9fe0bb266dc', variant: "primary-neutral", label: 'Button' }), h("nds-floating-button", { key: 'b4046ca70356e47f946d7035ff4572e5b68094fe', variant: "primary-brand", label: 'Button', iconOnly: true, icon: 'plus-sign' }), h("nds-floating-button", { key: '6850de062f4d3b57a80545aaaa81a33a57ae9532', variant: "primary-brand", label: 'Button' }), h("nds-floating-button", { key: '8923b990baeb8b96770324028108f07c8ba9df65', variant: "secondary-solid", label: 'Button', iconOnly: true, icon: 'plus-sign' }), h("nds-floating-button", { key: '5c847753cce63f00a72664dd26951a5bdcbfa087', variant: "secondary-solid", label: 'Button' })), h("div", { key: '977fb56e1b173eb335808d3279938154befd1053', style: { backgroundColor: '#074D31', padding: '20px', marginTop: '20px' } }, h("dga-flex", { key: '2f29434c6a9720731bf50c4b35b0b1c24d95ac4f', direction: "column", spacing: 16 }, h("nds-floating-button", { key: 'aae7230aed0efb57cacd5ed44c60924533f95b2d', variant: "primary-neutral", label: 'Button', iconOnly: true, icon: 'plus-sign', inverted: true }), h("nds-floating-button", { key: 'd350ac9cc4bf197528def370259bb9a09b51997b', variant: "primary-neutral", label: 'Button', inverted: true }), h("nds-floating-button", { key: '0d553f7e05ba667ca27139eef21c22c48362b0ec', variant: "primary-brand", label: 'Button', iconOnly: true, icon: 'plus-sign', inverted: true }), h("nds-floating-button", { key: 'de24454901b2d9f29dc034a9cadba10afef6e294', variant: "primary-brand", label: 'Button', inverted: true }), h("nds-floating-button", { key: '333514f4e6297be77e73f7725fcbae174cf379e0', variant: "secondary-solid", label: 'Button', iconOnly: true, icon: 'plus-sign', inverted: true }), h("nds-floating-button", { key: '001973bd2d981a3c6da7db63094f93d9aac909bb', variant: "secondary-solid", label: 'Button', inverted: true })))));
    }
}, [0, "nds-floating-button-usage"]);
function defineCustomElement$1() {
    if (typeof customElements === "undefined") {
        return;
    }
    const components = ["nds-floating-button-usage", "dga-flex", "dga-icon-v2", "nds-floating-button"];
    components.forEach(tagName => { switch (tagName) {
        case "nds-floating-button-usage":
            if (!customElements.get(tagName)) {
                customElements.define(tagName, NdsFloatingButtonUsage$1);
            }
            break;
        case "dga-flex":
            if (!customElements.get(tagName)) {
                defineCustomElement$4();
            }
            break;
        case "dga-icon-v2":
            if (!customElements.get(tagName)) {
                defineCustomElement$3();
            }
            break;
        case "nds-floating-button":
            if (!customElements.get(tagName)) {
                defineCustomElement$2();
            }
            break;
    } });
}

const NdsFloatingButtonUsage = NdsFloatingButtonUsage$1;
const defineCustomElement = defineCustomElement$1;

export { NdsFloatingButtonUsage, defineCustomElement };
