// @platformscode/core/dist/components/nds-button-usage.js
import { p as proxyCustomElement, H, h } from './p-BhqVmlhr.js';
import { d as defineCustomElement$6 } from './p-t2gkYg4G.js';
import { d as defineCustomElement$5 } from './p-B4U0lY4y.js';
import { d as defineCustomElement$4 } from './p-CCimZGO2.js';
import { d as defineCustomElement$3 } from './p-DBTOrdV2.js';
import { d as defineCustomElement$2 } from './p-Ck6ZXrKN.js';

const NdsButtonUsage$1 = /*@__PURE__*/ proxyCustomElement(class NdsButtonUsage extends H {
    constructor() {
        super();
        this.__registerHost();
        this.variant = 'neutral';
    }
    render() {
        return (h("div", { key: '56a7ef337ff09b7f1c3a08187c21b7f3aa7cb96c' }, h("h1", { key: 'ca9717cef5798d56c3eaad850e0501cb5cc87215' }, "Button Example"), h("div", { key: '7c7f2bb18c37b9878aa655b100a54a363bd650a3', style: { padding: '50px', display: 'flex', gap: '20px' } }, h("div", { key: '5431963690e4029f2b5fe10b7e2e248bf64ad86d', style: { display: 'flex', flexDirection: 'column', gap: '20px' } }, h("dga-flex", { key: 'b077dbe4db68b0bff9c1f7ff080972049b766cc4', direction: "column", spacing: 16 }, h("dga-button-v2", { key: '6ac388d692be98e218994016f44fdc53664e7ed8', label: "Button", variant: "primary", size: "lg" }), h("dga-button-v2", { key: '4e54e2f6413fd2b50cbcd9f2d10d9a797fcf8c0c', label: "Button", variant: "primary", size: "md" }), h("dga-button-v2", { key: 'd4728654f620ca8351993a2a9a185126ca9dee7d', label: "Button", variant: "primary", size: "sm" }), h("dga-button-v2", { key: 'c85d8145164da44e08d2a5498435065aa24c1546', label: "Button", variant: "neutral", size: "lg" }), h("dga-button-v2", { key: 'b0a20e61bbec22c48744a962153ee39046e0bbd7', label: "Button", variant: "neutral", size: "md" }), h("dga-button-v2", { key: '5e0838ef9974ec6342e199662a9d5671c41c0b2c', label: "Button", variant: "neutral", size: "sm" }), h("dga-button-v2", { key: 'd4e1dbeb76b97149e7a4cc5ebc976a21f47260bc', label: "Button", variant: "secondary-solid", size: "lg" }), h("dga-button-v2", { key: '5e708a54fb4762e6aec54699aecd1a3b5172919e', label: "Button", variant: "secondary-solid", size: "md" }), h("dga-button-v2", { key: 'c3e7bcf88f0c14a4df19ef5d0c074f53c17ad6b8', label: "Button", variant: "secondary-solid", size: "sm" }), h("dga-button-v2", { key: '6b295cea662acd66cdf58242b91758d620618f6d', label: "Button", variant: "primary", destructive: true, size: "lg" }), h("dga-button-v2", { key: '10f9427040f39a44f1be10342a8e7cefd5210ecd', label: "Button", variant: "primary", destructive: true, size: "md" }), h("dga-button-v2", { key: '3c534d208fd76cff93b545c2eed0054a386eba54', label: "Button", variant: "primary", destructive: true, size: "sm" })), h("dga-button-v2", { key: '82e31140b806cfc72557dce6027505ba63798508', label: "Button", variant: "primary", onColor: true }, h("dga-button-leading-icon", { key: '2ba45b4394c6059122515b0b7d75c21436f9342f', name: "link-square-01", size: 16, color: "red" })), h("dga-button-v2", { key: '8392c0b038dadd80540f3a649816ac7492b57a1f', label: "Source code", variant: "secondary-outline", leadIcon: true, leadIconProps: { name: 'source-code', variant: 'stroke', type: 'standard' } }), h("dga-button-v2", { key: '0690b4bf8b5a6756ed23e5399a0f29e8d60bd930',
            // className="absolute top-[50%] left-[50%] transform -translate-x-1/2 -translate-y-1/2 opacity-0 group-hover:opacity-100"
            label: "See live preview", size: "lg", variant: "secondary-outline", trailIcon: true, trailIconProps: {
                name: 'link-square-01',
                variant: 'stroke',
                type: 'standard',
            } })), h("div", { key: 'a30a63267b64c21d04f0dfdc1ada763c2bfe06fe', style: { display: 'flex', flexDirection: 'column', gap: '20px', backgroundColor: '#074D31', padding: '20px' } }, h("nds-button", { key: 'c0b8969ae1b54dfa538477e74d6010df4a2f298d', label: "Button", variant: "primary-brand", size: "lg" }), h("nds-button", { key: '14adca061189dc8ecaaf3ab6c094a17577c5b0dd', label: "Button", variant: "primary-brand", size: "lg", selected: true }), h("nds-button", { key: 'd4b933594e9225058ad996f0fe626f3c1e084071', leadingIcon: 'link-square-01', size: 'sm' }, h("dga-icon-v2", { key: 'c16f42b651dcfd75e5435fe45114eac0c7486917', slot: 'leading-icon', name: 'source-code', variant: 'stroke', type: 'rounded', size: 16, color: 'white' }))), h("dga-icon-v2", { key: '77517d27e1b0b26631c90362d57a45ca8d9eb35f', name: 'link-square-01', variant: 'stroke', type: 'rounded', color: 'red' }), h("div", { key: '14214c0e1f68c0e5a43ac307c832a234619c5b21', style: { display: 'flex', flexDirection: 'column', gap: '20px' } }))));
    }
}, [0, "nds-button-usage", {
        "variant": [32]
    }]);
function defineCustomElement$1() {
    if (typeof customElements === "undefined") {
        return;
    }
    const components = ["nds-button-usage", "dga-button-leading-icon", "dga-button-v2", "dga-flex", "dga-icon-v2", "nds-button"];
    components.forEach(tagName => { switch (tagName) {
        case "nds-button-usage":
            if (!customElements.get(tagName)) {
                customElements.define(tagName, NdsButtonUsage$1);
            }
            break;
        case "dga-button-leading-icon":
            if (!customElements.get(tagName)) {
                defineCustomElement$6();
            }
            break;
        case "dga-button-v2":
            if (!customElements.get(tagName)) {
                defineCustomElement$5();
            }
            break;
        case "dga-flex":
            if (!customElements.get(tagName)) {
                defineCustomElement$4();
            }
            break;
        case "dga-icon-v2":
            if (!customElements.get(tagName)) {
                defineCustomElement$3();
            }
            break;
        case "nds-button":
            if (!customElements.get(tagName)) {
                defineCustomElement$2();
            }
            break;
    } });
}

const NdsButtonUsage = NdsButtonUsage$1;
const defineCustomElement = defineCustomElement$1;

export { NdsButtonUsage, defineCustomElement };
