// @platformscode/core/dist/components/nds-link-usage.js
import { p as proxyCustomElement, H, h } from './p-BhqVmlhr.js';
import { d as defineCustomElement$5 } from './p-CCimZGO2.js';
import { d as defineCustomElement$4 } from './p-DBTOrdV2.js';
import { d as defineCustomElement$3 } from './p-Ck6ZXrKN.js';
import { d as defineCustomElement$2 } from './p-BRb_1tzT.js';

const NdsLinkUsage$1 = /*@__PURE__*/ proxyCustomElement(class NdsLinkUsage extends H {
    constructor() {
        super();
        this.__registerHost();
    }
    render() {
        return (h("div", { key: 'a27f7945ef4a777419bc07c853d55b3dc673b567' }, h("h1", { key: '54a680f61b5d8f9f8be59ecfed0d308ee741708e' }, "Button Example"), h("div", { key: '5098ca194a84accf70fe8d626d2ec9dc1e6112ed', style: { padding: '50px', display: 'flex', gap: '20px' } }, h("div", { key: '6abea75bf4980cc52c85a2eede3fd8d8f511c743', style: { display: 'flex', flexDirection: 'column', gap: '20px' } }, h("dga-flex", { key: 'e726ab8464b1edc4cb9193d6d310332d1cd936df', direction: "column", spacing: 16 }, h("nds-link", { key: '2650e63344a480cc03fbf28def4a4b8c9dc3a7ba', icon: "link-04" }), h("nds-link", { key: 'c14d8ab5bae8c36f8c1801586749d258c32610af', icon: "link-04", label: 'link from props' }), h("nds-link", { key: '9101d2ca20d6aaac7719ef13abf634fefe3894b4', icon: "link-04", label: 'link from props' }, "link from child"), h("nds-link", { key: 'd1891080e69454d7f213fc2d06f6633959dddffb', variant: "primary", href: "https://www.google.com", target: "_blank" }, "Primary Link"))), h("div", { key: 'c5b287c34cf264f1be7543aa44833b481443a54e', style: { display: 'flex', flexDirection: 'column', gap: '20px', backgroundColor: '#074D31', padding: '20px' } }, h("nds-button", { key: '43875a97796dfecdc5bbd44c8a24ff218c6ee047', label: "Button", variant: "primary-brand", size: "lg" }), h("nds-button", { key: 'a2c45882c55a5d6e84b6e5e69f3bd5618f46c3e4', label: "Button", variant: "primary-brand", size: "lg", selected: true }), h("nds-button", { key: '756c97529b96fdd6849e2964fdfa8bf6a66b11cb', leadingIcon: 'link-square-01', size: 'sm' }, h("dga-icon-v2", { key: 'fb1facdcb4ce219539c512bf0bf4bb5dd6c54099', slot: 'leading-icon', name: 'source-code', variant: 'stroke', type: 'rounded', size: 16, color: 'white' }))), h("dga-icon-v2", { key: 'debb37fcbbeb25cf0cce43a746ce36a8fec75323', name: 'link-square-01', variant: 'stroke', type: 'rounded', color: 'red' }), h("div", { key: '7139bfec3792e7be357204b5eec9dade4e149f71', style: { display: 'flex', flexDirection: 'column', gap: '20px' } }))));
    }
}, [0, "nds-link-usage"]);
function defineCustomElement$1() {
    if (typeof customElements === "undefined") {
        return;
    }
    const components = ["nds-link-usage", "dga-flex", "dga-icon-v2", "nds-button", "nds-link"];
    components.forEach(tagName => { switch (tagName) {
        case "nds-link-usage":
            if (!customElements.get(tagName)) {
                customElements.define(tagName, NdsLinkUsage$1);
            }
            break;
        case "dga-flex":
            if (!customElements.get(tagName)) {
                defineCustomElement$5();
            }
            break;
        case "dga-icon-v2":
            if (!customElements.get(tagName)) {
                defineCustomElement$4();
            }
            break;
        case "nds-button":
            if (!customElements.get(tagName)) {
                defineCustomElement$3();
            }
            break;
        case "nds-link":
            if (!customElements.get(tagName)) {
                defineCustomElement$2();
            }
            break;
    } });
}

const NdsLinkUsage = NdsLinkUsage$1;
const defineCustomElement = defineCustomElement$1;

export { NdsLinkUsage, defineCustomElement };
