// @platformscode/core/dist/components/nds-accordion-usage.js
import { p as proxyCustomElement, H, h } from './p-BhqVmlhr.js';
import { d as defineCustomElement$3 } from './p-DBTOrdV2.js';
import { d as defineCustomElement$2 } from './p-DIfpyTAV.js';

const NdsAccordionUsage$1 = /*@__PURE__*/ proxyCustomElement(class NdsAccordionUsage extends H {
    constructor() {
        super();
        this.__registerHost();
    }
    render() {
        return (h("div", { key: '7c41efaf3aed341b1d61f3832d265124e789d2f7' }, h("h1", { key: '267e710183605616f82ab4a85ca64ebc5f416475' }, "Accordion Example"), h("div", { key: '6127c5318bd6c420b0632fd6c6f7abe635170d62', style: { padding: '50px', display: 'flex', gap: '20px' } }, h("div", { key: 'c9f581d6af746db129fec4b97ea96833f60a292a', style: { display: 'flex', flexDirection: 'column', gap: '20px' } }, h("nds-accordion", { key: 'cde7d942982593a10128fb69e6a0aa2038b7e7f4', title: "Accordion Title", size: "lg", content: "The accordion component delivers large amounts of content in a small space through progressive disclosure. The user gets key details about the underlying content and can choose to expand that content within the constraints of the accordion." }), h("nds-accordion", { key: 'eb1948acc44b3d45e42dab520bf69a8e11c565fd', title: "Accordion Title", size: "lg", content: "The accordion component delivers large amounts of content in a small space through progressive disclosure. The user gets key details about the underlying content and can choose to expand that content within the constraints of the accordion." }, h("div", { key: '3e9d58012f444df3c5a31c3ddf7c00036a87d3a0' }, h("p", { key: '78b781b055379142cd273ed837e018a25402f2a0' }, "This is a ", h("span", { key: '8b253bec617aa4494e0a60241502d101ba888ebc', style: { color: "red" } }, "custom"), " content inside the accordion."), h("p", { key: 'be4d5da6b8f8b38eada62265f8b5c552ee0e8e29' }, "You can put any React nodes here.")))))));
    }
}, [0, "nds-accordion-usage"]);
function defineCustomElement$1() {
    if (typeof customElements === "undefined") {
        return;
    }
    const components = ["nds-accordion-usage", "dga-icon-v2", "nds-accordion"];
    components.forEach(tagName => { switch (tagName) {
        case "nds-accordion-usage":
            if (!customElements.get(tagName)) {
                customElements.define(tagName, NdsAccordionUsage$1);
            }
            break;
        case "dga-icon-v2":
            if (!customElements.get(tagName)) {
                defineCustomElement$3();
            }
            break;
        case "nds-accordion":
            if (!customElements.get(tagName)) {
                defineCustomElement$2();
            }
            break;
    } });
}

const NdsAccordionUsage = NdsAccordionUsage$1;
const defineCustomElement = defineCustomElement$1;

export { NdsAccordionUsage, defineCustomElement };
