// @platformscode/core/dist/components/p-BDegpq_a.js
import { p as proxyCustomElement, H, h } from './p-BhqVmlhr.js';
import { d as defineCustomElement$1 } from './p-DN3DJd5E.js';

const DgaHelperText = /*@__PURE__*/ proxyCustomElement(class DgaHelperText extends H {
    constructor() {
        super();
        this.__registerHost();
        this.helperText = '';
        this.error = false;
    }
    render() {
        var _a;
        return (h("div", { key: '9ce77fa2f456e20064b153aef3c1fa436563eb49', class: `dga-helper-text ${this.error ? "dga-helper-text--error" : ""}` }, h("span", { key: '07990ee96959b34986170620b219194844e30970', class: "dga-helper-text__icon" }, this.helperTextIcon && ((_a = this.helperTextIcon) === null || _a === void 0 ? void 0 : _a.name) && h("dga-icon", { key: '702f2185be7544b679b543a063f0c76049cbe7e3', icon: this.helperTextIcon.name, variant: this.helperTextIcon.variant, type: this.helperTextIcon.type, size: 16 })), h("span", { key: '5fb2b0ceacdda88772a4ade8ac9e94a514e79a22', class: "dga-helper-text__desc" }, this.helperText)));
    }
}, [0, "dga-helper-text", {
        "helperTextIcon": [16, "helper-text-icon"],
        "helperText": [1, "helper-text"],
        "error": [4]
    }]);
function defineCustomElement() {
    if (typeof customElements === "undefined") {
        return;
    }
    const components = ["dga-helper-text", "dga-icon"];
    components.forEach(tagName => { switch (tagName) {
        case "dga-helper-text":
            if (!customElements.get(tagName)) {
                customElements.define(tagName, DgaHelperText);
            }
            break;
        case "dga-icon":
            if (!customElements.get(tagName)) {
                defineCustomElement$1();
            }
            break;
    } });
}

export { DgaHelperText as D, defineCustomElement as d };
