// @platformscode/core/dist/components/p-BRUw79oz.js
import { p as proxyCustomElement, H, h, d as Host } from './p-BhqVmlhr.js';

const DgaSecondNavHeaderContent = /*@__PURE__*/ proxyCustomElement(class DgaSecondNavHeaderContent extends H {
    constructor() {
        super();
        this.__registerHost();
    }
    render() {
        return (h(Host, { key: 'd9b732973272bc20a9407bfb291293ff8936a792' }, h("div", { key: 'dce382719b0b96c984dee57dd4b054325ff4834d', class: `items-container` }, h("slot", { key: '851ec4474a4630d627326c6daf4da23f635fd268' }))));
    }
}, [4, "dga-second-nav-header-content"]);
function defineCustomElement() {
    if (typeof customElements === "undefined") {
        return;
    }
    const components = ["dga-second-nav-header-content"];
    components.forEach(tagName => { switch (tagName) {
        case "dga-second-nav-header-content":
            if (!customElements.get(tagName)) {
                customElements.define(tagName, DgaSecondNavHeaderContent);
            }
            break;
    } });
}

export { DgaSecondNavHeaderContent as D, defineCustomElement as d };
