// @platformscode/core/dist/components/p-BRtz_FTw.js
import { p as proxyCustomElement, H, h, d as Host } from './p-BhqVmlhr.js';

const DgaSecondNavHeader = /*@__PURE__*/ proxyCustomElement(class DgaSecondNavHeader extends H {
    constructor() {
        super();
        this.__registerHost();
        this.variant = 'primary';
    }
    componentWillLoad() {
        this.el.childNodes.forEach((item) => {
            const el = item;
            if (el.tagName === 'DGA-SECOND-NAV-HEADER-CONTENT' && this.hideContent === true) {
                el.style.display = 'none';
            }
            if (el.tagName === 'DGA-SECOND-NAV-HEADER-ACTIONS' && this.hideActions === true) {
                el.style.display = 'none';
            }
            if (el.tagName === 'DGA-SECOND-NAV-HEADER-CONTENT' && el.attributes.getNamedItem('slot') === null) {
                el.setAttribute('slot', 'content');
            }
            if (el.tagName === 'DGA-SECOND-NAV-HEADER-ACTIONS' && el.attributes.getNamedItem('slot') === null) {
                el.setAttribute('slot', 'actions');
            }
        });
    }
    render() {
        return (h(Host, { key: '4d6a56396f788b3f018b3774c84fb6bc0d0849a9' }, h("div", { key: 'f6c6cf9410c08dfdab2c9889852649634b763dbe', class: `wrapper ${this.variant} `, "aria-label": 'a secondary navigation navbar' }, h("div", { key: '4b851ce02ce31b959365e7a09374097d369973ac', class: `container ${this.hideDivider ? '' : 'border'}` }, h("div", { key: '6214ce6582acd5ac51e252425546acd6a5edad46', class: `content` }, h("slot", { key: '3a70b02c824ba50ca99427169697341dc781c5d5', name: "content" })), h("div", { key: '2193cb09f2bd21e7741460d91ab3ab38c723370b', class: `actions` }, h("slot", { key: 'fa36abcb4f2e101b986b746af0dad2af3fee7fa4', name: "actions" }))))));
    }
    get el() { return this; }
}, [6, "dga-second-nav-header", {
        "variant": [1],
        "hideDivider": [1028, "hide-divider"],
        "hideActions": [4, "hide-actions"],
        "hideContent": [4, "hide-content"]
    }]);
function defineCustomElement() {
    if (typeof customElements === "undefined") {
        return;
    }
    const components = ["dga-second-nav-header"];
    components.forEach(tagName => { switch (tagName) {
        case "dga-second-nav-header":
            if (!customElements.get(tagName)) {
                customElements.define(tagName, DgaSecondNavHeader);
            }
            break;
    } });
}

export { DgaSecondNavHeader as D, defineCustomElement as d };
