// @platformscode/core/dist/components/p-C99Dr3jt.js
import { h, p as proxyCustomElement, H, d as Host } from './p-BhqVmlhr.js';
import { g as guid } from './p-DEtoWGOc.js';
import { S as SxStyleManager } from './p-BFEqPAFn.js';
import { d as defineCustomElement$3 } from './p-BDegpq_a.js';
import { d as defineCustomElement$2 } from './p-DN3DJd5E.js';
import { d as defineCustomElement$1 } from './p-Cxhq_uxn.js';

const feedbackIconTypes = {
    success: h("dga-icon", { icon: "CheckmarkCircle02Icon", variant: "solid" }),
    error: h("dga-icon", { icon: "AlertCircleIcon", variant: "solid" }),
    warning: h("dga-icon", { icon: "Alert02Icon", variant: "solid" }),
};
const DgaTextInput = /*@__PURE__*/ proxyCustomElement(class DgaTextInput extends H {
    constructor() {
        super();
        this.__registerHost();
        this.size = 'lg';
        this.type = 'text';
        this.variant = 'default';
        this.error = false;
        this.feedbackIcon = false;
        this.feedbackIconType = 'success';
        this.feedbackIconRing = false;
        this.helperText = '';
        this.disabled = false;
        this.readonly = false;
        this.required = false;
        this.fullwidth = true;
        //   @Prop() id: string = guid();
        this.name = guid();
        this.value = '';
        this.isActive = false;
        this.isFocused = false;
        // Generate a unique ID if not provided
        this.componentId = `dga-button-${Math.random().toString(36).substring(2, 9)}`;
        // Computed inline styles
        this.rootStyles = {};
    }
    valueChanged(newValue, oldValue) {
        if (newValue !== oldValue) {
            this.textInputEl['value'] = newValue;
        }
    }
    handleSxChange() {
        this.processStyling();
    }
    componentWillLoad() {
        // Initialize style manager with element reference
        this.sxStyleManager = new SxStyleManager(this.componentId, this.el);
    }
    componentDidLoad() {
        // Process styling after the component is fully loaded and shadow DOM is ready
        this.processStyling();
    }
    componentDidUpdate() {
        // Re-process styling after updates
        this.processStyling();
    }
    processStyling() {
        if (!this.sx)
            return;
        // Handle both string and object formats
        const sxValue = typeof this.sx === 'string' ? JSON.parse(this.sx) : this.sx;
        this.rootStyles = this.sxStyleManager.processSxProp(sxValue);
    }
    disconnectedCallback() {
        var _a;
        // Clean up styles when component is removed
        (_a = this.sxStyleManager) === null || _a === void 0 ? void 0 : _a.cleanup();
    }
    render() {
        return (h(Host, { key: '0858f48662c1dec58bcf2b35f63501be68c9344a', id: this.componentId, style: this.rootStyles }, h("div", { key: '6e1f1d8717fcc5f8c1ae3da680f21075308d4563', class: { "dga-form-control": true, "dga-form-control--fullwidth": this.fullwidth } }, this.label && h("dga-label", { key: 'd957b562a4e04c7f93be7e2fdf07218e26270b62', label: this.label, size: this.size, required: this.required, disabled: this.disabled }), h("div", { key: '4cff6669f027958bad72f61ea825d3c68d50390b', class: `input input--${this.size} ${this.error ? 'input--error' : ''} ${this.variant !== 'default' ? `input--${this.variant}` : ''} ${this.isActive ? 'active' : ''} ${this.isFocused ? 'focus' : ''} ${this.disabled ? 'input--disabled' : ''} ${this.readonly ? 'readOnly' : ''}`, ref: el => (this.inputContainerRef = el) }, h("slot", { key: '6a72c1f0d8d1c7adf4e1813c0f17d11ee78ba0c7', name: "prefix" }), this.icon && (h("span", { key: '4126803f5157f8baca3ac2b7e31f0cda7434cf06', class: "input__icon" }, h("dga-icon", { key: '0119b859ecf61cb4e8f0a4b4844223646d215607', icon: this.icon.name, variant: this.icon.variant, type: this.icon.type, size: 20 }))), h("input", { key: 'a1011a70fef6d5f31c5e03467f84dbdfb290e34d',
            // id={id}
            ref: (el) => (this.textInputEl = el), value: this.value, name: this.name, type: this.type, onFocus: () => (this.isFocused = true), onBlur: e => {
                this.isFocused = false;
                this.onBlur(e);
            }, class: "input__field", placeholder: this.placeholder, disabled: this.disabled, readOnly: this.readonly, required: this.required, onMouseDown: () => (this.isActive = true), onMouseUp: () => (this.isActive = false), onMouseOut: () => (this.isActive = false), onChange: this.onChange, onInput: this.onInput }), this.error ? (h("span", { class: `input__feedback-icon input__feedback-icon--error ${this.feedbackIconRing ? 'input__feedback-icon--ring' : ''}` })) : this.feedbackIcon ? (h("span", { class: `input__feedback-icon input__feedback-icon--${this.feedbackIconType} ${this.feedbackIconRing ? 'input__feedback-icon--ring' : ''}` }, feedbackIconTypes[this.feedbackIconType])) : null, h("slot", { key: '068446d038a2bf4de9a072898eba807005994cac', name: "suffix" }), " "), this.helperText && h("dga-helper-text", { key: 'cf4e6eb83f1c4534b33b574301f0f4eba46c133b', helperText: this.helperText, helperTextIcon: this.helperTextIcon, error: this.error }))));
    }
    get el() { return this; }
    static get watchers() { return {
        "value": ["valueChanged"],
        "sx": ["handleSxChange"]
    }; }
}, [4, "dga-text-input", {
        "label": [1],
        "placeholder": [1],
        "size": [1],
        "type": [1],
        "variant": [1],
        "error": [4],
        "feedbackIcon": [4, "feedback-icon"],
        "feedbackIconType": [1, "feedback-icon-type"],
        "feedbackIconRing": [4, "feedback-icon-ring"],
        "helperTextIcon": [16, "helper-text-icon"],
        "helperText": [1, "helper-text"],
        "icon": [16],
        "disabled": [4],
        "readonly": [4],
        "required": [4],
        "fullwidth": [4],
        "name": [1],
        "value": [1],
        "onChange": [16, "on-change"],
        "onInput": [16, "on-input"],
        "onBlur": [16, "on-blur"],
        "sx": [1],
        "isActive": [32],
        "isFocused": [32]
    }, undefined, {
        "value": ["valueChanged"],
        "sx": ["handleSxChange"]
    }]);
function defineCustomElement() {
    if (typeof customElements === "undefined") {
        return;
    }
    const components = ["dga-text-input", "dga-helper-text", "dga-icon", "dga-label"];
    components.forEach(tagName => { switch (tagName) {
        case "dga-text-input":
            if (!customElements.get(tagName)) {
                customElements.define(tagName, DgaTextInput);
            }
            break;
        case "dga-helper-text":
            if (!customElements.get(tagName)) {
                defineCustomElement$3();
            }
            break;
        case "dga-icon":
            if (!customElements.get(tagName)) {
                defineCustomElement$2();
            }
            break;
        case "dga-label":
            if (!customElements.get(tagName)) {
                defineCustomElement$1();
            }
            break;
    } });
}

export { DgaTextInput as D, defineCustomElement as d };
