// @platformscode/core/dist/components/p-CD9sOu_w.js
import { p as proxyCustomElement, H, h } from './p-BhqVmlhr.js';
import { g as guid } from './p-DEtoWGOc.js';
import { d as defineCustomElement$5 } from './p-DBTOrdV2.js';
import { d as defineCustomElement$4 } from './p-XcGOP6S8.js';
import { d as defineCustomElement$3 } from './p-mAJwGPJG.js';
import { d as defineCustomElement$2 } from './p-BVvoCy50.js';
import { d as defineCustomElement$1 } from './p-Dua-gbUp.js';

const SidePanelMenu = /*@__PURE__*/ proxyCustomElement(class SidePanelMenu extends H {
    constructor() {
        super();
        this.__registerHost();
        this.routes = [];
    }
    render() {
        return (h("ul", { key: '8b2d550775b065135ab6588510c06543776420cd', class: "sidepanel__menu-list" }, this.routes.map((route, index) => (h("li", { key: guid() }, h("tab-component", { route: route }), index < this.routes.length - 1 && (h("div", { class: "sidepanel__menu-tab-divider" })))))));
    }
}, [0, "side-panel-menu", {
        "routes": [16]
    }]);
function defineCustomElement() {
    if (typeof customElements === "undefined") {
        return;
    }
    const components = ["side-panel-menu", "dga-icon-v2", "dga-tag", "submenu-component", "tab-component", "tab-link"];
    components.forEach(tagName => { switch (tagName) {
        case "side-panel-menu":
            if (!customElements.get(tagName)) {
                customElements.define(tagName, SidePanelMenu);
            }
            break;
        case "dga-icon-v2":
            if (!customElements.get(tagName)) {
                defineCustomElement$5();
            }
            break;
        case "dga-tag":
            if (!customElements.get(tagName)) {
                defineCustomElement$4();
            }
            break;
        case "submenu-component":
            if (!customElements.get(tagName)) {
                defineCustomElement$3();
            }
            break;
        case "tab-component":
            if (!customElements.get(tagName)) {
                defineCustomElement$2();
            }
            break;
        case "tab-link":
            if (!customElements.get(tagName)) {
                defineCustomElement$1();
            }
            break;
    } });
}

export { SidePanelMenu as S, defineCustomElement as d };
