// @platformscode/core/dist/components/p-Cxhq_uxn.js
import { p as proxyCustomElement, H, h } from './p-BhqVmlhr.js';

const DgaLabel = /*@__PURE__*/ proxyCustomElement(class DgaLabel extends H {
    constructor() {
        super();
        this.__registerHost();
        this.label = '';
        this.variant = 'default';
        this.size = 'md';
        this.disabled = false;
        this.required = false;
    }
    render() {
        return (h("label", { key: '4d40b590e3045b65d31807a4a02cbffdd5cb0a76', class: `dga-label dga-label--${this.size} ${this.disabled ? "disabled" : ""}` }, this.required && h("span", { key: '5a5decc5839d7ce035ec25dc84d492cf7e92dc6e', class: "star" }, "*"), this.label));
    }
}, [0, "dga-label", {
        "label": [1],
        "variant": [1],
        "size": [1],
        "disabled": [4],
        "required": [4]
    }]);
function defineCustomElement() {
    if (typeof customElements === "undefined") {
        return;
    }
    const components = ["dga-label"];
    components.forEach(tagName => { switch (tagName) {
        case "dga-label":
            if (!customElements.get(tagName)) {
                customElements.define(tagName, DgaLabel);
            }
            break;
    } });
}

export { DgaLabel as D, defineCustomElement as d };
