// @platformscode/core/dist/components/p-DsbIBJJJ.js
import { p as proxyCustomElement, H, h } from './p-BhqVmlhr.js';

const DgaMetricChart = /*@__PURE__*/ proxyCustomElement(class DgaMetricChart extends H {
    constructor() {
        super();
        this.__registerHost();
        this.type = 'realistic';
        this.trend = 'positive';
        this.size = 'lg';
        this.showMarker = false;
        this.data = [];
        this.lineColor = this.trend === 'positive' ? '#067647' : '#B42318';
        this.fillColor = this.trend === 'positive' ? '#079455' : '#D92D20';
        this.width = this.size === 'lg' ? 328 : 128;
        this.height = 64;
    }
    handleSizeChange() {
        this.width = this.size === 'lg' ? 328 : 128;
    }
    getLinePath() {
        const points = this.getPoints();
        switch (this.type) {
            case 'realistic':
                return this.getSmoothPath(points);
            case 'wavy':
                return this.getWavyPath(points);
            case 'straight':
                return this.getStraightPath(points);
            case 'layers':
                return this.getSmoothPath(points);
            default:
                return '';
        }
    }
    getPoints() {
        return this.data.map((d, i) => ({
            x: (i / (this.data.length - 1)) * this.width,
            y: this.height - (d / Math.max(...this.data)) * this.height
        }));
    }
    getSmoothPath(points) {
        return points.reduce((acc, point, i, arr) => {
            if (i === 0)
                return `M ${point.x},${point.y}`;
            const prev = arr[i - 1];
            const cX = (prev.x + point.x) / 2;
            return `${acc} C ${cX},${prev.y} ${cX},${point.y} ${point.x},${point.y}`;
        }, '');
    }
    getWavyPath(points) {
        return points.reduce((acc, point, i) => i === 0 ? `M ${point.x},${point.y}` : `${acc} Q ${point.x - 10},${point.y + 10} ${point.x},${point.y}`, '');
    }
    getStraightPath(points) {
        return points.reduce((acc, point, i) => i === 0 ? `M ${point.x},${point.y}` : `${acc} L ${point.x},${point.y}`, '');
    }
    getAreaPath() {
        const linePath = this.getLinePath();
        return `${linePath} L ${this.width},${this.height} L 0,${this.height} Z`;
    }
    render() {
        const linePath = this.getLinePath();
        const areaPath = this.getAreaPath();
        return (h("svg", { key: '2ad9c2c0939a122d3e3c1e727e9f33fe52ecd376', width: this.width, height: this.height, viewBox: `0 0 ${this.width} ${this.height}` }, h("defs", { key: '1dcfbe63f895bf76d6a8e8e4e54c9b2d6fa9fafb' }, h("linearGradient", { key: 'c7461f4ddf4587d347f132523448a025f9b1a017', id: "areaGradient", x1: "0", x2: "0", y1: "0", y2: "1" }, h("stop", { key: 'fa5fc33dbe033170dc884c122e403ab4ec316093', offset: "0%", "stop-color": this.fillColor, "stop-opacity": "0.25" }), h("stop", { key: 'c234bf4bedd32b97edef1bb0e69d7b8ca7b3fd7f', offset: "100%", "stop-color": this.fillColor, "stop-opacity": "0" }))), h("path", { key: '85aca411c2c387929308fa2814f47b4ab5b02271', d: areaPath, fill: "url(#areaGradient)" }), h("path", { key: '28018df6b9951eef64277075d0686bfe51176f07', d: linePath, fill: "none", stroke: this.lineColor, "stroke-width": "2" }), this.showMarker && this.getPoints().map((point) => (h("circle", { cx: point.x, cy: point.y, r: "4", fill: "white", stroke: this.lineColor, "stroke-width": "2" })))));
    }
    static get watchers() { return {
        "size": ["handleSizeChange"]
    }; }
}, [0, "dga-metric-chart", {
        "type": [1],
        "trend": [1],
        "size": [1],
        "showMarker": [4, "show-marker"],
        "data": [16],
        "lineColor": [32],
        "fillColor": [32],
        "width": [32],
        "height": [32]
    }, undefined, {
        "size": ["handleSizeChange"]
    }]);
function defineCustomElement() {
    if (typeof customElements === "undefined") {
        return;
    }
    const components = ["dga-metric-chart"];
    components.forEach(tagName => { switch (tagName) {
        case "dga-metric-chart":
            if (!customElements.get(tagName)) {
                customElements.define(tagName, DgaMetricChart);
            }
            break;
    } });
}

export { DgaMetricChart as D, defineCustomElement as d };
