// @platformscode/core/dist/components/p-DX_C6M6_.js
import { p as proxyCustomElement, H, h, d as Host } from './p-BhqVmlhr.js';

const DgaSecondNavHeaderItem = /*@__PURE__*/ proxyCustomElement(class DgaSecondNavHeaderItem extends H {
    constructor() {
        super();
        this.__registerHost();
    }
    connectedCallback() {
        this.el.childNodes.forEach((item) => {
            const el = item;
            if (el.tagName === 'DGA-ICON'.toLocaleUpperCase() && el.attributes.getNamedItem('slot') === null) {
                el.setAttribute('slot', 'icon');
            }
        });
    }
    render() {
        return (h(Host, { key: '87572880e04d67c78bbe24dce793d6147d166d37' }, h("slot", { key: 'e478e5be9f8231a25f8f76dc5b44b062552d3305', name: 'icon' }), h("slot", { key: 'bdf901885edd5f6edfe77ae63311748da8f6550e', name: 'label' }, this.label)));
    }
    get el() { return this; }
}, [4, "dga-second-nav-header-item", {
        "label": [1]
    }]);
function defineCustomElement() {
    if (typeof customElements === "undefined") {
        return;
    }
    const components = ["dga-second-nav-header-item"];
    components.forEach(tagName => { switch (tagName) {
        case "dga-second-nav-header-item":
            if (!customElements.get(tagName)) {
                customElements.define(tagName, DgaSecondNavHeaderItem);
            }
            break;
    } });
}

export { DgaSecondNavHeaderItem as D, defineCustomElement as d };
