// @platformscode/core/dist/components/p-OGfNpKA1.js
import { p as proxyCustomElement, H, h, d as Host } from './p-BhqVmlhr.js';

const DgaSecondNavHeaderActions = /*@__PURE__*/ proxyCustomElement(class DgaSecondNavHeaderActions extends H {
    constructor() {
        super();
        this.__registerHost();
    }
    render() {
        return (h(Host, { key: 'fec48112353fcd554b608780c7805c11ef383d51' }, h("div", { key: 'bb614b01662d01e7979638c6467014af2f5486e5', class: `items-container` }, h("slot", { key: '34dca111e83fd2489e3cb795dcf0083066a354a4' }))));
    }
}, [4, "dga-second-nav-header-actions"]);
function defineCustomElement() {
    if (typeof customElements === "undefined") {
        return;
    }
    const components = ["dga-second-nav-header-actions"];
    components.forEach(tagName => { switch (tagName) {
        case "dga-second-nav-header-actions":
            if (!customElements.get(tagName)) {
                customElements.define(tagName, DgaSecondNavHeaderActions);
            }
            break;
    } });
}

export { DgaSecondNavHeaderActions as D, defineCustomElement as d };
