// @platformscode/core/dist/components/p-t2gkYg4G.js
import { p as proxyCustomElement, H, h } from './p-BhqVmlhr.js';
import { d as defineCustomElement$1 } from './p-DBTOrdV2.js';

const DgaButtonLeadingIcon = /*@__PURE__*/ proxyCustomElement(class DgaButtonLeadingIcon extends H {
    constructor() {
        super();
        this.__registerHost();
        this.variant = 'bulk';
        this.type = 'rounded';
    }
    render() {
        // Check if there are any child elements
        return (h("span", { key: 'b182ed4a863d600e9c006533a8c79611d49caaf4', class: "dga-btn-icon dga-btn-icon--leading", "aria-hidden": "true", slot: "leading-icon" }, h("dga-icon-v2", { key: 'be52a7f7de8fbb6bd7d3474475b8620e635927c1', name: this.name, size: this.size, color: this.color, variant: this.variant, type: this.type })));
    }
    get el() { return this; }
}, [0, "dga-button-leading-icon", {
        "name": [1537],
        "size": [2],
        "color": [1],
        "variant": [1],
        "type": [1]
    }]);
function defineCustomElement() {
    if (typeof customElements === "undefined") {
        return;
    }
    const components = ["dga-button-leading-icon", "dga-icon-v2"];
    components.forEach(tagName => { switch (tagName) {
        case "dga-button-leading-icon":
            if (!customElements.get(tagName)) {
                customElements.define(tagName, DgaButtonLeadingIcon);
            }
            break;
        case "dga-icon-v2":
            if (!customElements.get(tagName)) {
                defineCustomElement$1();
            }
            break;
    } });
}

export { DgaButtonLeadingIcon as D, defineCustomElement as d };
