// @platformscode/core/dist/collection/components/dga-linear-progress-bar/dga-linear-progress-bar.js
import { h } from "@stencil/core";
export class DgaLinearProgressBar {
    constructor() {
        this.RTL = false;
        this.percentage = 0;
        this.size = 'small';
        this.progressStyle = 'neutral';
        this.showLabel = true;
        this.showHelperText = true;
        this.error = false;
        this.success = false;
        this.label = 'label';
        this.helperText = 'Help Text';
    }
    render() {
        const sizeMapper = {
            small: 'sm',
            medium: 'md',
            large: 'lg',
        };
        let progressStyle = this.error ? 'error' : this.success ? "success" : this.progressStyle;
        let progressPercentage = this.percentage;
        const icon = {
            neutral: (h("svg", { key: 'bdda62d38abb3ed96b9a0490ee955b153320b45e', width: "16", height: "16", viewBox: "0 0 16 16", fill: "none", xmlns: "http://www.w3.org/2000/svg" }, h("g", { key: '3f83a5e175c48a8fef598354242620aa0fef4e08', "clip-path": "url(#clip0_17869_405)" }, h("path", { key: '1956e5eaafabbb41254a5b8354704a8cb847300a', d: "M6.06004 6.00001C6.21678 5.55446 6.52614 5.17875 6.93334 4.93943C7.34055 4.70012 7.8193 4.61264 8.28483 4.69248C8.75035 4.77233 9.17259 5.01436 9.47676 5.3757C9.78093 5.73703 9.94741 6.19436 9.94671 6.66668C9.94671 8.00001 7.94671 8.66668 7.94671 8.66668M8.00004 11.3333H8.00671M14.6667 8.00001C14.6667 11.6819 11.6819 14.6667 8.00004 14.6667C4.31814 14.6667 1.33337 11.6819 1.33337 8.00001C1.33337 4.31811 4.31814 1.33334 8.00004 1.33334C11.6819 1.33334 14.6667 4.31811 14.6667 8.00001Z", stroke: "#384250", "stroke-width": "1.5", "stroke-linecap": "round", "stroke-linejoin": "round" })), h("defs", { key: 'defb7ec1fdabdde353223c98b208db7052abbcdb' }, h("clipPath", { key: 'ad66d7d8ec9f9b7b73448accb58bb49e6fbbba29', id: "clip0_17869_405" }, h("rect", { key: '864bd767649b615ca2ab568321b7c14e84f92899', width: "16", height: "16", fill: "white" }))))),
            primary: (h("svg", { key: '78f5a620dcc9d3242b427504d508276d5d60b143', width: "16", height: "16", viewBox: "0 0 16 16", fill: "none", xmlns: "http://www.w3.org/2000/svg" }, h("path", { key: '8c8db0ede7fa793bfb8543d04ddd8925556ffe10', "fill-rule": "evenodd", "clip-rule": "evenodd", d: "M8.00002 15.3334C12.0501 15.3334 15.3334 12.0501 15.3334 8.00002C15.3334 3.94993 12.0501 0.666687 8.00002 0.666687C3.94993 0.666687 0.666687 3.94993 0.666687 8.00002C0.666687 12.0501 3.94993 15.3334 8.00002 15.3334Z", fill: "white" }), h("path", { key: 'ef21c2d0b724634aaf16fa6266c24d12022cc729', "fill-rule": "evenodd", "clip-rule": "evenodd", d: "M8.00002 15.3334C12.0501 15.3334 15.3334 12.0501 15.3334 8.00002C15.3334 3.94993 12.0501 0.666687 8.00002 0.666687C3.94993 0.666687 0.666687 3.94993 0.666687 8.00002C0.666687 12.0501 3.94993 15.3334 8.00002 15.3334ZM4.70017 7.41423C4.43982 7.67458 4.43982 8.09669 4.70017 8.35704L6.58578 10.2427C6.84613 10.503 7.26824 10.503 7.52859 10.2427L11.2998 6.47143C11.5602 6.21108 11.5602 5.78897 11.2998 5.52862C11.0395 5.26827 10.6174 5.26827 10.357 5.52862L7.05719 8.82845L5.64297 7.41423C5.38263 7.15388 4.96052 7.15388 4.70017 7.41423Z", fill: "#067647" }))),
            error: (h("svg", { key: 'b77e72d4dabe8d9b8415ed1ea7a7c4cd8bd29ef0', width: "16", height: "16", viewBox: "0 0 16 16", fill: "none", xmlns: "http://www.w3.org/2000/svg" }, h("path", { key: '6652418740e62fcf5d3451817c39b93ab8b33a82', "fill-rule": "evenodd", "clip-rule": "evenodd", d: "M8.00002 15.3334C12.0501 15.3334 15.3334 12.0501 15.3334 8.00002C15.3334 3.94993 12.0501 0.666687 8.00002 0.666687C3.94993 0.666687 0.666687 3.94993 0.666687 8.00002C0.666687 12.0501 3.94993 15.3334 8.00002 15.3334Z", fill: "white" }), h("path", { key: '489992023ff62a23544f7b958771aa868b6b835f', "fill-rule": "evenodd", "clip-rule": "evenodd", d: "M15.3334 8.00002C15.3334 12.0502 12.0502 15.3334 8.00002 15.3334C3.94989 15.3334 0.666687 12.0502 0.666687 8.00002C0.666687 3.94989 3.94989 0.666687 8.00002 0.666687C12.0502 0.666687 15.3334 3.94989 15.3334 8.00002ZM5.80471 4.862C5.54445 4.60158 5.12225 4.60158 4.862 4.862C4.60158 5.12225 4.60158 5.54445 4.862 5.80471L7.05715 8.00002L4.862 10.1953C4.60158 10.4556 4.60158 10.8778 4.862 11.138C5.12225 11.3985 5.54445 11.3985 5.80471 11.138L8.00002 8.94289L10.1953 11.138C10.4556 11.3985 10.8778 11.3985 11.138 11.138C11.3985 10.8778 11.3985 10.4556 11.138 10.1953L8.94289 8.00002L11.138 5.80471C11.3985 5.54445 11.3985 5.12225 11.138 4.862C10.8778 4.60158 10.4556 4.60158 10.1953 4.862L8.00002 7.05715L5.80471 4.862Z", fill: "#B42318" }))),
        };
        return (h("div", { key: '3a2b0572bc8a58a5af8ace95d846abc00077b609', class: 'progress-wrapper', style: { direction: this.RTL ? "rtl" : "ltr" } }, this.showLabel && h("h3", { key: 'e6039a0f91fe20c6dafbbc6ee4976a0d8a323a24', class: `${sizeMapper[this.size]}` }, this.label), h("div", { key: '1d20d32ffe787f90ce28ef3b1258663d7b061011', class: `progress-bar-container ${sizeMapper[this.size]}`, title: `${progressPercentage.toString()}%` }, h("div", { key: '531270e5382633064a35044c3c105e3015a39a7f', class: `progress-bar ${progressStyle}`, style: { width: `${progressPercentage}%` } }, this.size === "large" && progressPercentage > 8 && h("span", { key: 'f9bfa8ae98a4d990789c78ae2039aedc4733f54f', class: "progress-bar-text" }, progressPercentage, "%"))), this.showHelperText && (h("div", { key: 'f1c14407be526e15673fae273ad9c3e8a434f5cc', class: "progress-bar__note" }, h("div", { key: 'd3ac0caddbd72327e6e56d44ce976b918cf53f63', class: "icon" }, this.error ? icon["error"] : this.success ? icon["primary"] : icon["neutral"]), h("div", { key: '8fec35fd4e879e8b8f3b799b5b0196ade376a1e3', class: "content" }, h("p", { key: '52c44ab296cd636fd58055eac107449b689fde2c', class: `progress-bar-content ${this.success && "success"} ${this.error && "error"} ${sizeMapper[this.size]}` }, this.helperText))))));
    }
    static get is() { return "dga-linear-progress-bar"; }
    static get originalStyleUrls() {
        return {
            "$": ["dga-linear-progress-bar.scss"]
        };
    }
    static get styleUrls() {
        return {
            "$": ["dga-linear-progress-bar.css"]
        };
    }
    static get properties() {
        return {
            "RTL": {
                "type": "boolean",
                "attribute": "r-t-l",
                "mutable": false,
                "complexType": {
                    "original": "boolean",
                    "resolved": "boolean",
                    "references": {}
                },
                "required": false,
                "optional": false,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false,
                "defaultValue": "false"
            },
            "percentage": {
                "type": "number",
                "attribute": "percentage",
                "mutable": false,
                "complexType": {
                    "original": "number",
                    "resolved": "number",
                    "references": {}
                },
                "required": false,
                "optional": false,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false,
                "defaultValue": "0"
            },
            "size": {
                "type": "string",
                "attribute": "size",
                "mutable": false,
                "complexType": {
                    "original": "'small' | 'medium' | 'large'",
                    "resolved": "\"large\" | \"medium\" | \"small\"",
                    "references": {}
                },
                "required": false,
                "optional": false,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false,
                "defaultValue": "'small'"
            },
            "progressStyle": {
                "type": "string",
                "attribute": "progress-style",
                "mutable": false,
                "complexType": {
                    "original": "'primary' | 'neutral' | 'error'",
                    "resolved": "\"error\" | \"neutral\" | \"primary\"",
                    "references": {}
                },
                "required": false,
                "optional": false,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false,
                "defaultValue": "'neutral'"
            },
            "showLabel": {
                "type": "boolean",
                "attribute": "show-label",
                "mutable": false,
                "complexType": {
                    "original": "boolean",
                    "resolved": "boolean",
                    "references": {}
                },
                "required": false,
                "optional": false,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false,
                "defaultValue": "true"
            },
            "showHelperText": {
                "type": "boolean",
                "attribute": "show-helper-text",
                "mutable": false,
                "complexType": {
                    "original": "boolean",
                    "resolved": "boolean",
                    "references": {}
                },
                "required": false,
                "optional": false,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false,
                "defaultValue": "true"
            },
            "error": {
                "type": "boolean",
                "attribute": "error",
                "mutable": false,
                "complexType": {
                    "original": "boolean",
                    "resolved": "boolean",
                    "references": {}
                },
                "required": false,
                "optional": false,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false,
                "defaultValue": "false"
            },
            "success": {
                "type": "boolean",
                "attribute": "success",
                "mutable": false,
                "complexType": {
                    "original": "boolean",
                    "resolved": "boolean",
                    "references": {}
                },
                "required": false,
                "optional": false,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false,
                "defaultValue": "false"
            },
            "label": {
                "type": "string",
                "attribute": "label",
                "mutable": false,
                "complexType": {
                    "original": "string",
                    "resolved": "string",
                    "references": {}
                },
                "required": false,
                "optional": false,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false,
                "defaultValue": "'label'"
            },
            "helperText": {
                "type": "string",
                "attribute": "helper-text",
                "mutable": false,
                "complexType": {
                    "original": "string",
                    "resolved": "string",
                    "references": {}
                },
                "required": false,
                "optional": false,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false,
                "defaultValue": "'Help Text'"
            }
        };
    }
}
