// @platformscode/core/dist/collection/components/dga-checkbox/dga-checkbox.js
import { h } from "@stencil/core";
import { guid } from "../../utils";
import { isRTL } from "../../utils/rtl";
import { tick03SolidRoundedIcon, alertCircleStrokeRoundedIcon, remove01SolidRoundedIcon, minusSignSolidRoundedIcon } from "../../assets/icons/index";
export class DgaCheckbox {
    constructor() {
        this.color = 'brand';
        this.size = 'md';
        this.disabled = false;
        this.name = guid();
        this.value = '';
        this.showBorderOnFocusAndFocusOute = true;
        this.checked = false;
        this.isChecked = this.checked;
        this.Indeterminate = false;
        // private handleActive = () => {
        //   this.labelRef?.classList.add('active');
        // };
        // private handleRemoveActive = () => {
        //   this.labelRef?.classList.remove('active');
        // };
        this.handleFocus = () => {
            var _a;
            if (!this.showBorderOnFocusAndFocusOute)
                return;
            (_a = this.labelRef) === null || _a === void 0 ? void 0 : _a.classList.add('focus');
        };
        this.handleFocusOut = () => {
            var _a;
            if (!this.showBorderOnFocusAndFocusOute)
                return;
            (_a = this.labelRef) === null || _a === void 0 ? void 0 : _a.classList.remove('focus');
        };
        this.handleEscape = (event) => {
            var _a, _b;
            if (event.key === 'Escape') {
                this.handleFocusOut();
                (_b = (_a = this.labelRef) === null || _a === void 0 ? void 0 : _a.querySelector('input')) === null || _b === void 0 ? void 0 : _b.blur();
            }
        };
    }
    componentWillLoad() {
        this.isChecked = this.checked;
    }
    componentDidLoad() {
        window.addEventListener('keydown', this.handleEscape);
    }
    watchMaxValue(newValue) {
        this.isChecked = newValue;
    }
    render() {
        const { el, disabled } = this;
        const dir = isRTL(el) ? 'rtl' : 'ltr';
        console.log(dir);
        return (h("label", { key: '89d197e6d492f63836cbbc1acdb11930bb78adbf', ref: el => (this.labelRef = el), class: `checkbox ${this.color ? `checkbox--${this.color}` : ''} checkbox--${this.size}`,
            // onMouseDown={this.handleActive}
            // onMouseUp={this.handleRemoveActive}
            // onMouseOut={this.handleRemoveActive}
            // onChange={() => this.handleRemoveActive}
            //   htmlFor={id}
            tabindex: disabled ? undefined : 0 }, h("div", { key: 'ad05d35d342e0056d62d9b43c3d2cc2a688e3c70', class: "checkbox-main" }, h("div", { key: '984821bbe1f32a5b2bfb47ce143be47b85531721', class: "checkbox-main__content" }, h("input", { key: 'b873bb169083b8420f0b4975f55efd12268aa287', type: "checkbox", name: this.name, value: this.value, disabled: this.disabled, onFocus: this.handleFocus, onBlur: this.handleFocusOut, onChange: e => {
                var _a;
                (_a = this.onChange) === null || _a === void 0 ? void 0 : _a.call(this, e);
                const target = e.target;
                this.isChecked = target.checked;
            }, onInput: this.onInput, checked: this.isChecked }), h("div", { key: '8ea4b43c69117f1614513d0c3cd671d79b210733', class: "checkbox-container" }, h("span", { key: '8543884ce9a28097cabdaa61869a2f219cc7f199', class: "checkbox-icon checkbox__checked-icon", style: { display: this.isChecked ? 'inline-block' : 'none' } }, this.Indeterminate ? h("dga-icon", { icon: minusSignSolidRoundedIcon }) : h("dga-icon", { icon: tick03SolidRoundedIcon })), h("span", { key: '4b7ad1bfffb3e853eb1f568852280bc6883fe13f', class: "checkbox-icon checkbox__indeterminate-icon", style: { display: this.isChecked ? 'none' : 'inline-block' } }, h("dga-icon", { key: 'd0d1dbf5a943144c29e8f5fd23e03d6d2e82984e', icon: remove01SolidRoundedIcon }))), !this.helperText ? (this.label && h("span", { class: "checkbox__label" }, this.label)) : (h("div", { class: "dga-flex-column" }, h("span", { class: "checkbox__label" }, this.label), h("span", { class: "checkbox__helper-text" }, this.helperText)))), this.alertMessage && (h("div", { key: '83124d4756a64f50e025f4f07e563d8a4c0d1394', class: "checkbox__warning" }, h("span", { key: '175c7e33cd767eb682130b374aa8f602527b0728', class: "checkbox__warning-icon" }, h("dga-icon", { key: '0dff6c6258871ddaf756a92069bf82a58488680b', icon: alertCircleStrokeRoundedIcon, size: 16 })), h("span", { key: 'c4d7d1d5e6483c9bb1976e12c664d3e06d1e1846', class: "checkbox__warning-text" }, this.alertText))))));
    }
    static get is() { return "dga-checkbox"; }
    static get encapsulation() { return "shadow"; }
    static get originalStyleUrls() {
        return {
            "$": ["dga-checkbox.scss"]
        };
    }
    static get styleUrls() {
        return {
            "$": ["dga-checkbox.css"]
        };
    }
    static get properties() {
        return {
            "color": {
                "type": "string",
                "attribute": "color",
                "mutable": false,
                "complexType": {
                    "original": "'brand' | 'neutral'",
                    "resolved": "\"brand\" | \"neutral\"",
                    "references": {}
                },
                "required": false,
                "optional": true,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false,
                "defaultValue": "'brand'"
            },
            "size": {
                "type": "string",
                "attribute": "size",
                "mutable": false,
                "complexType": {
                    "original": "'lg' | 'md' | 'sm'",
                    "resolved": "\"lg\" | \"md\" | \"sm\"",
                    "references": {}
                },
                "required": false,
                "optional": true,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false,
                "defaultValue": "'md'"
            },
            "label": {
                "type": "string",
                "attribute": "label",
                "mutable": false,
                "complexType": {
                    "original": "string",
                    "resolved": "string",
                    "references": {}
                },
                "required": false,
                "optional": true,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false
            },
            "helperText": {
                "type": "string",
                "attribute": "helper-text",
                "mutable": false,
                "complexType": {
                    "original": "string",
                    "resolved": "string",
                    "references": {}
                },
                "required": false,
                "optional": true,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false
            },
            "alertMessage": {
                "type": "boolean",
                "attribute": "alert-message",
                "mutable": false,
                "complexType": {
                    "original": "boolean",
                    "resolved": "boolean",
                    "references": {}
                },
                "required": false,
                "optional": true,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false
            },
            "alertText": {
                "type": "string",
                "attribute": "alert-text",
                "mutable": false,
                "complexType": {
                    "original": "string",
                    "resolved": "string",
                    "references": {}
                },
                "required": false,
                "optional": true,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false
            },
            "disabled": {
                "type": "boolean",
                "attribute": "disabled",
                "mutable": false,
                "complexType": {
                    "original": "boolean",
                    "resolved": "boolean",
                    "references": {}
                },
                "required": false,
                "optional": true,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false,
                "defaultValue": "false"
            },
            "name": {
                "type": "string",
                "attribute": "name",
                "mutable": false,
                "complexType": {
                    "original": "string",
                    "resolved": "string",
                    "references": {}
                },
                "required": false,
                "optional": true,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false,
                "defaultValue": "guid()"
            },
            "value": {
                "type": "string",
                "attribute": "value",
                "mutable": false,
                "complexType": {
                    "original": "string",
                    "resolved": "string",
                    "references": {}
                },
                "required": false,
                "optional": true,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false,
                "defaultValue": "''"
            },
            "showBorderOnFocusAndFocusOute": {
                "type": "boolean",
                "attribute": "show-border-on-focus-and-focus-oute",
                "mutable": false,
                "complexType": {
                    "original": "boolean",
                    "resolved": "boolean",
                    "references": {}
                },
                "required": false,
                "optional": true,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false,
                "defaultValue": "true"
            },
            "onChange": {
                "type": "unknown",
                "attribute": "on-change",
                "mutable": false,
                "complexType": {
                    "original": "(event?: any) => void",
                    "resolved": "(event?: any) => void",
                    "references": {}
                },
                "required": false,
                "optional": true,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false
            },
            "onInput": {
                "type": "unknown",
                "attribute": "on-input",
                "mutable": false,
                "complexType": {
                    "original": "(event?: any) => void",
                    "resolved": "(event?: any) => void",
                    "references": {}
                },
                "required": false,
                "optional": true,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false
            },
            "checked": {
                "type": "boolean",
                "attribute": "checked",
                "mutable": true,
                "complexType": {
                    "original": "boolean",
                    "resolved": "boolean",
                    "references": {}
                },
                "required": false,
                "optional": true,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": true,
                "defaultValue": "false"
            },
            "Indeterminate": {
                "type": "boolean",
                "attribute": "indeterminate",
                "mutable": false,
                "complexType": {
                    "original": "boolean",
                    "resolved": "boolean",
                    "references": {}
                },
                "required": false,
                "optional": true,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false,
                "defaultValue": "false"
            }
        };
    }
    static get states() {
        return {
            "isChecked": {}
        };
    }
    static get elementRef() { return "el"; }
    static get watchers() {
        return [{
                "propName": "checked",
                "methodName": "watchMaxValue"
            }];
    }
}
