// @platformscode/core/dist/cjs/dga-link-usage.cjs.entry.js
'use strict';

var index = require('./index-CgKhFr-4.js');

const DgaLinkUsage = class {
    constructor(hostRef) {
        index.registerInstance(this, hostRef);
        this.variant = 'neutral';
    }
    render() {
        return (index.h("div", { key: 'd3535c59ad7d9ee7b8c572371185d8a1d6f904d3' }, index.h("h1", { key: '69a9746a46fdff8283e019fcdded2b05e3c58d58' }, "Link Example"), index.h("div", { key: '4d38ef7601e90cfc0590614faaaf52a886e74ef2', style: { padding: '50px', display: 'flex', gap: '20px' } }, index.h("div", { key: 'a7474aa9ebdba9deaea67d947ce73399ed8052ff', style: { display: 'flex', flexDirection: 'column', gap: '20px' } }, index.h("dga-link", { key: 'bcd7aab3f2a8f87903495a7bcaade15363565c19', label: "Link", variant: "neutral" }), index.h("dga-link", { key: 'bd0b0a8cbacde7f58456d23895cae933f1c3cff0', label: "Link", variant: "on-color" }), index.h("dga-link", { key: '216150826f3a66dbdcb7e3f4dd57d43a9e64751c', label: "Link", variant: "primary" })), index.h("div", { key: '6e963385c72e429521625fc766d3ef30f7daa14b', style: { display: 'flex', flexDirection: 'column', gap: '20px' } }))));
    }
};

exports.dga_link_usage = DgaLinkUsage;
