// @platformscode/core/dist/cjs/dga-list.cjs.entry.js
'use strict';

var index = require('./index-CgKhFr-4.js');

const DgaList = class {
    constructor(hostRef) {
        index.registerInstance(this, hostRef);
        this.type = 'unordered';
        this.color = 'primary';
        //   @Prop() id: string = guid();
        this.title = "";
        this.items = [];
        this.renderIcon = (type) => {
            if (type === "with-icon") {
                return index.h("span", { class: "list-icon" }, "\u2713"); // Using checkmark icon for "with-icon" type
            }
            return null;
        };
    }
    render() {
        const Tag = this.type === 'ordered' ? 'ol' : 'ul';
        return (index.h("div", { key: '7d95361dcea178a59da32b4d08bf7f0560eb3d48', class: `list list--${this.type} list--${this.color}` }, this.title && index.h("h3", { key: '0344d56c05bb2edaee1716489128994fd076a0bf', class: "list-title" }, this.title), index.h(Tag, { key: 'd7e8b8e2a3f2162f9b0c6b25e9a6025d44f0ea71' }, this.items.map((item, index$1) => (index.h("li", { key: index$1, class: "list-item" }, this.type === "with-icon" && this.renderIcon(this.type), item))))));
    }
};

exports.dga_list = DgaList;
