// @platformscode/core/dist/cjs/dga-data-table-usage.cjs.entry.js
'use strict';

var index = require('./index-CgKhFr-4.js');

const DgaDateFieldUsage = class {
    constructor(hostRef) {
        index.registerInstance(this, hostRef);
        this.value = '';
    }
    render() {
        return (index.h("div", { key: 'f5634d14aa0af69dfec196238e9771ea4387afaf' }, index.h("h1", { key: '10b89f738ada8b9dffa038131f1b922fff9a7d2d' }, "Date Field Example"), index.h("div", { key: '27d4efdf00c4af04bd276891bed9d79dfc09ee97', style: { padding: '50px' } }, index.h("div", { key: '3a8e86317ca962d6aff3e87c35b18c638b44e07b', style: { display: 'flex', flexDirection: 'column', gap: '16px' } }, index.h("dga-data-table", { key: '320dfe63275dc14537c762771e4ce56814a8378a', cells: [
                {
                    colSpan: 1,
                    isFilter: true,
                    isSort: true,
                    propertyName: 'city',
                    sort: 'up',
                    title: 'city',
                    type: 'propertyName',
                },
                {
                    colSpan: 1,
                    isFilter: true,
                    isSort: true,
                    propertyName: 'price',
                    sort: 'down',
                    title: 'price',
                    type: 'propertyName',
                },
                {
                    colSpan: 1,
                    isFilter: true,
                    isSort: true,
                    propertyName: 'state',
                    title: 'state',
                    type: 'element', // <-- change this
                    El: row => index.h("dga-button", { label: row.state }, console.log(row, 'roooooow')), // <-- use a functional component
                },
            ], contained: true, data: [
                {
                    city: 'Dallas',
                    h_id: '3',
                    price: '162500',
                    selected: false,
                    state: 'TX',
                    zip: '75201',
                },
                {
                    city: 'Bevery Hills',
                    h_id: '4',
                    price: '319250',
                    selected: false,
                    state: 'CA',
                    zip: '90210',
                },
                {
                    city: 'New York',
                    h_id: '5',
                    price: '962500',
                    selected: false,
                    state: 'NY',
                    zip: '00010',
                },
            ], getSelectedRows: function vu() { }, pressOnFilter: function vu() { }, rowDivider: true, showSelectCheckBox: true })))));
    }
};

exports.dga_data_table_usage = DgaDateFieldUsage;
