// @platformscode/core/dist/cjs/dga-content-switcher_3.cjs.entry.js
'use strict';

var index = require('./index-CgKhFr-4.js');

const DgaContentSwitcher = class {
    constructor(hostRef) {
        index.registerInstance(this, hostRef);
        this.value = 0;
        this.onColor = false;
        this.size = 'md';
        this.items = [
            {
                label: 'Item 1',
                content: 'Content 1',
            },
            {
                label: 'Item 2',
                content: 'Content 2',
            },
            {
                label: 'Item 3',
                content: 'Content 3',
            },
        ];
        this.onChange = () => { };
        this.selected = this.value;
        this.handleSelect = (index) => {
            this.selected = index;
            this.onChange(index);
        };
    }
    render() {
        return (index.h("div", { key: '2466b35db0280a8325df5b74a22c71b192666141', class: "content-switcher" }, index.h("div", { key: '348081a64adfd115ef5715bb5f05004cf475ea5c', class: "content-switcher__items" }, this.items.map((item, index$1) => {
            const selected = index$1 === this.selected;
            const itemType = index$1 === 0 ? 'first' : index$1 === this.items.length - 1 ? 'last' : 'mid';
            const className = `content-switcher__item content-switcher__item--${itemType} ${this.onColor ? 'content-switcher__item--on-color' : ''} content-switcher__item--${this.size} ${selected ? 'content-switcher__item--selected' : ''}`;
            return (index.h("button", { key: index$1, type: "button", class: className, onClick: () => this.handleSelect(index$1) }, item.label));
        }))));
    }
};

const DgaContentSwitcherItem = class {
    constructor(hostRef) {
        index.registerInstance(this, hostRef);
    }
    render() {
        return index.h("slot", { key: 'e6e96a01a223b5899137acfa146a1c915477daae' });
    }
};

const DgaContentSwitcherPanel = class {
    constructor(hostRef) {
        index.registerInstance(this, hostRef);
    }
    render() {
        return (index.h("div", { key: '11b6c9cacf81551eb6b19325366c051032810d87', class: "content-switcher-panel", hidden: this.value !== this.index }, index.h("slot", { key: '2c437043353ecdfc7586f63010b8b9b3f6c1ffd2' })));
    }
};

exports.dga_content_switcher = DgaContentSwitcher;
exports.dga_content_switcher_item = DgaContentSwitcherItem;
exports.dga_content_switcher_panel = DgaContentSwitcherPanel;
