// @platformscode/core/dist/cjs/dga-structured-list-usage.cjs.entry.js
'use strict';

var index = require('./index-CgKhFr-4.js');

const DgaStructuredListUsage = class {
    constructor(hostRef) {
        index.registerInstance(this, hostRef);
    }
    render() {
        return (index.h("div", { key: '8af8dd832806e8928a197fa1988d1d86d1e30011' }, index.h("h1", { key: 'caa4445333d278a053ab424bd89ba765a82da8d4' }, "Structured List Example"), index.h("div", { key: '6089dad32ecf56764f7e23c926066fcd94ad49e8', style: { padding: "50px" } }, index.h("dga-structured-list", { key: '90b31d161cae14b9f478d5235975e82b06b33fbd', cells: [
                {
                    title: "ID",
                    propertyName: "id",
                    type: "propertyName",
                },
                {
                    title: "Name",
                    propertyName: "name",
                    type: "propertyName",
                },
            ], data: [
                { id: 1, name: "Alice", selected: false },
                { id: 2, name: "Bob", selected: false },
                { id: 3, name: "Charlie", selected: false },
            ], rowDivider: true, colDivider: false, selectable: true, getSelectedRows: selectedRows => console.log(selectedRows) }), index.h("dga-structured-list", { key: '20d63169ca065c07a6128e680d5bc988d09409a8', compact: true, cells: [
                {
                    title: "ID",
                    propertyName: "id",
                    type: "propertyName",
                },
                {
                    title: "Name",
                    propertyName: "name",
                    type: "propertyName",
                },
            ], data: [
                { id: 1, name: "Alice", selected: false },
                { id: 2, name: "Bob", selected: false },
                { id: 3, name: "Charlie", selected: false },
            ], rowDivider: true, selectable: true, getSelectedRows: selectedRows => console.log(selectedRows) }))));
    }
};

exports.dga_structured_list_usage = DgaStructuredListUsage;
