// @platformscode/core/dist/cjs/dga-slider-usage.cjs.entry.js
'use strict';

var index = require('./index-CgKhFr-4.js');

const DgaSliderUsage = class {
    constructor(hostRef) {
        index.registerInstance(this, hostRef);
        this.activeStep = 1;
        this.minValue = 10;
        this.maxValue = 100;
    }
    render() {
        return (index.h(index.Fragment, { key: '2dd0277992f2ccc09b234b39015acc9d3505d07d' }, index.h("br", { key: '2f8014793fbb6f09227dd05677935a595d18a401' }), index.h("dga-slider", { key: '786da1950c5ca8b4e38297c4054eef013b3c91a7', min: this.minValue, max: this.maxValue, step: 1, range: true, size: "medium", helperText: "Helper Text Example", label: "Test", sx: {
                backgroundColor: "yellow",
                marginTop: "200px",
                ".range-slider-inner-rail": {
                    backgroundColor: "red"
                }
            } }), ";", index.h("br", { key: '1a3ddb903fa675ee0cab4114b608fa2ac8d26488' }), this.minValue, " - ", this.maxValue, index.h("dga-slider", { key: 'eb9501d6b38fd40671fb138ccf8cfeb4b69a8ddb' })));
    }
};

exports.dga_slider_usage = DgaSliderUsage;
