// @platformscode/core/dist/cjs/dga-button-usage.cjs.entry.js
'use strict';

var index = require('./index-CgKhFr-4.js');

const DgaButtonUsage = class {
    constructor(hostRef) {
        index.registerInstance(this, hostRef);
        this.variant = 'neutral';
    }
    render() {
        return (index.h("div", { key: '42e4dcac6a8e5894473d5bb1e9d33657b97c6019' }, index.h("h1", { key: 'd5146ccfef34f67ce4bd8b12a03f8d2bfc81adb3' }, "Button Example"), index.h("div", { key: '5100816a09c96d79afec208186830a02027acf7c', style: { padding: '50px', display: 'flex', gap: '20px' } }, index.h("div", { key: '2c8aa1c5d07bacd0187db0eb4c993b4297a7235a', style: { display: 'flex', flexDirection: 'column', gap: '20px' } }, index.h("dga-button", { key: '30a1c6cdfb7addad54581223dc285d3ad6891124', label: "Click me", variant: "primary-brand", sx: {
                '&': {
                    fontSize: '18px',
                    fontWeight: 'bold',
                },
                '.dga-btn': {
                    backgroundColor: { xs: 'red', sm: 'blue', md: 'green' },
                    // backgroundColor: 'yellow',
                    color: { xs: 'black', sm: 'yellow', md: 'white' },
                },
                '.dga-btn:hover': {
                    backgroundColor: 'blue',
                },
            } }), index.h("dga-button", { key: '4f0af478bfea2622fbb41f068f5758e4ad9e8d5a', label: "Brown Button", variant: "primary-neutral" }), index.h("dga-button", { key: '66e35ddfdba9ac3f2e790987168c4caaf0ebafac', label: "Button", variant: "des-transparent" }), index.h("dga-button", { key: '2672bb3597dba7f408458b6c3bfe2677def2aa8b', label: "Source code", variant: "secondary-outline", leadIcon: true, leadIconType: "source-code", leadIconProps: { variant: 'stroke', type: 'standard' } }), index.h("dga-button", { key: 'e0f5b635ac8a3e8adc9bddbb99650c796130c363',
            // className="absolute top-[50%] left-[50%] transform -translate-x-1/2 -translate-y-1/2 opacity-0 group-hover:opacity-100"
            label: "See live preview", size: "lg", variant: "secondary-outline", trailIcon: true, trailIconType: "link-square-01", trailIconProps: {
                variant: 'stroke',
                type: 'standard',
            } })), index.h("div", { key: '4a0d497b275807dfefb313dbdf98ce95842b2214', style: { display: 'flex', flexDirection: 'column', gap: '20px' } }))));
    }
};

exports.dga_button_usage = DgaButtonUsage;
