// @platformscode/core/dist/cjs/dga-tooltip-usage.cjs.entry.js
'use strict';

var index = require('./index-CgKhFr-4.js');

const DgaTooltipUsage = class {
    constructor(hostRef) {
        index.registerInstance(this, hostRef);
        this.activeStep = 1;
    }
    render() {
        return (index.h("div", { key: '0064e5dff22f0ff8a3b5f569deb22d68e2b2c94c' }, index.h("h1", { key: '825e0479f5239adfe639b7b66da34de9725d7bc1' }, "Tooltip Example"), index.h("div", { key: '331afe44f9f531fabf614d567f9353f32ab5c092', style: { padding: "50px", display: "flex", gap: "20px" } }, index.h("div", { key: 'b3d786abd00af5ddf19ce70c866109a73a60bc4e', style: { display: "flex", flexDirection: "column", gap: "20px" } }, index.h("dga-tooltip", { key: '29a1d4ffc256c0e497f2b0e6641bc3fb5d992c3d', title: "Tooltip title", helperText: "Max width of tooltips is 240px - text will wrap automatically", direction: 'right', theme: 'dark', icon: true, beakAlignment: 'end', beakPlacement: 'left', feedbackIcon: {
                type: "error-!",
                style: "container",
                size: 24
            } }, "Hello")), index.h("div", { key: '8a42e1d104123e01fc04f7a1daee7beb20129488', style: { display: "flex", flexDirection: "column", gap: "20px" } }))));
    }
};

exports.dga_tooltip_usage = DgaTooltipUsage;
