// @platformscode/core/dist/cjs/dga-menu.cjs.entry.js
'use strict';

var index = require('./index-CgKhFr-4.js');

const DgaMenu = class {
    constructor(hostRef) {
        index.registerInstance(this, hostRef);
        this.anchorEl = null;
        this.open = false;
        this.onClose = () => { };
        // Method to detect outside click
        this.handleClickOutside = (event) => {
            if (this.anchorEl && !this.anchorEl.contains(event.target)) {
                this.onClose();
            }
        };
    }
    componentDidLoad() {
        // Add event listener to handle clicks outside the menu
        document.addEventListener('click', this.handleClickOutside);
    }
    disconnectedCallback() {
        // Cleanup event listener when the component is destroyed
        document.removeEventListener('click', this.handleClickOutside);
    }
    getMenuStyle() {
        var _a;
        const rect = (_a = this.anchorEl) === null || _a === void 0 ? void 0 : _a.getBoundingClientRect();
        return {
            position: 'absolute',
            top: `${(rect === null || rect === void 0 ? void 0 : rect.bottom) + window.scrollY}px`,
            left: `${(rect === null || rect === void 0 ? void 0 : rect.left) + window.scrollX}px`,
            zIndex: '1000',
        };
    }
    render() {
        return (index.h(index.Fragment, { key: '26bddcf78184796ce02fcf705488bdd613cf6949' }, this.open && this.anchorEl && (index.h("ul", { key: '6c65291898e68652253898973e165b59093c7aa1', class: "menu", style: this.getMenuStyle() }, index.h("slot", { key: '8aae03b4774f80523c472f18d8d023db87d2590b' })))));
    }
};

exports.dga_menu = DgaMenu;
