// @platformscode/core/dist/cjs/dga-avatar-usage.cjs.entry.js
'use strict';

var index = require('./index-CgKhFr-4.js');

const DgaAvatarUsage = class {
    constructor(hostRef) {
        index.registerInstance(this, hostRef);
        this.variant = 'neutral';
    }
    render() {
        return (index.h("div", { key: 'e245158aad21942e1be0fb88ef693055395148d4' }, index.h("h1", { key: 'a8759f1e8dd1e9175b940ac03e54aa28a2bbb6e1' }, "Avatar Example"), index.h("div", { key: 'a7d24af241d3ed261943dd93e17edf74d9a439d5', style: { padding: '50px', display: 'flex', gap: '20px' } }, index.h("div", { key: '98e6af7fe785f9e6e7137bd37df4a1ea3152e038', style: { display: 'flex', flexDirection: 'column', gap: '20px' } }, index.h("div", { key: '58d6e98a928a71a17421b19779bdb156cba74faf', style: { padding: "10px", display: 'flex', flexDirection: 'column', gap: '20px', backgroundColor: "black" } }, index.h("dga-avatar", { key: '0d00e519480ff976c78e2434cedcf61665b72289', size: 120, border: false, square: true, sx: {
                "& .dga-avatar": {
                    borderRadius: "100px"
                }
            } }), index.h("dga-avatar", { key: '8e458edc0c9441e26d87ff6a984f92335a8f7b8b', size: 120, border: true, square: true })), index.h("dga-avatar", { key: 'f315e6c5196045660ad11816783334b08d5d8880', type: "icon", size: 120, icon: {
                name: 'user',
                variant: "stroke",
                type: "rounded"
            } }), index.h("dga-avatar", { key: '513af1465f73602c3b85faec5a9e6352efcfb027', type: "icon", size: 80, icon: {
                name: 'user',
                variant: "stroke",
                type: "rounded"
            } }), index.h("dga-avatar", { key: 'c43c6de03e61b4d7165a5e1b9196caf6d0498b59', type: "icon", size: 68, icon: {
                name: 'user',
                variant: "stroke",
                type: "rounded"
            } }), index.h("dga-avatar", { key: '81727eae24c46330e5fe91192e778d0dbbde4435', type: "icon", size: 48, icon: {
                name: 'user',
                variant: "stroke",
                type: "rounded"
            } }), index.h("dga-avatar", { key: 'cd28d5122605132f3ef6da32088092602d97675f', type: "icon", size: 40, icon: {
                name: 'user',
                variant: "stroke",
                type: "rounded"
            } }), index.h("dga-avatar", { key: 'a89fb958d48fc07c0f0b93a575ca5af3a03524a2', type: "icon", size: 32, icon: {
                name: 'user',
                variant: "stroke",
                type: "rounded"
            } }), index.h("dga-avatar", { key: '413ef17f0a7ee9e9cedb01750e4e0a0d77567bd7', type: "icon", size: 24, icon: {
                name: 'user',
                variant: "stroke",
                type: "rounded"
            } })), index.h("dga-avatar-group", { key: '9cab23f29796c9420ba21e77eb60744039cb44f8', stacked: true }, index.h("dga-avatar", { key: 'aeb84ba1ae0eb899f7e4c35662f95c782a71eea0', type: "initials", text: 'AB', size: 40 }), index.h("dga-avatar", { key: 'e628821c70e828f9064d63d5f2138d606942799e', type: "initials", text: 'CD', size: 40 }), index.h("dga-avatar", { key: '386ce6fc1bdb9997562fbac4d49625394760804b', type: "initials", text: 'SB', size: 40 }), index.h("dga-avatar", { key: '7b4d37d4ced60ad0d7b5f41e699dcd16eaf6663f', type: "initials", text: 'WK', size: 40 }), index.h("dga-avatar", { key: '9bd4cbacf6d002d4620b89d53927728eaa09d998', type: 'initials', text: `+4`, size: 40 })), index.h("div", { key: '51d0d03e88610a78c53ce7c8ef8fc2f2824d7c61', style: { display: 'flex', flexDirection: 'column', gap: '20px' } }))));
    }
};

exports.dga_avatar_usage = DgaAvatarUsage;
