// @platformscode/core/dist/cjs/dga-radial-stepper-usage.cjs.entry.js
'use strict';

var index = require('./index-CgKhFr-4.js');

const DgaRadialStepperUsage = class {
    constructor(hostRef) {
        index.registerInstance(this, hostRef);
        this.percentage = 0;
        this.activeStep = 0;
    }
    handleInputChange(event) {
        const target = event.target;
        this.percentage = Number(target.value); // Convert the string value to a number
    }
    handleStepChange(event) {
        const target = event.target;
        this.activeStep = Number(target.value); // Convert the string value to a number
    }
    render() {
        return (index.h("div", { key: '3bfd206db81ed99dae9ec22401ce166ab9ea202d' }, index.h("h1", { key: 'e0ebc16837cf106a013ba87d0e6335a561a8ea89' }, "Radial Stepper Example"), index.h("div", { key: '4d62785358273ec5b138e2ae73a48f97fa255805', style: { padding: '50px', gap: '20px' } }, index.h("div", { key: 'a4c02d5dae77954998556f79d2f85138ac2f0ab7', style: { display: 'flex', flexDirection: 'column', gap: '20px' } }, index.h("dga-radial-stepper", { key: '46515a109fefa768232d04f46169729ff592e002', size: "120px", activeStep: this.activeStep, steps: [
                {
                    stepName: "Step One",
                    stepDescription: "this is step one description",
                    showPreStep: true,
                    showNextStep: true
                },
                {
                    stepName: "Step Two",
                    stepDescription: "this is step two description",
                    showPreStep: true,
                    showNextStep: true
                },
                {
                    stepName: "Step Three",
                    stepDescription: "this is step three description",
                    showPreStep: true,
                    showNextStep: true
                },
                {
                    stepName: "Step Four",
                    stepDescription: "this is step four description",
                    showPreStep: true,
                    showNextStep: true
                }
            ] }), index.h("dga-radial-stepper", { key: '7eb636bc195cf435c17cd9a45fbdcc306210e415', size: "80px", color: "success", activeStep: this.activeStep, steps: [
                {
                    stepName: "Step One",
                    stepDescription: "this is step one description",
                    showPreStep: true,
                    showNextStep: true
                },
                {
                    stepName: "Step Two",
                    stepDescription: "this is step two description",
                    showPreStep: true,
                    showNextStep: true
                },
                {
                    stepName: "Step Three",
                    stepDescription: "this is step three description",
                    showPreStep: true,
                    showNextStep: true
                },
                {
                    stepName: "Step Four",
                    stepDescription: "this is step four description",
                    showPreStep: true,
                    showNextStep: true
                }
            ] }), index.h("dga-radial-stepper", { key: '1b6c497328a00e4975a09fbbacddc7db4faa71f3', variant: 'neutral', size: "64px", activeStep: this.activeStep, steps: [
                {
                    stepName: "Step One",
                    stepDescription: "this is step one description",
                    showPreStep: true,
                    showNextStep: true
                },
                {
                    stepName: "Step Two",
                    stepDescription: "this is step two description",
                    showPreStep: true,
                    showNextStep: true
                },
                {
                    stepName: "Step Three",
                    stepDescription: "this is step three description",
                    showPreStep: true,
                    showNextStep: true
                },
                {
                    stepName: "Step Four",
                    stepDescription: "this is step four description",
                    showPreStep: true,
                    showNextStep: true
                }
            ] }), index.h("dga-radial-stepper", { key: 'e1eea3d7c88b596ce0e38c2150f2ac0aa7970e8e', size: "48px", activeStep: this.activeStep, steps: [
                {
                    stepName: "Step One",
                    stepDescription: "this is step one description",
                    showPreStep: true,
                    showNextStep: true
                },
                {
                    stepName: "Step Two",
                    stepDescription: "this is step two description",
                    showPreStep: true,
                    showNextStep: true
                },
                {
                    stepName: "Step Three",
                    stepDescription: "this is step three description",
                    showPreStep: true,
                    showNextStep: true
                },
                {
                    stepName: "Step Four",
                    stepDescription: "this is step four description",
                    showPreStep: true,
                    showNextStep: true
                }
            ] }), index.h("dga-radial-stepper", { key: '598021ae613ccabbf08d5533ffc440e8becb250c', size: "40px", activeStep: this.activeStep, steps: [
                {
                    stepName: "Step One",
                    stepDescription: "this is step one description",
                    showPreStep: true,
                    showNextStep: true
                },
                {
                    stepName: "Step Two",
                    stepDescription: "this is step two description",
                    showPreStep: true,
                    showNextStep: true
                },
                {
                    stepName: "Step Three",
                    stepDescription: "this is step three description",
                    showPreStep: true,
                    showNextStep: true
                },
                {
                    stepName: "Step Four",
                    stepDescription: "this is step four description",
                    showPreStep: true,
                    showNextStep: true
                }
            ] }), index.h("div", { key: '1fedca41452567be59be5c385a6755d56d80461f', style: {
                backgroundColor: "#1B8354",
                padding: "40px"
            } }, index.h("dga-radial-stepper", { key: 'ae1299b79c91f475dd75ef211930cd64a39b0589', variant: "on-color", size: "120px", activeStep: this.activeStep, steps: [
                {
                    stepName: "Step One",
                    stepDescription: "this is step one description",
                    showPreStep: true,
                    showNextStep: true
                },
                {
                    stepName: "Step Two",
                    stepDescription: "this is step two description",
                    showPreStep: true,
                    showNextStep: true
                },
                {
                    stepName: "Step Three",
                    stepDescription: "this is step three description",
                    showPreStep: true,
                    showNextStep: true
                },
                {
                    stepName: "Step Four",
                    stepDescription: "this is step four description",
                    showPreStep: true,
                    showNextStep: true
                },
                {
                    stepName: "Step Five",
                    stepDescription: "this is step four description",
                    showPreStep: true,
                    showNextStep: true
                },
                {
                    stepName: "Step Six",
                    stepDescription: "this is step four description",
                    showPreStep: true,
                    showNextStep: true
                }
            ], RTL: true })), index.h("button", { key: 'abe42640524b6901e6152fc9a411b8838344bd68', onClick: () => (this.activeStep += 1) }, "Next"), index.h("button", { key: '72142c1d3494195ecdd122b9f79e9a622f34d093', onClick: () => (this.activeStep -= 1) }, "Prev"), index.h("input", { key: '87c2d846e13106f4b8b5ddc7f0b4ba7eb7c2e152', type: "range", min: "0", max: "100", value: this.percentage, onInput: e => this.handleInputChange(e) })))));
    }
};

exports.dga_radial_stepper_usage = DgaRadialStepperUsage;
