// @platformscode/core/dist/cjs/dga-divider-usage.cjs.entry.js
'use strict';

var index = require('./index-CgKhFr-4.js');

const DgaDividerUsage = class {
    constructor(hostRef) {
        index.registerInstance(this, hostRef);
    }
    render() {
        return (index.h("div", { key: 'ff3ffee2d70a8c778f152859d05a64fde636f100' }, index.h("h1", { key: 'c915d5860e95872ea8208ee77384c947cb052752' }, "Divider Example"), index.h("div", { key: '6e0cbfbc0c5328beca2c1ef5c24d414108bb317d', style: { padding: '50px', display: 'flex', gap: '20px' } }, index.h("div", { key: '1be28c682c8037dfbdb980879e313371caeba521', style: { display: 'flex', flexDirection: 'column', gap: '20px' } }, index.h("div", { key: 'a723b61e53c4cb5c6bc15de3ba44e97df6d94066', style: {
                borderRadius: '20px',
                background: 'gray',
            } }, index.h("div", { key: '998827d8cee8154fa00768400b7cc3d2dbcdc43b', style: {
                width: '224px',
                height: '82px',
                background: '#fff',
                padding: '16px',
                display: 'flex',
                flexDirection: 'column',
                gap: '40px',
            } }, index.h("dga-divider", { key: '6d4a639c5a84397e9828f387a1952ed4bb37d0cb', color: "neutral" }), index.h("dga-divider", { key: '59672506be6126b1cc4320904923624b7351c19c', color: "primary" })), index.h("div", { key: '649700e28f0c39cac33654544529cc7dc43b55af', style: {
                width: '224px',
                height: '82px',
                background: '#fff',
                padding: '16px',
                display: 'flex',
                // flexDirection: 'column',
                gap: '40px',
            } }, index.h("dga-divider", { key: '6caa38e40aa960c2bb8e1a9a6e40717c428d564d', color: "neutral", orientation: 'vertical' }), index.h("dga-divider", { key: '3a182b9fa671fc31c44376ef710899d8c3af99dd', color: "primary", orientation: 'vertical' })), index.h("div", { key: '40c3818c30854ce4a0292b248be8d177f25d9abf', style: {
                width: '224px',
                height: '82px',
                background: '#1B8354',
                padding: '16px',
                display: 'flex',
                flexDirection: 'column',
                gap: '40px',
            } }, index.h("dga-divider", { key: 'e0ce8d59aa3b5d452fa039fe81b9307ca3b947a7', color: "alpha-white" }), index.h("dga-divider", { key: 'a0ccb20131b45e671e8cbf606dab29ba5f47f321', color: "white" })), index.h("div", { key: 'b3cf90384fda25f8d3ccb586aaebd538b2cc977e', style: {
                width: '224px',
                height: '82px',
                background: '#1B8354',
                padding: '16px',
                display: 'flex',
                // flexDirection: 'column',
                gap: '40px',
            } }, index.h("dga-divider", { key: 'fd0895a26c24b3a9986917f492f3a900ae48edff', color: "alpha-white", orientation: 'vertical' }), index.h("dga-divider", { key: 'e6ca9271afa5fd6666f176e2452e30ace2c4e72c', color: "white", orientation: 'vertical' })))), index.h("div", { key: 'b5d39e00e08c601b7cc016b2a8ec153630395ff2', style: { display: 'flex', flexDirection: 'column', gap: '20px' } }))));
    }
};

exports.dga_divider_usage = DgaDividerUsage;
