// @platformscode/core/dist/cjs/dga-collapse.cjs.entry.js
'use strict';

var index = require('./index-CgKhFr-4.js');

const DgaCollapse = class {
    constructor(hostRef) {
        index.registerInstance(this, hostRef);
        this.transitionDuration = '0.3s';
        this.transitionType = 'ease-in';
        this.height = 0;
    }
    componentDidLoad() {
        if (this.open && this.containerRef) {
            this.setHeight(this.containerRef.clientHeight);
            setTimeout(() => {
                this.setHeight('auto');
            }, parseFloat(this.transitionDuration) * 1000);
        }
    }
    watchOpen(newValue, oldValue) {
        if (newValue !== oldValue && this.containerRef) {
            if (newValue) {
                this.setHeight(this.containerRef.scrollHeight);
                setTimeout(() => {
                    this.setHeight('auto');
                }, 200);
            }
            else {
                // Handle closing animation
                this.setHeight(this.containerRef.clientHeight);
                setTimeout(() => {
                    this.setHeight(0);
                }, 20);
            }
        }
    }
    setHeight(value) {
        this.height = value;
    }
    render() {
        const styles = Object.assign(Object.assign({ overflow: 'hidden', transition: `height ${this.transitionDuration} ${this.transitionType}`, height: `${this.height}px` }, (this.height === 'auto' ? { height: 'auto' } : {})), this.customStyle);
        return (index.h("div", { key: 'b38eeb7b187504e9a8c91f09a5bf5cfb0033e45d', ref: (el) => this.containerRef = el, "aria-hidden": !this.open, style: styles }, index.h("slot", { key: '2bf0440b12d1cb7fd709691854ffee14d1df07e4' })));
    }
    static get watchers() { return {
        "open": ["watchOpen"]
    }; }
};

exports.dga_collapse = DgaCollapse;
