// @platformscode/core/dist/cjs/dga-helper-text_3.cjs.entry.js
'use strict';

var index = require('./index-CgKhFr-4.js');
var index$2 = require('./index-ByjaZgsn.js');
var sxStyleManager = require('./sx-style-manager-MFMlOw-g.js');
var index$1 = require('./index-D9eOytNW.js');

const DgaHelperText = class {
    constructor(hostRef) {
        index.registerInstance(this, hostRef);
        this.helperText = '';
        this.error = false;
    }
    render() {
        var _a;
        return (index.h("div", { key: 'f2a73d39268684cec9fba2210c59ebc7f5c5741c', class: `dga-helper-text ${this.error ? "dga-helper-text--error" : ""}` }, index.h("span", { key: '80f3bfa13dc22335d4fc636a069acbea322d3b3e', class: "dga-helper-text__icon" }, this.helperTextIcon && ((_a = this.helperTextIcon) === null || _a === void 0 ? void 0 : _a.name) && index.h("dga-icon", { key: '2bd2fe44c408bff2ffc26a71712b9717f94b6579', name: this.helperTextIcon.name, variant: this.helperTextIcon.variant, type: this.helperTextIcon.type, size: 16 })), index.h("span", { key: 'e886b1b9c9a018aa623df4920d53272615187e66', class: "dga-helper-text__desc" }, this.helperText)));
    }
};

const DgaLabel = class {
    constructor(hostRef) {
        index.registerInstance(this, hostRef);
        this.label = '';
        this.variant = 'default';
        this.size = 'md';
        this.disabled = false;
        this.required = false;
    }
    render() {
        return (index.h("label", { key: '972ee04f32e79084d35fd5c47f922ca01019c1ec', class: `dga-label dga-label--${this.size} ${this.disabled ? "disabled" : ""}` }, this.required && index.h("span", { key: '07cb58fcbc3eda4d621bda3a5263d87dc5a1583b', class: "star" }, "*"), this.label));
    }
};

const feedbackIconTypes = {
    success: index.h("dga-icon", { icon: index$1.checkmarkCircle02SolidRoundedIcon }),
    error: index.h("dga-icon", { icon: index$1.alertCircleSolidRoundedIcon }),
    warning: index.h("dga-icon", { icon: index$1.alert02SolidRoundedIcon }),
};
const DgaTextInput = class {
    constructor(hostRef) {
        index.registerInstance(this, hostRef);
        this.size = 'lg';
        this.type = 'text';
        this.variant = 'default';
        this.error = false;
        this.feedbackIcon = false;
        this.feedbackIconType = 'success';
        this.feedbackIconRing = false;
        this.helperText = '';
        this.disabled = false;
        this.readonly = false;
        this.required = false;
        this.fullwidth = true;
        //   @Prop() id: string = guid();
        this.name = index$2.guid();
        this.value = '';
        this.isActive = false;
        this.isFocused = false;
        // Generate a unique ID if not provided
        this.componentId = `dga-button-${Math.random().toString(36).substring(2, 9)}`;
        // Computed inline styles
        this.rootStyles = {};
    }
    valueChanged(newValue, oldValue) {
        if (newValue !== oldValue) {
            this.textInputEl['value'] = newValue;
        }
    }
    handleSxChange() {
        this.processStyling();
    }
    componentWillLoad() {
        // Initialize style manager with element reference
        this.sxStyleManager = new sxStyleManager.SxStyleManager(this.componentId, this.el);
    }
    componentDidLoad() {
        // Process styling after the component is fully loaded and shadow DOM is ready
        this.processStyling();
    }
    componentDidUpdate() {
        // Re-process styling after updates
        this.processStyling();
    }
    processStyling() {
        if (!this.sx)
            return;
        // Handle both string and object formats
        const sxValue = typeof this.sx === 'string' ? JSON.parse(this.sx) : this.sx;
        this.rootStyles = this.sxStyleManager.processSxProp(sxValue);
    }
    disconnectedCallback() {
        var _a;
        // Clean up styles when component is removed
        (_a = this.sxStyleManager) === null || _a === void 0 ? void 0 : _a.cleanup();
    }
    render() {
        return (index.h(index.Host, { key: '6a7418f442f89d250b71a49abb24329d63ab6634', id: this.componentId, style: this.rootStyles }, index.h("div", { key: '111513b59d70e0695cafdd3bad7b1a2cfdb4c471', class: { "dga-form-control": true, "dga-form-control--fullwidth": this.fullwidth } }, this.label && index.h("dga-label", { key: 'ecfdad326868959f54caa6ae050d893f88dd2891', label: this.label, size: this.size, required: this.required, disabled: this.disabled }), index.h("div", { key: 'fac3f3bc3c844263f1bc61a8c145a3527ec16bf7', class: `input input--${this.size} ${this.error ? 'input--error' : ''} ${this.variant !== 'default' ? `input--${this.variant}` : ''} ${this.isActive ? 'active' : ''} ${this.isFocused ? 'focus' : ''} ${this.disabled ? 'input--disabled' : ''} ${this.readonly ? 'readOnly' : ''}`, ref: el => (this.inputContainerRef = el) }, index.h("slot", { key: '7f984cb04e2f0595d9645e7ad75eec7ed7f35372', name: "prefix" }), this.icon && (index.h("span", { key: 'ff68fe935f634c0fbe46884e6da2d939547a48ac', class: "input__icon" }, index.h("dga-icon", { key: '0f73e6935cfcf8805153b4beca14be2dfb82daeb', name: this.icon.name, variant: this.icon.variant, type: this.icon.type, size: 20 }))), index.h("input", { key: '9fc19c8968a223fb7888e3ead0018901044a42c2',
            // id={id}
            ref: (el) => (this.textInputEl = el), value: this.value, name: this.name, type: this.type, onFocus: () => (this.isFocused = true), onBlur: e => {
                this.isFocused = false;
                this.onBlur(e);
            }, class: "input__field", placeholder: this.placeholder, disabled: this.disabled, readOnly: this.readonly, required: this.required, onMouseDown: () => (this.isActive = true), onMouseUp: () => (this.isActive = false), onMouseOut: () => (this.isActive = false), onChange: this.onChange, onInput: this.onInput }), this.error ? (index.h("span", { class: `input__feedback-icon input__feedback-icon--error ${this.feedbackIconRing ? 'input__feedback-icon--ring' : ''}` })) : this.feedbackIcon ? (index.h("span", { class: `input__feedback-icon input__feedback-icon--${this.feedbackIconType} ${this.feedbackIconRing ? 'input__feedback-icon--ring' : ''}` }, feedbackIconTypes[this.feedbackIconType])) : null, index.h("slot", { key: 'ec58f555af3ba84febf9358b96331e072c6616fc', name: "suffix" }), " "), this.helperText && index.h("dga-helper-text", { key: '10c91a2f4376725600f7b3172be27f57bb455dfc', helperText: this.helperText, helperTextIcon: this.helperTextIcon, error: this.error }))));
    }
    get el() { return index.getElement(this); }
    static get watchers() { return {
        "value": ["valueChanged"],
        "sx": ["handleSxChange"]
    }; }
};

exports.dga_helper_text = DgaHelperText;
exports.dga_label = DgaLabel;
exports.dga_text_input = DgaTextInput;
