// @platformscode/core/dist/cjs/dga-modal-usage.cjs.entry.js
'use strict';

var index = require('./index-CgKhFr-4.js');

const DgaModalUsage = class {
    constructor(hostRef) {
        index.registerInstance(this, hostRef);
        this.show = false;
    }
    render() {
        return (index.h("div", { key: '1d7c4cef1e1f98538810ce4b0eb77166f30cf030', style: { position: "relative" } }, index.h("h1", { key: 'a7ebea0aa5d67666454bfeecc4306a7ab7a71dfd' }, "Modal Example"), index.h("button", { key: 'ab9d06082be15e449e14b96eb2eac55b3f65a800', onClick: () => this.show = true }, "Open Modal"), index.h("dga-modal", { key: '7328e375daaf6abc83cfc846d8458b34f924f94b', buttonsList: [
                {
                    extraClass: '',
                    label: 'Close',
                    onClick: () => { },
                    title: 'Close Modal'
                }
            ], closeButton: true, modalTitle: "Basic Modal", name: "basicModal", onClose: () => { }, open: this.show }, "This is a simple modal with default configuration."), this.show ? "shown" : "hidden"));
    }
};

exports.dga_modal_usage = DgaModalUsage;
