// @platformscode/core/dist/collection/components/dga-flex/dga-flex.js
import { h, Host } from "@stencil/core";
import { SxStyleManager } from "../../utils/sx-style-manager";
export class DgaFlex {
    constructor() {
        /** Direction of stack items: 'row' (horizontal) or 'column' (vertical) */
        this.direction = 'row';
        /** Spacing between stack items */
        this.spacing = 0;
        /** Horizontal alignment of stack items */
        this.align = 'start';
        /** Vertical alignment of stack items */
        this.justify = 'start';
        /** Controls if items should wrap */
        this.wrap = false;
        // Generate a unique ID if not provided
        this.componentId = `dga-button-${Math.random().toString(36).substring(2, 9)}`;
        // Computed inline styles
        this.rootStyles = {};
    }
    handleSxChange() {
        this.processStyling();
    }
    componentWillLoad() {
        // Initialize style manager with element reference
        this.sxStyleManager = new SxStyleManager(this.componentId, this.el);
    }
    componentDidLoad() {
        // Process styling after the component is fully loaded and shadow DOM is ready
        this.processStyling();
    }
    componentDidUpdate() {
        // Re-process styling after updates
        this.processStyling();
    }
    processStyling() {
        if (!this.sx)
            return;
        // Handle both string and object formats
        const sxValue = typeof this.sx === 'string'
            ? JSON.parse(this.sx)
            : this.sx;
        this.rootStyles = this.sxStyleManager.processSxProp(sxValue);
    }
    disconnectedCallback() {
        var _a;
        // Clean up styles when component is removed
        (_a = this.sxStyleManager) === null || _a === void 0 ? void 0 : _a.cleanup();
    }
    render() {
        const styles = {
            display: 'flex',
            flexDirection: this.direction,
            gap: `${this.spacing}px`,
            alignItems: this.align,
            justifyContent: this.justify === 'between' ? 'space-between' : this.justify,
            flexWrap: this.wrap ? 'wrap' : 'nowrap'
        };
        return (h(Host, { key: 'acf4ab69b180d97f1f3285da9d62cec146ed4781', id: this.componentId, style: this.rootStyles }, h("div", { key: '4c34735d292bf2116ff842856ba07ac4900c6161', class: "dga-flex-container", style: styles }, h("slot", { key: '60dddf16983e037066496fcd86e014acee5dfd13' }))));
    }
    static get is() { return "dga-flex"; }
    static get encapsulation() { return "shadow"; }
    static get originalStyleUrls() {
        return {
            "$": ["dga-flex.scss"]
        };
    }
    static get styleUrls() {
        return {
            "$": ["dga-flex.css"]
        };
    }
    static get properties() {
        return {
            "direction": {
                "type": "string",
                "attribute": "direction",
                "mutable": false,
                "complexType": {
                    "original": "'row' | 'column'",
                    "resolved": "\"column\" | \"row\"",
                    "references": {}
                },
                "required": false,
                "optional": false,
                "docs": {
                    "tags": [],
                    "text": "Direction of stack items: 'row' (horizontal) or 'column' (vertical)"
                },
                "getter": false,
                "setter": false,
                "reflect": false,
                "defaultValue": "'row'"
            },
            "spacing": {
                "type": "number",
                "attribute": "spacing",
                "mutable": false,
                "complexType": {
                    "original": "number",
                    "resolved": "number",
                    "references": {}
                },
                "required": false,
                "optional": false,
                "docs": {
                    "tags": [],
                    "text": "Spacing between stack items"
                },
                "getter": false,
                "setter": false,
                "reflect": false,
                "defaultValue": "0"
            },
            "align": {
                "type": "string",
                "attribute": "align",
                "mutable": false,
                "complexType": {
                    "original": "'start' | 'center' | 'end' | 'stretch'",
                    "resolved": "\"center\" | \"end\" | \"start\" | \"stretch\"",
                    "references": {}
                },
                "required": false,
                "optional": false,
                "docs": {
                    "tags": [],
                    "text": "Horizontal alignment of stack items"
                },
                "getter": false,
                "setter": false,
                "reflect": false,
                "defaultValue": "'start'"
            },
            "justify": {
                "type": "string",
                "attribute": "justify",
                "mutable": false,
                "complexType": {
                    "original": "'start' | 'center' | 'end' | 'between'",
                    "resolved": "\"between\" | \"center\" | \"end\" | \"start\"",
                    "references": {}
                },
                "required": false,
                "optional": false,
                "docs": {
                    "tags": [],
                    "text": "Vertical alignment of stack items"
                },
                "getter": false,
                "setter": false,
                "reflect": false,
                "defaultValue": "'start'"
            },
            "wrap": {
                "type": "boolean",
                "attribute": "wrap",
                "mutable": false,
                "complexType": {
                    "original": "boolean",
                    "resolved": "boolean",
                    "references": {}
                },
                "required": false,
                "optional": false,
                "docs": {
                    "tags": [],
                    "text": "Controls if items should wrap"
                },
                "getter": false,
                "setter": false,
                "reflect": false,
                "defaultValue": "false"
            },
            "sx": {
                "type": "string",
                "attribute": "sx",
                "mutable": false,
                "complexType": {
                    "original": "SxProps | string",
                    "resolved": "string | { [x: string]: any; }",
                    "references": {
                        "SxProps": {
                            "location": "import",
                            "path": "../../utils/sx-style-manager",
                            "id": "src/utils/sx-style-manager.ts::SxProps"
                        }
                    }
                },
                "required": false,
                "optional": true,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false
            }
        };
    }
    static get elementRef() { return "el"; }
    static get watchers() {
        return [{
                "propName": "sx",
                "methodName": "handleSxChange"
            }];
    }
}
