// @platformscode/core/dist/collection/components/dga-footer/dga-footer.js
import { h } from "@stencil/core";
import { guid } from "../../utils";
export class DgaFooter {
    constructor() {
        this.mainTitle = '';
        this.mainDescription = '';
        this.mainLinkTilte = '';
        this.mainImage = '';
        this.copyright = '';
        this.groupLinks = [];
        this.socialMediaTitle = 'Social Media';
        this.socialMediaLinks = [];
        this.accessibilityTitle = 'Accessibility Tools';
        this.accessibilityLinks = [];
        this.basicLinks = [];
        this.extraLinks = [];
        this.bottomImages = [];
    }
    render() {
        return (h("footer", { key: '0bf3a72a79b89498d54b1e6390a08704818e944f', class: `footer ${this.background === 'DarkGreen' ? 'footer--bg-brand' : ''}` }, h("section", { key: '6c65c86684dda38b25dabbd50cb017e50f9c4a35', class: "footer-content" }, (this.mainLinkTilte || this.mainImage) && (h("div", { key: '3bdacc05cb45c9c8758728e2f0ea7a616d6efd3f', class: "footer-icon-section" }, h("a", { key: '0b47e7868c9c961667d7ff53be87d3fee4bfe3f3', class: this.background === 'DarkGreen' ? 'link--on-color' : '', href: this.mainLink || '#', target: "_blank", rel: "noopener noreferrer" }, this.mainLinkTilte))), (this.mainTitle || this.mainDescription) && (h("div", { key: 'e62e318fc34bf49f6ff35ef719ccf3343c9da1c2', class: "footer-title-description" }, h("h3", { key: '406ccbdbccef005931af02fff0569581a3b9f731', class: this.background === 'DarkGreen' ? 'link--on-color' : '' }, this.mainTitle), h("p", { key: '50f74421b0f14e804f76386931bd55d273fc8186', class: this.background === 'DarkGreen' ? 'link--on-color' : '' }, this.mainDescription))), this.NavLinks && (h("section", { key: '23da97a7b2abcfa121a2aa6f1faf89f233879b2a', class: "footer__nav-links" }, this.groupLinks.map(group => (h("div", { class: "footer__nav-links-column" }, h("div", { class: "footer__nav-links-column-title" }, group.title), h("ul", { class: "footer__nav-links-column-body" }, group.links.map(link => (h("li", null, h("a", { href: link.target, class: `link link--sm link--neutral ${this.background === 'DarkGreen' ? 'link--on-color' : ''}` }, h("span", { class: "link__label" }, link.name))))))))), h("div", { key: 'f0cbeab6b4b5f84fe79aa20d6cc8024e1c37002f', class: "footer__nav-links-column" }, this.socialMediaLinks.length > 0 && (h("div", { key: '16719b6440a15d09f69204e08d90bebf1bf86486' }, this.socialMediaTitle && h("div", { key: 'efa5bb08f24626231536fa2495abb55786c0a5d6', class: "footer__nav-links-column-title" }, this.socialMediaTitle), h("div", { key: '8edd8e4917395b96e55deba444776f8500e1ef7a', class: "dga-flex icons-container" }, this.socialMediaLinks.map(social => (h("a", { href: social.target, title: social.title, class: `dga-btn dga-btn--md dga-btn--secondary-outline dga-btn--icon ${this.background} ${this.background === 'DarkGreen' ? 'dga-btn--secondary-outline--on-color' : ''}` }, h("span", { class: "btn-icon" }, h("dga-icon-v2", Object.assign({}, social.icon, { size: 20 }))))))))), this.accessibilityLinks.length > 0 && (h("div", { key: 'b5d5b609fb7831457ff6f21d6649fcb8e602b99f' }, h("div", { key: '3d26b10c3a120a9291341801d0c5e1670f1d8cd5', class: "footer__nav-links-column-title" }, this.accessibilityTitle), h("div", { key: 'ea90e1944e5293a7282f743269df6ce3dc5cc90f', class: "dga-flex icons-container" }, this.accessibilityLinks.map(access => (h("a", { href: access.target, title: access.title, class: `dga-btn dga-btn--md dga-btn--secondary-outline dga-btn--icon ${this.background} ${this.background === 'DarkGreen' ? 'dga-btn--secondary-outline--on-color' : ''}` }, h("span", { class: "btn-icon" }, h("dga-icon-v2", Object.assign({}, access.icon, { size: 20 })))))))))))), h("section", { key: '3675304a0d05aa1f5c374f9c7b1476513055bc52', class: "footer__basic" }, h("div", { key: '670bb7ed690ceb2b4f568bf7ebeb52e00b1e0fe7', class: "footer__actions-copyright" }, h("ul", { key: 'afcbe74f270f07192d97833fa017c455ac7d6d03', class: "footer__actions" }, this.basicLinks.map(link => (h("li", null, h("a", { href: link.target, class: `link link--sm link--neutral link--inline ${this.background === 'DarkGreen' ? 'link--on-color' : ''}` }, h("span", { class: "link__label" }, link.name)))))), this.copyright && h("p", { key: 'c12fe9d6c8a6b3b64d10dd6f4dc1d7ed916a14bd', class: "footer__copyright text-sm-semibold" }, this.copyright), h("ul", { key: 'acef4f5a53fbf54e0aade3cdff877f3df06d7d9d', class: "footer__extra-actions" }, this.extraLinks.map(link => (h("li", null, h("a", { href: link.target, class: `link link--sm link--neutral ${this.background === 'DarkGreen' ? 'link--on-color' : ''}` }, h("span", { class: "link__label" }, link.name))))))), h("div", { key: '7939702071e0c435d58991271e40c0fb48de7264', class: "footer__logos" }, this.bottomImages.map(image => (h("img", { src: image, key: guid() }))))))));
    }
    static get is() { return "dga-footer"; }
    static get encapsulation() { return "shadow"; }
    static get originalStyleUrls() {
        return {
            "$": ["./footer.scss"]
        };
    }
    static get styleUrls() {
        return {
            "$": ["footer.css"]
        };
    }
    static get properties() {
        return {
            "background": {
                "type": "string",
                "attribute": "background",
                "mutable": false,
                "complexType": {
                    "original": "'DarkGreen' | 'Light'",
                    "resolved": "\"DarkGreen\" | \"Light\"",
                    "references": {}
                },
                "required": false,
                "optional": true,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false
            },
            "NavLinks": {
                "type": "boolean",
                "attribute": "nav-links",
                "mutable": false,
                "complexType": {
                    "original": "boolean",
                    "resolved": "boolean",
                    "references": {}
                },
                "required": false,
                "optional": true,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false
            },
            "mainTitle": {
                "type": "string",
                "attribute": "main-title",
                "mutable": false,
                "complexType": {
                    "original": "string",
                    "resolved": "string",
                    "references": {}
                },
                "required": false,
                "optional": false,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false,
                "defaultValue": "''"
            },
            "mainDescription": {
                "type": "string",
                "attribute": "main-description",
                "mutable": false,
                "complexType": {
                    "original": "string",
                    "resolved": "string",
                    "references": {}
                },
                "required": false,
                "optional": false,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false,
                "defaultValue": "''"
            },
            "mainLink": {
                "type": "string",
                "attribute": "main-link",
                "mutable": false,
                "complexType": {
                    "original": "string",
                    "resolved": "string",
                    "references": {}
                },
                "required": false,
                "optional": true,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false
            },
            "mainLinkTilte": {
                "type": "string",
                "attribute": "main-link-tilte",
                "mutable": false,
                "complexType": {
                    "original": "string",
                    "resolved": "string",
                    "references": {}
                },
                "required": false,
                "optional": false,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false,
                "defaultValue": "''"
            },
            "mainImage": {
                "type": "string",
                "attribute": "main-image",
                "mutable": false,
                "complexType": {
                    "original": "string",
                    "resolved": "string",
                    "references": {}
                },
                "required": false,
                "optional": false,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false,
                "defaultValue": "''"
            },
            "copyright": {
                "type": "string",
                "attribute": "copyright",
                "mutable": false,
                "complexType": {
                    "original": "string",
                    "resolved": "string",
                    "references": {}
                },
                "required": false,
                "optional": false,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false,
                "defaultValue": "''"
            },
            "groupLinks": {
                "type": "unknown",
                "attribute": "group-links",
                "mutable": false,
                "complexType": {
                    "original": "{ title: string; links: { name: string; target: string }[] }[]",
                    "resolved": "{ title: string; links: { name: string; target: string; }[]; }[]",
                    "references": {}
                },
                "required": false,
                "optional": false,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "defaultValue": "[]"
            },
            "socialMediaTitle": {
                "type": "string",
                "attribute": "social-media-title",
                "mutable": false,
                "complexType": {
                    "original": "string",
                    "resolved": "string",
                    "references": {}
                },
                "required": false,
                "optional": false,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false,
                "defaultValue": "'Social Media'"
            },
            "socialMediaLinks": {
                "type": "unknown",
                "attribute": "social-media-links",
                "mutable": false,
                "complexType": {
                    "original": "{ title?: string; target: string; icon: Icon }[]",
                    "resolved": "{ title?: string; target: string; icon: Icon; }[]",
                    "references": {
                        "Icon": {
                            "location": "import",
                            "path": "../dga-featured-icon/dga-featured-icon",
                            "id": "src/components/dga-featured-icon/dga-featured-icon.tsx::Icon"
                        }
                    }
                },
                "required": false,
                "optional": false,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "defaultValue": "[]"
            },
            "accessibilityTitle": {
                "type": "string",
                "attribute": "accessibility-title",
                "mutable": false,
                "complexType": {
                    "original": "string",
                    "resolved": "string",
                    "references": {}
                },
                "required": false,
                "optional": false,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false,
                "defaultValue": "'Accessibility Tools'"
            },
            "accessibilityLinks": {
                "type": "unknown",
                "attribute": "accessibility-links",
                "mutable": false,
                "complexType": {
                    "original": "{ title?: string; target: string; icon: Icon }[]",
                    "resolved": "{ title?: string; target: string; icon: Icon; }[]",
                    "references": {
                        "Icon": {
                            "location": "import",
                            "path": "../dga-featured-icon/dga-featured-icon",
                            "id": "src/components/dga-featured-icon/dga-featured-icon.tsx::Icon"
                        }
                    }
                },
                "required": false,
                "optional": false,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "defaultValue": "[]"
            },
            "basicLinks": {
                "type": "unknown",
                "attribute": "basic-links",
                "mutable": false,
                "complexType": {
                    "original": "{ name: string; target: string }[]",
                    "resolved": "{ name: string; target: string; }[]",
                    "references": {}
                },
                "required": false,
                "optional": false,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "defaultValue": "[]"
            },
            "extraLinks": {
                "type": "unknown",
                "attribute": "extra-links",
                "mutable": false,
                "complexType": {
                    "original": "{ name: string; target: string }[]",
                    "resolved": "{ name: string; target: string; }[]",
                    "references": {}
                },
                "required": false,
                "optional": false,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "defaultValue": "[]"
            },
            "bottomImages": {
                "type": "unknown",
                "attribute": "bottom-images",
                "mutable": false,
                "complexType": {
                    "original": "string[]",
                    "resolved": "string[]",
                    "references": {}
                },
                "required": false,
                "optional": false,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "defaultValue": "[]"
            }
        };
    }
}
