// @platformscode/core/dist/collection/components/dga-checkbox/dga-checkbox.js
import { h } from "@stencil/core";
import { guid } from "../../utils";
export class DgaCheckbox {
    constructor() {
        this.color = 'brand';
        this.size = 'md';
        this.disabled = false;
        this.name = guid();
        this.value = '';
        this.showBorderOnFocusAndFocusOute = true;
        this.checked = false;
        this.isChecked = this.checked;
        this.Indeterminate = false;
        // private handleActive = () => {
        //   this.labelRef?.classList.add('active');
        // };
        // private handleRemoveActive = () => {
        //   this.labelRef?.classList.remove('active');
        // };
        this.handleFocus = () => {
            var _a;
            if (!this.showBorderOnFocusAndFocusOute)
                return;
            (_a = this.labelRef) === null || _a === void 0 ? void 0 : _a.classList.add('focus');
        };
        this.handleFocusOut = () => {
            var _a;
            if (!this.showBorderOnFocusAndFocusOute)
                return;
            (_a = this.labelRef) === null || _a === void 0 ? void 0 : _a.classList.remove('focus');
        };
        this.handleEscape = (event) => {
            var _a, _b;
            if (event.key === 'Escape') {
                this.handleFocusOut();
                (_b = (_a = this.labelRef) === null || _a === void 0 ? void 0 : _a.querySelector('input')) === null || _b === void 0 ? void 0 : _b.blur();
            }
        };
    }
    componentWillLoad() {
        this.isChecked = this.checked;
    }
    componentDidLoad() {
        window.addEventListener('keydown', this.handleEscape);
    }
    watchMaxValue(newValue) {
        this.isChecked = newValue;
    }
    render() {
        return (h("label", { key: '787168ad4c73ee634969de203c69858e61bd5630', ref: el => (this.labelRef = el), class: `checkbox ${this.color ? `checkbox--${this.color}` : ''} checkbox--${this.size}` }, h("div", { key: '4917138cbb89b80158154ae1825c387e449dd913', class: "checkbox-main" }, h("div", { key: 'f46a45762ccb387876a05feb92de99a8e835eeb1', class: "checkbox-main__content" }, h("input", { key: '7dba10008da8ebca309c41d0d2ee883acce3e602', type: "checkbox", name: this.name, value: this.value,
            // id={id}
            // {...(customAttribute.key && {
            //   [customAttribute.key]: customAttribute.value
            //     ? customAttribute.value
            //     : "",
            // })}
            disabled: this.disabled, onFocus: this.handleFocus, onBlur: this.handleFocusOut, onChange: e => {
                var _a;
                (_a = this.onChange) === null || _a === void 0 ? void 0 : _a.call(this, e);
                const target = e.target;
                this.isChecked = target.checked;
            }, onInput: this.onInput, checked: this.isChecked }), h("div", { key: '787c1c2136f1d3088fe58974e86ff48ad4848734', class: "checkbox-container" }, h("span", { key: '32e71bf897db07d910a974352337bf602b39c1c1', class: "checkbox-icon checkbox__checked-icon", style: { display: this.isChecked ? 'inline-block' : 'none' } }, this.Indeterminate ? h("dga-icon", { icon: "minus-sign", variant: 'solid' }) : h("dga-icon", { icon: "tick-03", variant: 'solid' })), h("span", { key: 'ae21b8d5b4a9a5cdae989e86bab9d0901241d642', class: "checkbox-icon checkbox__indeterminate-icon", style: { display: this.isChecked ? 'none' : 'inline-block' } }, h("dga-icon-v2", { key: 'bc88eab4e863ab0923561331e1114609cb2c7ca4', name: 'remove-01', variant: 'solid' }))), !this.helperText ? (this.label && h("span", { class: "checkbox__label" }, this.label)) : (h("div", { class: "dga-flex-column" }, h("span", { class: "checkbox__label" }, this.label), h("span", { class: "checkbox__helper-text" }, this.helperText)))), this.alertMessage && (h("div", { key: '56795666001c8d48a4c10b39f3061263468f800a', class: "checkbox__warning" }, h("span", { key: 'bfa8e4457b68393c07ac84f6dc4e277f2eee2d2a', class: "checkbox__warning-icon" }, h("dga-icon-v2", { key: 'ca0454e322ca3b6470dcb007d8e161c96d31851d', name: 'alert-circle', variant: 'stroke', type: 'rounded', size: 16 })), h("span", { key: '4beb1f3e6f66b3fa5f38d927b98373c9c9c40b01', class: "checkbox__warning-text" }, this.alertText))))));
    }
    static get is() { return "dga-checkbox"; }
    static get encapsulation() { return "shadow"; }
    static get originalStyleUrls() {
        return {
            "$": ["dga-checkbox.scss"]
        };
    }
    static get styleUrls() {
        return {
            "$": ["dga-checkbox.css"]
        };
    }
    static get properties() {
        return {
            "color": {
                "type": "string",
                "attribute": "color",
                "mutable": false,
                "complexType": {
                    "original": "'brand' | 'neutral'",
                    "resolved": "\"brand\" | \"neutral\"",
                    "references": {}
                },
                "required": false,
                "optional": true,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false,
                "defaultValue": "'brand'"
            },
            "size": {
                "type": "string",
                "attribute": "size",
                "mutable": false,
                "complexType": {
                    "original": "'lg' | 'md' | 'sm'",
                    "resolved": "\"lg\" | \"md\" | \"sm\"",
                    "references": {}
                },
                "required": false,
                "optional": true,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false,
                "defaultValue": "'md'"
            },
            "label": {
                "type": "string",
                "attribute": "label",
                "mutable": false,
                "complexType": {
                    "original": "string",
                    "resolved": "string",
                    "references": {}
                },
                "required": false,
                "optional": true,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false
            },
            "helperText": {
                "type": "string",
                "attribute": "helper-text",
                "mutable": false,
                "complexType": {
                    "original": "string",
                    "resolved": "string",
                    "references": {}
                },
                "required": false,
                "optional": true,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false
            },
            "alertMessage": {
                "type": "boolean",
                "attribute": "alert-message",
                "mutable": false,
                "complexType": {
                    "original": "boolean",
                    "resolved": "boolean",
                    "references": {}
                },
                "required": false,
                "optional": true,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false
            },
            "alertText": {
                "type": "string",
                "attribute": "alert-text",
                "mutable": false,
                "complexType": {
                    "original": "string",
                    "resolved": "string",
                    "references": {}
                },
                "required": false,
                "optional": true,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false
            },
            "disabled": {
                "type": "boolean",
                "attribute": "disabled",
                "mutable": false,
                "complexType": {
                    "original": "boolean",
                    "resolved": "boolean",
                    "references": {}
                },
                "required": false,
                "optional": true,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false,
                "defaultValue": "false"
            },
            "name": {
                "type": "string",
                "attribute": "name",
                "mutable": false,
                "complexType": {
                    "original": "string",
                    "resolved": "string",
                    "references": {}
                },
                "required": false,
                "optional": true,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false,
                "defaultValue": "guid()"
            },
            "value": {
                "type": "string",
                "attribute": "value",
                "mutable": false,
                "complexType": {
                    "original": "string",
                    "resolved": "string",
                    "references": {}
                },
                "required": false,
                "optional": true,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false,
                "defaultValue": "''"
            },
            "showBorderOnFocusAndFocusOute": {
                "type": "boolean",
                "attribute": "show-border-on-focus-and-focus-oute",
                "mutable": false,
                "complexType": {
                    "original": "boolean",
                    "resolved": "boolean",
                    "references": {}
                },
                "required": false,
                "optional": true,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false,
                "defaultValue": "true"
            },
            "onChange": {
                "type": "unknown",
                "attribute": "on-change",
                "mutable": false,
                "complexType": {
                    "original": "(event?: any) => void",
                    "resolved": "(event?: any) => void",
                    "references": {}
                },
                "required": false,
                "optional": true,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false
            },
            "onInput": {
                "type": "unknown",
                "attribute": "on-input",
                "mutable": false,
                "complexType": {
                    "original": "(event?: any) => void",
                    "resolved": "(event?: any) => void",
                    "references": {}
                },
                "required": false,
                "optional": true,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false
            },
            "checked": {
                "type": "boolean",
                "attribute": "checked",
                "mutable": true,
                "complexType": {
                    "original": "boolean",
                    "resolved": "boolean",
                    "references": {}
                },
                "required": false,
                "optional": true,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": true,
                "defaultValue": "false"
            },
            "Indeterminate": {
                "type": "boolean",
                "attribute": "indeterminate",
                "mutable": false,
                "complexType": {
                    "original": "boolean",
                    "resolved": "boolean",
                    "references": {}
                },
                "required": false,
                "optional": true,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false,
                "defaultValue": "false"
            }
        };
    }
    static get states() {
        return {
            "isChecked": {}
        };
    }
    static get watchers() {
        return [{
                "propName": "checked",
                "methodName": "watchMaxValue"
            }];
    }
}
