// @platformscode/core/dist/collection/components/dga-quote/dga-quote.js
import { h } from "@stencil/core";
import { SxHandler } from "../../utils/sx-handler";
export class DgaQuote {
    constructor() {
        this.size = 'sm';
        this.whiteBackground = false;
        this.showQuoteTitle = true;
        this.quoteTitle = '';
        this.showQuoteDescription = true;
        this.quoteDescription = '';
        this.showAuthorDetails = true;
        this.authorName = '';
        this.authorDescription = '';
        this.showAvatar = false;
    }
    componentWillLoad() {
        this.sxHandler = new SxHandler(this.el, 'dga-slider');
    }
    componentDidLoad() {
        this.sxHandler.processSx(this.sx);
    }
    sxChanged() {
        this.sxHandler.processSx(this.sx);
    }
    disconnectedCallback() {
        this.sxHandler.cleanup();
    }
    render() {
        return (h("div", { key: '646789efc2ad38b7808ca82592939a6051f19c9f', class: { 'dga-quote': true, 'dga-quote--white-bg': this.whiteBackground } }, h("div", { key: '1837a07a10c5677cdd0c5c219bf38049946bc21d', class: "dga-quote__body" }, h("dga-flex", { key: '75ba955f74da6e195b1aa634d6b946d6fc07993f', spacing: 9 }, h("span", { key: '7e0a9caba28a5d9c21d852061eb9dcd1870d15dd', class: "quote-icon" }, "\u201D"), h("dga-flex", { key: '40644c053b578527da6903a1bdbb304cded58468', direction: "column", spacing: 16 }, this.showQuoteTitle && h("h2", { key: '92ef841ea8f0a55459d6fcc7fa40d46ffa996b3f', class: "dga-quote__body-title" }, this.quoteTitle), this.showQuoteDescription && h("p", { key: '52a97893f283ecdcc78c4648b234026d12f8c63a', class: "dga-quote__body-desc" }, this.quoteDescription)))), this.showAuthorDetails && (h("div", { key: '5083d38d5b396433df0435ae8fc7ad41d7edf580', class: "dga-quote__auther-details" }, h("dga-flex", { key: '8e73fea44dcb5d2a6e5c1cd79684b7d1d03dedf7', direction: "row", spacing: 16, align: "center" }, this.showAvatar && h("dga-avatar", { key: '9f0131bf583ad16c96c219b7d86f5d2b73c5cdfd', size: 80, type: "image", border: false }), h("dga-flex", { key: '32d24e14df4ca57a1bd1daa85b7f1648392408b0', direction: "column", spacing: 8 }, h("h3", { key: '9ca353846a3f53d6cff9e4a8eb1c54f216a6fb6a', class: "auther-title" }, this.authorName), h("p", { key: 'f0fa2d233583a74c5a012f405204f9d75de290d4', class: "auther-desc" }, this.authorDescription))), h("span", { key: 'affac155816d82093ad4341e9226c2fa780008d7', class: "quote-icon" }, "\u201C")))));
    }
    static get is() { return "dga-quote"; }
    static get originalStyleUrls() {
        return {
            "$": ["dga-quote.scss"]
        };
    }
    static get styleUrls() {
        return {
            "$": ["dga-quote.css"]
        };
    }
    static get properties() {
        return {
            "size": {
                "type": "string",
                "attribute": "size",
                "mutable": false,
                "complexType": {
                    "original": "'sm' | 'lg'",
                    "resolved": "\"lg\" | \"sm\"",
                    "references": {}
                },
                "required": false,
                "optional": true,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false,
                "defaultValue": "'sm'"
            },
            "whiteBackground": {
                "type": "boolean",
                "attribute": "white-background",
                "mutable": false,
                "complexType": {
                    "original": "boolean",
                    "resolved": "boolean",
                    "references": {}
                },
                "required": false,
                "optional": true,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false,
                "defaultValue": "false"
            },
            "showQuoteTitle": {
                "type": "boolean",
                "attribute": "show-quote-title",
                "mutable": false,
                "complexType": {
                    "original": "boolean",
                    "resolved": "boolean",
                    "references": {}
                },
                "required": false,
                "optional": true,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false,
                "defaultValue": "true"
            },
            "quoteTitle": {
                "type": "string",
                "attribute": "quote-title",
                "mutable": false,
                "complexType": {
                    "original": "string",
                    "resolved": "string",
                    "references": {}
                },
                "required": false,
                "optional": true,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false,
                "defaultValue": "''"
            },
            "showQuoteDescription": {
                "type": "boolean",
                "attribute": "show-quote-description",
                "mutable": false,
                "complexType": {
                    "original": "boolean",
                    "resolved": "boolean",
                    "references": {}
                },
                "required": false,
                "optional": true,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false,
                "defaultValue": "true"
            },
            "quoteDescription": {
                "type": "string",
                "attribute": "quote-description",
                "mutable": false,
                "complexType": {
                    "original": "string",
                    "resolved": "string",
                    "references": {}
                },
                "required": false,
                "optional": true,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false,
                "defaultValue": "''"
            },
            "showAuthorDetails": {
                "type": "boolean",
                "attribute": "show-author-details",
                "mutable": false,
                "complexType": {
                    "original": "boolean",
                    "resolved": "boolean",
                    "references": {}
                },
                "required": false,
                "optional": true,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false,
                "defaultValue": "true"
            },
            "authorName": {
                "type": "string",
                "attribute": "author-name",
                "mutable": false,
                "complexType": {
                    "original": "string",
                    "resolved": "string",
                    "references": {}
                },
                "required": false,
                "optional": true,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false,
                "defaultValue": "''"
            },
            "authorDescription": {
                "type": "string",
                "attribute": "author-description",
                "mutable": false,
                "complexType": {
                    "original": "string",
                    "resolved": "string",
                    "references": {}
                },
                "required": false,
                "optional": true,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false,
                "defaultValue": "''"
            },
            "showAvatar": {
                "type": "boolean",
                "attribute": "show-avatar",
                "mutable": false,
                "complexType": {
                    "original": "boolean",
                    "resolved": "boolean",
                    "references": {}
                },
                "required": false,
                "optional": true,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false,
                "defaultValue": "false"
            },
            "sx": {
                "type": "string",
                "attribute": "sx",
                "mutable": false,
                "complexType": {
                    "original": "SxProps | string",
                    "resolved": "SxProps | string",
                    "references": {
                        "SxProps": {
                            "location": "import",
                            "path": "../../utils/sx-handler",
                            "id": "src/utils/sx-handler.ts::SxProps"
                        }
                    }
                },
                "required": false,
                "optional": true,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false
            }
        };
    }
    static get elementRef() { return "el"; }
    static get watchers() {
        return [{
                "propName": "sx",
                "methodName": "sxChanged"
            }];
    }
}
