// @platformscode/core/dist/collection/components/dga-drawer/dga-drawer.js
import { Host, h } from "@stencil/core";
import { SxHandler } from "../../utils/sx-handler";
export class DgaDrawer {
    constructor() {
        this.background = 'brand';
        this.anchor = 'left';
        this.open = false;
    }
    componentWillLoad() {
        this.sxHandler = new SxHandler(this.el, 'dga-slider');
    }
    componentDidLoad() {
        this.sxHandler.processSx(this.sx);
    }
    sxChanged() {
        this.sxHandler.processSx(this.sx);
    }
    disconnectedCallback() {
        this.sxHandler.cleanup();
    }
    render() {
        return (h(Host, { key: '852c1b553acf13987eed86c49d6f237e119fc378', id: this.sxHandler.getId(), style: this.sxHandler.getStyles() }, h("div", { key: 'bfaa54d3f169a759aef88eb857f491bcc06f1857', class: `sidepanel sidepanel--${this.anchor} ${this.background ? `sidepanel--bg-${this.background}` : ''}
            ${this.overlay ? 'sidepanel--overlay' : ''}
            ${this.border ? 'sidepanel--border' : ''}
            ${this.open ? 'active' : ''}
            ` }, this.headerLogo && h("div", { key: 'c480128121823c203d9acfa71e32f2ea244849bb', class: "sidepanel__header" }, h("img", { key: '423d3ed5fb4efc982b2a4a43e34eb9356980de7a', src: this.headerLogo, alt: 'logo' })), h("side-panel-menu", { key: '59a4b075643abf8ec329ec9dd69e046b9263a870', routes: this.routes }), h("slot", { key: '6748cf132fc0e7872e030f10a7e0dcae25125ce4' }))));
    }
    static get is() { return "dga-drawer"; }
    static get encapsulation() { return "shadow"; }
    static get originalStyleUrls() {
        return {
            "$": ["dga-drawer.scss"]
        };
    }
    static get styleUrls() {
        return {
            "$": ["dga-drawer.css"]
        };
    }
    static get properties() {
        return {
            "background": {
                "type": "string",
                "attribute": "background",
                "mutable": false,
                "complexType": {
                    "original": "'gray' | 'brand'",
                    "resolved": "\"brand\" | \"gray\"",
                    "references": {}
                },
                "required": false,
                "optional": true,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false,
                "defaultValue": "'brand'"
            },
            "overlay": {
                "type": "boolean",
                "attribute": "overlay",
                "mutable": false,
                "complexType": {
                    "original": "boolean",
                    "resolved": "boolean",
                    "references": {}
                },
                "required": false,
                "optional": true,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false
            },
            "border": {
                "type": "boolean",
                "attribute": "border",
                "mutable": false,
                "complexType": {
                    "original": "boolean",
                    "resolved": "boolean",
                    "references": {}
                },
                "required": false,
                "optional": true,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false
            },
            "anchor": {
                "type": "string",
                "attribute": "anchor",
                "mutable": false,
                "complexType": {
                    "original": "'left'|'right'",
                    "resolved": "\"left\" | \"right\"",
                    "references": {}
                },
                "required": false,
                "optional": true,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false,
                "defaultValue": "'left'"
            },
            "open": {
                "type": "boolean",
                "attribute": "open",
                "mutable": true,
                "complexType": {
                    "original": "boolean",
                    "resolved": "boolean",
                    "references": {}
                },
                "required": false,
                "optional": true,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": true,
                "defaultValue": "false"
            },
            "routes": {
                "type": "unknown",
                "attribute": "routes",
                "mutable": false,
                "complexType": {
                    "original": "IRoute[]",
                    "resolved": "IRoute[]",
                    "references": {
                        "IRoute": {
                            "location": "local",
                            "path": "/home/mamoud/Documents/temp/components/platform-code-cross-platform-storybook/packages/core/src/components/dga-drawer/dga-drawer.tsx",
                            "id": "src/components/dga-drawer/dga-drawer.tsx::IRoute"
                        }
                    }
                },
                "required": false,
                "optional": false,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false
            },
            "headerLogo": {
                "type": "string",
                "attribute": "header-logo",
                "mutable": false,
                "complexType": {
                    "original": "string",
                    "resolved": "string",
                    "references": {}
                },
                "required": false,
                "optional": true,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false
            },
            "sx": {
                "type": "string",
                "attribute": "sx",
                "mutable": false,
                "complexType": {
                    "original": "SxProps | string",
                    "resolved": "SxProps | string",
                    "references": {
                        "SxProps": {
                            "location": "import",
                            "path": "../../utils/sx-handler",
                            "id": "src/utils/sx-handler.ts::SxProps"
                        }
                    }
                },
                "required": false,
                "optional": true,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false
            }
        };
    }
    static get elementRef() { return "el"; }
    static get watchers() {
        return [{
                "propName": "sx",
                "methodName": "sxChanged"
            }];
    }
}
