// @platformscode/core/dist/collection/components/dga-linear-progress-bar/dga-linear-progress-bar.js
import { h } from "@stencil/core";
export class DgaLinearProgressBar {
    constructor() {
        this.RTL = false;
        this.percentage = 0;
        this.size = 'small';
        this.progressStyle = 'neutral';
        this.showLabel = true;
        this.showHelperText = true;
        this.error = false;
        this.success = false;
        this.label = 'label';
        this.helperText = 'Help Text';
    }
    render() {
        const sizeMapper = {
            small: 'sm',
            medium: 'md',
            large: 'lg',
        };
        let progressStyle = this.error ? 'error' : this.success ? "success" : this.progressStyle;
        let progressPercentage = this.percentage;
        const icon = {
            neutral: (h("svg", { key: 'ad3cd132ce762b4ca230b3d061f8548dfc8e6743', width: "16", height: "16", viewBox: "0 0 16 16", fill: "none", xmlns: "http://www.w3.org/2000/svg" }, h("g", { key: 'ddf57d87c872b95f615ad514c32d6a9cdb2fcb89', "clip-path": "url(#clip0_17869_405)" }, h("path", { key: 'f099baff0c1983f75dcc22299f659541c8c05949', d: "M6.06004 6.00001C6.21678 5.55446 6.52614 5.17875 6.93334 4.93943C7.34055 4.70012 7.8193 4.61264 8.28483 4.69248C8.75035 4.77233 9.17259 5.01436 9.47676 5.3757C9.78093 5.73703 9.94741 6.19436 9.94671 6.66668C9.94671 8.00001 7.94671 8.66668 7.94671 8.66668M8.00004 11.3333H8.00671M14.6667 8.00001C14.6667 11.6819 11.6819 14.6667 8.00004 14.6667C4.31814 14.6667 1.33337 11.6819 1.33337 8.00001C1.33337 4.31811 4.31814 1.33334 8.00004 1.33334C11.6819 1.33334 14.6667 4.31811 14.6667 8.00001Z", stroke: "#384250", "stroke-width": "1.5", "stroke-linecap": "round", "stroke-linejoin": "round" })), h("defs", { key: 'd2caa8d45c1f3ccf746319965af923e42666b674' }, h("clipPath", { key: '892c8b48fd6273ed40c9c31bc37974b877985dd2', id: "clip0_17869_405" }, h("rect", { key: 'af315330c6ce48312464ab6f2682b6e61c0cb202', width: "16", height: "16", fill: "white" }))))),
            primary: (h("svg", { key: '2150fe4cad8756cfe448270a4969af9b0141501d', width: "16", height: "16", viewBox: "0 0 16 16", fill: "none", xmlns: "http://www.w3.org/2000/svg" }, h("path", { key: '7e43c870adcf2225c8e181477b765bd32408022b', "fill-rule": "evenodd", "clip-rule": "evenodd", d: "M8.00002 15.3334C12.0501 15.3334 15.3334 12.0501 15.3334 8.00002C15.3334 3.94993 12.0501 0.666687 8.00002 0.666687C3.94993 0.666687 0.666687 3.94993 0.666687 8.00002C0.666687 12.0501 3.94993 15.3334 8.00002 15.3334Z", fill: "white" }), h("path", { key: 'c6b1470eff9df4d63489b586f4c6dc6520696040', "fill-rule": "evenodd", "clip-rule": "evenodd", d: "M8.00002 15.3334C12.0501 15.3334 15.3334 12.0501 15.3334 8.00002C15.3334 3.94993 12.0501 0.666687 8.00002 0.666687C3.94993 0.666687 0.666687 3.94993 0.666687 8.00002C0.666687 12.0501 3.94993 15.3334 8.00002 15.3334ZM4.70017 7.41423C4.43982 7.67458 4.43982 8.09669 4.70017 8.35704L6.58578 10.2427C6.84613 10.503 7.26824 10.503 7.52859 10.2427L11.2998 6.47143C11.5602 6.21108 11.5602 5.78897 11.2998 5.52862C11.0395 5.26827 10.6174 5.26827 10.357 5.52862L7.05719 8.82845L5.64297 7.41423C5.38263 7.15388 4.96052 7.15388 4.70017 7.41423Z", fill: "#067647" }))),
            error: (h("svg", { key: '833d5409de4d166a775617c8a8aa6330adeb01ec', width: "16", height: "16", viewBox: "0 0 16 16", fill: "none", xmlns: "http://www.w3.org/2000/svg" }, h("path", { key: '3469dc6b5b91ba973b62aaaf8dd739ed14f0a2c7', "fill-rule": "evenodd", "clip-rule": "evenodd", d: "M8.00002 15.3334C12.0501 15.3334 15.3334 12.0501 15.3334 8.00002C15.3334 3.94993 12.0501 0.666687 8.00002 0.666687C3.94993 0.666687 0.666687 3.94993 0.666687 8.00002C0.666687 12.0501 3.94993 15.3334 8.00002 15.3334Z", fill: "white" }), h("path", { key: 'af5293f42df923600e77418e93a1783444d06b43', "fill-rule": "evenodd", "clip-rule": "evenodd", d: "M15.3334 8.00002C15.3334 12.0502 12.0502 15.3334 8.00002 15.3334C3.94989 15.3334 0.666687 12.0502 0.666687 8.00002C0.666687 3.94989 3.94989 0.666687 8.00002 0.666687C12.0502 0.666687 15.3334 3.94989 15.3334 8.00002ZM5.80471 4.862C5.54445 4.60158 5.12225 4.60158 4.862 4.862C4.60158 5.12225 4.60158 5.54445 4.862 5.80471L7.05715 8.00002L4.862 10.1953C4.60158 10.4556 4.60158 10.8778 4.862 11.138C5.12225 11.3985 5.54445 11.3985 5.80471 11.138L8.00002 8.94289L10.1953 11.138C10.4556 11.3985 10.8778 11.3985 11.138 11.138C11.3985 10.8778 11.3985 10.4556 11.138 10.1953L8.94289 8.00002L11.138 5.80471C11.3985 5.54445 11.3985 5.12225 11.138 4.862C10.8778 4.60158 10.4556 4.60158 10.1953 4.862L8.00002 7.05715L5.80471 4.862Z", fill: "#B42318" }))),
        };
        return (h("div", { key: '0ed22f0040c913ee8c47c8fcb6b09ad91a65516f', class: 'progress-wrapper', style: { direction: this.RTL ? "rtl" : "ltr" } }, this.showLabel && h("h3", { key: 'c20c5c2a65240ebc957f11d9140b0c913cba8d82', class: `${sizeMapper[this.size]}` }, this.label), h("div", { key: '0bfa2a34aadfee76ea1d389f01d45073e94fd719', class: `progress-bar-container ${sizeMapper[this.size]}`, title: `${progressPercentage.toString()}%` }, h("div", { key: '203f19f4b25fe4863d1003c0fef10e2fafc56016', class: `progress-bar ${progressStyle}`, style: { width: `${progressPercentage}%` } }, this.size === "large" && progressPercentage > 8 && h("span", { key: 'ec9b40fcaa22c7d95fd4af0d3ade0c93dc87d1cd', class: "progress-bar-text" }, progressPercentage, "%"))), this.showHelperText && (h("div", { key: '407ed36553eeb1b9efab3b20a80f46af0a4e387a', class: "progress-bar__note" }, h("div", { key: '99bcd695030873a3ece53c3a124f03a30c88baf0', class: "icon" }, this.error ? icon["error"] : this.success ? icon["primary"] : icon["neutral"]), h("div", { key: '9174d49de14f09c8dd0bdffcd38311dfb71a88ab', class: "content" }, h("p", { key: 'b5e63429f70fc732dd23a127ef3221364e220381', class: `progress-bar-content ${this.success && "success"} ${this.error && "error"} ${sizeMapper[this.size]}` }, this.helperText))))));
    }
    static get is() { return "dga-linear-progress-bar"; }
    static get originalStyleUrls() {
        return {
            "$": ["dga-linear-progress-bar.scss"]
        };
    }
    static get styleUrls() {
        return {
            "$": ["dga-linear-progress-bar.css"]
        };
    }
    static get properties() {
        return {
            "RTL": {
                "type": "boolean",
                "attribute": "r-t-l",
                "mutable": false,
                "complexType": {
                    "original": "boolean",
                    "resolved": "boolean",
                    "references": {}
                },
                "required": false,
                "optional": false,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false,
                "defaultValue": "false"
            },
            "percentage": {
                "type": "number",
                "attribute": "percentage",
                "mutable": false,
                "complexType": {
                    "original": "number",
                    "resolved": "number",
                    "references": {}
                },
                "required": false,
                "optional": false,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false,
                "defaultValue": "0"
            },
            "size": {
                "type": "string",
                "attribute": "size",
                "mutable": false,
                "complexType": {
                    "original": "'small' | 'medium' | 'large'",
                    "resolved": "\"large\" | \"medium\" | \"small\"",
                    "references": {}
                },
                "required": false,
                "optional": false,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false,
                "defaultValue": "'small'"
            },
            "progressStyle": {
                "type": "string",
                "attribute": "progress-style",
                "mutable": false,
                "complexType": {
                    "original": "'primary' | 'neutral' | 'error'",
                    "resolved": "\"error\" | \"neutral\" | \"primary\"",
                    "references": {}
                },
                "required": false,
                "optional": false,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false,
                "defaultValue": "'neutral'"
            },
            "showLabel": {
                "type": "boolean",
                "attribute": "show-label",
                "mutable": false,
                "complexType": {
                    "original": "boolean",
                    "resolved": "boolean",
                    "references": {}
                },
                "required": false,
                "optional": false,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false,
                "defaultValue": "true"
            },
            "showHelperText": {
                "type": "boolean",
                "attribute": "show-helper-text",
                "mutable": false,
                "complexType": {
                    "original": "boolean",
                    "resolved": "boolean",
                    "references": {}
                },
                "required": false,
                "optional": false,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false,
                "defaultValue": "true"
            },
            "error": {
                "type": "boolean",
                "attribute": "error",
                "mutable": false,
                "complexType": {
                    "original": "boolean",
                    "resolved": "boolean",
                    "references": {}
                },
                "required": false,
                "optional": false,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false,
                "defaultValue": "false"
            },
            "success": {
                "type": "boolean",
                "attribute": "success",
                "mutable": false,
                "complexType": {
                    "original": "boolean",
                    "resolved": "boolean",
                    "references": {}
                },
                "required": false,
                "optional": false,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false,
                "defaultValue": "false"
            },
            "label": {
                "type": "string",
                "attribute": "label",
                "mutable": false,
                "complexType": {
                    "original": "string",
                    "resolved": "string",
                    "references": {}
                },
                "required": false,
                "optional": false,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false,
                "defaultValue": "'label'"
            },
            "helperText": {
                "type": "string",
                "attribute": "helper-text",
                "mutable": false,
                "complexType": {
                    "original": "string",
                    "resolved": "string",
                    "references": {}
                },
                "required": false,
                "optional": false,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false,
                "defaultValue": "'Help Text'"
            }
        };
    }
}
