// @platformscode/core/dist/collection/components/dga-circular-progress-bar/dga-circular-progress-bar.js
import { h } from "@stencil/core";
import { checkmarkCircle02SolidRoundedIcon, multiplicationSignCircleSolidRoundedIcon } from "../../assets/icons/index";
export class DgaCircularProgressBar {
    constructor() {
        this.RTL = false;
        this.size = '64px';
        this.variant = 'primary';
        this.percentage = 0;
        this.sizes = {
            '200px': 200,
            '160px': 160,
            '120px': 120,
            '80px': 80,
            '64px': 64,
        };
    }
    render() {
        const strokeWidth = this.sizes[this.size] / 10;
        const radius = (this.sizes[this.size] - strokeWidth) / 2;
        const dashArray = 2 * Math.PI * radius;
        const dashOffset = dashArray * (1 - this.percentage / 100);
        return (h("div", { key: '291033816c03db4d77a13c4ac388a57d4a54b744', class: "circular-progress-bar", "aria-size": this.size, dir: this.RTL ? 'rtl' : 'ltr' }, h("div", { key: 'd26868af3a0401153c55090a9f443dd773aa2f02', class: "circular-progress-bar__label" }, this.variant === 'success' && h("dga-icon", { key: '28461b96e858abf4a408eb72f4ff3d4ec360ca4e', icon: checkmarkCircle02SolidRoundedIcon, size: 20 }), this.variant === 'error' && h("dga-icon", { key: '79373d85687effb8b0dab66a574dab976398c70b', icon: multiplicationSignCircleSolidRoundedIcon, size: 20 }), this.variant !== 'success' && this.variant !== 'error' && h("h4", { key: 'd5df43d3d7f0f5e796407f8e5a9cbda59cad70a3', class: "circular-progress-bar__label--percentage" }, this.percentage, "%"), this.text && h("h5", { key: '7c3ecccfcf8f82b5082c1e0b7b5bd24b11e4bb73', class: `circular-progress-bar__label--text ${this.variant}` }, this.text)), h("svg", { key: '6c61a192ccd9c5ea94dcb3c9679be3e2d4c51ad6', width: this.sizes[this.size], height: this.sizes[this.size], viewBox: `0 0 ${this.sizes[this.size]} ${this.sizes[this.size]}` }, h("circle", { key: '44db2d34749b0e27093ba0050f039e97fe1899a1', class: "track", fill: "none", stroke: "#e6e6e6", r: radius, cx: this.sizes[this.size] / 2, cy: this.sizes[this.size] / 2, "stroke-width": strokeWidth }), h("circle", { key: '42611fc781757fc90531f7a5ac41363a85c842ce', class: `tail tail--${this.variant}`, fill: "none", stroke: "currentColor", r: radius, cx: this.sizes[this.size] / 2, cy: this.sizes[this.size] / 2, "stroke-width": strokeWidth, "stroke-linecap": 'round', "stroke-dasharray": dashArray, "stroke-dashoffset": dashOffset, transform: `rotate(-180 ${this.sizes[this.size] / 2} ${this.sizes[this.size] / 2})`, style: {
                transition: 'stroke-dashoffset 0.5s ease',
            } }))));
    }
    static get is() { return "dga-circular-progress-bar"; }
    static get encapsulation() { return "shadow"; }
    static get originalStyleUrls() {
        return {
            "$": ["dga-circular-progress-bar.scss"]
        };
    }
    static get styleUrls() {
        return {
            "$": ["dga-circular-progress-bar.css"]
        };
    }
    static get properties() {
        return {
            "RTL": {
                "type": "boolean",
                "attribute": "r-t-l",
                "mutable": false,
                "complexType": {
                    "original": "boolean",
                    "resolved": "boolean",
                    "references": {}
                },
                "required": false,
                "optional": false,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false,
                "defaultValue": "false"
            },
            "size": {
                "type": "string",
                "attribute": "size",
                "mutable": false,
                "complexType": {
                    "original": "'200px' | '160px' | '120px' | '80px' | '64px'",
                    "resolved": "\"120px\" | \"160px\" | \"200px\" | \"64px\" | \"80px\"",
                    "references": {}
                },
                "required": false,
                "optional": false,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false,
                "defaultValue": "'64px'"
            },
            "text": {
                "type": "string",
                "attribute": "text",
                "mutable": false,
                "complexType": {
                    "original": "string",
                    "resolved": "string",
                    "references": {}
                },
                "required": false,
                "optional": true,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false
            },
            "variant": {
                "type": "string",
                "attribute": "variant",
                "mutable": false,
                "complexType": {
                    "original": "'primary' | 'neutral' | 'success' | 'error'",
                    "resolved": "\"error\" | \"neutral\" | \"primary\" | \"success\"",
                    "references": {}
                },
                "required": false,
                "optional": true,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false,
                "defaultValue": "'primary'"
            },
            "percentage": {
                "type": "number",
                "attribute": "percentage",
                "mutable": false,
                "complexType": {
                    "original": "number",
                    "resolved": "number",
                    "references": {}
                },
                "required": false,
                "optional": true,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false,
                "defaultValue": "0"
            }
        };
    }
}
