// @platformscode/core/dist/collection/components/dga-table-of-content/dga-table-of-content.js
import { Host, h } from "@stencil/core";
import { SxHandler } from "../../utils/sx-handler";
const THRESHOLD = 10;
export class DgaTableOfContent {
    constructor() {
        this.sections = [];
        this.subTitle = 'sub';
        this.tocTitle = 'main';
        this.rtl = false; // New prop to control RTL mode
        this.activeSectionId = '';
        this.tabRefs = new Map();
        this.isManuallyScrolling = false;
        this.scrollTimeout = null;
        this.handleScroll = () => {
            // Skip scroll handling if user just clicked (manual scrolling in progress)
            if (this.isManuallyScrolling)
                return;
            const scrollPosition = window.scrollY || window.pageYOffset;
            const scrollTop = window.scrollY || window.pageYOffset;
            const positions = {};
            const defineTargetsPositions = sections => {
                sections.forEach(section => {
                    const element = document.getElementById(section.Target);
                    if (element) {
                        const rect = element.getBoundingClientRect();
                        const elementTop = rect.top + scrollPosition;
                        positions[section.Target] = elementTop;
                        if (section.Children && section.Children.length > 0) {
                            defineTargetsPositions(section.Children);
                        }
                    }
                });
            };
            defineTargetsPositions(this.sections);
            const scrollRanges = this.convertToScrollRanges(positions);
            // Only proceed if we have ranges to check
            if (scrollRanges.length === 0)
                return;
            // Find the appropriate section
            let newActiveId = scrollRanges[0].title; // Default to first section
            for (const { title, range } of scrollRanges) {
                const [start, end] = range;
                if (scrollTop + THRESHOLD >= start && scrollTop < end) {
                    newActiveId = title;
                    break; // Stop once we find the matching section
                }
            }
            // Update active section if changed
            if (newActiveId !== this.activeSectionId) {
                this.activeSectionId = newActiveId;
            }
        };
    }
    sxChanged() {
        this.sxHandler.processSx(this.sx);
    }
    componentWillLoad() {
        // If defaultActiveId is provided, use it
        if (this.defaultActiveId) {
            this.activeSectionId = this.defaultActiveId;
        }
        this.sxHandler = new SxHandler(this.el, 'dga-table-of-content');
    }
    componentDidLoad() {
        // Initialize scroll handling after DOM is ready
        setTimeout(() => {
            this.handleScroll();
            window.addEventListener('scroll', this.handleScroll);
        }, 300);
        this.sxHandler.processSx(this.sx);
    }
    disconnectedCallback() {
        window.removeEventListener('scroll', this.handleScroll);
        if (this.scrollTimeout) {
            clearTimeout(this.scrollTimeout);
        }
        this.sxHandler.cleanup();
    }
    convertToScrollRanges(positions) {
        const titles = Object.keys(positions).sort((a, b) => positions[a] - positions[b]);
        if (titles.length === 0)
            return [];
        const scrollRanges = [];
        for (let i = 0; i < titles.length - 1; i++) {
            const title = titles[i];
            const nextTitle = titles[i + 1];
            const range = [positions[title], positions[nextTitle]];
            scrollRanges.push({ title, range });
        }
        // Handle the last title's range to Infinity
        const lastTitle = titles[titles.length - 1];
        const lastRange = [positions[lastTitle], Infinity];
        scrollRanges.push({ title: lastTitle, range: lastRange });
        return scrollRanges;
    }
    moveTOSelectedId(id) {
        try {
            // Set active section immediately on click
            this.activeSectionId = id;
            // Prevent scroll handler from changing the active section during programmatic scrolling
            this.isManuallyScrolling = true;
            const targetElement = document.getElementById(id);
            if (targetElement) {
                // Adjust for any fixed headers
                const headerOffset = 20;
                const elementPosition = targetElement.getBoundingClientRect().top;
                const offsetPosition = elementPosition + window.pageYOffset - headerOffset;
                window.scrollTo({
                    top: offsetPosition,
                    behavior: 'smooth',
                });
                // Re-enable scroll detection after scrolling completes
                if (this.scrollTimeout) {
                    clearTimeout(this.scrollTimeout);
                }
                this.scrollTimeout = setTimeout(() => {
                    this.isManuallyScrolling = false;
                }, 1000); // Adjust based on scroll animation duration
            }
        }
        catch (error) {
            console.error('Error moving to selected ID:', error);
            this.isManuallyScrolling = false;
        }
    }
    handleAddPressed(tabId) {
        const tabElement = this.tabRefs.get(tabId);
        tabElement === null || tabElement === void 0 ? void 0 : tabElement.classList.add('pressed');
    }
    handleRemovePressed(tabId) {
        const tabElement = this.tabRefs.get(tabId);
        tabElement === null || tabElement === void 0 ? void 0 : tabElement.classList.remove('pressed');
    }
    renderTab(tab) {
        return (h("li", null, h("a", { ref: el => el && this.tabRefs.set(tab.Target, el), onClick: e => {
                e.preventDefault();
                this.moveTOSelectedId(tab.Target);
            }, onMouseDown: () => this.handleAddPressed(tab.Target), onMouseUp: () => this.handleRemovePressed(tab.Target), onMouseOut: () => this.handleRemovePressed(tab.Target), class: `table-of-content__tab ${this.activeSectionId === tab.Target ? 'active' : ''}` }, h("span", { class: "table-of-content__tab-label" }, tab.Name))));
    }
    renderSubTabs(subTab) {
        var _a;
        return (h("ul", { class: "table-of-content__tab-sublist" }, this.renderTab(subTab), (_a = subTab.Children) === null || _a === void 0 ? void 0 :
            _a.map(subSubTab => (h("ul", { class: "table-of-content__tab-sub-sublist" }, this.renderTab(subSubTab))))));
    }
    render() {
        return (h(Host, { key: 'db353adc7d30b22d9bf9141c97256970f5999318', id: this.sxHandler.getId(), style: this.sxHandler.getStyles() }, h("aside", { key: '88678c2a577cf7ca5db1c2da355fe0a98191cfc5', class: `table-of-content ${this.rtl ? 'rtl' : ''}`, dir: this.rtl ? 'rtl' : 'ltr' }, h("div", { key: '62276b99ad64048930d38c142ade4a87b1aa01d3', class: "table-of-content__header" }, h("p", { key: '231ea7500bed46ccd686243b9e3c3936b8f22929', class: "text-sm-regular" }, this.subTitle), h("h4", { key: 'cec37a15457518eab9a94ea34b0c652067218424', class: "text-xl-semibold" }, this.tocTitle)), h("div", { key: '105c24aae4ab6dbe00871b8f420458c9c48f0d5e', class: "table-of-content__body" }, h("ul", { key: 'f83bd1593c3f3f2e7a5eba3744f3ef77128552ad', class: "table-of-content__tab-list" }, this.sections.map(tab => tab.Children ? (h("div", null, this.renderTab(tab), tab.Children.map(subTab => this.renderSubTabs(subTab)))) : (this.renderTab(tab))))))));
    }
    static get is() { return "dga-table-of-content"; }
    static get encapsulation() { return "shadow"; }
    static get originalStyleUrls() {
        return {
            "$": ["dga-table-of-content.scss"]
        };
    }
    static get styleUrls() {
        return {
            "$": ["dga-table-of-content.css"]
        };
    }
    static get properties() {
        return {
            "sections": {
                "type": "unknown",
                "attribute": "sections",
                "mutable": false,
                "complexType": {
                    "original": "TabType[]",
                    "resolved": "TabType[]",
                    "references": {
                        "TabType": {
                            "location": "local",
                            "path": "/home/mamoud/Documents/temp/components/platform-code-cross-platform-storybook/packages/core/src/components/dga-table-of-content/dga-table-of-content.tsx",
                            "id": "src/components/dga-table-of-content/dga-table-of-content.tsx::TabType"
                        }
                    }
                },
                "required": false,
                "optional": false,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "defaultValue": "[]"
            },
            "subTitle": {
                "type": "string",
                "attribute": "sub-title",
                "mutable": false,
                "complexType": {
                    "original": "string",
                    "resolved": "string",
                    "references": {}
                },
                "required": false,
                "optional": false,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false,
                "defaultValue": "'sub'"
            },
            "tocTitle": {
                "type": "string",
                "attribute": "title",
                "mutable": false,
                "complexType": {
                    "original": "string",
                    "resolved": "string",
                    "references": {}
                },
                "required": false,
                "optional": false,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false,
                "defaultValue": "'main'"
            },
            "defaultActiveId": {
                "type": "string",
                "attribute": "default-active-id",
                "mutable": false,
                "complexType": {
                    "original": "string",
                    "resolved": "string",
                    "references": {}
                },
                "required": false,
                "optional": true,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false
            },
            "rtl": {
                "type": "boolean",
                "attribute": "rtl",
                "mutable": false,
                "complexType": {
                    "original": "boolean",
                    "resolved": "boolean",
                    "references": {}
                },
                "required": false,
                "optional": false,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false,
                "defaultValue": "false"
            },
            "sx": {
                "type": "string",
                "attribute": "sx",
                "mutable": false,
                "complexType": {
                    "original": "SxProps | string | undefined",
                    "resolved": "SxProps | string",
                    "references": {
                        "SxProps": {
                            "location": "import",
                            "path": "../../utils/sx-handler",
                            "id": "src/utils/sx-handler.ts::SxProps"
                        }
                    }
                },
                "required": false,
                "optional": true,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false
            }
        };
    }
    static get states() {
        return {
            "activeSectionId": {}
        };
    }
    static get elementRef() { return "el"; }
    static get watchers() {
        return [{
                "propName": "sx",
                "methodName": "sxChanged"
            }];
    }
}
