// @platformscode/core/dist/collection/components/dga-quote/dga-quote.js
import { h } from "@stencil/core";
import { SxHandler } from "../../utils/sx-handler";
export class DgaQuote {
    constructor() {
        this.size = 'sm';
        this.whiteBackground = false;
        this.showQuoteTitle = true;
        this.quoteTitle = '';
        this.showQuoteDescription = true;
        this.quoteDescription = '';
        this.showAuthorDetails = true;
        this.authorName = '';
        this.authorDescription = '';
        this.showAvatar = false;
    }
    componentWillLoad() {
        this.sxHandler = new SxHandler(this.el, 'dga-slider');
    }
    componentDidLoad() {
        this.sxHandler.processSx(this.sx);
    }
    sxChanged() {
        this.sxHandler.processSx(this.sx);
    }
    disconnectedCallback() {
        this.sxHandler.cleanup();
    }
    render() {
        return (h("div", { key: 'fb35ac80585c933c25a5cb58d3087fdb5b8cd218', class: { 'dga-quote': true, 'dga-quote--white-bg': this.whiteBackground } }, h("div", { key: '463af797b482e462b7f6570b9b8c0e721f3310d3', class: "dga-quote__body" }, h("dga-flex", { key: '8d937c83d18bdc5904ba1eeb5a3f704ee9028cdc', spacing: 9 }, h("span", { key: '5401fb11a7e8c8f0f6dcd2990e7795d57d56d299', class: "quote-icon" }, "\u201D"), h("dga-flex", { key: '2d4f3441f4023ce7108979324cfbb12449584036', direction: "column", spacing: 16 }, this.showQuoteTitle && h("h2", { key: '7f3227d2a44ddf9849cdf0ec7fcff3f1abc8f4f6', class: "dga-quote__body-title" }, this.quoteTitle), this.showQuoteDescription && h("p", { key: 'b634b2084f309df61fb87136e1adacab787f45e9', class: "dga-quote__body-desc" }, this.quoteDescription)))), this.showAuthorDetails && (h("div", { key: '028cf1b9344eed5a37b60b7e079d13d63c1da9ba', class: "dga-quote__auther-details" }, h("dga-flex", { key: '197fb592768e52bf9b8917867f61d3eebf1658a1', direction: "row", spacing: 16, align: "center" }, this.showAvatar && h("dga-avatar", { key: 'af2a50edc81b2801db58b4e6843da7aa2d4d2c80', size: 80, type: "image", border: false }), h("dga-flex", { key: '870414b47d7f3530ca20711f25fd995182418558', direction: "column", spacing: 8 }, h("h3", { key: '0e786bdce329d14770327ca820f700476f2f397a', class: "auther-title" }, this.authorName), h("p", { key: '07fb34ce45805ac1d3861d29cef065095fee5f89', class: "auther-desc" }, this.authorDescription))), h("span", { key: 'd497031042b388dfaa7d6bd791d7c3356cc20ca0', class: "quote-icon" }, "\u201C")))));
    }
    static get is() { return "dga-quote"; }
    static get originalStyleUrls() {
        return {
            "$": ["dga-quote.scss"]
        };
    }
    static get styleUrls() {
        return {
            "$": ["dga-quote.css"]
        };
    }
    static get properties() {
        return {
            "size": {
                "type": "string",
                "attribute": "size",
                "mutable": false,
                "complexType": {
                    "original": "'sm' | 'lg'",
                    "resolved": "\"lg\" | \"sm\"",
                    "references": {}
                },
                "required": false,
                "optional": true,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false,
                "defaultValue": "'sm'"
            },
            "whiteBackground": {
                "type": "boolean",
                "attribute": "white-background",
                "mutable": false,
                "complexType": {
                    "original": "boolean",
                    "resolved": "boolean",
                    "references": {}
                },
                "required": false,
                "optional": true,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false,
                "defaultValue": "false"
            },
            "showQuoteTitle": {
                "type": "boolean",
                "attribute": "show-quote-title",
                "mutable": false,
                "complexType": {
                    "original": "boolean",
                    "resolved": "boolean",
                    "references": {}
                },
                "required": false,
                "optional": true,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false,
                "defaultValue": "true"
            },
            "quoteTitle": {
                "type": "string",
                "attribute": "quote-title",
                "mutable": false,
                "complexType": {
                    "original": "string",
                    "resolved": "string",
                    "references": {}
                },
                "required": false,
                "optional": true,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false,
                "defaultValue": "''"
            },
            "showQuoteDescription": {
                "type": "boolean",
                "attribute": "show-quote-description",
                "mutable": false,
                "complexType": {
                    "original": "boolean",
                    "resolved": "boolean",
                    "references": {}
                },
                "required": false,
                "optional": true,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false,
                "defaultValue": "true"
            },
            "quoteDescription": {
                "type": "string",
                "attribute": "quote-description",
                "mutable": false,
                "complexType": {
                    "original": "string",
                    "resolved": "string",
                    "references": {}
                },
                "required": false,
                "optional": true,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false,
                "defaultValue": "''"
            },
            "showAuthorDetails": {
                "type": "boolean",
                "attribute": "show-author-details",
                "mutable": false,
                "complexType": {
                    "original": "boolean",
                    "resolved": "boolean",
                    "references": {}
                },
                "required": false,
                "optional": true,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false,
                "defaultValue": "true"
            },
            "authorName": {
                "type": "string",
                "attribute": "author-name",
                "mutable": false,
                "complexType": {
                    "original": "string",
                    "resolved": "string",
                    "references": {}
                },
                "required": false,
                "optional": true,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false,
                "defaultValue": "''"
            },
            "authorDescription": {
                "type": "string",
                "attribute": "author-description",
                "mutable": false,
                "complexType": {
                    "original": "string",
                    "resolved": "string",
                    "references": {}
                },
                "required": false,
                "optional": true,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false,
                "defaultValue": "''"
            },
            "showAvatar": {
                "type": "boolean",
                "attribute": "show-avatar",
                "mutable": false,
                "complexType": {
                    "original": "boolean",
                    "resolved": "boolean",
                    "references": {}
                },
                "required": false,
                "optional": true,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false,
                "defaultValue": "false"
            },
            "sx": {
                "type": "string",
                "attribute": "sx",
                "mutable": false,
                "complexType": {
                    "original": "SxProps | string",
                    "resolved": "SxProps | string",
                    "references": {
                        "SxProps": {
                            "location": "import",
                            "path": "../../utils/sx-handler",
                            "id": "src/utils/sx-handler.ts::SxProps"
                        }
                    }
                },
                "required": false,
                "optional": true,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false
            }
        };
    }
    static get elementRef() { return "el"; }
    static get watchers() {
        return [{
                "propName": "sx",
                "methodName": "sxChanged"
            }];
    }
}
