// @platformscode/core/dist/collection/components/dga-datepicker/DateCell.js
// @ts-nocheck
import { h } from "@stencil/core";
export class DateCell {
    constructor() {
        this.midRange = false;
        this.isFirstRangeChild = false;
        this.isLastRangeChild = false;
        //   @Event() onSelect: EventEmitter<Date>;
        this.handleClick = () => {
            if (this.isCurrentMonth) {
                //   this.onSelect.emit(this.date);
                if (this.onSelect) {
                    this.onSelect(this.date);
                }
            }
        };
        this.handleAddPressed = (e) => {
            e.currentTarget.classList.add('active');
        };
        this.handleRemovePressed = (e) => {
            e.currentTarget.classList.remove('active');
        };
    }
    render() {
        let className = 'date-cell';
        if (!this.isCurrentMonth)
            className += ' date-cell--not-current-month';
        if (this.isSelected)
            className += ' date-cell--selected';
        if (this.isToday)
            className += ' date-cell--today';
        if (this.midRange)
            className += ' date-cell--mid-range';
        if (this.isFirstRangeChild)
            className += ' date-cell--mid-range--first-child';
        if (this.isLastRangeChild)
            className += ' date-cell--mid-range--last-child';
        return (h("button", { key: '3cfb2bbbe4b32af0cbe3f6a8c046f1a3df18e764', type: "button", class: className, onClick: () => this.handleClick(), onMouseDown: this.handleAddPressed, onMouseUp: this.handleRemovePressed, onMouseOut: this.handleRemovePressed, disabled: !this.isCurrentMonth }, h("div", { key: '639c4d2ad163e23494e0034016f39e294a4c8e23', class: "date-cell__label" }, h("span", { key: '4370802b3685b31fc702d8ab72ed8d8cb6a17d71' }, this.date.getDate()))));
    }
    static get is() { return "date-cell"; }
    static get originalStyleUrls() {
        return {
            "$": ["../../assets/styles/main.scss"]
        };
    }
    static get styleUrls() {
        return {
            "$": ["../../assets/styles/main.css"]
        };
    }
    static get properties() {
        return {
            "date": {
                "type": "unknown",
                "attribute": "date",
                "mutable": false,
                "complexType": {
                    "original": "Date",
                    "resolved": "Date",
                    "references": {
                        "Date": {
                            "location": "global",
                            "id": "global::Date"
                        }
                    }
                },
                "required": false,
                "optional": false,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false
            },
            "isCurrentMonth": {
                "type": "boolean",
                "attribute": "is-current-month",
                "mutable": false,
                "complexType": {
                    "original": "boolean",
                    "resolved": "boolean",
                    "references": {}
                },
                "required": false,
                "optional": false,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false
            },
            "isSelected": {
                "type": "boolean",
                "attribute": "is-selected",
                "mutable": true,
                "complexType": {
                    "original": "boolean",
                    "resolved": "boolean",
                    "references": {}
                },
                "required": false,
                "optional": false,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false
            },
            "isToday": {
                "type": "boolean",
                "attribute": "is-today",
                "mutable": false,
                "complexType": {
                    "original": "boolean",
                    "resolved": "boolean",
                    "references": {}
                },
                "required": false,
                "optional": false,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false
            },
            "midRange": {
                "type": "boolean",
                "attribute": "mid-range",
                "mutable": false,
                "complexType": {
                    "original": "boolean",
                    "resolved": "boolean",
                    "references": {}
                },
                "required": false,
                "optional": true,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false,
                "defaultValue": "false"
            },
            "isFirstRangeChild": {
                "type": "boolean",
                "attribute": "is-first-range-child",
                "mutable": false,
                "complexType": {
                    "original": "boolean",
                    "resolved": "boolean",
                    "references": {}
                },
                "required": false,
                "optional": false,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false,
                "defaultValue": "false"
            },
            "isLastRangeChild": {
                "type": "boolean",
                "attribute": "is-last-range-child",
                "mutable": false,
                "complexType": {
                    "original": "boolean",
                    "resolved": "boolean",
                    "references": {}
                },
                "required": false,
                "optional": false,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false,
                "defaultValue": "false"
            },
            "onSelect": {
                "type": "unknown",
                "attribute": "on-select",
                "mutable": false,
                "complexType": {
                    "original": "(date: Date) => void",
                    "resolved": "(date: Date) => void",
                    "references": {
                        "Date": {
                            "location": "global",
                            "id": "global::Date"
                        }
                    }
                },
                "required": false,
                "optional": false,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false
            }
        };
    }
}
