// @platformscode/core/dist/collection/components/dga-datepicker/MonthNavigator.js
import { h } from "@stencil/core";
// import { ArrowLeft01Icon, ArrowRight01Icon } from '@hugeicons/react-pro';
export class MonthNavigator {
    constructor() {
        this.RTL = false;
        this.showArrows = true;
        this.isYearsDropdownOpen = false;
        // private toggleYearsDropdown = () => {
        //   this.isYearsDropdownOpen = !this.isYearsDropdownOpen;
        // }
        // private handleYearSelect = (year: number) => {
        //   // Create a new date with the selected year but keep current month
        //   const newDate = new Date(
        //     year,
        //     this.currentMonth.getMonth(),
        //     1
        //   );
        //   this.onChange(newDate);
        //   this.toggleYearsDropdown()
        //   // this.isYearsDropdownOpen = false;
        // }
        this.handleYearSelect = (year, event) => {
            // Stop event propagation to prevent bubbling to the year-selector element
            event.stopPropagation();
            // Create a new date with the selected year but keep current month
            const newDate = new Date(year, this.currentMonth.getMonth(), 1);
            // Update the date first
            this.onChange(newDate);
            // Force the dropdown to close with a small delay to ensure state changes are processed
            setTimeout(() => {
                this.isYearsDropdownOpen = false;
            }, 5);
        };
        // Toggle the dropdown
        this.toggleYearsDropdown = () => {
            this.isYearsDropdownOpen = !this.isYearsDropdownOpen;
            // If opening the dropdown, focus on the selected year
            if (this.isYearsDropdownOpen) {
                // Use setTimeout to ensure the dropdown is rendered before scrolling
                setTimeout(() => this.scrollToSelectedYear(), 5);
            }
        };
        //   @Event() monthChange: EventEmitter<Date>;
        this.handlePrevClick = () => {
            const prevMonth = new Date(this.currentMonth.getFullYear(), this.currentMonth.getMonth() - 1, 1);
            // this.monthChange.emit(prevMonth);
            this.onChange(prevMonth);
        };
        this.handleNextClick = () => {
            const nextMonth = new Date(this.currentMonth.getFullYear(), this.currentMonth.getMonth() + 1, 1);
            // this.monthChange.emit(nextMonth);
            this.onChange(nextMonth);
        };
    }
    // Generate list of years (e.g., current year ± 10 years)
    get yearsList() {
        // const currentYear = this.currentMonth.getFullYear();
        const currentYear = new Date().getFullYear();
        const years = [];
        // Generate years (current year - 100 to current year + 100)
        for (let year = currentYear - 50; year <= currentYear + 50; year++) {
            years.push(year);
        }
        return years;
    }
    // Scroll to the currently selected year
    scrollToSelectedYear() {
        if (!this.yearDropdownRef)
            return;
        // Find the selected year element
        const selectedYearElement = this.yearDropdownRef.querySelector('.selected-year');
        if (selectedYearElement) {
            // Calculate position to center the selected year in the dropdown
            const dropdownHeight = this.yearDropdownRef.offsetHeight;
            const selectedYearTop = selectedYearElement.getBoundingClientRect().top;
            const dropdownTop = this.yearDropdownRef.getBoundingClientRect().top;
            const selectedYearRelativeTop = selectedYearTop - dropdownTop;
            // Scroll to center the selected year
            this.yearDropdownRef.scrollTop = selectedYearRelativeTop - (dropdownHeight / 2) + 20;
        }
    }
    render() {
        return (h("div", { key: '485e9d83547211470b319cea22e6ca699d8bf9f3', class: "date-picker__navigator" }, h("div", { key: '2f26355c7d7e66e5084d57bafe6d4ce45b61e967', class: "date-picker__navigator-label" }, this.currentMonth.toLocaleDateString(this.RTL ? 'ar' : 'default', {
            month: 'long',
            // year: 'numeric',
        }), " ", h("span", { key: '26c607af9d1119b823b83a0a77310b194cc48456', class: "year-selector", onClick: this.toggleYearsDropdown, style: {
                cursor: 'pointer',
                position: 'relative',
                display: 'inline-flex',
                alignItems: 'center'
            } }, this.currentMonth.getFullYear(), this.isYearsDropdownOpen && (
        // {true && (
        h("ul", { key: '2ef3876e85edabc3b1da6786495e0198f49aa5f7', class: "dga-breadcrumb-dropdown", ref: (el) => this.yearDropdownRef = el }, this.yearsList.map(year => (h("li", { key: year, class: `dga-breadcrumb-dropdown-item ${year === this.currentMonth.getFullYear() ? 'selected-year' : ''}` }, h("a", { onClick: (e) => this.handleYearSelect(year, e) }, h("span", { class: "link__label" }, year))))))))), this.showArrows && (h("div", { key: '8ed7afa9ed2cb230d5cca9baa525123b7e0d3231', class: "date-picker__navigator-buttons" }, h("dga-button-v2", { key: '985a28922d4029c7235b4e19c4cec0686f5fca9d', variant: "subtle", size: "lg", iconOnly: true, leadIcon: true, leadIconProps: {
                name: 'arrow-left-02',
                variant: 'stroke',
                type: 'rounded',
            }, onClick: this.handlePrevClick }), h("dga-button-v2", { key: 'd96b39a654ee99c5290fbacf5db3ce08a401fdc0', variant: "subtle", size: "lg", iconOnly: true, leadIcon: true, leadIconProps: {
                name: 'arrow-right-02',
                variant: 'stroke',
                type: 'rounded',
            }, onClick: this.handleNextClick })))));
    }
    static get is() { return "month-navigator"; }
    static get properties() {
        return {
            "RTL": {
                "type": "boolean",
                "attribute": "r-t-l",
                "mutable": false,
                "complexType": {
                    "original": "boolean",
                    "resolved": "boolean",
                    "references": {}
                },
                "required": false,
                "optional": false,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false,
                "defaultValue": "false"
            },
            "currentMonth": {
                "type": "unknown",
                "attribute": "current-month",
                "mutable": false,
                "complexType": {
                    "original": "Date",
                    "resolved": "Date",
                    "references": {
                        "Date": {
                            "location": "global",
                            "id": "global::Date"
                        }
                    }
                },
                "required": false,
                "optional": false,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false
            },
            "showArrows": {
                "type": "boolean",
                "attribute": "show-arrows",
                "mutable": false,
                "complexType": {
                    "original": "boolean",
                    "resolved": "boolean",
                    "references": {}
                },
                "required": false,
                "optional": false,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false,
                "defaultValue": "true"
            },
            "onChange": {
                "type": "unknown",
                "attribute": "on-change",
                "mutable": false,
                "complexType": {
                    "original": "(newMonth: Date) => void",
                    "resolved": "(newMonth: Date) => void",
                    "references": {
                        "Date": {
                            "location": "global",
                            "id": "global::Date"
                        }
                    }
                },
                "required": false,
                "optional": false,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false
            }
        };
    }
    static get states() {
        return {
            "isYearsDropdownOpen": {}
        };
    }
    static get elementRef() { return "el"; }
}
