// @platformscode/core/dist/collection/components/dga-file-upload/dga-file-upload.js
import { h } from "@stencil/core";
import { guid } from "../../utils";
export class DgaFileUpload {
    constructor() {
        this.id = guid();
        this.name = guid();
        this.title = 'Drag and drop files here to upload';
        this.fileTypesText = 'Maximum file size allowed is 2MB, supported file formats include .jpg, .png, and .pdf.';
        this.actionName = 'Browse Files';
        this.disabled = false;
        this.files = [];
        this.dragState = false;
        this.handleFileSelection = (event) => {
            const input = event.target;
            this.onChange && this.onChange(event);
            const selectedFiles = input.files;
            if (selectedFiles) {
                if (this.maximumFilesSize) {
                    const isFileOverSize = Array.from(selectedFiles).some(file => {
                        return file.size > this.maximumFilesSize;
                    });
                    if (isFileOverSize) {
                        this.uploadFileOverSize && this.uploadFileOverSize();
                        return;
                    }
                }
                const fileItems = Array.from(selectedFiles).map(file => ({
                    name: file.name,
                    uploadStatus: 'pending', // initial status
                }));
                this.files = [...this.files, ...fileItems];
                this.getUploadedFile && this.getUploadedFile([...this.files]);
            }
        };
        this.handleDrop = (event) => {
            event.preventDefault();
            event.stopPropagation();
            const droppedFiles = event.dataTransfer.files; // Accessing the dropped files
            if (droppedFiles) {
                const fileItems = Array.from(droppedFiles).map(file => ({
                    name: file.name,
                    uploadStatus: 'pending',
                }));
                this.files = [...this.files, ...fileItems]; // Update state with new files
            }
            this.dragState = false;
        };
        this.addDragOverClass = (event) => {
            event.preventDefault();
            event.stopPropagation();
            this.dragState = true;
        };
        this.removeDragOverClass = (event) => {
            event.preventDefault();
            event.stopPropagation();
            this.dragState = false;
        };
    }
    componentDidLoad() {
        const dropArea = document.querySelector('.file-upload__drop-area');
        dropArea.addEventListener('dragover', this.addDragOverClass);
        dropArea.addEventListener('dragleave', this.removeDragOverClass);
        dropArea.addEventListener('drop', this.handleDrop);
    }
    disconnectedCallback() {
        const dropArea = document.querySelector('.file-upload__drop-area');
        dropArea.removeEventListener('dragover', this.addDragOverClass);
        dropArea.removeEventListener('dragleave', this.removeDragOverClass);
        dropArea.removeEventListener('drop', this.handleDrop);
    }
    handleDeleteFile(index, file) {
        this.files = [...this.files.slice(0, index), ...this.files.slice(index + 1)];
        this.emitDeleteFile && this.emitDeleteFile(file, index);
    }
    renderFile(file, index) {
        if (file.uploadStatus === 'error') {
            return (h("div", { class: "file-upload__file-item--error" }, h("div", { class: "file-upload__file-item" }, h("span", { class: "file-upload__file-status file-upload__file-status--error" }, h("dga-icon", { name: "alert-circle", variant: "solid", size: 20 })), h("span", { class: "file-upload__file-name" }, file.name), h("button", { class: "btn btn--close btn--icon btn--sm", onClick: () => this.handleDeleteFile(index, file) }, h("span", { class: "btn-icon" }, h("dga-icon", { name: "multiplication-sign", variant: "stroke", size: 20 })))), h("div", { class: "file-upload__file-item-msg" }, file.errorMessage ? file.errorMessage : 'Here goes the helper text.')));
        }
        return (h("div", { class: "file-upload__file-item" }, h("span", { class: `file-upload__file-status file-upload__file-status--${file.uploadStatus}` }, file.uploadStatus === 'uploading' && file.progress && h("dga-loading", { size: "tiny", variant: "neutral" }), file.uploadStatus === 'complete' && h("dga-icon", { name: "checkmark-circle-02", variant: "solid", size: 20 })), h("span", { class: "file-upload__file-name" }, file.name), h("button", { class: "btn btn--close btn--icon btn--sm", onClick: () => this.handleDeleteFile(index, file) }, h("span", { class: "btn-icon" }, h("dga-icon", { name: "multiplication-sign", variant: "stroke", size: 20 })))));
    }
    render() {
        return (h("div", { key: '403b9bd7df4f837e789093c790163933718e3cfe', class: `file-upload ${this.dragState ? 'file-upload--drag-over' : ''} ${this.disabled ? 'file-upload--disabled' : ''}` }, h("input", { key: '7ba72bef2def2cec12f259b838f5916127b53e09', type: "file", id: this.id, name: this.name, class: "file-upload__input", accept: this.accept, multiple: this.multiple, "aria-hidden": "true", onChange: this.handleFileSelection, ref: el => (this.fileInputRef = el) }), h("label", { key: '43c441bd125795cfba70950bd75df41d08ece4f4', htmlFor: this.id, class: "file-upload__drop-area" }, h("span", { key: '9a843b0d167fd073b81428e817030c63633a35d1', class: "file-upload__icon" }, h("dga-icon", { key: 'ce39243d21b0ec0359e2eccaf6a489697f88ba0c', name: "file-upload", variant: "stroke", size: 20 })), h("div", { key: 'b7d6fd8b9613ad4d7e2fbb05ac8545f3e8c847d8', class: "file-upload__content" }, h("p", { key: '319c3ab7f3c806bc7b8271f10515c2a26c92536b', class: "file-upload__instructions" }, this.title), h("p", { key: '3fe784b51e9278662501654f43c9dfb3d31b6b4b', class: "file-upload__formats" }, this.fileTypesText)), h("button", { key: '2c02c041fc9f9c2ee97f0c6aa582a307d23de7b8', class: "btn btn--md btn--secondary", "aria-label": "Browse Files", disabled: this.dragState || this.disabled, type: "button", onClick: () => { var _a; return (_a = this.fileInputRef) === null || _a === void 0 ? void 0 : _a.click(); } }, h("span", { key: '6d39c744b101b9d2e66ab0180fe7d43bca10b96f' }, this.actionName))), h("div", { key: '1f91fcc34066b27848ee9d94df4f74ff189e0083', class: "file-upload__files-list" }, this.files.map((file, index) => this.renderFile(file, index)))));
    }
    static get is() { return "dga-file-upload"; }
    static get encapsulation() { return "shadow"; }
    static get originalStyleUrls() {
        return {
            "$": ["dga-file-upload.scss"]
        };
    }
    static get styleUrls() {
        return {
            "$": ["dga-file-upload.css"]
        };
    }
    static get properties() {
        return {
            "id": {
                "type": "string",
                "attribute": "id",
                "mutable": false,
                "complexType": {
                    "original": "string",
                    "resolved": "string",
                    "references": {}
                },
                "required": false,
                "optional": false,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false,
                "defaultValue": "guid()"
            },
            "name": {
                "type": "string",
                "attribute": "name",
                "mutable": false,
                "complexType": {
                    "original": "string",
                    "resolved": "string",
                    "references": {}
                },
                "required": false,
                "optional": false,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false,
                "defaultValue": "guid()"
            },
            "accept": {
                "type": "string",
                "attribute": "accept",
                "mutable": false,
                "complexType": {
                    "original": "string",
                    "resolved": "string",
                    "references": {}
                },
                "required": false,
                "optional": false,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false
            },
            "multiple": {
                "type": "boolean",
                "attribute": "multiple",
                "mutable": false,
                "complexType": {
                    "original": "boolean",
                    "resolved": "boolean",
                    "references": {}
                },
                "required": false,
                "optional": false,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false
            },
            "title": {
                "type": "string",
                "attribute": "title",
                "mutable": false,
                "complexType": {
                    "original": "string",
                    "resolved": "string",
                    "references": {}
                },
                "required": false,
                "optional": false,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false,
                "defaultValue": "'Drag and drop files here to upload'"
            },
            "fileTypesText": {
                "type": "string",
                "attribute": "file-types-text",
                "mutable": false,
                "complexType": {
                    "original": "string",
                    "resolved": "string",
                    "references": {}
                },
                "required": false,
                "optional": false,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false,
                "defaultValue": "'Maximum file size allowed is 2MB, supported file formats include .jpg, .png, and .pdf.'"
            },
            "actionName": {
                "type": "string",
                "attribute": "action-name",
                "mutable": false,
                "complexType": {
                    "original": "string",
                    "resolved": "string",
                    "references": {}
                },
                "required": false,
                "optional": false,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false,
                "defaultValue": "'Browse Files'"
            },
            "maximumFilesSize": {
                "type": "number",
                "attribute": "maximum-files-size",
                "mutable": false,
                "complexType": {
                    "original": "number",
                    "resolved": "number",
                    "references": {}
                },
                "required": false,
                "optional": true,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false
            },
            "disabled": {
                "type": "boolean",
                "attribute": "disabled",
                "mutable": false,
                "complexType": {
                    "original": "boolean",
                    "resolved": "boolean",
                    "references": {}
                },
                "required": false,
                "optional": true,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false,
                "defaultValue": "false"
            },
            "getUploadedFile": {
                "type": "unknown",
                "attribute": "get-uploaded-file",
                "mutable": false,
                "complexType": {
                    "original": "(\n    files: {\n      name: string;\n      uploadStatus: 'pending' | 'uploading' | 'complete' | 'error';\n      progress?: number; // from 0 to 100, if uploading\n      errorMessage?: string;\n    }[],\n  ) => void",
                    "resolved": "(files: { name: string; uploadStatus: \"error\" | \"pending\" | \"uploading\" | \"complete\"; progress?: number; errorMessage?: string; }[]) => void",
                    "references": {}
                },
                "required": false,
                "optional": true,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false
            },
            "uploadFileOverSize": {
                "type": "unknown",
                "attribute": "upload-file-over-size",
                "mutable": false,
                "complexType": {
                    "original": "() => void",
                    "resolved": "() => void",
                    "references": {}
                },
                "required": false,
                "optional": true,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false
            },
            "onChange": {
                "type": "unknown",
                "attribute": "on-change",
                "mutable": false,
                "complexType": {
                    "original": "(event: Event) => void",
                    "resolved": "(event: Event) => void",
                    "references": {
                        "Event": {
                            "location": "global",
                            "id": "global::Event"
                        }
                    }
                },
                "required": false,
                "optional": true,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false
            },
            "emitDeleteFile": {
                "type": "unknown",
                "attribute": "emit-delete-file",
                "mutable": false,
                "complexType": {
                    "original": "(\n    file: {\n      name: string;\n      uploadStatus: 'pending' | 'uploading' | 'complete' | 'error';\n      progress?: number; // from 0 to 100, if uploading\n      errorMessage?: string;\n    },\n    index: number,\n  ) => void",
                    "resolved": "(file: { name: string; uploadStatus: \"error\" | \"pending\" | \"uploading\" | \"complete\"; progress?: number; errorMessage?: string; }, index: number) => void",
                    "references": {}
                },
                "required": false,
                "optional": true,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false
            }
        };
    }
    static get states() {
        return {
            "files": {},
            "dragState": {}
        };
    }
}
